{
 "cells": [
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "Lee++ (NAR, 2016) says that IS*1* undertakes \"local hopping\".\n",
    "Here, I aim to analyze whether the distribution of essential genes contribute to the local hopping.\n",
    "\n",
    "# Overview\n",
    "\n",
    "1. Load data\n",
    "2. Identify esseitnal genes (using blastn)\n",
    "3. Calculate average distance from the closest essential gene and compare with deletion length\n",
    "\n",
    "### Input\n",
    "Deletion boundary and corresponding IS IDs.\n",
    "List of ranges of essential genes.\n",
    "\n",
    "### Procedure\n",
    "1. Identify new-existing type deletions (MA8, MA20).\n",
    "2. Identify homologous sequences to the essential regions\n",
    "\t1. blast homologous genes\n",
    "\t2. remove the duplicates\n",
    "\t3. output closest essential genes for each IS \n",
    "3. Identify distances to the closest essential region."
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "# Loading"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 1,
   "metadata": {},
   "outputs": [],
   "source": [
    "import os\n",
    "import subprocess\n",
    "import numpy as np\n",
    "import pandas as pd\n",
    "from Bio import SeqIO\n",
    "from Bio.Seq import Seq\n",
    "from Bio.SeqRecord import SeqRecord"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "## List of Essential genes\n",
    "note that gene_df... is in the coordinates of MDS42_IS1.fa starting from thrC."
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 2,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>gene</th>\n",
       "      <th>start</th>\n",
       "      <th>end</th>\n",
       "      <th>Essential_PEC</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>ribF</td>\n",
       "      <td>16232</td>\n",
       "      <td>17173</td>\n",
       "      <td>True</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>ileS</td>\n",
       "      <td>17216</td>\n",
       "      <td>20032</td>\n",
       "      <td>True</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>lspA</td>\n",
       "      <td>20032</td>\n",
       "      <td>20526</td>\n",
       "      <td>True</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>ispH</td>\n",
       "      <td>21102</td>\n",
       "      <td>22052</td>\n",
       "      <td>True</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>dapB</td>\n",
       "      <td>23199</td>\n",
       "      <td>24020</td>\n",
       "      <td>True</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "   gene  start    end Essential_PEC\n",
       "0  ribF  16232  17173          True\n",
       "1  ileS  17216  20032          True\n",
       "2  lspA  20032  20526          True\n",
       "3  ispH  21102  22052          True\n",
       "4  dapB  23199  24020          True"
      ]
     },
     "execution_count": 2,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "# Inputs\n",
    "# get essential genes\n",
    "essential_genes_file = os.path.join(\"..\",\"misc\",\"gene_df_ess_mds42_based_on_PEC.csv\")\n",
    "essential_genes = pd.read_csv(essential_genes_file)\n",
    "essential_genes = essential_genes[[\"gene\", \"start\", \"end\", \"Essential_PEC\"]]\n",
    "essential_genes = essential_genes[essential_genes[\"Essential_PEC\"] == True].\\\n",
    "\t\t\treset_index(drop=True)\n",
    "\n",
    "# get MDS42 sequences\n",
    "mds42_fasta = os.path.join(\"..\",\"fasta\",\"20230904\",\"MDS42_IS1.fa\")\n",
    "\n",
    "essential_genes.head()"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "## IS positions"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 44,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>start</th>\n",
       "      <th>end</th>\n",
       "      <th>IS_strand</th>\n",
       "      <th>max_alignment_length</th>\n",
       "      <th>length</th>\n",
       "      <th>cluster_id</th>\n",
       "      <th>line</th>\n",
       "      <th>gen_id</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>476928</td>\n",
       "      <td>480020</td>\n",
       "      <td>reverse</td>\n",
       "      <td>3093</td>\n",
       "      <td>3093</td>\n",
       "      <td>0</td>\n",
       "      <td>L01-1</td>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>557167</td>\n",
       "      <td>560259</td>\n",
       "      <td>reverse</td>\n",
       "      <td>3093</td>\n",
       "      <td>3093</td>\n",
       "      <td>1</td>\n",
       "      <td>L01-1</td>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>1187831</td>\n",
       "      <td>1192502</td>\n",
       "      <td>forward</td>\n",
       "      <td>3093</td>\n",
       "      <td>4672</td>\n",
       "      <td>2</td>\n",
       "      <td>L01-1</td>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>1405123</td>\n",
       "      <td>1408215</td>\n",
       "      <td>forward</td>\n",
       "      <td>3093</td>\n",
       "      <td>3093</td>\n",
       "      <td>3</td>\n",
       "      <td>L01-1</td>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>1499230</td>\n",
       "      <td>1503892</td>\n",
       "      <td>reverse</td>\n",
       "      <td>3093</td>\n",
       "      <td>4663</td>\n",
       "      <td>4</td>\n",
       "      <td>L01-1</td>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "     start      end IS_strand  max_alignment_length  length  cluster_id  \\\n",
       "0   476928   480020   reverse                  3093    3093           0   \n",
       "1   557167   560259   reverse                  3093    3093           1   \n",
       "2  1187831  1192502   forward                  3093    4672           2   \n",
       "3  1405123  1408215   forward                  3093    3093           3   \n",
       "4  1499230  1503892   reverse                  3093    4663           4   \n",
       "\n",
       "    line  gen_id  \n",
       "0  L01-1       1  \n",
       "1  L01-1       1  \n",
       "2  L01-1       1  \n",
       "3  L01-1       1  \n",
       "4  L01-1       1  "
      ]
     },
     "execution_count": 44,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "base_output_dir = os.path.join(\"exp\", \"multiple_runs11\")\n",
    "base_export = os.path.join(base_output_dir, \"export\")\n",
    "data_export_dir = os.path.join(\"exp\", \"data\", \"ins_del_analysis\")\n",
    "file_name_base = \"20240703_\"\n",
    "is_positions_df = os.path.join(base_export, \"classify_IS_events\", \"IS_positions.csv\")\n",
    "is_positions_df = pd.read_csv(is_positions_df)\n",
    "is_positions_df = is_positions_df.rename(columns = {\"Gen\":\"gen_id\", \"Line\":\"line\"})\n",
    "is_positions_df.head()"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "## Genome size"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 4,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>line</th>\n",
       "      <th>gen_id</th>\n",
       "      <th>genome_size</th>\n",
       "      <th>is_cnt</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>L01-1</td>\n",
       "      <td>1</td>\n",
       "      <td>4003851</td>\n",
       "      <td>10</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>L01-2</td>\n",
       "      <td>1</td>\n",
       "      <td>4003851</td>\n",
       "      <td>10</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>L01-3</td>\n",
       "      <td>1</td>\n",
       "      <td>4003851</td>\n",
       "      <td>10</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>L01-4</td>\n",
       "      <td>1</td>\n",
       "      <td>4003851</td>\n",
       "      <td>10</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>L02-1</td>\n",
       "      <td>1</td>\n",
       "      <td>4026846</td>\n",
       "      <td>15</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "    line  gen_id  genome_size  is_cnt\n",
       "0  L01-1       1      4003851      10\n",
       "1  L01-2       1      4003851      10\n",
       "2  L01-3       1      4003851      10\n",
       "3  L01-4       1      4003851      10\n",
       "4  L02-1       1      4026846      15"
      ]
     },
     "execution_count": 4,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "genome_sizes = os.path.join(base_export, \"classify_IS_events\", \"genome_stats.csv\")\n",
    "genome_sizes = pd.read_csv(genome_sizes)\n",
    "genome_sizes = genome_sizes[['Prefix', 'gen_id', 'genome_size', 'is_cnt']]\n",
    "genome_sizes = genome_sizes.rename(columns = {\"Prefix\":\"line\"})\n",
    "genome_sizes.head()"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "## Deletion length distributions"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 5,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>line</th>\n",
       "      <th>gen_id</th>\n",
       "      <th>group</th>\n",
       "      <th>start</th>\n",
       "      <th>end</th>\n",
       "      <th>is_start</th>\n",
       "      <th>is_end</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>L01-1</td>\n",
       "      <td>2</td>\n",
       "      <td>2</td>\n",
       "      <td>552531</td>\n",
       "      <td>560570</td>\n",
       "      <td>3</td>\n",
       "      <td>6</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>L01-1</td>\n",
       "      <td>2</td>\n",
       "      <td>4</td>\n",
       "      <td>2661331</td>\n",
       "      <td>2665538</td>\n",
       "      <td>21</td>\n",
       "      <td>22</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>6</th>\n",
       "      <td>L01-1</td>\n",
       "      <td>3</td>\n",
       "      <td>2</td>\n",
       "      <td>552045</td>\n",
       "      <td>552506</td>\n",
       "      <td>3</td>\n",
       "      <td>4</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>8</th>\n",
       "      <td>L01-1</td>\n",
       "      <td>3</td>\n",
       "      <td>4</td>\n",
       "      <td>1676236</td>\n",
       "      <td>1680732</td>\n",
       "      <td>18</td>\n",
       "      <td>21</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>10</th>\n",
       "      <td>L01-1</td>\n",
       "      <td>3</td>\n",
       "      <td>6</td>\n",
       "      <td>1791803</td>\n",
       "      <td>1792397</td>\n",
       "      <td>24</td>\n",
       "      <td>25</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "     line  gen_id  group    start      end  is_start  is_end\n",
       "1   L01-1       2      2   552531   560570         3       6\n",
       "3   L01-1       2      4  2661331  2665538        21      22\n",
       "6   L01-1       3      2   552045   552506         3       4\n",
       "8   L01-1       3      4  1676236  1680732        18      21\n",
       "10  L01-1       3      6  1791803  1792397        24      25"
      ]
     },
     "execution_count": 5,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "deletion_df = os.path.join(base_export, \"classify_IS_events\", \"depth_interval.csv\")\n",
    "deletion_df = pd.read_csv(deletion_df)\n",
    "deletion_df = deletion_df[['line', 'gen_id', 'group', 'start_fix', 'end_fix', 'is_start', 'is_end', 'depth']]\n",
    "deletion_df = deletion_df.rename(columns = {\"start_fix\":\"start\", \"end_fix\":\"end\"})\n",
    "deletion_df = deletion_df.query('depth == 0').drop(columns = ['depth'])\n",
    "deletion_df.head()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 6,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>gen_id</th>\n",
       "      <th>line</th>\n",
       "      <th>insert_id</th>\n",
       "      <th>genome</th>\n",
       "      <th>pos_id</th>\n",
       "      <th>IS_strand</th>\n",
       "      <th>position_status</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>22</th>\n",
       "      <td>2</td>\n",
       "      <td>L01-1</td>\n",
       "      <td>3</td>\n",
       "      <td>Query</td>\n",
       "      <td>1</td>\n",
       "      <td>False</td>\n",
       "      <td>new</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>23</th>\n",
       "      <td>2</td>\n",
       "      <td>L01-1</td>\n",
       "      <td>6</td>\n",
       "      <td>Query</td>\n",
       "      <td>1</td>\n",
       "      <td>False</td>\n",
       "      <td>new</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>24</th>\n",
       "      <td>2</td>\n",
       "      <td>L01-1</td>\n",
       "      <td>7</td>\n",
       "      <td>Query</td>\n",
       "      <td>2</td>\n",
       "      <td>False</td>\n",
       "      <td>new</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>25</th>\n",
       "      <td>2</td>\n",
       "      <td>L01-1</td>\n",
       "      <td>7</td>\n",
       "      <td>Query</td>\n",
       "      <td>2</td>\n",
       "      <td>False</td>\n",
       "      <td>new</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>27</th>\n",
       "      <td>2</td>\n",
       "      <td>L01-1</td>\n",
       "      <td>10</td>\n",
       "      <td>Query</td>\n",
       "      <td>3</td>\n",
       "      <td>False</td>\n",
       "      <td>new</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "    gen_id   line  insert_id genome  pos_id  IS_strand position_status\n",
       "22       2  L01-1          3  Query       1      False             new\n",
       "23       2  L01-1          6  Query       1      False             new\n",
       "24       2  L01-1          7  Query       2      False             new\n",
       "25       2  L01-1          7  Query       2      False             new\n",
       "27       2  L01-1         10  Query       3      False             new"
      ]
     },
     "execution_count": 6,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "is_pos_in_ref_genome = os.path.join(base_export, \"classify_IS_events\", \"IS_positions_in_ref_genome.csv\")\n",
    "is_pos_in_ref_genome = pd.read_csv(is_pos_in_ref_genome)\n",
    "# Line, Gen, insert_id, genome, pos_id, IS_strand, position_status\n",
    "is_pos_in_ref_genome = is_pos_in_ref_genome[['Gen', 'Line', 'insert_id', 'genome', 'pos_id', 'IS_strand', 'position_status']]\n",
    "is_pos_in_ref_genome = is_pos_in_ref_genome.rename(columns = {\"Gen\":\"gen_id\", \"Line\":\"line\"})\n",
    "is_pos_in_ref_genome = is_pos_in_ref_genome.query('genome == \"Query\"')\n",
    "is_pos_in_ref_genome.query('position_status == \"new\"').head()"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "# Identify essential genes\n",
    "\n",
    "### Procedure\n",
    "1. For each genome (line_id, gen_id)\n",
    "2. Build blastdb and run blast with essential genes of MDS42 (PEC database) as query\n",
    "3. Identify duplicated and missing essential genes\n",
    "\n",
    "### Output:\n",
    "- `merged_df`: Detailed alignment results including information about each significant match found during the BLAST searches.\n",
    "- `summarised_df`: A summary report indicating the number of duplicated and missing essential genes for each line and genome analyzed.\n",
    "\n",
    "## Misc functions"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 7,
   "metadata": {},
   "outputs": [],
   "source": [
    "def create_blast_db(ref_fasta, db_dir, prefix, overwrite=False):\n",
    "    \"\"\"\n",
    "    Create a BLAST database from a reference fasta file.\n",
    "    :param ref_fasta: Path to the reference fasta file.\n",
    "    :param db_dir: Directory where the BLAST database should be stored.\n",
    "    :param prefix: Prefix to identify the database.\n",
    "    \"\"\"\n",
    "    db_path = os.path.join(db_dir, prefix)\n",
    "    command = ['makeblastdb', '-in', ref_fasta, '-dbtype', 'nucl', '-out', db_path]\n",
    "    subprocess.run(command, check=True)\n",
    "    return db_path\n",
    "\n",
    "def run_blast(query_fasta, db_path, output_path):\n",
    "    \"\"\"\n",
    "    Run BLASTn using a query fasta file against a BLAST database.\n",
    "    :param query_fasta: Path to the query fasta file.\n",
    "    :param db_path: Path to the BLAST database.\n",
    "    :param output_path: Path to save the BLAST output.\n",
    "    \"\"\"\n",
    "    command = [\n",
    "        'blastn', '-query', query_fasta, '-db', db_path, \n",
    "        '-outfmt', '6 qseqid sseqid pident length mismatch gapopen qstart qend sstart send evalue bitscore qlen', \n",
    "        '-out', output_path\n",
    "    ]\n",
    "    subprocess.run(command, check=True)\n",
    "    # add column names to the csv. currently data starts from the first row\n",
    "    with open(output_path, 'r') as f:\n",
    "        data = f.read()\n",
    "    with open(output_path, 'w') as f:\n",
    "        #tsv\n",
    "        #f.write('qseqid sseqid pident length mismatch gapopen qstart qend sstart send evalue bitscore qlen\\n')\n",
    "        f.write('qseqid\\tsseqid\\tpident\\tlength\\tmismatch\\tgapopen\\tqstart\\tqend\\tsstart\\tsend\\tevalue\\tbitscore\\tqlen\\n')\n",
    "        f.write(data)\n",
    "    return output_path\n",
    "\n",
    "def create_query_fasta(df, reference_fasta, output_fasta):\n",
    "    \"\"\"\n",
    "    Create a query FASTA file from a pandas DataFrame containing start and end positions.\n",
    "    :param df: DataFrame with columns ['Name', 'Start', 'End'].\n",
    "    :param output_fasta: Path to the output FASTA file.\n",
    "    \"\"\"\n",
    "    os.makedirs(os.path.dirname(output_fasta), exist_ok=True)\n",
    "    records = []\n",
    "    reference_seq = SeqIO.read(reference_fasta, \"fasta\").seq\n",
    "    for index, row in df.iterrows():\n",
    "    \tsequence = reference_seq[row['start']:row['end']]\n",
    "    \trecord = SeqRecord(Seq(sequence), id=row['gene'], description=\"\")\n",
    "    \trecords.append(record)\n",
    "    SeqIO.write(records, output_fasta, \"fasta\")\n",
    "    return output_fasta\n",
    "\n",
    "def run_ess_blast(prefix, ref_fasta, query_fasta, tmp_dir):\n",
    "\tdb_dir = os.path.join(tmp_dir, 'blastdb')\n",
    "\toutput_dir = os.path.join(tmp_dir, 'blast-ess')\n",
    "\tresult_csv = os.path.join(output_dir, f'{prefix}.blast-ess.csv')\n",
    "\tos.makedirs(db_dir, exist_ok=True)\n",
    "\tos.makedirs(output_dir, exist_ok=True)\n",
    "\tdb_path = create_blast_db(ref_fasta, db_dir, prefix)\n",
    "\trun_blast(query_fasta, db_path, result_csv) # tsv\n",
    "\treturn result_csv"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "## Main function"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 8,
   "metadata": {},
   "outputs": [],
   "source": [
    "def identify_ess_genes(line_df, query_fasta, essential_genes, base_output_dir, tmp_dir):\n",
    "\tmerged_df = pd.DataFrame()\n",
    "\tsummarised_df = pd.DataFrame()\n",
    "\tfor index, row in line_df.iterrows():\n",
    "\t\tline_id = row['line']\n",
    "\t\tgen_id = row['gen_id']\n",
    "\n",
    "\t\tres_csv_ = run_ess_blast(f'{line_id}.{gen_id}',\\\n",
    "\t\t\tos.path.join(base_output_dir, \"tmp\", line_id, f'{line_id}_genome.{gen_id}.fasta'),\\\n",
    "\t\t\tquery_fasta, tmp_dir)\n",
    "\t\tres_ess_ = pd.read_csv(res_csv_, sep='\\t')\n",
    "\n",
    "\t\t# filter partial matches\n",
    "\t\tthreshold_length = 5\n",
    "\t\tthreshold_ratio = 0.95\n",
    "\t\tthreshold_evalue = 1e-10\n",
    "\t\tres_ess_ = res_ess_[(res_ess_['length'] > res_ess_['qlen'] - threshold_length) |\\\n",
    "\t\t\t(res_ess_['length'] / res_ess_['qlen'] > threshold_ratio)]\n",
    "\t\tres_ess_ = res_ess_[(res_ess_['evalue'] < threshold_evalue)]\n",
    "\n",
    "\t\tres_ess_['line'] = line_id\n",
    "\t\tres_ess_['gen_id'] = gen_id\n",
    "\t\t# reorder by length\n",
    "\t\tres_ess_ = res_ess_.sort_values(by=['length'], ascending=False)\n",
    "\t\tres_ess_['duplicated'] = res_ess_.duplicated(subset=['qseqid'], keep=False)\n",
    "\t\tres_ess_ = res_ess_.rename(columns={'qseqid': 'gene'}).\\\n",
    "\t\t\tdrop(columns=['sseqid'])\n",
    "\t\tmerged_df = pd.concat([merged_df, res_ess_], ignore_index=True)\n",
    "\n",
    "\t\tduplicated_genes = res_ess_[res_ess_['duplicated']]['gene'].unique()\n",
    "\t\tmissing_genes = essential_genes[~essential_genes['gene'].isin(res_ess_['gene'])]\n",
    "\t\tsummarised_df = pd.concat([summarised_df, pd.DataFrame({\n",
    "\t\t\t'line': [line_id],\n",
    "\t\t\t'gen_id': [gen_id],\n",
    "\t\t\t'duplicated_genes': [len(duplicated_genes)],\n",
    "\t\t\t'missing_genes': [len(missing_genes)], \n",
    "\t\t})], ignore_index=True)\n",
    "\n",
    "\t#reindex \n",
    "\tmerged_df = merged_df.reset_index(drop=True)\n",
    "\tsummarised_df = summarised_df.reset_index(drop=True)\n",
    "\treturn merged_df, summarised_df"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "## Run"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 9,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\n",
      "\n",
      "Building a new DB, current time: 07/06/2024 18:51:54\n",
      "New DB name:   /home/kanai/documents/analysis/myTELR/syri/ipynb/tmp/essential_gene_blast_identification/blastdb/L01-1.1\n",
      "New DB title:  exp/multiple_runs11/tmp/L01-1/L01-1_genome.1.fasta\n",
      "Sequence type: Nucleotide\n",
      "Deleted existing Nucleotide BLAST database named /home/kanai/documents/analysis/myTELR/syri/ipynb/tmp/essential_gene_blast_identification/blastdb/L01-1.1\n",
      "Keep MBits: T\n",
      "Maximum file size: 3000000000B\n",
      "Adding sequences from FASTA; added 1 sequences in 0.022016 seconds.\n",
      "\n",
      "\n",
      "\n",
      "\n",
      "Building a new DB, current time: 07/06/2024 18:51:54\n",
      "New DB name:   /home/kanai/documents/analysis/myTELR/syri/ipynb/tmp/essential_gene_blast_identification/blastdb/L01-1.2\n",
      "New DB title:  exp/multiple_runs11/tmp/L01-1/L01-1_genome.2.fasta\n",
      "Sequence type: Nucleotide\n",
      "Deleted existing Nucleotide BLAST database named /home/kanai/documents/analysis/myTELR/syri/ipynb/tmp/essential_gene_blast_identification/blastdb/L01-1.2\n",
      "Keep MBits: T\n",
      "Maximum file size: 3000000000B\n",
      "Adding sequences from FASTA; added 1 sequences in 0.0225828 seconds.\n",
      "\n",
      "\n",
      "\n",
      "\n",
      "Building a new DB, current time: 07/06/2024 18:51:54\n",
      "New DB name:   /home/kanai/documents/analysis/myTELR/syri/ipynb/tmp/essential_gene_blast_identification/blastdb/L01-1.3\n",
      "New DB title:  exp/multiple_runs11/tmp/L01-1/L01-1_genome.3.fasta\n",
      "Sequence type: Nucleotide\n",
      "Deleted existing Nucleotide BLAST database named /home/kanai/documents/analysis/myTELR/syri/ipynb/tmp/essential_gene_blast_identification/blastdb/L01-1.3\n",
      "Keep MBits: T\n",
      "Maximum file size: 3000000000B\n",
      "Adding sequences from FASTA; added 1 sequences in 0.021054 seconds.\n",
      "\n",
      "\n",
      "\n",
      "\n",
      "Building a new DB, current time: 07/06/2024 18:51:55\n",
      "New DB name:   /home/kanai/documents/analysis/myTELR/syri/ipynb/tmp/essential_gene_blast_identification/blastdb/L01-2.1\n",
      "New DB title:  exp/multiple_runs11/tmp/L01-2/L01-2_genome.1.fasta\n",
      "Sequence type: Nucleotide\n",
      "Deleted existing Nucleotide BLAST database named /home/kanai/documents/analysis/myTELR/syri/ipynb/tmp/essential_gene_blast_identification/blastdb/L01-2.1\n",
      "Keep MBits: T\n",
      "Maximum file size: 3000000000B\n",
      "Adding sequences from FASTA; added 1 sequences in 0.0222631 seconds.\n",
      "\n",
      "\n",
      "\n",
      "\n",
      "Building a new DB, current time: 07/06/2024 18:51:55\n",
      "New DB name:   /home/kanai/documents/analysis/myTELR/syri/ipynb/tmp/essential_gene_blast_identification/blastdb/L01-2.2\n",
      "New DB title:  exp/multiple_runs11/tmp/L01-2/L01-2_genome.2.fasta\n",
      "Sequence type: Nucleotide\n",
      "Deleted existing Nucleotide BLAST database named /home/kanai/documents/analysis/myTELR/syri/ipynb/tmp/essential_gene_blast_identification/blastdb/L01-2.2\n",
      "Keep MBits: T\n",
      "Maximum file size: 3000000000B\n",
      "Adding sequences from FASTA; added 1 sequences in 0.0223601 seconds.\n",
      "\n",
      "\n",
      "\n",
      "\n",
      "Building a new DB, current time: 07/06/2024 18:51:55\n",
      "New DB name:   /home/kanai/documents/analysis/myTELR/syri/ipynb/tmp/essential_gene_blast_identification/blastdb/L01-2.3\n",
      "New DB title:  exp/multiple_runs11/tmp/L01-2/L01-2_genome.3.fasta\n",
      "Sequence type: Nucleotide\n",
      "Deleted existing Nucleotide BLAST database named /home/kanai/documents/analysis/myTELR/syri/ipynb/tmp/essential_gene_blast_identification/blastdb/L01-2.3\n",
      "Keep MBits: T\n",
      "Maximum file size: 3000000000B\n",
      "Adding sequences from FASTA; added 1 sequences in 0.022244 seconds.\n",
      "\n",
      "\n",
      "\n",
      "\n",
      "Building a new DB, current time: 07/06/2024 18:51:55\n",
      "New DB name:   /home/kanai/documents/analysis/myTELR/syri/ipynb/tmp/essential_gene_blast_identification/blastdb/L01-3.1\n",
      "New DB title:  exp/multiple_runs11/tmp/L01-3/L01-3_genome.1.fasta\n",
      "Sequence type: Nucleotide\n",
      "Deleted existing Nucleotide BLAST database named /home/kanai/documents/analysis/myTELR/syri/ipynb/tmp/essential_gene_blast_identification/blastdb/L01-3.1\n",
      "Keep MBits: T\n",
      "Maximum file size: 3000000000B\n",
      "Adding sequences from FASTA; added 1 sequences in 0.0231249 seconds.\n",
      "\n",
      "\n",
      "\n",
      "\n",
      "Building a new DB, current time: 07/06/2024 18:51:55\n",
      "New DB name:   /home/kanai/documents/analysis/myTELR/syri/ipynb/tmp/essential_gene_blast_identification/blastdb/L01-3.2\n",
      "New DB title:  exp/multiple_runs11/tmp/L01-3/L01-3_genome.2.fasta\n",
      "Sequence type: Nucleotide\n",
      "Deleted existing Nucleotide BLAST database named /home/kanai/documents/analysis/myTELR/syri/ipynb/tmp/essential_gene_blast_identification/blastdb/L01-3.2\n",
      "Keep MBits: T\n",
      "Maximum file size: 3000000000B\n",
      "Adding sequences from FASTA; added 1 sequences in 0.0223792 seconds.\n",
      "\n",
      "\n",
      "\n",
      "\n",
      "Building a new DB, current time: 07/06/2024 18:51:55\n",
      "New DB name:   /home/kanai/documents/analysis/myTELR/syri/ipynb/tmp/essential_gene_blast_identification/blastdb/L01-3.3\n",
      "New DB title:  exp/multiple_runs11/tmp/L01-3/L01-3_genome.3.fasta\n",
      "Sequence type: Nucleotide\n",
      "Deleted existing Nucleotide BLAST database named /home/kanai/documents/analysis/myTELR/syri/ipynb/tmp/essential_gene_blast_identification/blastdb/L01-3.3\n",
      "Keep MBits: T\n",
      "Maximum file size: 3000000000B\n",
      "Adding sequences from FASTA; added 1 sequences in 0.025224 seconds.\n",
      "\n",
      "\n",
      "\n",
      "\n",
      "Building a new DB, current time: 07/06/2024 18:51:55\n",
      "New DB name:   /home/kanai/documents/analysis/myTELR/syri/ipynb/tmp/essential_gene_blast_identification/blastdb/L01-4.1\n",
      "New DB title:  exp/multiple_runs11/tmp/L01-4/L01-4_genome.1.fasta\n",
      "Sequence type: Nucleotide\n",
      "Deleted existing Nucleotide BLAST database named /home/kanai/documents/analysis/myTELR/syri/ipynb/tmp/essential_gene_blast_identification/blastdb/L01-4.1\n",
      "Keep MBits: T\n",
      "Maximum file size: 3000000000B\n",
      "Adding sequences from FASTA; added 1 sequences in 0.0230789 seconds.\n",
      "\n",
      "\n",
      "\n",
      "\n",
      "Building a new DB, current time: 07/06/2024 18:51:55\n",
      "New DB name:   /home/kanai/documents/analysis/myTELR/syri/ipynb/tmp/essential_gene_blast_identification/blastdb/L01-4.2\n",
      "New DB title:  exp/multiple_runs11/tmp/L01-4/L01-4_genome.2.fasta\n",
      "Sequence type: Nucleotide\n",
      "Deleted existing Nucleotide BLAST database named /home/kanai/documents/analysis/myTELR/syri/ipynb/tmp/essential_gene_blast_identification/blastdb/L01-4.2\n",
      "Keep MBits: T\n",
      "Maximum file size: 3000000000B\n",
      "Adding sequences from FASTA; added 1 sequences in 0.0218439 seconds.\n",
      "\n",
      "\n",
      "\n",
      "\n",
      "Building a new DB, current time: 07/06/2024 18:51:55\n",
      "New DB name:   /home/kanai/documents/analysis/myTELR/syri/ipynb/tmp/essential_gene_blast_identification/blastdb/L01-4.3\n",
      "New DB title:  exp/multiple_runs11/tmp/L01-4/L01-4_genome.3.fasta\n",
      "Sequence type: Nucleotide\n",
      "Deleted existing Nucleotide BLAST database named /home/kanai/documents/analysis/myTELR/syri/ipynb/tmp/essential_gene_blast_identification/blastdb/L01-4.3\n",
      "Keep MBits: T\n",
      "Maximum file size: 3000000000B\n",
      "Adding sequences from FASTA; added 1 sequences in 0.022531 seconds.\n",
      "\n",
      "\n",
      "\n",
      "\n",
      "Building a new DB, current time: 07/06/2024 18:51:56\n",
      "New DB name:   /home/kanai/documents/analysis/myTELR/syri/ipynb/tmp/essential_gene_blast_identification/blastdb/L02-1.1\n",
      "New DB title:  exp/multiple_runs11/tmp/L02-1/L02-1_genome.1.fasta\n",
      "Sequence type: Nucleotide\n",
      "Deleted existing Nucleotide BLAST database named /home/kanai/documents/analysis/myTELR/syri/ipynb/tmp/essential_gene_blast_identification/blastdb/L02-1.1\n",
      "Keep MBits: T\n",
      "Maximum file size: 3000000000B\n",
      "Adding sequences from FASTA; added 1 sequences in 0.0220702 seconds.\n",
      "\n",
      "\n",
      "\n",
      "\n",
      "Building a new DB, current time: 07/06/2024 18:51:56\n",
      "New DB name:   /home/kanai/documents/analysis/myTELR/syri/ipynb/tmp/essential_gene_blast_identification/blastdb/L02-1.2\n",
      "New DB title:  exp/multiple_runs11/tmp/L02-1/L02-1_genome.2.fasta\n",
      "Sequence type: Nucleotide\n",
      "Deleted existing Nucleotide BLAST database named /home/kanai/documents/analysis/myTELR/syri/ipynb/tmp/essential_gene_blast_identification/blastdb/L02-1.2\n",
      "Keep MBits: T\n",
      "Maximum file size: 3000000000B\n",
      "Adding sequences from FASTA; added 1 sequences in 0.0225959 seconds.\n",
      "\n",
      "\n",
      "\n",
      "\n",
      "Building a new DB, current time: 07/06/2024 18:51:56\n",
      "New DB name:   /home/kanai/documents/analysis/myTELR/syri/ipynb/tmp/essential_gene_blast_identification/blastdb/L02-1.3\n",
      "New DB title:  exp/multiple_runs11/tmp/L02-1/L02-1_genome.3.fasta\n",
      "Sequence type: Nucleotide\n",
      "Deleted existing Nucleotide BLAST database named /home/kanai/documents/analysis/myTELR/syri/ipynb/tmp/essential_gene_blast_identification/blastdb/L02-1.3\n",
      "Keep MBits: T\n",
      "Maximum file size: 3000000000B\n",
      "Adding sequences from FASTA; added 1 sequences in 0.0232768 seconds.\n",
      "\n",
      "\n",
      "\n",
      "\n",
      "Building a new DB, current time: 07/06/2024 18:51:56\n",
      "New DB name:   /home/kanai/documents/analysis/myTELR/syri/ipynb/tmp/essential_gene_blast_identification/blastdb/L02-2.1\n",
      "New DB title:  exp/multiple_runs11/tmp/L02-2/L02-2_genome.1.fasta\n",
      "Sequence type: Nucleotide\n",
      "Deleted existing Nucleotide BLAST database named /home/kanai/documents/analysis/myTELR/syri/ipynb/tmp/essential_gene_blast_identification/blastdb/L02-2.1\n",
      "Keep MBits: T\n",
      "Maximum file size: 3000000000B\n",
      "Adding sequences from FASTA; added 1 sequences in 0.0228109 seconds.\n",
      "\n",
      "\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\n",
      "\n",
      "Building a new DB, current time: 07/06/2024 18:51:56\n",
      "New DB name:   /home/kanai/documents/analysis/myTELR/syri/ipynb/tmp/essential_gene_blast_identification/blastdb/L02-2.2\n",
      "New DB title:  exp/multiple_runs11/tmp/L02-2/L02-2_genome.2.fasta\n",
      "Sequence type: Nucleotide\n",
      "Deleted existing Nucleotide BLAST database named /home/kanai/documents/analysis/myTELR/syri/ipynb/tmp/essential_gene_blast_identification/blastdb/L02-2.2\n",
      "Keep MBits: T\n",
      "Maximum file size: 3000000000B\n",
      "Adding sequences from FASTA; added 1 sequences in 0.0220981 seconds.\n",
      "\n",
      "\n",
      "\n",
      "\n",
      "Building a new DB, current time: 07/06/2024 18:51:56\n",
      "New DB name:   /home/kanai/documents/analysis/myTELR/syri/ipynb/tmp/essential_gene_blast_identification/blastdb/L02-2.3\n",
      "New DB title:  exp/multiple_runs11/tmp/L02-2/L02-2_genome.3.fasta\n",
      "Sequence type: Nucleotide\n",
      "Deleted existing Nucleotide BLAST database named /home/kanai/documents/analysis/myTELR/syri/ipynb/tmp/essential_gene_blast_identification/blastdb/L02-2.3\n",
      "Keep MBits: T\n",
      "Maximum file size: 3000000000B\n",
      "Adding sequences from FASTA; added 1 sequences in 0.024765 seconds.\n",
      "\n",
      "\n",
      "\n",
      "\n",
      "Building a new DB, current time: 07/06/2024 18:51:56\n",
      "New DB name:   /home/kanai/documents/analysis/myTELR/syri/ipynb/tmp/essential_gene_blast_identification/blastdb/L02-3.1\n",
      "New DB title:  exp/multiple_runs11/tmp/L02-3/L02-3_genome.1.fasta\n",
      "Sequence type: Nucleotide\n",
      "Deleted existing Nucleotide BLAST database named /home/kanai/documents/analysis/myTELR/syri/ipynb/tmp/essential_gene_blast_identification/blastdb/L02-3.1\n",
      "Keep MBits: T\n",
      "Maximum file size: 3000000000B\n",
      "Adding sequences from FASTA; added 1 sequences in 0.0229809 seconds.\n",
      "\n",
      "\n",
      "\n",
      "\n",
      "Building a new DB, current time: 07/06/2024 18:51:56\n",
      "New DB name:   /home/kanai/documents/analysis/myTELR/syri/ipynb/tmp/essential_gene_blast_identification/blastdb/L02-3.2\n",
      "New DB title:  exp/multiple_runs11/tmp/L02-3/L02-3_genome.2.fasta\n",
      "Sequence type: Nucleotide\n",
      "Deleted existing Nucleotide BLAST database named /home/kanai/documents/analysis/myTELR/syri/ipynb/tmp/essential_gene_blast_identification/blastdb/L02-3.2\n",
      "Keep MBits: T\n",
      "Maximum file size: 3000000000B\n",
      "Adding sequences from FASTA; added 1 sequences in 0.022562 seconds.\n",
      "\n",
      "\n",
      "\n",
      "\n",
      "Building a new DB, current time: 07/06/2024 18:51:56\n",
      "New DB name:   /home/kanai/documents/analysis/myTELR/syri/ipynb/tmp/essential_gene_blast_identification/blastdb/L02-3.3\n",
      "New DB title:  exp/multiple_runs11/tmp/L02-3/L02-3_genome.3.fasta\n",
      "Sequence type: Nucleotide\n",
      "Deleted existing Nucleotide BLAST database named /home/kanai/documents/analysis/myTELR/syri/ipynb/tmp/essential_gene_blast_identification/blastdb/L02-3.3\n",
      "Keep MBits: T\n",
      "Maximum file size: 3000000000B\n",
      "Adding sequences from FASTA; added 1 sequences in 0.0239658 seconds.\n",
      "\n",
      "\n",
      "\n",
      "\n",
      "Building a new DB, current time: 07/06/2024 18:51:57\n",
      "New DB name:   /home/kanai/documents/analysis/myTELR/syri/ipynb/tmp/essential_gene_blast_identification/blastdb/L02-4.1\n",
      "New DB title:  exp/multiple_runs11/tmp/L02-4/L02-4_genome.1.fasta\n",
      "Sequence type: Nucleotide\n",
      "Deleted existing Nucleotide BLAST database named /home/kanai/documents/analysis/myTELR/syri/ipynb/tmp/essential_gene_blast_identification/blastdb/L02-4.1\n",
      "Keep MBits: T\n",
      "Maximum file size: 3000000000B\n",
      "Adding sequences from FASTA; added 1 sequences in 0.0228012 seconds.\n",
      "\n",
      "\n",
      "\n",
      "\n",
      "Building a new DB, current time: 07/06/2024 18:51:57\n",
      "New DB name:   /home/kanai/documents/analysis/myTELR/syri/ipynb/tmp/essential_gene_blast_identification/blastdb/L02-4.2\n",
      "New DB title:  exp/multiple_runs11/tmp/L02-4/L02-4_genome.2.fasta\n",
      "Sequence type: Nucleotide\n",
      "Deleted existing Nucleotide BLAST database named /home/kanai/documents/analysis/myTELR/syri/ipynb/tmp/essential_gene_blast_identification/blastdb/L02-4.2\n",
      "Keep MBits: T\n",
      "Maximum file size: 3000000000B\n",
      "Adding sequences from FASTA; added 1 sequences in 0.022284 seconds.\n",
      "\n",
      "\n",
      "\n",
      "\n",
      "Building a new DB, current time: 07/06/2024 18:51:57\n",
      "New DB name:   /home/kanai/documents/analysis/myTELR/syri/ipynb/tmp/essential_gene_blast_identification/blastdb/L02-4.3\n",
      "New DB title:  exp/multiple_runs11/tmp/L02-4/L02-4_genome.3.fasta\n",
      "Sequence type: Nucleotide\n",
      "Deleted existing Nucleotide BLAST database named /home/kanai/documents/analysis/myTELR/syri/ipynb/tmp/essential_gene_blast_identification/blastdb/L02-4.3\n",
      "Keep MBits: T\n",
      "Maximum file size: 3000000000B\n",
      "Adding sequences from FASTA; added 1 sequences in 0.022573 seconds.\n",
      "\n",
      "\n",
      "\n",
      "\n",
      "Building a new DB, current time: 07/06/2024 18:51:57\n",
      "New DB name:   /home/kanai/documents/analysis/myTELR/syri/ipynb/tmp/essential_gene_blast_identification/blastdb/L03-1.1\n",
      "New DB title:  exp/multiple_runs11/tmp/L03-1/L03-1_genome.1.fasta\n",
      "Sequence type: Nucleotide\n",
      "Deleted existing Nucleotide BLAST database named /home/kanai/documents/analysis/myTELR/syri/ipynb/tmp/essential_gene_blast_identification/blastdb/L03-1.1\n",
      "Keep MBits: T\n",
      "Maximum file size: 3000000000B\n",
      "Adding sequences from FASTA; added 1 sequences in 0.0217259 seconds.\n",
      "\n",
      "\n",
      "\n",
      "\n",
      "Building a new DB, current time: 07/06/2024 18:51:57\n",
      "New DB name:   /home/kanai/documents/analysis/myTELR/syri/ipynb/tmp/essential_gene_blast_identification/blastdb/L03-1.2\n",
      "New DB title:  exp/multiple_runs11/tmp/L03-1/L03-1_genome.2.fasta\n",
      "Sequence type: Nucleotide\n",
      "Deleted existing Nucleotide BLAST database named /home/kanai/documents/analysis/myTELR/syri/ipynb/tmp/essential_gene_blast_identification/blastdb/L03-1.2\n",
      "Keep MBits: T\n",
      "Maximum file size: 3000000000B\n",
      "Adding sequences from FASTA; added 1 sequences in 0.0225379 seconds.\n",
      "\n",
      "\n",
      "\n",
      "\n",
      "Building a new DB, current time: 07/06/2024 18:51:57\n",
      "New DB name:   /home/kanai/documents/analysis/myTELR/syri/ipynb/tmp/essential_gene_blast_identification/blastdb/L03-1.3\n",
      "New DB title:  exp/multiple_runs11/tmp/L03-1/L03-1_genome.3.fasta\n",
      "Sequence type: Nucleotide\n",
      "Deleted existing Nucleotide BLAST database named /home/kanai/documents/analysis/myTELR/syri/ipynb/tmp/essential_gene_blast_identification/blastdb/L03-1.3\n",
      "Keep MBits: T\n",
      "Maximum file size: 3000000000B\n",
      "Adding sequences from FASTA; added 1 sequences in 0.0214858 seconds.\n",
      "\n",
      "\n",
      "\n",
      "\n",
      "Building a new DB, current time: 07/06/2024 18:51:57\n",
      "New DB name:   /home/kanai/documents/analysis/myTELR/syri/ipynb/tmp/essential_gene_blast_identification/blastdb/L03-2.1\n",
      "New DB title:  exp/multiple_runs11/tmp/L03-2/L03-2_genome.1.fasta\n",
      "Sequence type: Nucleotide\n",
      "Deleted existing Nucleotide BLAST database named /home/kanai/documents/analysis/myTELR/syri/ipynb/tmp/essential_gene_blast_identification/blastdb/L03-2.1\n",
      "Keep MBits: T\n",
      "Maximum file size: 3000000000B\n",
      "Adding sequences from FASTA; added 1 sequences in 0.0223801 seconds.\n",
      "\n",
      "\n",
      "\n",
      "\n",
      "Building a new DB, current time: 07/06/2024 18:51:57\n",
      "New DB name:   /home/kanai/documents/analysis/myTELR/syri/ipynb/tmp/essential_gene_blast_identification/blastdb/L03-2.2\n",
      "New DB title:  exp/multiple_runs11/tmp/L03-2/L03-2_genome.2.fasta\n",
      "Sequence type: Nucleotide\n",
      "Deleted existing Nucleotide BLAST database named /home/kanai/documents/analysis/myTELR/syri/ipynb/tmp/essential_gene_blast_identification/blastdb/L03-2.2\n",
      "Keep MBits: T\n",
      "Maximum file size: 3000000000B\n",
      "Adding sequences from FASTA; added 1 sequences in 0.022162 seconds.\n",
      "\n",
      "\n",
      "\n",
      "\n",
      "Building a new DB, current time: 07/06/2024 18:51:57\n",
      "New DB name:   /home/kanai/documents/analysis/myTELR/syri/ipynb/tmp/essential_gene_blast_identification/blastdb/L03-2.3\n",
      "New DB title:  exp/multiple_runs11/tmp/L03-2/L03-2_genome.3.fasta\n",
      "Sequence type: Nucleotide\n",
      "Deleted existing Nucleotide BLAST database named /home/kanai/documents/analysis/myTELR/syri/ipynb/tmp/essential_gene_blast_identification/blastdb/L03-2.3\n",
      "Keep MBits: T\n",
      "Maximum file size: 3000000000B\n",
      "Adding sequences from FASTA; added 1 sequences in 0.022686 seconds.\n",
      "\n",
      "\n",
      "\n",
      "\n",
      "Building a new DB, current time: 07/06/2024 18:51:57\n",
      "New DB name:   /home/kanai/documents/analysis/myTELR/syri/ipynb/tmp/essential_gene_blast_identification/blastdb/L03-3.1\n",
      "New DB title:  exp/multiple_runs11/tmp/L03-3/L03-3_genome.1.fasta\n",
      "Sequence type: Nucleotide\n",
      "Deleted existing Nucleotide BLAST database named /home/kanai/documents/analysis/myTELR/syri/ipynb/tmp/essential_gene_blast_identification/blastdb/L03-3.1\n",
      "Keep MBits: T\n",
      "Maximum file size: 3000000000B\n",
      "Adding sequences from FASTA; added 1 sequences in 0.022692 seconds.\n",
      "\n",
      "\n",
      "\n",
      "\n",
      "Building a new DB, current time: 07/06/2024 18:51:58\n",
      "New DB name:   /home/kanai/documents/analysis/myTELR/syri/ipynb/tmp/essential_gene_blast_identification/blastdb/L03-3.2\n",
      "New DB title:  exp/multiple_runs11/tmp/L03-3/L03-3_genome.2.fasta\n",
      "Sequence type: Nucleotide\n",
      "Deleted existing Nucleotide BLAST database named /home/kanai/documents/analysis/myTELR/syri/ipynb/tmp/essential_gene_blast_identification/blastdb/L03-3.2\n",
      "Keep MBits: T\n",
      "Maximum file size: 3000000000B\n",
      "Adding sequences from FASTA; added 1 sequences in 0.0225701 seconds.\n",
      "\n",
      "\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\n",
      "\n",
      "Building a new DB, current time: 07/06/2024 18:51:58\n",
      "New DB name:   /home/kanai/documents/analysis/myTELR/syri/ipynb/tmp/essential_gene_blast_identification/blastdb/L03-3.3\n",
      "New DB title:  exp/multiple_runs11/tmp/L03-3/L03-3_genome.3.fasta\n",
      "Sequence type: Nucleotide\n",
      "Deleted existing Nucleotide BLAST database named /home/kanai/documents/analysis/myTELR/syri/ipynb/tmp/essential_gene_blast_identification/blastdb/L03-3.3\n",
      "Keep MBits: T\n",
      "Maximum file size: 3000000000B\n",
      "Adding sequences from FASTA; added 1 sequences in 0.021244 seconds.\n",
      "\n",
      "\n",
      "\n",
      "\n",
      "Building a new DB, current time: 07/06/2024 18:51:58\n",
      "New DB name:   /home/kanai/documents/analysis/myTELR/syri/ipynb/tmp/essential_gene_blast_identification/blastdb/L03-4.1\n",
      "New DB title:  exp/multiple_runs11/tmp/L03-4/L03-4_genome.1.fasta\n",
      "Sequence type: Nucleotide\n",
      "Deleted existing Nucleotide BLAST database named /home/kanai/documents/analysis/myTELR/syri/ipynb/tmp/essential_gene_blast_identification/blastdb/L03-4.1\n",
      "Keep MBits: T\n",
      "Maximum file size: 3000000000B\n",
      "Adding sequences from FASTA; added 1 sequences in 0.022073 seconds.\n",
      "\n",
      "\n",
      "\n",
      "\n",
      "Building a new DB, current time: 07/06/2024 18:51:58\n",
      "New DB name:   /home/kanai/documents/analysis/myTELR/syri/ipynb/tmp/essential_gene_blast_identification/blastdb/L03-4.2\n",
      "New DB title:  exp/multiple_runs11/tmp/L03-4/L03-4_genome.2.fasta\n",
      "Sequence type: Nucleotide\n",
      "Deleted existing Nucleotide BLAST database named /home/kanai/documents/analysis/myTELR/syri/ipynb/tmp/essential_gene_blast_identification/blastdb/L03-4.2\n",
      "Keep MBits: T\n",
      "Maximum file size: 3000000000B\n",
      "Adding sequences from FASTA; added 1 sequences in 0.0219791 seconds.\n",
      "\n",
      "\n",
      "\n",
      "\n",
      "Building a new DB, current time: 07/06/2024 18:51:58\n",
      "New DB name:   /home/kanai/documents/analysis/myTELR/syri/ipynb/tmp/essential_gene_blast_identification/blastdb/L03-4.3\n",
      "New DB title:  exp/multiple_runs11/tmp/L03-4/L03-4_genome.3.fasta\n",
      "Sequence type: Nucleotide\n",
      "Deleted existing Nucleotide BLAST database named /home/kanai/documents/analysis/myTELR/syri/ipynb/tmp/essential_gene_blast_identification/blastdb/L03-4.3\n",
      "Keep MBits: T\n",
      "Maximum file size: 3000000000B\n",
      "Adding sequences from FASTA; added 1 sequences in 0.0227799 seconds.\n",
      "\n",
      "\n",
      "\n",
      "\n",
      "Building a new DB, current time: 07/06/2024 18:51:58\n",
      "New DB name:   /home/kanai/documents/analysis/myTELR/syri/ipynb/tmp/essential_gene_blast_identification/blastdb/L04-1.1\n",
      "New DB title:  exp/multiple_runs11/tmp/L04-1/L04-1_genome.1.fasta\n",
      "Sequence type: Nucleotide\n",
      "Deleted existing Nucleotide BLAST database named /home/kanai/documents/analysis/myTELR/syri/ipynb/tmp/essential_gene_blast_identification/blastdb/L04-1.1\n",
      "Keep MBits: T\n",
      "Maximum file size: 3000000000B\n",
      "Adding sequences from FASTA; added 1 sequences in 0.022712 seconds.\n",
      "\n",
      "\n",
      "\n",
      "\n",
      "Building a new DB, current time: 07/06/2024 18:51:58\n",
      "New DB name:   /home/kanai/documents/analysis/myTELR/syri/ipynb/tmp/essential_gene_blast_identification/blastdb/L04-1.2\n",
      "New DB title:  exp/multiple_runs11/tmp/L04-1/L04-1_genome.2.fasta\n",
      "Sequence type: Nucleotide\n",
      "Deleted existing Nucleotide BLAST database named /home/kanai/documents/analysis/myTELR/syri/ipynb/tmp/essential_gene_blast_identification/blastdb/L04-1.2\n",
      "Keep MBits: T\n",
      "Maximum file size: 3000000000B\n",
      "Adding sequences from FASTA; added 1 sequences in 0.022083 seconds.\n",
      "\n",
      "\n",
      "\n",
      "\n",
      "Building a new DB, current time: 07/06/2024 18:51:58\n",
      "New DB name:   /home/kanai/documents/analysis/myTELR/syri/ipynb/tmp/essential_gene_blast_identification/blastdb/L04-1.3\n",
      "New DB title:  exp/multiple_runs11/tmp/L04-1/L04-1_genome.3.fasta\n",
      "Sequence type: Nucleotide\n",
      "Deleted existing Nucleotide BLAST database named /home/kanai/documents/analysis/myTELR/syri/ipynb/tmp/essential_gene_blast_identification/blastdb/L04-1.3\n",
      "Keep MBits: T\n",
      "Maximum file size: 3000000000B\n",
      "Adding sequences from FASTA; added 1 sequences in 0.022202 seconds.\n",
      "\n",
      "\n",
      "\n",
      "\n",
      "Building a new DB, current time: 07/06/2024 18:51:58\n",
      "New DB name:   /home/kanai/documents/analysis/myTELR/syri/ipynb/tmp/essential_gene_blast_identification/blastdb/L04-2.1\n",
      "New DB title:  exp/multiple_runs11/tmp/L04-2/L04-2_genome.1.fasta\n",
      "Sequence type: Nucleotide\n",
      "Deleted existing Nucleotide BLAST database named /home/kanai/documents/analysis/myTELR/syri/ipynb/tmp/essential_gene_blast_identification/blastdb/L04-2.1\n",
      "Keep MBits: T\n",
      "Maximum file size: 3000000000B\n",
      "Adding sequences from FASTA; added 1 sequences in 0.0235281 seconds.\n",
      "\n",
      "\n",
      "\n",
      "\n",
      "Building a new DB, current time: 07/06/2024 18:51:59\n",
      "New DB name:   /home/kanai/documents/analysis/myTELR/syri/ipynb/tmp/essential_gene_blast_identification/blastdb/L04-2.2\n",
      "New DB title:  exp/multiple_runs11/tmp/L04-2/L04-2_genome.2.fasta\n",
      "Sequence type: Nucleotide\n",
      "Deleted existing Nucleotide BLAST database named /home/kanai/documents/analysis/myTELR/syri/ipynb/tmp/essential_gene_blast_identification/blastdb/L04-2.2\n",
      "Keep MBits: T\n",
      "Maximum file size: 3000000000B\n",
      "Adding sequences from FASTA; added 1 sequences in 0.0218852 seconds.\n",
      "\n",
      "\n",
      "\n",
      "\n",
      "Building a new DB, current time: 07/06/2024 18:51:59\n",
      "New DB name:   /home/kanai/documents/analysis/myTELR/syri/ipynb/tmp/essential_gene_blast_identification/blastdb/L04-2.3\n",
      "New DB title:  exp/multiple_runs11/tmp/L04-2/L04-2_genome.3.fasta\n",
      "Sequence type: Nucleotide\n",
      "Deleted existing Nucleotide BLAST database named /home/kanai/documents/analysis/myTELR/syri/ipynb/tmp/essential_gene_blast_identification/blastdb/L04-2.3\n",
      "Keep MBits: T\n",
      "Maximum file size: 3000000000B\n",
      "Adding sequences from FASTA; added 1 sequences in 0.0206161 seconds.\n",
      "\n",
      "\n",
      "\n",
      "\n",
      "Building a new DB, current time: 07/06/2024 18:51:59\n",
      "New DB name:   /home/kanai/documents/analysis/myTELR/syri/ipynb/tmp/essential_gene_blast_identification/blastdb/L04-3.1\n",
      "New DB title:  exp/multiple_runs11/tmp/L04-3/L04-3_genome.1.fasta\n",
      "Sequence type: Nucleotide\n",
      "Deleted existing Nucleotide BLAST database named /home/kanai/documents/analysis/myTELR/syri/ipynb/tmp/essential_gene_blast_identification/blastdb/L04-3.1\n",
      "Keep MBits: T\n",
      "Maximum file size: 3000000000B\n",
      "Adding sequences from FASTA; added 1 sequences in 0.0226128 seconds.\n",
      "\n",
      "\n",
      "\n",
      "\n",
      "Building a new DB, current time: 07/06/2024 18:51:59\n",
      "New DB name:   /home/kanai/documents/analysis/myTELR/syri/ipynb/tmp/essential_gene_blast_identification/blastdb/L04-3.2\n",
      "New DB title:  exp/multiple_runs11/tmp/L04-3/L04-3_genome.2.fasta\n",
      "Sequence type: Nucleotide\n",
      "Deleted existing Nucleotide BLAST database named /home/kanai/documents/analysis/myTELR/syri/ipynb/tmp/essential_gene_blast_identification/blastdb/L04-3.2\n",
      "Keep MBits: T\n",
      "Maximum file size: 3000000000B\n",
      "Adding sequences from FASTA; added 1 sequences in 0.0243862 seconds.\n",
      "\n",
      "\n",
      "\n",
      "\n",
      "Building a new DB, current time: 07/06/2024 18:51:59\n",
      "New DB name:   /home/kanai/documents/analysis/myTELR/syri/ipynb/tmp/essential_gene_blast_identification/blastdb/L04-3.3\n",
      "New DB title:  exp/multiple_runs11/tmp/L04-3/L04-3_genome.3.fasta\n",
      "Sequence type: Nucleotide\n",
      "Deleted existing Nucleotide BLAST database named /home/kanai/documents/analysis/myTELR/syri/ipynb/tmp/essential_gene_blast_identification/blastdb/L04-3.3\n",
      "Keep MBits: T\n",
      "Maximum file size: 3000000000B\n",
      "Adding sequences from FASTA; added 1 sequences in 0.024045 seconds.\n",
      "\n",
      "\n",
      "\n",
      "\n",
      "Building a new DB, current time: 07/06/2024 18:51:59\n",
      "New DB name:   /home/kanai/documents/analysis/myTELR/syri/ipynb/tmp/essential_gene_blast_identification/blastdb/L04-4.1\n",
      "New DB title:  exp/multiple_runs11/tmp/L04-4/L04-4_genome.1.fasta\n",
      "Sequence type: Nucleotide\n",
      "Deleted existing Nucleotide BLAST database named /home/kanai/documents/analysis/myTELR/syri/ipynb/tmp/essential_gene_blast_identification/blastdb/L04-4.1\n",
      "Keep MBits: T\n",
      "Maximum file size: 3000000000B\n",
      "Adding sequences from FASTA; added 1 sequences in 0.0218861 seconds.\n",
      "\n",
      "\n",
      "\n",
      "\n",
      "Building a new DB, current time: 07/06/2024 18:51:59\n",
      "New DB name:   /home/kanai/documents/analysis/myTELR/syri/ipynb/tmp/essential_gene_blast_identification/blastdb/L04-4.2\n",
      "New DB title:  exp/multiple_runs11/tmp/L04-4/L04-4_genome.2.fasta\n",
      "Sequence type: Nucleotide\n",
      "Deleted existing Nucleotide BLAST database named /home/kanai/documents/analysis/myTELR/syri/ipynb/tmp/essential_gene_blast_identification/blastdb/L04-4.2\n",
      "Keep MBits: T\n",
      "Maximum file size: 3000000000B\n",
      "Adding sequences from FASTA; added 1 sequences in 0.0228939 seconds.\n",
      "\n",
      "\n",
      "\n",
      "\n",
      "Building a new DB, current time: 07/06/2024 18:51:59\n",
      "New DB name:   /home/kanai/documents/analysis/myTELR/syri/ipynb/tmp/essential_gene_blast_identification/blastdb/L04-4.3\n",
      "New DB title:  exp/multiple_runs11/tmp/L04-4/L04-4_genome.3.fasta\n",
      "Sequence type: Nucleotide\n",
      "Deleted existing Nucleotide BLAST database named /home/kanai/documents/analysis/myTELR/syri/ipynb/tmp/essential_gene_blast_identification/blastdb/L04-4.3\n",
      "Keep MBits: T\n",
      "Maximum file size: 3000000000B\n",
      "Adding sequences from FASTA; added 1 sequences in 0.023253 seconds.\n",
      "\n",
      "\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\n",
      "\n",
      "Building a new DB, current time: 07/06/2024 18:51:59\n",
      "New DB name:   /home/kanai/documents/analysis/myTELR/syri/ipynb/tmp/essential_gene_blast_identification/blastdb/L05-1.1\n",
      "New DB title:  exp/multiple_runs11/tmp/L05-1/L05-1_genome.1.fasta\n",
      "Sequence type: Nucleotide\n",
      "Deleted existing Nucleotide BLAST database named /home/kanai/documents/analysis/myTELR/syri/ipynb/tmp/essential_gene_blast_identification/blastdb/L05-1.1\n",
      "Keep MBits: T\n",
      "Maximum file size: 3000000000B\n",
      "Adding sequences from FASTA; added 1 sequences in 0.0216098 seconds.\n",
      "\n",
      "\n",
      "\n",
      "\n",
      "Building a new DB, current time: 07/06/2024 18:52:00\n",
      "New DB name:   /home/kanai/documents/analysis/myTELR/syri/ipynb/tmp/essential_gene_blast_identification/blastdb/L05-1.2\n",
      "New DB title:  exp/multiple_runs11/tmp/L05-1/L05-1_genome.2.fasta\n",
      "Sequence type: Nucleotide\n",
      "Deleted existing Nucleotide BLAST database named /home/kanai/documents/analysis/myTELR/syri/ipynb/tmp/essential_gene_blast_identification/blastdb/L05-1.2\n",
      "Keep MBits: T\n",
      "Maximum file size: 3000000000B\n",
      "Adding sequences from FASTA; added 1 sequences in 0.0218441 seconds.\n",
      "\n",
      "\n",
      "\n",
      "\n",
      "Building a new DB, current time: 07/06/2024 18:52:00\n",
      "New DB name:   /home/kanai/documents/analysis/myTELR/syri/ipynb/tmp/essential_gene_blast_identification/blastdb/L05-1.3\n",
      "New DB title:  exp/multiple_runs11/tmp/L05-1/L05-1_genome.3.fasta\n",
      "Sequence type: Nucleotide\n",
      "Deleted existing Nucleotide BLAST database named /home/kanai/documents/analysis/myTELR/syri/ipynb/tmp/essential_gene_blast_identification/blastdb/L05-1.3\n",
      "Keep MBits: T\n",
      "Maximum file size: 3000000000B\n",
      "Adding sequences from FASTA; added 1 sequences in 0.0224071 seconds.\n",
      "\n",
      "\n",
      "\n",
      "\n",
      "Building a new DB, current time: 07/06/2024 18:52:00\n",
      "New DB name:   /home/kanai/documents/analysis/myTELR/syri/ipynb/tmp/essential_gene_blast_identification/blastdb/L05-2.1\n",
      "New DB title:  exp/multiple_runs11/tmp/L05-2/L05-2_genome.1.fasta\n",
      "Sequence type: Nucleotide\n",
      "Deleted existing Nucleotide BLAST database named /home/kanai/documents/analysis/myTELR/syri/ipynb/tmp/essential_gene_blast_identification/blastdb/L05-2.1\n",
      "Keep MBits: T\n",
      "Maximum file size: 3000000000B\n",
      "Adding sequences from FASTA; added 1 sequences in 0.0217059 seconds.\n",
      "\n",
      "\n",
      "\n",
      "\n",
      "Building a new DB, current time: 07/06/2024 18:52:00\n",
      "New DB name:   /home/kanai/documents/analysis/myTELR/syri/ipynb/tmp/essential_gene_blast_identification/blastdb/L05-2.2\n",
      "New DB title:  exp/multiple_runs11/tmp/L05-2/L05-2_genome.2.fasta\n",
      "Sequence type: Nucleotide\n",
      "Deleted existing Nucleotide BLAST database named /home/kanai/documents/analysis/myTELR/syri/ipynb/tmp/essential_gene_blast_identification/blastdb/L05-2.2\n",
      "Keep MBits: T\n",
      "Maximum file size: 3000000000B\n",
      "Adding sequences from FASTA; added 1 sequences in 0.0219209 seconds.\n",
      "\n",
      "\n",
      "\n",
      "\n",
      "Building a new DB, current time: 07/06/2024 18:52:00\n",
      "New DB name:   /home/kanai/documents/analysis/myTELR/syri/ipynb/tmp/essential_gene_blast_identification/blastdb/L05-2.3\n",
      "New DB title:  exp/multiple_runs11/tmp/L05-2/L05-2_genome.3.fasta\n",
      "Sequence type: Nucleotide\n",
      "Deleted existing Nucleotide BLAST database named /home/kanai/documents/analysis/myTELR/syri/ipynb/tmp/essential_gene_blast_identification/blastdb/L05-2.3\n",
      "Keep MBits: T\n",
      "Maximum file size: 3000000000B\n",
      "Adding sequences from FASTA; added 1 sequences in 0.0217109 seconds.\n",
      "\n",
      "\n",
      "\n",
      "\n",
      "Building a new DB, current time: 07/06/2024 18:52:00\n",
      "New DB name:   /home/kanai/documents/analysis/myTELR/syri/ipynb/tmp/essential_gene_blast_identification/blastdb/L05-3.1\n",
      "New DB title:  exp/multiple_runs11/tmp/L05-3/L05-3_genome.1.fasta\n",
      "Sequence type: Nucleotide\n",
      "Deleted existing Nucleotide BLAST database named /home/kanai/documents/analysis/myTELR/syri/ipynb/tmp/essential_gene_blast_identification/blastdb/L05-3.1\n",
      "Keep MBits: T\n",
      "Maximum file size: 3000000000B\n",
      "Adding sequences from FASTA; added 1 sequences in 0.0216429 seconds.\n",
      "\n",
      "\n",
      "\n",
      "\n",
      "Building a new DB, current time: 07/06/2024 18:52:00\n",
      "New DB name:   /home/kanai/documents/analysis/myTELR/syri/ipynb/tmp/essential_gene_blast_identification/blastdb/L05-3.2\n",
      "New DB title:  exp/multiple_runs11/tmp/L05-3/L05-3_genome.2.fasta\n",
      "Sequence type: Nucleotide\n",
      "Deleted existing Nucleotide BLAST database named /home/kanai/documents/analysis/myTELR/syri/ipynb/tmp/essential_gene_blast_identification/blastdb/L05-3.2\n",
      "Keep MBits: T\n",
      "Maximum file size: 3000000000B\n",
      "Adding sequences from FASTA; added 1 sequences in 0.0204101 seconds.\n",
      "\n",
      "\n",
      "\n",
      "\n",
      "Building a new DB, current time: 07/06/2024 18:52:00\n",
      "New DB name:   /home/kanai/documents/analysis/myTELR/syri/ipynb/tmp/essential_gene_blast_identification/blastdb/L05-3.3\n",
      "New DB title:  exp/multiple_runs11/tmp/L05-3/L05-3_genome.3.fasta\n",
      "Sequence type: Nucleotide\n",
      "Deleted existing Nucleotide BLAST database named /home/kanai/documents/analysis/myTELR/syri/ipynb/tmp/essential_gene_blast_identification/blastdb/L05-3.3\n",
      "Keep MBits: T\n",
      "Maximum file size: 3000000000B\n",
      "Adding sequences from FASTA; added 1 sequences in 0.019345 seconds.\n",
      "\n",
      "\n",
      "\n",
      "\n",
      "Building a new DB, current time: 07/06/2024 18:52:00\n",
      "New DB name:   /home/kanai/documents/analysis/myTELR/syri/ipynb/tmp/essential_gene_blast_identification/blastdb/L05-4.1\n",
      "New DB title:  exp/multiple_runs11/tmp/L05-4/L05-4_genome.1.fasta\n",
      "Sequence type: Nucleotide\n",
      "Deleted existing Nucleotide BLAST database named /home/kanai/documents/analysis/myTELR/syri/ipynb/tmp/essential_gene_blast_identification/blastdb/L05-4.1\n",
      "Keep MBits: T\n",
      "Maximum file size: 3000000000B\n",
      "Adding sequences from FASTA; added 1 sequences in 0.0224941 seconds.\n",
      "\n",
      "\n",
      "\n",
      "\n",
      "Building a new DB, current time: 07/06/2024 18:52:01\n",
      "New DB name:   /home/kanai/documents/analysis/myTELR/syri/ipynb/tmp/essential_gene_blast_identification/blastdb/L05-4.2\n",
      "New DB title:  exp/multiple_runs11/tmp/L05-4/L05-4_genome.2.fasta\n",
      "Sequence type: Nucleotide\n",
      "Deleted existing Nucleotide BLAST database named /home/kanai/documents/analysis/myTELR/syri/ipynb/tmp/essential_gene_blast_identification/blastdb/L05-4.2\n",
      "Keep MBits: T\n",
      "Maximum file size: 3000000000B\n",
      "Adding sequences from FASTA; added 1 sequences in 0.0222919 seconds.\n",
      "\n",
      "\n",
      "\n",
      "\n",
      "Building a new DB, current time: 07/06/2024 18:52:01\n",
      "New DB name:   /home/kanai/documents/analysis/myTELR/syri/ipynb/tmp/essential_gene_blast_identification/blastdb/L05-4.3\n",
      "New DB title:  exp/multiple_runs11/tmp/L05-4/L05-4_genome.3.fasta\n",
      "Sequence type: Nucleotide\n",
      "Deleted existing Nucleotide BLAST database named /home/kanai/documents/analysis/myTELR/syri/ipynb/tmp/essential_gene_blast_identification/blastdb/L05-4.3\n",
      "Keep MBits: T\n",
      "Maximum file size: 3000000000B\n",
      "Adding sequences from FASTA; added 1 sequences in 0.0217199 seconds.\n",
      "\n",
      "\n",
      "\n",
      "\n",
      "Building a new DB, current time: 07/06/2024 18:52:01\n",
      "New DB name:   /home/kanai/documents/analysis/myTELR/syri/ipynb/tmp/essential_gene_blast_identification/blastdb/L06-1.1\n",
      "New DB title:  exp/multiple_runs11/tmp/L06-1/L06-1_genome.1.fasta\n",
      "Sequence type: Nucleotide\n",
      "Deleted existing Nucleotide BLAST database named /home/kanai/documents/analysis/myTELR/syri/ipynb/tmp/essential_gene_blast_identification/blastdb/L06-1.1\n",
      "Keep MBits: T\n",
      "Maximum file size: 3000000000B\n",
      "Adding sequences from FASTA; added 1 sequences in 0.0218289 seconds.\n",
      "\n",
      "\n",
      "\n",
      "\n",
      "Building a new DB, current time: 07/06/2024 18:52:01\n",
      "New DB name:   /home/kanai/documents/analysis/myTELR/syri/ipynb/tmp/essential_gene_blast_identification/blastdb/L06-1.2\n",
      "New DB title:  exp/multiple_runs11/tmp/L06-1/L06-1_genome.2.fasta\n",
      "Sequence type: Nucleotide\n",
      "Deleted existing Nucleotide BLAST database named /home/kanai/documents/analysis/myTELR/syri/ipynb/tmp/essential_gene_blast_identification/blastdb/L06-1.2\n",
      "Keep MBits: T\n",
      "Maximum file size: 3000000000B\n",
      "Adding sequences from FASTA; added 1 sequences in 0.0212419 seconds.\n",
      "\n",
      "\n",
      "\n",
      "\n",
      "Building a new DB, current time: 07/06/2024 18:52:01\n",
      "New DB name:   /home/kanai/documents/analysis/myTELR/syri/ipynb/tmp/essential_gene_blast_identification/blastdb/L06-1.3\n",
      "New DB title:  exp/multiple_runs11/tmp/L06-1/L06-1_genome.3.fasta\n",
      "Sequence type: Nucleotide\n",
      "Deleted existing Nucleotide BLAST database named /home/kanai/documents/analysis/myTELR/syri/ipynb/tmp/essential_gene_blast_identification/blastdb/L06-1.3\n",
      "Keep MBits: T\n",
      "Maximum file size: 3000000000B\n",
      "Adding sequences from FASTA; added 1 sequences in 0.022645 seconds.\n",
      "\n",
      "\n",
      "\n",
      "\n",
      "Building a new DB, current time: 07/06/2024 18:52:01\n",
      "New DB name:   /home/kanai/documents/analysis/myTELR/syri/ipynb/tmp/essential_gene_blast_identification/blastdb/L06-2.1\n",
      "New DB title:  exp/multiple_runs11/tmp/L06-2/L06-2_genome.1.fasta\n",
      "Sequence type: Nucleotide\n",
      "Deleted existing Nucleotide BLAST database named /home/kanai/documents/analysis/myTELR/syri/ipynb/tmp/essential_gene_blast_identification/blastdb/L06-2.1\n",
      "Keep MBits: T\n",
      "Maximum file size: 3000000000B\n",
      "Adding sequences from FASTA; added 1 sequences in 0.0224731 seconds.\n",
      "\n",
      "\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\n",
      "\n",
      "Building a new DB, current time: 07/06/2024 18:52:01\n",
      "New DB name:   /home/kanai/documents/analysis/myTELR/syri/ipynb/tmp/essential_gene_blast_identification/blastdb/L06-2.2\n",
      "New DB title:  exp/multiple_runs11/tmp/L06-2/L06-2_genome.2.fasta\n",
      "Sequence type: Nucleotide\n",
      "Deleted existing Nucleotide BLAST database named /home/kanai/documents/analysis/myTELR/syri/ipynb/tmp/essential_gene_blast_identification/blastdb/L06-2.2\n",
      "Keep MBits: T\n",
      "Maximum file size: 3000000000B\n",
      "Adding sequences from FASTA; added 1 sequences in 0.0212419 seconds.\n",
      "\n",
      "\n",
      "\n",
      "\n",
      "Building a new DB, current time: 07/06/2024 18:52:01\n",
      "New DB name:   /home/kanai/documents/analysis/myTELR/syri/ipynb/tmp/essential_gene_blast_identification/blastdb/L06-2.3\n",
      "New DB title:  exp/multiple_runs11/tmp/L06-2/L06-2_genome.3.fasta\n",
      "Sequence type: Nucleotide\n",
      "Deleted existing Nucleotide BLAST database named /home/kanai/documents/analysis/myTELR/syri/ipynb/tmp/essential_gene_blast_identification/blastdb/L06-2.3\n",
      "Keep MBits: T\n",
      "Maximum file size: 3000000000B\n",
      "Adding sequences from FASTA; added 1 sequences in 0.0219312 seconds.\n",
      "\n",
      "\n",
      "\n",
      "\n",
      "Building a new DB, current time: 07/06/2024 18:52:01\n",
      "New DB name:   /home/kanai/documents/analysis/myTELR/syri/ipynb/tmp/essential_gene_blast_identification/blastdb/L06-3.1\n",
      "New DB title:  exp/multiple_runs11/tmp/L06-3/L06-3_genome.1.fasta\n",
      "Sequence type: Nucleotide\n",
      "Deleted existing Nucleotide BLAST database named /home/kanai/documents/analysis/myTELR/syri/ipynb/tmp/essential_gene_blast_identification/blastdb/L06-3.1\n",
      "Keep MBits: T\n",
      "Maximum file size: 3000000000B\n",
      "Adding sequences from FASTA; added 1 sequences in 0.0233419 seconds.\n",
      "\n",
      "\n",
      "\n",
      "\n",
      "Building a new DB, current time: 07/06/2024 18:52:01\n",
      "New DB name:   /home/kanai/documents/analysis/myTELR/syri/ipynb/tmp/essential_gene_blast_identification/blastdb/L06-3.2\n",
      "New DB title:  exp/multiple_runs11/tmp/L06-3/L06-3_genome.2.fasta\n",
      "Sequence type: Nucleotide\n",
      "Deleted existing Nucleotide BLAST database named /home/kanai/documents/analysis/myTELR/syri/ipynb/tmp/essential_gene_blast_identification/blastdb/L06-3.2\n",
      "Keep MBits: T\n",
      "Maximum file size: 3000000000B\n",
      "Adding sequences from FASTA; added 1 sequences in 0.022512 seconds.\n",
      "\n",
      "\n",
      "\n",
      "\n",
      "Building a new DB, current time: 07/06/2024 18:52:02\n",
      "New DB name:   /home/kanai/documents/analysis/myTELR/syri/ipynb/tmp/essential_gene_blast_identification/blastdb/L06-3.3\n",
      "New DB title:  exp/multiple_runs11/tmp/L06-3/L06-3_genome.3.fasta\n",
      "Sequence type: Nucleotide\n",
      "Deleted existing Nucleotide BLAST database named /home/kanai/documents/analysis/myTELR/syri/ipynb/tmp/essential_gene_blast_identification/blastdb/L06-3.3\n",
      "Keep MBits: T\n",
      "Maximum file size: 3000000000B\n",
      "Adding sequences from FASTA; added 1 sequences in 0.0220509 seconds.\n",
      "\n",
      "\n",
      "\n",
      "\n",
      "Building a new DB, current time: 07/06/2024 18:52:02\n",
      "New DB name:   /home/kanai/documents/analysis/myTELR/syri/ipynb/tmp/essential_gene_blast_identification/blastdb/L06-4.1\n",
      "New DB title:  exp/multiple_runs11/tmp/L06-4/L06-4_genome.1.fasta\n",
      "Sequence type: Nucleotide\n",
      "Deleted existing Nucleotide BLAST database named /home/kanai/documents/analysis/myTELR/syri/ipynb/tmp/essential_gene_blast_identification/blastdb/L06-4.1\n",
      "Keep MBits: T\n",
      "Maximum file size: 3000000000B\n",
      "Adding sequences from FASTA; added 1 sequences in 0.0218041 seconds.\n",
      "\n",
      "\n",
      "\n",
      "\n",
      "Building a new DB, current time: 07/06/2024 18:52:02\n",
      "New DB name:   /home/kanai/documents/analysis/myTELR/syri/ipynb/tmp/essential_gene_blast_identification/blastdb/L06-4.2\n",
      "New DB title:  exp/multiple_runs11/tmp/L06-4/L06-4_genome.2.fasta\n",
      "Sequence type: Nucleotide\n",
      "Deleted existing Nucleotide BLAST database named /home/kanai/documents/analysis/myTELR/syri/ipynb/tmp/essential_gene_blast_identification/blastdb/L06-4.2\n",
      "Keep MBits: T\n",
      "Maximum file size: 3000000000B\n",
      "Adding sequences from FASTA; added 1 sequences in 0.021517 seconds.\n",
      "\n",
      "\n",
      "\n",
      "\n",
      "Building a new DB, current time: 07/06/2024 18:52:02\n",
      "New DB name:   /home/kanai/documents/analysis/myTELR/syri/ipynb/tmp/essential_gene_blast_identification/blastdb/L06-4.3\n",
      "New DB title:  exp/multiple_runs11/tmp/L06-4/L06-4_genome.3.fasta\n",
      "Sequence type: Nucleotide\n",
      "Deleted existing Nucleotide BLAST database named /home/kanai/documents/analysis/myTELR/syri/ipynb/tmp/essential_gene_blast_identification/blastdb/L06-4.3\n",
      "Keep MBits: T\n",
      "Maximum file size: 3000000000B\n",
      "Adding sequences from FASTA; added 1 sequences in 0.020978 seconds.\n",
      "\n",
      "\n",
      "\n",
      "\n",
      "Building a new DB, current time: 07/06/2024 18:52:02\n",
      "New DB name:   /home/kanai/documents/analysis/myTELR/syri/ipynb/tmp/essential_gene_blast_identification/blastdb/L07-1.1\n",
      "New DB title:  exp/multiple_runs11/tmp/L07-1/L07-1_genome.1.fasta\n",
      "Sequence type: Nucleotide\n",
      "Deleted existing Nucleotide BLAST database named /home/kanai/documents/analysis/myTELR/syri/ipynb/tmp/essential_gene_blast_identification/blastdb/L07-1.1\n",
      "Keep MBits: T\n",
      "Maximum file size: 3000000000B\n",
      "Adding sequences from FASTA; added 1 sequences in 0.021873 seconds.\n",
      "\n",
      "\n",
      "\n",
      "\n",
      "Building a new DB, current time: 07/06/2024 18:52:02\n",
      "New DB name:   /home/kanai/documents/analysis/myTELR/syri/ipynb/tmp/essential_gene_blast_identification/blastdb/L07-1.2\n",
      "New DB title:  exp/multiple_runs11/tmp/L07-1/L07-1_genome.2.fasta\n",
      "Sequence type: Nucleotide\n",
      "Deleted existing Nucleotide BLAST database named /home/kanai/documents/analysis/myTELR/syri/ipynb/tmp/essential_gene_blast_identification/blastdb/L07-1.2\n",
      "Keep MBits: T\n",
      "Maximum file size: 3000000000B\n",
      "Adding sequences from FASTA; added 1 sequences in 0.022258 seconds.\n",
      "\n",
      "\n",
      "\n",
      "\n",
      "Building a new DB, current time: 07/06/2024 18:52:02\n",
      "New DB name:   /home/kanai/documents/analysis/myTELR/syri/ipynb/tmp/essential_gene_blast_identification/blastdb/L07-1.3\n",
      "New DB title:  exp/multiple_runs11/tmp/L07-1/L07-1_genome.3.fasta\n",
      "Sequence type: Nucleotide\n",
      "Deleted existing Nucleotide BLAST database named /home/kanai/documents/analysis/myTELR/syri/ipynb/tmp/essential_gene_blast_identification/blastdb/L07-1.3\n",
      "Keep MBits: T\n",
      "Maximum file size: 3000000000B\n",
      "Adding sequences from FASTA; added 1 sequences in 0.0211582 seconds.\n",
      "\n",
      "\n",
      "\n",
      "\n",
      "Building a new DB, current time: 07/06/2024 18:52:02\n",
      "New DB name:   /home/kanai/documents/analysis/myTELR/syri/ipynb/tmp/essential_gene_blast_identification/blastdb/L07-2.1\n",
      "New DB title:  exp/multiple_runs11/tmp/L07-2/L07-2_genome.1.fasta\n",
      "Sequence type: Nucleotide\n",
      "Deleted existing Nucleotide BLAST database named /home/kanai/documents/analysis/myTELR/syri/ipynb/tmp/essential_gene_blast_identification/blastdb/L07-2.1\n",
      "Keep MBits: T\n",
      "Maximum file size: 3000000000B\n",
      "Adding sequences from FASTA; added 1 sequences in 0.0221732 seconds.\n",
      "\n",
      "\n",
      "\n",
      "\n",
      "Building a new DB, current time: 07/06/2024 18:52:02\n",
      "New DB name:   /home/kanai/documents/analysis/myTELR/syri/ipynb/tmp/essential_gene_blast_identification/blastdb/L07-2.2\n",
      "New DB title:  exp/multiple_runs11/tmp/L07-2/L07-2_genome.2.fasta\n",
      "Sequence type: Nucleotide\n",
      "Deleted existing Nucleotide BLAST database named /home/kanai/documents/analysis/myTELR/syri/ipynb/tmp/essential_gene_blast_identification/blastdb/L07-2.2\n",
      "Keep MBits: T\n",
      "Maximum file size: 3000000000B\n",
      "Adding sequences from FASTA; added 1 sequences in 0.022661 seconds.\n",
      "\n",
      "\n",
      "\n",
      "\n",
      "Building a new DB, current time: 07/06/2024 18:52:03\n",
      "New DB name:   /home/kanai/documents/analysis/myTELR/syri/ipynb/tmp/essential_gene_blast_identification/blastdb/L07-2.3\n",
      "New DB title:  exp/multiple_runs11/tmp/L07-2/L07-2_genome.3.fasta\n",
      "Sequence type: Nucleotide\n",
      "Deleted existing Nucleotide BLAST database named /home/kanai/documents/analysis/myTELR/syri/ipynb/tmp/essential_gene_blast_identification/blastdb/L07-2.3\n",
      "Keep MBits: T\n",
      "Maximum file size: 3000000000B\n",
      "Adding sequences from FASTA; added 1 sequences in 0.021174 seconds.\n",
      "\n",
      "\n",
      "\n",
      "\n",
      "Building a new DB, current time: 07/06/2024 18:52:03\n",
      "New DB name:   /home/kanai/documents/analysis/myTELR/syri/ipynb/tmp/essential_gene_blast_identification/blastdb/L07-3.1\n",
      "New DB title:  exp/multiple_runs11/tmp/L07-3/L07-3_genome.1.fasta\n",
      "Sequence type: Nucleotide\n",
      "Deleted existing Nucleotide BLAST database named /home/kanai/documents/analysis/myTELR/syri/ipynb/tmp/essential_gene_blast_identification/blastdb/L07-3.1\n",
      "Keep MBits: T\n",
      "Maximum file size: 3000000000B\n",
      "Adding sequences from FASTA; added 1 sequences in 0.021889 seconds.\n",
      "\n",
      "\n",
      "\n",
      "\n",
      "Building a new DB, current time: 07/06/2024 18:52:03\n",
      "New DB name:   /home/kanai/documents/analysis/myTELR/syri/ipynb/tmp/essential_gene_blast_identification/blastdb/L07-3.2\n",
      "New DB title:  exp/multiple_runs11/tmp/L07-3/L07-3_genome.2.fasta\n",
      "Sequence type: Nucleotide\n",
      "Deleted existing Nucleotide BLAST database named /home/kanai/documents/analysis/myTELR/syri/ipynb/tmp/essential_gene_blast_identification/blastdb/L07-3.2\n",
      "Keep MBits: T\n",
      "Maximum file size: 3000000000B\n",
      "Adding sequences from FASTA; added 1 sequences in 0.0211661 seconds.\n",
      "\n",
      "\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\n",
      "\n",
      "Building a new DB, current time: 07/06/2024 18:52:03\n",
      "New DB name:   /home/kanai/documents/analysis/myTELR/syri/ipynb/tmp/essential_gene_blast_identification/blastdb/L07-3.3\n",
      "New DB title:  exp/multiple_runs11/tmp/L07-3/L07-3_genome.3.fasta\n",
      "Sequence type: Nucleotide\n",
      "Deleted existing Nucleotide BLAST database named /home/kanai/documents/analysis/myTELR/syri/ipynb/tmp/essential_gene_blast_identification/blastdb/L07-3.3\n",
      "Keep MBits: T\n",
      "Maximum file size: 3000000000B\n",
      "Adding sequences from FASTA; added 1 sequences in 0.0219769 seconds.\n",
      "\n",
      "\n",
      "\n",
      "\n",
      "Building a new DB, current time: 07/06/2024 18:52:03\n",
      "New DB name:   /home/kanai/documents/analysis/myTELR/syri/ipynb/tmp/essential_gene_blast_identification/blastdb/L07-4.1\n",
      "New DB title:  exp/multiple_runs11/tmp/L07-4/L07-4_genome.1.fasta\n",
      "Sequence type: Nucleotide\n",
      "Deleted existing Nucleotide BLAST database named /home/kanai/documents/analysis/myTELR/syri/ipynb/tmp/essential_gene_blast_identification/blastdb/L07-4.1\n",
      "Keep MBits: T\n",
      "Maximum file size: 3000000000B\n",
      "Adding sequences from FASTA; added 1 sequences in 0.0212591 seconds.\n",
      "\n",
      "\n",
      "\n",
      "\n",
      "Building a new DB, current time: 07/06/2024 18:52:03\n",
      "New DB name:   /home/kanai/documents/analysis/myTELR/syri/ipynb/tmp/essential_gene_blast_identification/blastdb/L07-4.2\n",
      "New DB title:  exp/multiple_runs11/tmp/L07-4/L07-4_genome.2.fasta\n",
      "Sequence type: Nucleotide\n",
      "Deleted existing Nucleotide BLAST database named /home/kanai/documents/analysis/myTELR/syri/ipynb/tmp/essential_gene_blast_identification/blastdb/L07-4.2\n",
      "Keep MBits: T\n",
      "Maximum file size: 3000000000B\n",
      "Adding sequences from FASTA; added 1 sequences in 0.0218132 seconds.\n",
      "\n",
      "\n",
      "\n",
      "\n",
      "Building a new DB, current time: 07/06/2024 18:52:03\n",
      "New DB name:   /home/kanai/documents/analysis/myTELR/syri/ipynb/tmp/essential_gene_blast_identification/blastdb/L07-4.3\n",
      "New DB title:  exp/multiple_runs11/tmp/L07-4/L07-4_genome.3.fasta\n",
      "Sequence type: Nucleotide\n",
      "Deleted existing Nucleotide BLAST database named /home/kanai/documents/analysis/myTELR/syri/ipynb/tmp/essential_gene_blast_identification/blastdb/L07-4.3\n",
      "Keep MBits: T\n",
      "Maximum file size: 3000000000B\n",
      "Adding sequences from FASTA; added 1 sequences in 0.0210309 seconds.\n",
      "\n",
      "\n",
      "\n",
      "\n",
      "Building a new DB, current time: 07/06/2024 18:52:03\n",
      "New DB name:   /home/kanai/documents/analysis/myTELR/syri/ipynb/tmp/essential_gene_blast_identification/blastdb/L08-1.1\n",
      "New DB title:  exp/multiple_runs11/tmp/L08-1/L08-1_genome.1.fasta\n",
      "Sequence type: Nucleotide\n",
      "Deleted existing Nucleotide BLAST database named /home/kanai/documents/analysis/myTELR/syri/ipynb/tmp/essential_gene_blast_identification/blastdb/L08-1.1\n",
      "Keep MBits: T\n",
      "Maximum file size: 3000000000B\n",
      "Adding sequences from FASTA; added 1 sequences in 0.0220339 seconds.\n",
      "\n",
      "\n",
      "\n",
      "\n",
      "Building a new DB, current time: 07/06/2024 18:52:03\n",
      "New DB name:   /home/kanai/documents/analysis/myTELR/syri/ipynb/tmp/essential_gene_blast_identification/blastdb/L08-1.2\n",
      "New DB title:  exp/multiple_runs11/tmp/L08-1/L08-1_genome.2.fasta\n",
      "Sequence type: Nucleotide\n",
      "Deleted existing Nucleotide BLAST database named /home/kanai/documents/analysis/myTELR/syri/ipynb/tmp/essential_gene_blast_identification/blastdb/L08-1.2\n",
      "Keep MBits: T\n",
      "Maximum file size: 3000000000B\n",
      "Adding sequences from FASTA; added 1 sequences in 0.020072 seconds.\n",
      "\n",
      "\n",
      "\n",
      "\n",
      "Building a new DB, current time: 07/06/2024 18:52:04\n",
      "New DB name:   /home/kanai/documents/analysis/myTELR/syri/ipynb/tmp/essential_gene_blast_identification/blastdb/L08-1.3\n",
      "New DB title:  exp/multiple_runs11/tmp/L08-1/L08-1_genome.3.fasta\n",
      "Sequence type: Nucleotide\n",
      "Deleted existing Nucleotide BLAST database named /home/kanai/documents/analysis/myTELR/syri/ipynb/tmp/essential_gene_blast_identification/blastdb/L08-1.3\n",
      "Keep MBits: T\n",
      "Maximum file size: 3000000000B\n",
      "Adding sequences from FASTA; added 1 sequences in 0.0205309 seconds.\n",
      "\n",
      "\n",
      "\n",
      "\n",
      "Building a new DB, current time: 07/06/2024 18:52:04\n",
      "New DB name:   /home/kanai/documents/analysis/myTELR/syri/ipynb/tmp/essential_gene_blast_identification/blastdb/L08-2.1\n",
      "New DB title:  exp/multiple_runs11/tmp/L08-2/L08-2_genome.1.fasta\n",
      "Sequence type: Nucleotide\n",
      "Deleted existing Nucleotide BLAST database named /home/kanai/documents/analysis/myTELR/syri/ipynb/tmp/essential_gene_blast_identification/blastdb/L08-2.1\n",
      "Keep MBits: T\n",
      "Maximum file size: 3000000000B\n",
      "Adding sequences from FASTA; added 1 sequences in 0.0215631 seconds.\n",
      "\n",
      "\n",
      "\n",
      "\n",
      "Building a new DB, current time: 07/06/2024 18:52:04\n",
      "New DB name:   /home/kanai/documents/analysis/myTELR/syri/ipynb/tmp/essential_gene_blast_identification/blastdb/L08-2.2\n",
      "New DB title:  exp/multiple_runs11/tmp/L08-2/L08-2_genome.2.fasta\n",
      "Sequence type: Nucleotide\n",
      "Deleted existing Nucleotide BLAST database named /home/kanai/documents/analysis/myTELR/syri/ipynb/tmp/essential_gene_blast_identification/blastdb/L08-2.2\n",
      "Keep MBits: T\n",
      "Maximum file size: 3000000000B\n",
      "Adding sequences from FASTA; added 1 sequences in 0.021822 seconds.\n",
      "\n",
      "\n",
      "\n",
      "\n",
      "Building a new DB, current time: 07/06/2024 18:52:04\n",
      "New DB name:   /home/kanai/documents/analysis/myTELR/syri/ipynb/tmp/essential_gene_blast_identification/blastdb/L08-2.3\n",
      "New DB title:  exp/multiple_runs11/tmp/L08-2/L08-2_genome.3.fasta\n",
      "Sequence type: Nucleotide\n",
      "Deleted existing Nucleotide BLAST database named /home/kanai/documents/analysis/myTELR/syri/ipynb/tmp/essential_gene_blast_identification/blastdb/L08-2.3\n",
      "Keep MBits: T\n",
      "Maximum file size: 3000000000B\n",
      "Adding sequences from FASTA; added 1 sequences in 0.021322 seconds.\n",
      "\n",
      "\n",
      "\n",
      "\n",
      "Building a new DB, current time: 07/06/2024 18:52:04\n",
      "New DB name:   /home/kanai/documents/analysis/myTELR/syri/ipynb/tmp/essential_gene_blast_identification/blastdb/L08-3.1\n",
      "New DB title:  exp/multiple_runs11/tmp/L08-3/L08-3_genome.1.fasta\n",
      "Sequence type: Nucleotide\n",
      "Deleted existing Nucleotide BLAST database named /home/kanai/documents/analysis/myTELR/syri/ipynb/tmp/essential_gene_blast_identification/blastdb/L08-3.1\n",
      "Keep MBits: T\n",
      "Maximum file size: 3000000000B\n",
      "Adding sequences from FASTA; added 1 sequences in 0.0219741 seconds.\n",
      "\n",
      "\n",
      "\n",
      "\n",
      "Building a new DB, current time: 07/06/2024 18:52:04\n",
      "New DB name:   /home/kanai/documents/analysis/myTELR/syri/ipynb/tmp/essential_gene_blast_identification/blastdb/L08-3.2\n",
      "New DB title:  exp/multiple_runs11/tmp/L08-3/L08-3_genome.2.fasta\n",
      "Sequence type: Nucleotide\n",
      "Deleted existing Nucleotide BLAST database named /home/kanai/documents/analysis/myTELR/syri/ipynb/tmp/essential_gene_blast_identification/blastdb/L08-3.2\n",
      "Keep MBits: T\n",
      "Maximum file size: 3000000000B\n",
      "Adding sequences from FASTA; added 1 sequences in 0.0225689 seconds.\n",
      "\n",
      "\n",
      "\n",
      "\n",
      "Building a new DB, current time: 07/06/2024 18:52:04\n",
      "New DB name:   /home/kanai/documents/analysis/myTELR/syri/ipynb/tmp/essential_gene_blast_identification/blastdb/L08-3.3\n",
      "New DB title:  exp/multiple_runs11/tmp/L08-3/L08-3_genome.3.fasta\n",
      "Sequence type: Nucleotide\n",
      "Deleted existing Nucleotide BLAST database named /home/kanai/documents/analysis/myTELR/syri/ipynb/tmp/essential_gene_blast_identification/blastdb/L08-3.3\n",
      "Keep MBits: T\n",
      "Maximum file size: 3000000000B\n",
      "Adding sequences from FASTA; added 1 sequences in 0.0222988 seconds.\n",
      "\n",
      "\n",
      "\n",
      "\n",
      "Building a new DB, current time: 07/06/2024 18:52:04\n",
      "New DB name:   /home/kanai/documents/analysis/myTELR/syri/ipynb/tmp/essential_gene_blast_identification/blastdb/L08-4.1\n",
      "New DB title:  exp/multiple_runs11/tmp/L08-4/L08-4_genome.1.fasta\n",
      "Sequence type: Nucleotide\n",
      "Deleted existing Nucleotide BLAST database named /home/kanai/documents/analysis/myTELR/syri/ipynb/tmp/essential_gene_blast_identification/blastdb/L08-4.1\n",
      "Keep MBits: T\n",
      "Maximum file size: 3000000000B\n",
      "Adding sequences from FASTA; added 1 sequences in 0.0212591 seconds.\n",
      "\n",
      "\n",
      "\n",
      "\n",
      "Building a new DB, current time: 07/06/2024 18:52:04\n",
      "New DB name:   /home/kanai/documents/analysis/myTELR/syri/ipynb/tmp/essential_gene_blast_identification/blastdb/L08-4.2\n",
      "New DB title:  exp/multiple_runs11/tmp/L08-4/L08-4_genome.2.fasta\n",
      "Sequence type: Nucleotide\n",
      "Deleted existing Nucleotide BLAST database named /home/kanai/documents/analysis/myTELR/syri/ipynb/tmp/essential_gene_blast_identification/blastdb/L08-4.2\n",
      "Keep MBits: T\n",
      "Maximum file size: 3000000000B\n",
      "Adding sequences from FASTA; added 1 sequences in 0.0219469 seconds.\n",
      "\n",
      "\n",
      "\n",
      "\n",
      "Building a new DB, current time: 07/06/2024 18:52:04\n",
      "New DB name:   /home/kanai/documents/analysis/myTELR/syri/ipynb/tmp/essential_gene_blast_identification/blastdb/L08-4.3\n",
      "New DB title:  exp/multiple_runs11/tmp/L08-4/L08-4_genome.3.fasta\n",
      "Sequence type: Nucleotide\n",
      "Deleted existing Nucleotide BLAST database named /home/kanai/documents/analysis/myTELR/syri/ipynb/tmp/essential_gene_blast_identification/blastdb/L08-4.3\n",
      "Keep MBits: T\n",
      "Maximum file size: 3000000000B\n",
      "Adding sequences from FASTA; added 1 sequences in 0.0214329 seconds.\n",
      "\n",
      "\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\n",
      "\n",
      "Building a new DB, current time: 07/06/2024 18:52:05\n",
      "New DB name:   /home/kanai/documents/analysis/myTELR/syri/ipynb/tmp/essential_gene_blast_identification/blastdb/L09-1.1\n",
      "New DB title:  exp/multiple_runs11/tmp/L09-1/L09-1_genome.1.fasta\n",
      "Sequence type: Nucleotide\n",
      "Deleted existing Nucleotide BLAST database named /home/kanai/documents/analysis/myTELR/syri/ipynb/tmp/essential_gene_blast_identification/blastdb/L09-1.1\n",
      "Keep MBits: T\n",
      "Maximum file size: 3000000000B\n",
      "Adding sequences from FASTA; added 1 sequences in 0.0212591 seconds.\n",
      "\n",
      "\n",
      "\n",
      "\n",
      "Building a new DB, current time: 07/06/2024 18:52:05\n",
      "New DB name:   /home/kanai/documents/analysis/myTELR/syri/ipynb/tmp/essential_gene_blast_identification/blastdb/L09-1.2\n",
      "New DB title:  exp/multiple_runs11/tmp/L09-1/L09-1_genome.2.fasta\n",
      "Sequence type: Nucleotide\n",
      "Deleted existing Nucleotide BLAST database named /home/kanai/documents/analysis/myTELR/syri/ipynb/tmp/essential_gene_blast_identification/blastdb/L09-1.2\n",
      "Keep MBits: T\n",
      "Maximum file size: 3000000000B\n",
      "Adding sequences from FASTA; added 1 sequences in 0.0229399 seconds.\n",
      "\n",
      "\n",
      "\n",
      "\n",
      "Building a new DB, current time: 07/06/2024 18:52:05\n",
      "New DB name:   /home/kanai/documents/analysis/myTELR/syri/ipynb/tmp/essential_gene_blast_identification/blastdb/L09-1.3\n",
      "New DB title:  exp/multiple_runs11/tmp/L09-1/L09-1_genome.3.fasta\n",
      "Sequence type: Nucleotide\n",
      "Deleted existing Nucleotide BLAST database named /home/kanai/documents/analysis/myTELR/syri/ipynb/tmp/essential_gene_blast_identification/blastdb/L09-1.3\n",
      "Keep MBits: T\n",
      "Maximum file size: 3000000000B\n",
      "Adding sequences from FASTA; added 1 sequences in 0.021229 seconds.\n",
      "\n",
      "\n",
      "\n",
      "\n",
      "Building a new DB, current time: 07/06/2024 18:52:05\n",
      "New DB name:   /home/kanai/documents/analysis/myTELR/syri/ipynb/tmp/essential_gene_blast_identification/blastdb/L09-2.1\n",
      "New DB title:  exp/multiple_runs11/tmp/L09-2/L09-2_genome.1.fasta\n",
      "Sequence type: Nucleotide\n",
      "Deleted existing Nucleotide BLAST database named /home/kanai/documents/analysis/myTELR/syri/ipynb/tmp/essential_gene_blast_identification/blastdb/L09-2.1\n",
      "Keep MBits: T\n",
      "Maximum file size: 3000000000B\n",
      "Adding sequences from FASTA; added 1 sequences in 0.0233309 seconds.\n",
      "\n",
      "\n",
      "\n",
      "\n",
      "Building a new DB, current time: 07/06/2024 18:52:05\n",
      "New DB name:   /home/kanai/documents/analysis/myTELR/syri/ipynb/tmp/essential_gene_blast_identification/blastdb/L09-2.2\n",
      "New DB title:  exp/multiple_runs11/tmp/L09-2/L09-2_genome.2.fasta\n",
      "Sequence type: Nucleotide\n",
      "Deleted existing Nucleotide BLAST database named /home/kanai/documents/analysis/myTELR/syri/ipynb/tmp/essential_gene_blast_identification/blastdb/L09-2.2\n",
      "Keep MBits: T\n",
      "Maximum file size: 3000000000B\n",
      "Adding sequences from FASTA; added 1 sequences in 0.021101 seconds.\n",
      "\n",
      "\n",
      "\n",
      "\n",
      "Building a new DB, current time: 07/06/2024 18:52:05\n",
      "New DB name:   /home/kanai/documents/analysis/myTELR/syri/ipynb/tmp/essential_gene_blast_identification/blastdb/L09-2.3\n",
      "New DB title:  exp/multiple_runs11/tmp/L09-2/L09-2_genome.3.fasta\n",
      "Sequence type: Nucleotide\n",
      "Deleted existing Nucleotide BLAST database named /home/kanai/documents/analysis/myTELR/syri/ipynb/tmp/essential_gene_blast_identification/blastdb/L09-2.3\n",
      "Keep MBits: T\n",
      "Maximum file size: 3000000000B\n",
      "Adding sequences from FASTA; added 1 sequences in 0.022289 seconds.\n",
      "\n",
      "\n",
      "\n",
      "\n",
      "Building a new DB, current time: 07/06/2024 18:52:05\n",
      "New DB name:   /home/kanai/documents/analysis/myTELR/syri/ipynb/tmp/essential_gene_blast_identification/blastdb/L09-3.1\n",
      "New DB title:  exp/multiple_runs11/tmp/L09-3/L09-3_genome.1.fasta\n",
      "Sequence type: Nucleotide\n",
      "Deleted existing Nucleotide BLAST database named /home/kanai/documents/analysis/myTELR/syri/ipynb/tmp/essential_gene_blast_identification/blastdb/L09-3.1\n",
      "Keep MBits: T\n",
      "Maximum file size: 3000000000B\n",
      "Adding sequences from FASTA; added 1 sequences in 0.0214882 seconds.\n",
      "\n",
      "\n",
      "\n",
      "\n",
      "Building a new DB, current time: 07/06/2024 18:52:05\n",
      "New DB name:   /home/kanai/documents/analysis/myTELR/syri/ipynb/tmp/essential_gene_blast_identification/blastdb/L09-3.2\n",
      "New DB title:  exp/multiple_runs11/tmp/L09-3/L09-3_genome.2.fasta\n",
      "Sequence type: Nucleotide\n",
      "Deleted existing Nucleotide BLAST database named /home/kanai/documents/analysis/myTELR/syri/ipynb/tmp/essential_gene_blast_identification/blastdb/L09-3.2\n",
      "Keep MBits: T\n",
      "Maximum file size: 3000000000B\n",
      "Adding sequences from FASTA; added 1 sequences in 0.022305 seconds.\n",
      "\n",
      "\n",
      "\n",
      "\n",
      "Building a new DB, current time: 07/06/2024 18:52:05\n",
      "New DB name:   /home/kanai/documents/analysis/myTELR/syri/ipynb/tmp/essential_gene_blast_identification/blastdb/L09-3.3\n",
      "New DB title:  exp/multiple_runs11/tmp/L09-3/L09-3_genome.3.fasta\n",
      "Sequence type: Nucleotide\n",
      "Deleted existing Nucleotide BLAST database named /home/kanai/documents/analysis/myTELR/syri/ipynb/tmp/essential_gene_blast_identification/blastdb/L09-3.3\n",
      "Keep MBits: T\n",
      "Maximum file size: 3000000000B\n",
      "Adding sequences from FASTA; added 1 sequences in 0.0213861 seconds.\n",
      "\n",
      "\n",
      "\n",
      "\n",
      "Building a new DB, current time: 07/06/2024 18:52:06\n",
      "New DB name:   /home/kanai/documents/analysis/myTELR/syri/ipynb/tmp/essential_gene_blast_identification/blastdb/L09-4.1\n",
      "New DB title:  exp/multiple_runs11/tmp/L09-4/L09-4_genome.1.fasta\n",
      "Sequence type: Nucleotide\n",
      "Deleted existing Nucleotide BLAST database named /home/kanai/documents/analysis/myTELR/syri/ipynb/tmp/essential_gene_blast_identification/blastdb/L09-4.1\n",
      "Keep MBits: T\n",
      "Maximum file size: 3000000000B\n",
      "Adding sequences from FASTA; added 1 sequences in 0.0220232 seconds.\n",
      "\n",
      "\n",
      "\n",
      "\n",
      "Building a new DB, current time: 07/06/2024 18:52:06\n",
      "New DB name:   /home/kanai/documents/analysis/myTELR/syri/ipynb/tmp/essential_gene_blast_identification/blastdb/L09-4.2\n",
      "New DB title:  exp/multiple_runs11/tmp/L09-4/L09-4_genome.2.fasta\n",
      "Sequence type: Nucleotide\n",
      "Deleted existing Nucleotide BLAST database named /home/kanai/documents/analysis/myTELR/syri/ipynb/tmp/essential_gene_blast_identification/blastdb/L09-4.2\n",
      "Keep MBits: T\n",
      "Maximum file size: 3000000000B\n",
      "Adding sequences from FASTA; added 1 sequences in 0.0224628 seconds.\n",
      "\n",
      "\n",
      "\n",
      "\n",
      "Building a new DB, current time: 07/06/2024 18:52:06\n",
      "New DB name:   /home/kanai/documents/analysis/myTELR/syri/ipynb/tmp/essential_gene_blast_identification/blastdb/L09-4.3\n",
      "New DB title:  exp/multiple_runs11/tmp/L09-4/L09-4_genome.3.fasta\n",
      "Sequence type: Nucleotide\n",
      "Deleted existing Nucleotide BLAST database named /home/kanai/documents/analysis/myTELR/syri/ipynb/tmp/essential_gene_blast_identification/blastdb/L09-4.3\n",
      "Keep MBits: T\n",
      "Maximum file size: 3000000000B\n",
      "Adding sequences from FASTA; added 1 sequences in 0.02227 seconds.\n",
      "\n",
      "\n",
      "\n",
      "\n",
      "Building a new DB, current time: 07/06/2024 18:52:06\n",
      "New DB name:   /home/kanai/documents/analysis/myTELR/syri/ipynb/tmp/essential_gene_blast_identification/blastdb/L10-1.1\n",
      "New DB title:  exp/multiple_runs11/tmp/L10-1/L10-1_genome.1.fasta\n",
      "Sequence type: Nucleotide\n",
      "Deleted existing Nucleotide BLAST database named /home/kanai/documents/analysis/myTELR/syri/ipynb/tmp/essential_gene_blast_identification/blastdb/L10-1.1\n",
      "Keep MBits: T\n",
      "Maximum file size: 3000000000B\n",
      "Adding sequences from FASTA; added 1 sequences in 0.0212932 seconds.\n",
      "\n",
      "\n",
      "\n",
      "\n",
      "Building a new DB, current time: 07/06/2024 18:52:06\n",
      "New DB name:   /home/kanai/documents/analysis/myTELR/syri/ipynb/tmp/essential_gene_blast_identification/blastdb/L10-1.2\n",
      "New DB title:  exp/multiple_runs11/tmp/L10-1/L10-1_genome.2.fasta\n",
      "Sequence type: Nucleotide\n",
      "Deleted existing Nucleotide BLAST database named /home/kanai/documents/analysis/myTELR/syri/ipynb/tmp/essential_gene_blast_identification/blastdb/L10-1.2\n",
      "Keep MBits: T\n",
      "Maximum file size: 3000000000B\n",
      "Adding sequences from FASTA; added 1 sequences in 0.0219059 seconds.\n",
      "\n",
      "\n",
      "\n",
      "\n",
      "Building a new DB, current time: 07/06/2024 18:52:06\n",
      "New DB name:   /home/kanai/documents/analysis/myTELR/syri/ipynb/tmp/essential_gene_blast_identification/blastdb/L10-1.3\n",
      "New DB title:  exp/multiple_runs11/tmp/L10-1/L10-1_genome.3.fasta\n",
      "Sequence type: Nucleotide\n",
      "Deleted existing Nucleotide BLAST database named /home/kanai/documents/analysis/myTELR/syri/ipynb/tmp/essential_gene_blast_identification/blastdb/L10-1.3\n",
      "Keep MBits: T\n",
      "Maximum file size: 3000000000B\n",
      "Adding sequences from FASTA; added 1 sequences in 0.0210571 seconds.\n",
      "\n",
      "\n",
      "\n",
      "\n",
      "Building a new DB, current time: 07/06/2024 18:52:06\n",
      "New DB name:   /home/kanai/documents/analysis/myTELR/syri/ipynb/tmp/essential_gene_blast_identification/blastdb/L10-2.1\n",
      "New DB title:  exp/multiple_runs11/tmp/L10-2/L10-2_genome.1.fasta\n",
      "Sequence type: Nucleotide\n",
      "Deleted existing Nucleotide BLAST database named /home/kanai/documents/analysis/myTELR/syri/ipynb/tmp/essential_gene_blast_identification/blastdb/L10-2.1\n",
      "Keep MBits: T\n",
      "Maximum file size: 3000000000B\n",
      "Adding sequences from FASTA; added 1 sequences in 0.0218711 seconds.\n",
      "\n",
      "\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\n",
      "\n",
      "Building a new DB, current time: 07/06/2024 18:52:06\n",
      "New DB name:   /home/kanai/documents/analysis/myTELR/syri/ipynb/tmp/essential_gene_blast_identification/blastdb/L10-2.2\n",
      "New DB title:  exp/multiple_runs11/tmp/L10-2/L10-2_genome.2.fasta\n",
      "Sequence type: Nucleotide\n",
      "Deleted existing Nucleotide BLAST database named /home/kanai/documents/analysis/myTELR/syri/ipynb/tmp/essential_gene_blast_identification/blastdb/L10-2.2\n",
      "Keep MBits: T\n",
      "Maximum file size: 3000000000B\n",
      "Adding sequences from FASTA; added 1 sequences in 0.0213559 seconds.\n",
      "\n",
      "\n",
      "\n",
      "\n",
      "Building a new DB, current time: 07/06/2024 18:52:06\n",
      "New DB name:   /home/kanai/documents/analysis/myTELR/syri/ipynb/tmp/essential_gene_blast_identification/blastdb/L10-2.3\n",
      "New DB title:  exp/multiple_runs11/tmp/L10-2/L10-2_genome.3.fasta\n",
      "Sequence type: Nucleotide\n",
      "Deleted existing Nucleotide BLAST database named /home/kanai/documents/analysis/myTELR/syri/ipynb/tmp/essential_gene_blast_identification/blastdb/L10-2.3\n",
      "Keep MBits: T\n",
      "Maximum file size: 3000000000B\n",
      "Adding sequences from FASTA; added 1 sequences in 0.0219679 seconds.\n",
      "\n",
      "\n",
      "\n",
      "\n",
      "Building a new DB, current time: 07/06/2024 18:52:07\n",
      "New DB name:   /home/kanai/documents/analysis/myTELR/syri/ipynb/tmp/essential_gene_blast_identification/blastdb/L10-3.1\n",
      "New DB title:  exp/multiple_runs11/tmp/L10-3/L10-3_genome.1.fasta\n",
      "Sequence type: Nucleotide\n",
      "Deleted existing Nucleotide BLAST database named /home/kanai/documents/analysis/myTELR/syri/ipynb/tmp/essential_gene_blast_identification/blastdb/L10-3.1\n",
      "Keep MBits: T\n",
      "Maximum file size: 3000000000B\n",
      "Adding sequences from FASTA; added 1 sequences in 0.021601 seconds.\n",
      "\n",
      "\n",
      "\n",
      "\n",
      "Building a new DB, current time: 07/06/2024 18:52:07\n",
      "New DB name:   /home/kanai/documents/analysis/myTELR/syri/ipynb/tmp/essential_gene_blast_identification/blastdb/L10-3.2\n",
      "New DB title:  exp/multiple_runs11/tmp/L10-3/L10-3_genome.2.fasta\n",
      "Sequence type: Nucleotide\n",
      "Deleted existing Nucleotide BLAST database named /home/kanai/documents/analysis/myTELR/syri/ipynb/tmp/essential_gene_blast_identification/blastdb/L10-3.2\n",
      "Keep MBits: T\n",
      "Maximum file size: 3000000000B\n",
      "Adding sequences from FASTA; added 1 sequences in 0.0223429 seconds.\n",
      "\n",
      "\n",
      "\n",
      "\n",
      "Building a new DB, current time: 07/06/2024 18:52:07\n",
      "New DB name:   /home/kanai/documents/analysis/myTELR/syri/ipynb/tmp/essential_gene_blast_identification/blastdb/L10-3.3\n",
      "New DB title:  exp/multiple_runs11/tmp/L10-3/L10-3_genome.3.fasta\n",
      "Sequence type: Nucleotide\n",
      "Deleted existing Nucleotide BLAST database named /home/kanai/documents/analysis/myTELR/syri/ipynb/tmp/essential_gene_blast_identification/blastdb/L10-3.3\n",
      "Keep MBits: T\n",
      "Maximum file size: 3000000000B\n",
      "Adding sequences from FASTA; added 1 sequences in 0.022336 seconds.\n",
      "\n",
      "\n",
      "\n",
      "\n",
      "Building a new DB, current time: 07/06/2024 18:52:07\n",
      "New DB name:   /home/kanai/documents/analysis/myTELR/syri/ipynb/tmp/essential_gene_blast_identification/blastdb/L10-4.1\n",
      "New DB title:  exp/multiple_runs11/tmp/L10-4/L10-4_genome.1.fasta\n",
      "Sequence type: Nucleotide\n",
      "Deleted existing Nucleotide BLAST database named /home/kanai/documents/analysis/myTELR/syri/ipynb/tmp/essential_gene_blast_identification/blastdb/L10-4.1\n",
      "Keep MBits: T\n",
      "Maximum file size: 3000000000B\n",
      "Adding sequences from FASTA; added 1 sequences in 0.022516 seconds.\n",
      "\n",
      "\n",
      "\n",
      "\n",
      "Building a new DB, current time: 07/06/2024 18:52:07\n",
      "New DB name:   /home/kanai/documents/analysis/myTELR/syri/ipynb/tmp/essential_gene_blast_identification/blastdb/L10-4.2\n",
      "New DB title:  exp/multiple_runs11/tmp/L10-4/L10-4_genome.2.fasta\n",
      "Sequence type: Nucleotide\n",
      "Deleted existing Nucleotide BLAST database named /home/kanai/documents/analysis/myTELR/syri/ipynb/tmp/essential_gene_blast_identification/blastdb/L10-4.2\n",
      "Keep MBits: T\n",
      "Maximum file size: 3000000000B\n",
      "Adding sequences from FASTA; added 1 sequences in 0.0221348 seconds.\n",
      "\n",
      "\n",
      "\n",
      "\n",
      "Building a new DB, current time: 07/06/2024 18:52:07\n",
      "New DB name:   /home/kanai/documents/analysis/myTELR/syri/ipynb/tmp/essential_gene_blast_identification/blastdb/L10-4.3\n",
      "New DB title:  exp/multiple_runs11/tmp/L10-4/L10-4_genome.3.fasta\n",
      "Sequence type: Nucleotide\n",
      "Deleted existing Nucleotide BLAST database named /home/kanai/documents/analysis/myTELR/syri/ipynb/tmp/essential_gene_blast_identification/blastdb/L10-4.3\n",
      "Keep MBits: T\n",
      "Maximum file size: 3000000000B\n",
      "Adding sequences from FASTA; added 1 sequences in 0.021667 seconds.\n",
      "\n",
      "\n",
      "\n",
      "\n",
      "Building a new DB, current time: 07/06/2024 18:52:07\n",
      "New DB name:   /home/kanai/documents/analysis/myTELR/syri/ipynb/tmp/essential_gene_blast_identification/blastdb/L11-1.1\n",
      "New DB title:  exp/multiple_runs11/tmp/L11-1/L11-1_genome.1.fasta\n",
      "Sequence type: Nucleotide\n",
      "Deleted existing Nucleotide BLAST database named /home/kanai/documents/analysis/myTELR/syri/ipynb/tmp/essential_gene_blast_identification/blastdb/L11-1.1\n",
      "Keep MBits: T\n",
      "Maximum file size: 3000000000B\n",
      "Adding sequences from FASTA; added 1 sequences in 0.0223329 seconds.\n",
      "\n",
      "\n",
      "\n",
      "\n",
      "Building a new DB, current time: 07/06/2024 18:52:07\n",
      "New DB name:   /home/kanai/documents/analysis/myTELR/syri/ipynb/tmp/essential_gene_blast_identification/blastdb/L11-1.2\n",
      "New DB title:  exp/multiple_runs11/tmp/L11-1/L11-1_genome.2.fasta\n",
      "Sequence type: Nucleotide\n",
      "Deleted existing Nucleotide BLAST database named /home/kanai/documents/analysis/myTELR/syri/ipynb/tmp/essential_gene_blast_identification/blastdb/L11-1.2\n",
      "Keep MBits: T\n",
      "Maximum file size: 3000000000B\n",
      "Adding sequences from FASTA; added 1 sequences in 0.022871 seconds.\n",
      "\n",
      "\n",
      "\n",
      "\n",
      "Building a new DB, current time: 07/06/2024 18:52:07\n",
      "New DB name:   /home/kanai/documents/analysis/myTELR/syri/ipynb/tmp/essential_gene_blast_identification/blastdb/L11-1.3\n",
      "New DB title:  exp/multiple_runs11/tmp/L11-1/L11-1_genome.3.fasta\n",
      "Sequence type: Nucleotide\n",
      "Deleted existing Nucleotide BLAST database named /home/kanai/documents/analysis/myTELR/syri/ipynb/tmp/essential_gene_blast_identification/blastdb/L11-1.3\n",
      "Keep MBits: T\n",
      "Maximum file size: 3000000000B\n",
      "Adding sequences from FASTA; added 1 sequences in 0.0213301 seconds.\n",
      "\n",
      "\n",
      "\n",
      "\n",
      "Building a new DB, current time: 07/06/2024 18:52:08\n",
      "New DB name:   /home/kanai/documents/analysis/myTELR/syri/ipynb/tmp/essential_gene_blast_identification/blastdb/L11-2.1\n",
      "New DB title:  exp/multiple_runs11/tmp/L11-2/L11-2_genome.1.fasta\n",
      "Sequence type: Nucleotide\n",
      "Deleted existing Nucleotide BLAST database named /home/kanai/documents/analysis/myTELR/syri/ipynb/tmp/essential_gene_blast_identification/blastdb/L11-2.1\n",
      "Keep MBits: T\n",
      "Maximum file size: 3000000000B\n",
      "Adding sequences from FASTA; added 1 sequences in 0.022203 seconds.\n",
      "\n",
      "\n",
      "\n",
      "\n",
      "Building a new DB, current time: 07/06/2024 18:52:08\n",
      "New DB name:   /home/kanai/documents/analysis/myTELR/syri/ipynb/tmp/essential_gene_blast_identification/blastdb/L11-2.2\n",
      "New DB title:  exp/multiple_runs11/tmp/L11-2/L11-2_genome.2.fasta\n",
      "Sequence type: Nucleotide\n",
      "Deleted existing Nucleotide BLAST database named /home/kanai/documents/analysis/myTELR/syri/ipynb/tmp/essential_gene_blast_identification/blastdb/L11-2.2\n",
      "Keep MBits: T\n",
      "Maximum file size: 3000000000B\n",
      "Adding sequences from FASTA; added 1 sequences in 0.02163 seconds.\n",
      "\n",
      "\n",
      "\n",
      "\n",
      "Building a new DB, current time: 07/06/2024 18:52:08\n",
      "New DB name:   /home/kanai/documents/analysis/myTELR/syri/ipynb/tmp/essential_gene_blast_identification/blastdb/L11-2.3\n",
      "New DB title:  exp/multiple_runs11/tmp/L11-2/L11-2_genome.3.fasta\n",
      "Sequence type: Nucleotide\n",
      "Deleted existing Nucleotide BLAST database named /home/kanai/documents/analysis/myTELR/syri/ipynb/tmp/essential_gene_blast_identification/blastdb/L11-2.3\n",
      "Keep MBits: T\n",
      "Maximum file size: 3000000000B\n",
      "Adding sequences from FASTA; added 1 sequences in 0.0233059 seconds.\n",
      "\n",
      "\n",
      "\n",
      "\n",
      "Building a new DB, current time: 07/06/2024 18:52:08\n",
      "New DB name:   /home/kanai/documents/analysis/myTELR/syri/ipynb/tmp/essential_gene_blast_identification/blastdb/L11-3.1\n",
      "New DB title:  exp/multiple_runs11/tmp/L11-3/L11-3_genome.1.fasta\n",
      "Sequence type: Nucleotide\n",
      "Deleted existing Nucleotide BLAST database named /home/kanai/documents/analysis/myTELR/syri/ipynb/tmp/essential_gene_blast_identification/blastdb/L11-3.1\n",
      "Keep MBits: T\n",
      "Maximum file size: 3000000000B\n",
      "Adding sequences from FASTA; added 1 sequences in 0.0215731 seconds.\n",
      "\n",
      "\n",
      "\n",
      "\n",
      "Building a new DB, current time: 07/06/2024 18:52:08\n",
      "New DB name:   /home/kanai/documents/analysis/myTELR/syri/ipynb/tmp/essential_gene_blast_identification/blastdb/L11-3.2\n",
      "New DB title:  exp/multiple_runs11/tmp/L11-3/L11-3_genome.2.fasta\n",
      "Sequence type: Nucleotide\n",
      "Deleted existing Nucleotide BLAST database named /home/kanai/documents/analysis/myTELR/syri/ipynb/tmp/essential_gene_blast_identification/blastdb/L11-3.2\n",
      "Keep MBits: T\n",
      "Maximum file size: 3000000000B\n",
      "Adding sequences from FASTA; added 1 sequences in 0.0224559 seconds.\n",
      "\n",
      "\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\n",
      "\n",
      "Building a new DB, current time: 07/06/2024 18:52:08\n",
      "New DB name:   /home/kanai/documents/analysis/myTELR/syri/ipynb/tmp/essential_gene_blast_identification/blastdb/L11-3.3\n",
      "New DB title:  exp/multiple_runs11/tmp/L11-3/L11-3_genome.3.fasta\n",
      "Sequence type: Nucleotide\n",
      "Deleted existing Nucleotide BLAST database named /home/kanai/documents/analysis/myTELR/syri/ipynb/tmp/essential_gene_blast_identification/blastdb/L11-3.3\n",
      "Keep MBits: T\n",
      "Maximum file size: 3000000000B\n",
      "Adding sequences from FASTA; added 1 sequences in 0.0210679 seconds.\n",
      "\n",
      "\n",
      "\n",
      "\n",
      "Building a new DB, current time: 07/06/2024 18:52:08\n",
      "New DB name:   /home/kanai/documents/analysis/myTELR/syri/ipynb/tmp/essential_gene_blast_identification/blastdb/L11-4.1\n",
      "New DB title:  exp/multiple_runs11/tmp/L11-4/L11-4_genome.1.fasta\n",
      "Sequence type: Nucleotide\n",
      "Deleted existing Nucleotide BLAST database named /home/kanai/documents/analysis/myTELR/syri/ipynb/tmp/essential_gene_blast_identification/blastdb/L11-4.1\n",
      "Keep MBits: T\n",
      "Maximum file size: 3000000000B\n",
      "Adding sequences from FASTA; added 1 sequences in 0.022541 seconds.\n",
      "\n",
      "\n",
      "\n",
      "\n",
      "Building a new DB, current time: 07/06/2024 18:52:08\n",
      "New DB name:   /home/kanai/documents/analysis/myTELR/syri/ipynb/tmp/essential_gene_blast_identification/blastdb/L11-4.2\n",
      "New DB title:  exp/multiple_runs11/tmp/L11-4/L11-4_genome.2.fasta\n",
      "Sequence type: Nucleotide\n",
      "Deleted existing Nucleotide BLAST database named /home/kanai/documents/analysis/myTELR/syri/ipynb/tmp/essential_gene_blast_identification/blastdb/L11-4.2\n",
      "Keep MBits: T\n",
      "Maximum file size: 3000000000B\n",
      "Adding sequences from FASTA; added 1 sequences in 0.022516 seconds.\n",
      "\n",
      "\n",
      "\n",
      "\n",
      "Building a new DB, current time: 07/06/2024 18:52:08\n",
      "New DB name:   /home/kanai/documents/analysis/myTELR/syri/ipynb/tmp/essential_gene_blast_identification/blastdb/L11-4.3\n",
      "New DB title:  exp/multiple_runs11/tmp/L11-4/L11-4_genome.3.fasta\n",
      "Sequence type: Nucleotide\n",
      "Deleted existing Nucleotide BLAST database named /home/kanai/documents/analysis/myTELR/syri/ipynb/tmp/essential_gene_blast_identification/blastdb/L11-4.3\n",
      "Keep MBits: T\n",
      "Maximum file size: 3000000000B\n",
      "Adding sequences from FASTA; added 1 sequences in 0.023181 seconds.\n",
      "\n",
      "\n"
     ]
    }
   ],
   "source": [
    "#L01-1, L01-2, ... L01-4, L02-1, ... L11-4\n",
    "lines = ['L01', 'L02', 'L03', 'L04', 'L05', 'L06', 'L07', 'L08', 'L09', 'L10', 'L11']\n",
    "#lines = ['L01', 'L02']\n",
    "lines = [f'{line}-{i}' for line in lines for i in range(1, 5)]\n",
    "gen_id = [1, 2, 3]\n",
    "\n",
    "tmp_dir = os.path.join(\"tmp\", \"essential_gene_blast_identification\")\n",
    "query_fasta = os.path.join(tmp_dir, \"mds42_ess.fasta\")\n",
    "create_query_fasta(essential_genes, mds42_fasta, query_fasta)\n",
    "\n",
    "# lines for every gen-id\n",
    "line_df = pd.DataFrame({'line': np.repeat(lines, len(gen_id)), 'gen_id': np.tile(gen_id, len(lines))})\n",
    "merged_df, sumamrised_df = identify_ess_genes(line_df, query_fasta, essential_genes, base_output_dir, tmp_dir)"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "## Result (no missing)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 10,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>line</th>\n",
       "      <th>gen_id</th>\n",
       "      <th>duplicated_genes</th>\n",
       "      <th>missing_genes</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>L01-2</td>\n",
       "      <td>2</td>\n",
       "      <td>2</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>5</th>\n",
       "      <td>L01-2</td>\n",
       "      <td>3</td>\n",
       "      <td>2</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>14</th>\n",
       "      <td>L02-1</td>\n",
       "      <td>3</td>\n",
       "      <td>13</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>16</th>\n",
       "      <td>L02-2</td>\n",
       "      <td>2</td>\n",
       "      <td>2</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>17</th>\n",
       "      <td>L02-2</td>\n",
       "      <td>3</td>\n",
       "      <td>14</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>20</th>\n",
       "      <td>L02-3</td>\n",
       "      <td>3</td>\n",
       "      <td>9</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>23</th>\n",
       "      <td>L02-4</td>\n",
       "      <td>3</td>\n",
       "      <td>3</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>29</th>\n",
       "      <td>L03-2</td>\n",
       "      <td>3</td>\n",
       "      <td>7</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>35</th>\n",
       "      <td>L03-4</td>\n",
       "      <td>3</td>\n",
       "      <td>3</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>37</th>\n",
       "      <td>L04-1</td>\n",
       "      <td>2</td>\n",
       "      <td>4</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>43</th>\n",
       "      <td>L04-3</td>\n",
       "      <td>2</td>\n",
       "      <td>10</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>44</th>\n",
       "      <td>L04-3</td>\n",
       "      <td>3</td>\n",
       "      <td>6</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>46</th>\n",
       "      <td>L04-4</td>\n",
       "      <td>2</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>47</th>\n",
       "      <td>L04-4</td>\n",
       "      <td>3</td>\n",
       "      <td>4</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>58</th>\n",
       "      <td>L05-4</td>\n",
       "      <td>2</td>\n",
       "      <td>2</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>68</th>\n",
       "      <td>L06-3</td>\n",
       "      <td>3</td>\n",
       "      <td>6</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>112</th>\n",
       "      <td>L10-2</td>\n",
       "      <td>2</td>\n",
       "      <td>2</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>116</th>\n",
       "      <td>L10-3</td>\n",
       "      <td>3</td>\n",
       "      <td>33</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>125</th>\n",
       "      <td>L11-2</td>\n",
       "      <td>3</td>\n",
       "      <td>5</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>131</th>\n",
       "      <td>L11-4</td>\n",
       "      <td>3</td>\n",
       "      <td>19</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "      line  gen_id  duplicated_genes  missing_genes\n",
       "4    L01-2       2                 2              0\n",
       "5    L01-2       3                 2              0\n",
       "14   L02-1       3                13              0\n",
       "16   L02-2       2                 2              0\n",
       "17   L02-2       3                14              0\n",
       "20   L02-3       3                 9              0\n",
       "23   L02-4       3                 3              0\n",
       "29   L03-2       3                 7              0\n",
       "35   L03-4       3                 3              0\n",
       "37   L04-1       2                 4              0\n",
       "43   L04-3       2                10              0\n",
       "44   L04-3       3                 6              0\n",
       "46   L04-4       2                 1              0\n",
       "47   L04-4       3                 4              0\n",
       "58   L05-4       2                 2              0\n",
       "68   L06-3       3                 6              0\n",
       "112  L10-2       2                 2              0\n",
       "116  L10-3       3                33              0\n",
       "125  L11-2       3                 5              0\n",
       "131  L11-4       3                19              0"
      ]
     },
     "execution_count": 10,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "# all essential genes are found in the list\n",
    "sumamrised_df.query(\"duplicated_genes > 0 | missing_genes > 0\")"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "## Test run (L04-3)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 11,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\n",
      "\n",
      "Building a new DB, current time: 07/06/2024 18:52:09\n",
      "New DB name:   /home/kanai/documents/analysis/myTELR/syri/ipynb/tmp/essential_gene_blast_identification/blastdb/L04-3.2\n",
      "New DB title:  exp/multiple_runs11/tmp/L04-3/L04-3_genome.2.fasta\n",
      "Sequence type: Nucleotide\n",
      "Deleted existing Nucleotide BLAST database named /home/kanai/documents/analysis/myTELR/syri/ipynb/tmp/essential_gene_blast_identification/blastdb/L04-3.2\n",
      "Keep MBits: T\n",
      "Maximum file size: 3000000000B\n",
      "Adding sequences from FASTA; added 1 sequences in 0.022994 seconds.\n",
      "\n",
      "\n"
     ]
    },
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>qseqid</th>\n",
       "      <th>sseqid</th>\n",
       "      <th>pident</th>\n",
       "      <th>length</th>\n",
       "      <th>mismatch</th>\n",
       "      <th>gapopen</th>\n",
       "      <th>qstart</th>\n",
       "      <th>qend</th>\n",
       "      <th>sstart</th>\n",
       "      <th>send</th>\n",
       "      <th>evalue</th>\n",
       "      <th>bitscore</th>\n",
       "      <th>qlen</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>ribF</td>\n",
       "      <td>edge_3-,edge_6+,edge_2-,edge_4+,edge_5-,edge_4...</td>\n",
       "      <td>100.0</td>\n",
       "      <td>941</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>941</td>\n",
       "      <td>575672</td>\n",
       "      <td>576612</td>\n",
       "      <td>0.0</td>\n",
       "      <td>1738</td>\n",
       "      <td>941</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>ileS</td>\n",
       "      <td>edge_3-,edge_6+,edge_2-,edge_4+,edge_5-,edge_4...</td>\n",
       "      <td>100.0</td>\n",
       "      <td>2816</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>2816</td>\n",
       "      <td>576656</td>\n",
       "      <td>579471</td>\n",
       "      <td>0.0</td>\n",
       "      <td>5201</td>\n",
       "      <td>2816</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>lspA</td>\n",
       "      <td>edge_3-,edge_6+,edge_2-,edge_4+,edge_5-,edge_4...</td>\n",
       "      <td>100.0</td>\n",
       "      <td>494</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>494</td>\n",
       "      <td>579472</td>\n",
       "      <td>579965</td>\n",
       "      <td>0.0</td>\n",
       "      <td>913</td>\n",
       "      <td>494</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>ispH</td>\n",
       "      <td>edge_3-,edge_6+,edge_2-,edge_4+,edge_5-,edge_4...</td>\n",
       "      <td>100.0</td>\n",
       "      <td>950</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>950</td>\n",
       "      <td>580542</td>\n",
       "      <td>581491</td>\n",
       "      <td>0.0</td>\n",
       "      <td>1755</td>\n",
       "      <td>950</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>dapB</td>\n",
       "      <td>edge_3-,edge_6+,edge_2-,edge_4+,edge_5-,edge_4...</td>\n",
       "      <td>100.0</td>\n",
       "      <td>821</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>821</td>\n",
       "      <td>582639</td>\n",
       "      <td>583459</td>\n",
       "      <td>0.0</td>\n",
       "      <td>1517</td>\n",
       "      <td>821</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "  qseqid                                             sseqid  pident  length  \\\n",
       "0   ribF  edge_3-,edge_6+,edge_2-,edge_4+,edge_5-,edge_4...   100.0     941   \n",
       "1   ileS  edge_3-,edge_6+,edge_2-,edge_4+,edge_5-,edge_4...   100.0    2816   \n",
       "2   lspA  edge_3-,edge_6+,edge_2-,edge_4+,edge_5-,edge_4...   100.0     494   \n",
       "3   ispH  edge_3-,edge_6+,edge_2-,edge_4+,edge_5-,edge_4...   100.0     950   \n",
       "4   dapB  edge_3-,edge_6+,edge_2-,edge_4+,edge_5-,edge_4...   100.0     821   \n",
       "\n",
       "   mismatch  gapopen  qstart  qend  sstart    send  evalue  bitscore  qlen  \n",
       "0         0        0       1   941  575672  576612     0.0      1738   941  \n",
       "1         0        0       1  2816  576656  579471     0.0      5201  2816  \n",
       "2         0        0       1   494  579472  579965     0.0       913   494  \n",
       "3         0        0       1   950  580542  581491     0.0      1755   950  \n",
       "4         0        0       1   821  582639  583459     0.0      1517   821  "
      ]
     },
     "execution_count": 11,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "# L04-3 should have duplicated essential genes. I use this to test the code\n",
    "line_id = 'L04-3'\n",
    "gen_id = 2\n",
    "res_csv_ = run_ess_blast(f'{line_id}.{gen_id}',\\\n",
    "\tos.path.join(base_output_dir, \"tmp\", line_id, f'{line_id}_genome.{gen_id}.fasta'),\\\n",
    "\tquery_fasta, tmp_dir)\n",
    "\n",
    "res_ess_ = pd.read_csv(res_csv_, sep='\\t')\n",
    "res_ess_.head()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 12,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Number of unique hits: 292\n",
      "Number of missed essential genes: 0\n",
      "Number of duplicated hits: 10\n"
     ]
    }
   ],
   "source": [
    "res_ess_['is_duplicated'] = res_ess_.duplicated(subset=['qseqid'], keep=False)\n",
    "unique_hits = res_ess_[~res_ess_['is_duplicated']]\n",
    "\n",
    "duplicated_genes = res_ess_[res_ess_['is_duplicated']]['qseqid'].unique()\n",
    "missing_genes = essential_genes[~essential_genes['gene'].isin(res_ess_['qseqid'])]\n",
    "\n",
    "print(f\"Number of unique hits: {len(unique_hits)}\")\n",
    "print(f\"Number of missed essential genes: {len(missing_genes)}\")\n",
    "print(f\"Number of duplicated hits: {len(duplicated_genes)}\")"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "# Calculate average distance from the closest essential gene\n",
    "\n",
    "For a region to be deleted by NE type IS deletion(recombination between Newly inserted IS with a preExisting one), which was the majority in Lee 2016,\n",
    "the region must be non-essential.\n",
    "\n",
    "Here, I identify if the local hopping is due to the distribution of essential genes\n",
    "or due to an innate property of the IS."
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 13,
   "metadata": {},
   "outputs": [],
   "source": [
    "# by chatgpt\n",
    "import pandas as pd\n",
    "\n",
    "# Function to find the closest essential gene and IS end\n",
    "def find_closest_gene(is_row, genes_df):\n",
    "    # Initialize minimum distance and details for the closest approach\n",
    "    min_distance = float('inf')\n",
    "    closest_gene = None\n",
    "    gene_end = None\n",
    "    is_end_involved = None\n",
    "    \n",
    "    for _, gene_row in genes_df.iterrows():\n",
    "        # Distances from IS start and end to both ends of the gene\n",
    "        distances = {\n",
    "            ('5_prime', 'start'): abs(is_row['start'] - gene_row['sstart']),\n",
    "            ('5_prime', 'end'): abs(is_row['end'] - gene_row['sstart']),\n",
    "            ('3_prime', 'start'): abs(is_row['start'] - gene_row['send']),\n",
    "            ('3_prime', 'end'): abs(is_row['end'] - gene_row['send'])\n",
    "        }\n",
    "        \n",
    "        # Identify the minimum distance and corresponding details\n",
    "        for (gene_side, is_side), distance in distances.items():\n",
    "            if distance < min_distance:\n",
    "                min_distance = distance\n",
    "                closest_gene = gene_row['gene']\n",
    "                gene_end = gene_side\n",
    "                is_end_involved = is_side\n",
    "                position = gene_row['sstart'] if gene_side == '5_prime' else gene_row['send']\n",
    "                \n",
    "    # Return details including which IS end (start or end) is closest to the gene\n",
    "    return pd.Series([closest_gene, gene_end, is_end_involved, min_distance, position],\\\n",
    "\t\t\t index=['closest_gene', 'gene_end', 'IS_end', 'distance', 'position'])"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 14,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>start</th>\n",
       "      <th>end</th>\n",
       "      <th>IS_strand</th>\n",
       "      <th>max_alignment_length</th>\n",
       "      <th>length</th>\n",
       "      <th>cluster_id</th>\n",
       "      <th>line</th>\n",
       "      <th>gen_id</th>\n",
       "      <th>closest_gene</th>\n",
       "      <th>gene_end</th>\n",
       "      <th>IS_end</th>\n",
       "      <th>distance</th>\n",
       "      <th>position</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>831</th>\n",
       "      <td>42391</td>\n",
       "      <td>45482</td>\n",
       "      <td>reverse</td>\n",
       "      <td>3093</td>\n",
       "      <td>3092</td>\n",
       "      <td>0</td>\n",
       "      <td>L04-3</td>\n",
       "      <td>2</td>\n",
       "      <td>rho</td>\n",
       "      <td>3_prime</td>\n",
       "      <td>start</td>\n",
       "      <td>466</td>\n",
       "      <td>41925</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>832</th>\n",
       "      <td>121558</td>\n",
       "      <td>124650</td>\n",
       "      <td>forward</td>\n",
       "      <td>3093</td>\n",
       "      <td>3093</td>\n",
       "      <td>1</td>\n",
       "      <td>L04-3</td>\n",
       "      <td>2</td>\n",
       "      <td>polA</td>\n",
       "      <td>5_prime</td>\n",
       "      <td>end</td>\n",
       "      <td>2763</td>\n",
       "      <td>127413</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>833</th>\n",
       "      <td>355703</td>\n",
       "      <td>358793</td>\n",
       "      <td>reverse</td>\n",
       "      <td>3092</td>\n",
       "      <td>3091</td>\n",
       "      <td>2</td>\n",
       "      <td>L04-3</td>\n",
       "      <td>2</td>\n",
       "      <td>ssb</td>\n",
       "      <td>3_prime</td>\n",
       "      <td>start</td>\n",
       "      <td>608</td>\n",
       "      <td>355095</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>834</th>\n",
       "      <td>444530</td>\n",
       "      <td>447621</td>\n",
       "      <td>reverse</td>\n",
       "      <td>3092</td>\n",
       "      <td>3092</td>\n",
       "      <td>3</td>\n",
       "      <td>L04-3</td>\n",
       "      <td>2</td>\n",
       "      <td>ppa</td>\n",
       "      <td>5_prime</td>\n",
       "      <td>end</td>\n",
       "      <td>10760</td>\n",
       "      <td>458381</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>835</th>\n",
       "      <td>552106</td>\n",
       "      <td>555198</td>\n",
       "      <td>reverse</td>\n",
       "      <td>3093</td>\n",
       "      <td>3093</td>\n",
       "      <td>4</td>\n",
       "      <td>L04-3</td>\n",
       "      <td>2</td>\n",
       "      <td>ribF</td>\n",
       "      <td>5_prime</td>\n",
       "      <td>end</td>\n",
       "      <td>20474</td>\n",
       "      <td>575672</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "      start     end IS_strand  max_alignment_length  length  cluster_id  \\\n",
       "831   42391   45482   reverse                  3093    3092           0   \n",
       "832  121558  124650   forward                  3093    3093           1   \n",
       "833  355703  358793   reverse                  3092    3091           2   \n",
       "834  444530  447621   reverse                  3092    3092           3   \n",
       "835  552106  555198   reverse                  3093    3093           4   \n",
       "\n",
       "      line  gen_id closest_gene gene_end IS_end  distance  position  \n",
       "831  L04-3       2          rho  3_prime  start       466     41925  \n",
       "832  L04-3       2         polA  5_prime    end      2763    127413  \n",
       "833  L04-3       2          ssb  3_prime  start       608    355095  \n",
       "834  L04-3       2          ppa  5_prime    end     10760    458381  \n",
       "835  L04-3       2         ribF  5_prime    end     20474    575672  "
      ]
     },
     "execution_count": 14,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "# test\n",
    "is_df = is_positions_df.query(\"line == 'L04-3' and gen_id == 2\")\n",
    "essential_genes_df = merged_df.query(\"line == 'L04-3' and gen_id == 2\")\n",
    "\n",
    "# Apply the function to each IS row\n",
    "#is_df[['closest_gene', 'gene_end', 'is_end','distance']] = is_df.apply(find_closest_gene, genes_df=essential_genes_df, axis=1)\n",
    "# use loc\n",
    "is_df.loc[:, ['closest_gene', 'gene_end', 'IS_end','distance', 'position']] = \\\n",
    "\tis_df.apply(find_closest_gene, genes_df=essential_genes_df, axis=1)\n",
    "is_df.head()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 15,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>gene</th>\n",
       "      <th>pident</th>\n",
       "      <th>length</th>\n",
       "      <th>mismatch</th>\n",
       "      <th>gapopen</th>\n",
       "      <th>qstart</th>\n",
       "      <th>qend</th>\n",
       "      <th>sstart</th>\n",
       "      <th>send</th>\n",
       "      <th>evalue</th>\n",
       "      <th>bitscore</th>\n",
       "      <th>qlen</th>\n",
       "      <th>line</th>\n",
       "      <th>gen_id</th>\n",
       "      <th>duplicated</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>13048</th>\n",
       "      <td>mukB</td>\n",
       "      <td>100.0</td>\n",
       "      <td>4460</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>4460</td>\n",
       "      <td>1418203</td>\n",
       "      <td>1422662</td>\n",
       "      <td>0.000000e+00</td>\n",
       "      <td>8237</td>\n",
       "      <td>4460</td>\n",
       "      <td>L04-3</td>\n",
       "      <td>2</td>\n",
       "      <td>False</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>13049</th>\n",
       "      <td>rpoC</td>\n",
       "      <td>100.0</td>\n",
       "      <td>4223</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>4223</td>\n",
       "      <td>265789</td>\n",
       "      <td>270011</td>\n",
       "      <td>0.000000e+00</td>\n",
       "      <td>7799</td>\n",
       "      <td>4223</td>\n",
       "      <td>L04-3</td>\n",
       "      <td>2</td>\n",
       "      <td>False</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>13050</th>\n",
       "      <td>rpoB</td>\n",
       "      <td>100.0</td>\n",
       "      <td>4028</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>4028</td>\n",
       "      <td>261684</td>\n",
       "      <td>265711</td>\n",
       "      <td>0.000000e+00</td>\n",
       "      <td>7439</td>\n",
       "      <td>4028</td>\n",
       "      <td>L04-3</td>\n",
       "      <td>2</td>\n",
       "      <td>False</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>13051</th>\n",
       "      <td>ftsK</td>\n",
       "      <td>100.0</td>\n",
       "      <td>3989</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>3989</td>\n",
       "      <td>1929228</td>\n",
       "      <td>1925240</td>\n",
       "      <td>0.000000e+00</td>\n",
       "      <td>7367</td>\n",
       "      <td>3989</td>\n",
       "      <td>L04-3</td>\n",
       "      <td>2</td>\n",
       "      <td>True</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>13052</th>\n",
       "      <td>ftsK</td>\n",
       "      <td>100.0</td>\n",
       "      <td>3989</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>3989</td>\n",
       "      <td>1372000</td>\n",
       "      <td>1375988</td>\n",
       "      <td>0.000000e+00</td>\n",
       "      <td>7367</td>\n",
       "      <td>3989</td>\n",
       "      <td>L04-3</td>\n",
       "      <td>2</td>\n",
       "      <td>True</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>13355</th>\n",
       "      <td>hisR</td>\n",
       "      <td>100.0</td>\n",
       "      <td>76</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>76</td>\n",
       "      <td>59860</td>\n",
       "      <td>59935</td>\n",
       "      <td>6.390000e-35</td>\n",
       "      <td>141</td>\n",
       "      <td>76</td>\n",
       "      <td>L04-3</td>\n",
       "      <td>2</td>\n",
       "      <td>False</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>13356</th>\n",
       "      <td>trpT</td>\n",
       "      <td>100.0</td>\n",
       "      <td>75</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>75</td>\n",
       "      <td>21211</td>\n",
       "      <td>21285</td>\n",
       "      <td>2.260000e-34</td>\n",
       "      <td>139</td>\n",
       "      <td>75</td>\n",
       "      <td>L04-3</td>\n",
       "      <td>2</td>\n",
       "      <td>False</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>13357</th>\n",
       "      <td>thrU</td>\n",
       "      <td>100.0</td>\n",
       "      <td>75</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>75</td>\n",
       "      <td>255827</td>\n",
       "      <td>255901</td>\n",
       "      <td>2.260000e-34</td>\n",
       "      <td>139</td>\n",
       "      <td>75</td>\n",
       "      <td>L04-3</td>\n",
       "      <td>2</td>\n",
       "      <td>False</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>13358</th>\n",
       "      <td>glyT</td>\n",
       "      <td>100.0</td>\n",
       "      <td>74</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>74</td>\n",
       "      <td>256112</td>\n",
       "      <td>256185</td>\n",
       "      <td>7.970000e-34</td>\n",
       "      <td>137</td>\n",
       "      <td>74</td>\n",
       "      <td>L04-3</td>\n",
       "      <td>2</td>\n",
       "      <td>False</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>13359</th>\n",
       "      <td>cysT</td>\n",
       "      <td>100.0</td>\n",
       "      <td>73</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>73</td>\n",
       "      <td>2593132</td>\n",
       "      <td>2593204</td>\n",
       "      <td>2.810000e-33</td>\n",
       "      <td>135</td>\n",
       "      <td>73</td>\n",
       "      <td>L04-3</td>\n",
       "      <td>2</td>\n",
       "      <td>False</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>312 rows  15 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "       gene  pident  length  mismatch  gapopen  qstart  qend   sstart  \\\n",
       "13048  mukB   100.0    4460         0        0       1  4460  1418203   \n",
       "13049  rpoC   100.0    4223         0        0       1  4223   265789   \n",
       "13050  rpoB   100.0    4028         0        0       1  4028   261684   \n",
       "13051  ftsK   100.0    3989         0        0       1  3989  1929228   \n",
       "13052  ftsK   100.0    3989         0        0       1  3989  1372000   \n",
       "...     ...     ...     ...       ...      ...     ...   ...      ...   \n",
       "13355  hisR   100.0      76         0        0       1    76    59860   \n",
       "13356  trpT   100.0      75         0        0       1    75    21211   \n",
       "13357  thrU   100.0      75         0        0       1    75   255827   \n",
       "13358  glyT   100.0      74         0        0       1    74   256112   \n",
       "13359  cysT   100.0      73         0        0       1    73  2593132   \n",
       "\n",
       "          send        evalue  bitscore  qlen   line  gen_id  duplicated  \n",
       "13048  1422662  0.000000e+00      8237  4460  L04-3       2       False  \n",
       "13049   270011  0.000000e+00      7799  4223  L04-3       2       False  \n",
       "13050   265711  0.000000e+00      7439  4028  L04-3       2       False  \n",
       "13051  1925240  0.000000e+00      7367  3989  L04-3       2        True  \n",
       "13052  1375988  0.000000e+00      7367  3989  L04-3       2        True  \n",
       "...        ...           ...       ...   ...    ...     ...         ...  \n",
       "13355    59935  6.390000e-35       141    76  L04-3       2       False  \n",
       "13356    21285  2.260000e-34       139    75  L04-3       2       False  \n",
       "13357   255901  2.260000e-34       139    75  L04-3       2       False  \n",
       "13358   256185  7.970000e-34       137    74  L04-3       2       False  \n",
       "13359  2593204  2.810000e-33       135    73  L04-3       2       False  \n",
       "\n",
       "[312 rows x 15 columns]"
      ]
     },
     "execution_count": 15,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "essential_genes_df"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 16,
   "metadata": {},
   "outputs": [],
   "source": [
    "import pandas as pd\n",
    "\n",
    "# Example dataframes\n",
    "# Assuming these are already defined: is_df and essential_genes_df\n",
    "\n",
    "# Additional input\n",
    "genome_size = 5000000  # Example genome size, adjust as necessary\n",
    "\n",
    "def calculate_circular_distance(start, end, genome_size):\n",
    "    \"\"\"\n",
    "    Calculate the shortest distance between two points in a circular genome.\n",
    "    \"\"\"\n",
    "    direct_distance = abs(end - start)\n",
    "    return min(direct_distance, genome_size - direct_distance)\n",
    "\n",
    "def find_closest_genes_around_is(is_row, genes_df, genome_size):\n",
    "    \"\"\"\n",
    "    Find the closest essential genes to both the 5' and 3' ends of an IS, considering circular genome.\n",
    "    \"\"\"\n",
    "    # Initialize variables to track the closest genes and distances\n",
    "    closest_gene_5_prime = None\n",
    "    closest_gene_3_prime = None\n",
    "    min_distance_5_prime = float('inf')\n",
    "    min_distance_3_prime = float('inf')\n",
    "    position_5_prime = None\n",
    "    position_3_prime = None\n",
    "    \n",
    "    # Loop through each essential gene to find the closest to 5' and 3' ends of IS\n",
    "    for _, gene_row in genes_df.iterrows():\n",
    "        # For 5' end, consider genes at lower indexes (considering circularity)\n",
    "        if gene_row['send'] < is_row['start']:\n",
    "            distance_5_prime = calculate_circular_distance(gene_row['send'], is_row['start'], genome_size)\n",
    "            if distance_5_prime < min_distance_5_prime:\n",
    "                min_distance_5_prime = distance_5_prime\n",
    "                closest_gene_5_prime = gene_row['gene']\n",
    "                position_5_prime = gene_row['send']\n",
    "        # For 3' end, consider genes at higher indexes (considering circularity)\n",
    "        elif gene_row['sstart'] > is_row['end']:\n",
    "            distance_3_prime = calculate_circular_distance(is_row['end'], gene_row['sstart'], genome_size)\n",
    "            if distance_3_prime < min_distance_3_prime:\n",
    "                min_distance_3_prime = distance_3_prime\n",
    "                closest_gene_3_prime = gene_row['gene']\n",
    "                position_3_prime = gene_row['sstart']\n",
    "    \n",
    "    return pd.Series([closest_gene_5_prime, position_5_prime, min_distance_5_prime, closest_gene_3_prime, position_3_prime, min_distance_3_prime],\\\n",
    "                     index=['closest_gene_5_prime', 'position_5_prime', 'distance_5_prime', 'closest_gene_3_prime', 'position_3_prime', 'distance_3_prime'])"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 17,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>start</th>\n",
       "      <th>end</th>\n",
       "      <th>IS_strand</th>\n",
       "      <th>max_alignment_length</th>\n",
       "      <th>length</th>\n",
       "      <th>cluster_id</th>\n",
       "      <th>line</th>\n",
       "      <th>gen_id</th>\n",
       "      <th>closest_gene_5_prime</th>\n",
       "      <th>position_5_prime</th>\n",
       "      <th>distance_5_prime</th>\n",
       "      <th>closest_gene_3_prime</th>\n",
       "      <th>position_3_prime</th>\n",
       "      <th>distance_3_prime</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>831</th>\n",
       "      <td>42391</td>\n",
       "      <td>45482</td>\n",
       "      <td>reverse</td>\n",
       "      <td>3093</td>\n",
       "      <td>3092</td>\n",
       "      <td>0</td>\n",
       "      <td>L04-3</td>\n",
       "      <td>2</td>\n",
       "      <td>rho</td>\n",
       "      <td>41925</td>\n",
       "      <td>466</td>\n",
       "      <td>argX</td>\n",
       "      <td>59726</td>\n",
       "      <td>14244</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>832</th>\n",
       "      <td>121558</td>\n",
       "      <td>124650</td>\n",
       "      <td>forward</td>\n",
       "      <td>3093</td>\n",
       "      <td>3093</td>\n",
       "      <td>1</td>\n",
       "      <td>L04-3</td>\n",
       "      <td>2</td>\n",
       "      <td>hemG</td>\n",
       "      <td>112499</td>\n",
       "      <td>9059</td>\n",
       "      <td>polA</td>\n",
       "      <td>127413</td>\n",
       "      <td>2763</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>833</th>\n",
       "      <td>355703</td>\n",
       "      <td>358793</td>\n",
       "      <td>reverse</td>\n",
       "      <td>3092</td>\n",
       "      <td>3091</td>\n",
       "      <td>2</td>\n",
       "      <td>L04-3</td>\n",
       "      <td>2</td>\n",
       "      <td>ssb</td>\n",
       "      <td>355095</td>\n",
       "      <td>608</td>\n",
       "      <td>groS</td>\n",
       "      <td>377254</td>\n",
       "      <td>18461</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>834</th>\n",
       "      <td>444530</td>\n",
       "      <td>447621</td>\n",
       "      <td>reverse</td>\n",
       "      <td>3092</td>\n",
       "      <td>3092</td>\n",
       "      <td>3</td>\n",
       "      <td>L04-3</td>\n",
       "      <td>2</td>\n",
       "      <td>rpsR</td>\n",
       "      <td>432628</td>\n",
       "      <td>11902</td>\n",
       "      <td>ppa</td>\n",
       "      <td>458381</td>\n",
       "      <td>10760</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>835</th>\n",
       "      <td>552106</td>\n",
       "      <td>555198</td>\n",
       "      <td>reverse</td>\n",
       "      <td>3093</td>\n",
       "      <td>3093</td>\n",
       "      <td>4</td>\n",
       "      <td>L04-3</td>\n",
       "      <td>2</td>\n",
       "      <td>dnaT</td>\n",
       "      <td>516211</td>\n",
       "      <td>35895</td>\n",
       "      <td>ribF</td>\n",
       "      <td>575672</td>\n",
       "      <td>20474</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "      start     end IS_strand  max_alignment_length  length  cluster_id  \\\n",
       "831   42391   45482   reverse                  3093    3092           0   \n",
       "832  121558  124650   forward                  3093    3093           1   \n",
       "833  355703  358793   reverse                  3092    3091           2   \n",
       "834  444530  447621   reverse                  3092    3092           3   \n",
       "835  552106  555198   reverse                  3093    3093           4   \n",
       "\n",
       "      line  gen_id closest_gene_5_prime  position_5_prime  distance_5_prime  \\\n",
       "831  L04-3       2                  rho             41925               466   \n",
       "832  L04-3       2                 hemG            112499              9059   \n",
       "833  L04-3       2                  ssb            355095               608   \n",
       "834  L04-3       2                 rpsR            432628             11902   \n",
       "835  L04-3       2                 dnaT            516211             35895   \n",
       "\n",
       "    closest_gene_3_prime  position_3_prime  distance_3_prime  \n",
       "831                 argX             59726             14244  \n",
       "832                 polA            127413              2763  \n",
       "833                 groS            377254             18461  \n",
       "834                  ppa            458381             10760  \n",
       "835                 ribF            575672             20474  "
      ]
     },
     "execution_count": 17,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "# test I manually checked the first two is right.\n",
    "is_df = is_positions_df.query(\"line == 'L04-3' and gen_id == 2\")\n",
    "essential_genes_df = merged_df.query(\"line == 'L04-3' and gen_id == 2 and not duplicated\")[['gene', 'sstart', 'send']]\n",
    "essential_genes_df = pd.concat([essential_genes_df,\\\n",
    "\t\tpd.DataFrame({'gene': ['oriC', 'oriC'], 'sstart': [1, genome_size], 'send': [1, genome_size]})], ignore_index=True)\n",
    "\n",
    "#is_df[['closest_gene_5_prime', 'position_5_prime', 'distance_5_prime', 'closest_gene_3_prime', 'position_3_prime', 'distance_3_prime']] = is_df.apply(find_closest_genes_around_is, genes_df=essential_genes_df, genome_size=genome_size, axis=1)\n",
    "# use loc\n",
    "is_df.loc[:, ['closest_gene_5_prime', 'position_5_prime', 'distance_5_prime', 'closest_gene_3_prime', 'position_3_prime', 'distance_3_prime']] = \\\n",
    "\tis_df.apply(find_closest_genes_around_is, genes_df=essential_genes_df, genome_size=genome_size, axis=1)\n",
    "is_df.head()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 18,
   "metadata": {},
   "outputs": [],
   "source": [
    "# genome size should not matter much as deletions can't cover oriC\n",
    "def find_closest_genes_around_is_batch(line_df, genome_sizes, merged_essential_genes, is_positions_df):\n",
    "\toutput_df = pd.DataFrame()\n",
    "\tfor _, row in line_df.iterrows():\n",
    "\t\tline_id = row['line']\n",
    "\t\tgen_id = row['gen_id']\n",
    "\t\tgenome_size = genome_sizes.query(f\"line == '{line_id}' and gen_id == {gen_id}\")['genome_size'].values[0]\n",
    "\n",
    "\t\tessential_genes_df = merged_essential_genes.query(f\"line == '{line_id}' and gen_id == {gen_id} and not duplicated\")[['gene', 'sstart', 'send']]\n",
    "\t\tessential_genes_df = pd.concat([essential_genes_df,\\\n",
    "\t\t\tpd.DataFrame({'gene': ['oriC', 'oriC'], 'sstart': [1, genome_size], 'send': [1, genome_size]})], ignore_index=True)\n",
    "\n",
    "\t\tis_df = is_positions_df.query(f\"line == '{line_id}' and gen_id == {gen_id}\")\n",
    "\t\tis_df.loc[:, ['closest_gene_5_prime', 'position_5_prime', 'distance_5_prime', 'closest_gene_3_prime', 'position_3_prime', 'distance_3_prime']] = \\\n",
    "\t\t\tis_df.apply(find_closest_genes_around_is, genes_df=essential_genes_df, genome_size=genome_size, axis=1)\n",
    "\n",
    "\t\toutput_df = pd.concat([output_df, is_df], ignore_index=True)\n",
    "\treturn output_df"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 19,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>start</th>\n",
       "      <th>end</th>\n",
       "      <th>IS_strand</th>\n",
       "      <th>max_alignment_length</th>\n",
       "      <th>length</th>\n",
       "      <th>cluster_id</th>\n",
       "      <th>line</th>\n",
       "      <th>gen_id</th>\n",
       "      <th>closest_gene_5_prime</th>\n",
       "      <th>position_5_prime</th>\n",
       "      <th>distance_5_prime</th>\n",
       "      <th>closest_gene_3_prime</th>\n",
       "      <th>position_3_prime</th>\n",
       "      <th>distance_3_prime</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>476928</td>\n",
       "      <td>480020</td>\n",
       "      <td>reverse</td>\n",
       "      <td>3093</td>\n",
       "      <td>3093</td>\n",
       "      <td>0</td>\n",
       "      <td>L01-1</td>\n",
       "      <td>1</td>\n",
       "      <td>yjeE</td>\n",
       "      <td>470186</td>\n",
       "      <td>6742</td>\n",
       "      <td>rpsR</td>\n",
       "      <td>503080</td>\n",
       "      <td>23060</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>557167</td>\n",
       "      <td>560259</td>\n",
       "      <td>reverse</td>\n",
       "      <td>3093</td>\n",
       "      <td>3093</td>\n",
       "      <td>1</td>\n",
       "      <td>L01-1</td>\n",
       "      <td>1</td>\n",
       "      <td>ppa</td>\n",
       "      <td>526892</td>\n",
       "      <td>30275</td>\n",
       "      <td>valS</td>\n",
       "      <td>561324</td>\n",
       "      <td>1065</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>1187831</td>\n",
       "      <td>1192502</td>\n",
       "      <td>forward</td>\n",
       "      <td>3093</td>\n",
       "      <td>4672</td>\n",
       "      <td>2</td>\n",
       "      <td>L01-1</td>\n",
       "      <td>1</td>\n",
       "      <td>leuS</td>\n",
       "      <td>1175771</td>\n",
       "      <td>12060</td>\n",
       "      <td>asnS</td>\n",
       "      <td>1205042</td>\n",
       "      <td>12540</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>1405123</td>\n",
       "      <td>1408215</td>\n",
       "      <td>forward</td>\n",
       "      <td>3093</td>\n",
       "      <td>3093</td>\n",
       "      <td>3</td>\n",
       "      <td>L01-1</td>\n",
       "      <td>1</td>\n",
       "      <td>infA</td>\n",
       "      <td>1266184</td>\n",
       "      <td>138939</td>\n",
       "      <td>fldA</td>\n",
       "      <td>1476321</td>\n",
       "      <td>68106</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>1499230</td>\n",
       "      <td>1503892</td>\n",
       "      <td>reverse</td>\n",
       "      <td>3093</td>\n",
       "      <td>4663</td>\n",
       "      <td>4</td>\n",
       "      <td>L01-1</td>\n",
       "      <td>1</td>\n",
       "      <td>lnt</td>\n",
       "      <td>1496376</td>\n",
       "      <td>2854</td>\n",
       "      <td>fabA</td>\n",
       "      <td>1519050</td>\n",
       "      <td>15158</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "     start      end IS_strand  max_alignment_length  length  cluster_id  \\\n",
       "0   476928   480020   reverse                  3093    3093           0   \n",
       "1   557167   560259   reverse                  3093    3093           1   \n",
       "2  1187831  1192502   forward                  3093    4672           2   \n",
       "3  1405123  1408215   forward                  3093    3093           3   \n",
       "4  1499230  1503892   reverse                  3093    4663           4   \n",
       "\n",
       "    line  gen_id closest_gene_5_prime  position_5_prime  distance_5_prime  \\\n",
       "0  L01-1       1                 yjeE            470186              6742   \n",
       "1  L01-1       1                  ppa            526892             30275   \n",
       "2  L01-1       1                 leuS           1175771             12060   \n",
       "3  L01-1       1                 infA           1266184            138939   \n",
       "4  L01-1       1                  lnt           1496376              2854   \n",
       "\n",
       "  closest_gene_3_prime  position_3_prime  distance_3_prime  \n",
       "0                 rpsR            503080             23060  \n",
       "1                 valS            561324              1065  \n",
       "2                 asnS           1205042             12540  \n",
       "3                 fldA           1476321             68106  \n",
       "4                 fabA           1519050             15158  "
      ]
     },
     "execution_count": 19,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "is_distance_df = find_closest_genes_around_is_batch(line_df, genome_sizes, merged_df, is_positions_df)\n",
    "is_distance_df.head()"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "## Analyse results\n",
    "group by {line, gen_id} and calculate average, median merged {distance_5_prime, distance_3_prime}\n",
    "Merging the 5' and 3' prime distances into one column"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 20,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>line</th>\n",
       "      <th>gen_id</th>\n",
       "      <th>average_distance</th>\n",
       "      <th>median_distance</th>\n",
       "      <th>expected</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>L01-1</td>\n",
       "      <td>1</td>\n",
       "      <td>37956.400000</td>\n",
       "      <td>14958.0</td>\n",
       "      <td>50612.642789</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>L01-1</td>\n",
       "      <td>2</td>\n",
       "      <td>38750.375000</td>\n",
       "      <td>14957.5</td>\n",
       "      <td>49624.484261</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>L01-1</td>\n",
       "      <td>3</td>\n",
       "      <td>30962.933333</td>\n",
       "      <td>18428.5</td>\n",
       "      <td>32979.744036</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>L01-2</td>\n",
       "      <td>1</td>\n",
       "      <td>37956.400000</td>\n",
       "      <td>14958.0</td>\n",
       "      <td>50612.642789</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>L01-2</td>\n",
       "      <td>2</td>\n",
       "      <td>32031.406250</td>\n",
       "      <td>14956.5</td>\n",
       "      <td>43147.040845</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>5</th>\n",
       "      <td>L01-2</td>\n",
       "      <td>3</td>\n",
       "      <td>35587.261905</td>\n",
       "      <td>14958.0</td>\n",
       "      <td>47751.338572</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>6</th>\n",
       "      <td>L01-3</td>\n",
       "      <td>1</td>\n",
       "      <td>37956.400000</td>\n",
       "      <td>14958.0</td>\n",
       "      <td>50612.642789</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>7</th>\n",
       "      <td>L01-3</td>\n",
       "      <td>2</td>\n",
       "      <td>42077.833333</td>\n",
       "      <td>21731.5</td>\n",
       "      <td>53974.845497</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>8</th>\n",
       "      <td>L01-3</td>\n",
       "      <td>3</td>\n",
       "      <td>35247.043478</td>\n",
       "      <td>19606.5</td>\n",
       "      <td>48134.267662</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>9</th>\n",
       "      <td>L01-4</td>\n",
       "      <td>1</td>\n",
       "      <td>37956.400000</td>\n",
       "      <td>14958.0</td>\n",
       "      <td>50612.642789</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "    line  gen_id  average_distance  median_distance      expected\n",
       "0  L01-1       1      37956.400000          14958.0  50612.642789\n",
       "1  L01-1       2      38750.375000          14957.5  49624.484261\n",
       "2  L01-1       3      30962.933333          18428.5  32979.744036\n",
       "3  L01-2       1      37956.400000          14958.0  50612.642789\n",
       "4  L01-2       2      32031.406250          14956.5  43147.040845\n",
       "5  L01-2       3      35587.261905          14958.0  47751.338572\n",
       "6  L01-3       1      37956.400000          14958.0  50612.642789\n",
       "7  L01-3       2      42077.833333          21731.5  53974.845497\n",
       "8  L01-3       3      35247.043478          19606.5  48134.267662\n",
       "9  L01-4       1      37956.400000          14958.0  50612.642789"
      ]
     },
     "execution_count": 20,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "merged_distances = pd.melt(is_distance_df, id_vars=['line', 'gen_id'], value_vars=['distance_5_prime', 'distance_3_prime'], \n",
    "                           var_name='prime_end', value_name='distance')\n",
    "\n",
    "# Group by line and gen_id, then calculate average and median\n",
    "grouped = merged_distances.groupby(['line', 'gen_id'])\n",
    "avg_distance = grouped['distance'].mean().reset_index(name='average_distance')\n",
    "median_distance = grouped['distance'].median().reset_index(name='median_distance')\n",
    "\n",
    "# Calculating expected value\n",
    "expected_value = grouped.apply(lambda x: x['distance'].pow(2).sum() / x['distance'].sum() / 2).reset_index(name='expected')\n",
    "\n",
    "# Merging the results\n",
    "is_distance_stats = avg_distance.merge(median_distance, on=['line', 'gen_id']).merge(expected_value, on=['line', 'gen_id'])\n",
    "is_distance_stats.head(10)"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "## median of expected distance  (not used)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 21,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "10869.794110541989\n",
      "The median (IQR) of the expected distances between IS elements and the closest essential genes is 40831.26 (37021.08-47950.56).\n",
      "Number of genomes: 87\n"
     ]
    }
   ],
   "source": [
    "# median of expected distances\n",
    "expected_distances = is_distance_stats.query('gen_id == 1 | gen_id == 2').\\\n",
    "\tquery('not(line == \"L04-1\" and gen_id == 2)').expected\n",
    "print(np.percentile(is_distance_stats.query('gen_id == 1 | gen_id == 2').expected, 75) - np.percentile(is_distance_stats.query('gen_id == 1 | gen_id == 2').expected, 25))\n",
    "# in text\n",
    "print(f'The median (IQR) of the expected distances between IS elements and the closest essential genes is {np.median(expected_distances):.2f} ({np.percentile(expected_distances, 25):.2f}-{np.percentile(expected_distances, 75):.2f}).')\n",
    "print(f'Number of genomes: {len(expected_distances)}')"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "## median of expected median distance (not used)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 22,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>line</th>\n",
       "      <th>gen_id</th>\n",
       "      <th>average_distance</th>\n",
       "      <th>median_distance</th>\n",
       "      <th>expected</th>\n",
       "      <th>weighted_median</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>L01-1</td>\n",
       "      <td>1</td>\n",
       "      <td>37956.400000</td>\n",
       "      <td>14958.0</td>\n",
       "      <td>50612.642789</td>\n",
       "      <td>48877.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>L01-1</td>\n",
       "      <td>2</td>\n",
       "      <td>38750.375000</td>\n",
       "      <td>14957.5</td>\n",
       "      <td>49624.484261</td>\n",
       "      <td>48875.5</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>L01-1</td>\n",
       "      <td>3</td>\n",
       "      <td>30962.933333</td>\n",
       "      <td>18428.5</td>\n",
       "      <td>32979.744036</td>\n",
       "      <td>34049.5</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>L01-2</td>\n",
       "      <td>1</td>\n",
       "      <td>37956.400000</td>\n",
       "      <td>14958.0</td>\n",
       "      <td>50612.642789</td>\n",
       "      <td>48877.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>L01-2</td>\n",
       "      <td>2</td>\n",
       "      <td>32031.406250</td>\n",
       "      <td>14956.5</td>\n",
       "      <td>43147.040845</td>\n",
       "      <td>34051.0</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "    line  gen_id  average_distance  median_distance      expected  \\\n",
       "0  L01-1       1      37956.400000          14958.0  50612.642789   \n",
       "1  L01-1       2      38750.375000          14957.5  49624.484261   \n",
       "2  L01-1       3      30962.933333          18428.5  32979.744036   \n",
       "3  L01-2       1      37956.400000          14958.0  50612.642789   \n",
       "4  L01-2       2      32031.406250          14956.5  43147.040845   \n",
       "\n",
       "   weighted_median  \n",
       "0          48877.0  \n",
       "1          48875.5  \n",
       "2          34049.5  \n",
       "3          48877.0  \n",
       "4          34051.0  "
      ]
     },
     "execution_count": 22,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "# calculate the expected median\n",
    "def calculate_expected_median(group):\n",
    "    # Sort the group by distance\n",
    "    sorted_group = group.sort_values(by='distance')\n",
    "    \n",
    "    # Calculate the cumulative sum of distances\n",
    "    cumulative_sum = sorted_group['distance'].cumsum()\n",
    "    \n",
    "    # Total sum of distances for the group\n",
    "    total_sum = cumulative_sum.iloc[-1]\n",
    "    \n",
    "    # Find the median value: where the cumulative sum reaches half of the total sum\n",
    "    # At such point, half the events will be related to ISs with shorter distances and the other half with longer distances\n",
    "    median_index = cumulative_sum[cumulative_sum >= total_sum / 2].index[0]\n",
    "    weighted_median = sorted_group.loc[median_index, 'distance'] / 2\n",
    "    \n",
    "    return weighted_median\n",
    "\n",
    "# Apply the weighted median calculation to each group\n",
    "median_values = grouped.apply(calculate_expected_median).reset_index(name='weighted_median')\n",
    "\n",
    "# Merging the weighted median with the previous results\n",
    "final_results = is_distance_stats.merge(median_values, on=['line', 'gen_id'])\n",
    "\n",
    "final_results.head()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 23,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "The median (IQR) of the expected distances between IS elements and the closest essential genes is:\n",
      "33787.50 (32818.50-47095.25).\n",
      "Number of genomes: 87\n"
     ]
    }
   ],
   "source": [
    "# IQR of weighed medians\n",
    "expected_distances = final_results.query('gen_id == 1 | gen_id == 2').\\\n",
    "\tquery('not(line == \"L04-1\" and gen_id == 2)').weighted_median\n",
    "# in text\n",
    "print(f'The median (IQR) of the expected distances between IS elements and the closest essential genes is:')\n",
    "print(f'{np.median(expected_distances):.2f} ({np.percentile(expected_distances, 25):.2f}-{np.percentile(expected_distances, 75):.2f}).')\n",
    "print(f'Number of genomes: {len(expected_distances)}')"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "## Expected median deletions by normalizing the weights for each genome"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 33,
   "metadata": {},
   "outputs": [],
   "source": [
    "merged_distances_median = merged_distances.query('gen_id == 1 | gen_id == 2').\\\n",
    "\tquery('not(line == \"L04-1\" and gen_id == 2)').\\\n",
    "\tsort_values(by='distance')"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 34,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "       line  gen_id         prime_end  distance  initial_weight  factor  \\\n",
      "1941  L09-4       1  distance_5_prime        23              23       8   \n",
      "1953  L09-4       2  distance_5_prime        23              23      12   \n",
      "1891  L09-3       1  distance_5_prime        23              23       8   \n",
      "1865  L09-2       2  distance_5_prime        23              23      12   \n",
      "1905  L09-3       2  distance_5_prime        23              23      12   \n",
      "...     ...     ...               ...       ...             ...     ...   \n",
      "2084  L10-3       2  distance_5_prime    239737          239737      12   \n",
      "2128  L10-4       2  distance_5_prime    239740          239740      12   \n",
      "1990  L10-1       2  distance_5_prime    239742          239742      12   \n",
      "843   L04-3       2  distance_5_prime    340675          340675      12   \n",
      "844   L04-3       2  distance_5_prime    345583          345583      12   \n",
      "\n",
      "      normalized_weight  \n",
      "1941           0.000272  \n",
      "1953           0.000361  \n",
      "1891           0.000250  \n",
      "1865           0.000336  \n",
      "1905           0.000230  \n",
      "...                 ...  \n",
      "2084           2.046176  \n",
      "2128           2.311581  \n",
      "1990           2.411804  \n",
      "843            1.219398  \n",
      "844            1.236966  \n",
      "\n",
      "[2632 rows x 7 columns]\n"
     ]
    }
   ],
   "source": [
    "# Calculate initial weights based on distance\n",
    "merged_distances_median['initial_weight'] = merged_distances_median['distance']\n",
    "\n",
    "# Determine factors based on gen_id (8 for gen_id = 1, 12 for gen_id = 2)\n",
    "merged_distances_median['factor'] = np.where(merged_distances_median['gen_id'] == 1, 8, 12)\n",
    "\n",
    "# Group by 'line' and 'gen_id', and calculate the sum of weights for normalization\n",
    "grouped_weights = merged_distances_median.groupby(['line', 'gen_id'])['initial_weight'].transform('sum')\n",
    "\n",
    "# Normalize weights within each group and apply factor\n",
    "merged_distances_median['normalized_weight'] = (merged_distances_median['initial_weight'] / grouped_weights) * merged_distances_median['factor']\n",
    "\n",
    "# Print DataFrame to see the normalized weights\n",
    "print(merged_distances_median)"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "## STATS: weighed median"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 35,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Analytically estimated expected median deletion size: 34338.5\n"
     ]
    }
   ],
   "source": [
    "# Calculate cumulative weight to find the median\n",
    "# This approach assumes uniform distribution of deletion sizes from 1 to distance\n",
    "total_weight = merged_distances_median['normalized_weight'].sum()\n",
    "merged_distances_median['cumulative_weight'] = merged_distances_median['normalized_weight'].cumsum()\n",
    "\n",
    "# Locate the median by finding where the cumulative weight exceeds half of the total weight\n",
    "half_total_weight = total_weight / 2\n",
    "median_row = merged_distances_median[merged_distances_median['cumulative_weight'] >= half_total_weight].iloc[0]\n",
    "median_deletion_size = median_row['distance'] / 2 + 20  # Median of uniform distribution from 1 to distance\n",
    "\n",
    "# Display the analytically estimated expected median deletion size\n",
    "print(\"Analytically estimated expected median deletion size:\", median_deletion_size)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 36,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>line</th>\n",
       "      <th>gen_id</th>\n",
       "      <th>prime_end</th>\n",
       "      <th>distance</th>\n",
       "      <th>initial_weight</th>\n",
       "      <th>factor</th>\n",
       "      <th>normalized_weight</th>\n",
       "      <th>cumulative_weight</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>1941</th>\n",
       "      <td>L09-4</td>\n",
       "      <td>1</td>\n",
       "      <td>distance_5_prime</td>\n",
       "      <td>23</td>\n",
       "      <td>23</td>\n",
       "      <td>8</td>\n",
       "      <td>0.000272</td>\n",
       "      <td>0.000272</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1953</th>\n",
       "      <td>L09-4</td>\n",
       "      <td>2</td>\n",
       "      <td>distance_5_prime</td>\n",
       "      <td>23</td>\n",
       "      <td>23</td>\n",
       "      <td>12</td>\n",
       "      <td>0.000361</td>\n",
       "      <td>0.000634</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1891</th>\n",
       "      <td>L09-3</td>\n",
       "      <td>1</td>\n",
       "      <td>distance_5_prime</td>\n",
       "      <td>23</td>\n",
       "      <td>23</td>\n",
       "      <td>8</td>\n",
       "      <td>0.000250</td>\n",
       "      <td>0.000883</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1865</th>\n",
       "      <td>L09-2</td>\n",
       "      <td>2</td>\n",
       "      <td>distance_5_prime</td>\n",
       "      <td>23</td>\n",
       "      <td>23</td>\n",
       "      <td>12</td>\n",
       "      <td>0.000336</td>\n",
       "      <td>0.001220</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1905</th>\n",
       "      <td>L09-3</td>\n",
       "      <td>2</td>\n",
       "      <td>distance_5_prime</td>\n",
       "      <td>23</td>\n",
       "      <td>23</td>\n",
       "      <td>12</td>\n",
       "      <td>0.000230</td>\n",
       "      <td>0.001449</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2084</th>\n",
       "      <td>L10-3</td>\n",
       "      <td>2</td>\n",
       "      <td>distance_5_prime</td>\n",
       "      <td>239737</td>\n",
       "      <td>239737</td>\n",
       "      <td>12</td>\n",
       "      <td>2.046176</td>\n",
       "      <td>860.820251</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2128</th>\n",
       "      <td>L10-4</td>\n",
       "      <td>2</td>\n",
       "      <td>distance_5_prime</td>\n",
       "      <td>239740</td>\n",
       "      <td>239740</td>\n",
       "      <td>12</td>\n",
       "      <td>2.311581</td>\n",
       "      <td>863.131831</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1990</th>\n",
       "      <td>L10-1</td>\n",
       "      <td>2</td>\n",
       "      <td>distance_5_prime</td>\n",
       "      <td>239742</td>\n",
       "      <td>239742</td>\n",
       "      <td>12</td>\n",
       "      <td>2.411804</td>\n",
       "      <td>865.543636</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>843</th>\n",
       "      <td>L04-3</td>\n",
       "      <td>2</td>\n",
       "      <td>distance_5_prime</td>\n",
       "      <td>340675</td>\n",
       "      <td>340675</td>\n",
       "      <td>12</td>\n",
       "      <td>1.219398</td>\n",
       "      <td>866.763034</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>844</th>\n",
       "      <td>L04-3</td>\n",
       "      <td>2</td>\n",
       "      <td>distance_5_prime</td>\n",
       "      <td>345583</td>\n",
       "      <td>345583</td>\n",
       "      <td>12</td>\n",
       "      <td>1.236966</td>\n",
       "      <td>868.000000</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>2632 rows  8 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "       line  gen_id         prime_end  distance  initial_weight  factor  \\\n",
       "1941  L09-4       1  distance_5_prime        23              23       8   \n",
       "1953  L09-4       2  distance_5_prime        23              23      12   \n",
       "1891  L09-3       1  distance_5_prime        23              23       8   \n",
       "1865  L09-2       2  distance_5_prime        23              23      12   \n",
       "1905  L09-3       2  distance_5_prime        23              23      12   \n",
       "...     ...     ...               ...       ...             ...     ...   \n",
       "2084  L10-3       2  distance_5_prime    239737          239737      12   \n",
       "2128  L10-4       2  distance_5_prime    239740          239740      12   \n",
       "1990  L10-1       2  distance_5_prime    239742          239742      12   \n",
       "843   L04-3       2  distance_5_prime    340675          340675      12   \n",
       "844   L04-3       2  distance_5_prime    345583          345583      12   \n",
       "\n",
       "      normalized_weight  cumulative_weight  \n",
       "1941           0.000272           0.000272  \n",
       "1953           0.000361           0.000634  \n",
       "1891           0.000250           0.000883  \n",
       "1865           0.000336           0.001220  \n",
       "1905           0.000230           0.001449  \n",
       "...                 ...                ...  \n",
       "2084           2.046176         860.820251  \n",
       "2128           2.311581         863.131831  \n",
       "1990           2.411804         865.543636  \n",
       "843            1.219398         866.763034  \n",
       "844            1.236966         868.000000  \n",
       "\n",
       "[2632 rows x 8 columns]"
      ]
     },
     "execution_count": 36,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "merged_distances_median"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "#### Bootstrap with percentile method (not used)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 37,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "95% Confidence Interval for the Weighted Median of Distance: [33807. 37755.]\n"
     ]
    }
   ],
   "source": [
    "# Function to calculate the weighted median for a given DataFrame\n",
    "def weighted_median(df, value_column, weight_column):\n",
    "    df_sorted = df.sort_values(value_column)\n",
    "    cumsum = df_sorted[weight_column].cumsum()\n",
    "    cutoff = df_sorted[weight_column].sum() / 2.0\n",
    "    return df_sorted[cumsum >= cutoff][value_column].iloc[0]/2 + 20\n",
    "\n",
    "# Bootstrapping function to resample and calculate the weighted median\n",
    "def bootstrap_sample(df, n_iterations=1000):\n",
    "    results = []\n",
    "    for _ in range(n_iterations):\n",
    "        # Sample with replacement\n",
    "        sample = df.sample(n=len(df), replace=True)\n",
    "        # Calculate the weighted median of 'distance' using 'normalized_weight'\n",
    "        median = weighted_median(sample, 'distance', 'normalized_weight')\n",
    "        results.append(median)\n",
    "    return np.percentile(results, [2.5, 97.5])  # 95% confidence interval\n",
    "\n",
    "# Perform bootstrapping\n",
    "confidence_interval = bootstrap_sample(merged_distances_median)\n",
    "\n",
    "print(\"95% Confidence Interval for the Weighted Median of Distance:\", confidence_interval)"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "#### STATS: Bootstrap with BCa"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 38,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "BCa Confidence Interval for the Weighted Median of Distance: ConfidenceInterval(low=33807.0, high=38628.0)\n"
     ]
    }
   ],
   "source": [
    "import numpy as np\n",
    "import pandas as pd\n",
    "from scipy.stats import bootstrap\n",
    "\n",
    "# Function to calculate the weighted median\n",
    "def weighted_median(data, weights):\n",
    "    # Ensure data and weights are numpy arrays for efficient computation\n",
    "    data, weights = np.array(data), np.array(weights)\n",
    "    # Sort data and corresponding weights\n",
    "    sorted_indices = np.argsort(data)\n",
    "    sorted_data = data[sorted_indices]\n",
    "    sorted_weights = weights[sorted_indices]\n",
    "    # Compute cumulative weights\n",
    "    cumsum_weights = np.cumsum(sorted_weights)\n",
    "    # Find the index where the cumulative weight is greater than or equal to half the total weight\n",
    "    cutoff = cumsum_weights[-1] / 2.0 # -1 returns total weight\n",
    "    median_index = np.searchsorted(cumsum_weights, cutoff) # a[i-1] < v <= a[i]\n",
    "    return sorted_data[median_index] / 2.0 + 20  # given the distance the expected median is halft + 1 for unifromal distribution\n",
    "\n",
    "# Function to perform bootstrapping using the BCa method\n",
    "def bootstrap_bca_median(df, value_column, weight_column, n_resamples=1000):\n",
    "    # Extract values and weights from DataFrame\n",
    "    values = df[value_column].to_numpy()\n",
    "    weights = df[weight_column].to_numpy()\n",
    "\n",
    "    # Define a function for use with bootstrap that computes the weighted median\n",
    "    def compute_median(indices):\n",
    "        # indices to sample data and weights for the bootstrap sample\n",
    "        sampled_values = values[indices]\n",
    "        sampled_weights = weights[indices]\n",
    "        return weighted_median(sampled_values, sampled_weights)\n",
    "\n",
    "    # Perform bootstrap with BCa method\n",
    "    res = bootstrap((np.arange(len(values)),), statistic=compute_median, \n",
    "                    n_resamples=n_resamples, method='BCa', vectorized=False)\n",
    "    return res.confidence_interval\n",
    "\n",
    "# Perform bootstrapping and print the confidence interval\n",
    "confidence_interval = bootstrap_bca_median(merged_distances_median, 'distance', 'normalized_weight')\n",
    "\n",
    "print(\"BCa Confidence Interval for the Weighted Median of Distance:\", confidence_interval)"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "### Save weight data"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 45,
   "metadata": {},
   "outputs": [],
   "source": [
    "merged_distances_median.to_csv(os.path.join(data_export_dir, file_name_base + \"essential_gene_distance_weighted.csv\"))"
   ]
  }
 ],
 "metadata": {
  "kernelspec": {
   "display_name": "Python 3 (ipykernel)",
   "language": "python",
   "name": "python3"
  },
  "language_info": {
   "codemirror_mode": {
    "name": "ipython",
    "version": 3
   },
   "file_extension": ".py",
   "mimetype": "text/x-python",
   "name": "python",
   "nbconvert_exporter": "python",
   "pygments_lexer": "ipython3",
   "version": "3.9.17"
  }
 },
 "nbformat": 4,
 "nbformat_minor": 2
}
