{
 "cells": [
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "Given fasta files as specified in file_list, run SyRI_IS to identify the IS positions (SyRI_IS), SVs  (SyRI_IS) and the relationship among ISs of different time-points (classify_IS_events).\n",
    "\n",
    "# Overview\n",
    "\n",
    "1. load file_list\n",
    "2. run SyRI_IS\n",
    "3. run classify_IS_events against ...\n",
    "    1. consecutive rounds of sequence\n",
    "    2. MA ancestor (after FACS enrichment)\n",
    "    3. MDS42 + IS1\n",
    "4. Aggeregate stats for other ipynb\n",
    "    1. Genomesize (03)\n",
    "    2. Distance of insertions from the closest preexisting IS (04)\n",
    "    3. position of iss in the gneome (02)\n",
    "    4. Find identical mutations in different runs to avoid over counting\n",
    "    5. get statistics of each genome\n",
    "    6. aggregate sv data (MDS) (02)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 1,
   "metadata": {},
   "outputs": [],
   "source": [
    "%load_ext autoreload\n",
    "%autoreload 2\n",
    "\n",
    "import pandas as pd\n",
    "import numpy as np\n",
    "from datetime import datetime, timedelta\n",
    "import sys\n",
    "import os\n",
    "import glob\n",
    "from Bio import SeqIO\n",
    "from Bio.Seq import Seq\n",
    "from Bio.SeqRecord import SeqRecord\n",
    "sys.path.append('../src/SyRI_IS/')\n",
    "import importlib, subprocess\n",
    "import run_utils"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 2,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>IS_Detect_ID</th>\n",
       "      <th>ParentLine</th>\n",
       "      <th>SubLine</th>\n",
       "      <th>gen</th>\n",
       "      <th>file_name</th>\n",
       "      <th>Anc</th>\n",
       "      <th>RecA</th>\n",
       "      <th>Prefix</th>\n",
       "      <th>sample_name_raw</th>\n",
       "      <th>Contig_Date</th>\n",
       "      <th>Complete</th>\n",
       "      <th>Folder</th>\n",
       "      <th>Unnamed: 12</th>\n",
       "      <th>Unnamed: 13</th>\n",
       "      <th>Unnamed: 14</th>\n",
       "      <th>Unnamed: 15</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td></td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>Anc</td>\n",
       "      <td>20230904/MDS42_IS1.fa</td>\n",
       "      <td></td>\n",
       "      <td>1</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>L01_Anc</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>FACS</td>\n",
       "      <td>20231004/L01_Anc_m1.fasta</td>\n",
       "      <td>R01</td>\n",
       "      <td>1</td>\n",
       "      <td>L01-1</td>\n",
       "      <td>L01_Anc</td>\n",
       "      <td>20220418</td>\n",
       "      <td>TRUE</td>\n",
       "      <td>20231004</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>L01_Anc</td>\n",
       "      <td>1</td>\n",
       "      <td>2</td>\n",
       "      <td>FACS</td>\n",
       "      <td>20231004/L01_Anc_m1.fasta</td>\n",
       "      <td>R01</td>\n",
       "      <td>1</td>\n",
       "      <td>L01-2</td>\n",
       "      <td>L01_Anc</td>\n",
       "      <td>20220418</td>\n",
       "      <td>TRUE</td>\n",
       "      <td>20231004</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>2</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>L01_Anc</td>\n",
       "      <td>1</td>\n",
       "      <td>3</td>\n",
       "      <td>FACS</td>\n",
       "      <td>20231004/L01_Anc_m1.fasta</td>\n",
       "      <td>R01</td>\n",
       "      <td>1</td>\n",
       "      <td>L01-3</td>\n",
       "      <td>L01_Anc</td>\n",
       "      <td>20220418</td>\n",
       "      <td>TRUE</td>\n",
       "      <td>20231004</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>3</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>L01_Anc</td>\n",
       "      <td>1</td>\n",
       "      <td>4</td>\n",
       "      <td>FACS</td>\n",
       "      <td>20231004/L01_Anc_m1.fasta</td>\n",
       "      <td>R01</td>\n",
       "      <td>1</td>\n",
       "      <td>L01-4</td>\n",
       "      <td>L01_Anc</td>\n",
       "      <td>20220418</td>\n",
       "      <td>TRUE</td>\n",
       "      <td>20231004</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>4</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "  IS_Detect_ID  ParentLine  SubLine   gen                  file_name  Anc  \\\n",
       "0                        0        0   Anc      20230904/MDS42_IS1.fa        \n",
       "1      L01_Anc           1        1  FACS  20231004/L01_Anc_m1.fasta  R01   \n",
       "2      L01_Anc           1        2  FACS  20231004/L01_Anc_m1.fasta  R01   \n",
       "3      L01_Anc           1        3  FACS  20231004/L01_Anc_m1.fasta  R01   \n",
       "4      L01_Anc           1        4  FACS  20231004/L01_Anc_m1.fasta  R01   \n",
       "\n",
       "   RecA Prefix sample_name_raw Contig_Date Complete    Folder Unnamed: 12  \\\n",
       "0     1                                                                     \n",
       "1     1  L01-1         L01_Anc    20220418     TRUE  20231004               \n",
       "2     1  L01-2         L01_Anc    20220418     TRUE  20231004               \n",
       "3     1  L01-3         L01_Anc    20220418     TRUE  20231004               \n",
       "4     1  L01-4         L01_Anc    20220418     TRUE  20231004               \n",
       "\n",
       "  Unnamed: 13 Unnamed: 14 Unnamed: 15  \n",
       "0                                      \n",
       "1                                   1  \n",
       "2                                   2  \n",
       "3                                   3  \n",
       "4                                   4  "
      ]
     },
     "execution_count": 2,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "file_df = pd.read_csv('../fasta/File_list_20240703.csv', na_filter=False)\n",
    "file_df.head()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 3,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>IS_Detect_ID</th>\n",
       "      <th>ParentLine</th>\n",
       "      <th>SubLine</th>\n",
       "      <th>gen</th>\n",
       "      <th>file_name</th>\n",
       "      <th>Anc</th>\n",
       "      <th>RecA</th>\n",
       "      <th>Prefix</th>\n",
       "      <th>sample_name_raw</th>\n",
       "      <th>Contig_Date</th>\n",
       "      <th>Complete</th>\n",
       "      <th>Folder</th>\n",
       "      <th>Unnamed: 12</th>\n",
       "      <th>Unnamed: 13</th>\n",
       "      <th>Unnamed: 14</th>\n",
       "      <th>Unnamed: 15</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>128</th>\n",
       "      <td>L10-4_G20</td>\n",
       "      <td>10</td>\n",
       "      <td>4</td>\n",
       "      <td>20</td>\n",
       "      <td>20231003/L10-4_G20.fasta</td>\n",
       "      <td>r04</td>\n",
       "      <td>0</td>\n",
       "      <td>L10-4</td>\n",
       "      <td>L10-4_G20</td>\n",
       "      <td>20230909</td>\n",
       "      <td>TRUE</td>\n",
       "      <td>20231003</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>128</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>129</th>\n",
       "      <td>L11-1_G20</td>\n",
       "      <td>11</td>\n",
       "      <td>1</td>\n",
       "      <td>20</td>\n",
       "      <td>20231003/L11-1_G20.fasta</td>\n",
       "      <td>r05</td>\n",
       "      <td>0</td>\n",
       "      <td>L11-1</td>\n",
       "      <td>L11-1_G20</td>\n",
       "      <td>20230909</td>\n",
       "      <td>TRUE</td>\n",
       "      <td>20231003</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>129</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>130</th>\n",
       "      <td>L11-2_G20</td>\n",
       "      <td>11</td>\n",
       "      <td>2</td>\n",
       "      <td>20</td>\n",
       "      <td>20231003/L11-2_G20.fasta</td>\n",
       "      <td>r05</td>\n",
       "      <td>0</td>\n",
       "      <td>L11-2</td>\n",
       "      <td>L11-2_G20</td>\n",
       "      <td>20230909</td>\n",
       "      <td>TRUE</td>\n",
       "      <td>20231003</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>130</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>131</th>\n",
       "      <td>L11-3_G20</td>\n",
       "      <td>11</td>\n",
       "      <td>3</td>\n",
       "      <td>20</td>\n",
       "      <td>20231003/L11-3_G20.fasta</td>\n",
       "      <td>r05</td>\n",
       "      <td>0</td>\n",
       "      <td>L11-3</td>\n",
       "      <td>L11-3_G20</td>\n",
       "      <td>20230909</td>\n",
       "      <td>TRUE</td>\n",
       "      <td>20231003</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>131</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>132</th>\n",
       "      <td>L11-4_G20</td>\n",
       "      <td>11</td>\n",
       "      <td>4</td>\n",
       "      <td>20</td>\n",
       "      <td>20231003/L11-4_G20.fasta</td>\n",
       "      <td>r05</td>\n",
       "      <td>0</td>\n",
       "      <td>L11-4</td>\n",
       "      <td>L11-4_G20</td>\n",
       "      <td>20230909</td>\n",
       "      <td>FALSE</td>\n",
       "      <td>20231003</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>132</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "    IS_Detect_ID  ParentLine  SubLine gen                 file_name  Anc  \\\n",
       "128    L10-4_G20          10        4  20  20231003/L10-4_G20.fasta  r04   \n",
       "129    L11-1_G20          11        1  20  20231003/L11-1_G20.fasta  r05   \n",
       "130    L11-2_G20          11        2  20  20231003/L11-2_G20.fasta  r05   \n",
       "131    L11-3_G20          11        3  20  20231003/L11-3_G20.fasta  r05   \n",
       "132    L11-4_G20          11        4  20  20231003/L11-4_G20.fasta  r05   \n",
       "\n",
       "     RecA Prefix sample_name_raw Contig_Date Complete    Folder Unnamed: 12  \\\n",
       "128     0  L10-4       L10-4_G20    20230909     TRUE  20231003               \n",
       "129     0  L11-1       L11-1_G20    20230909     TRUE  20231003               \n",
       "130     0  L11-2       L11-2_G20    20230909     TRUE  20231003               \n",
       "131     0  L11-3       L11-3_G20    20230909     TRUE  20231003               \n",
       "132     0  L11-4       L11-4_G20    20230909    FALSE  20231003               \n",
       "\n",
       "    Unnamed: 13 Unnamed: 14 Unnamed: 15  \n",
       "128                                 128  \n",
       "129                                 129  \n",
       "130                                 130  \n",
       "131                                 131  \n",
       "132                                 132  "
      ]
     },
     "execution_count": 3,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "file_df.tail()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 4,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "['L01-1', 'L01-2', 'L01-3', 'L01-4', 'L02-1'] 44\n"
     ]
    }
   ],
   "source": [
    "master_dir = 'exp/multiple_runs11'\n",
    "is_detect_dir = 'exp/is_detect11'\n",
    "tmpdir = os.path.join(master_dir, 'tmp')\n",
    "exportdir = os.path.join(master_dir, 'export')\n",
    "logdir = os.path.join(master_dir, 'log')\n",
    "stderr_out = os.path.join(logdir, 'stderr_out')\n",
    "os.makedirs(stderr_out, exist_ok=True)\n",
    "fastadir = '../fasta/'\n",
    "lines = file_df.Prefix.unique()\n",
    "lines = [line for line in lines if line != '']\n",
    "is_fasta = os.path.join(fastadir, 'IS1.fasta')\n",
    "print(lines[0:5], len(lines))\n",
    "os.makedirs(logdir, exist_ok=True)"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "# Run SyRI_IS"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 5,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "../fasta/20231004/L01_Anc_m1.fasta\n",
      "contig_cnt 1\n",
      "../fasta/20231013/L01-1_G8_m.fasta\n",
      "contig_cnt 1\n",
      "../fasta/20231003/L01-1_G20.fasta\n",
      "contig_cnt 1\n",
      "python3 ../src/SyRI_IS/runSyRI_IS.py -i ../fasta/IS1.fasta -p L01-1 --replicationOrigin 3359187 --tmp exp/multiple_runs11/tmp --out exp/multiple_runs11/export --log exp/multiple_runs11/log --rerun  -g ../fasta/20230904/MDS42_IS1.fa -g ../fasta/20231004/L01_Anc_m1.fasta -g ../fasta/20231013/L01-1_G8_m.fasta -g ../fasta/20231003/L01-1_G20.fasta\n",
      "../fasta/20231004/L01_Anc_m1.fasta\n",
      "contig_cnt 1\n",
      "../fasta/20231003/L01-2_G8.fasta\n",
      "contig_cnt 1\n",
      "../fasta/20231003/L01-2_G20.fasta\n",
      "contig_cnt 1\n",
      "python3 ../src/SyRI_IS/runSyRI_IS.py -i ../fasta/IS1.fasta -p L01-2 --replicationOrigin 3359187 --tmp exp/multiple_runs11/tmp --out exp/multiple_runs11/export --log exp/multiple_runs11/log --rerun  -g ../fasta/20230904/MDS42_IS1.fa -g ../fasta/20231004/L01_Anc_m1.fasta -g ../fasta/20231003/L01-2_G8.fasta -g ../fasta/20231003/L01-2_G20.fasta\n",
      "../fasta/20231004/L01_Anc_m1.fasta\n",
      "contig_cnt 1\n",
      "../fasta/20231003/L01-3_G8.fasta\n",
      "contig_cnt 1\n",
      "../fasta/20231003/L01-3_G20.fasta\n",
      "contig_cnt 1\n",
      "python3 ../src/SyRI_IS/runSyRI_IS.py -i ../fasta/IS1.fasta -p L01-3 --replicationOrigin 3359187 --tmp exp/multiple_runs11/tmp --out exp/multiple_runs11/export --log exp/multiple_runs11/log --rerun  -g ../fasta/20230904/MDS42_IS1.fa -g ../fasta/20231004/L01_Anc_m1.fasta -g ../fasta/20231003/L01-3_G8.fasta -g ../fasta/20231003/L01-3_G20.fasta\n",
      "../fasta/20231004/L01_Anc_m1.fasta\n",
      "contig_cnt 1\n",
      "../fasta/20231003/L01-4_G8.fasta\n",
      "contig_cnt 1\n",
      "../fasta/20231003/L01-4_G20.fasta\n",
      "contig_cnt 1\n",
      "python3 ../src/SyRI_IS/runSyRI_IS.py -i ../fasta/IS1.fasta -p L01-4 --replicationOrigin 3359187 --tmp exp/multiple_runs11/tmp --out exp/multiple_runs11/export --log exp/multiple_runs11/log --rerun  -g ../fasta/20230904/MDS42_IS1.fa -g ../fasta/20231004/L01_Anc_m1.fasta -g ../fasta/20231003/L01-4_G8.fasta -g ../fasta/20231003/L01-4_G20.fasta\n",
      "../fasta/20231003/L02_Anc.fasta\n",
      "contig_cnt 1\n",
      "../fasta/20231003/L02-1_G8.fasta\n",
      "contig_cnt 1\n",
      "../fasta/20231003/L02-1_G20.fasta\n",
      "contig_cnt 1\n",
      "python3 ../src/SyRI_IS/runSyRI_IS.py -i ../fasta/IS1.fasta -p L02-1 --replicationOrigin 3359187 --tmp exp/multiple_runs11/tmp --out exp/multiple_runs11/export --log exp/multiple_runs11/log --rerun  -g ../fasta/20230904/MDS42_IS1.fa -g ../fasta/20231003/L02_Anc.fasta -g ../fasta/20231003/L02-1_G8.fasta -g ../fasta/20231003/L02-1_G20.fasta\n",
      "../fasta/20231003/L02_Anc.fasta\n",
      "contig_cnt 1\n",
      "../fasta/20231003/L02-2_G8.fasta\n",
      "contig_cnt 1\n",
      "../fasta/20231003/L02-2_G20.fasta\n",
      "contig_cnt 1\n",
      "python3 ../src/SyRI_IS/runSyRI_IS.py -i ../fasta/IS1.fasta -p L02-2 --replicationOrigin 3359187 --tmp exp/multiple_runs11/tmp --out exp/multiple_runs11/export --log exp/multiple_runs11/log --rerun  -g ../fasta/20230904/MDS42_IS1.fa -g ../fasta/20231003/L02_Anc.fasta -g ../fasta/20231003/L02-2_G8.fasta -g ../fasta/20231003/L02-2_G20.fasta\n",
      "../fasta/20231003/L02_Anc.fasta\n",
      "contig_cnt 1\n",
      "../fasta/20231003/L02-3_G8.fasta\n",
      "contig_cnt 1\n",
      "../fasta/20231003/L02-3_G20.fasta\n",
      "contig_cnt 1\n",
      "python3 ../src/SyRI_IS/runSyRI_IS.py -i ../fasta/IS1.fasta -p L02-3 --replicationOrigin 3359187 --tmp exp/multiple_runs11/tmp --out exp/multiple_runs11/export --log exp/multiple_runs11/log --rerun  -g ../fasta/20230904/MDS42_IS1.fa -g ../fasta/20231003/L02_Anc.fasta -g ../fasta/20231003/L02-3_G8.fasta -g ../fasta/20231003/L02-3_G20.fasta\n",
      "../fasta/20231003/L02_Anc.fasta\n",
      "contig_cnt 1\n",
      "../fasta/20231003/L02-4_G8.fasta\n",
      "contig_cnt 1\n",
      "../fasta/20231003/L02-4_G20.fasta\n",
      "contig_cnt 1\n",
      "python3 ../src/SyRI_IS/runSyRI_IS.py -i ../fasta/IS1.fasta -p L02-4 --replicationOrigin 3359187 --tmp exp/multiple_runs11/tmp --out exp/multiple_runs11/export --log exp/multiple_runs11/log --rerun  -g ../fasta/20230904/MDS42_IS1.fa -g ../fasta/20231003/L02_Anc.fasta -g ../fasta/20231003/L02-4_G8.fasta -g ../fasta/20231003/L02-4_G20.fasta\n",
      "../fasta/20240703/L03_Anc_m1.fasta\n",
      "contig_cnt 1\n",
      "../fasta/20231003/L03-1_G8.fasta\n",
      "contig_cnt 1\n",
      "../fasta/20231004/L03-1_G20_m20231003.fasta\n",
      "contig_cnt 1\n",
      "python3 ../src/SyRI_IS/runSyRI_IS.py -i ../fasta/IS1.fasta -p L03-1 --replicationOrigin 3359187 --tmp exp/multiple_runs11/tmp --out exp/multiple_runs11/export --log exp/multiple_runs11/log --rerun  -g ../fasta/20230904/MDS42_IS1.fa -g ../fasta/20240703/L03_Anc_m1.fasta -g ../fasta/20231003/L03-1_G8.fasta -g ../fasta/20231004/L03-1_G20_m20231003.fasta\n",
      "../fasta/20240703/L03_Anc_m1.fasta\n",
      "contig_cnt 1\n",
      "../fasta/20231003/L03-2_G8.fasta\n",
      "contig_cnt 1\n",
      "../fasta/20231003/L03-2_G20.fasta\n",
      "contig_cnt 1\n",
      "python3 ../src/SyRI_IS/runSyRI_IS.py -i ../fasta/IS1.fasta -p L03-2 --replicationOrigin 3359187 --tmp exp/multiple_runs11/tmp --out exp/multiple_runs11/export --log exp/multiple_runs11/log --rerun  -g ../fasta/20230904/MDS42_IS1.fa -g ../fasta/20240703/L03_Anc_m1.fasta -g ../fasta/20231003/L03-2_G8.fasta -g ../fasta/20231003/L03-2_G20.fasta\n",
      "../fasta/20240703/L03_Anc_m1.fasta\n",
      "contig_cnt 1\n",
      "../fasta/20231003/L03-3_G8.fasta\n",
      "contig_cnt 1\n",
      "../fasta/20231003/L03-3_G20.fasta\n",
      "contig_cnt 1\n",
      "python3 ../src/SyRI_IS/runSyRI_IS.py -i ../fasta/IS1.fasta -p L03-3 --replicationOrigin 3359187 --tmp exp/multiple_runs11/tmp --out exp/multiple_runs11/export --log exp/multiple_runs11/log --rerun  -g ../fasta/20230904/MDS42_IS1.fa -g ../fasta/20240703/L03_Anc_m1.fasta -g ../fasta/20231003/L03-3_G8.fasta -g ../fasta/20231003/L03-3_G20.fasta\n",
      "../fasta/20240703/L03_Anc_m1.fasta\n",
      "contig_cnt 1\n",
      "../fasta/20231003/L03-4_G8.fasta\n",
      "contig_cnt 1\n",
      "../fasta/20240413/L03-4_G20.fasta\n",
      "contig_cnt 1\n",
      "python3 ../src/SyRI_IS/runSyRI_IS.py -i ../fasta/IS1.fasta -p L03-4 --replicationOrigin 3359187 --tmp exp/multiple_runs11/tmp --out exp/multiple_runs11/export --log exp/multiple_runs11/log --rerun  -g ../fasta/20230904/MDS42_IS1.fa -g ../fasta/20240703/L03_Anc_m1.fasta -g ../fasta/20231003/L03-4_G8.fasta -g ../fasta/20240413/L03-4_G20.fasta\n",
      "../fasta/20231004/L04_Anc_m1.fasta\n",
      "contig_cnt 1\n",
      "../fasta/20231003/L04-1_G8.fasta\n",
      "contig_cnt 1\n",
      "../fasta/20231003/L04-1_G20.fasta\n",
      "contig_cnt 1\n",
      "python3 ../src/SyRI_IS/runSyRI_IS.py -i ../fasta/IS1.fasta -p L04-1 --replicationOrigin 3359187 --tmp exp/multiple_runs11/tmp --out exp/multiple_runs11/export --log exp/multiple_runs11/log --rerun  -g ../fasta/20230904/MDS42_IS1.fa -g ../fasta/20231004/L04_Anc_m1.fasta -g ../fasta/20231003/L04-1_G8.fasta -g ../fasta/20231003/L04-1_G20.fasta\n",
      "../fasta/20231004/L04_Anc_m1.fasta\n",
      "contig_cnt 1\n",
      "../fasta/20231003/L04-2_G8.fasta\n",
      "contig_cnt 1\n",
      "../fasta/20231003/L04-2_G20.fasta\n",
      "contig_cnt 1\n",
      "python3 ../src/SyRI_IS/runSyRI_IS.py -i ../fasta/IS1.fasta -p L04-2 --replicationOrigin 3359187 --tmp exp/multiple_runs11/tmp --out exp/multiple_runs11/export --log exp/multiple_runs11/log --rerun  -g ../fasta/20230904/MDS42_IS1.fa -g ../fasta/20231004/L04_Anc_m1.fasta -g ../fasta/20231003/L04-2_G8.fasta -g ../fasta/20231003/L04-2_G20.fasta\n",
      "../fasta/20231004/L04_Anc_m1.fasta\n",
      "contig_cnt 1\n",
      "../fasta/20231003/L04-3_G8.fasta\n",
      "contig_cnt 1\n",
      "../fasta/20231003/L04-3_G20.fasta\n",
      "contig_cnt 1\n",
      "python3 ../src/SyRI_IS/runSyRI_IS.py -i ../fasta/IS1.fasta -p L04-3 --replicationOrigin 3359187 --tmp exp/multiple_runs11/tmp --out exp/multiple_runs11/export --log exp/multiple_runs11/log --rerun  -g ../fasta/20230904/MDS42_IS1.fa -g ../fasta/20231004/L04_Anc_m1.fasta -g ../fasta/20231003/L04-3_G8.fasta -g ../fasta/20231003/L04-3_G20.fasta\n",
      "../fasta/20231004/L04_Anc_m1.fasta\n",
      "contig_cnt 1\n",
      "../fasta/20231003/L04-4_G8.fasta\n",
      "contig_cnt 1\n",
      "../fasta/20231003/L04-4_G20.fasta\n",
      "contig_cnt 1\n",
      "python3 ../src/SyRI_IS/runSyRI_IS.py -i ../fasta/IS1.fasta -p L04-4 --replicationOrigin 3359187 --tmp exp/multiple_runs11/tmp --out exp/multiple_runs11/export --log exp/multiple_runs11/log --rerun  -g ../fasta/20230904/MDS42_IS1.fa -g ../fasta/20231004/L04_Anc_m1.fasta -g ../fasta/20231003/L04-4_G8.fasta -g ../fasta/20231003/L04-4_G20.fasta\n",
      "../fasta/20231004/L05_Anc_m1.fasta\n",
      "contig_cnt 1\n",
      "../fasta/20231003/L05-1_G8.fasta\n",
      "contig_cnt 1\n",
      "../fasta/20231003/L05-1_G20.fasta\n",
      "contig_cnt 1\n",
      "python3 ../src/SyRI_IS/runSyRI_IS.py -i ../fasta/IS1.fasta -p L05-1 --replicationOrigin 3359187 --tmp exp/multiple_runs11/tmp --out exp/multiple_runs11/export --log exp/multiple_runs11/log --rerun  -g ../fasta/20230904/MDS42_IS1.fa -g ../fasta/20231004/L05_Anc_m1.fasta -g ../fasta/20231003/L05-1_G8.fasta -g ../fasta/20231003/L05-1_G20.fasta\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "../fasta/20240703/L05_Anc_m2.fasta\n",
      "contig_cnt 1\n",
      "../fasta/20231003/L05-2_G8.fasta\n",
      "contig_cnt 1\n",
      "../fasta/20231003/L05-2_G20.fasta\n",
      "contig_cnt 1\n",
      "python3 ../src/SyRI_IS/runSyRI_IS.py -i ../fasta/IS1.fasta -p L05-2 --replicationOrigin 3359187 --tmp exp/multiple_runs11/tmp --out exp/multiple_runs11/export --log exp/multiple_runs11/log --rerun  -g ../fasta/20230904/MDS42_IS1.fa -g ../fasta/20240703/L05_Anc_m2.fasta -g ../fasta/20231003/L05-2_G8.fasta -g ../fasta/20231003/L05-2_G20.fasta\n",
      "../fasta/20240703/L05_Anc_m2.fasta\n",
      "contig_cnt 1\n",
      "../fasta/20231003/L05-3_G8.fasta\n",
      "contig_cnt 1\n",
      "../fasta/20231003/L05-3_G20.fasta\n",
      "contig_cnt 1\n",
      "python3 ../src/SyRI_IS/runSyRI_IS.py -i ../fasta/IS1.fasta -p L05-3 --replicationOrigin 3359187 --tmp exp/multiple_runs11/tmp --out exp/multiple_runs11/export --log exp/multiple_runs11/log --rerun  -g ../fasta/20230904/MDS42_IS1.fa -g ../fasta/20240703/L05_Anc_m2.fasta -g ../fasta/20231003/L05-3_G8.fasta -g ../fasta/20231003/L05-3_G20.fasta\n",
      "../fasta/20231004/L05_Anc_m1.fasta\n",
      "contig_cnt 1\n",
      "../fasta/20231003/L05-4_G8.fasta\n",
      "contig_cnt 1\n",
      "../fasta/20231003/L05-4_G20.fasta\n",
      "contig_cnt 1\n",
      "python3 ../src/SyRI_IS/runSyRI_IS.py -i ../fasta/IS1.fasta -p L05-4 --replicationOrigin 3359187 --tmp exp/multiple_runs11/tmp --out exp/multiple_runs11/export --log exp/multiple_runs11/log --rerun  -g ../fasta/20230904/MDS42_IS1.fa -g ../fasta/20231004/L05_Anc_m1.fasta -g ../fasta/20231003/L05-4_G8.fasta -g ../fasta/20231003/L05-4_G20.fasta\n",
      "../fasta/20231004/L06_Anc_m1.fasta\n",
      "contig_cnt 1\n",
      "../fasta/20231003/L06-1_G8.fasta\n",
      "contig_cnt 1\n",
      "../fasta/20231003/L06-1_G20.fasta\n",
      "contig_cnt 1\n",
      "python3 ../src/SyRI_IS/runSyRI_IS.py -i ../fasta/IS1.fasta -p L06-1 --replicationOrigin 3359187 --tmp exp/multiple_runs11/tmp --out exp/multiple_runs11/export --log exp/multiple_runs11/log --rerun  -g ../fasta/20230904/MDS42_IS1.fa -g ../fasta/20231004/L06_Anc_m1.fasta -g ../fasta/20231003/L06-1_G8.fasta -g ../fasta/20231003/L06-1_G20.fasta\n",
      "../fasta/20231004/L06_Anc_m1.fasta\n",
      "contig_cnt 1\n",
      "../fasta/20231003/L06-2_G8.fasta\n",
      "contig_cnt 1\n",
      "../fasta/20231003/L06-2_G20.fasta\n",
      "contig_cnt 1\n",
      "python3 ../src/SyRI_IS/runSyRI_IS.py -i ../fasta/IS1.fasta -p L06-2 --replicationOrigin 3359187 --tmp exp/multiple_runs11/tmp --out exp/multiple_runs11/export --log exp/multiple_runs11/log --rerun  -g ../fasta/20230904/MDS42_IS1.fa -g ../fasta/20231004/L06_Anc_m1.fasta -g ../fasta/20231003/L06-2_G8.fasta -g ../fasta/20231003/L06-2_G20.fasta\n",
      "../fasta/20231004/L06_Anc_m1.fasta\n",
      "contig_cnt 1\n",
      "../fasta/20231003/L06-3_G8.fasta\n",
      "contig_cnt 1\n",
      "../fasta/20231003/L06-3_G20.fasta\n",
      "contig_cnt 1\n",
      "python3 ../src/SyRI_IS/runSyRI_IS.py -i ../fasta/IS1.fasta -p L06-3 --replicationOrigin 3359187 --tmp exp/multiple_runs11/tmp --out exp/multiple_runs11/export --log exp/multiple_runs11/log --rerun  -g ../fasta/20230904/MDS42_IS1.fa -g ../fasta/20231004/L06_Anc_m1.fasta -g ../fasta/20231003/L06-3_G8.fasta -g ../fasta/20231003/L06-3_G20.fasta\n",
      "../fasta/20231004/L06_Anc_m1.fasta\n",
      "contig_cnt 1\n",
      "../fasta/20231003/L06-4_G8.fasta\n",
      "contig_cnt 1\n",
      "../fasta/20231003/L06-4_G20.fasta\n",
      "contig_cnt 1\n",
      "python3 ../src/SyRI_IS/runSyRI_IS.py -i ../fasta/IS1.fasta -p L06-4 --replicationOrigin 3359187 --tmp exp/multiple_runs11/tmp --out exp/multiple_runs11/export --log exp/multiple_runs11/log --rerun  -g ../fasta/20230904/MDS42_IS1.fa -g ../fasta/20231004/L06_Anc_m1.fasta -g ../fasta/20231003/L06-4_G8.fasta -g ../fasta/20231003/L06-4_G20.fasta\n",
      "../fasta/20240703/L07_Anc_m1.fasta\n",
      "contig_cnt 1\n",
      "../fasta/20231003/L07-1_G8.fasta\n",
      "contig_cnt 1\n",
      "../fasta/20231003/L07-1_G20.fasta\n",
      "contig_cnt 1\n",
      "python3 ../src/SyRI_IS/runSyRI_IS.py -i ../fasta/IS1.fasta -p L07-1 --replicationOrigin 3359187 --tmp exp/multiple_runs11/tmp --out exp/multiple_runs11/export --log exp/multiple_runs11/log --rerun  -g ../fasta/20230904/MDS42_IS1.fa -g ../fasta/20240703/L07_Anc_m1.fasta -g ../fasta/20231003/L07-1_G8.fasta -g ../fasta/20231003/L07-1_G20.fasta\n",
      "../fasta/20240703/L07_Anc_m1.fasta\n",
      "contig_cnt 1\n",
      "../fasta/20231003/L07-2_G8.fasta\n",
      "contig_cnt 1\n",
      "../fasta/20231003/L07-2_G20.fasta\n",
      "contig_cnt 1\n",
      "python3 ../src/SyRI_IS/runSyRI_IS.py -i ../fasta/IS1.fasta -p L07-2 --replicationOrigin 3359187 --tmp exp/multiple_runs11/tmp --out exp/multiple_runs11/export --log exp/multiple_runs11/log --rerun  -g ../fasta/20230904/MDS42_IS1.fa -g ../fasta/20240703/L07_Anc_m1.fasta -g ../fasta/20231003/L07-2_G8.fasta -g ../fasta/20231003/L07-2_G20.fasta\n",
      "../fasta/20240703/L07_Anc_m1.fasta\n",
      "contig_cnt 1\n",
      "../fasta/20231003/L07-3_G8.fasta\n",
      "contig_cnt 1\n",
      "../fasta/20231003/L07-3_G20.fasta\n",
      "contig_cnt 1\n",
      "python3 ../src/SyRI_IS/runSyRI_IS.py -i ../fasta/IS1.fasta -p L07-3 --replicationOrigin 3359187 --tmp exp/multiple_runs11/tmp --out exp/multiple_runs11/export --log exp/multiple_runs11/log --rerun  -g ../fasta/20230904/MDS42_IS1.fa -g ../fasta/20240703/L07_Anc_m1.fasta -g ../fasta/20231003/L07-3_G8.fasta -g ../fasta/20231003/L07-3_G20.fasta\n",
      "../fasta/20240703/L07_Anc_m2.fasta\n",
      "contig_cnt 1\n",
      "../fasta/20231004/L07-4_G8_m.fasta\n",
      "contig_cnt 1\n",
      "../fasta/20231003/L07-4_G20.fasta\n",
      "contig_cnt 1\n",
      "python3 ../src/SyRI_IS/runSyRI_IS.py -i ../fasta/IS1.fasta -p L07-4 --replicationOrigin 3359187 --tmp exp/multiple_runs11/tmp --out exp/multiple_runs11/export --log exp/multiple_runs11/log --rerun  -g ../fasta/20230904/MDS42_IS1.fa -g ../fasta/20240703/L07_Anc_m2.fasta -g ../fasta/20231004/L07-4_G8_m.fasta -g ../fasta/20231003/L07-4_G20.fasta\n",
      "../fasta/20231003/L08_Anc.fasta\n",
      "contig_cnt 1\n",
      "../fasta/20231003/L08-1_G8.fasta\n",
      "contig_cnt 1\n",
      "../fasta/20231003/L08-1_G20.fasta\n",
      "contig_cnt 1\n",
      "python3 ../src/SyRI_IS/runSyRI_IS.py -i ../fasta/IS1.fasta -p L08-1 --replicationOrigin 3359187 --tmp exp/multiple_runs11/tmp --out exp/multiple_runs11/export --log exp/multiple_runs11/log --rerun  -g ../fasta/20230904/MDS42_IS1.fa -g ../fasta/20231003/L08_Anc.fasta -g ../fasta/20231003/L08-1_G8.fasta -g ../fasta/20231003/L08-1_G20.fasta\n",
      "../fasta/20231003/L08_Anc.fasta\n",
      "contig_cnt 1\n",
      "../fasta/20231003/L08-2_G8.fasta\n",
      "contig_cnt 1\n",
      "../fasta/20231003/L08-2_G20.fasta\n",
      "contig_cnt 1\n",
      "python3 ../src/SyRI_IS/runSyRI_IS.py -i ../fasta/IS1.fasta -p L08-2 --replicationOrigin 3359187 --tmp exp/multiple_runs11/tmp --out exp/multiple_runs11/export --log exp/multiple_runs11/log --rerun  -g ../fasta/20230904/MDS42_IS1.fa -g ../fasta/20231003/L08_Anc.fasta -g ../fasta/20231003/L08-2_G8.fasta -g ../fasta/20231003/L08-2_G20.fasta\n",
      "../fasta/20231003/L08_Anc.fasta\n",
      "contig_cnt 1\n",
      "../fasta/20231003/L08-3_G8.fasta\n",
      "contig_cnt 1\n",
      "../fasta/20231003/L08-3_G20.fasta\n",
      "contig_cnt 1\n",
      "python3 ../src/SyRI_IS/runSyRI_IS.py -i ../fasta/IS1.fasta -p L08-3 --replicationOrigin 3359187 --tmp exp/multiple_runs11/tmp --out exp/multiple_runs11/export --log exp/multiple_runs11/log --rerun  -g ../fasta/20230904/MDS42_IS1.fa -g ../fasta/20231003/L08_Anc.fasta -g ../fasta/20231003/L08-3_G8.fasta -g ../fasta/20231003/L08-3_G20.fasta\n",
      "../fasta/20231003/L08_Anc.fasta\n",
      "contig_cnt 1\n",
      "../fasta/20231003/L08-4_G8.fasta\n",
      "contig_cnt 1\n",
      "../fasta/20231003/L08-4_G20.fasta\n",
      "contig_cnt 1\n",
      "python3 ../src/SyRI_IS/runSyRI_IS.py -i ../fasta/IS1.fasta -p L08-4 --replicationOrigin 3359187 --tmp exp/multiple_runs11/tmp --out exp/multiple_runs11/export --log exp/multiple_runs11/log --rerun  -g ../fasta/20230904/MDS42_IS1.fa -g ../fasta/20231003/L08_Anc.fasta -g ../fasta/20231003/L08-4_G8.fasta -g ../fasta/20231003/L08-4_G20.fasta\n",
      "../fasta/20231004/L09_Anc_m1.fasta\n",
      "contig_cnt 1\n",
      "../fasta/20231003/L09-1_G8.fasta\n",
      "contig_cnt 1\n",
      "../fasta/20231003/L09-1_G20.fasta\n",
      "contig_cnt 1\n",
      "python3 ../src/SyRI_IS/runSyRI_IS.py -i ../fasta/IS1.fasta -p L09-1 --replicationOrigin 3359187 --tmp exp/multiple_runs11/tmp --out exp/multiple_runs11/export --log exp/multiple_runs11/log --rerun  -g ../fasta/20230904/MDS42_IS1.fa -g ../fasta/20231004/L09_Anc_m1.fasta -g ../fasta/20231003/L09-1_G8.fasta -g ../fasta/20231003/L09-1_G20.fasta\n",
      "../fasta/20231004/L09_Anc_m2.fasta\n",
      "contig_cnt 1\n",
      "../fasta/20231003/L09-2_G8.fasta\n",
      "contig_cnt 1\n",
      "../fasta/20231003/L09-2_G20.fasta\n",
      "contig_cnt 1\n",
      "python3 ../src/SyRI_IS/runSyRI_IS.py -i ../fasta/IS1.fasta -p L09-2 --replicationOrigin 3359187 --tmp exp/multiple_runs11/tmp --out exp/multiple_runs11/export --log exp/multiple_runs11/log --rerun  -g ../fasta/20230904/MDS42_IS1.fa -g ../fasta/20231004/L09_Anc_m2.fasta -g ../fasta/20231003/L09-2_G8.fasta -g ../fasta/20231003/L09-2_G20.fasta\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "../fasta/20231004/L09_Anc_m1.fasta\n",
      "contig_cnt 1\n",
      "../fasta/20231003/L09-3_G8.fasta\n",
      "contig_cnt 1\n",
      "../fasta/20231003/L09-3_G20.fasta\n",
      "contig_cnt 1\n",
      "python3 ../src/SyRI_IS/runSyRI_IS.py -i ../fasta/IS1.fasta -p L09-3 --replicationOrigin 3359187 --tmp exp/multiple_runs11/tmp --out exp/multiple_runs11/export --log exp/multiple_runs11/log --rerun  -g ../fasta/20230904/MDS42_IS1.fa -g ../fasta/20231004/L09_Anc_m1.fasta -g ../fasta/20231003/L09-3_G8.fasta -g ../fasta/20231003/L09-3_G20.fasta\n",
      "../fasta/20231004/L09_Anc_m2.fasta\n",
      "contig_cnt 1\n",
      "../fasta/20231003/L09-4_G8.fasta\n",
      "contig_cnt 1\n",
      "../fasta/20231003/L09-4_G20.fasta\n",
      "contig_cnt 1\n",
      "python3 ../src/SyRI_IS/runSyRI_IS.py -i ../fasta/IS1.fasta -p L09-4 --replicationOrigin 3359187 --tmp exp/multiple_runs11/tmp --out exp/multiple_runs11/export --log exp/multiple_runs11/log --rerun  -g ../fasta/20230904/MDS42_IS1.fa -g ../fasta/20231004/L09_Anc_m2.fasta -g ../fasta/20231003/L09-4_G8.fasta -g ../fasta/20231003/L09-4_G20.fasta\n",
      "../fasta/20231003/L10_Anc.fasta\n",
      "contig_cnt 1\n",
      "../fasta/20231003/L10-1_G8.fasta\n",
      "contig_cnt 1\n",
      "../fasta/20231003/L10-1_G20.fasta\n",
      "contig_cnt 1\n",
      "python3 ../src/SyRI_IS/runSyRI_IS.py -i ../fasta/IS1.fasta -p L10-1 --replicationOrigin 3359187 --tmp exp/multiple_runs11/tmp --out exp/multiple_runs11/export --log exp/multiple_runs11/log --rerun  -g ../fasta/20230904/MDS42_IS1.fa -g ../fasta/20231003/L10_Anc.fasta -g ../fasta/20231003/L10-1_G8.fasta -g ../fasta/20231003/L10-1_G20.fasta\n",
      "../fasta/20231003/L10_Anc.fasta\n",
      "contig_cnt 1\n",
      "../fasta/20231003/L10-2_G8.fasta\n",
      "contig_cnt 1\n",
      "../fasta/20231003/L10-2_G20.fasta\n",
      "contig_cnt 1\n",
      "python3 ../src/SyRI_IS/runSyRI_IS.py -i ../fasta/IS1.fasta -p L10-2 --replicationOrigin 3359187 --tmp exp/multiple_runs11/tmp --out exp/multiple_runs11/export --log exp/multiple_runs11/log --rerun  -g ../fasta/20230904/MDS42_IS1.fa -g ../fasta/20231003/L10_Anc.fasta -g ../fasta/20231003/L10-2_G8.fasta -g ../fasta/20231003/L10-2_G20.fasta\n",
      "../fasta/20231003/L10_Anc.fasta\n",
      "contig_cnt 1\n",
      "../fasta/20231004/L10-3_G8_m20231003.fasta\n",
      "contig_cnt 1\n",
      "../fasta/20231003/L10-3_G20.fasta\n",
      "contig_cnt 1\n",
      "python3 ../src/SyRI_IS/runSyRI_IS.py -i ../fasta/IS1.fasta -p L10-3 --replicationOrigin 3359187 --tmp exp/multiple_runs11/tmp --out exp/multiple_runs11/export --log exp/multiple_runs11/log --rerun  -g ../fasta/20230904/MDS42_IS1.fa -g ../fasta/20231003/L10_Anc.fasta -g ../fasta/20231004/L10-3_G8_m20231003.fasta -g ../fasta/20231003/L10-3_G20.fasta\n",
      "../fasta/20231003/L10_Anc.fasta\n",
      "contig_cnt 1\n",
      "../fasta/20231003/L10-4_G8.fasta\n",
      "contig_cnt 1\n",
      "../fasta/20231003/L10-4_G20.fasta\n",
      "contig_cnt 1\n",
      "python3 ../src/SyRI_IS/runSyRI_IS.py -i ../fasta/IS1.fasta -p L10-4 --replicationOrigin 3359187 --tmp exp/multiple_runs11/tmp --out exp/multiple_runs11/export --log exp/multiple_runs11/log --rerun  -g ../fasta/20230904/MDS42_IS1.fa -g ../fasta/20231003/L10_Anc.fasta -g ../fasta/20231003/L10-4_G8.fasta -g ../fasta/20231003/L10-4_G20.fasta\n",
      "../fasta/20231003/L11_Anc.fasta\n",
      "contig_cnt 1\n",
      "../fasta/20231003/L11-1_G8.fasta\n",
      "contig_cnt 1\n",
      "../fasta/20231003/L11-1_G20.fasta\n",
      "contig_cnt 1\n",
      "python3 ../src/SyRI_IS/runSyRI_IS.py -i ../fasta/IS1.fasta -p L11-1 --replicationOrigin 3359187 --tmp exp/multiple_runs11/tmp --out exp/multiple_runs11/export --log exp/multiple_runs11/log --rerun  -g ../fasta/20230904/MDS42_IS1.fa -g ../fasta/20231003/L11_Anc.fasta -g ../fasta/20231003/L11-1_G8.fasta -g ../fasta/20231003/L11-1_G20.fasta\n",
      "../fasta/20231003/L11_Anc.fasta\n",
      "contig_cnt 1\n",
      "../fasta/20231003/L11-2_G8.fasta\n",
      "contig_cnt 1\n",
      "../fasta/20231003/L11-2_G20.fasta\n",
      "contig_cnt 1\n",
      "python3 ../src/SyRI_IS/runSyRI_IS.py -i ../fasta/IS1.fasta -p L11-2 --replicationOrigin 3359187 --tmp exp/multiple_runs11/tmp --out exp/multiple_runs11/export --log exp/multiple_runs11/log --rerun  -g ../fasta/20230904/MDS42_IS1.fa -g ../fasta/20231003/L11_Anc.fasta -g ../fasta/20231003/L11-2_G8.fasta -g ../fasta/20231003/L11-2_G20.fasta\n",
      "../fasta/20231003/L11_Anc.fasta\n",
      "contig_cnt 1\n",
      "../fasta/20231003/L11-3_G8.fasta\n",
      "contig_cnt 1\n",
      "../fasta/20231003/L11-3_G20.fasta\n",
      "contig_cnt 1\n",
      "python3 ../src/SyRI_IS/runSyRI_IS.py -i ../fasta/IS1.fasta -p L11-3 --replicationOrigin 3359187 --tmp exp/multiple_runs11/tmp --out exp/multiple_runs11/export --log exp/multiple_runs11/log --rerun  -g ../fasta/20230904/MDS42_IS1.fa -g ../fasta/20231003/L11_Anc.fasta -g ../fasta/20231003/L11-3_G8.fasta -g ../fasta/20231003/L11-3_G20.fasta\n",
      "../fasta/20231003/L11_Anc.fasta\n",
      "contig_cnt 1\n",
      "../fasta/20231003/L11-4_G8.fasta\n",
      "contig_cnt 1\n",
      "../fasta/20231003/L11-4_G20.fasta\n",
      "contig_cnt 1\n",
      "python3 ../src/SyRI_IS/runSyRI_IS.py -i ../fasta/IS1.fasta -p L11-4 --replicationOrigin 3359187 --tmp exp/multiple_runs11/tmp --out exp/multiple_runs11/export --log exp/multiple_runs11/log --rerun  -g ../fasta/20230904/MDS42_IS1.fa -g ../fasta/20231003/L11_Anc.fasta -g ../fasta/20231003/L11-4_G8.fasta -g ../fasta/20231003/L11-4_G20.fasta\n"
     ]
    }
   ],
   "source": [
    "import subprocess\n",
    "facs_fasta_files = []\n",
    "ma_fasta_files = []\n",
    "gen_ids = []\n",
    "is_detect_ids = []\n",
    "#lines_ = [line for line in lines if line not in ['L01-3', 'L01-4', 'L10-1', 'L10-2']]\n",
    "#lines_ = ['L01-1', 'L01-2']\n",
    "lines_ = lines\n",
    "for line in lines_:\n",
    "\t# example python3 ../src/SyRI_IS/runSyRI_IS.py -i ../fasta/IS1.fasta -g ../fasta/MDS42_IS1.fa -g ../fasta/L01_Anc.fasta -g ../fasta/L01-2_G8.fasta -p L01-2 --replicationOrigin 3359187\n",
    "\tmds_fasta = os.path.join(fastadir, file_df.file_name[file_df.gen == 'Anc'][0])\n",
    "\t#parent_line = file_df[file_df.Prefix == line].iloc[0].ParentLine\n",
    "\tC = 'python3 ../src/SyRI_IS/runSyRI_IS.py -i ' + is_fasta \n",
    "\tC += ' -p ' + line + ' --replicationOrigin 3359187'\n",
    "\tC += ' --tmp ' + tmpdir + ' --out ' + exportdir\n",
    "\tC += ' --log ' + logdir\n",
    "\tC += ' --rerun '\n",
    "\t#C += ' --rerun_align_two_genomes '\n",
    "\t#C += ' --rerun_run_syri '\n",
    "\tC += ' -g ' + mds_fasta\n",
    "\tanc_fasta = mds_fasta\n",
    "\tfor gen_id, g in enumerate(file_df[file_df.Prefix == line].file_name):\n",
    "\t\tfasta_file = os.path.join(fastadir, g)\n",
    "\t\tprint(fasta_file)\n",
    "\t\tfasta_file = run_utils.get_full_contig(fasta_file, tmpdir=tmpdir, basefasta=anc_fasta, skip_if_exists=False)\n",
    "\t\tma_fasta_files.append(fasta_file)\n",
    "\t\tgen_ids.append(gen_id+2) # exclude the ancstors\n",
    "\t\tis_detect_ids.append(file_df.query('file_name == @g').iloc[0].IS_Detect_ID)\n",
    "\t\tC += ' -g ' + fasta_file\n",
    "\t\tanc_fasta = fasta_file\n",
    "\tprint(C)\n",
    "\t# Run the command\n",
    "\tproc = subprocess.Popen(C, stdout=subprocess.PIPE, stderr=subprocess.PIPE, shell=True)\n",
    "\tout, err = proc.communicate()\n",
    "\n",
    "\t# Write stdout and stderr to files\n",
    "\twith open(os.path.join(stderr_out, 'std_out_' + line + '.log'), 'w') as f:\n",
    "\t\tf.write(out.decode('utf-8'))\n",
    "\n",
    "\twith open(os.path.join(stderr_out, 'std_err_' + line + '.log'), 'w') as f:\n",
    "\t\tf.write(err.decode('utf-8'))"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 6,
   "metadata": {},
   "outputs": [],
   "source": [
    "if False: # just for updating vectors\n",
    "\tfacs_fasta_files = []\n",
    "\tma_fasta_files = []\n",
    "\tprefixes = []\n",
    "\tgen_ids = []\n",
    "\tis_detect_ids = []\n",
    "\t#lines_ = [line for line in lines if line not in ['L01-3', 'L01-4', 'L10-1', 'L10-2']]\n",
    "\tlines_ = lines\n",
    "\tfor line in lines_:\n",
    "\t\t# example python3 ../src/SyRI_IS/runSyRI_IS.py -i ../fasta/IS1.fasta -g ../fasta/MDS42_IS1.fa -g ../fasta/L01_Anc.fasta -g ../fasta/L01-2_G8.fasta -p L01-2 --replicationOrigin 3359187\n",
    "\t\tmds_fasta = os.path.join(fastadir, file_df.file_name[file_df.gen == 'Anc'][0])\n",
    "\t\t#parent_line = file_df[file_df.Prefix == line].iloc[0].ParentLine\n",
    "\t\tC = 'python3 ../src/SyRI_IS/runSyRI_IS.py -i ' + is_fasta \n",
    "\t\tC += ' -p ' + line + ' --replicationOrigin 3359187'\n",
    "\t\tC += ' --tmp ' + tmpdir + ' --out ' + exportdir\n",
    "\t\tC += ' --log ' + logdir\n",
    "\t\tC += ' --rerun '\n",
    "\t\tC += ' -g ' + mds_fasta\n",
    "\t\tanc_fasta = mds_fasta\n",
    "\t\tfor gen_id, g in enumerate(file_df[file_df.Prefix == line].file_name):\n",
    "\t\t\tfasta_file = os.path.join(fastadir, g)\n",
    "\t\t\tfasta_file = run_utils.get_full_contig(fasta_file, tmpdir=tmpdir, basefasta=anc_fasta, skip_if_exists=True)\n",
    "\t\t\tma_fasta_files.append(fasta_file)\n",
    "\t\t\tprefixes.append(line)\n",
    "\t\t\tgen_ids.append(gen_id+2) # exclude the ancstors\n",
    "\t\t\tis_detect_ids.append(file_df.query('file_name == @g').iloc[0].IS_Detect_ID)\n",
    "\t\t\tC += ' -g ' + fasta_file\n",
    "\t\t\tanc_fasta = fasta_file\n",
    "\t\tprint(C)"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "## merge plots"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 7,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "python3 ../src/SyRI_IS/connect_multiple_runs_plotsr.py exp/multiple_runs11/tmp exp/multiple_runs11/export --tmp exp/multiple_runs11/tmp/merged --out exp/multiple_runs11/export/merged -p L01-1 -p L01-2 -p L01-3 -p L01-4 -p L02-1 -p L02-2 -p L02-3 -p L02-4 -p L03-1 -p L03-2 -p L03-3 -p L03-4 -p L04-1 -p L04-2 -p L04-3 -p L04-4 -p L05-1 -p L05-2 -p L05-3 -p L05-4 -p L06-1 -p L06-2 -p L06-3 -p L06-4 -p L07-1 -p L07-2 -p L07-3 -p L07-4 -p L08-1 -p L08-2 -p L08-3 -p L08-4 -p L09-1 -p L09-2 -p L09-3 -p L09-4 -p L10-1 -p L10-2 -p L10-3 -p L10-4 -p L11-1 -p L11-2 -p L11-3 -p L11-4 -i MDS42_IS1 -i FACS -i MA_G08 -i MA_G20 --log exp/multiple_runs11/log/merged\n",
      "['--sr', 'exp/multiple_runs11/tmp/merged/syri/merged.0.syri.out', '--sr', 'exp/multiple_runs11/tmp/merged/syri/merged.1.syri.out', '--sr', 'exp/multiple_runs11/tmp/merged/syri/merged.2.syri.out', '--genomes', 'exp/multiple_runs11/tmp/merged/genome_info.txt', '--markers', 'exp/multiple_runs11/tmp/merged/merged_IS_tag.bed', '-s', '100', '-o', 'exp/multiple_runs11/export/merged/merged_plotsr.pdf']\n",
      "\u001b[0;49;32m2024-07-03 12:58:23,135 - Plotsr - INFO - Starting\u001b[0m\n",
      "\u001b[0;49;32m2024-07-03 12:58:37,096 - Plotsr - INFO - Plot exp/multiple_runs11/export/merged/merged_plotsr.pdf generated.\u001b[0m\n",
      "\u001b[0;49;32m2024-07-03 12:58:37,096 - Plotsr - INFO - Finished\u001b[0m\n",
      "['--sr', 'exp/multiple_runs11/tmp/merged/syri/merged.0.syri.local.out', '--sr', 'exp/multiple_runs11/tmp/merged/syri/merged.1.syri.local.out', '--sr', 'exp/multiple_runs11/tmp/merged/syri/merged.2.syri.local.out', '--genomes', 'exp/multiple_runs11/tmp/merged/genome_info.txt', '--markers', 'exp/multiple_runs11/tmp/merged/merged_IS_tag.bed', '-s', '100', '-o', 'exp/multiple_runs11/export/merged/merged-local_plotsr.pdf']\n",
      "\u001b[0;49;32m2024-07-03 12:58:37,097 - Plotsr - INFO - Starting\u001b[0m\n",
      "\u001b[0;49;32m2024-07-03 12:58:37,097 - Plotsr - INFO - Starting\u001b[0m\n",
      "\u001b[0;49;32m2024-07-03 12:58:51,116 - Plotsr - INFO - Plot exp/multiple_runs11/export/merged/merged-local_plotsr.pdf generated.\u001b[0m\n",
      "\u001b[0;49;32m2024-07-03 12:58:51,116 - Plotsr - INFO - Plot exp/multiple_runs11/export/merged/merged-local_plotsr.pdf generated.\u001b[0m\n",
      "\u001b[0;49;32m2024-07-03 12:58:51,116 - Plotsr - INFO - Finished\u001b[0m\n",
      "\u001b[0;49;32m2024-07-03 12:58:51,116 - Plotsr - INFO - Finished\u001b[0m\n"
     ]
    }
   ],
   "source": [
    "# concatenate the runs\n",
    "merged_out = os.path.join(exportdir, 'merged')\n",
    "merged_tmp = os.path.join(tmpdir, 'merged')\n",
    "\n",
    "#lines_ = [line for line in lines if line not in ['L05-1', 'L05-2', 'L05-3', 'L05-4']]\n",
    "lines_ = lines\n",
    "C = 'python3 ../src/SyRI_IS/connect_multiple_runs_plotsr.py ' + tmpdir + ' ' + exportdir + ' --tmp ' + merged_tmp + ' --out ' + merged_out\n",
    "C += ' -p ' + ' -p '.join(lines_)\n",
    "C += ' -i MDS42_IS1 -i FACS -i MA_G08 -i MA_G20'\n",
    "C += ' --log ' + os.path.join(logdir, 'merged')\n",
    "print(C)\n",
    "!{C}"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 8,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "python3 ../src/SyRI_IS/connect_multiple_runs_plotsr.py exp/multiple_runs11/tmp exp/multiple_runs11/export --tmp exp/multiple_runs11/tmp/merged/L01 --out exp/multiple_runs11/export/merged/L01 -p L01-1 -p L01-2 -p L01-3 -p L01-4 -i MDS42_IS1 -i FACS -i MA_G08 -i MA_G20 --log exp/multiple_runs11/log/merged/L01 --plotsr \" -W 3 -H 4 -f 10 --cfg ./20231001_plotsr.cfg --xlim 4490000 --noyaxislabel --noxaxislabel\"\n",
      "['--sr', 'exp/multiple_runs11/tmp/merged/L01/syri/merged.0.syri.out', '--sr', 'exp/multiple_runs11/tmp/merged/L01/syri/merged.1.syri.out', '--sr', 'exp/multiple_runs11/tmp/merged/L01/syri/merged.2.syri.out', '--genomes', 'exp/multiple_runs11/tmp/merged/L01/genome_info.txt', '--markers', 'exp/multiple_runs11/tmp/merged/L01/merged_IS_tag.bed', '-s', '100', '-o', 'exp/multiple_runs11/export/merged/L01/merged_plotsr.pdf', '-W', '3', '-H', '4', '-f', '10', '--cfg', './20231001_plotsr.cfg', '--xlim', '4490000', '--noyaxislabel', '--noxaxislabel']\n",
      "\u001b[0;49;32m2024-07-03 12:58:53,068 - Plotsr - INFO - Starting\u001b[0m\n",
      "readbasecfg - ERROR - # is not a valid config parameter. Using default value.\n",
      "0 -1 4490000 (0.0, 4014776.0)\n",
      "\u001b[0;49;32m2024-07-03 12:58:54,560 - Plotsr - INFO - Plot exp/multiple_runs11/export/merged/L01/merged_plotsr.pdf generated.\u001b[0m\n",
      "\u001b[0;49;32m2024-07-03 12:58:54,560 - Plotsr - INFO - Finished\u001b[0m\n",
      "['--sr', 'exp/multiple_runs11/tmp/merged/L01/syri/merged.0.syri.local.out', '--sr', 'exp/multiple_runs11/tmp/merged/L01/syri/merged.1.syri.local.out', '--sr', 'exp/multiple_runs11/tmp/merged/L01/syri/merged.2.syri.local.out', '--genomes', 'exp/multiple_runs11/tmp/merged/L01/genome_info.txt', '--markers', 'exp/multiple_runs11/tmp/merged/L01/merged_IS_tag.bed', '-s', '100', '-o', 'exp/multiple_runs11/export/merged/L01/merged-local_plotsr.pdf', '-W', '3', '-H', '4', '-f', '10', '--cfg', './20231001_plotsr.cfg', '--xlim', '4490000', '--noyaxislabel', '--noxaxislabel']\n",
      "\u001b[0;49;32m2024-07-03 12:58:54,561 - Plotsr - INFO - Starting\u001b[0m\n",
      "\u001b[0;49;32m2024-07-03 12:58:54,561 - Plotsr - INFO - Starting\u001b[0m\n",
      "readbasecfg - ERROR - # is not a valid config parameter. Using default value.\n",
      "0 -1 4490000 (0.0, 4014776.0)\n",
      "\u001b[0;49;32m2024-07-03 12:58:55,862 - Plotsr - INFO - Plot exp/multiple_runs11/export/merged/L01/merged-local_plotsr.pdf generated.\u001b[0m\n",
      "\u001b[0;49;32m2024-07-03 12:58:55,862 - Plotsr - INFO - Plot exp/multiple_runs11/export/merged/L01/merged-local_plotsr.pdf generated.\u001b[0m\n",
      "\u001b[0;49;32m2024-07-03 12:58:55,863 - Plotsr - INFO - Finished\u001b[0m\n",
      "\u001b[0;49;32m2024-07-03 12:58:55,863 - Plotsr - INFO - Finished\u001b[0m\n",
      "python3 ../src/SyRI_IS/connect_multiple_runs_plotsr.py exp/multiple_runs11/tmp exp/multiple_runs11/export --tmp exp/multiple_runs11/tmp/merged/L02 --out exp/multiple_runs11/export/merged/L02 -p L02-1 -p L02-2 -p L02-3 -p L02-4 -i MDS42_IS1 -i FACS -i MA_G08 -i MA_G20 --log exp/multiple_runs11/log/merged/L02 --plotsr \" -W 3 -H 4 -f 10 --cfg ./20231001_plotsr.cfg --xlim 4490000 --noyaxislabel --noxaxislabel\"\n",
      "['--sr', 'exp/multiple_runs11/tmp/merged/L02/syri/merged.0.syri.out', '--sr', 'exp/multiple_runs11/tmp/merged/L02/syri/merged.1.syri.out', '--sr', 'exp/multiple_runs11/tmp/merged/L02/syri/merged.2.syri.out', '--genomes', 'exp/multiple_runs11/tmp/merged/L02/genome_info.txt', '--markers', 'exp/multiple_runs11/tmp/merged/L02/merged_IS_tag.bed', '-s', '100', '-o', 'exp/multiple_runs11/export/merged/L02/merged_plotsr.pdf', '-W', '3', '-H', '4', '-f', '10', '--cfg', './20231001_plotsr.cfg', '--xlim', '4490000', '--noyaxislabel', '--noxaxislabel']\n",
      "\u001b[0;49;32m2024-07-03 12:58:57,494 - Plotsr - INFO - Starting\u001b[0m\n",
      "readbasecfg - ERROR - # is not a valid config parameter. Using default value.\n",
      "0 -1 4490000 (0.0, 4396104.0)\n",
      "\u001b[0;49;32m2024-07-03 12:58:59,032 - Plotsr - INFO - Plot exp/multiple_runs11/export/merged/L02/merged_plotsr.pdf generated.\u001b[0m\n",
      "\u001b[0;49;32m2024-07-03 12:58:59,032 - Plotsr - INFO - Finished\u001b[0m\n",
      "['--sr', 'exp/multiple_runs11/tmp/merged/L02/syri/merged.0.syri.local.out', '--sr', 'exp/multiple_runs11/tmp/merged/L02/syri/merged.1.syri.local.out', '--sr', 'exp/multiple_runs11/tmp/merged/L02/syri/merged.2.syri.local.out', '--genomes', 'exp/multiple_runs11/tmp/merged/L02/genome_info.txt', '--markers', 'exp/multiple_runs11/tmp/merged/L02/merged_IS_tag.bed', '-s', '100', '-o', 'exp/multiple_runs11/export/merged/L02/merged-local_plotsr.pdf', '-W', '3', '-H', '4', '-f', '10', '--cfg', './20231001_plotsr.cfg', '--xlim', '4490000', '--noyaxislabel', '--noxaxislabel']\n",
      "\u001b[0;49;32m2024-07-03 12:58:59,033 - Plotsr - INFO - Starting\u001b[0m\n",
      "\u001b[0;49;32m2024-07-03 12:58:59,033 - Plotsr - INFO - Starting\u001b[0m\n",
      "readbasecfg - ERROR - # is not a valid config parameter. Using default value.\n",
      "0 -1 4490000 (0.0, 4396104.0)\n",
      "\u001b[0;49;32m2024-07-03 12:59:00,379 - Plotsr - INFO - Plot exp/multiple_runs11/export/merged/L02/merged-local_plotsr.pdf generated.\u001b[0m\n",
      "\u001b[0;49;32m2024-07-03 12:59:00,379 - Plotsr - INFO - Plot exp/multiple_runs11/export/merged/L02/merged-local_plotsr.pdf generated.\u001b[0m\n",
      "\u001b[0;49;32m2024-07-03 12:59:00,379 - Plotsr - INFO - Finished\u001b[0m\n",
      "\u001b[0;49;32m2024-07-03 12:59:00,379 - Plotsr - INFO - Finished\u001b[0m\n",
      "python3 ../src/SyRI_IS/connect_multiple_runs_plotsr.py exp/multiple_runs11/tmp exp/multiple_runs11/export --tmp exp/multiple_runs11/tmp/merged/L03 --out exp/multiple_runs11/export/merged/L03 -p L03-1 -p L03-2 -p L03-3 -p L03-4 -i MDS42_IS1 -i FACS -i MA_G08 -i MA_G20 --log exp/multiple_runs11/log/merged/L03 --plotsr \" -W 3 -H 4 -f 10 --cfg ./20231001_plotsr.cfg --xlim 4490000 --noyaxislabel --noxaxislabel\"\n",
      "['--sr', 'exp/multiple_runs11/tmp/merged/L03/syri/merged.0.syri.out', '--sr', 'exp/multiple_runs11/tmp/merged/L03/syri/merged.1.syri.out', '--sr', 'exp/multiple_runs11/tmp/merged/L03/syri/merged.2.syri.out', '--genomes', 'exp/multiple_runs11/tmp/merged/L03/genome_info.txt', '--markers', 'exp/multiple_runs11/tmp/merged/L03/merged_IS_tag.bed', '-s', '100', '-o', 'exp/multiple_runs11/export/merged/L03/merged_plotsr.pdf', '-W', '3', '-H', '4', '-f', '10', '--cfg', './20231001_plotsr.cfg', '--xlim', '4490000', '--noyaxislabel', '--noxaxislabel']\n",
      "\u001b[0;49;32m2024-07-03 12:59:01,860 - Plotsr - INFO - Starting\u001b[0m\n",
      "readbasecfg - ERROR - # is not a valid config parameter. Using default value.\n",
      "0 -1 4490000 (0.0, 4041838.0)\n",
      "\u001b[0;49;32m2024-07-03 12:59:03,337 - Plotsr - INFO - Plot exp/multiple_runs11/export/merged/L03/merged_plotsr.pdf generated.\u001b[0m\n",
      "\u001b[0;49;32m2024-07-03 12:59:03,337 - Plotsr - INFO - Finished\u001b[0m\n",
      "['--sr', 'exp/multiple_runs11/tmp/merged/L03/syri/merged.0.syri.local.out', '--sr', 'exp/multiple_runs11/tmp/merged/L03/syri/merged.1.syri.local.out', '--sr', 'exp/multiple_runs11/tmp/merged/L03/syri/merged.2.syri.local.out', '--genomes', 'exp/multiple_runs11/tmp/merged/L03/genome_info.txt', '--markers', 'exp/multiple_runs11/tmp/merged/L03/merged_IS_tag.bed', '-s', '100', '-o', 'exp/multiple_runs11/export/merged/L03/merged-local_plotsr.pdf', '-W', '3', '-H', '4', '-f', '10', '--cfg', './20231001_plotsr.cfg', '--xlim', '4490000', '--noyaxislabel', '--noxaxislabel']\n",
      "\u001b[0;49;32m2024-07-03 12:59:03,338 - Plotsr - INFO - Starting\u001b[0m\n",
      "\u001b[0;49;32m2024-07-03 12:59:03,338 - Plotsr - INFO - Starting\u001b[0m\n",
      "readbasecfg - ERROR - # is not a valid config parameter. Using default value.\n",
      "0 -1 4490000 (0.0, 4041838.0)\n",
      "\u001b[0;49;32m2024-07-03 12:59:04,614 - Plotsr - INFO - Plot exp/multiple_runs11/export/merged/L03/merged-local_plotsr.pdf generated.\u001b[0m\n",
      "\u001b[0;49;32m2024-07-03 12:59:04,614 - Plotsr - INFO - Plot exp/multiple_runs11/export/merged/L03/merged-local_plotsr.pdf generated.\u001b[0m\n",
      "\u001b[0;49;32m2024-07-03 12:59:04,614 - Plotsr - INFO - Finished\u001b[0m\n",
      "\u001b[0;49;32m2024-07-03 12:59:04,614 - Plotsr - INFO - Finished\u001b[0m\n",
      "python3 ../src/SyRI_IS/connect_multiple_runs_plotsr.py exp/multiple_runs11/tmp exp/multiple_runs11/export --tmp exp/multiple_runs11/tmp/merged/L04 --out exp/multiple_runs11/export/merged/L04 -p L04-1 -p L04-2 -p L04-3 -p L04-4 -i MDS42_IS1 -i FACS -i MA_G08 -i MA_G20 --log exp/multiple_runs11/log/merged/L04 --plotsr \" -W 3 -H 4 -f 10 --cfg ./20231001_plotsr.cfg --xlim 4490000 --noyaxislabel --noxaxislabel\"\n",
      "['--sr', 'exp/multiple_runs11/tmp/merged/L04/syri/merged.0.syri.out', '--sr', 'exp/multiple_runs11/tmp/merged/L04/syri/merged.1.syri.out', '--sr', 'exp/multiple_runs11/tmp/merged/L04/syri/merged.2.syri.out', '--genomes', 'exp/multiple_runs11/tmp/merged/L04/genome_info.txt', '--markers', 'exp/multiple_runs11/tmp/merged/L04/merged_IS_tag.bed', '-s', '100', '-o', 'exp/multiple_runs11/export/merged/L04/merged_plotsr.pdf', '-W', '3', '-H', '4', '-f', '10', '--cfg', './20231001_plotsr.cfg', '--xlim', '4490000', '--noyaxislabel', '--noxaxislabel']\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\u001b[0;49;32m2024-07-03 12:59:06,138 - Plotsr - INFO - Starting\u001b[0m\n",
      "readbasecfg - ERROR - # is not a valid config parameter. Using default value.\n",
      "0 -1 4490000 (0.0, 4313004.0)\n",
      "\u001b[0;49;32m2024-07-03 12:59:07,632 - Plotsr - INFO - Plot exp/multiple_runs11/export/merged/L04/merged_plotsr.pdf generated.\u001b[0m\n",
      "\u001b[0;49;32m2024-07-03 12:59:07,632 - Plotsr - INFO - Finished\u001b[0m\n",
      "['--sr', 'exp/multiple_runs11/tmp/merged/L04/syri/merged.0.syri.local.out', '--sr', 'exp/multiple_runs11/tmp/merged/L04/syri/merged.1.syri.local.out', '--sr', 'exp/multiple_runs11/tmp/merged/L04/syri/merged.2.syri.local.out', '--genomes', 'exp/multiple_runs11/tmp/merged/L04/genome_info.txt', '--markers', 'exp/multiple_runs11/tmp/merged/L04/merged_IS_tag.bed', '-s', '100', '-o', 'exp/multiple_runs11/export/merged/L04/merged-local_plotsr.pdf', '-W', '3', '-H', '4', '-f', '10', '--cfg', './20231001_plotsr.cfg', '--xlim', '4490000', '--noyaxislabel', '--noxaxislabel']\n",
      "\u001b[0;49;32m2024-07-03 12:59:07,633 - Plotsr - INFO - Starting\u001b[0m\n",
      "\u001b[0;49;32m2024-07-03 12:59:07,633 - Plotsr - INFO - Starting\u001b[0m\n",
      "readbasecfg - ERROR - # is not a valid config parameter. Using default value.\n",
      "0 -1 4490000 (0.0, 4313004.0)\n",
      "\u001b[0;49;32m2024-07-03 12:59:08,942 - Plotsr - INFO - Plot exp/multiple_runs11/export/merged/L04/merged-local_plotsr.pdf generated.\u001b[0m\n",
      "\u001b[0;49;32m2024-07-03 12:59:08,942 - Plotsr - INFO - Plot exp/multiple_runs11/export/merged/L04/merged-local_plotsr.pdf generated.\u001b[0m\n",
      "\u001b[0;49;32m2024-07-03 12:59:08,942 - Plotsr - INFO - Finished\u001b[0m\n",
      "\u001b[0;49;32m2024-07-03 12:59:08,942 - Plotsr - INFO - Finished\u001b[0m\n",
      "python3 ../src/SyRI_IS/connect_multiple_runs_plotsr.py exp/multiple_runs11/tmp exp/multiple_runs11/export --tmp exp/multiple_runs11/tmp/merged/L05 --out exp/multiple_runs11/export/merged/L05 -p L05-1 -p L05-2 -p L05-3 -p L05-4 -i MDS42_IS1 -i FACS -i MA_G08 -i MA_G20 --log exp/multiple_runs11/log/merged/L05 --plotsr \" -W 3 -H 4 -f 10 --cfg ./20231001_plotsr.cfg --xlim 4490000 --noyaxislabel --noxaxislabel\"\n",
      "['--sr', 'exp/multiple_runs11/tmp/merged/L05/syri/merged.0.syri.out', '--sr', 'exp/multiple_runs11/tmp/merged/L05/syri/merged.1.syri.out', '--sr', 'exp/multiple_runs11/tmp/merged/L05/syri/merged.2.syri.out', '--genomes', 'exp/multiple_runs11/tmp/merged/L05/genome_info.txt', '--markers', 'exp/multiple_runs11/tmp/merged/L05/merged_IS_tag.bed', '-s', '100', '-o', 'exp/multiple_runs11/export/merged/L05/merged_plotsr.pdf', '-W', '3', '-H', '4', '-f', '10', '--cfg', './20231001_plotsr.cfg', '--xlim', '4490000', '--noyaxislabel', '--noxaxislabel']\n",
      "\u001b[0;49;32m2024-07-03 12:59:10,454 - Plotsr - INFO - Starting\u001b[0m\n",
      "readbasecfg - ERROR - # is not a valid config parameter. Using default value.\n",
      "0 -1 4490000 (0.0, 3980425.0)\n",
      "\u001b[0;49;32m2024-07-03 12:59:11,931 - Plotsr - INFO - Plot exp/multiple_runs11/export/merged/L05/merged_plotsr.pdf generated.\u001b[0m\n",
      "\u001b[0;49;32m2024-07-03 12:59:11,931 - Plotsr - INFO - Finished\u001b[0m\n",
      "['--sr', 'exp/multiple_runs11/tmp/merged/L05/syri/merged.0.syri.local.out', '--sr', 'exp/multiple_runs11/tmp/merged/L05/syri/merged.1.syri.local.out', '--sr', 'exp/multiple_runs11/tmp/merged/L05/syri/merged.2.syri.local.out', '--genomes', 'exp/multiple_runs11/tmp/merged/L05/genome_info.txt', '--markers', 'exp/multiple_runs11/tmp/merged/L05/merged_IS_tag.bed', '-s', '100', '-o', 'exp/multiple_runs11/export/merged/L05/merged-local_plotsr.pdf', '-W', '3', '-H', '4', '-f', '10', '--cfg', './20231001_plotsr.cfg', '--xlim', '4490000', '--noyaxislabel', '--noxaxislabel']\n",
      "\u001b[0;49;32m2024-07-03 12:59:11,932 - Plotsr - INFO - Starting\u001b[0m\n",
      "\u001b[0;49;32m2024-07-03 12:59:11,932 - Plotsr - INFO - Starting\u001b[0m\n",
      "readbasecfg - ERROR - # is not a valid config parameter. Using default value.\n",
      "0 -1 4490000 (0.0, 3980425.0)\n",
      "\u001b[0;49;32m2024-07-03 12:59:13,226 - Plotsr - INFO - Plot exp/multiple_runs11/export/merged/L05/merged-local_plotsr.pdf generated.\u001b[0m\n",
      "\u001b[0;49;32m2024-07-03 12:59:13,226 - Plotsr - INFO - Plot exp/multiple_runs11/export/merged/L05/merged-local_plotsr.pdf generated.\u001b[0m\n",
      "\u001b[0;49;32m2024-07-03 12:59:13,226 - Plotsr - INFO - Finished\u001b[0m\n",
      "\u001b[0;49;32m2024-07-03 12:59:13,226 - Plotsr - INFO - Finished\u001b[0m\n",
      "python3 ../src/SyRI_IS/connect_multiple_runs_plotsr.py exp/multiple_runs11/tmp exp/multiple_runs11/export --tmp exp/multiple_runs11/tmp/merged/L06 --out exp/multiple_runs11/export/merged/L06 -p L06-1 -p L06-2 -p L06-3 -p L06-4 -i MDS42_IS1 -i FACS -i MA_G08 -i MA_G20 --log exp/multiple_runs11/log/merged/L06 --plotsr \" -W 3 -H 4 -f 10 --cfg ./20231001_plotsr.cfg --xlim 4490000 --noyaxislabel --noxaxislabel\"\n",
      "['--sr', 'exp/multiple_runs11/tmp/merged/L06/syri/merged.0.syri.out', '--sr', 'exp/multiple_runs11/tmp/merged/L06/syri/merged.1.syri.out', '--sr', 'exp/multiple_runs11/tmp/merged/L06/syri/merged.2.syri.out', '--genomes', 'exp/multiple_runs11/tmp/merged/L06/genome_info.txt', '--markers', 'exp/multiple_runs11/tmp/merged/L06/merged_IS_tag.bed', '-s', '100', '-o', 'exp/multiple_runs11/export/merged/L06/merged_plotsr.pdf', '-W', '3', '-H', '4', '-f', '10', '--cfg', './20231001_plotsr.cfg', '--xlim', '4490000', '--noyaxislabel', '--noxaxislabel']\n",
      "\u001b[0;49;32m2024-07-03 12:59:14,725 - Plotsr - INFO - Starting\u001b[0m\n",
      "readbasecfg - ERROR - # is not a valid config parameter. Using default value.\n",
      "0 -1 4490000 (0.0, 4145548.0)\n",
      "\u001b[0;49;32m2024-07-03 12:59:16,197 - Plotsr - INFO - Plot exp/multiple_runs11/export/merged/L06/merged_plotsr.pdf generated.\u001b[0m\n",
      "\u001b[0;49;32m2024-07-03 12:59:16,197 - Plotsr - INFO - Finished\u001b[0m\n",
      "['--sr', 'exp/multiple_runs11/tmp/merged/L06/syri/merged.0.syri.local.out', '--sr', 'exp/multiple_runs11/tmp/merged/L06/syri/merged.1.syri.local.out', '--sr', 'exp/multiple_runs11/tmp/merged/L06/syri/merged.2.syri.local.out', '--genomes', 'exp/multiple_runs11/tmp/merged/L06/genome_info.txt', '--markers', 'exp/multiple_runs11/tmp/merged/L06/merged_IS_tag.bed', '-s', '100', '-o', 'exp/multiple_runs11/export/merged/L06/merged-local_plotsr.pdf', '-W', '3', '-H', '4', '-f', '10', '--cfg', './20231001_plotsr.cfg', '--xlim', '4490000', '--noyaxislabel', '--noxaxislabel']\n",
      "\u001b[0;49;32m2024-07-03 12:59:16,198 - Plotsr - INFO - Starting\u001b[0m\n",
      "\u001b[0;49;32m2024-07-03 12:59:16,198 - Plotsr - INFO - Starting\u001b[0m\n",
      "readbasecfg - ERROR - # is not a valid config parameter. Using default value.\n",
      "0 -1 4490000 (0.0, 4145548.0)\n",
      "\u001b[0;49;32m2024-07-03 12:59:17,480 - Plotsr - INFO - Plot exp/multiple_runs11/export/merged/L06/merged-local_plotsr.pdf generated.\u001b[0m\n",
      "\u001b[0;49;32m2024-07-03 12:59:17,480 - Plotsr - INFO - Plot exp/multiple_runs11/export/merged/L06/merged-local_plotsr.pdf generated.\u001b[0m\n",
      "\u001b[0;49;32m2024-07-03 12:59:17,481 - Plotsr - INFO - Finished\u001b[0m\n",
      "\u001b[0;49;32m2024-07-03 12:59:17,481 - Plotsr - INFO - Finished\u001b[0m\n",
      "python3 ../src/SyRI_IS/connect_multiple_runs_plotsr.py exp/multiple_runs11/tmp exp/multiple_runs11/export --tmp exp/multiple_runs11/tmp/merged/L07 --out exp/multiple_runs11/export/merged/L07 -p L07-1 -p L07-2 -p L07-3 -p L07-4 -i MDS42_IS1 -i FACS -i MA_G08 -i MA_G20 --log exp/multiple_runs11/log/merged/L07 --plotsr \" -W 3 -H 4 -f 10 --cfg ./20231001_plotsr.cfg --xlim 4490000 --noyaxislabel --noxaxislabel\"\n",
      "['--sr', 'exp/multiple_runs11/tmp/merged/L07/syri/merged.0.syri.out', '--sr', 'exp/multiple_runs11/tmp/merged/L07/syri/merged.1.syri.out', '--sr', 'exp/multiple_runs11/tmp/merged/L07/syri/merged.2.syri.out', '--genomes', 'exp/multiple_runs11/tmp/merged/L07/genome_info.txt', '--markers', 'exp/multiple_runs11/tmp/merged/L07/merged_IS_tag.bed', '-s', '100', '-o', 'exp/multiple_runs11/export/merged/L07/merged_plotsr.pdf', '-W', '3', '-H', '4', '-f', '10', '--cfg', './20231001_plotsr.cfg', '--xlim', '4490000', '--noyaxislabel', '--noxaxislabel']\n",
      "\u001b[0;49;32m2024-07-03 12:59:18,939 - Plotsr - INFO - Starting\u001b[0m\n",
      "readbasecfg - ERROR - # is not a valid config parameter. Using default value.\n",
      "0 -1 4490000 (0.0, 3998098.0)\n",
      "\u001b[0;49;32m2024-07-03 12:59:20,399 - Plotsr - INFO - Plot exp/multiple_runs11/export/merged/L07/merged_plotsr.pdf generated.\u001b[0m\n",
      "\u001b[0;49;32m2024-07-03 12:59:20,400 - Plotsr - INFO - Finished\u001b[0m\n",
      "['--sr', 'exp/multiple_runs11/tmp/merged/L07/syri/merged.0.syri.local.out', '--sr', 'exp/multiple_runs11/tmp/merged/L07/syri/merged.1.syri.local.out', '--sr', 'exp/multiple_runs11/tmp/merged/L07/syri/merged.2.syri.local.out', '--genomes', 'exp/multiple_runs11/tmp/merged/L07/genome_info.txt', '--markers', 'exp/multiple_runs11/tmp/merged/L07/merged_IS_tag.bed', '-s', '100', '-o', 'exp/multiple_runs11/export/merged/L07/merged-local_plotsr.pdf', '-W', '3', '-H', '4', '-f', '10', '--cfg', './20231001_plotsr.cfg', '--xlim', '4490000', '--noyaxislabel', '--noxaxislabel']\n",
      "\u001b[0;49;32m2024-07-03 12:59:20,401 - Plotsr - INFO - Starting\u001b[0m\n",
      "\u001b[0;49;32m2024-07-03 12:59:20,401 - Plotsr - INFO - Starting\u001b[0m\n",
      "readbasecfg - ERROR - # is not a valid config parameter. Using default value.\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "0 -1 4490000 (0.0, 3998098.0)\n",
      "\u001b[0;49;32m2024-07-03 12:59:21,662 - Plotsr - INFO - Plot exp/multiple_runs11/export/merged/L07/merged-local_plotsr.pdf generated.\u001b[0m\n",
      "\u001b[0;49;32m2024-07-03 12:59:21,662 - Plotsr - INFO - Plot exp/multiple_runs11/export/merged/L07/merged-local_plotsr.pdf generated.\u001b[0m\n",
      "\u001b[0;49;32m2024-07-03 12:59:21,662 - Plotsr - INFO - Finished\u001b[0m\n",
      "\u001b[0;49;32m2024-07-03 12:59:21,662 - Plotsr - INFO - Finished\u001b[0m\n",
      "python3 ../src/SyRI_IS/connect_multiple_runs_plotsr.py exp/multiple_runs11/tmp exp/multiple_runs11/export --tmp exp/multiple_runs11/tmp/merged/L08 --out exp/multiple_runs11/export/merged/L08 -p L08-1 -p L08-2 -p L08-3 -p L08-4 -i MDS42_IS1 -i FACS -i MA_G08 -i MA_G20 --log exp/multiple_runs11/log/merged/L08 --plotsr \" -W 3 -H 4 -f 10 --cfg ./20231001_plotsr.cfg --xlim 4490000 --noyaxislabel --noxaxislabel\"\n",
      "['--sr', 'exp/multiple_runs11/tmp/merged/L08/syri/merged.0.syri.out', '--sr', 'exp/multiple_runs11/tmp/merged/L08/syri/merged.1.syri.out', '--sr', 'exp/multiple_runs11/tmp/merged/L08/syri/merged.2.syri.out', '--genomes', 'exp/multiple_runs11/tmp/merged/L08/genome_info.txt', '--markers', 'exp/multiple_runs11/tmp/merged/L08/merged_IS_tag.bed', '-s', '100', '-o', 'exp/multiple_runs11/export/merged/L08/merged_plotsr.pdf', '-W', '3', '-H', '4', '-f', '10', '--cfg', './20231001_plotsr.cfg', '--xlim', '4490000', '--noyaxislabel', '--noxaxislabel']\n",
      "\u001b[0;49;32m2024-07-03 12:59:23,066 - Plotsr - INFO - Starting\u001b[0m\n",
      "readbasecfg - ERROR - # is not a valid config parameter. Using default value.\n",
      "0 -1 4490000 (0.0, 4001201.0)\n",
      "\u001b[0;49;32m2024-07-03 12:59:24,480 - Plotsr - INFO - Plot exp/multiple_runs11/export/merged/L08/merged_plotsr.pdf generated.\u001b[0m\n",
      "\u001b[0;49;32m2024-07-03 12:59:24,480 - Plotsr - INFO - Finished\u001b[0m\n",
      "['--sr', 'exp/multiple_runs11/tmp/merged/L08/syri/merged.0.syri.local.out', '--sr', 'exp/multiple_runs11/tmp/merged/L08/syri/merged.1.syri.local.out', '--sr', 'exp/multiple_runs11/tmp/merged/L08/syri/merged.2.syri.local.out', '--genomes', 'exp/multiple_runs11/tmp/merged/L08/genome_info.txt', '--markers', 'exp/multiple_runs11/tmp/merged/L08/merged_IS_tag.bed', '-s', '100', '-o', 'exp/multiple_runs11/export/merged/L08/merged-local_plotsr.pdf', '-W', '3', '-H', '4', '-f', '10', '--cfg', './20231001_plotsr.cfg', '--xlim', '4490000', '--noyaxislabel', '--noxaxislabel']\n",
      "\u001b[0;49;32m2024-07-03 12:59:24,481 - Plotsr - INFO - Starting\u001b[0m\n",
      "\u001b[0;49;32m2024-07-03 12:59:24,481 - Plotsr - INFO - Starting\u001b[0m\n",
      "readbasecfg - ERROR - # is not a valid config parameter. Using default value.\n",
      "0 -1 4490000 (0.0, 4001201.0)\n",
      "\u001b[0;49;32m2024-07-03 12:59:25,712 - Plotsr - INFO - Plot exp/multiple_runs11/export/merged/L08/merged-local_plotsr.pdf generated.\u001b[0m\n",
      "\u001b[0;49;32m2024-07-03 12:59:25,712 - Plotsr - INFO - Plot exp/multiple_runs11/export/merged/L08/merged-local_plotsr.pdf generated.\u001b[0m\n",
      "\u001b[0;49;32m2024-07-03 12:59:25,712 - Plotsr - INFO - Finished\u001b[0m\n",
      "\u001b[0;49;32m2024-07-03 12:59:25,712 - Plotsr - INFO - Finished\u001b[0m\n",
      "python3 ../src/SyRI_IS/connect_multiple_runs_plotsr.py exp/multiple_runs11/tmp exp/multiple_runs11/export --tmp exp/multiple_runs11/tmp/merged/L09 --out exp/multiple_runs11/export/merged/L09 -p L09-1 -p L09-2 -p L09-3 -p L09-4 -i MDS42_IS1 -i FACS -i MA_G08 -i MA_G20 --log exp/multiple_runs11/log/merged/L09 --plotsr \" -W 3 -H 4 -f 10 --cfg ./20231001_plotsr.cfg --xlim 4490000 --noyaxislabel --noxaxislabel\"\n",
      "['--sr', 'exp/multiple_runs11/tmp/merged/L09/syri/merged.0.syri.out', '--sr', 'exp/multiple_runs11/tmp/merged/L09/syri/merged.1.syri.out', '--sr', 'exp/multiple_runs11/tmp/merged/L09/syri/merged.2.syri.out', '--genomes', 'exp/multiple_runs11/tmp/merged/L09/genome_info.txt', '--markers', 'exp/multiple_runs11/tmp/merged/L09/merged_IS_tag.bed', '-s', '100', '-o', 'exp/multiple_runs11/export/merged/L09/merged_plotsr.pdf', '-W', '3', '-H', '4', '-f', '10', '--cfg', './20231001_plotsr.cfg', '--xlim', '4490000', '--noyaxislabel', '--noxaxislabel']\n",
      "\u001b[0;49;32m2024-07-03 12:59:27,224 - Plotsr - INFO - Starting\u001b[0m\n",
      "readbasecfg - ERROR - # is not a valid config parameter. Using default value.\n",
      "0 -1 4490000 (0.0, 4013673.0)\n",
      "\u001b[0;49;32m2024-07-03 12:59:28,704 - Plotsr - INFO - Plot exp/multiple_runs11/export/merged/L09/merged_plotsr.pdf generated.\u001b[0m\n",
      "\u001b[0;49;32m2024-07-03 12:59:28,704 - Plotsr - INFO - Finished\u001b[0m\n",
      "['--sr', 'exp/multiple_runs11/tmp/merged/L09/syri/merged.0.syri.local.out', '--sr', 'exp/multiple_runs11/tmp/merged/L09/syri/merged.1.syri.local.out', '--sr', 'exp/multiple_runs11/tmp/merged/L09/syri/merged.2.syri.local.out', '--genomes', 'exp/multiple_runs11/tmp/merged/L09/genome_info.txt', '--markers', 'exp/multiple_runs11/tmp/merged/L09/merged_IS_tag.bed', '-s', '100', '-o', 'exp/multiple_runs11/export/merged/L09/merged-local_plotsr.pdf', '-W', '3', '-H', '4', '-f', '10', '--cfg', './20231001_plotsr.cfg', '--xlim', '4490000', '--noyaxislabel', '--noxaxislabel']\n",
      "\u001b[0;49;32m2024-07-03 12:59:28,705 - Plotsr - INFO - Starting\u001b[0m\n",
      "\u001b[0;49;32m2024-07-03 12:59:28,705 - Plotsr - INFO - Starting\u001b[0m\n",
      "readbasecfg - ERROR - # is not a valid config parameter. Using default value.\n",
      "0 -1 4490000 (0.0, 4013673.0)\n",
      "\u001b[0;49;32m2024-07-03 12:59:29,978 - Plotsr - INFO - Plot exp/multiple_runs11/export/merged/L09/merged-local_plotsr.pdf generated.\u001b[0m\n",
      "\u001b[0;49;32m2024-07-03 12:59:29,978 - Plotsr - INFO - Plot exp/multiple_runs11/export/merged/L09/merged-local_plotsr.pdf generated.\u001b[0m\n",
      "\u001b[0;49;32m2024-07-03 12:59:29,978 - Plotsr - INFO - Finished\u001b[0m\n",
      "\u001b[0;49;32m2024-07-03 12:59:29,978 - Plotsr - INFO - Finished\u001b[0m\n",
      "python3 ../src/SyRI_IS/connect_multiple_runs_plotsr.py exp/multiple_runs11/tmp exp/multiple_runs11/export --tmp exp/multiple_runs11/tmp/merged/L10 --out exp/multiple_runs11/export/merged/L10 -p L10-1 -p L10-2 -p L10-3 -p L10-4 -i MDS42_IS1 -i FACS -i MA_G08 -i MA_G20 --log exp/multiple_runs11/log/merged/L10 --plotsr \" -W 3 -H 4 -f 10 --cfg ./20231001_plotsr.cfg --xlim 4490000 --noyaxislabel --noxaxislabel\"\n",
      "['--sr', 'exp/multiple_runs11/tmp/merged/L10/syri/merged.0.syri.out', '--sr', 'exp/multiple_runs11/tmp/merged/L10/syri/merged.1.syri.out', '--sr', 'exp/multiple_runs11/tmp/merged/L10/syri/merged.2.syri.out', '--genomes', 'exp/multiple_runs11/tmp/merged/L10/genome_info.txt', '--markers', 'exp/multiple_runs11/tmp/merged/L10/merged_IS_tag.bed', '-s', '100', '-o', 'exp/multiple_runs11/export/merged/L10/merged_plotsr.pdf', '-W', '3', '-H', '4', '-f', '10', '--cfg', './20231001_plotsr.cfg', '--xlim', '4490000', '--noyaxislabel', '--noxaxislabel']\n",
      "\u001b[0;49;32m2024-07-03 12:59:31,552 - Plotsr - INFO - Starting\u001b[0m\n",
      "readbasecfg - ERROR - # is not a valid config parameter. Using default value.\n",
      "0 -1 4490000 (0.0, 4074199.0)\n",
      "\u001b[0;49;32m2024-07-03 12:59:33,059 - Plotsr - INFO - Plot exp/multiple_runs11/export/merged/L10/merged_plotsr.pdf generated.\u001b[0m\n",
      "\u001b[0;49;32m2024-07-03 12:59:33,059 - Plotsr - INFO - Finished\u001b[0m\n",
      "['--sr', 'exp/multiple_runs11/tmp/merged/L10/syri/merged.0.syri.local.out', '--sr', 'exp/multiple_runs11/tmp/merged/L10/syri/merged.1.syri.local.out', '--sr', 'exp/multiple_runs11/tmp/merged/L10/syri/merged.2.syri.local.out', '--genomes', 'exp/multiple_runs11/tmp/merged/L10/genome_info.txt', '--markers', 'exp/multiple_runs11/tmp/merged/L10/merged_IS_tag.bed', '-s', '100', '-o', 'exp/multiple_runs11/export/merged/L10/merged-local_plotsr.pdf', '-W', '3', '-H', '4', '-f', '10', '--cfg', './20231001_plotsr.cfg', '--xlim', '4490000', '--noyaxislabel', '--noxaxislabel']\n",
      "\u001b[0;49;32m2024-07-03 12:59:33,060 - Plotsr - INFO - Starting\u001b[0m\n",
      "\u001b[0;49;32m2024-07-03 12:59:33,060 - Plotsr - INFO - Starting\u001b[0m\n",
      "readbasecfg - ERROR - # is not a valid config parameter. Using default value.\n",
      "0 -1 4490000 (0.0, 4074199.0)\n",
      "\u001b[0;49;32m2024-07-03 12:59:34,370 - Plotsr - INFO - Plot exp/multiple_runs11/export/merged/L10/merged-local_plotsr.pdf generated.\u001b[0m\n",
      "\u001b[0;49;32m2024-07-03 12:59:34,370 - Plotsr - INFO - Plot exp/multiple_runs11/export/merged/L10/merged-local_plotsr.pdf generated.\u001b[0m\n",
      "\u001b[0;49;32m2024-07-03 12:59:34,371 - Plotsr - INFO - Finished\u001b[0m\n",
      "\u001b[0;49;32m2024-07-03 12:59:34,371 - Plotsr - INFO - Finished\u001b[0m\n",
      "python3 ../src/SyRI_IS/connect_multiple_runs_plotsr.py exp/multiple_runs11/tmp exp/multiple_runs11/export --tmp exp/multiple_runs11/tmp/merged/L11 --out exp/multiple_runs11/export/merged/L11 -p L11-1 -p L11-2 -p L11-3 -p L11-4 -i MDS42_IS1 -i FACS -i MA_G08 -i MA_G20 --log exp/multiple_runs11/log/merged/L11 --plotsr \" -W 3 -H 4 -f 10 --cfg ./20231001_plotsr.cfg --xlim 4490000 --noyaxislabel --noxaxislabel\"\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "['--sr', 'exp/multiple_runs11/tmp/merged/L11/syri/merged.0.syri.out', '--sr', 'exp/multiple_runs11/tmp/merged/L11/syri/merged.1.syri.out', '--sr', 'exp/multiple_runs11/tmp/merged/L11/syri/merged.2.syri.out', '--genomes', 'exp/multiple_runs11/tmp/merged/L11/genome_info.txt', '--markers', 'exp/multiple_runs11/tmp/merged/L11/merged_IS_tag.bed', '-s', '100', '-o', 'exp/multiple_runs11/export/merged/L11/merged_plotsr.pdf', '-W', '3', '-H', '4', '-f', '10', '--cfg', './20231001_plotsr.cfg', '--xlim', '4490000', '--noyaxislabel', '--noxaxislabel']\n",
      "\u001b[0;49;32m2024-07-03 12:59:35,926 - Plotsr - INFO - Starting\u001b[0m\n",
      "readbasecfg - ERROR - # is not a valid config parameter. Using default value.\n",
      "0 -1 4490000 (0.0, 4248613.0)\n",
      "\u001b[0;49;32m2024-07-03 12:59:37,418 - Plotsr - INFO - Plot exp/multiple_runs11/export/merged/L11/merged_plotsr.pdf generated.\u001b[0m\n",
      "\u001b[0;49;32m2024-07-03 12:59:37,418 - Plotsr - INFO - Finished\u001b[0m\n",
      "['--sr', 'exp/multiple_runs11/tmp/merged/L11/syri/merged.0.syri.local.out', '--sr', 'exp/multiple_runs11/tmp/merged/L11/syri/merged.1.syri.local.out', '--sr', 'exp/multiple_runs11/tmp/merged/L11/syri/merged.2.syri.local.out', '--genomes', 'exp/multiple_runs11/tmp/merged/L11/genome_info.txt', '--markers', 'exp/multiple_runs11/tmp/merged/L11/merged_IS_tag.bed', '-s', '100', '-o', 'exp/multiple_runs11/export/merged/L11/merged-local_plotsr.pdf', '-W', '3', '-H', '4', '-f', '10', '--cfg', './20231001_plotsr.cfg', '--xlim', '4490000', '--noyaxislabel', '--noxaxislabel']\n",
      "\u001b[0;49;32m2024-07-03 12:59:37,419 - Plotsr - INFO - Starting\u001b[0m\n",
      "\u001b[0;49;32m2024-07-03 12:59:37,419 - Plotsr - INFO - Starting\u001b[0m\n",
      "readbasecfg - ERROR - # is not a valid config parameter. Using default value.\n",
      "0 -1 4490000 (0.0, 4248613.0)\n",
      "\u001b[0;49;32m2024-07-03 12:59:38,697 - Plotsr - INFO - Plot exp/multiple_runs11/export/merged/L11/merged-local_plotsr.pdf generated.\u001b[0m\n",
      "\u001b[0;49;32m2024-07-03 12:59:38,697 - Plotsr - INFO - Plot exp/multiple_runs11/export/merged/L11/merged-local_plotsr.pdf generated.\u001b[0m\n",
      "\u001b[0;49;32m2024-07-03 12:59:38,698 - Plotsr - INFO - Finished\u001b[0m\n",
      "\u001b[0;49;32m2024-07-03 12:59:38,698 - Plotsr - INFO - Finished\u001b[0m\n"
     ]
    }
   ],
   "source": [
    "# ouput svs for supplemntary figure\n",
    "for parent in ['L01', 'L02', 'L03', 'L04', 'L05', 'L06', 'L07', 'L08', 'L09', 'L10', 'L11']:\n",
    "\tlines__ = [line for line in lines if line.startswith(parent)]\n",
    "\tC = 'python3 ../src/SyRI_IS/connect_multiple_runs_plotsr.py '+ tmpdir + ' ' + exportdir \n",
    "\tC += ' --tmp ' + os.path.join(merged_tmp, parent) + ' --out ' + os.path.join(merged_out, parent) \n",
    "\tC += ' -p ' + ' -p '.join(lines__)\n",
    "\tC += ' -i MDS42_IS1 -i FACS -i MA_G08 -i MA_G20'\n",
    "\tC += ' --log ' + os.path.join(logdir, 'merged', parent)\n",
    "\tC += ' --plotsr \" -W 3 -H 4 -f 10 --cfg ./20231001_plotsr.cfg --xlim 4490000 --noyaxislabel --noxaxislabel\"'\n",
    "\tprint(C)\n",
    "\t!{C}"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 9,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "pdfunite exp/multiple_runs11/export/merged/L01/merged-local_plotsr.pdf exp/multiple_runs11/export/merged/L02/merged-local_plotsr.pdf exp/multiple_runs11/export/merged/L03/merged-local_plotsr.pdf exp/multiple_runs11/export/merged/L04/merged-local_plotsr.pdf exp/multiple_runs11/export/merged/L05/merged-local_plotsr.pdf exp/multiple_runs11/export/merged/L06/merged-local_plotsr.pdf exp/multiple_runs11/export/merged/L07/merged-local_plotsr.pdf exp/multiple_runs11/export/merged/L08/merged-local_plotsr.pdf exp/multiple_runs11/export/merged/L09/merged-local_plotsr.pdf exp/multiple_runs11/export/merged/L10/merged-local_plotsr.pdf exp/multiple_runs11/export/merged/L11/merged-local_plotsr.pdf  ./combine_pdf/combined_plotsr.pdf\n",
      "This is pdfTeX, Version 3.14159265-2.6-1.40.18 (TeX Live 2017/Debian) (preloaded format=pdflatex)\n",
      " restricted \\write18 enabled.\n",
      "entering extended mode\n",
      "(./latex_script_to_combine_plotsr_pdfs.tex\n",
      "LaTeX2e <2017-04-15>\n",
      "Babel <3.18> and hyphenation patterns for 3 language(s) loaded.\n",
      "(/usr/share/texlive/texmf-dist/tex/latex/base/article.cls\n",
      "Document Class: article 2014/09/29 v1.4h Standard LaTeX document class\n",
      "(/usr/share/texlive/texmf-dist/tex/latex/base/size10.clo))\n",
      "(/usr/share/texlive/texmf-dist/tex/latex/graphics/color.sty\n",
      "(/usr/share/texlive/texmf-dist/tex/latex/graphics-cfg/color.cfg)\n",
      "(/usr/share/texlive/texmf-dist/tex/latex/graphics-def/pdftex.def))\n",
      "(/usr/share/texlive/texmf-dist/tex/latex/pdfpages/pdfpages.sty\n",
      "(/usr/share/texlive/texmf-dist/tex/latex/base/ifthen.sty)\n",
      "(/usr/share/texlive/texmf-dist/tex/latex/tools/calc.sty)\n",
      "(/usr/share/texlive/texmf-dist/tex/latex/eso-pic/eso-pic.sty\n",
      "(/usr/share/texlive/texmf-dist/tex/generic/oberdiek/atbegshi.sty\n",
      "(/usr/share/texlive/texmf-dist/tex/generic/oberdiek/infwarerr.sty)\n",
      "(/usr/share/texlive/texmf-dist/tex/generic/oberdiek/ltxcmds.sty)\n",
      "(/usr/share/texlive/texmf-dist/tex/generic/oberdiek/ifpdf.sty))\n",
      "(/usr/share/texlive/texmf-dist/tex/latex/graphics/keyval.sty))\n",
      "(/usr/share/texlive/texmf-dist/tex/latex/graphics/graphicx.sty\n",
      "(/usr/share/texlive/texmf-dist/tex/latex/graphics/graphics.sty\n",
      "(/usr/share/texlive/texmf-dist/tex/latex/graphics/trig.sty)\n",
      "(/usr/share/texlive/texmf-dist/tex/latex/graphics-cfg/graphics.cfg)))\n",
      "(/usr/share/texlive/texmf-dist/tex/latex/pdfpages/pppdftex.def))\n",
      "(/usr/share/texlive/texmf-dist/tex/latex/geometry/geometry.sty\n",
      "(/usr/share/texlive/texmf-dist/tex/generic/oberdiek/ifvtex.sty)\n",
      "(/usr/share/texlive/texmf-dist/tex/generic/ifxetex/ifxetex.sty))\n",
      "(./latex_script_to_combine_plotsr_pdfs.aux)\n",
      "(/usr/share/texlive/texmf-dist/tex/context/base/mkii/supp-pdf.mkii\n",
      "[Loading MPS to PDF converter (version 2006.09.02).]\n",
      ") (/usr/share/texlive/texmf-dist/tex/latex/oberdiek/epstopdf-base.sty\n",
      "(/usr/share/texlive/texmf-dist/tex/latex/oberdiek/grfext.sty\n",
      "(/usr/share/texlive/texmf-dist/tex/generic/oberdiek/kvdefinekeys.sty))\n",
      "(/usr/share/texlive/texmf-dist/tex/latex/oberdiek/kvoptions.sty\n",
      "(/usr/share/texlive/texmf-dist/tex/generic/oberdiek/kvsetkeys.sty\n",
      "(/usr/share/texlive/texmf-dist/tex/generic/oberdiek/etexcmds.sty\n",
      "(/usr/share/texlive/texmf-dist/tex/generic/oberdiek/ifluatex.sty))))\n",
      "(/usr/share/texlive/texmf-dist/tex/generic/oberdiek/pdftexcmds.sty)\n",
      "(/usr/share/texlive/texmf-dist/tex/latex/latexconfig/epstopdf-sys.cfg))\n",
      "(/usr/share/texlive/texmf-dist/tex/latex/oberdiek/pdflscape.sty\n",
      "(/usr/share/texlive/texmf-dist/tex/latex/graphics/lscape.sty))\n",
      "*geometry* driver: auto-detecting\n",
      "*geometry* detected driver: pdftex\n",
      "[1 <./combined_plotsr.pdf> <./combined_plotsr.pdf> <./combined_plotsr.pdf> <./c\n",
      "ombined_plotsr.pdf>] [2 <./combined_plotsr.pdf> <./combined_plotsr.pdf> <./comb\n",
      "ined_plotsr.pdf> <./combined_plotsr.pdf>] [3 <./combined_plotsr.pdf> <./combine\n",
      "d_plotsr.pdf> <./combined_plotsr.pdf>]\n",
      "(./latex_script_to_combine_plotsr_pdfs.aux) )\n",
      "Output written on latex_script_to_combine_plotsr_pdfs.pdf (3 pages, 95797 bytes\n",
      ").\n",
      "Transcript written on latex_script_to_combine_plotsr_pdfs.log.\n"
     ]
    }
   ],
   "source": [
    "# Combine every four pdfs to one page\n",
    "merged_out = os.path.join(exportdir, 'merged')\n",
    "merged_tmp = os.path.join(tmpdir, 'merged')\n",
    "C = 'pdfunite '\n",
    "for parent in ['L01', 'L02', 'L03', 'L04', 'L05', 'L06', 'L07', 'L08', 'L09', 'L10', 'L11']:\n",
    "\tC += os.path.join(merged_out, parent, 'merged-local_plotsr.pdf') + ' '\n",
    "C += ' ./combine_pdf/combined_plotsr.pdf'\n",
    "print(C)\n",
    "!{C}\n",
    "!{'cd ./combine_pdf && pdflatex latex_script_to_combine_plotsr_pdfs.tex'}"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "# Classify the events\n",
    "Determine dup, del. Connect ISs of different generations. Determine simple insertions etc."
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 10,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "python3 ../src/SyRI_IS/classify_IS_events.py  --prefix L01-1 --out exp/multiple_runs11/export/classify_IS_events --log exp/multiple_runs11/log --input exp/multiple_runs11/tmp/L01-1 --calc_depth_interval  --rid 1 --qid 2\n"
     ]
    }
   ],
   "source": [
    "lines_ = lines\n",
    "for i, line in enumerate(lines_):\n",
    "\tC_ = 'python3 ../src/SyRI_IS/classify_IS_events.py '\n",
    "\tC_ += ' --prefix ' + line \n",
    "\tC_ += ' --out ' + os.path.join(exportdir, 'classify_IS_events')\n",
    "\tC_ += ' --log ' + logdir\n",
    "\tC_ += ' --input ' + os.path.join(tmpdir, line)\n",
    "\tC_ += ' --calc_depth_interval '\n",
    "\tfor gen_id, g in enumerate(file_df[file_df.Prefix == line].file_name):\n",
    "\t\tif gen_id == 0: # 0: FACS, 1: MA_G08, 2: MA_G20\n",
    "\t\t\tcontinue\n",
    "\t\tC = C_ + ' --rid ' + str(gen_id+0) + ' --qid ' + str(gen_id+1)\n",
    "\t\tif i ==0 and gen_id == 1:\n",
    "\t\t\tprint(C)\n",
    "\t\tproc = subprocess.Popen(C, stdout=subprocess.PIPE, stderr=subprocess.PIPE, shell=True)\n",
    "\t\tout, err = proc.communicate()\n",
    "\n",
    "\t\t# Write stdout and stderr to files\n",
    "\t\twith open(os.path.join(stderr_out, 'std_out_' + line + '.log'), 'w') as f:\n",
    "\t\t\tf.write(out.decode('utf-8'))\n",
    "\n",
    "\t\twith open(os.path.join(stderr_out, 'std_err_' + line + '.log'), 'w') as f:\n",
    "\t\t\tf.write(err.decode('utf-8'))"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 11,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>group</th>\n",
       "      <th>start</th>\n",
       "      <th>end</th>\n",
       "      <th>depth</th>\n",
       "      <th>length</th>\n",
       "      <th>start_fix</th>\n",
       "      <th>end_fix</th>\n",
       "      <th>is_start</th>\n",
       "      <th>is_end</th>\n",
       "      <th>line</th>\n",
       "      <th>gen_id</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>549437</td>\n",
       "      <td>1.0</td>\n",
       "      <td>549437</td>\n",
       "      <td>1</td>\n",
       "      <td>552530</td>\n",
       "      <td>-1</td>\n",
       "      <td>3</td>\n",
       "      <td>L01-1</td>\n",
       "      <td>2</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>2</td>\n",
       "      <td>549438</td>\n",
       "      <td>554384</td>\n",
       "      <td>0.0</td>\n",
       "      <td>4947</td>\n",
       "      <td>552531</td>\n",
       "      <td>560570</td>\n",
       "      <td>3</td>\n",
       "      <td>6</td>\n",
       "      <td>L01-1</td>\n",
       "      <td>2</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>3</td>\n",
       "      <td>554385</td>\n",
       "      <td>2633390</td>\n",
       "      <td>1.0</td>\n",
       "      <td>2079006</td>\n",
       "      <td>560571</td>\n",
       "      <td>2661330</td>\n",
       "      <td>6</td>\n",
       "      <td>21</td>\n",
       "      <td>L01-1</td>\n",
       "      <td>2</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>4</td>\n",
       "      <td>2633391</td>\n",
       "      <td>2637598</td>\n",
       "      <td>0.0</td>\n",
       "      <td>4208</td>\n",
       "      <td>2661331</td>\n",
       "      <td>2665538</td>\n",
       "      <td>21</td>\n",
       "      <td>22</td>\n",
       "      <td>L01-1</td>\n",
       "      <td>2</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>5</td>\n",
       "      <td>2637599</td>\n",
       "      <td>3965065</td>\n",
       "      <td>1.0</td>\n",
       "      <td>1327467</td>\n",
       "      <td>2665539</td>\n",
       "      <td>4003851</td>\n",
       "      <td>22</td>\n",
       "      <td>-1</td>\n",
       "      <td>L01-1</td>\n",
       "      <td>2</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "   group    start      end  depth   length  start_fix  end_fix  is_start  \\\n",
       "0      1        1   549437    1.0   549437          1   552530        -1   \n",
       "1      2   549438   554384    0.0     4947     552531   560570         3   \n",
       "2      3   554385  2633390    1.0  2079006     560571  2661330         6   \n",
       "3      4  2633391  2637598    0.0     4208    2661331  2665538        21   \n",
       "4      5  2637599  3965065    1.0  1327467    2665539  4003851        22   \n",
       "\n",
       "   is_end   line  gen_id  \n",
       "0       3  L01-1       2  \n",
       "1       6  L01-1       2  \n",
       "2      21  L01-1       2  \n",
       "3      22  L01-1       2  \n",
       "4      -1  L01-1       2  "
      ]
     },
     "execution_count": 11,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "depth_interval_df_evo = pd.DataFrame()\n",
    "lines_ = lines\n",
    "for line in lines_:\n",
    "\tfor qid in [2,3]:\n",
    "\t\tdf_ = pd.read_csv(os.path.join(exportdir, 'classify_IS_events', line, f'{line}_genome.{qid}.depth_interval.csv'))\n",
    "\t\tdf_['line'] = line\n",
    "\t\tdf_['gen_id'] = qid\n",
    "\t\tdepth_interval_df_evo = pd.concat([depth_interval_df_evo, df_], axis=0)\n",
    "depth_interval_df_evo= depth_interval_df_evo.reset_index(drop=True)\n",
    "depth_interval_df_evo['gen_id'] = depth_interval_df_evo['gen_id'].astype(int)\n",
    "depth_interval_df_evo.to_csv(os.path.join(exportdir, 'classify_IS_events', 'depth_interval.csv'), index=False)\n",
    "depth_interval_df_evo.head()"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "## IS events compared to FACS generation\n",
    "The previous analysis was based on the comparison between two consecutive generations. Here we compare the IS events to the FACS generation.\n",
    "currently, simple transposition is ignored"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 12,
   "metadata": {},
   "outputs": [
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "[M::mm_idx_gen::0.057*1.01] collected minimizers\n",
      "[M::mm_idx_gen::0.076*1.00] sorted minimizers\n",
      "[M::main::0.076*1.00] loaded/built the index for 1 target sequence(s)\n",
      "[M::mm_mapopt_update::0.076*1.00] mid_occ = 100\n",
      "[M::mm_idx_stat] kmer size: 19; skip: 10; is_hpc: 1; #seq: 1\n",
      "[M::mm_idx_stat::0.080*1.00] distinct minimizers: 523880 (99.33% are singletons); average occurrences: 1.015; average spacing: 7.460; total length: 3965065\n",
      "[M::worker_pipeline::0.620*0.99] mapped 1 sequences\n",
      "[M::main] Version: 2.19-r1057\n",
      "[M::main] CMD: minimap2 -ax asm20 --eqx -H -f 100 -r1k,10k --rmq=no -t 1 exp/multiple_runs11/tmp/L01-1/is_deleted/L01-1_genome.deleted_IS.1.fasta exp/multiple_runs11/tmp/L01-1/is_deleted/L01-1_genome.deleted_IS.2.fasta\n",
      "[M::main] Real time: 0.623 sec; CPU: 0.616 sec; Peak RSS: 0.087 GB\n",
      "local_variation - WARNING - Finished syri\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "TLOut.txt  is empty. Skipping analysing it.\n",
      "invTLOut.txt  is empty. Skipping analysing it.\n",
      "ctxOut.txt is empty. Skipping analysing it.\n",
      "TL Out.txt is empty. Skipping analysing it.\n",
      "invTL Out.txt is empty. Skipping analysing it.\n",
      "dup Out.txt is empty. Skipping analysing it.\n",
      "invDup Out.txt is empty. Skipping analysing it.\n",
      "ctxOut.txt is empty. Skipping analysing it.\n",
      "TLOut.txt  is empty. Skipping analysing it.\n",
      "invTLOut.txt  is empty. Skipping analysing it.\n",
      "ctxOut.txt is empty. Skipping analysing it.\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "[M::mm_idx_gen::0.057*1.01] collected minimizers\n",
      "[M::mm_idx_gen::0.076*1.00] sorted minimizers\n",
      "[M::main::0.076*1.00] loaded/built the index for 1 target sequence(s)\n",
      "[M::mm_mapopt_update::0.076*1.00] mid_occ = 100\n",
      "[M::mm_idx_stat] kmer size: 19; skip: 10; is_hpc: 1; #seq: 1\n",
      "[M::mm_idx_stat::0.080*1.00] distinct minimizers: 523880 (99.33% are singletons); average occurrences: 1.015; average spacing: 7.460; total length: 3965065\n",
      "[M::worker_pipeline::0.605*0.99] mapped 1 sequences\n",
      "[M::main] Version: 2.19-r1057\n",
      "[M::main] CMD: minimap2 -ax asm20 --eqx -H -f 100 -r1k,10k --rmq=no -t 1 exp/multiple_runs11/tmp/L01-1/is_deleted/L01-1_genome.deleted_IS.1.fasta exp/multiple_runs11/tmp/L01-1/is_deleted/L01-1_genome.deleted_IS.3.fasta\n",
      "[M::main] Real time: 0.608 sec; CPU: 0.601 sec; Peak RSS: 0.087 GB\n",
      "local_variation - WARNING - Finished syri\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "invTLOut.txt  is empty. Skipping analysing it.\n",
      "ctxOut.txt is empty. Skipping analysing it.\n",
      "invTL Out.txt is empty. Skipping analysing it.\n",
      "dup Out.txt is empty. Skipping analysing it.\n",
      "invDup Out.txt is empty. Skipping analysing it.\n",
      "ctxOut.txt is empty. Skipping analysing it.\n",
      "invTLOut.txt  is empty. Skipping analysing it.\n",
      "ctxOut.txt is empty. Skipping analysing it.\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "[M::mm_idx_gen::0.058*1.01] collected minimizers\n",
      "[M::mm_idx_gen::0.077*1.00] sorted minimizers\n",
      "[M::main::0.077*1.00] loaded/built the index for 1 target sequence(s)\n",
      "[M::mm_mapopt_update::0.077*1.00] mid_occ = 100\n",
      "[M::mm_idx_stat] kmer size: 19; skip: 10; is_hpc: 1; #seq: 1\n",
      "[M::mm_idx_stat::0.081*1.00] distinct minimizers: 523880 (99.33% are singletons); average occurrences: 1.015; average spacing: 7.460; total length: 3965065\n",
      "[M::worker_pipeline::0.621*0.99] mapped 1 sequences\n",
      "[M::main] Version: 2.19-r1057\n",
      "[M::main] CMD: minimap2 -ax asm20 --eqx -H -f 100 -r1k,10k --rmq=no -t 1 exp/multiple_runs11/tmp/L01-2/is_deleted/L01-2_genome.deleted_IS.1.fasta exp/multiple_runs11/tmp/L01-2/is_deleted/L01-2_genome.deleted_IS.2.fasta\n",
      "[M::main] Real time: 0.624 sec; CPU: 0.617 sec; Peak RSS: 0.087 GB\n",
      "local_variation - WARNING - Finished syri\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "TLOut.txt  is empty. Skipping analysing it.\n",
      "ctxOut.txt is empty. Skipping analysing it.\n",
      "TL Out.txt is empty. Skipping analysing it.\n",
      "dup Out.txt is empty. Skipping analysing it.\n",
      "invDup Out.txt is empty. Skipping analysing it.\n",
      "ctxOut.txt is empty. Skipping analysing it.\n",
      "TLOut.txt  is empty. Skipping analysing it.\n",
      "ctxOut.txt is empty. Skipping analysing it.\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "[M::mm_idx_gen::0.057*1.01] collected minimizers\n",
      "[M::mm_idx_gen::0.076*1.00] sorted minimizers\n",
      "[M::main::0.076*1.00] loaded/built the index for 1 target sequence(s)\n",
      "[M::mm_mapopt_update::0.076*1.00] mid_occ = 100\n",
      "[M::mm_idx_stat] kmer size: 19; skip: 10; is_hpc: 1; #seq: 1\n",
      "[M::mm_idx_stat::0.080*1.00] distinct minimizers: 523880 (99.33% are singletons); average occurrences: 1.015; average spacing: 7.460; total length: 3965065\n",
      "[M::worker_pipeline::0.615*0.99] mapped 1 sequences\n",
      "[M::main] Version: 2.19-r1057\n",
      "[M::main] CMD: minimap2 -ax asm20 --eqx -H -f 100 -r1k,10k --rmq=no -t 1 exp/multiple_runs11/tmp/L01-2/is_deleted/L01-2_genome.deleted_IS.1.fasta exp/multiple_runs11/tmp/L01-2/is_deleted/L01-2_genome.deleted_IS.3.fasta\n",
      "[M::main] Real time: 0.619 sec; CPU: 0.612 sec; Peak RSS: 0.087 GB\n",
      "local_variation - WARNING - Finished syri\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "TLOut.txt  is empty. Skipping analysing it.\n",
      "ctxOut.txt is empty. Skipping analysing it.\n",
      "TL Out.txt is empty. Skipping analysing it.\n",
      "dup Out.txt is empty. Skipping analysing it.\n",
      "invDup Out.txt is empty. Skipping analysing it.\n",
      "ctxOut.txt is empty. Skipping analysing it.\n",
      "TLOut.txt  is empty. Skipping analysing it.\n",
      "ctxOut.txt is empty. Skipping analysing it.\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "[M::mm_idx_gen::0.064*1.01] collected minimizers\n",
      "[M::mm_idx_gen::0.082*1.00] sorted minimizers\n",
      "[M::main::0.082*1.00] loaded/built the index for 1 target sequence(s)\n",
      "[M::mm_mapopt_update::0.082*1.00] mid_occ = 100\n",
      "[M::mm_idx_stat] kmer size: 19; skip: 10; is_hpc: 1; #seq: 1\n",
      "[M::mm_idx_stat::0.086*1.00] distinct minimizers: 523880 (99.33% are singletons); average occurrences: 1.015; average spacing: 7.460; total length: 3965065\n",
      "[M::worker_pipeline::0.623*0.99] mapped 1 sequences\n",
      "[M::main] Version: 2.19-r1057\n",
      "[M::main] CMD: minimap2 -ax asm20 --eqx -H -f 100 -r1k,10k --rmq=no -t 1 exp/multiple_runs11/tmp/L01-3/is_deleted/L01-3_genome.deleted_IS.1.fasta exp/multiple_runs11/tmp/L01-3/is_deleted/L01-3_genome.deleted_IS.2.fasta\n",
      "[M::main] Real time: 0.626 sec; CPU: 0.619 sec; Peak RSS: 0.088 GB\n",
      "local_variation - WARNING - Finished syri\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "TLOut.txt  is empty. Skipping analysing it.\n",
      "invTLOut.txt  is empty. Skipping analysing it.\n",
      "ctxOut.txt is empty. Skipping analysing it.\n",
      "TL Out.txt is empty. Skipping analysing it.\n",
      "invTL Out.txt is empty. Skipping analysing it.\n",
      "dup Out.txt is empty. Skipping analysing it.\n",
      "invDup Out.txt is empty. Skipping analysing it.\n",
      "ctxOut.txt is empty. Skipping analysing it.\n",
      "TLOut.txt  is empty. Skipping analysing it.\n",
      "invTLOut.txt  is empty. Skipping analysing it.\n",
      "ctxOut.txt is empty. Skipping analysing it.\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "[M::mm_idx_gen::0.057*1.01] collected minimizers\n",
      "[M::mm_idx_gen::0.076*1.00] sorted minimizers\n",
      "[M::main::0.076*1.00] loaded/built the index for 1 target sequence(s)\n",
      "[M::mm_mapopt_update::0.076*1.00] mid_occ = 100\n",
      "[M::mm_idx_stat] kmer size: 19; skip: 10; is_hpc: 1; #seq: 1\n",
      "[M::mm_idx_stat::0.080*1.00] distinct minimizers: 523880 (99.33% are singletons); average occurrences: 1.015; average spacing: 7.460; total length: 3965065\n",
      "[M::worker_pipeline::0.624*0.99] mapped 1 sequences\n",
      "[M::main] Version: 2.19-r1057\n",
      "[M::main] CMD: minimap2 -ax asm20 --eqx -H -f 100 -r1k,10k --rmq=no -t 1 exp/multiple_runs11/tmp/L01-3/is_deleted/L01-3_genome.deleted_IS.1.fasta exp/multiple_runs11/tmp/L01-3/is_deleted/L01-3_genome.deleted_IS.3.fasta\n",
      "[M::main] Real time: 0.628 sec; CPU: 0.621 sec; Peak RSS: 0.121 GB\n",
      "local_variation - WARNING - Finished syri\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "TLOut.txt  is empty. Skipping analysing it.\n",
      "invTLOut.txt  is empty. Skipping analysing it.\n",
      "ctxOut.txt is empty. Skipping analysing it.\n",
      "TL Out.txt is empty. Skipping analysing it.\n",
      "invTL Out.txt is empty. Skipping analysing it.\n",
      "dup Out.txt is empty. Skipping analysing it.\n",
      "invDup Out.txt is empty. Skipping analysing it.\n",
      "ctxOut.txt is empty. Skipping analysing it.\n",
      "TLOut.txt  is empty. Skipping analysing it.\n",
      "invTLOut.txt  is empty. Skipping analysing it.\n",
      "ctxOut.txt is empty. Skipping analysing it.\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "[M::mm_idx_gen::0.058*1.01] collected minimizers\n",
      "[M::mm_idx_gen::0.076*1.00] sorted minimizers\n",
      "[M::main::0.076*1.00] loaded/built the index for 1 target sequence(s)\n",
      "[M::mm_mapopt_update::0.076*1.00] mid_occ = 100\n",
      "[M::mm_idx_stat] kmer size: 19; skip: 10; is_hpc: 1; #seq: 1\n",
      "[M::mm_idx_stat::0.080*1.00] distinct minimizers: 523880 (99.33% are singletons); average occurrences: 1.015; average spacing: 7.460; total length: 3965065\n",
      "[M::worker_pipeline::0.618*0.99] mapped 1 sequences\n",
      "[M::main] Version: 2.19-r1057\n",
      "[M::main] CMD: minimap2 -ax asm20 --eqx -H -f 100 -r1k,10k --rmq=no -t 1 exp/multiple_runs11/tmp/L01-4/is_deleted/L01-4_genome.deleted_IS.1.fasta exp/multiple_runs11/tmp/L01-4/is_deleted/L01-4_genome.deleted_IS.2.fasta\n",
      "[M::main] Real time: 0.620 sec; CPU: 0.615 sec; Peak RSS: 0.086 GB\n",
      "local_variation - WARNING - Finished syri\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "TLOut.txt  is empty. Skipping analysing it.\n",
      "invTLOut.txt  is empty. Skipping analysing it.\n",
      "ctxOut.txt is empty. Skipping analysing it.\n",
      "TL Out.txt is empty. Skipping analysing it.\n",
      "invTL Out.txt is empty. Skipping analysing it.\n",
      "dup Out.txt is empty. Skipping analysing it.\n",
      "invDup Out.txt is empty. Skipping analysing it.\n",
      "ctxOut.txt is empty. Skipping analysing it.\n",
      "TLOut.txt  is empty. Skipping analysing it.\n",
      "invTLOut.txt  is empty. Skipping analysing it.\n",
      "ctxOut.txt is empty. Skipping analysing it.\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "[M::mm_idx_gen::0.057*1.01] collected minimizers\n",
      "[M::mm_idx_gen::0.076*1.00] sorted minimizers\n",
      "[M::main::0.076*1.00] loaded/built the index for 1 target sequence(s)\n",
      "[M::mm_mapopt_update::0.076*1.00] mid_occ = 100\n",
      "[M::mm_idx_stat] kmer size: 19; skip: 10; is_hpc: 1; #seq: 1\n",
      "[M::mm_idx_stat::0.080*1.00] distinct minimizers: 523880 (99.33% are singletons); average occurrences: 1.015; average spacing: 7.460; total length: 3965065\n",
      "[M::worker_pipeline::0.615*0.99] mapped 1 sequences\n",
      "[M::main] Version: 2.19-r1057\n",
      "[M::main] CMD: minimap2 -ax asm20 --eqx -H -f 100 -r1k,10k --rmq=no -t 1 exp/multiple_runs11/tmp/L01-4/is_deleted/L01-4_genome.deleted_IS.1.fasta exp/multiple_runs11/tmp/L01-4/is_deleted/L01-4_genome.deleted_IS.3.fasta\n",
      "[M::main] Real time: 0.618 sec; CPU: 0.611 sec; Peak RSS: 0.086 GB\n",
      "local_variation - WARNING - Finished syri\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "TLOut.txt  is empty. Skipping analysing it.\n",
      "invTLOut.txt  is empty. Skipping analysing it.\n",
      "ctxOut.txt is empty. Skipping analysing it.\n",
      "TL Out.txt is empty. Skipping analysing it.\n",
      "invTL Out.txt is empty. Skipping analysing it.\n",
      "dup Out.txt is empty. Skipping analysing it.\n",
      "invDup Out.txt is empty. Skipping analysing it.\n",
      "ctxOut.txt is empty. Skipping analysing it.\n",
      "TLOut.txt  is empty. Skipping analysing it.\n",
      "invTLOut.txt  is empty. Skipping analysing it.\n",
      "ctxOut.txt is empty. Skipping analysing it.\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "[M::mm_idx_gen::0.057*1.01] collected minimizers\n",
      "[M::mm_idx_gen::0.075*1.00] sorted minimizers\n",
      "[M::main::0.075*1.00] loaded/built the index for 1 target sequence(s)\n",
      "[M::mm_mapopt_update::0.075*1.00] mid_occ = 100\n",
      "[M::mm_idx_stat] kmer size: 19; skip: 10; is_hpc: 1; #seq: 1\n",
      "[M::mm_idx_stat::0.079*1.00] distinct minimizers: 525472 (99.33% are singletons); average occurrences: 1.015; average spacing: 7.461; total length: 3977353\n",
      "[M::worker_pipeline::0.607*0.99] mapped 1 sequences\n",
      "[M::main] Version: 2.19-r1057\n",
      "[M::main] CMD: minimap2 -ax asm20 --eqx -H -f 100 -r1k,10k --rmq=no -t 1 exp/multiple_runs11/tmp/L02-1/is_deleted/L02-1_genome.deleted_IS.1.fasta exp/multiple_runs11/tmp/L02-1/is_deleted/L02-1_genome.deleted_IS.2.fasta\n",
      "[M::main] Real time: 0.610 sec; CPU: 0.604 sec; Peak RSS: 0.087 GB\n",
      "local_variation - WARNING - Finished syri\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "invOut.txt  is empty. Skipping analysing it.\n",
      "TLOut.txt  is empty. Skipping analysing it.\n",
      "invTLOut.txt  is empty. Skipping analysing it.\n",
      "ctxOut.txt is empty. Skipping analysing it.\n",
      "inv Out.txt is empty. Skipping analysing it.\n",
      "TL Out.txt is empty. Skipping analysing it.\n",
      "invTL Out.txt is empty. Skipping analysing it.\n",
      "dup Out.txt is empty. Skipping analysing it.\n",
      "invDup Out.txt is empty. Skipping analysing it.\n",
      "ctxOut.txt is empty. Skipping analysing it.\n",
      "invOut.txt  is empty. Skipping analysing it.\n",
      "TLOut.txt  is empty. Skipping analysing it.\n",
      "invTLOut.txt  is empty. Skipping analysing it.\n",
      "ctxOut.txt is empty. Skipping analysing it.\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "[M::mm_idx_gen::0.063*1.01] collected minimizers\n",
      "[M::mm_idx_gen::0.081*1.00] sorted minimizers\n",
      "[M::main::0.081*1.00] loaded/built the index for 1 target sequence(s)\n",
      "[M::mm_mapopt_update::0.081*1.00] mid_occ = 100\n",
      "[M::mm_idx_stat] kmer size: 19; skip: 10; is_hpc: 1; #seq: 1\n",
      "[M::mm_idx_stat::0.085*1.00] distinct minimizers: 525472 (99.33% are singletons); average occurrences: 1.015; average spacing: 7.461; total length: 3977353\n",
      "[M::worker_pipeline::0.642*0.99] mapped 1 sequences\n",
      "[M::main] Version: 2.19-r1057\n",
      "[M::main] CMD: minimap2 -ax asm20 --eqx -H -f 100 -r1k,10k --rmq=no -t 1 exp/multiple_runs11/tmp/L02-1/is_deleted/L02-1_genome.deleted_IS.1.fasta exp/multiple_runs11/tmp/L02-1/is_deleted/L02-1_genome.deleted_IS.3.fasta\n",
      "[M::main] Real time: 0.645 sec; CPU: 0.638 sec; Peak RSS: 0.089 GB\n",
      "local_variation - WARNING - Finished syri\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "invOut.txt  is empty. Skipping analysing it.\n",
      "TLOut.txt  is empty. Skipping analysing it.\n",
      "ctxOut.txt is empty. Skipping analysing it.\n",
      "inv Out.txt is empty. Skipping analysing it.\n",
      "TL Out.txt is empty. Skipping analysing it.\n",
      "dup Out.txt is empty. Skipping analysing it.\n",
      "invDup Out.txt is empty. Skipping analysing it.\n",
      "ctxOut.txt is empty. Skipping analysing it.\n",
      "invOut.txt  is empty. Skipping analysing it.\n",
      "TLOut.txt  is empty. Skipping analysing it.\n",
      "ctxOut.txt is empty. Skipping analysing it.\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "[M::mm_idx_gen::0.064*1.01] collected minimizers\n",
      "[M::mm_idx_gen::0.083*1.00] sorted minimizers\n",
      "[M::main::0.083*1.00] loaded/built the index for 1 target sequence(s)\n",
      "[M::mm_mapopt_update::0.083*1.00] mid_occ = 100\n",
      "[M::mm_idx_stat] kmer size: 19; skip: 10; is_hpc: 1; #seq: 1\n",
      "[M::mm_idx_stat::0.086*1.00] distinct minimizers: 525472 (99.33% are singletons); average occurrences: 1.015; average spacing: 7.461; total length: 3977353\n",
      "[M::worker_pipeline::0.625*0.99] mapped 1 sequences\n",
      "[M::main] Version: 2.19-r1057\n",
      "[M::main] CMD: minimap2 -ax asm20 --eqx -H -f 100 -r1k,10k --rmq=no -t 1 exp/multiple_runs11/tmp/L02-2/is_deleted/L02-2_genome.deleted_IS.1.fasta exp/multiple_runs11/tmp/L02-2/is_deleted/L02-2_genome.deleted_IS.2.fasta\n",
      "[M::main] Real time: 0.627 sec; CPU: 0.621 sec; Peak RSS: 0.087 GB\n",
      "local_variation - WARNING - Finished syri\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "TLOut.txt  is empty. Skipping analysing it.\n",
      "invTLOut.txt  is empty. Skipping analysing it.\n",
      "ctxOut.txt is empty. Skipping analysing it.\n",
      "TL Out.txt is empty. Skipping analysing it.\n",
      "invTL Out.txt is empty. Skipping analysing it.\n",
      "invDup Out.txt is empty. Skipping analysing it.\n",
      "ctxOut.txt is empty. Skipping analysing it.\n",
      "TLOut.txt  is empty. Skipping analysing it.\n",
      "invTLOut.txt  is empty. Skipping analysing it.\n",
      "ctxOut.txt is empty. Skipping analysing it.\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "[M::mm_idx_gen::0.062*1.01] collected minimizers\n",
      "[M::mm_idx_gen::0.081*1.00] sorted minimizers\n",
      "[M::main::0.081*1.00] loaded/built the index for 1 target sequence(s)\n",
      "[M::mm_mapopt_update::0.081*1.00] mid_occ = 100\n",
      "[M::mm_idx_stat] kmer size: 19; skip: 10; is_hpc: 1; #seq: 1\n",
      "[M::mm_idx_stat::0.085*1.00] distinct minimizers: 525472 (99.33% are singletons); average occurrences: 1.015; average spacing: 7.461; total length: 3977353\n",
      "[M::worker_pipeline::0.682*0.99] mapped 1 sequences\n",
      "[M::main] Version: 2.19-r1057\n",
      "[M::main] CMD: minimap2 -ax asm20 --eqx -H -f 100 -r1k,10k --rmq=no -t 1 exp/multiple_runs11/tmp/L02-2/is_deleted/L02-2_genome.deleted_IS.1.fasta exp/multiple_runs11/tmp/L02-2/is_deleted/L02-2_genome.deleted_IS.3.fasta\n",
      "[M::main] Real time: 0.684 sec; CPU: 0.677 sec; Peak RSS: 0.091 GB\n",
      "local_variation - WARNING - Finished syri\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "TLOut.txt  is empty. Skipping analysing it.\n",
      "invTLOut.txt  is empty. Skipping analysing it.\n",
      "ctxOut.txt is empty. Skipping analysing it.\n",
      "TL Out.txt is empty. Skipping analysing it.\n",
      "invTL Out.txt is empty. Skipping analysing it.\n",
      "invDup Out.txt is empty. Skipping analysing it.\n",
      "ctxOut.txt is empty. Skipping analysing it.\n",
      "TLOut.txt  is empty. Skipping analysing it.\n",
      "invTLOut.txt  is empty. Skipping analysing it.\n",
      "ctxOut.txt is empty. Skipping analysing it.\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "[M::mm_idx_gen::0.058*1.01] collected minimizers\n",
      "[M::mm_idx_gen::0.076*1.00] sorted minimizers\n",
      "[M::main::0.076*1.00] loaded/built the index for 1 target sequence(s)\n",
      "[M::mm_mapopt_update::0.076*1.00] mid_occ = 100\n",
      "[M::mm_idx_stat] kmer size: 19; skip: 10; is_hpc: 1; #seq: 1\n",
      "[M::mm_idx_stat::0.080*1.00] distinct minimizers: 525472 (99.33% are singletons); average occurrences: 1.015; average spacing: 7.461; total length: 3977353\n",
      "[M::worker_pipeline::0.619*0.99] mapped 1 sequences\n",
      "[M::main] Version: 2.19-r1057\n",
      "[M::main] CMD: minimap2 -ax asm20 --eqx -H -f 100 -r1k,10k --rmq=no -t 1 exp/multiple_runs11/tmp/L02-3/is_deleted/L02-3_genome.deleted_IS.1.fasta exp/multiple_runs11/tmp/L02-3/is_deleted/L02-3_genome.deleted_IS.2.fasta\n",
      "[M::main] Real time: 0.622 sec; CPU: 0.615 sec; Peak RSS: 0.086 GB\n",
      "local_variation - WARNING - Finished syri\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "invOut.txt  is empty. Skipping analysing it.\n",
      "TLOut.txt  is empty. Skipping analysing it.\n",
      "invTLOut.txt  is empty. Skipping analysing it.\n",
      "ctxOut.txt is empty. Skipping analysing it.\n",
      "inv Out.txt is empty. Skipping analysing it.\n",
      "TL Out.txt is empty. Skipping analysing it.\n",
      "invTL Out.txt is empty. Skipping analysing it.\n",
      "dup Out.txt is empty. Skipping analysing it.\n",
      "invDup Out.txt is empty. Skipping analysing it.\n",
      "ctxOut.txt is empty. Skipping analysing it.\n",
      "invOut.txt  is empty. Skipping analysing it.\n",
      "TLOut.txt  is empty. Skipping analysing it.\n",
      "invTLOut.txt  is empty. Skipping analysing it.\n",
      "ctxOut.txt is empty. Skipping analysing it.\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "[M::mm_idx_gen::0.058*1.00] collected minimizers\n",
      "[M::mm_idx_gen::0.077*1.00] sorted minimizers\n",
      "[M::main::0.077*1.00] loaded/built the index for 1 target sequence(s)\n",
      "[M::mm_mapopt_update::0.077*1.00] mid_occ = 100\n",
      "[M::mm_idx_stat] kmer size: 19; skip: 10; is_hpc: 1; #seq: 1\n",
      "[M::mm_idx_stat::0.080*1.00] distinct minimizers: 525472 (99.33% are singletons); average occurrences: 1.015; average spacing: 7.461; total length: 3977353\n",
      "[M::worker_pipeline::0.647*0.99] mapped 1 sequences\n",
      "[M::main] Version: 2.19-r1057\n",
      "[M::main] CMD: minimap2 -ax asm20 --eqx -H -f 100 -r1k,10k --rmq=no -t 1 exp/multiple_runs11/tmp/L02-3/is_deleted/L02-3_genome.deleted_IS.1.fasta exp/multiple_runs11/tmp/L02-3/is_deleted/L02-3_genome.deleted_IS.3.fasta\n",
      "[M::main] Real time: 0.649 sec; CPU: 0.642 sec; Peak RSS: 0.089 GB\n",
      "local_variation - WARNING - Finished syri\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "TLOut.txt  is empty. Skipping analysing it.\n",
      "ctxOut.txt is empty. Skipping analysing it.\n",
      "TL Out.txt is empty. Skipping analysing it.\n",
      "dup Out.txt is empty. Skipping analysing it.\n",
      "invDup Out.txt is empty. Skipping analysing it.\n",
      "ctxOut.txt is empty. Skipping analysing it.\n",
      "TLOut.txt  is empty. Skipping analysing it.\n",
      "ctxOut.txt is empty. Skipping analysing it.\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "[M::mm_idx_gen::0.060*1.01] collected minimizers\n",
      "[M::mm_idx_gen::0.078*1.00] sorted minimizers\n",
      "[M::main::0.078*1.00] loaded/built the index for 1 target sequence(s)\n",
      "[M::mm_mapopt_update::0.078*1.00] mid_occ = 100\n",
      "[M::mm_idx_stat] kmer size: 19; skip: 10; is_hpc: 1; #seq: 1\n",
      "[M::mm_idx_stat::0.082*1.00] distinct minimizers: 525472 (99.33% are singletons); average occurrences: 1.015; average spacing: 7.461; total length: 3977353\n",
      "[M::worker_pipeline::0.612*0.99] mapped 1 sequences\n",
      "[M::main] Version: 2.19-r1057\n",
      "[M::main] CMD: minimap2 -ax asm20 --eqx -H -f 100 -r1k,10k --rmq=no -t 1 exp/multiple_runs11/tmp/L02-4/is_deleted/L02-4_genome.deleted_IS.1.fasta exp/multiple_runs11/tmp/L02-4/is_deleted/L02-4_genome.deleted_IS.2.fasta\n",
      "[M::main] Real time: 0.615 sec; CPU: 0.610 sec; Peak RSS: 0.086 GB\n",
      "local_variation - WARNING - Finished syri\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "TLOut.txt  is empty. Skipping analysing it.\n",
      "invTLOut.txt  is empty. Skipping analysing it.\n",
      "ctxOut.txt is empty. Skipping analysing it.\n",
      "TL Out.txt is empty. Skipping analysing it.\n",
      "invTL Out.txt is empty. Skipping analysing it.\n",
      "dup Out.txt is empty. Skipping analysing it.\n",
      "invDup Out.txt is empty. Skipping analysing it.\n",
      "ctxOut.txt is empty. Skipping analysing it.\n",
      "TLOut.txt  is empty. Skipping analysing it.\n",
      "invTLOut.txt  is empty. Skipping analysing it.\n",
      "ctxOut.txt is empty. Skipping analysing it.\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "[M::mm_idx_gen::0.059*1.01] collected minimizers\n",
      "[M::mm_idx_gen::0.077*1.00] sorted minimizers\n",
      "[M::main::0.077*1.00] loaded/built the index for 1 target sequence(s)\n",
      "[M::mm_mapopt_update::0.077*1.00] mid_occ = 100\n",
      "[M::mm_idx_stat] kmer size: 19; skip: 10; is_hpc: 1; #seq: 1\n",
      "[M::mm_idx_stat::0.081*1.00] distinct minimizers: 525472 (99.33% are singletons); average occurrences: 1.015; average spacing: 7.461; total length: 3977353\n",
      "[M::worker_pipeline::0.617*0.99] mapped 1 sequences\n",
      "[M::main] Version: 2.19-r1057\n",
      "[M::main] CMD: minimap2 -ax asm20 --eqx -H -f 100 -r1k,10k --rmq=no -t 1 exp/multiple_runs11/tmp/L02-4/is_deleted/L02-4_genome.deleted_IS.1.fasta exp/multiple_runs11/tmp/L02-4/is_deleted/L02-4_genome.deleted_IS.3.fasta\n",
      "[M::main] Real time: 0.619 sec; CPU: 0.613 sec; Peak RSS: 0.089 GB\n",
      "local_variation - WARNING - Finished syri\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "TLOut.txt  is empty. Skipping analysing it.\n",
      "invTLOut.txt  is empty. Skipping analysing it.\n",
      "ctxOut.txt is empty. Skipping analysing it.\n",
      "TL Out.txt is empty. Skipping analysing it.\n",
      "invTL Out.txt is empty. Skipping analysing it.\n",
      "dup Out.txt is empty. Skipping analysing it.\n",
      "ctxOut.txt is empty. Skipping analysing it.\n",
      "TLOut.txt  is empty. Skipping analysing it.\n",
      "invTLOut.txt  is empty. Skipping analysing it.\n",
      "ctxOut.txt is empty. Skipping analysing it.\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "[M::mm_idx_gen::0.062*1.01] collected minimizers\n",
      "[M::mm_idx_gen::0.080*1.00] sorted minimizers\n",
      "[M::main::0.080*1.00] loaded/built the index for 1 target sequence(s)\n",
      "[M::mm_mapopt_update::0.080*1.00] mid_occ = 100\n",
      "[M::mm_idx_stat] kmer size: 19; skip: 10; is_hpc: 1; #seq: 1\n",
      "[M::mm_idx_stat::0.084*1.00] distinct minimizers: 517524 (99.37% are singletons); average occurrences: 1.014; average spacing: 7.460; total length: 3916002\n",
      "[M::worker_pipeline::0.616*1.00] mapped 1 sequences\n",
      "[M::main] Version: 2.19-r1057\n",
      "[M::main] CMD: minimap2 -ax asm20 --eqx -H -f 100 -r1k,10k --rmq=no -t 1 exp/multiple_runs11/tmp/L03-1/is_deleted/L03-1_genome.deleted_IS.1.fasta exp/multiple_runs11/tmp/L03-1/is_deleted/L03-1_genome.deleted_IS.2.fasta\n",
      "[M::main] Real time: 0.618 sec; CPU: 0.617 sec; Peak RSS: 0.086 GB\n",
      "local_variation - WARNING - Finished syri\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "invOut.txt  is empty. Skipping analysing it.\n",
      "TLOut.txt  is empty. Skipping analysing it.\n",
      "invTLOut.txt  is empty. Skipping analysing it.\n",
      "ctxOut.txt is empty. Skipping analysing it.\n",
      "inv Out.txt is empty. Skipping analysing it.\n",
      "TL Out.txt is empty. Skipping analysing it.\n",
      "invTL Out.txt is empty. Skipping analysing it.\n",
      "dup Out.txt is empty. Skipping analysing it.\n",
      "invDup Out.txt is empty. Skipping analysing it.\n",
      "ctxOut.txt is empty. Skipping analysing it.\n",
      "invOut.txt  is empty. Skipping analysing it.\n",
      "TLOut.txt  is empty. Skipping analysing it.\n",
      "invTLOut.txt  is empty. Skipping analysing it.\n",
      "ctxOut.txt is empty. Skipping analysing it.\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "[M::mm_idx_gen::0.061*1.01] collected minimizers\n",
      "[M::mm_idx_gen::0.079*1.00] sorted minimizers\n",
      "[M::main::0.079*1.00] loaded/built the index for 1 target sequence(s)\n",
      "[M::mm_mapopt_update::0.079*1.00] mid_occ = 100\n",
      "[M::mm_idx_stat] kmer size: 19; skip: 10; is_hpc: 1; #seq: 1\n",
      "[M::mm_idx_stat::0.083*1.00] distinct minimizers: 517524 (99.37% are singletons); average occurrences: 1.014; average spacing: 7.460; total length: 3916002\n",
      "[M::worker_pipeline::0.613*0.99] mapped 1 sequences\n",
      "[M::main] Version: 2.19-r1057\n",
      "[M::main] CMD: minimap2 -ax asm20 --eqx -H -f 100 -r1k,10k --rmq=no -t 1 exp/multiple_runs11/tmp/L03-1/is_deleted/L03-1_genome.deleted_IS.1.fasta exp/multiple_runs11/tmp/L03-1/is_deleted/L03-1_genome.deleted_IS.3.fasta\n",
      "[M::main] Real time: 0.616 sec; CPU: 0.610 sec; Peak RSS: 0.086 GB\n",
      "local_variation - WARNING - Finished syri\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "TLOut.txt  is empty. Skipping analysing it.\n",
      "invTLOut.txt  is empty. Skipping analysing it.\n",
      "ctxOut.txt is empty. Skipping analysing it.\n",
      "TL Out.txt is empty. Skipping analysing it.\n",
      "invTL Out.txt is empty. Skipping analysing it.\n",
      "dup Out.txt is empty. Skipping analysing it.\n",
      "invDup Out.txt is empty. Skipping analysing it.\n",
      "ctxOut.txt is empty. Skipping analysing it.\n",
      "TLOut.txt  is empty. Skipping analysing it.\n",
      "invTLOut.txt  is empty. Skipping analysing it.\n",
      "ctxOut.txt is empty. Skipping analysing it.\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "[M::mm_idx_gen::0.057*1.01] collected minimizers\n",
      "[M::mm_idx_gen::0.075*1.00] sorted minimizers\n",
      "[M::main::0.075*1.00] loaded/built the index for 1 target sequence(s)\n",
      "[M::mm_mapopt_update::0.075*1.00] mid_occ = 100\n",
      "[M::mm_idx_stat] kmer size: 19; skip: 10; is_hpc: 1; #seq: 1\n",
      "[M::mm_idx_stat::0.079*1.00] distinct minimizers: 517524 (99.37% are singletons); average occurrences: 1.014; average spacing: 7.460; total length: 3916002\n",
      "[M::worker_pipeline::0.616*0.99] mapped 1 sequences\n",
      "[M::main] Version: 2.19-r1057\n",
      "[M::main] CMD: minimap2 -ax asm20 --eqx -H -f 100 -r1k,10k --rmq=no -t 1 exp/multiple_runs11/tmp/L03-2/is_deleted/L03-2_genome.deleted_IS.1.fasta exp/multiple_runs11/tmp/L03-2/is_deleted/L03-2_genome.deleted_IS.2.fasta\n",
      "[M::main] Real time: 0.619 sec; CPU: 0.613 sec; Peak RSS: 0.085 GB\n",
      "local_variation - WARNING - Finished syri\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "TLOut.txt  is empty. Skipping analysing it.\n",
      "invTLOut.txt  is empty. Skipping analysing it.\n",
      "ctxOut.txt is empty. Skipping analysing it.\n",
      "TL Out.txt is empty. Skipping analysing it.\n",
      "invTL Out.txt is empty. Skipping analysing it.\n",
      "dup Out.txt is empty. Skipping analysing it.\n",
      "invDup Out.txt is empty. Skipping analysing it.\n",
      "ctxOut.txt is empty. Skipping analysing it.\n",
      "TLOut.txt  is empty. Skipping analysing it.\n",
      "invTLOut.txt  is empty. Skipping analysing it.\n",
      "ctxOut.txt is empty. Skipping analysing it.\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "[M::mm_idx_gen::0.057*1.00] collected minimizers\n",
      "[M::mm_idx_gen::0.076*1.00] sorted minimizers\n",
      "[M::main::0.076*1.00] loaded/built the index for 1 target sequence(s)\n",
      "[M::mm_mapopt_update::0.076*1.00] mid_occ = 100\n",
      "[M::mm_idx_stat] kmer size: 19; skip: 10; is_hpc: 1; #seq: 1\n",
      "[M::mm_idx_stat::0.079*1.00] distinct minimizers: 517524 (99.37% are singletons); average occurrences: 1.014; average spacing: 7.460; total length: 3916002\n",
      "[M::worker_pipeline::0.620*0.99] mapped 1 sequences\n",
      "[M::main] Version: 2.19-r1057\n",
      "[M::main] CMD: minimap2 -ax asm20 --eqx -H -f 100 -r1k,10k --rmq=no -t 1 exp/multiple_runs11/tmp/L03-2/is_deleted/L03-2_genome.deleted_IS.1.fasta exp/multiple_runs11/tmp/L03-2/is_deleted/L03-2_genome.deleted_IS.3.fasta\n",
      "[M::main] Real time: 0.622 sec; CPU: 0.616 sec; Peak RSS: 0.086 GB\n",
      "local_variation - WARNING - Finished syri\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "invTLOut.txt  is empty. Skipping analysing it.\n",
      "ctxOut.txt is empty. Skipping analysing it.\n",
      "invTL Out.txt is empty. Skipping analysing it.\n",
      "dup Out.txt is empty. Skipping analysing it.\n",
      "invDup Out.txt is empty. Skipping analysing it.\n",
      "ctxOut.txt is empty. Skipping analysing it.\n",
      "invTLOut.txt  is empty. Skipping analysing it.\n",
      "ctxOut.txt is empty. Skipping analysing it.\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "[M::mm_idx_gen::0.062*1.01] collected minimizers\n",
      "[M::mm_idx_gen::0.080*1.00] sorted minimizers\n",
      "[M::main::0.080*1.00] loaded/built the index for 1 target sequence(s)\n",
      "[M::mm_mapopt_update::0.080*1.00] mid_occ = 100\n",
      "[M::mm_idx_stat] kmer size: 19; skip: 10; is_hpc: 1; #seq: 1\n",
      "[M::mm_idx_stat::0.084*1.00] distinct minimizers: 517524 (99.37% are singletons); average occurrences: 1.014; average spacing: 7.460; total length: 3916002\n",
      "[M::worker_pipeline::0.626*0.99] mapped 1 sequences\n",
      "[M::main] Version: 2.19-r1057\n",
      "[M::main] CMD: minimap2 -ax asm20 --eqx -H -f 100 -r1k,10k --rmq=no -t 1 exp/multiple_runs11/tmp/L03-3/is_deleted/L03-3_genome.deleted_IS.1.fasta exp/multiple_runs11/tmp/L03-3/is_deleted/L03-3_genome.deleted_IS.2.fasta\n",
      "[M::main] Real time: 0.629 sec; CPU: 0.622 sec; Peak RSS: 0.086 GB\n",
      "local_variation - WARNING - Finished syri\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "invOut.txt  is empty. Skipping analysing it.\n",
      "TLOut.txt  is empty. Skipping analysing it.\n",
      "invTLOut.txt  is empty. Skipping analysing it.\n",
      "ctxOut.txt is empty. Skipping analysing it.\n",
      "inv Out.txt is empty. Skipping analysing it.\n",
      "TL Out.txt is empty. Skipping analysing it.\n",
      "invTL Out.txt is empty. Skipping analysing it.\n",
      "dup Out.txt is empty. Skipping analysing it.\n",
      "ctxOut.txt is empty. Skipping analysing it.\n",
      "invOut.txt  is empty. Skipping analysing it.\n",
      "TLOut.txt  is empty. Skipping analysing it.\n",
      "invTLOut.txt  is empty. Skipping analysing it.\n",
      "ctxOut.txt is empty. Skipping analysing it.\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "[M::mm_idx_gen::0.061*1.01] collected minimizers\n",
      "[M::mm_idx_gen::0.079*1.00] sorted minimizers\n",
      "[M::main::0.079*1.00] loaded/built the index for 1 target sequence(s)\n",
      "[M::mm_mapopt_update::0.079*1.00] mid_occ = 100\n",
      "[M::mm_idx_stat] kmer size: 19; skip: 10; is_hpc: 1; #seq: 1\n",
      "[M::mm_idx_stat::0.083*1.00] distinct minimizers: 517524 (99.37% are singletons); average occurrences: 1.014; average spacing: 7.460; total length: 3916002\n",
      "[M::worker_pipeline::0.610*0.99] mapped 1 sequences\n",
      "[M::main] Version: 2.19-r1057\n",
      "[M::main] CMD: minimap2 -ax asm20 --eqx -H -f 100 -r1k,10k --rmq=no -t 1 exp/multiple_runs11/tmp/L03-3/is_deleted/L03-3_genome.deleted_IS.1.fasta exp/multiple_runs11/tmp/L03-3/is_deleted/L03-3_genome.deleted_IS.3.fasta\n",
      "[M::main] Real time: 0.613 sec; CPU: 0.607 sec; Peak RSS: 0.086 GB\n",
      "local_variation - WARNING - Finished syri\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "invOut.txt  is empty. Skipping analysing it.\n",
      "invTLOut.txt  is empty. Skipping analysing it.\n",
      "ctxOut.txt is empty. Skipping analysing it.\n",
      "inv Out.txt is empty. Skipping analysing it.\n",
      "invTL Out.txt is empty. Skipping analysing it.\n",
      "dup Out.txt is empty. Skipping analysing it.\n",
      "invDup Out.txt is empty. Skipping analysing it.\n",
      "ctxOut.txt is empty. Skipping analysing it.\n",
      "invOut.txt  is empty. Skipping analysing it.\n",
      "invTLOut.txt  is empty. Skipping analysing it.\n",
      "ctxOut.txt is empty. Skipping analysing it.\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "[M::mm_idx_gen::0.058*1.01] collected minimizers\n",
      "[M::mm_idx_gen::0.076*1.00] sorted minimizers\n",
      "[M::main::0.076*1.00] loaded/built the index for 1 target sequence(s)\n",
      "[M::mm_mapopt_update::0.076*1.00] mid_occ = 100\n",
      "[M::mm_idx_stat] kmer size: 19; skip: 10; is_hpc: 1; #seq: 1\n",
      "[M::mm_idx_stat::0.079*1.00] distinct minimizers: 517524 (99.37% are singletons); average occurrences: 1.014; average spacing: 7.460; total length: 3916002\n",
      "[M::worker_pipeline::0.613*1.00] mapped 1 sequences\n",
      "[M::main] Version: 2.19-r1057\n",
      "[M::main] CMD: minimap2 -ax asm20 --eqx -H -f 100 -r1k,10k --rmq=no -t 1 exp/multiple_runs11/tmp/L03-4/is_deleted/L03-4_genome.deleted_IS.1.fasta exp/multiple_runs11/tmp/L03-4/is_deleted/L03-4_genome.deleted_IS.2.fasta\n",
      "[M::main] Real time: 0.615 sec; CPU: 0.614 sec; Peak RSS: 0.086 GB\n",
      "local_variation - WARNING - Finished syri\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "invOut.txt  is empty. Skipping analysing it.\n",
      "TLOut.txt  is empty. Skipping analysing it.\n",
      "invTLOut.txt  is empty. Skipping analysing it.\n",
      "ctxOut.txt is empty. Skipping analysing it.\n",
      "inv Out.txt is empty. Skipping analysing it.\n",
      "TL Out.txt is empty. Skipping analysing it.\n",
      "invTL Out.txt is empty. Skipping analysing it.\n",
      "dup Out.txt is empty. Skipping analysing it.\n",
      "invDup Out.txt is empty. Skipping analysing it.\n",
      "ctxOut.txt is empty. Skipping analysing it.\n",
      "invOut.txt  is empty. Skipping analysing it.\n",
      "TLOut.txt  is empty. Skipping analysing it.\n",
      "invTLOut.txt  is empty. Skipping analysing it.\n",
      "ctxOut.txt is empty. Skipping analysing it.\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "[M::mm_idx_gen::0.057*1.01] collected minimizers\n",
      "[M::mm_idx_gen::0.076*1.01] sorted minimizers\n",
      "[M::main::0.076*1.01] loaded/built the index for 1 target sequence(s)\n",
      "[M::mm_mapopt_update::0.076*1.01] mid_occ = 100\n",
      "[M::mm_idx_stat] kmer size: 19; skip: 10; is_hpc: 1; #seq: 1\n",
      "[M::mm_idx_stat::0.079*1.00] distinct minimizers: 517524 (99.37% are singletons); average occurrences: 1.014; average spacing: 7.460; total length: 3916002\n",
      "[M::worker_pipeline::0.623*0.99] mapped 1 sequences\n",
      "[M::main] Version: 2.19-r1057\n",
      "[M::main] CMD: minimap2 -ax asm20 --eqx -H -f 100 -r1k,10k --rmq=no -t 1 exp/multiple_runs11/tmp/L03-4/is_deleted/L03-4_genome.deleted_IS.1.fasta exp/multiple_runs11/tmp/L03-4/is_deleted/L03-4_genome.deleted_IS.3.fasta\n",
      "[M::main] Real time: 0.625 sec; CPU: 0.620 sec; Peak RSS: 0.085 GB\n",
      "local_variation - WARNING - Finished syri\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "invOut.txt  is empty. Skipping analysing it.\n",
      "TLOut.txt  is empty. Skipping analysing it.\n",
      "invTLOut.txt  is empty. Skipping analysing it.\n",
      "ctxOut.txt is empty. Skipping analysing it.\n",
      "inv Out.txt is empty. Skipping analysing it.\n",
      "TL Out.txt is empty. Skipping analysing it.\n",
      "invTL Out.txt is empty. Skipping analysing it.\n",
      "invDup Out.txt is empty. Skipping analysing it.\n",
      "ctxOut.txt is empty. Skipping analysing it.\n",
      "invOut.txt  is empty. Skipping analysing it.\n",
      "TLOut.txt  is empty. Skipping analysing it.\n",
      "invTLOut.txt  is empty. Skipping analysing it.\n",
      "ctxOut.txt is empty. Skipping analysing it.\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "[M::mm_idx_gen::0.058*1.01] collected minimizers\n",
      "[M::mm_idx_gen::0.077*1.00] sorted minimizers\n",
      "[M::main::0.077*1.00] loaded/built the index for 1 target sequence(s)\n",
      "[M::mm_mapopt_update::0.077*1.00] mid_occ = 100\n",
      "[M::mm_idx_stat] kmer size: 19; skip: 10; is_hpc: 1; #seq: 1\n",
      "[M::mm_idx_stat::0.081*1.00] distinct minimizers: 521853 (99.34% are singletons); average occurrences: 1.014; average spacing: 7.460; total length: 3949534\n",
      "[M::worker_pipeline::0.625*0.99] mapped 1 sequences\n",
      "[M::main] Version: 2.19-r1057\n",
      "[M::main] CMD: minimap2 -ax asm20 --eqx -H -f 100 -r1k,10k --rmq=no -t 1 exp/multiple_runs11/tmp/L04-1/is_deleted/L04-1_genome.deleted_IS.1.fasta exp/multiple_runs11/tmp/L04-1/is_deleted/L04-1_genome.deleted_IS.2.fasta\n",
      "[M::main] Real time: 0.627 sec; CPU: 0.622 sec; Peak RSS: 0.087 GB\n",
      "Reading Coords - WARNING - Reference chromosome contig_1 has high fraction of inverted alignments with its homologous chromosome in the query genome (contig_1). Ensure that same chromosome-strands are being compared in the two genomes, as different strand can result in unexpected errors.\n",
      "local_variation - WARNING - Finished syri\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "TLOut.txt  is empty. Skipping analysing it.\n",
      "invTLOut.txt  is empty. Skipping analysing it.\n",
      "ctxOut.txt is empty. Skipping analysing it.\n",
      "TL Out.txt is empty. Skipping analysing it.\n",
      "invTL Out.txt is empty. Skipping analysing it.\n",
      "dup Out.txt is empty. Skipping analysing it.\n",
      "invDup Out.txt is empty. Skipping analysing it.\n",
      "ctxOut.txt is empty. Skipping analysing it.\n",
      "TLOut.txt  is empty. Skipping analysing it.\n",
      "invTLOut.txt  is empty. Skipping analysing it.\n",
      "ctxOut.txt is empty. Skipping analysing it.\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "[M::mm_idx_gen::0.058*1.01] collected minimizers\n",
      "[M::mm_idx_gen::0.077*1.00] sorted minimizers\n",
      "[M::main::0.077*1.00] loaded/built the index for 1 target sequence(s)\n",
      "[M::mm_mapopt_update::0.077*1.00] mid_occ = 100\n",
      "[M::mm_idx_stat] kmer size: 19; skip: 10; is_hpc: 1; #seq: 1\n",
      "[M::mm_idx_stat::0.081*1.00] distinct minimizers: 521853 (99.34% are singletons); average occurrences: 1.014; average spacing: 7.460; total length: 3949534\n",
      "[M::worker_pipeline::0.614*0.99] mapped 1 sequences\n",
      "[M::main] Version: 2.19-r1057\n",
      "[M::main] CMD: minimap2 -ax asm20 --eqx -H -f 100 -r1k,10k --rmq=no -t 1 exp/multiple_runs11/tmp/L04-1/is_deleted/L04-1_genome.deleted_IS.1.fasta exp/multiple_runs11/tmp/L04-1/is_deleted/L04-1_genome.deleted_IS.3.fasta\n",
      "[M::main] Real time: 0.617 sec; CPU: 0.610 sec; Peak RSS: 0.088 GB\n",
      "local_variation - WARNING - Finished syri\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "TLOut.txt  is empty. Skipping analysing it.\n",
      "invTLOut.txt  is empty. Skipping analysing it.\n",
      "ctxOut.txt is empty. Skipping analysing it.\n",
      "TL Out.txt is empty. Skipping analysing it.\n",
      "invTL Out.txt is empty. Skipping analysing it.\n",
      "dup Out.txt is empty. Skipping analysing it.\n",
      "invDup Out.txt is empty. Skipping analysing it.\n",
      "ctxOut.txt is empty. Skipping analysing it.\n",
      "TLOut.txt  is empty. Skipping analysing it.\n",
      "invTLOut.txt  is empty. Skipping analysing it.\n",
      "ctxOut.txt is empty. Skipping analysing it.\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "[M::mm_idx_gen::0.061*1.01] collected minimizers\n",
      "[M::mm_idx_gen::0.080*1.00] sorted minimizers\n",
      "[M::main::0.080*1.00] loaded/built the index for 1 target sequence(s)\n",
      "[M::mm_mapopt_update::0.080*1.00] mid_occ = 100\n",
      "[M::mm_idx_stat] kmer size: 19; skip: 10; is_hpc: 1; #seq: 1\n",
      "[M::mm_idx_stat::0.083*1.00] distinct minimizers: 521853 (99.34% are singletons); average occurrences: 1.014; average spacing: 7.460; total length: 3949534\n",
      "[M::worker_pipeline::0.617*0.99] mapped 1 sequences\n",
      "[M::main] Version: 2.19-r1057\n",
      "[M::main] CMD: minimap2 -ax asm20 --eqx -H -f 100 -r1k,10k --rmq=no -t 1 exp/multiple_runs11/tmp/L04-2/is_deleted/L04-2_genome.deleted_IS.1.fasta exp/multiple_runs11/tmp/L04-2/is_deleted/L04-2_genome.deleted_IS.2.fasta\n",
      "[M::main] Real time: 0.620 sec; CPU: 0.614 sec; Peak RSS: 0.088 GB\n",
      "local_variation - WARNING - Finished syri\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "invOut.txt  is empty. Skipping analysing it.\n",
      "TLOut.txt  is empty. Skipping analysing it.\n",
      "invTLOut.txt  is empty. Skipping analysing it.\n",
      "ctxOut.txt is empty. Skipping analysing it.\n",
      "inv Out.txt is empty. Skipping analysing it.\n",
      "TL Out.txt is empty. Skipping analysing it.\n",
      "invTL Out.txt is empty. Skipping analysing it.\n",
      "dup Out.txt is empty. Skipping analysing it.\n",
      "invDup Out.txt is empty. Skipping analysing it.\n",
      "ctxOut.txt is empty. Skipping analysing it.\n",
      "invOut.txt  is empty. Skipping analysing it.\n",
      "TLOut.txt  is empty. Skipping analysing it.\n",
      "invTLOut.txt  is empty. Skipping analysing it.\n",
      "ctxOut.txt is empty. Skipping analysing it.\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "[M::mm_idx_gen::0.062*1.01] collected minimizers\n",
      "[M::mm_idx_gen::0.081*1.00] sorted minimizers\n",
      "[M::main::0.081*1.00] loaded/built the index for 1 target sequence(s)\n",
      "[M::mm_mapopt_update::0.081*1.00] mid_occ = 100\n",
      "[M::mm_idx_stat] kmer size: 19; skip: 10; is_hpc: 1; #seq: 1\n",
      "[M::mm_idx_stat::0.084*1.00] distinct minimizers: 521853 (99.34% are singletons); average occurrences: 1.014; average spacing: 7.460; total length: 3949534\n",
      "[M::worker_pipeline::0.617*0.99] mapped 1 sequences\n",
      "[M::main] Version: 2.19-r1057\n",
      "[M::main] CMD: minimap2 -ax asm20 --eqx -H -f 100 -r1k,10k --rmq=no -t 1 exp/multiple_runs11/tmp/L04-2/is_deleted/L04-2_genome.deleted_IS.1.fasta exp/multiple_runs11/tmp/L04-2/is_deleted/L04-2_genome.deleted_IS.3.fasta\n",
      "[M::main] Real time: 0.621 sec; CPU: 0.614 sec; Peak RSS: 0.088 GB\n",
      "local_variation - WARNING - Finished syri\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "TLOut.txt  is empty. Skipping analysing it.\n",
      "invTLOut.txt  is empty. Skipping analysing it.\n",
      "ctxOut.txt is empty. Skipping analysing it.\n",
      "TL Out.txt is empty. Skipping analysing it.\n",
      "invTL Out.txt is empty. Skipping analysing it.\n",
      "dup Out.txt is empty. Skipping analysing it.\n",
      "invDup Out.txt is empty. Skipping analysing it.\n",
      "ctxOut.txt is empty. Skipping analysing it.\n",
      "TLOut.txt  is empty. Skipping analysing it.\n",
      "invTLOut.txt  is empty. Skipping analysing it.\n",
      "ctxOut.txt is empty. Skipping analysing it.\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "[M::mm_idx_gen::0.063*1.01] collected minimizers\n",
      "[M::mm_idx_gen::0.082*1.00] sorted minimizers\n",
      "[M::main::0.082*1.00] loaded/built the index for 1 target sequence(s)\n",
      "[M::mm_mapopt_update::0.082*1.00] mid_occ = 100\n",
      "[M::mm_idx_stat] kmer size: 19; skip: 10; is_hpc: 1; #seq: 1\n",
      "[M::mm_idx_stat::0.085*1.00] distinct minimizers: 521853 (99.34% are singletons); average occurrences: 1.014; average spacing: 7.460; total length: 3949534\n",
      "[M::worker_pipeline::0.668*0.99] mapped 1 sequences\n",
      "[M::main] Version: 2.19-r1057\n",
      "[M::main] CMD: minimap2 -ax asm20 --eqx -H -f 100 -r1k,10k --rmq=no -t 1 exp/multiple_runs11/tmp/L04-3/is_deleted/L04-3_genome.deleted_IS.1.fasta exp/multiple_runs11/tmp/L04-3/is_deleted/L04-3_genome.deleted_IS.2.fasta\n",
      "[M::main] Real time: 0.671 sec; CPU: 0.664 sec; Peak RSS: 0.088 GB\n",
      "local_variation - WARNING - Finished syri\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "invOut.txt  is empty. Skipping analysing it.\n",
      "TLOut.txt  is empty. Skipping analysing it.\n",
      "invTLOut.txt  is empty. Skipping analysing it.\n",
      "ctxOut.txt is empty. Skipping analysing it.\n",
      "inv Out.txt is empty. Skipping analysing it.\n",
      "TL Out.txt is empty. Skipping analysing it.\n",
      "invTL Out.txt is empty. Skipping analysing it.\n",
      "ctxOut.txt is empty. Skipping analysing it.\n",
      "invOut.txt  is empty. Skipping analysing it.\n",
      "TLOut.txt  is empty. Skipping analysing it.\n",
      "invTLOut.txt  is empty. Skipping analysing it.\n",
      "ctxOut.txt is empty. Skipping analysing it.\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "[M::mm_idx_gen::0.062*1.01] collected minimizers\n",
      "[M::mm_idx_gen::0.080*1.00] sorted minimizers\n",
      "[M::main::0.080*1.00] loaded/built the index for 1 target sequence(s)\n",
      "[M::mm_mapopt_update::0.080*1.00] mid_occ = 100\n",
      "[M::mm_idx_stat] kmer size: 19; skip: 10; is_hpc: 1; #seq: 1\n",
      "[M::mm_idx_stat::0.084*1.00] distinct minimizers: 521853 (99.34% are singletons); average occurrences: 1.014; average spacing: 7.460; total length: 3949534\n",
      "[M::worker_pipeline::0.643*0.99] mapped 1 sequences\n",
      "[M::main] Version: 2.19-r1057\n",
      "[M::main] CMD: minimap2 -ax asm20 --eqx -H -f 100 -r1k,10k --rmq=no -t 1 exp/multiple_runs11/tmp/L04-3/is_deleted/L04-3_genome.deleted_IS.1.fasta exp/multiple_runs11/tmp/L04-3/is_deleted/L04-3_genome.deleted_IS.3.fasta\n",
      "[M::main] Real time: 0.645 sec; CPU: 0.639 sec; Peak RSS: 0.088 GB\n",
      "local_variation - WARNING - Finished syri\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "TLOut.txt  is empty. Skipping analysing it.\n",
      "invTLOut.txt  is empty. Skipping analysing it.\n",
      "ctxOut.txt is empty. Skipping analysing it.\n",
      "TL Out.txt is empty. Skipping analysing it.\n",
      "invTL Out.txt is empty. Skipping analysing it.\n",
      "invDup Out.txt is empty. Skipping analysing it.\n",
      "ctxOut.txt is empty. Skipping analysing it.\n",
      "TLOut.txt  is empty. Skipping analysing it.\n",
      "invTLOut.txt  is empty. Skipping analysing it.\n",
      "ctxOut.txt is empty. Skipping analysing it.\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "[M::mm_idx_gen::0.064*1.01] collected minimizers\n",
      "[M::mm_idx_gen::0.083*1.01] sorted minimizers\n",
      "[M::main::0.083*1.01] loaded/built the index for 1 target sequence(s)\n",
      "[M::mm_mapopt_update::0.083*1.01] mid_occ = 100\n",
      "[M::mm_idx_stat] kmer size: 19; skip: 10; is_hpc: 1; #seq: 1\n",
      "[M::mm_idx_stat::0.086*1.01] distinct minimizers: 521853 (99.34% are singletons); average occurrences: 1.014; average spacing: 7.460; total length: 3949534\n",
      "[M::worker_pipeline::0.626*0.99] mapped 1 sequences\n",
      "[M::main] Version: 2.19-r1057\n",
      "[M::main] CMD: minimap2 -ax asm20 --eqx -H -f 100 -r1k,10k --rmq=no -t 1 exp/multiple_runs11/tmp/L04-4/is_deleted/L04-4_genome.deleted_IS.1.fasta exp/multiple_runs11/tmp/L04-4/is_deleted/L04-4_genome.deleted_IS.2.fasta\n",
      "[M::main] Real time: 0.629 sec; CPU: 0.622 sec; Peak RSS: 0.087 GB\n",
      "local_variation - WARNING - Finished syri\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "invOut.txt  is empty. Skipping analysing it.\n",
      "TLOut.txt  is empty. Skipping analysing it.\n",
      "invTLOut.txt  is empty. Skipping analysing it.\n",
      "ctxOut.txt is empty. Skipping analysing it.\n",
      "inv Out.txt is empty. Skipping analysing it.\n",
      "TL Out.txt is empty. Skipping analysing it.\n",
      "invTL Out.txt is empty. Skipping analysing it.\n",
      "dup Out.txt is empty. Skipping analysing it.\n",
      "ctxOut.txt is empty. Skipping analysing it.\n",
      "invOut.txt  is empty. Skipping analysing it.\n",
      "TLOut.txt  is empty. Skipping analysing it.\n",
      "invTLOut.txt  is empty. Skipping analysing it.\n",
      "ctxOut.txt is empty. Skipping analysing it.\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "[M::mm_idx_gen::0.058*1.01] collected minimizers\n",
      "[M::mm_idx_gen::0.076*1.00] sorted minimizers\n",
      "[M::main::0.076*1.00] loaded/built the index for 1 target sequence(s)\n",
      "[M::mm_mapopt_update::0.076*1.00] mid_occ = 100\n",
      "[M::mm_idx_stat] kmer size: 19; skip: 10; is_hpc: 1; #seq: 1\n",
      "[M::mm_idx_stat::0.080*1.00] distinct minimizers: 521853 (99.34% are singletons); average occurrences: 1.014; average spacing: 7.460; total length: 3949534\n",
      "[M::worker_pipeline::0.631*0.99] mapped 1 sequences\n",
      "[M::main] Version: 2.19-r1057\n",
      "[M::main] CMD: minimap2 -ax asm20 --eqx -H -f 100 -r1k,10k --rmq=no -t 1 exp/multiple_runs11/tmp/L04-4/is_deleted/L04-4_genome.deleted_IS.1.fasta exp/multiple_runs11/tmp/L04-4/is_deleted/L04-4_genome.deleted_IS.3.fasta\n",
      "[M::main] Real time: 0.633 sec; CPU: 0.626 sec; Peak RSS: 0.089 GB\n",
      "local_variation - WARNING - Finished syri\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "TLOut.txt  is empty. Skipping analysing it.\n",
      "invTLOut.txt  is empty. Skipping analysing it.\n",
      "ctxOut.txt is empty. Skipping analysing it.\n",
      "TL Out.txt is empty. Skipping analysing it.\n",
      "invTL Out.txt is empty. Skipping analysing it.\n",
      "ctxOut.txt is empty. Skipping analysing it.\n",
      "TLOut.txt  is empty. Skipping analysing it.\n",
      "invTLOut.txt  is empty. Skipping analysing it.\n",
      "ctxOut.txt is empty. Skipping analysing it.\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "[M::mm_idx_gen::0.059*1.01] collected minimizers\n",
      "[M::mm_idx_gen::0.077*1.00] sorted minimizers\n",
      "[M::main::0.077*1.00] loaded/built the index for 1 target sequence(s)\n",
      "[M::mm_mapopt_update::0.077*1.00] mid_occ = 100\n",
      "[M::mm_idx_stat] kmer size: 19; skip: 10; is_hpc: 1; #seq: 1\n",
      "[M::mm_idx_stat::0.081*1.00] distinct minimizers: 517453 (99.34% are singletons); average occurrences: 1.015; average spacing: 7.460; total length: 3916230\n",
      "[M::worker_pipeline::0.617*0.99] mapped 1 sequences\n",
      "[M::main] Version: 2.19-r1057\n",
      "[M::main] CMD: minimap2 -ax asm20 --eqx -H -f 100 -r1k,10k --rmq=no -t 1 exp/multiple_runs11/tmp/L05-1/is_deleted/L05-1_genome.deleted_IS.1.fasta exp/multiple_runs11/tmp/L05-1/is_deleted/L05-1_genome.deleted_IS.2.fasta\n",
      "[M::main] Real time: 0.620 sec; CPU: 0.614 sec; Peak RSS: 0.085 GB\n",
      "local_variation - WARNING - Finished syri\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "invOut.txt  is empty. Skipping analysing it.\n",
      "TLOut.txt  is empty. Skipping analysing it.\n",
      "invTLOut.txt  is empty. Skipping analysing it.\n",
      "ctxOut.txt is empty. Skipping analysing it.\n",
      "inv Out.txt is empty. Skipping analysing it.\n",
      "TL Out.txt is empty. Skipping analysing it.\n",
      "invTL Out.txt is empty. Skipping analysing it.\n",
      "dup Out.txt is empty. Skipping analysing it.\n",
      "invDup Out.txt is empty. Skipping analysing it.\n",
      "ctxOut.txt is empty. Skipping analysing it.\n",
      "invOut.txt  is empty. Skipping analysing it.\n",
      "TLOut.txt  is empty. Skipping analysing it.\n",
      "invTLOut.txt  is empty. Skipping analysing it.\n",
      "ctxOut.txt is empty. Skipping analysing it.\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "[M::mm_idx_gen::0.058*1.01] collected minimizers\n",
      "[M::mm_idx_gen::0.076*1.00] sorted minimizers\n",
      "[M::main::0.076*1.00] loaded/built the index for 1 target sequence(s)\n",
      "[M::mm_mapopt_update::0.076*1.00] mid_occ = 100\n",
      "[M::mm_idx_stat] kmer size: 19; skip: 10; is_hpc: 1; #seq: 1\n",
      "[M::mm_idx_stat::0.080*1.00] distinct minimizers: 517453 (99.34% are singletons); average occurrences: 1.015; average spacing: 7.460; total length: 3916230\n",
      "[M::worker_pipeline::0.610*0.99] mapped 1 sequences\n",
      "[M::main] Version: 2.19-r1057\n",
      "[M::main] CMD: minimap2 -ax asm20 --eqx -H -f 100 -r1k,10k --rmq=no -t 1 exp/multiple_runs11/tmp/L05-1/is_deleted/L05-1_genome.deleted_IS.1.fasta exp/multiple_runs11/tmp/L05-1/is_deleted/L05-1_genome.deleted_IS.3.fasta\n",
      "[M::main] Real time: 0.614 sec; CPU: 0.607 sec; Peak RSS: 0.087 GB\n",
      "local_variation - WARNING - Finished syri\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "invOut.txt  is empty. Skipping analysing it.\n",
      "TLOut.txt  is empty. Skipping analysing it.\n",
      "invTLOut.txt  is empty. Skipping analysing it.\n",
      "ctxOut.txt is empty. Skipping analysing it.\n",
      "inv Out.txt is empty. Skipping analysing it.\n",
      "TL Out.txt is empty. Skipping analysing it.\n",
      "invTL Out.txt is empty. Skipping analysing it.\n",
      "dup Out.txt is empty. Skipping analysing it.\n",
      "invDup Out.txt is empty. Skipping analysing it.\n",
      "ctxOut.txt is empty. Skipping analysing it.\n",
      "invOut.txt  is empty. Skipping analysing it.\n",
      "TLOut.txt  is empty. Skipping analysing it.\n",
      "invTLOut.txt  is empty. Skipping analysing it.\n",
      "ctxOut.txt is empty. Skipping analysing it.\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "[M::mm_idx_gen::0.059*1.01] collected minimizers\n",
      "[M::mm_idx_gen::0.078*1.00] sorted minimizers\n",
      "[M::main::0.078*1.00] loaded/built the index for 1 target sequence(s)\n",
      "[M::mm_mapopt_update::0.078*1.00] mid_occ = 100\n",
      "[M::mm_idx_stat] kmer size: 19; skip: 10; is_hpc: 1; #seq: 1\n",
      "[M::mm_idx_stat::0.082*1.00] distinct minimizers: 517505 (99.34% are singletons); average occurrences: 1.015; average spacing: 7.460; total length: 3916618\n",
      "[M::worker_pipeline::0.613*0.99] mapped 1 sequences\n",
      "[M::main] Version: 2.19-r1057\n",
      "[M::main] CMD: minimap2 -ax asm20 --eqx -H -f 100 -r1k,10k --rmq=no -t 1 exp/multiple_runs11/tmp/L05-2/is_deleted/L05-2_genome.deleted_IS.1.fasta exp/multiple_runs11/tmp/L05-2/is_deleted/L05-2_genome.deleted_IS.2.fasta\n",
      "[M::main] Real time: 0.616 sec; CPU: 0.609 sec; Peak RSS: 0.086 GB\n",
      "local_variation - WARNING - Finished syri\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "TLOut.txt  is empty. Skipping analysing it.\n",
      "ctxOut.txt is empty. Skipping analysing it.\n",
      "TL Out.txt is empty. Skipping analysing it.\n",
      "dup Out.txt is empty. Skipping analysing it.\n",
      "invDup Out.txt is empty. Skipping analysing it.\n",
      "ctxOut.txt is empty. Skipping analysing it.\n",
      "TLOut.txt  is empty. Skipping analysing it.\n",
      "ctxOut.txt is empty. Skipping analysing it.\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "[M::mm_idx_gen::0.059*1.01] collected minimizers\n",
      "[M::mm_idx_gen::0.078*1.00] sorted minimizers\n",
      "[M::main::0.078*1.00] loaded/built the index for 1 target sequence(s)\n",
      "[M::mm_mapopt_update::0.078*1.00] mid_occ = 100\n",
      "[M::mm_idx_stat] kmer size: 19; skip: 10; is_hpc: 1; #seq: 1\n",
      "[M::mm_idx_stat::0.082*1.00] distinct minimizers: 517505 (99.34% are singletons); average occurrences: 1.015; average spacing: 7.460; total length: 3916618\n",
      "[M::worker_pipeline::0.609*0.99] mapped 1 sequences\n",
      "[M::main] Version: 2.19-r1057\n",
      "[M::main] CMD: minimap2 -ax asm20 --eqx -H -f 100 -r1k,10k --rmq=no -t 1 exp/multiple_runs11/tmp/L05-2/is_deleted/L05-2_genome.deleted_IS.1.fasta exp/multiple_runs11/tmp/L05-2/is_deleted/L05-2_genome.deleted_IS.3.fasta\n",
      "[M::main] Real time: 0.612 sec; CPU: 0.606 sec; Peak RSS: 0.085 GB\n",
      "local_variation - WARNING - Finished syri\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "TLOut.txt  is empty. Skipping analysing it.\n",
      "invTLOut.txt  is empty. Skipping analysing it.\n",
      "ctxOut.txt is empty. Skipping analysing it.\n",
      "TL Out.txt is empty. Skipping analysing it.\n",
      "invTL Out.txt is empty. Skipping analysing it.\n",
      "invDup Out.txt is empty. Skipping analysing it.\n",
      "ctxOut.txt is empty. Skipping analysing it.\n",
      "TLOut.txt  is empty. Skipping analysing it.\n",
      "invTLOut.txt  is empty. Skipping analysing it.\n",
      "ctxOut.txt is empty. Skipping analysing it.\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "[M::mm_idx_gen::0.058*1.01] collected minimizers\n",
      "[M::mm_idx_gen::0.076*1.00] sorted minimizers\n",
      "[M::main::0.076*1.00] loaded/built the index for 1 target sequence(s)\n",
      "[M::mm_mapopt_update::0.076*1.00] mid_occ = 100\n",
      "[M::mm_idx_stat] kmer size: 19; skip: 10; is_hpc: 1; #seq: 1\n",
      "[M::mm_idx_stat::0.080*1.00] distinct minimizers: 517505 (99.34% are singletons); average occurrences: 1.015; average spacing: 7.460; total length: 3916618\n",
      "[M::worker_pipeline::0.599*0.99] mapped 1 sequences\n",
      "[M::main] Version: 2.19-r1057\n",
      "[M::main] CMD: minimap2 -ax asm20 --eqx -H -f 100 -r1k,10k --rmq=no -t 1 exp/multiple_runs11/tmp/L05-3/is_deleted/L05-3_genome.deleted_IS.1.fasta exp/multiple_runs11/tmp/L05-3/is_deleted/L05-3_genome.deleted_IS.2.fasta\n",
      "[M::main] Real time: 0.602 sec; CPU: 0.596 sec; Peak RSS: 0.085 GB\n",
      "local_variation - WARNING - Finished syri\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "TLOut.txt  is empty. Skipping analysing it.\n",
      "invTLOut.txt  is empty. Skipping analysing it.\n",
      "ctxOut.txt is empty. Skipping analysing it.\n",
      "TL Out.txt is empty. Skipping analysing it.\n",
      "invTL Out.txt is empty. Skipping analysing it.\n",
      "dup Out.txt is empty. Skipping analysing it.\n",
      "invDup Out.txt is empty. Skipping analysing it.\n",
      "ctxOut.txt is empty. Skipping analysing it.\n",
      "TLOut.txt  is empty. Skipping analysing it.\n",
      "invTLOut.txt  is empty. Skipping analysing it.\n",
      "ctxOut.txt is empty. Skipping analysing it.\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "[M::mm_idx_gen::0.057*1.01] collected minimizers\n",
      "[M::mm_idx_gen::0.075*1.00] sorted minimizers\n",
      "[M::main::0.075*1.00] loaded/built the index for 1 target sequence(s)\n",
      "[M::mm_mapopt_update::0.075*1.01] mid_occ = 100\n",
      "[M::mm_idx_stat] kmer size: 19; skip: 10; is_hpc: 1; #seq: 1\n",
      "[M::mm_idx_stat::0.079*1.00] distinct minimizers: 517505 (99.34% are singletons); average occurrences: 1.015; average spacing: 7.460; total length: 3916618\n",
      "[M::worker_pipeline::0.587*0.99] mapped 1 sequences\n",
      "[M::main] Version: 2.19-r1057\n",
      "[M::main] CMD: minimap2 -ax asm20 --eqx -H -f 100 -r1k,10k --rmq=no -t 1 exp/multiple_runs11/tmp/L05-3/is_deleted/L05-3_genome.deleted_IS.1.fasta exp/multiple_runs11/tmp/L05-3/is_deleted/L05-3_genome.deleted_IS.3.fasta\n",
      "[M::main] Real time: 0.590 sec; CPU: 0.584 sec; Peak RSS: 0.088 GB\n",
      "local_variation - WARNING - Finished syri\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "TLOut.txt  is empty. Skipping analysing it.\n",
      "invTLOut.txt  is empty. Skipping analysing it.\n",
      "ctxOut.txt is empty. Skipping analysing it.\n",
      "TL Out.txt is empty. Skipping analysing it.\n",
      "invTL Out.txt is empty. Skipping analysing it.\n",
      "dup Out.txt is empty. Skipping analysing it.\n",
      "invDup Out.txt is empty. Skipping analysing it.\n",
      "ctxOut.txt is empty. Skipping analysing it.\n",
      "TLOut.txt  is empty. Skipping analysing it.\n",
      "invTLOut.txt  is empty. Skipping analysing it.\n",
      "ctxOut.txt is empty. Skipping analysing it.\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "[M::mm_idx_gen::0.062*1.01] collected minimizers\n",
      "[M::mm_idx_gen::0.081*1.00] sorted minimizers\n",
      "[M::main::0.081*1.00] loaded/built the index for 1 target sequence(s)\n",
      "[M::mm_mapopt_update::0.081*1.00] mid_occ = 100\n",
      "[M::mm_idx_stat] kmer size: 19; skip: 10; is_hpc: 1; #seq: 1\n",
      "[M::mm_idx_stat::0.085*1.00] distinct minimizers: 517453 (99.34% are singletons); average occurrences: 1.015; average spacing: 7.460; total length: 3916230\n",
      "[M::worker_pipeline::0.626*0.99] mapped 1 sequences\n",
      "[M::main] Version: 2.19-r1057\n",
      "[M::main] CMD: minimap2 -ax asm20 --eqx -H -f 100 -r1k,10k --rmq=no -t 1 exp/multiple_runs11/tmp/L05-4/is_deleted/L05-4_genome.deleted_IS.1.fasta exp/multiple_runs11/tmp/L05-4/is_deleted/L05-4_genome.deleted_IS.2.fasta\n",
      "[M::main] Real time: 0.628 sec; CPU: 0.622 sec; Peak RSS: 0.087 GB\n",
      "Reading Coords - WARNING - Reference chromosome contig_1 has high fraction of inverted alignments with its homologous chromosome in the query genome (contig_1). Ensure that same chromosome-strands are being compared in the two genomes, as different strand can result in unexpected errors.\n",
      "local_variation - WARNING - Finished syri\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "TLOut.txt  is empty. Skipping analysing it.\n",
      "invTLOut.txt  is empty. Skipping analysing it.\n",
      "ctxOut.txt is empty. Skipping analysing it.\n",
      "TL Out.txt is empty. Skipping analysing it.\n",
      "invTL Out.txt is empty. Skipping analysing it.\n",
      "ctxOut.txt is empty. Skipping analysing it.\n",
      "TLOut.txt  is empty. Skipping analysing it.\n",
      "invTLOut.txt  is empty. Skipping analysing it.\n",
      "ctxOut.txt is empty. Skipping analysing it.\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "[M::mm_idx_gen::0.062*1.01] collected minimizers\n",
      "[M::mm_idx_gen::0.081*1.00] sorted minimizers\n",
      "[M::main::0.081*1.00] loaded/built the index for 1 target sequence(s)\n",
      "[M::mm_mapopt_update::0.081*1.00] mid_occ = 100\n",
      "[M::mm_idx_stat] kmer size: 19; skip: 10; is_hpc: 1; #seq: 1\n",
      "[M::mm_idx_stat::0.084*1.00] distinct minimizers: 517453 (99.34% are singletons); average occurrences: 1.015; average spacing: 7.460; total length: 3916230\n",
      "[M::worker_pipeline::0.603*0.99] mapped 1 sequences\n",
      "[M::main] Version: 2.19-r1057\n",
      "[M::main] CMD: minimap2 -ax asm20 --eqx -H -f 100 -r1k,10k --rmq=no -t 1 exp/multiple_runs11/tmp/L05-4/is_deleted/L05-4_genome.deleted_IS.1.fasta exp/multiple_runs11/tmp/L05-4/is_deleted/L05-4_genome.deleted_IS.3.fasta\n",
      "[M::main] Real time: 0.605 sec; CPU: 0.599 sec; Peak RSS: 0.087 GB\n",
      "local_variation - WARNING - Finished syri\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "TLOut.txt  is empty. Skipping analysing it.\n",
      "invTLOut.txt  is empty. Skipping analysing it.\n",
      "ctxOut.txt is empty. Skipping analysing it.\n",
      "TL Out.txt is empty. Skipping analysing it.\n",
      "invTL Out.txt is empty. Skipping analysing it.\n",
      "ctxOut.txt is empty. Skipping analysing it.\n",
      "TLOut.txt  is empty. Skipping analysing it.\n",
      "invTLOut.txt  is empty. Skipping analysing it.\n",
      "ctxOut.txt is empty. Skipping analysing it.\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "[M::mm_idx_gen::0.057*1.01] collected minimizers\n",
      "[M::mm_idx_gen::0.075*1.00] sorted minimizers\n",
      "[M::main::0.075*1.00] loaded/built the index for 1 target sequence(s)\n",
      "[M::mm_mapopt_update::0.075*1.00] mid_occ = 100\n",
      "[M::mm_idx_stat] kmer size: 19; skip: 10; is_hpc: 1; #seq: 1\n",
      "[M::mm_idx_stat::0.079*1.00] distinct minimizers: 521558 (99.33% are singletons); average occurrences: 1.015; average spacing: 7.460; total length: 3947678\n",
      "[M::worker_pipeline::0.606*0.99] mapped 1 sequences\n",
      "[M::main] Version: 2.19-r1057\n",
      "[M::main] CMD: minimap2 -ax asm20 --eqx -H -f 100 -r1k,10k --rmq=no -t 1 exp/multiple_runs11/tmp/L06-1/is_deleted/L06-1_genome.deleted_IS.1.fasta exp/multiple_runs11/tmp/L06-1/is_deleted/L06-1_genome.deleted_IS.2.fasta\n",
      "[M::main] Real time: 0.609 sec; CPU: 0.603 sec; Peak RSS: 0.087 GB\n",
      "local_variation - WARNING - Finished syri\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "TLOut.txt  is empty. Skipping analysing it.\n",
      "invTLOut.txt  is empty. Skipping analysing it.\n",
      "ctxOut.txt is empty. Skipping analysing it.\n",
      "TL Out.txt is empty. Skipping analysing it.\n",
      "invTL Out.txt is empty. Skipping analysing it.\n",
      "dup Out.txt is empty. Skipping analysing it.\n",
      "invDup Out.txt is empty. Skipping analysing it.\n",
      "ctxOut.txt is empty. Skipping analysing it.\n",
      "TLOut.txt  is empty. Skipping analysing it.\n",
      "invTLOut.txt  is empty. Skipping analysing it.\n",
      "ctxOut.txt is empty. Skipping analysing it.\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "[M::mm_idx_gen::0.058*1.01] collected minimizers\n",
      "[M::mm_idx_gen::0.076*1.00] sorted minimizers\n",
      "[M::main::0.076*1.00] loaded/built the index for 1 target sequence(s)\n",
      "[M::mm_mapopt_update::0.076*1.00] mid_occ = 100\n",
      "[M::mm_idx_stat] kmer size: 19; skip: 10; is_hpc: 1; #seq: 1\n",
      "[M::mm_idx_stat::0.080*1.00] distinct minimizers: 521558 (99.33% are singletons); average occurrences: 1.015; average spacing: 7.460; total length: 3947678\n",
      "[M::worker_pipeline::0.615*0.99] mapped 1 sequences\n",
      "[M::main] Version: 2.19-r1057\n",
      "[M::main] CMD: minimap2 -ax asm20 --eqx -H -f 100 -r1k,10k --rmq=no -t 1 exp/multiple_runs11/tmp/L06-1/is_deleted/L06-1_genome.deleted_IS.1.fasta exp/multiple_runs11/tmp/L06-1/is_deleted/L06-1_genome.deleted_IS.3.fasta\n",
      "[M::main] Real time: 0.618 sec; CPU: 0.612 sec; Peak RSS: 0.087 GB\n",
      "local_variation - WARNING - Finished syri\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "invTLOut.txt  is empty. Skipping analysing it.\n",
      "ctxOut.txt is empty. Skipping analysing it.\n",
      "invTL Out.txt is empty. Skipping analysing it.\n",
      "dup Out.txt is empty. Skipping analysing it.\n",
      "invDup Out.txt is empty. Skipping analysing it.\n",
      "ctxOut.txt is empty. Skipping analysing it.\n",
      "invTLOut.txt  is empty. Skipping analysing it.\n",
      "ctxOut.txt is empty. Skipping analysing it.\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "[M::mm_idx_gen::0.058*1.01] collected minimizers\n",
      "[M::mm_idx_gen::0.076*1.00] sorted minimizers\n",
      "[M::main::0.077*1.00] loaded/built the index for 1 target sequence(s)\n",
      "[M::mm_mapopt_update::0.077*1.00] mid_occ = 100\n",
      "[M::mm_idx_stat] kmer size: 19; skip: 10; is_hpc: 1; #seq: 1\n",
      "[M::mm_idx_stat::0.080*1.00] distinct minimizers: 521558 (99.33% are singletons); average occurrences: 1.015; average spacing: 7.460; total length: 3947678\n",
      "[M::worker_pipeline::0.701*0.99] mapped 1 sequences\n",
      "[M::main] Version: 2.19-r1057\n",
      "[M::main] CMD: minimap2 -ax asm20 --eqx -H -f 100 -r1k,10k --rmq=no -t 1 exp/multiple_runs11/tmp/L06-2/is_deleted/L06-2_genome.deleted_IS.1.fasta exp/multiple_runs11/tmp/L06-2/is_deleted/L06-2_genome.deleted_IS.2.fasta\n",
      "[M::main] Real time: 0.703 sec; CPU: 0.697 sec; Peak RSS: 0.215 GB\n",
      "local_variation - WARNING - Finished syri\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "TLOut.txt  is empty. Skipping analysing it.\n",
      "invTLOut.txt  is empty. Skipping analysing it.\n",
      "ctxOut.txt is empty. Skipping analysing it.\n",
      "TL Out.txt is empty. Skipping analysing it.\n",
      "invTL Out.txt is empty. Skipping analysing it.\n",
      "dup Out.txt is empty. Skipping analysing it.\n",
      "invDup Out.txt is empty. Skipping analysing it.\n",
      "ctxOut.txt is empty. Skipping analysing it.\n",
      "TLOut.txt  is empty. Skipping analysing it.\n",
      "invTLOut.txt  is empty. Skipping analysing it.\n",
      "ctxOut.txt is empty. Skipping analysing it.\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "[M::mm_idx_gen::0.063*1.01] collected minimizers\n",
      "[M::mm_idx_gen::0.082*1.00] sorted minimizers\n",
      "[M::main::0.082*1.00] loaded/built the index for 1 target sequence(s)\n",
      "[M::mm_mapopt_update::0.082*1.00] mid_occ = 100\n",
      "[M::mm_idx_stat] kmer size: 19; skip: 10; is_hpc: 1; #seq: 1\n",
      "[M::mm_idx_stat::0.085*1.00] distinct minimizers: 521558 (99.33% are singletons); average occurrences: 1.015; average spacing: 7.460; total length: 3947678\n",
      "[M::worker_pipeline::0.619*0.99] mapped 1 sequences\n",
      "[M::main] Version: 2.19-r1057\n",
      "[M::main] CMD: minimap2 -ax asm20 --eqx -H -f 100 -r1k,10k --rmq=no -t 1 exp/multiple_runs11/tmp/L06-2/is_deleted/L06-2_genome.deleted_IS.1.fasta exp/multiple_runs11/tmp/L06-2/is_deleted/L06-2_genome.deleted_IS.3.fasta\n",
      "[M::main] Real time: 0.622 sec; CPU: 0.615 sec; Peak RSS: 0.089 GB\n",
      "local_variation - WARNING - Finished syri\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "TLOut.txt  is empty. Skipping analysing it.\n",
      "ctxOut.txt is empty. Skipping analysing it.\n",
      "TL Out.txt is empty. Skipping analysing it.\n",
      "dup Out.txt is empty. Skipping analysing it.\n",
      "invDup Out.txt is empty. Skipping analysing it.\n",
      "ctxOut.txt is empty. Skipping analysing it.\n",
      "TLOut.txt  is empty. Skipping analysing it.\n",
      "ctxOut.txt is empty. Skipping analysing it.\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "[M::mm_idx_gen::0.064*1.01] collected minimizers\n",
      "[M::mm_idx_gen::0.082*1.00] sorted minimizers\n",
      "[M::main::0.082*1.00] loaded/built the index for 1 target sequence(s)\n",
      "[M::mm_mapopt_update::0.082*1.00] mid_occ = 100\n",
      "[M::mm_idx_stat] kmer size: 19; skip: 10; is_hpc: 1; #seq: 1\n",
      "[M::mm_idx_stat::0.086*1.00] distinct minimizers: 521558 (99.33% are singletons); average occurrences: 1.015; average spacing: 7.460; total length: 3947678\n",
      "[M::worker_pipeline::0.622*0.99] mapped 1 sequences\n",
      "[M::main] Version: 2.19-r1057\n",
      "[M::main] CMD: minimap2 -ax asm20 --eqx -H -f 100 -r1k,10k --rmq=no -t 1 exp/multiple_runs11/tmp/L06-3/is_deleted/L06-3_genome.deleted_IS.1.fasta exp/multiple_runs11/tmp/L06-3/is_deleted/L06-3_genome.deleted_IS.2.fasta\n",
      "[M::main] Real time: 0.625 sec; CPU: 0.619 sec; Peak RSS: 0.086 GB\n",
      "Reading Coords - WARNING - Reference chromosome contig_1 has high fraction of inverted alignments with its homologous chromosome in the query genome (contig_1). Ensure that same chromosome-strands are being compared in the two genomes, as different strand can result in unexpected errors.\n",
      "local_variation - WARNING - Finished syri\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "TLOut.txt  is empty. Skipping analysing it.\n",
      "invTLOut.txt  is empty. Skipping analysing it.\n",
      "ctxOut.txt is empty. Skipping analysing it.\n",
      "TL Out.txt is empty. Skipping analysing it.\n",
      "invTL Out.txt is empty. Skipping analysing it.\n",
      "dup Out.txt is empty. Skipping analysing it.\n",
      "invDup Out.txt is empty. Skipping analysing it.\n",
      "ctxOut.txt is empty. Skipping analysing it.\n",
      "TLOut.txt  is empty. Skipping analysing it.\n",
      "invTLOut.txt  is empty. Skipping analysing it.\n",
      "ctxOut.txt is empty. Skipping analysing it.\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "[M::mm_idx_gen::0.057*1.01] collected minimizers\n",
      "[M::mm_idx_gen::0.076*1.00] sorted minimizers\n",
      "[M::main::0.076*1.00] loaded/built the index for 1 target sequence(s)\n",
      "[M::mm_mapopt_update::0.076*1.00] mid_occ = 100\n",
      "[M::mm_idx_stat] kmer size: 19; skip: 10; is_hpc: 1; #seq: 1\n",
      "[M::mm_idx_stat::0.080*1.00] distinct minimizers: 521558 (99.33% are singletons); average occurrences: 1.015; average spacing: 7.460; total length: 3947678\n",
      "[M::worker_pipeline::0.644*0.99] mapped 1 sequences\n",
      "[M::main] Version: 2.19-r1057\n",
      "[M::main] CMD: minimap2 -ax asm20 --eqx -H -f 100 -r1k,10k --rmq=no -t 1 exp/multiple_runs11/tmp/L06-3/is_deleted/L06-3_genome.deleted_IS.1.fasta exp/multiple_runs11/tmp/L06-3/is_deleted/L06-3_genome.deleted_IS.3.fasta\n",
      "[M::main] Real time: 0.647 sec; CPU: 0.640 sec; Peak RSS: 0.089 GB\n",
      "Reading Coords - WARNING - Reference chromosome contig_1 has high fraction of inverted alignments with its homologous chromosome in the query genome (contig_1). Ensure that same chromosome-strands are being compared in the two genomes, as different strand can result in unexpected errors.\n",
      "local_variation - WARNING - Finished syri\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "TLOut.txt  is empty. Skipping analysing it.\n",
      "invTLOut.txt  is empty. Skipping analysing it.\n",
      "ctxOut.txt is empty. Skipping analysing it.\n",
      "TL Out.txt is empty. Skipping analysing it.\n",
      "invTL Out.txt is empty. Skipping analysing it.\n",
      "dup Out.txt is empty. Skipping analysing it.\n",
      "invDup Out.txt is empty. Skipping analysing it.\n",
      "ctxOut.txt is empty. Skipping analysing it.\n",
      "TLOut.txt  is empty. Skipping analysing it.\n",
      "invTLOut.txt  is empty. Skipping analysing it.\n",
      "ctxOut.txt is empty. Skipping analysing it.\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "[M::mm_idx_gen::0.059*1.01] collected minimizers\n",
      "[M::mm_idx_gen::0.078*1.00] sorted minimizers\n",
      "[M::main::0.078*1.00] loaded/built the index for 1 target sequence(s)\n",
      "[M::mm_mapopt_update::0.078*1.00] mid_occ = 100\n",
      "[M::mm_idx_stat] kmer size: 19; skip: 10; is_hpc: 1; #seq: 1\n",
      "[M::mm_idx_stat::0.082*1.00] distinct minimizers: 521558 (99.33% are singletons); average occurrences: 1.015; average spacing: 7.460; total length: 3947678\n",
      "[M::worker_pipeline::0.614*0.99] mapped 1 sequences\n",
      "[M::main] Version: 2.19-r1057\n",
      "[M::main] CMD: minimap2 -ax asm20 --eqx -H -f 100 -r1k,10k --rmq=no -t 1 exp/multiple_runs11/tmp/L06-4/is_deleted/L06-4_genome.deleted_IS.1.fasta exp/multiple_runs11/tmp/L06-4/is_deleted/L06-4_genome.deleted_IS.2.fasta\n",
      "[M::main] Real time: 0.617 sec; CPU: 0.611 sec; Peak RSS: 0.087 GB\n",
      "local_variation - WARNING - Finished syri\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "TLOut.txt  is empty. Skipping analysing it.\n",
      "invTLOut.txt  is empty. Skipping analysing it.\n",
      "ctxOut.txt is empty. Skipping analysing it.\n",
      "TL Out.txt is empty. Skipping analysing it.\n",
      "invTL Out.txt is empty. Skipping analysing it.\n",
      "dup Out.txt is empty. Skipping analysing it.\n",
      "invDup Out.txt is empty. Skipping analysing it.\n",
      "ctxOut.txt is empty. Skipping analysing it.\n",
      "TLOut.txt  is empty. Skipping analysing it.\n",
      "invTLOut.txt  is empty. Skipping analysing it.\n",
      "ctxOut.txt is empty. Skipping analysing it.\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "[M::mm_idx_gen::0.057*1.01] collected minimizers\n",
      "[M::mm_idx_gen::0.076*1.00] sorted minimizers\n",
      "[M::main::0.076*1.00] loaded/built the index for 1 target sequence(s)\n",
      "[M::mm_mapopt_update::0.076*1.00] mid_occ = 100\n",
      "[M::mm_idx_stat] kmer size: 19; skip: 10; is_hpc: 1; #seq: 1\n",
      "[M::mm_idx_stat::0.080*1.00] distinct minimizers: 521558 (99.33% are singletons); average occurrences: 1.015; average spacing: 7.460; total length: 3947678\n",
      "[M::worker_pipeline::0.604*0.99] mapped 1 sequences\n",
      "[M::main] Version: 2.19-r1057\n",
      "[M::main] CMD: minimap2 -ax asm20 --eqx -H -f 100 -r1k,10k --rmq=no -t 1 exp/multiple_runs11/tmp/L06-4/is_deleted/L06-4_genome.deleted_IS.1.fasta exp/multiple_runs11/tmp/L06-4/is_deleted/L06-4_genome.deleted_IS.3.fasta\n",
      "[M::main] Real time: 0.608 sec; CPU: 0.601 sec; Peak RSS: 0.086 GB\n",
      "local_variation - WARNING - Finished syri\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "TLOut.txt  is empty. Skipping analysing it.\n",
      "invTLOut.txt  is empty. Skipping analysing it.\n",
      "ctxOut.txt is empty. Skipping analysing it.\n",
      "TL Out.txt is empty. Skipping analysing it.\n",
      "invTL Out.txt is empty. Skipping analysing it.\n",
      "dup Out.txt is empty. Skipping analysing it.\n",
      "invDup Out.txt is empty. Skipping analysing it.\n",
      "ctxOut.txt is empty. Skipping analysing it.\n",
      "TLOut.txt  is empty. Skipping analysing it.\n",
      "invTLOut.txt  is empty. Skipping analysing it.\n",
      "ctxOut.txt is empty. Skipping analysing it.\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "[M::mm_idx_gen::0.062*1.01] collected minimizers\n",
      "[M::mm_idx_gen::0.081*1.00] sorted minimizers\n",
      "[M::main::0.081*1.00] loaded/built the index for 1 target sequence(s)\n",
      "[M::mm_mapopt_update::0.081*1.00] mid_occ = 100\n",
      "[M::mm_idx_stat] kmer size: 19; skip: 10; is_hpc: 1; #seq: 1\n",
      "[M::mm_idx_stat::0.085*1.00] distinct minimizers: 520925 (99.33% are singletons); average occurrences: 1.015; average spacing: 7.459; total length: 3941898\n",
      "[M::worker_pipeline::0.622*0.99] mapped 1 sequences\n",
      "[M::main] Version: 2.19-r1057\n",
      "[M::main] CMD: minimap2 -ax asm20 --eqx -H -f 100 -r1k,10k --rmq=no -t 1 exp/multiple_runs11/tmp/L07-1/is_deleted/L07-1_genome.deleted_IS.1.fasta exp/multiple_runs11/tmp/L07-1/is_deleted/L07-1_genome.deleted_IS.2.fasta\n",
      "[M::main] Real time: 0.625 sec; CPU: 0.619 sec; Peak RSS: 0.085 GB\n",
      "local_variation - WARNING - Finished syri\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "invOut.txt  is empty. Skipping analysing it.\n",
      "TLOut.txt  is empty. Skipping analysing it.\n",
      "invTLOut.txt  is empty. Skipping analysing it.\n",
      "ctxOut.txt is empty. Skipping analysing it.\n",
      "inv Out.txt is empty. Skipping analysing it.\n",
      "TL Out.txt is empty. Skipping analysing it.\n",
      "invTL Out.txt is empty. Skipping analysing it.\n",
      "dup Out.txt is empty. Skipping analysing it.\n",
      "invDup Out.txt is empty. Skipping analysing it.\n",
      "ctxOut.txt is empty. Skipping analysing it.\n",
      "invOut.txt  is empty. Skipping analysing it.\n",
      "TLOut.txt  is empty. Skipping analysing it.\n",
      "invTLOut.txt  is empty. Skipping analysing it.\n",
      "ctxOut.txt is empty. Skipping analysing it.\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "[M::mm_idx_gen::0.056*1.01] collected minimizers\n",
      "[M::mm_idx_gen::0.074*1.00] sorted minimizers\n",
      "[M::main::0.074*1.00] loaded/built the index for 1 target sequence(s)\n",
      "[M::mm_mapopt_update::0.074*1.00] mid_occ = 100\n",
      "[M::mm_idx_stat] kmer size: 19; skip: 10; is_hpc: 1; #seq: 1\n",
      "[M::mm_idx_stat::0.078*1.00] distinct minimizers: 520925 (99.33% are singletons); average occurrences: 1.015; average spacing: 7.459; total length: 3941898\n",
      "[M::worker_pipeline::0.617*0.99] mapped 1 sequences\n",
      "[M::main] Version: 2.19-r1057\n",
      "[M::main] CMD: minimap2 -ax asm20 --eqx -H -f 100 -r1k,10k --rmq=no -t 1 exp/multiple_runs11/tmp/L07-1/is_deleted/L07-1_genome.deleted_IS.1.fasta exp/multiple_runs11/tmp/L07-1/is_deleted/L07-1_genome.deleted_IS.3.fasta\n",
      "[M::main] Real time: 0.620 sec; CPU: 0.613 sec; Peak RSS: 0.087 GB\n",
      "local_variation - WARNING - Finished syri\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "TLOut.txt  is empty. Skipping analysing it.\n",
      "invTLOut.txt  is empty. Skipping analysing it.\n",
      "ctxOut.txt is empty. Skipping analysing it.\n",
      "TL Out.txt is empty. Skipping analysing it.\n",
      "invTL Out.txt is empty. Skipping analysing it.\n",
      "dup Out.txt is empty. Skipping analysing it.\n",
      "invDup Out.txt is empty. Skipping analysing it.\n",
      "ctxOut.txt is empty. Skipping analysing it.\n",
      "TLOut.txt  is empty. Skipping analysing it.\n",
      "invTLOut.txt  is empty. Skipping analysing it.\n",
      "ctxOut.txt is empty. Skipping analysing it.\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "[M::mm_idx_gen::0.058*1.01] collected minimizers\n",
      "[M::mm_idx_gen::0.076*1.01] sorted minimizers\n",
      "[M::main::0.076*1.01] loaded/built the index for 1 target sequence(s)\n",
      "[M::mm_mapopt_update::0.076*1.01] mid_occ = 100\n",
      "[M::mm_idx_stat] kmer size: 19; skip: 10; is_hpc: 1; #seq: 1\n",
      "[M::mm_idx_stat::0.080*1.00] distinct minimizers: 520925 (99.33% are singletons); average occurrences: 1.015; average spacing: 7.459; total length: 3941898\n",
      "[M::worker_pipeline::0.621*0.99] mapped 1 sequences\n",
      "[M::main] Version: 2.19-r1057\n",
      "[M::main] CMD: minimap2 -ax asm20 --eqx -H -f 100 -r1k,10k --rmq=no -t 1 exp/multiple_runs11/tmp/L07-2/is_deleted/L07-2_genome.deleted_IS.1.fasta exp/multiple_runs11/tmp/L07-2/is_deleted/L07-2_genome.deleted_IS.2.fasta\n",
      "[M::main] Real time: 0.624 sec; CPU: 0.618 sec; Peak RSS: 0.088 GB\n",
      "local_variation - WARNING - Finished syri\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "invOut.txt  is empty. Skipping analysing it.\n",
      "TLOut.txt  is empty. Skipping analysing it.\n",
      "invTLOut.txt  is empty. Skipping analysing it.\n",
      "ctxOut.txt is empty. Skipping analysing it.\n",
      "inv Out.txt is empty. Skipping analysing it.\n",
      "TL Out.txt is empty. Skipping analysing it.\n",
      "invTL Out.txt is empty. Skipping analysing it.\n",
      "dup Out.txt is empty. Skipping analysing it.\n",
      "invDup Out.txt is empty. Skipping analysing it.\n",
      "ctxOut.txt is empty. Skipping analysing it.\n",
      "invOut.txt  is empty. Skipping analysing it.\n",
      "TLOut.txt  is empty. Skipping analysing it.\n",
      "invTLOut.txt  is empty. Skipping analysing it.\n",
      "ctxOut.txt is empty. Skipping analysing it.\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "[M::mm_idx_gen::0.063*1.01] collected minimizers\n",
      "[M::mm_idx_gen::0.082*1.00] sorted minimizers\n",
      "[M::main::0.082*1.00] loaded/built the index for 1 target sequence(s)\n",
      "[M::mm_mapopt_update::0.082*1.00] mid_occ = 100\n",
      "[M::mm_idx_stat] kmer size: 19; skip: 10; is_hpc: 1; #seq: 1\n",
      "[M::mm_idx_stat::0.086*1.00] distinct minimizers: 520925 (99.33% are singletons); average occurrences: 1.015; average spacing: 7.459; total length: 3941898\n",
      "[M::worker_pipeline::0.623*0.99] mapped 1 sequences\n",
      "[M::main] Version: 2.19-r1057\n",
      "[M::main] CMD: minimap2 -ax asm20 --eqx -H -f 100 -r1k,10k --rmq=no -t 1 exp/multiple_runs11/tmp/L07-2/is_deleted/L07-2_genome.deleted_IS.1.fasta exp/multiple_runs11/tmp/L07-2/is_deleted/L07-2_genome.deleted_IS.3.fasta\n",
      "[M::main] Real time: 0.626 sec; CPU: 0.619 sec; Peak RSS: 0.089 GB\n",
      "local_variation - WARNING - Finished syri\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "invOut.txt  is empty. Skipping analysing it.\n",
      "TLOut.txt  is empty. Skipping analysing it.\n",
      "invTLOut.txt  is empty. Skipping analysing it.\n",
      "ctxOut.txt is empty. Skipping analysing it.\n",
      "inv Out.txt is empty. Skipping analysing it.\n",
      "TL Out.txt is empty. Skipping analysing it.\n",
      "invTL Out.txt is empty. Skipping analysing it.\n",
      "dup Out.txt is empty. Skipping analysing it.\n",
      "invDup Out.txt is empty. Skipping analysing it.\n",
      "ctxOut.txt is empty. Skipping analysing it.\n",
      "invOut.txt  is empty. Skipping analysing it.\n",
      "TLOut.txt  is empty. Skipping analysing it.\n",
      "invTLOut.txt  is empty. Skipping analysing it.\n",
      "ctxOut.txt is empty. Skipping analysing it.\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "[M::mm_idx_gen::0.060*1.01] collected minimizers\n",
      "[M::mm_idx_gen::0.078*1.01] sorted minimizers\n",
      "[M::main::0.078*1.01] loaded/built the index for 1 target sequence(s)\n",
      "[M::mm_mapopt_update::0.078*1.01] mid_occ = 100\n",
      "[M::mm_idx_stat] kmer size: 19; skip: 10; is_hpc: 1; #seq: 1\n",
      "[M::mm_idx_stat::0.082*1.01] distinct minimizers: 520925 (99.33% are singletons); average occurrences: 1.015; average spacing: 7.459; total length: 3941898\n",
      "[M::worker_pipeline::0.619*0.99] mapped 1 sequences\n",
      "[M::main] Version: 2.19-r1057\n",
      "[M::main] CMD: minimap2 -ax asm20 --eqx -H -f 100 -r1k,10k --rmq=no -t 1 exp/multiple_runs11/tmp/L07-3/is_deleted/L07-3_genome.deleted_IS.1.fasta exp/multiple_runs11/tmp/L07-3/is_deleted/L07-3_genome.deleted_IS.2.fasta\n",
      "[M::main] Real time: 0.622 sec; CPU: 0.616 sec; Peak RSS: 0.086 GB\n",
      "local_variation - WARNING - Finished syri\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "TLOut.txt  is empty. Skipping analysing it.\n",
      "invTLOut.txt  is empty. Skipping analysing it.\n",
      "ctxOut.txt is empty. Skipping analysing it.\n",
      "TL Out.txt is empty. Skipping analysing it.\n",
      "invTL Out.txt is empty. Skipping analysing it.\n",
      "dup Out.txt is empty. Skipping analysing it.\n",
      "invDup Out.txt is empty. Skipping analysing it.\n",
      "ctxOut.txt is empty. Skipping analysing it.\n",
      "TLOut.txt  is empty. Skipping analysing it.\n",
      "invTLOut.txt  is empty. Skipping analysing it.\n",
      "ctxOut.txt is empty. Skipping analysing it.\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "[M::mm_idx_gen::0.063*1.01] collected minimizers\n",
      "[M::mm_idx_gen::0.082*1.00] sorted minimizers\n",
      "[M::main::0.082*1.00] loaded/built the index for 1 target sequence(s)\n",
      "[M::mm_mapopt_update::0.082*1.00] mid_occ = 100\n",
      "[M::mm_idx_stat] kmer size: 19; skip: 10; is_hpc: 1; #seq: 1\n",
      "[M::mm_idx_stat::0.085*1.00] distinct minimizers: 520925 (99.33% are singletons); average occurrences: 1.015; average spacing: 7.459; total length: 3941898\n",
      "[M::worker_pipeline::0.622*0.99] mapped 1 sequences\n",
      "[M::main] Version: 2.19-r1057\n",
      "[M::main] CMD: minimap2 -ax asm20 --eqx -H -f 100 -r1k,10k --rmq=no -t 1 exp/multiple_runs11/tmp/L07-3/is_deleted/L07-3_genome.deleted_IS.1.fasta exp/multiple_runs11/tmp/L07-3/is_deleted/L07-3_genome.deleted_IS.3.fasta\n",
      "[M::main] Real time: 0.625 sec; CPU: 0.619 sec; Peak RSS: 0.087 GB\n",
      "local_variation - WARNING - Finished syri\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "TLOut.txt  is empty. Skipping analysing it.\n",
      "invTLOut.txt  is empty. Skipping analysing it.\n",
      "ctxOut.txt is empty. Skipping analysing it.\n",
      "TL Out.txt is empty. Skipping analysing it.\n",
      "invTL Out.txt is empty. Skipping analysing it.\n",
      "dup Out.txt is empty. Skipping analysing it.\n",
      "invDup Out.txt is empty. Skipping analysing it.\n",
      "ctxOut.txt is empty. Skipping analysing it.\n",
      "TLOut.txt  is empty. Skipping analysing it.\n",
      "invTLOut.txt  is empty. Skipping analysing it.\n",
      "ctxOut.txt is empty. Skipping analysing it.\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "[M::mm_idx_gen::0.058*1.01] collected minimizers\n",
      "[M::mm_idx_gen::0.077*1.00] sorted minimizers\n",
      "[M::main::0.077*1.00] loaded/built the index for 1 target sequence(s)\n",
      "[M::mm_mapopt_update::0.077*1.00] mid_occ = 100\n",
      "[M::mm_idx_stat] kmer size: 19; skip: 10; is_hpc: 1; #seq: 1\n",
      "[M::mm_idx_stat::0.080*1.00] distinct minimizers: 522574 (99.33% are singletons); average occurrences: 1.015; average spacing: 7.460; total length: 3954801\n",
      "[M::worker_pipeline::0.615*0.99] mapped 1 sequences\n",
      "[M::main] Version: 2.19-r1057\n",
      "[M::main] CMD: minimap2 -ax asm20 --eqx -H -f 100 -r1k,10k --rmq=no -t 1 exp/multiple_runs11/tmp/L07-4/is_deleted/L07-4_genome.deleted_IS.1.fasta exp/multiple_runs11/tmp/L07-4/is_deleted/L07-4_genome.deleted_IS.2.fasta\n",
      "[M::main] Real time: 0.618 sec; CPU: 0.612 sec; Peak RSS: 0.086 GB\n",
      "local_variation - WARNING - Finished syri\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "invOut.txt  is empty. Skipping analysing it.\n",
      "TLOut.txt  is empty. Skipping analysing it.\n",
      "invTLOut.txt  is empty. Skipping analysing it.\n",
      "ctxOut.txt is empty. Skipping analysing it.\n",
      "inv Out.txt is empty. Skipping analysing it.\n",
      "TL Out.txt is empty. Skipping analysing it.\n",
      "invTL Out.txt is empty. Skipping analysing it.\n",
      "dup Out.txt is empty. Skipping analysing it.\n",
      "invDup Out.txt is empty. Skipping analysing it.\n",
      "ctxOut.txt is empty. Skipping analysing it.\n",
      "invOut.txt  is empty. Skipping analysing it.\n",
      "TLOut.txt  is empty. Skipping analysing it.\n",
      "invTLOut.txt  is empty. Skipping analysing it.\n",
      "ctxOut.txt is empty. Skipping analysing it.\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "[M::mm_idx_gen::0.058*1.01] collected minimizers\n",
      "[M::mm_idx_gen::0.076*1.00] sorted minimizers\n",
      "[M::main::0.076*1.00] loaded/built the index for 1 target sequence(s)\n",
      "[M::mm_mapopt_update::0.076*1.00] mid_occ = 100\n",
      "[M::mm_idx_stat] kmer size: 19; skip: 10; is_hpc: 1; #seq: 1\n",
      "[M::mm_idx_stat::0.080*1.00] distinct minimizers: 522574 (99.33% are singletons); average occurrences: 1.015; average spacing: 7.460; total length: 3954801\n",
      "[M::worker_pipeline::0.615*0.99] mapped 1 sequences\n",
      "[M::main] Version: 2.19-r1057\n",
      "[M::main] CMD: minimap2 -ax asm20 --eqx -H -f 100 -r1k,10k --rmq=no -t 1 exp/multiple_runs11/tmp/L07-4/is_deleted/L07-4_genome.deleted_IS.1.fasta exp/multiple_runs11/tmp/L07-4/is_deleted/L07-4_genome.deleted_IS.3.fasta\n",
      "[M::main] Real time: 0.618 sec; CPU: 0.611 sec; Peak RSS: 0.087 GB\n",
      "local_variation - WARNING - Finished syri\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "TLOut.txt  is empty. Skipping analysing it.\n",
      "invTLOut.txt  is empty. Skipping analysing it.\n",
      "ctxOut.txt is empty. Skipping analysing it.\n",
      "TL Out.txt is empty. Skipping analysing it.\n",
      "invTL Out.txt is empty. Skipping analysing it.\n",
      "dup Out.txt is empty. Skipping analysing it.\n",
      "invDup Out.txt is empty. Skipping analysing it.\n",
      "ctxOut.txt is empty. Skipping analysing it.\n",
      "TLOut.txt  is empty. Skipping analysing it.\n",
      "invTLOut.txt  is empty. Skipping analysing it.\n",
      "ctxOut.txt is empty. Skipping analysing it.\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "[M::mm_idx_gen::0.058*1.01] collected minimizers\n",
      "[M::mm_idx_gen::0.076*1.00] sorted minimizers\n",
      "[M::main::0.076*1.00] loaded/built the index for 1 target sequence(s)\n",
      "[M::mm_mapopt_update::0.076*1.00] mid_occ = 100\n",
      "[M::mm_idx_stat] kmer size: 19; skip: 10; is_hpc: 1; #seq: 1\n",
      "[M::mm_idx_stat::0.080*1.00] distinct minimizers: 522574 (99.33% are singletons); average occurrences: 1.015; average spacing: 7.460; total length: 3954810\n",
      "[M::worker_pipeline::0.614*0.99] mapped 1 sequences\n",
      "[M::main] Version: 2.19-r1057\n",
      "[M::main] CMD: minimap2 -ax asm20 --eqx -H -f 100 -r1k,10k --rmq=no -t 1 exp/multiple_runs11/tmp/L08-1/is_deleted/L08-1_genome.deleted_IS.1.fasta exp/multiple_runs11/tmp/L08-1/is_deleted/L08-1_genome.deleted_IS.2.fasta\n",
      "[M::main] Real time: 0.617 sec; CPU: 0.611 sec; Peak RSS: 0.087 GB\n",
      "local_variation - WARNING - Finished syri\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "invOut.txt  is empty. Skipping analysing it.\n",
      "TLOut.txt  is empty. Skipping analysing it.\n",
      "invTLOut.txt  is empty. Skipping analysing it.\n",
      "ctxOut.txt is empty. Skipping analysing it.\n",
      "inv Out.txt is empty. Skipping analysing it.\n",
      "TL Out.txt is empty. Skipping analysing it.\n",
      "invTL Out.txt is empty. Skipping analysing it.\n",
      "dup Out.txt is empty. Skipping analysing it.\n",
      "invDup Out.txt is empty. Skipping analysing it.\n",
      "ctxOut.txt is empty. Skipping analysing it.\n",
      "invOut.txt  is empty. Skipping analysing it.\n",
      "TLOut.txt  is empty. Skipping analysing it.\n",
      "invTLOut.txt  is empty. Skipping analysing it.\n",
      "ctxOut.txt is empty. Skipping analysing it.\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "[M::mm_idx_gen::0.058*1.01] collected minimizers\n",
      "[M::mm_idx_gen::0.077*1.00] sorted minimizers\n",
      "[M::main::0.077*1.00] loaded/built the index for 1 target sequence(s)\n",
      "[M::mm_mapopt_update::0.077*1.00] mid_occ = 100\n",
      "[M::mm_idx_stat] kmer size: 19; skip: 10; is_hpc: 1; #seq: 1\n",
      "[M::mm_idx_stat::0.081*1.00] distinct minimizers: 522574 (99.33% are singletons); average occurrences: 1.015; average spacing: 7.460; total length: 3954810\n",
      "[M::worker_pipeline::0.608*0.99] mapped 1 sequences\n",
      "[M::main] Version: 2.19-r1057\n",
      "[M::main] CMD: minimap2 -ax asm20 --eqx -H -f 100 -r1k,10k --rmq=no -t 1 exp/multiple_runs11/tmp/L08-1/is_deleted/L08-1_genome.deleted_IS.1.fasta exp/multiple_runs11/tmp/L08-1/is_deleted/L08-1_genome.deleted_IS.3.fasta\n",
      "[M::main] Real time: 0.612 sec; CPU: 0.605 sec; Peak RSS: 0.087 GB\n",
      "local_variation - WARNING - Finished syri\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "invOut.txt  is empty. Skipping analysing it.\n",
      "TLOut.txt  is empty. Skipping analysing it.\n",
      "invTLOut.txt  is empty. Skipping analysing it.\n",
      "ctxOut.txt is empty. Skipping analysing it.\n",
      "inv Out.txt is empty. Skipping analysing it.\n",
      "TL Out.txt is empty. Skipping analysing it.\n",
      "invTL Out.txt is empty. Skipping analysing it.\n",
      "dup Out.txt is empty. Skipping analysing it.\n",
      "invDup Out.txt is empty. Skipping analysing it.\n",
      "ctxOut.txt is empty. Skipping analysing it.\n",
      "invOut.txt  is empty. Skipping analysing it.\n",
      "TLOut.txt  is empty. Skipping analysing it.\n",
      "invTLOut.txt  is empty. Skipping analysing it.\n",
      "ctxOut.txt is empty. Skipping analysing it.\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "[M::mm_idx_gen::0.058*1.01] collected minimizers\n",
      "[M::mm_idx_gen::0.077*1.00] sorted minimizers\n",
      "[M::main::0.077*1.00] loaded/built the index for 1 target sequence(s)\n",
      "[M::mm_mapopt_update::0.077*1.00] mid_occ = 100\n",
      "[M::mm_idx_stat] kmer size: 19; skip: 10; is_hpc: 1; #seq: 1\n",
      "[M::mm_idx_stat::0.080*1.00] distinct minimizers: 522574 (99.33% are singletons); average occurrences: 1.015; average spacing: 7.460; total length: 3954810\n",
      "[M::worker_pipeline::0.613*0.99] mapped 1 sequences\n",
      "[M::main] Version: 2.19-r1057\n",
      "[M::main] CMD: minimap2 -ax asm20 --eqx -H -f 100 -r1k,10k --rmq=no -t 1 exp/multiple_runs11/tmp/L08-2/is_deleted/L08-2_genome.deleted_IS.1.fasta exp/multiple_runs11/tmp/L08-2/is_deleted/L08-2_genome.deleted_IS.2.fasta\n",
      "[M::main] Real time: 0.617 sec; CPU: 0.611 sec; Peak RSS: 0.085 GB\n",
      "local_variation - WARNING - Finished syri\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "invOut.txt  is empty. Skipping analysing it.\n",
      "TLOut.txt  is empty. Skipping analysing it.\n",
      "invTLOut.txt  is empty. Skipping analysing it.\n",
      "ctxOut.txt is empty. Skipping analysing it.\n",
      "inv Out.txt is empty. Skipping analysing it.\n",
      "TL Out.txt is empty. Skipping analysing it.\n",
      "invTL Out.txt is empty. Skipping analysing it.\n",
      "dup Out.txt is empty. Skipping analysing it.\n",
      "invDup Out.txt is empty. Skipping analysing it.\n",
      "ctxOut.txt is empty. Skipping analysing it.\n",
      "invOut.txt  is empty. Skipping analysing it.\n",
      "TLOut.txt  is empty. Skipping analysing it.\n",
      "invTLOut.txt  is empty. Skipping analysing it.\n",
      "ctxOut.txt is empty. Skipping analysing it.\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "[M::mm_idx_gen::0.057*1.01] collected minimizers\n",
      "[M::mm_idx_gen::0.076*1.00] sorted minimizers\n",
      "[M::main::0.076*1.00] loaded/built the index for 1 target sequence(s)\n",
      "[M::mm_mapopt_update::0.076*1.00] mid_occ = 100\n",
      "[M::mm_idx_stat] kmer size: 19; skip: 10; is_hpc: 1; #seq: 1\n",
      "[M::mm_idx_stat::0.079*1.00] distinct minimizers: 522574 (99.33% are singletons); average occurrences: 1.015; average spacing: 7.460; total length: 3954810\n",
      "[M::worker_pipeline::0.612*0.99] mapped 1 sequences\n",
      "[M::main] Version: 2.19-r1057\n",
      "[M::main] CMD: minimap2 -ax asm20 --eqx -H -f 100 -r1k,10k --rmq=no -t 1 exp/multiple_runs11/tmp/L08-2/is_deleted/L08-2_genome.deleted_IS.1.fasta exp/multiple_runs11/tmp/L08-2/is_deleted/L08-2_genome.deleted_IS.3.fasta\n",
      "[M::main] Real time: 0.616 sec; CPU: 0.610 sec; Peak RSS: 0.086 GB\n",
      "local_variation - WARNING - Finished syri\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "invOut.txt  is empty. Skipping analysing it.\n",
      "TLOut.txt  is empty. Skipping analysing it.\n",
      "invTLOut.txt  is empty. Skipping analysing it.\n",
      "ctxOut.txt is empty. Skipping analysing it.\n",
      "inv Out.txt is empty. Skipping analysing it.\n",
      "TL Out.txt is empty. Skipping analysing it.\n",
      "invTL Out.txt is empty. Skipping analysing it.\n",
      "invDup Out.txt is empty. Skipping analysing it.\n",
      "ctxOut.txt is empty. Skipping analysing it.\n",
      "invOut.txt  is empty. Skipping analysing it.\n",
      "TLOut.txt  is empty. Skipping analysing it.\n",
      "invTLOut.txt  is empty. Skipping analysing it.\n",
      "ctxOut.txt is empty. Skipping analysing it.\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "[M::mm_idx_gen::0.064*1.01] collected minimizers\n",
      "[M::mm_idx_gen::0.083*1.00] sorted minimizers\n",
      "[M::main::0.083*1.00] loaded/built the index for 1 target sequence(s)\n",
      "[M::mm_mapopt_update::0.083*1.00] mid_occ = 100\n",
      "[M::mm_idx_stat] kmer size: 19; skip: 10; is_hpc: 1; #seq: 1\n",
      "[M::mm_idx_stat::0.087*1.00] distinct minimizers: 522574 (99.33% are singletons); average occurrences: 1.015; average spacing: 7.460; total length: 3954810\n",
      "[M::worker_pipeline::0.619*0.99] mapped 1 sequences\n",
      "[M::main] Version: 2.19-r1057\n",
      "[M::main] CMD: minimap2 -ax asm20 --eqx -H -f 100 -r1k,10k --rmq=no -t 1 exp/multiple_runs11/tmp/L08-3/is_deleted/L08-3_genome.deleted_IS.1.fasta exp/multiple_runs11/tmp/L08-3/is_deleted/L08-3_genome.deleted_IS.2.fasta\n",
      "[M::main] Real time: 0.622 sec; CPU: 0.617 sec; Peak RSS: 0.086 GB\n",
      "local_variation - WARNING - Finished syri\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "invOut.txt  is empty. Skipping analysing it.\n",
      "TLOut.txt  is empty. Skipping analysing it.\n",
      "invTLOut.txt  is empty. Skipping analysing it.\n",
      "ctxOut.txt is empty. Skipping analysing it.\n",
      "inv Out.txt is empty. Skipping analysing it.\n",
      "TL Out.txt is empty. Skipping analysing it.\n",
      "invTL Out.txt is empty. Skipping analysing it.\n",
      "dup Out.txt is empty. Skipping analysing it.\n",
      "invDup Out.txt is empty. Skipping analysing it.\n",
      "ctxOut.txt is empty. Skipping analysing it.\n",
      "invOut.txt  is empty. Skipping analysing it.\n",
      "TLOut.txt  is empty. Skipping analysing it.\n",
      "invTLOut.txt  is empty. Skipping analysing it.\n",
      "ctxOut.txt is empty. Skipping analysing it.\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "[M::mm_idx_gen::0.058*1.01] collected minimizers\n",
      "[M::mm_idx_gen::0.077*1.00] sorted minimizers\n",
      "[M::main::0.077*1.00] loaded/built the index for 1 target sequence(s)\n",
      "[M::mm_mapopt_update::0.077*1.00] mid_occ = 100\n",
      "[M::mm_idx_stat] kmer size: 19; skip: 10; is_hpc: 1; #seq: 1\n",
      "[M::mm_idx_stat::0.081*1.00] distinct minimizers: 522574 (99.33% are singletons); average occurrences: 1.015; average spacing: 7.460; total length: 3954810\n",
      "[M::worker_pipeline::0.611*0.99] mapped 1 sequences\n",
      "[M::main] Version: 2.19-r1057\n",
      "[M::main] CMD: minimap2 -ax asm20 --eqx -H -f 100 -r1k,10k --rmq=no -t 1 exp/multiple_runs11/tmp/L08-3/is_deleted/L08-3_genome.deleted_IS.1.fasta exp/multiple_runs11/tmp/L08-3/is_deleted/L08-3_genome.deleted_IS.3.fasta\n",
      "[M::main] Real time: 0.615 sec; CPU: 0.610 sec; Peak RSS: 0.087 GB\n",
      "local_variation - WARNING - Finished syri\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "invOut.txt  is empty. Skipping analysing it.\n",
      "TLOut.txt  is empty. Skipping analysing it.\n",
      "invTLOut.txt  is empty. Skipping analysing it.\n",
      "ctxOut.txt is empty. Skipping analysing it.\n",
      "inv Out.txt is empty. Skipping analysing it.\n",
      "TL Out.txt is empty. Skipping analysing it.\n",
      "invTL Out.txt is empty. Skipping analysing it.\n",
      "dup Out.txt is empty. Skipping analysing it.\n",
      "invDup Out.txt is empty. Skipping analysing it.\n",
      "ctxOut.txt is empty. Skipping analysing it.\n",
      "invOut.txt  is empty. Skipping analysing it.\n",
      "TLOut.txt  is empty. Skipping analysing it.\n",
      "invTLOut.txt  is empty. Skipping analysing it.\n",
      "ctxOut.txt is empty. Skipping analysing it.\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "[M::mm_idx_gen::0.057*1.01] collected minimizers\n",
      "[M::mm_idx_gen::0.076*1.00] sorted minimizers\n",
      "[M::main::0.076*1.00] loaded/built the index for 1 target sequence(s)\n",
      "[M::mm_mapopt_update::0.076*1.00] mid_occ = 100\n",
      "[M::mm_idx_stat] kmer size: 19; skip: 10; is_hpc: 1; #seq: 1\n",
      "[M::mm_idx_stat::0.080*1.00] distinct minimizers: 522574 (99.33% are singletons); average occurrences: 1.015; average spacing: 7.460; total length: 3954810\n",
      "[M::worker_pipeline::0.617*0.99] mapped 1 sequences\n",
      "[M::main] Version: 2.19-r1057\n",
      "[M::main] CMD: minimap2 -ax asm20 --eqx -H -f 100 -r1k,10k --rmq=no -t 1 exp/multiple_runs11/tmp/L08-4/is_deleted/L08-4_genome.deleted_IS.1.fasta exp/multiple_runs11/tmp/L08-4/is_deleted/L08-4_genome.deleted_IS.2.fasta\n",
      "[M::main] Real time: 0.620 sec; CPU: 0.614 sec; Peak RSS: 0.086 GB\n",
      "local_variation - WARNING - Finished syri\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "invOut.txt  is empty. Skipping analysing it.\n",
      "TLOut.txt  is empty. Skipping analysing it.\n",
      "invTLOut.txt  is empty. Skipping analysing it.\n",
      "ctxOut.txt is empty. Skipping analysing it.\n",
      "inv Out.txt is empty. Skipping analysing it.\n",
      "TL Out.txt is empty. Skipping analysing it.\n",
      "invTL Out.txt is empty. Skipping analysing it.\n",
      "dup Out.txt is empty. Skipping analysing it.\n",
      "invDup Out.txt is empty. Skipping analysing it.\n",
      "ctxOut.txt is empty. Skipping analysing it.\n",
      "invOut.txt  is empty. Skipping analysing it.\n",
      "TLOut.txt  is empty. Skipping analysing it.\n",
      "invTLOut.txt  is empty. Skipping analysing it.\n",
      "ctxOut.txt is empty. Skipping analysing it.\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "[M::mm_idx_gen::0.057*1.01] collected minimizers\n",
      "[M::mm_idx_gen::0.076*1.00] sorted minimizers\n",
      "[M::main::0.076*1.00] loaded/built the index for 1 target sequence(s)\n",
      "[M::mm_mapopt_update::0.076*1.00] mid_occ = 100\n",
      "[M::mm_idx_stat] kmer size: 19; skip: 10; is_hpc: 1; #seq: 1\n",
      "[M::mm_idx_stat::0.079*1.00] distinct minimizers: 522574 (99.33% are singletons); average occurrences: 1.015; average spacing: 7.460; total length: 3954810\n",
      "[M::worker_pipeline::0.615*0.99] mapped 1 sequences\n",
      "[M::main] Version: 2.19-r1057\n",
      "[M::main] CMD: minimap2 -ax asm20 --eqx -H -f 100 -r1k,10k --rmq=no -t 1 exp/multiple_runs11/tmp/L08-4/is_deleted/L08-4_genome.deleted_IS.1.fasta exp/multiple_runs11/tmp/L08-4/is_deleted/L08-4_genome.deleted_IS.3.fasta\n",
      "[M::main] Real time: 0.617 sec; CPU: 0.612 sec; Peak RSS: 0.086 GB\n",
      "local_variation - WARNING - Finished syri\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "invOut.txt  is empty. Skipping analysing it.\n",
      "TLOut.txt  is empty. Skipping analysing it.\n",
      "invTLOut.txt  is empty. Skipping analysing it.\n",
      "ctxOut.txt is empty. Skipping analysing it.\n",
      "inv Out.txt is empty. Skipping analysing it.\n",
      "TL Out.txt is empty. Skipping analysing it.\n",
      "invTL Out.txt is empty. Skipping analysing it.\n",
      "dup Out.txt is empty. Skipping analysing it.\n",
      "invDup Out.txt is empty. Skipping analysing it.\n",
      "ctxOut.txt is empty. Skipping analysing it.\n",
      "invOut.txt  is empty. Skipping analysing it.\n",
      "TLOut.txt  is empty. Skipping analysing it.\n",
      "invTLOut.txt  is empty. Skipping analysing it.\n",
      "ctxOut.txt is empty. Skipping analysing it.\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "[M::mm_idx_gen::0.057*1.01] collected minimizers\n",
      "[M::mm_idx_gen::0.075*1.00] sorted minimizers\n",
      "[M::main::0.075*1.00] loaded/built the index for 1 target sequence(s)\n",
      "[M::mm_mapopt_update::0.075*1.00] mid_occ = 100\n",
      "[M::mm_idx_stat] kmer size: 19; skip: 10; is_hpc: 1; #seq: 1\n",
      "[M::mm_idx_stat::0.079*1.00] distinct minimizers: 521564 (99.34% are singletons); average occurrences: 1.014; average spacing: 7.461; total length: 3947629\n",
      "[M::worker_pipeline::0.598*1.00] mapped 1 sequences\n",
      "[M::main] Version: 2.19-r1057\n",
      "[M::main] CMD: minimap2 -ax asm20 --eqx -H -f 100 -r1k,10k --rmq=no -t 1 exp/multiple_runs11/tmp/L09-1/is_deleted/L09-1_genome.deleted_IS.1.fasta exp/multiple_runs11/tmp/L09-1/is_deleted/L09-1_genome.deleted_IS.2.fasta\n",
      "[M::main] Real time: 0.601 sec; CPU: 0.600 sec; Peak RSS: 0.087 GB\n",
      "local_variation - WARNING - Finished syri\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "invOut.txt  is empty. Skipping analysing it.\n",
      "TLOut.txt  is empty. Skipping analysing it.\n",
      "invTLOut.txt  is empty. Skipping analysing it.\n",
      "ctxOut.txt is empty. Skipping analysing it.\n",
      "inv Out.txt is empty. Skipping analysing it.\n",
      "TL Out.txt is empty. Skipping analysing it.\n",
      "invTL Out.txt is empty. Skipping analysing it.\n",
      "dup Out.txt is empty. Skipping analysing it.\n",
      "invDup Out.txt is empty. Skipping analysing it.\n",
      "ctxOut.txt is empty. Skipping analysing it.\n",
      "invOut.txt  is empty. Skipping analysing it.\n",
      "TLOut.txt  is empty. Skipping analysing it.\n",
      "invTLOut.txt  is empty. Skipping analysing it.\n",
      "ctxOut.txt is empty. Skipping analysing it.\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "[M::mm_idx_gen::0.057*1.01] collected minimizers\n",
      "[M::mm_idx_gen::0.075*1.00] sorted minimizers\n",
      "[M::main::0.075*1.00] loaded/built the index for 1 target sequence(s)\n",
      "[M::mm_mapopt_update::0.075*1.00] mid_occ = 100\n",
      "[M::mm_idx_stat] kmer size: 19; skip: 10; is_hpc: 1; #seq: 1\n",
      "[M::mm_idx_stat::0.079*1.00] distinct minimizers: 521564 (99.34% are singletons); average occurrences: 1.014; average spacing: 7.461; total length: 3947629\n",
      "[M::worker_pipeline::0.612*1.00] mapped 1 sequences\n",
      "[M::main] Version: 2.19-r1057\n",
      "[M::main] CMD: minimap2 -ax asm20 --eqx -H -f 100 -r1k,10k --rmq=no -t 1 exp/multiple_runs11/tmp/L09-1/is_deleted/L09-1_genome.deleted_IS.1.fasta exp/multiple_runs11/tmp/L09-1/is_deleted/L09-1_genome.deleted_IS.3.fasta\n",
      "[M::main] Real time: 0.615 sec; CPU: 0.613 sec; Peak RSS: 0.085 GB\n",
      "local_variation - WARNING - Finished syri\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "invOut.txt  is empty. Skipping analysing it.\n",
      "TLOut.txt  is empty. Skipping analysing it.\n",
      "invTLOut.txt  is empty. Skipping analysing it.\n",
      "ctxOut.txt is empty. Skipping analysing it.\n",
      "inv Out.txt is empty. Skipping analysing it.\n",
      "TL Out.txt is empty. Skipping analysing it.\n",
      "invTL Out.txt is empty. Skipping analysing it.\n",
      "dup Out.txt is empty. Skipping analysing it.\n",
      "invDup Out.txt is empty. Skipping analysing it.\n",
      "ctxOut.txt is empty. Skipping analysing it.\n",
      "invOut.txt  is empty. Skipping analysing it.\n",
      "TLOut.txt  is empty. Skipping analysing it.\n",
      "invTLOut.txt  is empty. Skipping analysing it.\n",
      "ctxOut.txt is empty. Skipping analysing it.\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "[M::mm_idx_gen::0.060*1.01] collected minimizers\n",
      "[M::mm_idx_gen::0.078*1.00] sorted minimizers\n",
      "[M::main::0.078*1.00] loaded/built the index for 1 target sequence(s)\n",
      "[M::mm_mapopt_update::0.078*1.00] mid_occ = 100\n",
      "[M::mm_idx_stat] kmer size: 19; skip: 10; is_hpc: 1; #seq: 1\n",
      "[M::mm_idx_stat::0.082*1.00] distinct minimizers: 521562 (99.34% are singletons); average occurrences: 1.014; average spacing: 7.461; total length: 3947620\n",
      "[M::worker_pipeline::0.616*0.99] mapped 1 sequences\n",
      "[M::main] Version: 2.19-r1057\n",
      "[M::main] CMD: minimap2 -ax asm20 --eqx -H -f 100 -r1k,10k --rmq=no -t 1 exp/multiple_runs11/tmp/L09-2/is_deleted/L09-2_genome.deleted_IS.1.fasta exp/multiple_runs11/tmp/L09-2/is_deleted/L09-2_genome.deleted_IS.2.fasta\n",
      "[M::main] Real time: 0.619 sec; CPU: 0.613 sec; Peak RSS: 0.087 GB\n",
      "local_variation - WARNING - Finished syri\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "TLOut.txt  is empty. Skipping analysing it.\n",
      "invTLOut.txt  is empty. Skipping analysing it.\n",
      "ctxOut.txt is empty. Skipping analysing it.\n",
      "TL Out.txt is empty. Skipping analysing it.\n",
      "invTL Out.txt is empty. Skipping analysing it.\n",
      "dup Out.txt is empty. Skipping analysing it.\n",
      "invDup Out.txt is empty. Skipping analysing it.\n",
      "ctxOut.txt is empty. Skipping analysing it.\n",
      "TLOut.txt  is empty. Skipping analysing it.\n",
      "invTLOut.txt  is empty. Skipping analysing it.\n",
      "ctxOut.txt is empty. Skipping analysing it.\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "[M::mm_idx_gen::0.057*1.00] collected minimizers\n",
      "[M::mm_idx_gen::0.075*1.00] sorted minimizers\n",
      "[M::main::0.075*1.00] loaded/built the index for 1 target sequence(s)\n",
      "[M::mm_mapopt_update::0.075*1.00] mid_occ = 100\n",
      "[M::mm_idx_stat] kmer size: 19; skip: 10; is_hpc: 1; #seq: 1\n",
      "[M::mm_idx_stat::0.079*1.00] distinct minimizers: 521562 (99.34% are singletons); average occurrences: 1.014; average spacing: 7.461; total length: 3947620\n",
      "[M::worker_pipeline::0.602*0.99] mapped 1 sequences\n",
      "[M::main] Version: 2.19-r1057\n",
      "[M::main] CMD: minimap2 -ax asm20 --eqx -H -f 100 -r1k,10k --rmq=no -t 1 exp/multiple_runs11/tmp/L09-2/is_deleted/L09-2_genome.deleted_IS.1.fasta exp/multiple_runs11/tmp/L09-2/is_deleted/L09-2_genome.deleted_IS.3.fasta\n",
      "[M::main] Real time: 0.605 sec; CPU: 0.599 sec; Peak RSS: 0.085 GB\n",
      "local_variation - WARNING - Finished syri\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "TLOut.txt  is empty. Skipping analysing it.\n",
      "invTLOut.txt  is empty. Skipping analysing it.\n",
      "ctxOut.txt is empty. Skipping analysing it.\n",
      "TL Out.txt is empty. Skipping analysing it.\n",
      "invTL Out.txt is empty. Skipping analysing it.\n",
      "dup Out.txt is empty. Skipping analysing it.\n",
      "invDup Out.txt is empty. Skipping analysing it.\n",
      "ctxOut.txt is empty. Skipping analysing it.\n",
      "TLOut.txt  is empty. Skipping analysing it.\n",
      "invTLOut.txt  is empty. Skipping analysing it.\n",
      "ctxOut.txt is empty. Skipping analysing it.\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "[M::mm_idx_gen::0.062*1.01] collected minimizers\n",
      "[M::mm_idx_gen::0.080*1.00] sorted minimizers\n",
      "[M::main::0.081*1.00] loaded/built the index for 1 target sequence(s)\n",
      "[M::mm_mapopt_update::0.081*1.00] mid_occ = 100\n",
      "[M::mm_idx_stat] kmer size: 19; skip: 10; is_hpc: 1; #seq: 1\n",
      "[M::mm_idx_stat::0.084*1.00] distinct minimizers: 521564 (99.34% are singletons); average occurrences: 1.014; average spacing: 7.461; total length: 3947629\n",
      "[M::worker_pipeline::0.612*0.99] mapped 1 sequences\n",
      "[M::main] Version: 2.19-r1057\n",
      "[M::main] CMD: minimap2 -ax asm20 --eqx -H -f 100 -r1k,10k --rmq=no -t 1 exp/multiple_runs11/tmp/L09-3/is_deleted/L09-3_genome.deleted_IS.1.fasta exp/multiple_runs11/tmp/L09-3/is_deleted/L09-3_genome.deleted_IS.2.fasta\n",
      "[M::main] Real time: 0.615 sec; CPU: 0.610 sec; Peak RSS: 0.086 GB\n",
      "local_variation - WARNING - Finished syri\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "invOut.txt  is empty. Skipping analysing it.\n",
      "TLOut.txt  is empty. Skipping analysing it.\n",
      "invTLOut.txt  is empty. Skipping analysing it.\n",
      "ctxOut.txt is empty. Skipping analysing it.\n",
      "inv Out.txt is empty. Skipping analysing it.\n",
      "TL Out.txt is empty. Skipping analysing it.\n",
      "invTL Out.txt is empty. Skipping analysing it.\n",
      "dup Out.txt is empty. Skipping analysing it.\n",
      "invDup Out.txt is empty. Skipping analysing it.\n",
      "ctxOut.txt is empty. Skipping analysing it.\n",
      "invOut.txt  is empty. Skipping analysing it.\n",
      "TLOut.txt  is empty. Skipping analysing it.\n",
      "invTLOut.txt  is empty. Skipping analysing it.\n",
      "ctxOut.txt is empty. Skipping analysing it.\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "[M::mm_idx_gen::0.057*1.01] collected minimizers\n",
      "[M::mm_idx_gen::0.075*1.00] sorted minimizers\n",
      "[M::main::0.075*1.00] loaded/built the index for 1 target sequence(s)\n",
      "[M::mm_mapopt_update::0.075*1.00] mid_occ = 100\n",
      "[M::mm_idx_stat] kmer size: 19; skip: 10; is_hpc: 1; #seq: 1\n",
      "[M::mm_idx_stat::0.079*1.00] distinct minimizers: 521564 (99.34% are singletons); average occurrences: 1.014; average spacing: 7.461; total length: 3947629\n",
      "[M::worker_pipeline::0.617*0.99] mapped 1 sequences\n",
      "[M::main] Version: 2.19-r1057\n",
      "[M::main] CMD: minimap2 -ax asm20 --eqx -H -f 100 -r1k,10k --rmq=no -t 1 exp/multiple_runs11/tmp/L09-3/is_deleted/L09-3_genome.deleted_IS.1.fasta exp/multiple_runs11/tmp/L09-3/is_deleted/L09-3_genome.deleted_IS.3.fasta\n",
      "[M::main] Real time: 0.620 sec; CPU: 0.614 sec; Peak RSS: 0.085 GB\n",
      "local_variation - WARNING - Finished syri\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "invOut.txt  is empty. Skipping analysing it.\n",
      "TLOut.txt  is empty. Skipping analysing it.\n",
      "invTLOut.txt  is empty. Skipping analysing it.\n",
      "ctxOut.txt is empty. Skipping analysing it.\n",
      "inv Out.txt is empty. Skipping analysing it.\n",
      "TL Out.txt is empty. Skipping analysing it.\n",
      "invTL Out.txt is empty. Skipping analysing it.\n",
      "dup Out.txt is empty. Skipping analysing it.\n",
      "invDup Out.txt is empty. Skipping analysing it.\n",
      "ctxOut.txt is empty. Skipping analysing it.\n",
      "invOut.txt  is empty. Skipping analysing it.\n",
      "TLOut.txt  is empty. Skipping analysing it.\n",
      "invTLOut.txt  is empty. Skipping analysing it.\n",
      "ctxOut.txt is empty. Skipping analysing it.\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "[M::mm_idx_gen::0.056*1.01] collected minimizers\n",
      "[M::mm_idx_gen::0.075*1.00] sorted minimizers\n",
      "[M::main::0.075*1.00] loaded/built the index for 1 target sequence(s)\n",
      "[M::mm_mapopt_update::0.075*1.00] mid_occ = 100\n",
      "[M::mm_idx_stat] kmer size: 19; skip: 10; is_hpc: 1; #seq: 1\n",
      "[M::mm_idx_stat::0.079*1.00] distinct minimizers: 521562 (99.34% are singletons); average occurrences: 1.014; average spacing: 7.461; total length: 3947620\n",
      "[M::worker_pipeline::0.611*1.00] mapped 1 sequences\n",
      "[M::main] Version: 2.19-r1057\n",
      "[M::main] CMD: minimap2 -ax asm20 --eqx -H -f 100 -r1k,10k --rmq=no -t 1 exp/multiple_runs11/tmp/L09-4/is_deleted/L09-4_genome.deleted_IS.1.fasta exp/multiple_runs11/tmp/L09-4/is_deleted/L09-4_genome.deleted_IS.2.fasta\n",
      "[M::main] Real time: 0.614 sec; CPU: 0.612 sec; Peak RSS: 0.085 GB\n",
      "local_variation - WARNING - Finished syri\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "invOut.txt  is empty. Skipping analysing it.\n",
      "TLOut.txt  is empty. Skipping analysing it.\n",
      "invTLOut.txt  is empty. Skipping analysing it.\n",
      "ctxOut.txt is empty. Skipping analysing it.\n",
      "inv Out.txt is empty. Skipping analysing it.\n",
      "TL Out.txt is empty. Skipping analysing it.\n",
      "invTL Out.txt is empty. Skipping analysing it.\n",
      "dup Out.txt is empty. Skipping analysing it.\n",
      "invDup Out.txt is empty. Skipping analysing it.\n",
      "ctxOut.txt is empty. Skipping analysing it.\n",
      "invOut.txt  is empty. Skipping analysing it.\n",
      "TLOut.txt  is empty. Skipping analysing it.\n",
      "invTLOut.txt  is empty. Skipping analysing it.\n",
      "ctxOut.txt is empty. Skipping analysing it.\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "[M::mm_idx_gen::0.058*1.00] collected minimizers\n",
      "[M::mm_idx_gen::0.076*1.00] sorted minimizers\n",
      "[M::main::0.076*1.00] loaded/built the index for 1 target sequence(s)\n",
      "[M::mm_mapopt_update::0.076*1.00] mid_occ = 100\n",
      "[M::mm_idx_stat] kmer size: 19; skip: 10; is_hpc: 1; #seq: 1\n",
      "[M::mm_idx_stat::0.080*1.00] distinct minimizers: 521562 (99.34% are singletons); average occurrences: 1.014; average spacing: 7.461; total length: 3947620\n",
      "[M::worker_pipeline::0.612*1.00] mapped 1 sequences\n",
      "[M::main] Version: 2.19-r1057\n",
      "[M::main] CMD: minimap2 -ax asm20 --eqx -H -f 100 -r1k,10k --rmq=no -t 1 exp/multiple_runs11/tmp/L09-4/is_deleted/L09-4_genome.deleted_IS.1.fasta exp/multiple_runs11/tmp/L09-4/is_deleted/L09-4_genome.deleted_IS.3.fasta\n",
      "[M::main] Real time: 0.615 sec; CPU: 0.613 sec; Peak RSS: 0.086 GB\n",
      "local_variation - WARNING - Finished syri\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "invOut.txt  is empty. Skipping analysing it.\n",
      "TLOut.txt  is empty. Skipping analysing it.\n",
      "invTLOut.txt  is empty. Skipping analysing it.\n",
      "ctxOut.txt is empty. Skipping analysing it.\n",
      "inv Out.txt is empty. Skipping analysing it.\n",
      "TL Out.txt is empty. Skipping analysing it.\n",
      "invTL Out.txt is empty. Skipping analysing it.\n",
      "dup Out.txt is empty. Skipping analysing it.\n",
      "invDup Out.txt is empty. Skipping analysing it.\n",
      "ctxOut.txt is empty. Skipping analysing it.\n",
      "invOut.txt  is empty. Skipping analysing it.\n",
      "TLOut.txt  is empty. Skipping analysing it.\n",
      "invTLOut.txt  is empty. Skipping analysing it.\n",
      "ctxOut.txt is empty. Skipping analysing it.\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "[M::mm_idx_gen::0.063*1.01] collected minimizers\n",
      "[M::mm_idx_gen::0.081*1.00] sorted minimizers\n",
      "[M::main::0.081*1.00] loaded/built the index for 1 target sequence(s)\n",
      "[M::mm_mapopt_update::0.081*1.00] mid_occ = 100\n",
      "[M::mm_idx_stat] kmer size: 19; skip: 10; is_hpc: 1; #seq: 1\n",
      "[M::mm_idx_stat::0.085*1.00] distinct minimizers: 521662 (99.33% are singletons); average occurrences: 1.015; average spacing: 7.460; total length: 3948368\n",
      "[M::worker_pipeline::0.620*1.00] mapped 1 sequences\n",
      "[M::main] Version: 2.19-r1057\n",
      "[M::main] CMD: minimap2 -ax asm20 --eqx -H -f 100 -r1k,10k --rmq=no -t 1 exp/multiple_runs11/tmp/L10-1/is_deleted/L10-1_genome.deleted_IS.1.fasta exp/multiple_runs11/tmp/L10-1/is_deleted/L10-1_genome.deleted_IS.2.fasta\n",
      "[M::main] Real time: 0.623 sec; CPU: 0.622 sec; Peak RSS: 0.086 GB\n",
      "local_variation - WARNING - Finished syri\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "invOut.txt  is empty. Skipping analysing it.\n",
      "TLOut.txt  is empty. Skipping analysing it.\n",
      "invTLOut.txt  is empty. Skipping analysing it.\n",
      "ctxOut.txt is empty. Skipping analysing it.\n",
      "inv Out.txt is empty. Skipping analysing it.\n",
      "TL Out.txt is empty. Skipping analysing it.\n",
      "invTL Out.txt is empty. Skipping analysing it.\n",
      "dup Out.txt is empty. Skipping analysing it.\n",
      "invDup Out.txt is empty. Skipping analysing it.\n",
      "ctxOut.txt is empty. Skipping analysing it.\n",
      "invOut.txt  is empty. Skipping analysing it.\n",
      "TLOut.txt  is empty. Skipping analysing it.\n",
      "invTLOut.txt  is empty. Skipping analysing it.\n",
      "ctxOut.txt is empty. Skipping analysing it.\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "[M::mm_idx_gen::0.058*1.01] collected minimizers\n",
      "[M::mm_idx_gen::0.077*1.00] sorted minimizers\n",
      "[M::main::0.077*1.00] loaded/built the index for 1 target sequence(s)\n",
      "[M::mm_mapopt_update::0.077*1.00] mid_occ = 100\n",
      "[M::mm_idx_stat] kmer size: 19; skip: 10; is_hpc: 1; #seq: 1\n",
      "[M::mm_idx_stat::0.080*1.00] distinct minimizers: 521662 (99.33% are singletons); average occurrences: 1.015; average spacing: 7.460; total length: 3948368\n",
      "[M::worker_pipeline::0.615*0.99] mapped 1 sequences\n",
      "[M::main] Version: 2.19-r1057\n",
      "[M::main] CMD: minimap2 -ax asm20 --eqx -H -f 100 -r1k,10k --rmq=no -t 1 exp/multiple_runs11/tmp/L10-1/is_deleted/L10-1_genome.deleted_IS.1.fasta exp/multiple_runs11/tmp/L10-1/is_deleted/L10-1_genome.deleted_IS.3.fasta\n",
      "[M::main] Real time: 0.619 sec; CPU: 0.613 sec; Peak RSS: 0.088 GB\n",
      "local_variation - WARNING - Finished syri\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "invOut.txt  is empty. Skipping analysing it.\n",
      "TLOut.txt  is empty. Skipping analysing it.\n",
      "invTLOut.txt  is empty. Skipping analysing it.\n",
      "ctxOut.txt is empty. Skipping analysing it.\n",
      "inv Out.txt is empty. Skipping analysing it.\n",
      "TL Out.txt is empty. Skipping analysing it.\n",
      "invTL Out.txt is empty. Skipping analysing it.\n",
      "dup Out.txt is empty. Skipping analysing it.\n",
      "invDup Out.txt is empty. Skipping analysing it.\n",
      "ctxOut.txt is empty. Skipping analysing it.\n",
      "invOut.txt  is empty. Skipping analysing it.\n",
      "TLOut.txt  is empty. Skipping analysing it.\n",
      "invTLOut.txt  is empty. Skipping analysing it.\n",
      "ctxOut.txt is empty. Skipping analysing it.\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "[M::mm_idx_gen::0.063*1.01] collected minimizers\n",
      "[M::mm_idx_gen::0.082*1.00] sorted minimizers\n",
      "[M::main::0.082*1.00] loaded/built the index for 1 target sequence(s)\n",
      "[M::mm_mapopt_update::0.082*1.00] mid_occ = 100\n",
      "[M::mm_idx_stat] kmer size: 19; skip: 10; is_hpc: 1; #seq: 1\n",
      "[M::mm_idx_stat::0.086*1.00] distinct minimizers: 521662 (99.33% are singletons); average occurrences: 1.015; average spacing: 7.460; total length: 3948368\n",
      "[M::worker_pipeline::0.626*0.99] mapped 1 sequences\n",
      "[M::main] Version: 2.19-r1057\n",
      "[M::main] CMD: minimap2 -ax asm20 --eqx -H -f 100 -r1k,10k --rmq=no -t 1 exp/multiple_runs11/tmp/L10-2/is_deleted/L10-2_genome.deleted_IS.1.fasta exp/multiple_runs11/tmp/L10-2/is_deleted/L10-2_genome.deleted_IS.2.fasta\n",
      "[M::main] Real time: 0.629 sec; CPU: 0.623 sec; Peak RSS: 0.088 GB\n",
      "local_variation - WARNING - Finished syri\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "invOut.txt  is empty. Skipping analysing it.\n",
      "TLOut.txt  is empty. Skipping analysing it.\n",
      "invTLOut.txt  is empty. Skipping analysing it.\n",
      "ctxOut.txt is empty. Skipping analysing it.\n",
      "inv Out.txt is empty. Skipping analysing it.\n",
      "TL Out.txt is empty. Skipping analysing it.\n",
      "invTL Out.txt is empty. Skipping analysing it.\n",
      "dup Out.txt is empty. Skipping analysing it.\n",
      "ctxOut.txt is empty. Skipping analysing it.\n",
      "invOut.txt  is empty. Skipping analysing it.\n",
      "TLOut.txt  is empty. Skipping analysing it.\n",
      "invTLOut.txt  is empty. Skipping analysing it.\n",
      "ctxOut.txt is empty. Skipping analysing it.\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "[M::mm_idx_gen::0.058*1.01] collected minimizers\n",
      "[M::mm_idx_gen::0.076*1.00] sorted minimizers\n",
      "[M::main::0.076*1.00] loaded/built the index for 1 target sequence(s)\n",
      "[M::mm_mapopt_update::0.076*1.00] mid_occ = 100\n",
      "[M::mm_idx_stat] kmer size: 19; skip: 10; is_hpc: 1; #seq: 1\n",
      "[M::mm_idx_stat::0.080*1.00] distinct minimizers: 521662 (99.33% are singletons); average occurrences: 1.015; average spacing: 7.460; total length: 3948368\n",
      "[M::worker_pipeline::0.615*0.99] mapped 1 sequences\n",
      "[M::main] Version: 2.19-r1057\n",
      "[M::main] CMD: minimap2 -ax asm20 --eqx -H -f 100 -r1k,10k --rmq=no -t 1 exp/multiple_runs11/tmp/L10-2/is_deleted/L10-2_genome.deleted_IS.1.fasta exp/multiple_runs11/tmp/L10-2/is_deleted/L10-2_genome.deleted_IS.3.fasta\n",
      "[M::main] Real time: 0.617 sec; CPU: 0.611 sec; Peak RSS: 0.087 GB\n",
      "local_variation - WARNING - Finished syri\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "TLOut.txt  is empty. Skipping analysing it.\n",
      "ctxOut.txt is empty. Skipping analysing it.\n",
      "TL Out.txt is empty. Skipping analysing it.\n",
      "dup Out.txt is empty. Skipping analysing it.\n",
      "invDup Out.txt is empty. Skipping analysing it.\n",
      "ctxOut.txt is empty. Skipping analysing it.\n",
      "TLOut.txt  is empty. Skipping analysing it.\n",
      "ctxOut.txt is empty. Skipping analysing it.\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "[M::mm_idx_gen::0.063*1.01] collected minimizers\n",
      "[M::mm_idx_gen::0.082*1.00] sorted minimizers\n",
      "[M::main::0.082*1.00] loaded/built the index for 1 target sequence(s)\n",
      "[M::mm_mapopt_update::0.082*1.00] mid_occ = 100\n",
      "[M::mm_idx_stat] kmer size: 19; skip: 10; is_hpc: 1; #seq: 1\n",
      "[M::mm_idx_stat::0.085*1.00] distinct minimizers: 521662 (99.33% are singletons); average occurrences: 1.015; average spacing: 7.460; total length: 3948368\n",
      "[M::worker_pipeline::0.633*0.99] mapped 1 sequences\n",
      "[M::main] Version: 2.19-r1057\n",
      "[M::main] CMD: minimap2 -ax asm20 --eqx -H -f 100 -r1k,10k --rmq=no -t 1 exp/multiple_runs11/tmp/L10-3/is_deleted/L10-3_genome.deleted_IS.1.fasta exp/multiple_runs11/tmp/L10-3/is_deleted/L10-3_genome.deleted_IS.2.fasta\n",
      "[M::main] Real time: 0.636 sec; CPU: 0.629 sec; Peak RSS: 0.106 GB\n",
      "local_variation - WARNING - Finished syri\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "TLOut.txt  is empty. Skipping analysing it.\n",
      "invTLOut.txt  is empty. Skipping analysing it.\n",
      "ctxOut.txt is empty. Skipping analysing it.\n",
      "TL Out.txt is empty. Skipping analysing it.\n",
      "invTL Out.txt is empty. Skipping analysing it.\n",
      "dup Out.txt is empty. Skipping analysing it.\n",
      "invDup Out.txt is empty. Skipping analysing it.\n",
      "ctxOut.txt is empty. Skipping analysing it.\n",
      "TLOut.txt  is empty. Skipping analysing it.\n",
      "invTLOut.txt  is empty. Skipping analysing it.\n",
      "ctxOut.txt is empty. Skipping analysing it.\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "[M::mm_idx_gen::0.058*1.01] collected minimizers\n",
      "[M::mm_idx_gen::0.077*1.00] sorted minimizers\n",
      "[M::main::0.077*1.00] loaded/built the index for 1 target sequence(s)\n",
      "[M::mm_mapopt_update::0.077*1.00] mid_occ = 100\n",
      "[M::mm_idx_stat] kmer size: 19; skip: 10; is_hpc: 1; #seq: 1\n",
      "[M::mm_idx_stat::0.080*1.00] distinct minimizers: 521662 (99.33% are singletons); average occurrences: 1.015; average spacing: 7.460; total length: 3948368\n",
      "[M::worker_pipeline::0.623*0.99] mapped 1 sequences\n",
      "[M::main] Version: 2.19-r1057\n",
      "[M::main] CMD: minimap2 -ax asm20 --eqx -H -f 100 -r1k,10k --rmq=no -t 1 exp/multiple_runs11/tmp/L10-3/is_deleted/L10-3_genome.deleted_IS.1.fasta exp/multiple_runs11/tmp/L10-3/is_deleted/L10-3_genome.deleted_IS.3.fasta\n",
      "[M::main] Real time: 0.625 sec; CPU: 0.619 sec; Peak RSS: 0.086 GB\n",
      "local_variation - WARNING - Finished syri\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "TLOut.txt  is empty. Skipping analysing it.\n",
      "invTLOut.txt  is empty. Skipping analysing it.\n",
      "ctxOut.txt is empty. Skipping analysing it.\n",
      "TL Out.txt is empty. Skipping analysing it.\n",
      "invTL Out.txt is empty. Skipping analysing it.\n",
      "dup Out.txt is empty. Skipping analysing it.\n",
      "invDup Out.txt is empty. Skipping analysing it.\n",
      "ctxOut.txt is empty. Skipping analysing it.\n",
      "TLOut.txt  is empty. Skipping analysing it.\n",
      "invTLOut.txt  is empty. Skipping analysing it.\n",
      "ctxOut.txt is empty. Skipping analysing it.\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "[M::mm_idx_gen::0.060*1.01] collected minimizers\n",
      "[M::mm_idx_gen::0.079*1.00] sorted minimizers\n",
      "[M::main::0.079*1.00] loaded/built the index for 1 target sequence(s)\n",
      "[M::mm_mapopt_update::0.079*1.00] mid_occ = 100\n",
      "[M::mm_idx_stat] kmer size: 19; skip: 10; is_hpc: 1; #seq: 1\n",
      "[M::mm_idx_stat::0.083*1.00] distinct minimizers: 521662 (99.33% are singletons); average occurrences: 1.015; average spacing: 7.460; total length: 3948368\n",
      "[M::worker_pipeline::0.621*0.99] mapped 1 sequences\n",
      "[M::main] Version: 2.19-r1057\n",
      "[M::main] CMD: minimap2 -ax asm20 --eqx -H -f 100 -r1k,10k --rmq=no -t 1 exp/multiple_runs11/tmp/L10-4/is_deleted/L10-4_genome.deleted_IS.1.fasta exp/multiple_runs11/tmp/L10-4/is_deleted/L10-4_genome.deleted_IS.2.fasta\n",
      "[M::main] Real time: 0.623 sec; CPU: 0.617 sec; Peak RSS: 0.086 GB\n",
      "local_variation - WARNING - Finished syri\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "TLOut.txt  is empty. Skipping analysing it.\n",
      "invTLOut.txt  is empty. Skipping analysing it.\n",
      "ctxOut.txt is empty. Skipping analysing it.\n",
      "TL Out.txt is empty. Skipping analysing it.\n",
      "invTL Out.txt is empty. Skipping analysing it.\n",
      "dup Out.txt is empty. Skipping analysing it.\n",
      "invDup Out.txt is empty. Skipping analysing it.\n",
      "ctxOut.txt is empty. Skipping analysing it.\n",
      "TLOut.txt  is empty. Skipping analysing it.\n",
      "invTLOut.txt  is empty. Skipping analysing it.\n",
      "ctxOut.txt is empty. Skipping analysing it.\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "[M::mm_idx_gen::0.057*1.01] collected minimizers\n",
      "[M::mm_idx_gen::0.075*1.00] sorted minimizers\n",
      "[M::main::0.076*1.00] loaded/built the index for 1 target sequence(s)\n",
      "[M::mm_mapopt_update::0.076*1.00] mid_occ = 100\n",
      "[M::mm_idx_stat] kmer size: 19; skip: 10; is_hpc: 1; #seq: 1\n",
      "[M::mm_idx_stat::0.079*1.00] distinct minimizers: 521662 (99.33% are singletons); average occurrences: 1.015; average spacing: 7.460; total length: 3948368\n",
      "[M::worker_pipeline::0.623*0.99] mapped 1 sequences\n",
      "[M::main] Version: 2.19-r1057\n",
      "[M::main] CMD: minimap2 -ax asm20 --eqx -H -f 100 -r1k,10k --rmq=no -t 1 exp/multiple_runs11/tmp/L10-4/is_deleted/L10-4_genome.deleted_IS.1.fasta exp/multiple_runs11/tmp/L10-4/is_deleted/L10-4_genome.deleted_IS.3.fasta\n",
      "[M::main] Real time: 0.625 sec; CPU: 0.619 sec; Peak RSS: 0.086 GB\n",
      "local_variation - WARNING - Finished syri\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "TLOut.txt  is empty. Skipping analysing it.\n",
      "invTLOut.txt  is empty. Skipping analysing it.\n",
      "ctxOut.txt is empty. Skipping analysing it.\n",
      "TL Out.txt is empty. Skipping analysing it.\n",
      "invTL Out.txt is empty. Skipping analysing it.\n",
      "invDup Out.txt is empty. Skipping analysing it.\n",
      "ctxOut.txt is empty. Skipping analysing it.\n",
      "TLOut.txt  is empty. Skipping analysing it.\n",
      "invTLOut.txt  is empty. Skipping analysing it.\n",
      "ctxOut.txt is empty. Skipping analysing it.\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "[M::mm_idx_gen::0.058*1.01] collected minimizers\n",
      "[M::mm_idx_gen::0.076*1.00] sorted minimizers\n",
      "[M::main::0.076*1.00] loaded/built the index for 1 target sequence(s)\n",
      "[M::mm_mapopt_update::0.076*1.00] mid_occ = 100\n",
      "[M::mm_idx_stat] kmer size: 19; skip: 10; is_hpc: 1; #seq: 1\n",
      "[M::mm_idx_stat::0.080*1.00] distinct minimizers: 524338 (99.34% are singletons); average occurrences: 1.014; average spacing: 7.460; total length: 3967244\n",
      "[M::worker_pipeline::0.616*1.00] mapped 1 sequences\n",
      "[M::main] Version: 2.19-r1057\n",
      "[M::main] CMD: minimap2 -ax asm20 --eqx -H -f 100 -r1k,10k --rmq=no -t 1 exp/multiple_runs11/tmp/L11-1/is_deleted/L11-1_genome.deleted_IS.1.fasta exp/multiple_runs11/tmp/L11-1/is_deleted/L11-1_genome.deleted_IS.2.fasta\n",
      "[M::main] Real time: 0.618 sec; CPU: 0.617 sec; Peak RSS: 0.087 GB\n",
      "local_variation - WARNING - Finished syri\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "invOut.txt  is empty. Skipping analysing it.\n",
      "TLOut.txt  is empty. Skipping analysing it.\n",
      "invTLOut.txt  is empty. Skipping analysing it.\n",
      "ctxOut.txt is empty. Skipping analysing it.\n",
      "inv Out.txt is empty. Skipping analysing it.\n",
      "TL Out.txt is empty. Skipping analysing it.\n",
      "invTL Out.txt is empty. Skipping analysing it.\n",
      "dup Out.txt is empty. Skipping analysing it.\n",
      "invDup Out.txt is empty. Skipping analysing it.\n",
      "ctxOut.txt is empty. Skipping analysing it.\n",
      "invOut.txt  is empty. Skipping analysing it.\n",
      "TLOut.txt  is empty. Skipping analysing it.\n",
      "invTLOut.txt  is empty. Skipping analysing it.\n",
      "ctxOut.txt is empty. Skipping analysing it.\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "[M::mm_idx_gen::0.058*1.01] collected minimizers\n",
      "[M::mm_idx_gen::0.076*1.00] sorted minimizers\n",
      "[M::main::0.076*1.00] loaded/built the index for 1 target sequence(s)\n",
      "[M::mm_mapopt_update::0.076*1.00] mid_occ = 100\n",
      "[M::mm_idx_stat] kmer size: 19; skip: 10; is_hpc: 1; #seq: 1\n",
      "[M::mm_idx_stat::0.080*1.00] distinct minimizers: 524338 (99.34% are singletons); average occurrences: 1.014; average spacing: 7.460; total length: 3967244\n",
      "[M::worker_pipeline::0.613*0.99] mapped 1 sequences\n",
      "[M::main] Version: 2.19-r1057\n",
      "[M::main] CMD: minimap2 -ax asm20 --eqx -H -f 100 -r1k,10k --rmq=no -t 1 exp/multiple_runs11/tmp/L11-1/is_deleted/L11-1_genome.deleted_IS.1.fasta exp/multiple_runs11/tmp/L11-1/is_deleted/L11-1_genome.deleted_IS.3.fasta\n",
      "[M::main] Real time: 0.615 sec; CPU: 0.609 sec; Peak RSS: 0.088 GB\n",
      "local_variation - WARNING - Finished syri\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "invOut.txt  is empty. Skipping analysing it.\n",
      "TLOut.txt  is empty. Skipping analysing it.\n",
      "invTLOut.txt  is empty. Skipping analysing it.\n",
      "ctxOut.txt is empty. Skipping analysing it.\n",
      "inv Out.txt is empty. Skipping analysing it.\n",
      "TL Out.txt is empty. Skipping analysing it.\n",
      "invTL Out.txt is empty. Skipping analysing it.\n",
      "dup Out.txt is empty. Skipping analysing it.\n",
      "invDup Out.txt is empty. Skipping analysing it.\n",
      "ctxOut.txt is empty. Skipping analysing it.\n",
      "invOut.txt  is empty. Skipping analysing it.\n",
      "TLOut.txt  is empty. Skipping analysing it.\n",
      "invTLOut.txt  is empty. Skipping analysing it.\n",
      "ctxOut.txt is empty. Skipping analysing it.\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "[M::mm_idx_gen::0.061*1.01] collected minimizers\n",
      "[M::mm_idx_gen::0.080*1.01] sorted minimizers\n",
      "[M::main::0.080*1.01] loaded/built the index for 1 target sequence(s)\n",
      "[M::mm_mapopt_update::0.080*1.01] mid_occ = 100\n",
      "[M::mm_idx_stat] kmer size: 19; skip: 10; is_hpc: 1; #seq: 1\n",
      "[M::mm_idx_stat::0.084*1.01] distinct minimizers: 524338 (99.34% are singletons); average occurrences: 1.014; average spacing: 7.460; total length: 3967244\n",
      "[M::worker_pipeline::0.621*0.99] mapped 1 sequences\n",
      "[M::main] Version: 2.19-r1057\n",
      "[M::main] CMD: minimap2 -ax asm20 --eqx -H -f 100 -r1k,10k --rmq=no -t 1 exp/multiple_runs11/tmp/L11-2/is_deleted/L11-2_genome.deleted_IS.1.fasta exp/multiple_runs11/tmp/L11-2/is_deleted/L11-2_genome.deleted_IS.2.fasta\n",
      "[M::main] Real time: 0.624 sec; CPU: 0.617 sec; Peak RSS: 0.087 GB\n",
      "local_variation - WARNING - Finished syri\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "invOut.txt  is empty. Skipping analysing it.\n",
      "TLOut.txt  is empty. Skipping analysing it.\n",
      "invTLOut.txt  is empty. Skipping analysing it.\n",
      "ctxOut.txt is empty. Skipping analysing it.\n",
      "inv Out.txt is empty. Skipping analysing it.\n",
      "TL Out.txt is empty. Skipping analysing it.\n",
      "invTL Out.txt is empty. Skipping analysing it.\n",
      "ctxOut.txt is empty. Skipping analysing it.\n",
      "invOut.txt  is empty. Skipping analysing it.\n",
      "TLOut.txt  is empty. Skipping analysing it.\n",
      "invTLOut.txt  is empty. Skipping analysing it.\n",
      "ctxOut.txt is empty. Skipping analysing it.\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "[M::mm_idx_gen::0.063*1.01] collected minimizers\n",
      "[M::mm_idx_gen::0.082*1.00] sorted minimizers\n",
      "[M::main::0.082*1.00] loaded/built the index for 1 target sequence(s)\n",
      "[M::mm_mapopt_update::0.082*1.00] mid_occ = 100\n",
      "[M::mm_idx_stat] kmer size: 19; skip: 10; is_hpc: 1; #seq: 1\n",
      "[M::mm_idx_stat::0.086*1.00] distinct minimizers: 524338 (99.34% are singletons); average occurrences: 1.014; average spacing: 7.460; total length: 3967244\n",
      "[M::worker_pipeline::0.640*0.99] mapped 1 sequences\n",
      "[M::main] Version: 2.19-r1057\n",
      "[M::main] CMD: minimap2 -ax asm20 --eqx -H -f 100 -r1k,10k --rmq=no -t 1 exp/multiple_runs11/tmp/L11-2/is_deleted/L11-2_genome.deleted_IS.1.fasta exp/multiple_runs11/tmp/L11-2/is_deleted/L11-2_genome.deleted_IS.3.fasta\n",
      "[M::main] Real time: 0.642 sec; CPU: 0.636 sec; Peak RSS: 0.088 GB\n",
      "local_variation - WARNING - Finished syri\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "TLOut.txt  is empty. Skipping analysing it.\n",
      "invTLOut.txt  is empty. Skipping analysing it.\n",
      "ctxOut.txt is empty. Skipping analysing it.\n",
      "TL Out.txt is empty. Skipping analysing it.\n",
      "invTL Out.txt is empty. Skipping analysing it.\n",
      "ctxOut.txt is empty. Skipping analysing it.\n",
      "TLOut.txt  is empty. Skipping analysing it.\n",
      "invTLOut.txt  is empty. Skipping analysing it.\n",
      "ctxOut.txt is empty. Skipping analysing it.\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "[M::mm_idx_gen::0.056*1.01] collected minimizers\n",
      "[M::mm_idx_gen::0.075*1.00] sorted minimizers\n",
      "[M::main::0.075*1.00] loaded/built the index for 1 target sequence(s)\n",
      "[M::mm_mapopt_update::0.075*1.00] mid_occ = 100\n",
      "[M::mm_idx_stat] kmer size: 19; skip: 10; is_hpc: 1; #seq: 1\n",
      "[M::mm_idx_stat::0.079*1.00] distinct minimizers: 524338 (99.34% are singletons); average occurrences: 1.014; average spacing: 7.460; total length: 3967244\n",
      "[M::worker_pipeline::0.616*0.99] mapped 1 sequences\n",
      "[M::main] Version: 2.19-r1057\n",
      "[M::main] CMD: minimap2 -ax asm20 --eqx -H -f 100 -r1k,10k --rmq=no -t 1 exp/multiple_runs11/tmp/L11-3/is_deleted/L11-3_genome.deleted_IS.1.fasta exp/multiple_runs11/tmp/L11-3/is_deleted/L11-3_genome.deleted_IS.2.fasta\n",
      "[M::main] Real time: 0.619 sec; CPU: 0.614 sec; Peak RSS: 0.086 GB\n",
      "local_variation - WARNING - Finished syri\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "invOut.txt  is empty. Skipping analysing it.\n",
      "TLOut.txt  is empty. Skipping analysing it.\n",
      "invTLOut.txt  is empty. Skipping analysing it.\n",
      "ctxOut.txt is empty. Skipping analysing it.\n",
      "inv Out.txt is empty. Skipping analysing it.\n",
      "TL Out.txt is empty. Skipping analysing it.\n",
      "invTL Out.txt is empty. Skipping analysing it.\n",
      "dup Out.txt is empty. Skipping analysing it.\n",
      "invDup Out.txt is empty. Skipping analysing it.\n",
      "ctxOut.txt is empty. Skipping analysing it.\n",
      "invOut.txt  is empty. Skipping analysing it.\n",
      "TLOut.txt  is empty. Skipping analysing it.\n",
      "invTLOut.txt  is empty. Skipping analysing it.\n",
      "ctxOut.txt is empty. Skipping analysing it.\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "[M::mm_idx_gen::0.061*1.01] collected minimizers\n",
      "[M::mm_idx_gen::0.080*1.00] sorted minimizers\n",
      "[M::main::0.080*1.00] loaded/built the index for 1 target sequence(s)\n",
      "[M::mm_mapopt_update::0.080*1.00] mid_occ = 100\n",
      "[M::mm_idx_stat] kmer size: 19; skip: 10; is_hpc: 1; #seq: 1\n",
      "[M::mm_idx_stat::0.084*1.00] distinct minimizers: 524338 (99.34% are singletons); average occurrences: 1.014; average spacing: 7.460; total length: 3967244\n",
      "[M::worker_pipeline::0.611*0.99] mapped 1 sequences\n",
      "[M::main] Version: 2.19-r1057\n",
      "[M::main] CMD: minimap2 -ax asm20 --eqx -H -f 100 -r1k,10k --rmq=no -t 1 exp/multiple_runs11/tmp/L11-3/is_deleted/L11-3_genome.deleted_IS.1.fasta exp/multiple_runs11/tmp/L11-3/is_deleted/L11-3_genome.deleted_IS.3.fasta\n",
      "[M::main] Real time: 0.613 sec; CPU: 0.608 sec; Peak RSS: 0.087 GB\n",
      "local_variation - WARNING - Finished syri\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "invOut.txt  is empty. Skipping analysing it.\n",
      "TLOut.txt  is empty. Skipping analysing it.\n",
      "invTLOut.txt  is empty. Skipping analysing it.\n",
      "ctxOut.txt is empty. Skipping analysing it.\n",
      "inv Out.txt is empty. Skipping analysing it.\n",
      "TL Out.txt is empty. Skipping analysing it.\n",
      "invTL Out.txt is empty. Skipping analysing it.\n",
      "dup Out.txt is empty. Skipping analysing it.\n",
      "invDup Out.txt is empty. Skipping analysing it.\n",
      "ctxOut.txt is empty. Skipping analysing it.\n",
      "invOut.txt  is empty. Skipping analysing it.\n",
      "TLOut.txt  is empty. Skipping analysing it.\n",
      "invTLOut.txt  is empty. Skipping analysing it.\n",
      "ctxOut.txt is empty. Skipping analysing it.\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "[M::mm_idx_gen::0.057*1.01] collected minimizers\n",
      "[M::mm_idx_gen::0.076*1.00] sorted minimizers\n",
      "[M::main::0.076*1.00] loaded/built the index for 1 target sequence(s)\n",
      "[M::mm_mapopt_update::0.076*1.00] mid_occ = 100\n",
      "[M::mm_idx_stat] kmer size: 19; skip: 10; is_hpc: 1; #seq: 1\n",
      "[M::mm_idx_stat::0.079*1.00] distinct minimizers: 524338 (99.34% are singletons); average occurrences: 1.014; average spacing: 7.460; total length: 3967244\n",
      "[M::worker_pipeline::0.620*0.99] mapped 1 sequences\n",
      "[M::main] Version: 2.19-r1057\n",
      "[M::main] CMD: minimap2 -ax asm20 --eqx -H -f 100 -r1k,10k --rmq=no -t 1 exp/multiple_runs11/tmp/L11-4/is_deleted/L11-4_genome.deleted_IS.1.fasta exp/multiple_runs11/tmp/L11-4/is_deleted/L11-4_genome.deleted_IS.2.fasta\n",
      "[M::main] Real time: 0.623 sec; CPU: 0.617 sec; Peak RSS: 0.087 GB\n",
      "local_variation - WARNING - Finished syri\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "TLOut.txt  is empty. Skipping analysing it.\n",
      "invTLOut.txt  is empty. Skipping analysing it.\n",
      "ctxOut.txt is empty. Skipping analysing it.\n",
      "TL Out.txt is empty. Skipping analysing it.\n",
      "invTL Out.txt is empty. Skipping analysing it.\n",
      "dup Out.txt is empty. Skipping analysing it.\n",
      "invDup Out.txt is empty. Skipping analysing it.\n",
      "ctxOut.txt is empty. Skipping analysing it.\n",
      "TLOut.txt  is empty. Skipping analysing it.\n",
      "invTLOut.txt  is empty. Skipping analysing it.\n",
      "ctxOut.txt is empty. Skipping analysing it.\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "[M::mm_idx_gen::0.056*1.01] collected minimizers\n",
      "[M::mm_idx_gen::0.075*1.00] sorted minimizers\n",
      "[M::main::0.075*1.00] loaded/built the index for 1 target sequence(s)\n",
      "[M::mm_mapopt_update::0.075*1.00] mid_occ = 100\n",
      "[M::mm_idx_stat] kmer size: 19; skip: 10; is_hpc: 1; #seq: 1\n",
      "[M::mm_idx_stat::0.079*1.00] distinct minimizers: 524338 (99.34% are singletons); average occurrences: 1.014; average spacing: 7.460; total length: 3967244\n",
      "[M::worker_pipeline::0.654*0.99] mapped 1 sequences\n",
      "[M::main] Version: 2.19-r1057\n",
      "[M::main] CMD: minimap2 -ax asm20 --eqx -H -f 100 -r1k,10k --rmq=no -t 1 exp/multiple_runs11/tmp/L11-4/is_deleted/L11-4_genome.deleted_IS.1.fasta exp/multiple_runs11/tmp/L11-4/is_deleted/L11-4_genome.deleted_IS.3.fasta\n",
      "[M::main] Real time: 0.657 sec; CPU: 0.650 sec; Peak RSS: 0.090 GB\n",
      "local_variation - WARNING - Finished syri\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "TLOut.txt  is empty. Skipping analysing it.\n",
      "invTLOut.txt  is empty. Skipping analysing it.\n",
      "ctxOut.txt is empty. Skipping analysing it.\n",
      "TL Out.txt is empty. Skipping analysing it.\n",
      "invTL Out.txt is empty. Skipping analysing it.\n",
      "dup Out.txt is empty. Skipping analysing it.\n",
      "ctxOut.txt is empty. Skipping analysing it.\n",
      "TLOut.txt  is empty. Skipping analysing it.\n",
      "invTLOut.txt  is empty. Skipping analysing it.\n",
      "ctxOut.txt is empty. Skipping analysing it.\n"
     ]
    }
   ],
   "source": [
    "import is_event_classification_func as iecf\n",
    "max_run = 1\n",
    "run_cnt =0\n",
    "for i, line in enumerate(lines_):\n",
    "\tC_ = 'python3 ../src/SyRI_IS/classify_IS_events.py '\n",
    "\tC_ += ' --prefix ' + line \n",
    "\tC_ += ' --out ' + os.path.join(exportdir, 'classify_IS_events_Anc')\n",
    "\tC_ += ' --log ' + logdir\n",
    "\tC_ += ' --input ' + os.path.join(tmpdir, line)\n",
    "\tC_ += ' --calc_depth_interval ' \n",
    "\tfor gen_id, g in enumerate(file_df[file_df.Prefix == line].file_name):\n",
    "\t\t# genid 0 is the ancestor obtained from FACS\n",
    "\t\tif gen_id == 0:\n",
    "\t\t\tcontinue\n",
    "\t\tC = C_ + ' --rid ' + str(1) + ' --qid ' + str(gen_id+1)\n",
    "\t\tif i ==0 and gen_id == 0:\n",
    "\t\t\tprint(C)\n",
    "\t\tiecf.run_syri_isrm_for_specific_id(tmpdir, line, 1, gen_id+1)\n",
    "\t\tos.system(C)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 13,
   "metadata": {},
   "outputs": [],
   "source": [
    "lines_ = lines\n",
    "depth_interval_df_anc = pd.DataFrame()\n",
    "for line in lines_:\n",
    "\tfor qid in [2,3]:\n",
    "\t\tdf_ = pd.read_csv(os.path.join(exportdir, 'classify_IS_events_Anc', line, f'{line}_genome.{qid}.depth_interval.csv'))\n",
    "\t\tdf_['line'] = line\n",
    "\t\tdf_['gen_id'] = qid\n",
    "\t\tdepth_interval_df_anc = pd.concat([depth_interval_df_anc, df_], axis=0)\n",
    "depth_interval_df_anc= depth_interval_df_anc.reset_index(drop=True)\n",
    "depth_interval_df_anc['gen_id'] = depth_interval_df_anc['gen_id'].astype(int)\n",
    "depth_interval_df_anc.to_csv(os.path.join(exportdir, 'classify_IS_events_Anc', 'depth_interval.asm20.csv'), index=False)"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "## IS events compared to MDS42\n",
    "The previous analysis was based on the comparison between two consecutive generations. Here we compare the IS events to the MDS42 generation."
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 14,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "python3 ../src/SyRI_IS/classify_IS_events.py  --prefix L01-1 --out exp/multiple_runs11/export/classify_IS_events_MDS --log exp/multiple_runs11/log --input exp/multiple_runs11/tmp/L01-1 --calc_depth_interval  --rid 0 --qid 1\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "[M::mm_idx_gen::0.058*1.01] collected minimizers\n",
      "[M::mm_idx_gen::0.076*1.00] sorted minimizers\n",
      "[M::main::0.076*1.00] loaded/built the index for 1 target sequence(s)\n",
      "[M::mm_mapopt_update::0.076*1.00] mid_occ = 100\n",
      "[M::mm_idx_stat] kmer size: 19; skip: 10; is_hpc: 1; #seq: 1\n",
      "[M::mm_idx_stat::0.080*1.00] distinct minimizers: 525478 (99.34% are singletons); average occurrences: 1.015; average spacing: 7.461; total length: 3977333\n",
      "[M::worker_pipeline::0.611*0.99] mapped 1 sequences\n",
      "[M::main] Version: 2.19-r1057\n",
      "[M::main] CMD: minimap2 -ax asm20 --eqx -H -f 100 -r1k,10k --rmq=no -t 1 exp/multiple_runs11/tmp/L01-1/is_deleted/L01-1_genome.deleted_IS.0.fasta exp/multiple_runs11/tmp/L01-1/is_deleted/L01-1_genome.deleted_IS.1.fasta\n",
      "[M::main] Real time: 0.614 sec; CPU: 0.608 sec; Peak RSS: 0.087 GB\n",
      "local_variation - WARNING - Finished syri\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "TLOut.txt  is empty. Skipping analysing it.\n",
      "invTLOut.txt  is empty. Skipping analysing it.\n",
      "ctxOut.txt is empty. Skipping analysing it.\n",
      "TL Out.txt is empty. Skipping analysing it.\n",
      "invTL Out.txt is empty. Skipping analysing it.\n",
      "dup Out.txt is empty. Skipping analysing it.\n",
      "invDup Out.txt is empty. Skipping analysing it.\n",
      "ctxOut.txt is empty. Skipping analysing it.\n",
      "TLOut.txt  is empty. Skipping analysing it.\n",
      "invTLOut.txt  is empty. Skipping analysing it.\n",
      "ctxOut.txt is empty. Skipping analysing it.\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "[M::mm_idx_gen::0.057*1.01] collected minimizers\n",
      "[M::mm_idx_gen::0.075*1.00] sorted minimizers\n",
      "[M::main::0.075*1.00] loaded/built the index for 1 target sequence(s)\n",
      "[M::mm_mapopt_update::0.075*1.00] mid_occ = 100\n",
      "[M::mm_idx_stat] kmer size: 19; skip: 10; is_hpc: 1; #seq: 1\n",
      "[M::mm_idx_stat::0.079*1.00] distinct minimizers: 525478 (99.34% are singletons); average occurrences: 1.015; average spacing: 7.461; total length: 3977333\n",
      "[M::worker_pipeline::0.611*0.99] mapped 1 sequences\n",
      "[M::main] Version: 2.19-r1057\n",
      "[M::main] CMD: minimap2 -ax asm20 --eqx -H -f 100 -r1k,10k --rmq=no -t 1 exp/multiple_runs11/tmp/L01-1/is_deleted/L01-1_genome.deleted_IS.0.fasta exp/multiple_runs11/tmp/L01-1/is_deleted/L01-1_genome.deleted_IS.2.fasta\n",
      "[M::main] Real time: 0.613 sec; CPU: 0.607 sec; Peak RSS: 0.090 GB\n",
      "local_variation - WARNING - Finished syri\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "invTLOut.txt  is empty. Skipping analysing it.\n",
      "ctxOut.txt is empty. Skipping analysing it.\n",
      "invTL Out.txt is empty. Skipping analysing it.\n",
      "dup Out.txt is empty. Skipping analysing it.\n",
      "invDup Out.txt is empty. Skipping analysing it.\n",
      "ctxOut.txt is empty. Skipping analysing it.\n",
      "invTLOut.txt  is empty. Skipping analysing it.\n",
      "ctxOut.txt is empty. Skipping analysing it.\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "[M::mm_idx_gen::0.056*1.01] collected minimizers\n",
      "[M::mm_idx_gen::0.075*1.00] sorted minimizers\n",
      "[M::main::0.075*1.00] loaded/built the index for 1 target sequence(s)\n",
      "[M::mm_mapopt_update::0.075*1.00] mid_occ = 100\n",
      "[M::mm_idx_stat] kmer size: 19; skip: 10; is_hpc: 1; #seq: 1\n",
      "[M::mm_idx_stat::0.079*1.00] distinct minimizers: 525478 (99.34% are singletons); average occurrences: 1.015; average spacing: 7.461; total length: 3977333\n",
      "[M::worker_pipeline::0.594*0.99] mapped 1 sequences\n",
      "[M::main] Version: 2.19-r1057\n",
      "[M::main] CMD: minimap2 -ax asm20 --eqx -H -f 100 -r1k,10k --rmq=no -t 1 exp/multiple_runs11/tmp/L01-1/is_deleted/L01-1_genome.deleted_IS.0.fasta exp/multiple_runs11/tmp/L01-1/is_deleted/L01-1_genome.deleted_IS.3.fasta\n",
      "[M::main] Real time: 0.597 sec; CPU: 0.591 sec; Peak RSS: 0.088 GB\n",
      "local_variation - WARNING - Finished syri\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "TLOut.txt  is empty. Skipping analysing it.\n",
      "ctxOut.txt is empty. Skipping analysing it.\n",
      "TL Out.txt is empty. Skipping analysing it.\n",
      "dup Out.txt is empty. Skipping analysing it.\n",
      "invDup Out.txt is empty. Skipping analysing it.\n",
      "ctxOut.txt is empty. Skipping analysing it.\n",
      "TLOut.txt  is empty. Skipping analysing it.\n",
      "ctxOut.txt is empty. Skipping analysing it.\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "[M::mm_idx_gen::0.057*1.01] collected minimizers\n",
      "[M::mm_idx_gen::0.076*1.00] sorted minimizers\n",
      "[M::main::0.076*1.00] loaded/built the index for 1 target sequence(s)\n",
      "[M::mm_mapopt_update::0.076*1.00] mid_occ = 100\n",
      "[M::mm_idx_stat] kmer size: 19; skip: 10; is_hpc: 1; #seq: 1\n",
      "[M::mm_idx_stat::0.080*1.00] distinct minimizers: 525478 (99.34% are singletons); average occurrences: 1.015; average spacing: 7.461; total length: 3977333\n",
      "[M::worker_pipeline::0.611*0.99] mapped 1 sequences\n",
      "[M::main] Version: 2.19-r1057\n",
      "[M::main] CMD: minimap2 -ax asm20 --eqx -H -f 100 -r1k,10k --rmq=no -t 1 exp/multiple_runs11/tmp/L01-2/is_deleted/L01-2_genome.deleted_IS.0.fasta exp/multiple_runs11/tmp/L01-2/is_deleted/L01-2_genome.deleted_IS.1.fasta\n",
      "[M::main] Real time: 0.613 sec; CPU: 0.607 sec; Peak RSS: 0.087 GB\n",
      "local_variation - WARNING - Finished syri\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "TLOut.txt  is empty. Skipping analysing it.\n",
      "invTLOut.txt  is empty. Skipping analysing it.\n",
      "ctxOut.txt is empty. Skipping analysing it.\n",
      "TL Out.txt is empty. Skipping analysing it.\n",
      "invTL Out.txt is empty. Skipping analysing it.\n",
      "dup Out.txt is empty. Skipping analysing it.\n",
      "invDup Out.txt is empty. Skipping analysing it.\n",
      "ctxOut.txt is empty. Skipping analysing it.\n",
      "TLOut.txt  is empty. Skipping analysing it.\n",
      "invTLOut.txt  is empty. Skipping analysing it.\n",
      "ctxOut.txt is empty. Skipping analysing it.\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "[M::mm_idx_gen::0.061*1.01] collected minimizers\n",
      "[M::mm_idx_gen::0.080*1.00] sorted minimizers\n",
      "[M::main::0.080*1.00] loaded/built the index for 1 target sequence(s)\n",
      "[M::mm_mapopt_update::0.080*1.00] mid_occ = 100\n",
      "[M::mm_idx_stat] kmer size: 19; skip: 10; is_hpc: 1; #seq: 1\n",
      "[M::mm_idx_stat::0.084*1.00] distinct minimizers: 525478 (99.34% are singletons); average occurrences: 1.015; average spacing: 7.461; total length: 3977333\n",
      "[M::worker_pipeline::0.614*0.99] mapped 1 sequences\n",
      "[M::main] Version: 2.19-r1057\n",
      "[M::main] CMD: minimap2 -ax asm20 --eqx -H -f 100 -r1k,10k --rmq=no -t 1 exp/multiple_runs11/tmp/L01-2/is_deleted/L01-2_genome.deleted_IS.0.fasta exp/multiple_runs11/tmp/L01-2/is_deleted/L01-2_genome.deleted_IS.2.fasta\n",
      "[M::main] Real time: 0.616 sec; CPU: 0.609 sec; Peak RSS: 0.088 GB\n",
      "local_variation - WARNING - Finished syri\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "ctxOut.txt is empty. Skipping analysing it.\n",
      "dup Out.txt is empty. Skipping analysing it.\n",
      "invDup Out.txt is empty. Skipping analysing it.\n",
      "ctxOut.txt is empty. Skipping analysing it.\n",
      "ctxOut.txt is empty. Skipping analysing it.\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "[M::mm_idx_gen::0.063*1.01] collected minimizers\n",
      "[M::mm_idx_gen::0.081*1.00] sorted minimizers\n",
      "[M::main::0.081*1.00] loaded/built the index for 1 target sequence(s)\n",
      "[M::mm_mapopt_update::0.081*1.00] mid_occ = 100\n",
      "[M::mm_idx_stat] kmer size: 19; skip: 10; is_hpc: 1; #seq: 1\n",
      "[M::mm_idx_stat::0.085*1.00] distinct minimizers: 525478 (99.34% are singletons); average occurrences: 1.015; average spacing: 7.461; total length: 3977333\n",
      "[M::worker_pipeline::0.621*0.99] mapped 1 sequences\n",
      "[M::main] Version: 2.19-r1057\n",
      "[M::main] CMD: minimap2 -ax asm20 --eqx -H -f 100 -r1k,10k --rmq=no -t 1 exp/multiple_runs11/tmp/L01-2/is_deleted/L01-2_genome.deleted_IS.0.fasta exp/multiple_runs11/tmp/L01-2/is_deleted/L01-2_genome.deleted_IS.3.fasta\n",
      "[M::main] Real time: 0.624 sec; CPU: 0.618 sec; Peak RSS: 0.089 GB\n",
      "local_variation - WARNING - Finished syri\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "ctxOut.txt is empty. Skipping analysing it.\n",
      "dup Out.txt is empty. Skipping analysing it.\n",
      "invDup Out.txt is empty. Skipping analysing it.\n",
      "ctxOut.txt is empty. Skipping analysing it.\n",
      "ctxOut.txt is empty. Skipping analysing it.\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "[M::mm_idx_gen::0.058*1.01] collected minimizers\n",
      "[M::mm_idx_gen::0.076*1.00] sorted minimizers\n",
      "[M::main::0.076*1.00] loaded/built the index for 1 target sequence(s)\n",
      "[M::mm_mapopt_update::0.076*1.00] mid_occ = 100\n",
      "[M::mm_idx_stat] kmer size: 19; skip: 10; is_hpc: 1; #seq: 1\n",
      "[M::mm_idx_stat::0.080*1.00] distinct minimizers: 525478 (99.34% are singletons); average occurrences: 1.015; average spacing: 7.461; total length: 3977333\n",
      "[M::worker_pipeline::0.610*0.99] mapped 1 sequences\n",
      "[M::main] Version: 2.19-r1057\n",
      "[M::main] CMD: minimap2 -ax asm20 --eqx -H -f 100 -r1k,10k --rmq=no -t 1 exp/multiple_runs11/tmp/L01-3/is_deleted/L01-3_genome.deleted_IS.0.fasta exp/multiple_runs11/tmp/L01-3/is_deleted/L01-3_genome.deleted_IS.1.fasta\n",
      "[M::main] Real time: 0.613 sec; CPU: 0.607 sec; Peak RSS: 0.087 GB\n",
      "local_variation - WARNING - Finished syri\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "TLOut.txt  is empty. Skipping analysing it.\n",
      "invTLOut.txt  is empty. Skipping analysing it.\n",
      "ctxOut.txt is empty. Skipping analysing it.\n",
      "TL Out.txt is empty. Skipping analysing it.\n",
      "invTL Out.txt is empty. Skipping analysing it.\n",
      "dup Out.txt is empty. Skipping analysing it.\n",
      "invDup Out.txt is empty. Skipping analysing it.\n",
      "ctxOut.txt is empty. Skipping analysing it.\n",
      "TLOut.txt  is empty. Skipping analysing it.\n",
      "invTLOut.txt  is empty. Skipping analysing it.\n",
      "ctxOut.txt is empty. Skipping analysing it.\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "[M::mm_idx_gen::0.063*1.01] collected minimizers\n",
      "[M::mm_idx_gen::0.081*1.00] sorted minimizers\n",
      "[M::main::0.081*1.00] loaded/built the index for 1 target sequence(s)\n",
      "[M::mm_mapopt_update::0.081*1.00] mid_occ = 100\n",
      "[M::mm_idx_stat] kmer size: 19; skip: 10; is_hpc: 1; #seq: 1\n",
      "[M::mm_idx_stat::0.085*1.00] distinct minimizers: 525478 (99.34% are singletons); average occurrences: 1.015; average spacing: 7.461; total length: 3977333\n",
      "[M::worker_pipeline::0.623*0.99] mapped 1 sequences\n",
      "[M::main] Version: 2.19-r1057\n",
      "[M::main] CMD: minimap2 -ax asm20 --eqx -H -f 100 -r1k,10k --rmq=no -t 1 exp/multiple_runs11/tmp/L01-3/is_deleted/L01-3_genome.deleted_IS.0.fasta exp/multiple_runs11/tmp/L01-3/is_deleted/L01-3_genome.deleted_IS.2.fasta\n",
      "[M::main] Real time: 0.626 sec; CPU: 0.619 sec; Peak RSS: 0.088 GB\n",
      "local_variation - WARNING - Finished syri\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "TLOut.txt  is empty. Skipping analysing it.\n",
      "invTLOut.txt  is empty. Skipping analysing it.\n",
      "ctxOut.txt is empty. Skipping analysing it.\n",
      "TL Out.txt is empty. Skipping analysing it.\n",
      "invTL Out.txt is empty. Skipping analysing it.\n",
      "dup Out.txt is empty. Skipping analysing it.\n",
      "invDup Out.txt is empty. Skipping analysing it.\n",
      "ctxOut.txt is empty. Skipping analysing it.\n",
      "TLOut.txt  is empty. Skipping analysing it.\n",
      "invTLOut.txt  is empty. Skipping analysing it.\n",
      "ctxOut.txt is empty. Skipping analysing it.\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "[M::mm_idx_gen::0.058*1.01] collected minimizers\n",
      "[M::mm_idx_gen::0.076*1.00] sorted minimizers\n",
      "[M::main::0.076*1.00] loaded/built the index for 1 target sequence(s)\n",
      "[M::mm_mapopt_update::0.076*1.00] mid_occ = 100\n",
      "[M::mm_idx_stat] kmer size: 19; skip: 10; is_hpc: 1; #seq: 1\n",
      "[M::mm_idx_stat::0.080*1.00] distinct minimizers: 525478 (99.34% are singletons); average occurrences: 1.015; average spacing: 7.461; total length: 3977333\n",
      "[M::worker_pipeline::0.625*0.99] mapped 1 sequences\n",
      "[M::main] Version: 2.19-r1057\n",
      "[M::main] CMD: minimap2 -ax asm20 --eqx -H -f 100 -r1k,10k --rmq=no -t 1 exp/multiple_runs11/tmp/L01-3/is_deleted/L01-3_genome.deleted_IS.0.fasta exp/multiple_runs11/tmp/L01-3/is_deleted/L01-3_genome.deleted_IS.3.fasta\n",
      "[M::main] Real time: 0.628 sec; CPU: 0.622 sec; Peak RSS: 0.121 GB\n",
      "local_variation - WARNING - Finished syri\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "TLOut.txt  is empty. Skipping analysing it.\n",
      "invTLOut.txt  is empty. Skipping analysing it.\n",
      "ctxOut.txt is empty. Skipping analysing it.\n",
      "TL Out.txt is empty. Skipping analysing it.\n",
      "invTL Out.txt is empty. Skipping analysing it.\n",
      "dup Out.txt is empty. Skipping analysing it.\n",
      "invDup Out.txt is empty. Skipping analysing it.\n",
      "ctxOut.txt is empty. Skipping analysing it.\n",
      "TLOut.txt  is empty. Skipping analysing it.\n",
      "invTLOut.txt  is empty. Skipping analysing it.\n",
      "ctxOut.txt is empty. Skipping analysing it.\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "[M::mm_idx_gen::0.060*1.01] collected minimizers\n",
      "[M::mm_idx_gen::0.078*1.00] sorted minimizers\n",
      "[M::main::0.078*1.00] loaded/built the index for 1 target sequence(s)\n",
      "[M::mm_mapopt_update::0.078*1.00] mid_occ = 100\n",
      "[M::mm_idx_stat] kmer size: 19; skip: 10; is_hpc: 1; #seq: 1\n",
      "[M::mm_idx_stat::0.082*1.00] distinct minimizers: 525478 (99.34% are singletons); average occurrences: 1.015; average spacing: 7.461; total length: 3977333\n",
      "[M::worker_pipeline::0.625*0.99] mapped 1 sequences\n",
      "[M::main] Version: 2.19-r1057\n",
      "[M::main] CMD: minimap2 -ax asm20 --eqx -H -f 100 -r1k,10k --rmq=no -t 1 exp/multiple_runs11/tmp/L01-4/is_deleted/L01-4_genome.deleted_IS.0.fasta exp/multiple_runs11/tmp/L01-4/is_deleted/L01-4_genome.deleted_IS.1.fasta\n",
      "[M::main] Real time: 0.628 sec; CPU: 0.622 sec; Peak RSS: 0.088 GB\n",
      "local_variation - WARNING - Finished syri\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "TLOut.txt  is empty. Skipping analysing it.\n",
      "invTLOut.txt  is empty. Skipping analysing it.\n",
      "ctxOut.txt is empty. Skipping analysing it.\n",
      "TL Out.txt is empty. Skipping analysing it.\n",
      "invTL Out.txt is empty. Skipping analysing it.\n",
      "dup Out.txt is empty. Skipping analysing it.\n",
      "invDup Out.txt is empty. Skipping analysing it.\n",
      "ctxOut.txt is empty. Skipping analysing it.\n",
      "TLOut.txt  is empty. Skipping analysing it.\n",
      "invTLOut.txt  is empty. Skipping analysing it.\n",
      "ctxOut.txt is empty. Skipping analysing it.\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "[M::mm_idx_gen::0.058*1.01] collected minimizers\n",
      "[M::mm_idx_gen::0.077*1.00] sorted minimizers\n",
      "[M::main::0.077*1.00] loaded/built the index for 1 target sequence(s)\n",
      "[M::mm_mapopt_update::0.077*1.00] mid_occ = 100\n",
      "[M::mm_idx_stat] kmer size: 19; skip: 10; is_hpc: 1; #seq: 1\n",
      "[M::mm_idx_stat::0.081*1.00] distinct minimizers: 525478 (99.34% are singletons); average occurrences: 1.015; average spacing: 7.461; total length: 3977333\n",
      "[M::worker_pipeline::0.621*0.99] mapped 1 sequences\n",
      "[M::main] Version: 2.19-r1057\n",
      "[M::main] CMD: minimap2 -ax asm20 --eqx -H -f 100 -r1k,10k --rmq=no -t 1 exp/multiple_runs11/tmp/L01-4/is_deleted/L01-4_genome.deleted_IS.0.fasta exp/multiple_runs11/tmp/L01-4/is_deleted/L01-4_genome.deleted_IS.2.fasta\n",
      "[M::main] Real time: 0.623 sec; CPU: 0.617 sec; Peak RSS: 0.087 GB\n",
      "local_variation - WARNING - Finished syri\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "invTLOut.txt  is empty. Skipping analysing it.\n",
      "ctxOut.txt is empty. Skipping analysing it.\n",
      "invTL Out.txt is empty. Skipping analysing it.\n",
      "dup Out.txt is empty. Skipping analysing it.\n",
      "invDup Out.txt is empty. Skipping analysing it.\n",
      "ctxOut.txt is empty. Skipping analysing it.\n",
      "invTLOut.txt  is empty. Skipping analysing it.\n",
      "ctxOut.txt is empty. Skipping analysing it.\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "[M::mm_idx_gen::0.058*1.01] collected minimizers\n",
      "[M::mm_idx_gen::0.077*1.00] sorted minimizers\n",
      "[M::main::0.077*1.00] loaded/built the index for 1 target sequence(s)\n",
      "[M::mm_mapopt_update::0.077*1.00] mid_occ = 100\n",
      "[M::mm_idx_stat] kmer size: 19; skip: 10; is_hpc: 1; #seq: 1\n",
      "[M::mm_idx_stat::0.081*1.00] distinct minimizers: 525478 (99.34% are singletons); average occurrences: 1.015; average spacing: 7.461; total length: 3977333\n",
      "[M::worker_pipeline::0.619*0.99] mapped 1 sequences\n",
      "[M::main] Version: 2.19-r1057\n",
      "[M::main] CMD: minimap2 -ax asm20 --eqx -H -f 100 -r1k,10k --rmq=no -t 1 exp/multiple_runs11/tmp/L01-4/is_deleted/L01-4_genome.deleted_IS.0.fasta exp/multiple_runs11/tmp/L01-4/is_deleted/L01-4_genome.deleted_IS.3.fasta\n",
      "[M::main] Real time: 0.622 sec; CPU: 0.615 sec; Peak RSS: 0.090 GB\n",
      "local_variation - WARNING - Finished syri\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "invTLOut.txt  is empty. Skipping analysing it.\n",
      "ctxOut.txt is empty. Skipping analysing it.\n",
      "invTL Out.txt is empty. Skipping analysing it.\n",
      "dup Out.txt is empty. Skipping analysing it.\n",
      "invDup Out.txt is empty. Skipping analysing it.\n",
      "ctxOut.txt is empty. Skipping analysing it.\n",
      "invTLOut.txt  is empty. Skipping analysing it.\n",
      "ctxOut.txt is empty. Skipping analysing it.\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "[M::mm_idx_gen::0.066*1.01] collected minimizers\n",
      "[M::mm_idx_gen::0.084*1.01] sorted minimizers\n",
      "[M::main::0.084*1.01] loaded/built the index for 1 target sequence(s)\n",
      "[M::mm_mapopt_update::0.084*1.01] mid_occ = 100\n",
      "[M::mm_idx_stat] kmer size: 19; skip: 10; is_hpc: 1; #seq: 1\n",
      "[M::mm_idx_stat::0.088*1.01] distinct minimizers: 525478 (99.34% are singletons); average occurrences: 1.015; average spacing: 7.461; total length: 3977333\n",
      "[M::worker_pipeline::0.624*1.00] mapped 1 sequences\n",
      "[M::main] Version: 2.19-r1057\n",
      "[M::main] CMD: minimap2 -ax asm20 --eqx -H -f 100 -r1k,10k --rmq=no -t 1 exp/multiple_runs11/tmp/L02-1/is_deleted/L02-1_genome.deleted_IS.0.fasta exp/multiple_runs11/tmp/L02-1/is_deleted/L02-1_genome.deleted_IS.1.fasta\n",
      "[M::main] Real time: 0.627 sec; CPU: 0.626 sec; Peak RSS: 0.085 GB\n",
      "local_variation - WARNING - Finished syri\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "invOut.txt  is empty. Skipping analysing it.\n",
      "TLOut.txt  is empty. Skipping analysing it.\n",
      "invTLOut.txt  is empty. Skipping analysing it.\n",
      "ctxOut.txt is empty. Skipping analysing it.\n",
      "inv Out.txt is empty. Skipping analysing it.\n",
      "TL Out.txt is empty. Skipping analysing it.\n",
      "invTL Out.txt is empty. Skipping analysing it.\n",
      "dup Out.txt is empty. Skipping analysing it.\n",
      "invDup Out.txt is empty. Skipping analysing it.\n",
      "ctxOut.txt is empty. Skipping analysing it.\n",
      "invOut.txt  is empty. Skipping analysing it.\n",
      "TLOut.txt  is empty. Skipping analysing it.\n",
      "invTLOut.txt  is empty. Skipping analysing it.\n",
      "ctxOut.txt is empty. Skipping analysing it.\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "[M::mm_idx_gen::0.060*1.01] collected minimizers\n",
      "[M::mm_idx_gen::0.079*1.01] sorted minimizers\n",
      "[M::main::0.079*1.01] loaded/built the index for 1 target sequence(s)\n",
      "[M::mm_mapopt_update::0.079*1.01] mid_occ = 100\n",
      "[M::mm_idx_stat] kmer size: 19; skip: 10; is_hpc: 1; #seq: 1\n",
      "[M::mm_idx_stat::0.083*1.01] distinct minimizers: 525478 (99.34% are singletons); average occurrences: 1.015; average spacing: 7.461; total length: 3977333\n",
      "[M::worker_pipeline::0.625*0.99] mapped 1 sequences\n",
      "[M::main] Version: 2.19-r1057\n",
      "[M::main] CMD: minimap2 -ax asm20 --eqx -H -f 100 -r1k,10k --rmq=no -t 1 exp/multiple_runs11/tmp/L02-1/is_deleted/L02-1_genome.deleted_IS.0.fasta exp/multiple_runs11/tmp/L02-1/is_deleted/L02-1_genome.deleted_IS.2.fasta\n",
      "[M::main] Real time: 0.628 sec; CPU: 0.621 sec; Peak RSS: 0.087 GB\n",
      "local_variation - WARNING - Finished syri\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "invOut.txt  is empty. Skipping analysing it.\n",
      "TLOut.txt  is empty. Skipping analysing it.\n",
      "invTLOut.txt  is empty. Skipping analysing it.\n",
      "ctxOut.txt is empty. Skipping analysing it.\n",
      "inv Out.txt is empty. Skipping analysing it.\n",
      "TL Out.txt is empty. Skipping analysing it.\n",
      "invTL Out.txt is empty. Skipping analysing it.\n",
      "dup Out.txt is empty. Skipping analysing it.\n",
      "invDup Out.txt is empty. Skipping analysing it.\n",
      "ctxOut.txt is empty. Skipping analysing it.\n",
      "invOut.txt  is empty. Skipping analysing it.\n",
      "TLOut.txt  is empty. Skipping analysing it.\n",
      "invTLOut.txt  is empty. Skipping analysing it.\n",
      "ctxOut.txt is empty. Skipping analysing it.\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "[M::mm_idx_gen::0.059*1.01] collected minimizers\n",
      "[M::mm_idx_gen::0.077*1.01] sorted minimizers\n",
      "[M::main::0.077*1.01] loaded/built the index for 1 target sequence(s)\n",
      "[M::mm_mapopt_update::0.077*1.01] mid_occ = 100\n",
      "[M::mm_idx_stat] kmer size: 19; skip: 10; is_hpc: 1; #seq: 1\n",
      "[M::mm_idx_stat::0.081*1.01] distinct minimizers: 525478 (99.34% are singletons); average occurrences: 1.015; average spacing: 7.461; total length: 3977333\n",
      "[M::worker_pipeline::0.637*0.99] mapped 1 sequences\n",
      "[M::main] Version: 2.19-r1057\n",
      "[M::main] CMD: minimap2 -ax asm20 --eqx -H -f 100 -r1k,10k --rmq=no -t 1 exp/multiple_runs11/tmp/L02-1/is_deleted/L02-1_genome.deleted_IS.0.fasta exp/multiple_runs11/tmp/L02-1/is_deleted/L02-1_genome.deleted_IS.3.fasta\n",
      "[M::main] Real time: 0.640 sec; CPU: 0.633 sec; Peak RSS: 0.090 GB\n",
      "local_variation - WARNING - Finished syri\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "invOut.txt  is empty. Skipping analysing it.\n",
      "TLOut.txt  is empty. Skipping analysing it.\n",
      "ctxOut.txt is empty. Skipping analysing it.\n",
      "inv Out.txt is empty. Skipping analysing it.\n",
      "TL Out.txt is empty. Skipping analysing it.\n",
      "dup Out.txt is empty. Skipping analysing it.\n",
      "invDup Out.txt is empty. Skipping analysing it.\n",
      "ctxOut.txt is empty. Skipping analysing it.\n",
      "invOut.txt  is empty. Skipping analysing it.\n",
      "TLOut.txt  is empty. Skipping analysing it.\n",
      "ctxOut.txt is empty. Skipping analysing it.\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "[M::mm_idx_gen::0.063*1.01] collected minimizers\n",
      "[M::mm_idx_gen::0.081*1.00] sorted minimizers\n",
      "[M::main::0.081*1.00] loaded/built the index for 1 target sequence(s)\n",
      "[M::mm_mapopt_update::0.081*1.00] mid_occ = 100\n",
      "[M::mm_idx_stat] kmer size: 19; skip: 10; is_hpc: 1; #seq: 1\n",
      "[M::mm_idx_stat::0.085*1.00] distinct minimizers: 525478 (99.34% are singletons); average occurrences: 1.015; average spacing: 7.461; total length: 3977333\n",
      "[M::worker_pipeline::0.621*1.00] mapped 1 sequences\n",
      "[M::main] Version: 2.19-r1057\n",
      "[M::main] CMD: minimap2 -ax asm20 --eqx -H -f 100 -r1k,10k --rmq=no -t 1 exp/multiple_runs11/tmp/L02-2/is_deleted/L02-2_genome.deleted_IS.0.fasta exp/multiple_runs11/tmp/L02-2/is_deleted/L02-2_genome.deleted_IS.1.fasta\n",
      "[M::main] Real time: 0.624 sec; CPU: 0.623 sec; Peak RSS: 0.085 GB\n",
      "local_variation - WARNING - Finished syri\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "invOut.txt  is empty. Skipping analysing it.\n",
      "TLOut.txt  is empty. Skipping analysing it.\n",
      "invTLOut.txt  is empty. Skipping analysing it.\n",
      "ctxOut.txt is empty. Skipping analysing it.\n",
      "inv Out.txt is empty. Skipping analysing it.\n",
      "TL Out.txt is empty. Skipping analysing it.\n",
      "invTL Out.txt is empty. Skipping analysing it.\n",
      "dup Out.txt is empty. Skipping analysing it.\n",
      "invDup Out.txt is empty. Skipping analysing it.\n",
      "ctxOut.txt is empty. Skipping analysing it.\n",
      "invOut.txt  is empty. Skipping analysing it.\n",
      "TLOut.txt  is empty. Skipping analysing it.\n",
      "invTLOut.txt  is empty. Skipping analysing it.\n",
      "ctxOut.txt is empty. Skipping analysing it.\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "[M::mm_idx_gen::0.058*1.01] collected minimizers\n",
      "[M::mm_idx_gen::0.077*1.00] sorted minimizers\n",
      "[M::main::0.077*1.00] loaded/built the index for 1 target sequence(s)\n",
      "[M::mm_mapopt_update::0.077*1.00] mid_occ = 100\n",
      "[M::mm_idx_stat] kmer size: 19; skip: 10; is_hpc: 1; #seq: 1\n",
      "[M::mm_idx_stat::0.081*1.00] distinct minimizers: 525478 (99.34% are singletons); average occurrences: 1.015; average spacing: 7.461; total length: 3977333\n",
      "[M::worker_pipeline::0.619*0.99] mapped 1 sequences\n",
      "[M::main] Version: 2.19-r1057\n",
      "[M::main] CMD: minimap2 -ax asm20 --eqx -H -f 100 -r1k,10k --rmq=no -t 1 exp/multiple_runs11/tmp/L02-2/is_deleted/L02-2_genome.deleted_IS.0.fasta exp/multiple_runs11/tmp/L02-2/is_deleted/L02-2_genome.deleted_IS.2.fasta\n",
      "[M::main] Real time: 0.622 sec; CPU: 0.615 sec; Peak RSS: 0.089 GB\n",
      "local_variation - WARNING - Finished syri\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "TLOut.txt  is empty. Skipping analysing it.\n",
      "invTLOut.txt  is empty. Skipping analysing it.\n",
      "ctxOut.txt is empty. Skipping analysing it.\n",
      "TL Out.txt is empty. Skipping analysing it.\n",
      "invTL Out.txt is empty. Skipping analysing it.\n",
      "invDup Out.txt is empty. Skipping analysing it.\n",
      "ctxOut.txt is empty. Skipping analysing it.\n",
      "TLOut.txt  is empty. Skipping analysing it.\n",
      "invTLOut.txt  is empty. Skipping analysing it.\n",
      "ctxOut.txt is empty. Skipping analysing it.\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "[M::mm_idx_gen::0.058*1.01] collected minimizers\n",
      "[M::mm_idx_gen::0.077*1.00] sorted minimizers\n",
      "[M::main::0.077*1.00] loaded/built the index for 1 target sequence(s)\n",
      "[M::mm_mapopt_update::0.077*1.00] mid_occ = 100\n",
      "[M::mm_idx_stat] kmer size: 19; skip: 10; is_hpc: 1; #seq: 1\n",
      "[M::mm_idx_stat::0.080*1.00] distinct minimizers: 525478 (99.34% are singletons); average occurrences: 1.015; average spacing: 7.461; total length: 3977333\n",
      "[M::worker_pipeline::0.679*0.99] mapped 1 sequences\n",
      "[M::main] Version: 2.19-r1057\n",
      "[M::main] CMD: minimap2 -ax asm20 --eqx -H -f 100 -r1k,10k --rmq=no -t 1 exp/multiple_runs11/tmp/L02-2/is_deleted/L02-2_genome.deleted_IS.0.fasta exp/multiple_runs11/tmp/L02-2/is_deleted/L02-2_genome.deleted_IS.3.fasta\n",
      "[M::main] Real time: 0.682 sec; CPU: 0.675 sec; Peak RSS: 0.091 GB\n",
      "local_variation - WARNING - Finished syri\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "TLOut.txt  is empty. Skipping analysing it.\n",
      "invTLOut.txt  is empty. Skipping analysing it.\n",
      "ctxOut.txt is empty. Skipping analysing it.\n",
      "TL Out.txt is empty. Skipping analysing it.\n",
      "invTL Out.txt is empty. Skipping analysing it.\n",
      "invDup Out.txt is empty. Skipping analysing it.\n",
      "ctxOut.txt is empty. Skipping analysing it.\n",
      "TLOut.txt  is empty. Skipping analysing it.\n",
      "invTLOut.txt  is empty. Skipping analysing it.\n",
      "ctxOut.txt is empty. Skipping analysing it.\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "[M::mm_idx_gen::0.059*1.01] collected minimizers\n",
      "[M::mm_idx_gen::0.077*1.00] sorted minimizers\n",
      "[M::main::0.077*1.00] loaded/built the index for 1 target sequence(s)\n",
      "[M::mm_mapopt_update::0.077*1.00] mid_occ = 100\n",
      "[M::mm_idx_stat] kmer size: 19; skip: 10; is_hpc: 1; #seq: 1\n",
      "[M::mm_idx_stat::0.081*1.00] distinct minimizers: 525478 (99.34% are singletons); average occurrences: 1.015; average spacing: 7.461; total length: 3977333\n",
      "[M::worker_pipeline::0.617*1.00] mapped 1 sequences\n",
      "[M::main] Version: 2.19-r1057\n",
      "[M::main] CMD: minimap2 -ax asm20 --eqx -H -f 100 -r1k,10k --rmq=no -t 1 exp/multiple_runs11/tmp/L02-3/is_deleted/L02-3_genome.deleted_IS.0.fasta exp/multiple_runs11/tmp/L02-3/is_deleted/L02-3_genome.deleted_IS.1.fasta\n",
      "[M::main] Real time: 0.619 sec; CPU: 0.618 sec; Peak RSS: 0.085 GB\n",
      "local_variation - WARNING - Finished syri\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "invOut.txt  is empty. Skipping analysing it.\n",
      "TLOut.txt  is empty. Skipping analysing it.\n",
      "invTLOut.txt  is empty. Skipping analysing it.\n",
      "ctxOut.txt is empty. Skipping analysing it.\n",
      "inv Out.txt is empty. Skipping analysing it.\n",
      "TL Out.txt is empty. Skipping analysing it.\n",
      "invTL Out.txt is empty. Skipping analysing it.\n",
      "dup Out.txt is empty. Skipping analysing it.\n",
      "invDup Out.txt is empty. Skipping analysing it.\n",
      "ctxOut.txt is empty. Skipping analysing it.\n",
      "invOut.txt  is empty. Skipping analysing it.\n",
      "TLOut.txt  is empty. Skipping analysing it.\n",
      "invTLOut.txt  is empty. Skipping analysing it.\n",
      "ctxOut.txt is empty. Skipping analysing it.\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "[M::mm_idx_gen::0.058*1.01] collected minimizers\n",
      "[M::mm_idx_gen::0.076*1.00] sorted minimizers\n",
      "[M::main::0.076*1.00] loaded/built the index for 1 target sequence(s)\n",
      "[M::mm_mapopt_update::0.076*1.00] mid_occ = 100\n",
      "[M::mm_idx_stat] kmer size: 19; skip: 10; is_hpc: 1; #seq: 1\n",
      "[M::mm_idx_stat::0.080*1.00] distinct minimizers: 525478 (99.34% are singletons); average occurrences: 1.015; average spacing: 7.461; total length: 3977333\n",
      "[M::worker_pipeline::0.616*0.99] mapped 1 sequences\n",
      "[M::main] Version: 2.19-r1057\n",
      "[M::main] CMD: minimap2 -ax asm20 --eqx -H -f 100 -r1k,10k --rmq=no -t 1 exp/multiple_runs11/tmp/L02-3/is_deleted/L02-3_genome.deleted_IS.0.fasta exp/multiple_runs11/tmp/L02-3/is_deleted/L02-3_genome.deleted_IS.2.fasta\n",
      "[M::main] Real time: 0.619 sec; CPU: 0.612 sec; Peak RSS: 0.086 GB\n",
      "local_variation - WARNING - Finished syri\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "invOut.txt  is empty. Skipping analysing it.\n",
      "TLOut.txt  is empty. Skipping analysing it.\n",
      "invTLOut.txt  is empty. Skipping analysing it.\n",
      "ctxOut.txt is empty. Skipping analysing it.\n",
      "inv Out.txt is empty. Skipping analysing it.\n",
      "TL Out.txt is empty. Skipping analysing it.\n",
      "invTL Out.txt is empty. Skipping analysing it.\n",
      "dup Out.txt is empty. Skipping analysing it.\n",
      "invDup Out.txt is empty. Skipping analysing it.\n",
      "ctxOut.txt is empty. Skipping analysing it.\n",
      "invOut.txt  is empty. Skipping analysing it.\n",
      "TLOut.txt  is empty. Skipping analysing it.\n",
      "invTLOut.txt  is empty. Skipping analysing it.\n",
      "ctxOut.txt is empty. Skipping analysing it.\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "[M::mm_idx_gen::0.064*1.01] collected minimizers\n",
      "[M::mm_idx_gen::0.082*1.00] sorted minimizers\n",
      "[M::main::0.082*1.00] loaded/built the index for 1 target sequence(s)\n",
      "[M::mm_mapopt_update::0.082*1.00] mid_occ = 100\n",
      "[M::mm_idx_stat] kmer size: 19; skip: 10; is_hpc: 1; #seq: 1\n",
      "[M::mm_idx_stat::0.086*1.00] distinct minimizers: 525478 (99.34% are singletons); average occurrences: 1.015; average spacing: 7.461; total length: 3977333\n",
      "[M::worker_pipeline::0.653*0.99] mapped 1 sequences\n",
      "[M::main] Version: 2.19-r1057\n",
      "[M::main] CMD: minimap2 -ax asm20 --eqx -H -f 100 -r1k,10k --rmq=no -t 1 exp/multiple_runs11/tmp/L02-3/is_deleted/L02-3_genome.deleted_IS.0.fasta exp/multiple_runs11/tmp/L02-3/is_deleted/L02-3_genome.deleted_IS.3.fasta\n",
      "[M::main] Real time: 0.655 sec; CPU: 0.648 sec; Peak RSS: 0.089 GB\n",
      "local_variation - WARNING - Finished syri\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "TLOut.txt  is empty. Skipping analysing it.\n",
      "ctxOut.txt is empty. Skipping analysing it.\n",
      "TL Out.txt is empty. Skipping analysing it.\n",
      "dup Out.txt is empty. Skipping analysing it.\n",
      "invDup Out.txt is empty. Skipping analysing it.\n",
      "ctxOut.txt is empty. Skipping analysing it.\n",
      "TLOut.txt  is empty. Skipping analysing it.\n",
      "ctxOut.txt is empty. Skipping analysing it.\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "[M::mm_idx_gen::0.059*1.01] collected minimizers\n",
      "[M::mm_idx_gen::0.077*1.00] sorted minimizers\n",
      "[M::main::0.077*1.00] loaded/built the index for 1 target sequence(s)\n",
      "[M::mm_mapopt_update::0.077*1.00] mid_occ = 100\n",
      "[M::mm_idx_stat] kmer size: 19; skip: 10; is_hpc: 1; #seq: 1\n",
      "[M::mm_idx_stat::0.081*1.00] distinct minimizers: 525478 (99.34% are singletons); average occurrences: 1.015; average spacing: 7.461; total length: 3977333\n",
      "[M::worker_pipeline::0.618*1.00] mapped 1 sequences\n",
      "[M::main] Version: 2.19-r1057\n",
      "[M::main] CMD: minimap2 -ax asm20 --eqx -H -f 100 -r1k,10k --rmq=no -t 1 exp/multiple_runs11/tmp/L02-4/is_deleted/L02-4_genome.deleted_IS.0.fasta exp/multiple_runs11/tmp/L02-4/is_deleted/L02-4_genome.deleted_IS.1.fasta\n",
      "[M::main] Real time: 0.620 sec; CPU: 0.619 sec; Peak RSS: 0.085 GB\n",
      "local_variation - WARNING - Finished syri\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "invOut.txt  is empty. Skipping analysing it.\n",
      "TLOut.txt  is empty. Skipping analysing it.\n",
      "invTLOut.txt  is empty. Skipping analysing it.\n",
      "ctxOut.txt is empty. Skipping analysing it.\n",
      "inv Out.txt is empty. Skipping analysing it.\n",
      "TL Out.txt is empty. Skipping analysing it.\n",
      "invTL Out.txt is empty. Skipping analysing it.\n",
      "dup Out.txt is empty. Skipping analysing it.\n",
      "invDup Out.txt is empty. Skipping analysing it.\n",
      "ctxOut.txt is empty. Skipping analysing it.\n",
      "invOut.txt  is empty. Skipping analysing it.\n",
      "TLOut.txt  is empty. Skipping analysing it.\n",
      "invTLOut.txt  is empty. Skipping analysing it.\n",
      "ctxOut.txt is empty. Skipping analysing it.\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "[M::mm_idx_gen::0.059*1.01] collected minimizers\n",
      "[M::mm_idx_gen::0.077*1.00] sorted minimizers\n",
      "[M::main::0.077*1.00] loaded/built the index for 1 target sequence(s)\n",
      "[M::mm_mapopt_update::0.077*1.00] mid_occ = 100\n",
      "[M::mm_idx_stat] kmer size: 19; skip: 10; is_hpc: 1; #seq: 1\n",
      "[M::mm_idx_stat::0.081*1.00] distinct minimizers: 525478 (99.34% are singletons); average occurrences: 1.015; average spacing: 7.461; total length: 3977333\n",
      "[M::worker_pipeline::0.612*0.99] mapped 1 sequences\n",
      "[M::main] Version: 2.19-r1057\n",
      "[M::main] CMD: minimap2 -ax asm20 --eqx -H -f 100 -r1k,10k --rmq=no -t 1 exp/multiple_runs11/tmp/L02-4/is_deleted/L02-4_genome.deleted_IS.0.fasta exp/multiple_runs11/tmp/L02-4/is_deleted/L02-4_genome.deleted_IS.2.fasta\n",
      "[M::main] Real time: 0.616 sec; CPU: 0.610 sec; Peak RSS: 0.087 GB\n",
      "local_variation - WARNING - Finished syri\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "TLOut.txt  is empty. Skipping analysing it.\n",
      "invTLOut.txt  is empty. Skipping analysing it.\n",
      "ctxOut.txt is empty. Skipping analysing it.\n",
      "TL Out.txt is empty. Skipping analysing it.\n",
      "invTL Out.txt is empty. Skipping analysing it.\n",
      "dup Out.txt is empty. Skipping analysing it.\n",
      "invDup Out.txt is empty. Skipping analysing it.\n",
      "ctxOut.txt is empty. Skipping analysing it.\n",
      "TLOut.txt  is empty. Skipping analysing it.\n",
      "invTLOut.txt  is empty. Skipping analysing it.\n",
      "ctxOut.txt is empty. Skipping analysing it.\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "[M::mm_idx_gen::0.058*1.01] collected minimizers\n",
      "[M::mm_idx_gen::0.076*1.00] sorted minimizers\n",
      "[M::main::0.076*1.00] loaded/built the index for 1 target sequence(s)\n",
      "[M::mm_mapopt_update::0.076*1.00] mid_occ = 100\n",
      "[M::mm_idx_stat] kmer size: 19; skip: 10; is_hpc: 1; #seq: 1\n",
      "[M::mm_idx_stat::0.080*1.00] distinct minimizers: 525478 (99.34% are singletons); average occurrences: 1.015; average spacing: 7.461; total length: 3977333\n",
      "[M::worker_pipeline::0.617*0.99] mapped 1 sequences\n",
      "[M::main] Version: 2.19-r1057\n",
      "[M::main] CMD: minimap2 -ax asm20 --eqx -H -f 100 -r1k,10k --rmq=no -t 1 exp/multiple_runs11/tmp/L02-4/is_deleted/L02-4_genome.deleted_IS.0.fasta exp/multiple_runs11/tmp/L02-4/is_deleted/L02-4_genome.deleted_IS.3.fasta\n",
      "[M::main] Real time: 0.619 sec; CPU: 0.613 sec; Peak RSS: 0.090 GB\n",
      "local_variation - WARNING - Finished syri\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "TLOut.txt  is empty. Skipping analysing it.\n",
      "invTLOut.txt  is empty. Skipping analysing it.\n",
      "ctxOut.txt is empty. Skipping analysing it.\n",
      "TL Out.txt is empty. Skipping analysing it.\n",
      "invTL Out.txt is empty. Skipping analysing it.\n",
      "dup Out.txt is empty. Skipping analysing it.\n",
      "ctxOut.txt is empty. Skipping analysing it.\n",
      "TLOut.txt  is empty. Skipping analysing it.\n",
      "invTLOut.txt  is empty. Skipping analysing it.\n",
      "ctxOut.txt is empty. Skipping analysing it.\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "[M::mm_idx_gen::0.058*1.01] collected minimizers\n",
      "[M::mm_idx_gen::0.077*1.00] sorted minimizers\n",
      "[M::main::0.077*1.00] loaded/built the index for 1 target sequence(s)\n",
      "[M::mm_mapopt_update::0.077*1.00] mid_occ = 100\n",
      "[M::mm_idx_stat] kmer size: 19; skip: 10; is_hpc: 1; #seq: 1\n",
      "[M::mm_idx_stat::0.081*1.00] distinct minimizers: 525478 (99.34% are singletons); average occurrences: 1.015; average spacing: 7.461; total length: 3977333\n",
      "[M::worker_pipeline::0.620*0.99] mapped 1 sequences\n",
      "[M::main] Version: 2.19-r1057\n",
      "[M::main] CMD: minimap2 -ax asm20 --eqx -H -f 100 -r1k,10k --rmq=no -t 1 exp/multiple_runs11/tmp/L03-1/is_deleted/L03-1_genome.deleted_IS.0.fasta exp/multiple_runs11/tmp/L03-1/is_deleted/L03-1_genome.deleted_IS.1.fasta\n",
      "[M::main] Real time: 0.622 sec; CPU: 0.616 sec; Peak RSS: 0.086 GB\n",
      "local_variation - WARNING - Finished syri\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "TLOut.txt  is empty. Skipping analysing it.\n",
      "invTLOut.txt  is empty. Skipping analysing it.\n",
      "ctxOut.txt is empty. Skipping analysing it.\n",
      "TL Out.txt is empty. Skipping analysing it.\n",
      "invTL Out.txt is empty. Skipping analysing it.\n",
      "dup Out.txt is empty. Skipping analysing it.\n",
      "invDup Out.txt is empty. Skipping analysing it.\n",
      "ctxOut.txt is empty. Skipping analysing it.\n",
      "TLOut.txt  is empty. Skipping analysing it.\n",
      "invTLOut.txt  is empty. Skipping analysing it.\n",
      "ctxOut.txt is empty. Skipping analysing it.\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "[M::mm_idx_gen::0.065*1.01] collected minimizers\n",
      "[M::mm_idx_gen::0.083*1.01] sorted minimizers\n",
      "[M::main::0.083*1.01] loaded/built the index for 1 target sequence(s)\n",
      "[M::mm_mapopt_update::0.083*1.01] mid_occ = 100\n",
      "[M::mm_idx_stat] kmer size: 19; skip: 10; is_hpc: 1; #seq: 1\n",
      "[M::mm_idx_stat::0.087*1.01] distinct minimizers: 525478 (99.34% are singletons); average occurrences: 1.015; average spacing: 7.461; total length: 3977333\n",
      "[M::worker_pipeline::0.626*0.99] mapped 1 sequences\n",
      "[M::main] Version: 2.19-r1057\n",
      "[M::main] CMD: minimap2 -ax asm20 --eqx -H -f 100 -r1k,10k --rmq=no -t 1 exp/multiple_runs11/tmp/L03-1/is_deleted/L03-1_genome.deleted_IS.0.fasta exp/multiple_runs11/tmp/L03-1/is_deleted/L03-1_genome.deleted_IS.2.fasta\n",
      "[M::main] Real time: 0.628 sec; CPU: 0.622 sec; Peak RSS: 0.086 GB\n",
      "local_variation - WARNING - Finished syri\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "TLOut.txt  is empty. Skipping analysing it.\n",
      "invTLOut.txt  is empty. Skipping analysing it.\n",
      "ctxOut.txt is empty. Skipping analysing it.\n",
      "TL Out.txt is empty. Skipping analysing it.\n",
      "invTL Out.txt is empty. Skipping analysing it.\n",
      "dup Out.txt is empty. Skipping analysing it.\n",
      "invDup Out.txt is empty. Skipping analysing it.\n",
      "ctxOut.txt is empty. Skipping analysing it.\n",
      "TLOut.txt  is empty. Skipping analysing it.\n",
      "invTLOut.txt  is empty. Skipping analysing it.\n",
      "ctxOut.txt is empty. Skipping analysing it.\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "[M::mm_idx_gen::0.058*1.01] collected minimizers\n",
      "[M::mm_idx_gen::0.077*1.00] sorted minimizers\n",
      "[M::main::0.077*1.00] loaded/built the index for 1 target sequence(s)\n",
      "[M::mm_mapopt_update::0.077*1.00] mid_occ = 100\n",
      "[M::mm_idx_stat] kmer size: 19; skip: 10; is_hpc: 1; #seq: 1\n",
      "[M::mm_idx_stat::0.081*1.00] distinct minimizers: 525478 (99.34% are singletons); average occurrences: 1.015; average spacing: 7.461; total length: 3977333\n",
      "[M::worker_pipeline::0.612*0.99] mapped 1 sequences\n",
      "[M::main] Version: 2.19-r1057\n",
      "[M::main] CMD: minimap2 -ax asm20 --eqx -H -f 100 -r1k,10k --rmq=no -t 1 exp/multiple_runs11/tmp/L03-1/is_deleted/L03-1_genome.deleted_IS.0.fasta exp/multiple_runs11/tmp/L03-1/is_deleted/L03-1_genome.deleted_IS.3.fasta\n",
      "[M::main] Real time: 0.616 sec; CPU: 0.609 sec; Peak RSS: 0.088 GB\n",
      "local_variation - WARNING - Finished syri\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "TLOut.txt  is empty. Skipping analysing it.\n",
      "invTLOut.txt  is empty. Skipping analysing it.\n",
      "ctxOut.txt is empty. Skipping analysing it.\n",
      "TL Out.txt is empty. Skipping analysing it.\n",
      "invTL Out.txt is empty. Skipping analysing it.\n",
      "dup Out.txt is empty. Skipping analysing it.\n",
      "invDup Out.txt is empty. Skipping analysing it.\n",
      "ctxOut.txt is empty. Skipping analysing it.\n",
      "TLOut.txt  is empty. Skipping analysing it.\n",
      "invTLOut.txt  is empty. Skipping analysing it.\n",
      "ctxOut.txt is empty. Skipping analysing it.\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "[M::mm_idx_gen::0.063*1.01] collected minimizers\n",
      "[M::mm_idx_gen::0.082*1.00] sorted minimizers\n",
      "[M::main::0.082*1.00] loaded/built the index for 1 target sequence(s)\n",
      "[M::mm_mapopt_update::0.082*1.00] mid_occ = 100\n",
      "[M::mm_idx_stat] kmer size: 19; skip: 10; is_hpc: 1; #seq: 1\n",
      "[M::mm_idx_stat::0.086*1.00] distinct minimizers: 525478 (99.34% are singletons); average occurrences: 1.015; average spacing: 7.461; total length: 3977333\n",
      "[M::worker_pipeline::0.623*0.99] mapped 1 sequences\n",
      "[M::main] Version: 2.19-r1057\n",
      "[M::main] CMD: minimap2 -ax asm20 --eqx -H -f 100 -r1k,10k --rmq=no -t 1 exp/multiple_runs11/tmp/L03-2/is_deleted/L03-2_genome.deleted_IS.0.fasta exp/multiple_runs11/tmp/L03-2/is_deleted/L03-2_genome.deleted_IS.1.fasta\n",
      "[M::main] Real time: 0.626 sec; CPU: 0.619 sec; Peak RSS: 0.086 GB\n",
      "local_variation - WARNING - Finished syri\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "TLOut.txt  is empty. Skipping analysing it.\n",
      "invTLOut.txt  is empty. Skipping analysing it.\n",
      "ctxOut.txt is empty. Skipping analysing it.\n",
      "TL Out.txt is empty. Skipping analysing it.\n",
      "invTL Out.txt is empty. Skipping analysing it.\n",
      "dup Out.txt is empty. Skipping analysing it.\n",
      "invDup Out.txt is empty. Skipping analysing it.\n",
      "ctxOut.txt is empty. Skipping analysing it.\n",
      "TLOut.txt  is empty. Skipping analysing it.\n",
      "invTLOut.txt  is empty. Skipping analysing it.\n",
      "ctxOut.txt is empty. Skipping analysing it.\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "[M::mm_idx_gen::0.063*1.01] collected minimizers\n",
      "[M::mm_idx_gen::0.081*1.00] sorted minimizers\n",
      "[M::main::0.081*1.00] loaded/built the index for 1 target sequence(s)\n",
      "[M::mm_mapopt_update::0.081*1.00] mid_occ = 100\n",
      "[M::mm_idx_stat] kmer size: 19; skip: 10; is_hpc: 1; #seq: 1\n",
      "[M::mm_idx_stat::0.085*1.00] distinct minimizers: 525478 (99.34% are singletons); average occurrences: 1.015; average spacing: 7.461; total length: 3977333\n",
      "[M::worker_pipeline::0.624*0.99] mapped 1 sequences\n",
      "[M::main] Version: 2.19-r1057\n",
      "[M::main] CMD: minimap2 -ax asm20 --eqx -H -f 100 -r1k,10k --rmq=no -t 1 exp/multiple_runs11/tmp/L03-2/is_deleted/L03-2_genome.deleted_IS.0.fasta exp/multiple_runs11/tmp/L03-2/is_deleted/L03-2_genome.deleted_IS.2.fasta\n",
      "[M::main] Real time: 0.627 sec; CPU: 0.620 sec; Peak RSS: 0.087 GB\n",
      "local_variation - WARNING - Finished syri\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "invTLOut.txt  is empty. Skipping analysing it.\n",
      "ctxOut.txt is empty. Skipping analysing it.\n",
      "invTL Out.txt is empty. Skipping analysing it.\n",
      "dup Out.txt is empty. Skipping analysing it.\n",
      "invDup Out.txt is empty. Skipping analysing it.\n",
      "ctxOut.txt is empty. Skipping analysing it.\n",
      "invTLOut.txt  is empty. Skipping analysing it.\n",
      "ctxOut.txt is empty. Skipping analysing it.\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "[M::mm_idx_gen::0.063*1.01] collected minimizers\n",
      "[M::mm_idx_gen::0.082*1.00] sorted minimizers\n",
      "[M::main::0.082*1.00] loaded/built the index for 1 target sequence(s)\n",
      "[M::mm_mapopt_update::0.082*1.00] mid_occ = 100\n",
      "[M::mm_idx_stat] kmer size: 19; skip: 10; is_hpc: 1; #seq: 1\n",
      "[M::mm_idx_stat::0.085*1.00] distinct minimizers: 525478 (99.34% are singletons); average occurrences: 1.015; average spacing: 7.461; total length: 3977333\n",
      "[M::worker_pipeline::0.626*0.99] mapped 1 sequences\n",
      "[M::main] Version: 2.19-r1057\n",
      "[M::main] CMD: minimap2 -ax asm20 --eqx -H -f 100 -r1k,10k --rmq=no -t 1 exp/multiple_runs11/tmp/L03-2/is_deleted/L03-2_genome.deleted_IS.0.fasta exp/multiple_runs11/tmp/L03-2/is_deleted/L03-2_genome.deleted_IS.3.fasta\n",
      "[M::main] Real time: 0.629 sec; CPU: 0.622 sec; Peak RSS: 0.087 GB\n",
      "local_variation - WARNING - Finished syri\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "ctxOut.txt is empty. Skipping analysing it.\n",
      "dup Out.txt is empty. Skipping analysing it.\n",
      "invDup Out.txt is empty. Skipping analysing it.\n",
      "ctxOut.txt is empty. Skipping analysing it.\n",
      "ctxOut.txt is empty. Skipping analysing it.\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "[M::mm_idx_gen::0.058*1.01] collected minimizers\n",
      "[M::mm_idx_gen::0.077*1.01] sorted minimizers\n",
      "[M::main::0.077*1.01] loaded/built the index for 1 target sequence(s)\n",
      "[M::mm_mapopt_update::0.077*1.01] mid_occ = 100\n",
      "[M::mm_idx_stat] kmer size: 19; skip: 10; is_hpc: 1; #seq: 1\n",
      "[M::mm_idx_stat::0.080*1.00] distinct minimizers: 525478 (99.34% are singletons); average occurrences: 1.015; average spacing: 7.461; total length: 3977333\n",
      "[M::worker_pipeline::0.617*0.99] mapped 1 sequences\n",
      "[M::main] Version: 2.19-r1057\n",
      "[M::main] CMD: minimap2 -ax asm20 --eqx -H -f 100 -r1k,10k --rmq=no -t 1 exp/multiple_runs11/tmp/L03-3/is_deleted/L03-3_genome.deleted_IS.0.fasta exp/multiple_runs11/tmp/L03-3/is_deleted/L03-3_genome.deleted_IS.1.fasta\n",
      "[M::main] Real time: 0.619 sec; CPU: 0.613 sec; Peak RSS: 0.086 GB\n",
      "local_variation - WARNING - Finished syri\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "TLOut.txt  is empty. Skipping analysing it.\n",
      "invTLOut.txt  is empty. Skipping analysing it.\n",
      "ctxOut.txt is empty. Skipping analysing it.\n",
      "TL Out.txt is empty. Skipping analysing it.\n",
      "invTL Out.txt is empty. Skipping analysing it.\n",
      "dup Out.txt is empty. Skipping analysing it.\n",
      "invDup Out.txt is empty. Skipping analysing it.\n",
      "ctxOut.txt is empty. Skipping analysing it.\n",
      "TLOut.txt  is empty. Skipping analysing it.\n",
      "invTLOut.txt  is empty. Skipping analysing it.\n",
      "ctxOut.txt is empty. Skipping analysing it.\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "[M::mm_idx_gen::0.056*1.01] collected minimizers\n",
      "[M::mm_idx_gen::0.075*1.00] sorted minimizers\n",
      "[M::main::0.075*1.00] loaded/built the index for 1 target sequence(s)\n",
      "[M::mm_mapopt_update::0.075*1.00] mid_occ = 100\n",
      "[M::mm_idx_stat] kmer size: 19; skip: 10; is_hpc: 1; #seq: 1\n",
      "[M::mm_idx_stat::0.079*1.00] distinct minimizers: 525478 (99.34% are singletons); average occurrences: 1.015; average spacing: 7.461; total length: 3977333\n",
      "[M::worker_pipeline::0.607*0.99] mapped 1 sequences\n",
      "[M::main] Version: 2.19-r1057\n",
      "[M::main] CMD: minimap2 -ax asm20 --eqx -H -f 100 -r1k,10k --rmq=no -t 1 exp/multiple_runs11/tmp/L03-3/is_deleted/L03-3_genome.deleted_IS.0.fasta exp/multiple_runs11/tmp/L03-3/is_deleted/L03-3_genome.deleted_IS.2.fasta\n",
      "[M::main] Real time: 0.610 sec; CPU: 0.603 sec; Peak RSS: 0.086 GB\n",
      "local_variation - WARNING - Finished syri\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "TLOut.txt  is empty. Skipping analysing it.\n",
      "invTLOut.txt  is empty. Skipping analysing it.\n",
      "ctxOut.txt is empty. Skipping analysing it.\n",
      "TL Out.txt is empty. Skipping analysing it.\n",
      "invTL Out.txt is empty. Skipping analysing it.\n",
      "dup Out.txt is empty. Skipping analysing it.\n",
      "ctxOut.txt is empty. Skipping analysing it.\n",
      "TLOut.txt  is empty. Skipping analysing it.\n",
      "invTLOut.txt  is empty. Skipping analysing it.\n",
      "ctxOut.txt is empty. Skipping analysing it.\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "[M::mm_idx_gen::0.062*1.01] collected minimizers\n",
      "[M::mm_idx_gen::0.081*1.00] sorted minimizers\n",
      "[M::main::0.081*1.00] loaded/built the index for 1 target sequence(s)\n",
      "[M::mm_mapopt_update::0.081*1.00] mid_occ = 100\n",
      "[M::mm_idx_stat] kmer size: 19; skip: 10; is_hpc: 1; #seq: 1\n",
      "[M::mm_idx_stat::0.085*1.00] distinct minimizers: 525478 (99.34% are singletons); average occurrences: 1.015; average spacing: 7.461; total length: 3977333\n",
      "[M::worker_pipeline::0.698*0.99] mapped 1 sequences\n",
      "[M::main] Version: 2.19-r1057\n",
      "[M::main] CMD: minimap2 -ax asm20 --eqx -H -f 100 -r1k,10k --rmq=no -t 1 exp/multiple_runs11/tmp/L03-3/is_deleted/L03-3_genome.deleted_IS.0.fasta exp/multiple_runs11/tmp/L03-3/is_deleted/L03-3_genome.deleted_IS.3.fasta\n",
      "[M::main] Real time: 0.702 sec; CPU: 0.695 sec; Peak RSS: 0.086 GB\n",
      "local_variation - WARNING - Finished syri\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "invTLOut.txt  is empty. Skipping analysing it.\n",
      "ctxOut.txt is empty. Skipping analysing it.\n",
      "invTL Out.txt is empty. Skipping analysing it.\n",
      "dup Out.txt is empty. Skipping analysing it.\n",
      "invDup Out.txt is empty. Skipping analysing it.\n",
      "ctxOut.txt is empty. Skipping analysing it.\n",
      "invTLOut.txt  is empty. Skipping analysing it.\n",
      "ctxOut.txt is empty. Skipping analysing it.\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "[M::mm_idx_gen::0.057*1.01] collected minimizers\n",
      "[M::mm_idx_gen::0.075*1.00] sorted minimizers\n",
      "[M::main::0.075*1.00] loaded/built the index for 1 target sequence(s)\n",
      "[M::mm_mapopt_update::0.075*1.00] mid_occ = 100\n",
      "[M::mm_idx_stat] kmer size: 19; skip: 10; is_hpc: 1; #seq: 1\n",
      "[M::mm_idx_stat::0.079*1.00] distinct minimizers: 525478 (99.34% are singletons); average occurrences: 1.015; average spacing: 7.461; total length: 3977333\n",
      "[M::worker_pipeline::0.607*0.99] mapped 1 sequences\n",
      "[M::main] Version: 2.19-r1057\n",
      "[M::main] CMD: minimap2 -ax asm20 --eqx -H -f 100 -r1k,10k --rmq=no -t 1 exp/multiple_runs11/tmp/L03-4/is_deleted/L03-4_genome.deleted_IS.0.fasta exp/multiple_runs11/tmp/L03-4/is_deleted/L03-4_genome.deleted_IS.1.fasta\n",
      "[M::main] Real time: 0.609 sec; CPU: 0.603 sec; Peak RSS: 0.087 GB\n",
      "local_variation - WARNING - Finished syri\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "TLOut.txt  is empty. Skipping analysing it.\n",
      "invTLOut.txt  is empty. Skipping analysing it.\n",
      "ctxOut.txt is empty. Skipping analysing it.\n",
      "TL Out.txt is empty. Skipping analysing it.\n",
      "invTL Out.txt is empty. Skipping analysing it.\n",
      "dup Out.txt is empty. Skipping analysing it.\n",
      "invDup Out.txt is empty. Skipping analysing it.\n",
      "ctxOut.txt is empty. Skipping analysing it.\n",
      "TLOut.txt  is empty. Skipping analysing it.\n",
      "invTLOut.txt  is empty. Skipping analysing it.\n",
      "ctxOut.txt is empty. Skipping analysing it.\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "[M::mm_idx_gen::0.062*1.01] collected minimizers\n",
      "[M::mm_idx_gen::0.080*1.00] sorted minimizers\n",
      "[M::main::0.080*1.00] loaded/built the index for 1 target sequence(s)\n",
      "[M::mm_mapopt_update::0.080*1.00] mid_occ = 100\n",
      "[M::mm_idx_stat] kmer size: 19; skip: 10; is_hpc: 1; #seq: 1\n",
      "[M::mm_idx_stat::0.084*1.00] distinct minimizers: 525478 (99.34% are singletons); average occurrences: 1.015; average spacing: 7.461; total length: 3977333\n",
      "[M::worker_pipeline::0.611*0.99] mapped 1 sequences\n",
      "[M::main] Version: 2.19-r1057\n",
      "[M::main] CMD: minimap2 -ax asm20 --eqx -H -f 100 -r1k,10k --rmq=no -t 1 exp/multiple_runs11/tmp/L03-4/is_deleted/L03-4_genome.deleted_IS.0.fasta exp/multiple_runs11/tmp/L03-4/is_deleted/L03-4_genome.deleted_IS.2.fasta\n",
      "[M::main] Real time: 0.613 sec; CPU: 0.607 sec; Peak RSS: 0.088 GB\n",
      "local_variation - WARNING - Finished syri\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "TLOut.txt  is empty. Skipping analysing it.\n",
      "invTLOut.txt  is empty. Skipping analysing it.\n",
      "ctxOut.txt is empty. Skipping analysing it.\n",
      "TL Out.txt is empty. Skipping analysing it.\n",
      "invTL Out.txt is empty. Skipping analysing it.\n",
      "dup Out.txt is empty. Skipping analysing it.\n",
      "invDup Out.txt is empty. Skipping analysing it.\n",
      "ctxOut.txt is empty. Skipping analysing it.\n",
      "TLOut.txt  is empty. Skipping analysing it.\n",
      "invTLOut.txt  is empty. Skipping analysing it.\n",
      "ctxOut.txt is empty. Skipping analysing it.\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "[M::mm_idx_gen::0.062*1.01] collected minimizers\n",
      "[M::mm_idx_gen::0.081*1.00] sorted minimizers\n",
      "[M::main::0.081*1.00] loaded/built the index for 1 target sequence(s)\n",
      "[M::mm_mapopt_update::0.081*1.00] mid_occ = 100\n",
      "[M::mm_idx_stat] kmer size: 19; skip: 10; is_hpc: 1; #seq: 1\n",
      "[M::mm_idx_stat::0.085*1.00] distinct minimizers: 525478 (99.34% are singletons); average occurrences: 1.015; average spacing: 7.461; total length: 3977333\n",
      "[M::worker_pipeline::0.618*0.99] mapped 1 sequences\n",
      "[M::main] Version: 2.19-r1057\n",
      "[M::main] CMD: minimap2 -ax asm20 --eqx -H -f 100 -r1k,10k --rmq=no -t 1 exp/multiple_runs11/tmp/L03-4/is_deleted/L03-4_genome.deleted_IS.0.fasta exp/multiple_runs11/tmp/L03-4/is_deleted/L03-4_genome.deleted_IS.3.fasta\n",
      "[M::main] Real time: 0.620 sec; CPU: 0.614 sec; Peak RSS: 0.088 GB\n",
      "local_variation - WARNING - Finished syri\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "TLOut.txt  is empty. Skipping analysing it.\n",
      "invTLOut.txt  is empty. Skipping analysing it.\n",
      "ctxOut.txt is empty. Skipping analysing it.\n",
      "TL Out.txt is empty. Skipping analysing it.\n",
      "invTL Out.txt is empty. Skipping analysing it.\n",
      "invDup Out.txt is empty. Skipping analysing it.\n",
      "ctxOut.txt is empty. Skipping analysing it.\n",
      "TLOut.txt  is empty. Skipping analysing it.\n",
      "invTLOut.txt  is empty. Skipping analysing it.\n",
      "ctxOut.txt is empty. Skipping analysing it.\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "[M::mm_idx_gen::0.064*1.01] collected minimizers\n",
      "[M::mm_idx_gen::0.083*1.00] sorted minimizers\n",
      "[M::main::0.083*1.00] loaded/built the index for 1 target sequence(s)\n",
      "[M::mm_mapopt_update::0.083*1.00] mid_occ = 100\n",
      "[M::mm_idx_stat] kmer size: 19; skip: 10; is_hpc: 1; #seq: 1\n",
      "[M::mm_idx_stat::0.086*1.00] distinct minimizers: 525478 (99.34% are singletons); average occurrences: 1.015; average spacing: 7.461; total length: 3977333\n",
      "[M::worker_pipeline::0.619*1.00] mapped 1 sequences\n",
      "[M::main] Version: 2.19-r1057\n",
      "[M::main] CMD: minimap2 -ax asm20 --eqx -H -f 100 -r1k,10k --rmq=no -t 1 exp/multiple_runs11/tmp/L04-1/is_deleted/L04-1_genome.deleted_IS.0.fasta exp/multiple_runs11/tmp/L04-1/is_deleted/L04-1_genome.deleted_IS.1.fasta\n",
      "[M::main] Real time: 0.622 sec; CPU: 0.620 sec; Peak RSS: 0.086 GB\n",
      "local_variation - WARNING - Finished syri\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "invOut.txt  is empty. Skipping analysing it.\n",
      "TLOut.txt  is empty. Skipping analysing it.\n",
      "invTLOut.txt  is empty. Skipping analysing it.\n",
      "ctxOut.txt is empty. Skipping analysing it.\n",
      "inv Out.txt is empty. Skipping analysing it.\n",
      "TL Out.txt is empty. Skipping analysing it.\n",
      "invTL Out.txt is empty. Skipping analysing it.\n",
      "dup Out.txt is empty. Skipping analysing it.\n",
      "invDup Out.txt is empty. Skipping analysing it.\n",
      "ctxOut.txt is empty. Skipping analysing it.\n",
      "invOut.txt  is empty. Skipping analysing it.\n",
      "TLOut.txt  is empty. Skipping analysing it.\n",
      "invTLOut.txt  is empty. Skipping analysing it.\n",
      "ctxOut.txt is empty. Skipping analysing it.\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "[M::mm_idx_gen::0.057*1.01] collected minimizers\n",
      "[M::mm_idx_gen::0.076*1.00] sorted minimizers\n",
      "[M::main::0.076*1.00] loaded/built the index for 1 target sequence(s)\n",
      "[M::mm_mapopt_update::0.076*1.00] mid_occ = 100\n",
      "[M::mm_idx_stat] kmer size: 19; skip: 10; is_hpc: 1; #seq: 1\n",
      "[M::mm_idx_stat::0.080*1.00] distinct minimizers: 525478 (99.34% are singletons); average occurrences: 1.015; average spacing: 7.461; total length: 3977333\n",
      "[M::worker_pipeline::0.613*0.99] mapped 1 sequences\n",
      "[M::main] Version: 2.19-r1057\n",
      "[M::main] CMD: minimap2 -ax asm20 --eqx -H -f 100 -r1k,10k --rmq=no -t 1 exp/multiple_runs11/tmp/L04-1/is_deleted/L04-1_genome.deleted_IS.0.fasta exp/multiple_runs11/tmp/L04-1/is_deleted/L04-1_genome.deleted_IS.2.fasta\n",
      "[M::main] Real time: 0.615 sec; CPU: 0.609 sec; Peak RSS: 0.089 GB\n",
      "Reading Coords - WARNING - Reference chromosome contig_1 has high fraction of inverted alignments with its homologous chromosome in the query genome (contig_1). Ensure that same chromosome-strands are being compared in the two genomes, as different strand can result in unexpected errors.\n",
      "local_variation - WARNING - Finished syri\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "TLOut.txt  is empty. Skipping analysing it.\n",
      "invTLOut.txt  is empty. Skipping analysing it.\n",
      "ctxOut.txt is empty. Skipping analysing it.\n",
      "TL Out.txt is empty. Skipping analysing it.\n",
      "invTL Out.txt is empty. Skipping analysing it.\n",
      "dup Out.txt is empty. Skipping analysing it.\n",
      "invDup Out.txt is empty. Skipping analysing it.\n",
      "ctxOut.txt is empty. Skipping analysing it.\n",
      "TLOut.txt  is empty. Skipping analysing it.\n",
      "invTLOut.txt  is empty. Skipping analysing it.\n",
      "ctxOut.txt is empty. Skipping analysing it.\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "[M::mm_idx_gen::0.058*1.01] collected minimizers\n",
      "[M::mm_idx_gen::0.077*1.00] sorted minimizers\n",
      "[M::main::0.077*1.00] loaded/built the index for 1 target sequence(s)\n",
      "[M::mm_mapopt_update::0.077*1.00] mid_occ = 100\n",
      "[M::mm_idx_stat] kmer size: 19; skip: 10; is_hpc: 1; #seq: 1\n",
      "[M::mm_idx_stat::0.080*1.00] distinct minimizers: 525478 (99.34% are singletons); average occurrences: 1.015; average spacing: 7.461; total length: 3977333\n",
      "[M::worker_pipeline::0.614*0.99] mapped 1 sequences\n",
      "[M::main] Version: 2.19-r1057\n",
      "[M::main] CMD: minimap2 -ax asm20 --eqx -H -f 100 -r1k,10k --rmq=no -t 1 exp/multiple_runs11/tmp/L04-1/is_deleted/L04-1_genome.deleted_IS.0.fasta exp/multiple_runs11/tmp/L04-1/is_deleted/L04-1_genome.deleted_IS.3.fasta\n",
      "[M::main] Real time: 0.617 sec; CPU: 0.610 sec; Peak RSS: 0.088 GB\n",
      "local_variation - WARNING - Finished syri\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "TLOut.txt  is empty. Skipping analysing it.\n",
      "invTLOut.txt  is empty. Skipping analysing it.\n",
      "ctxOut.txt is empty. Skipping analysing it.\n",
      "TL Out.txt is empty. Skipping analysing it.\n",
      "invTL Out.txt is empty. Skipping analysing it.\n",
      "dup Out.txt is empty. Skipping analysing it.\n",
      "invDup Out.txt is empty. Skipping analysing it.\n",
      "ctxOut.txt is empty. Skipping analysing it.\n",
      "TLOut.txt  is empty. Skipping analysing it.\n",
      "invTLOut.txt  is empty. Skipping analysing it.\n",
      "ctxOut.txt is empty. Skipping analysing it.\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "[M::mm_idx_gen::0.057*1.01] collected minimizers\n",
      "[M::mm_idx_gen::0.076*1.00] sorted minimizers\n",
      "[M::main::0.076*1.00] loaded/built the index for 1 target sequence(s)\n",
      "[M::mm_mapopt_update::0.076*1.00] mid_occ = 100\n",
      "[M::mm_idx_stat] kmer size: 19; skip: 10; is_hpc: 1; #seq: 1\n",
      "[M::mm_idx_stat::0.080*1.00] distinct minimizers: 525478 (99.34% are singletons); average occurrences: 1.015; average spacing: 7.461; total length: 3977333\n",
      "[M::worker_pipeline::0.612*1.00] mapped 1 sequences\n",
      "[M::main] Version: 2.19-r1057\n",
      "[M::main] CMD: minimap2 -ax asm20 --eqx -H -f 100 -r1k,10k --rmq=no -t 1 exp/multiple_runs11/tmp/L04-2/is_deleted/L04-2_genome.deleted_IS.0.fasta exp/multiple_runs11/tmp/L04-2/is_deleted/L04-2_genome.deleted_IS.1.fasta\n",
      "[M::main] Real time: 0.615 sec; CPU: 0.613 sec; Peak RSS: 0.086 GB\n",
      "local_variation - WARNING - Finished syri\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "invOut.txt  is empty. Skipping analysing it.\n",
      "TLOut.txt  is empty. Skipping analysing it.\n",
      "invTLOut.txt  is empty. Skipping analysing it.\n",
      "ctxOut.txt is empty. Skipping analysing it.\n",
      "inv Out.txt is empty. Skipping analysing it.\n",
      "TL Out.txt is empty. Skipping analysing it.\n",
      "invTL Out.txt is empty. Skipping analysing it.\n",
      "dup Out.txt is empty. Skipping analysing it.\n",
      "invDup Out.txt is empty. Skipping analysing it.\n",
      "ctxOut.txt is empty. Skipping analysing it.\n",
      "invOut.txt  is empty. Skipping analysing it.\n",
      "TLOut.txt  is empty. Skipping analysing it.\n",
      "invTLOut.txt  is empty. Skipping analysing it.\n",
      "ctxOut.txt is empty. Skipping analysing it.\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "[M::mm_idx_gen::0.058*1.01] collected minimizers\n",
      "[M::mm_idx_gen::0.076*1.00] sorted minimizers\n",
      "[M::main::0.076*1.00] loaded/built the index for 1 target sequence(s)\n",
      "[M::mm_mapopt_update::0.076*1.00] mid_occ = 100\n",
      "[M::mm_idx_stat] kmer size: 19; skip: 10; is_hpc: 1; #seq: 1\n",
      "[M::mm_idx_stat::0.080*1.00] distinct minimizers: 525478 (99.34% are singletons); average occurrences: 1.015; average spacing: 7.461; total length: 3977333\n",
      "[M::worker_pipeline::0.615*0.99] mapped 1 sequences\n",
      "[M::main] Version: 2.19-r1057\n",
      "[M::main] CMD: minimap2 -ax asm20 --eqx -H -f 100 -r1k,10k --rmq=no -t 1 exp/multiple_runs11/tmp/L04-2/is_deleted/L04-2_genome.deleted_IS.0.fasta exp/multiple_runs11/tmp/L04-2/is_deleted/L04-2_genome.deleted_IS.2.fasta\n",
      "[M::main] Real time: 0.618 sec; CPU: 0.612 sec; Peak RSS: 0.087 GB\n",
      "local_variation - WARNING - Finished syri\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "invOut.txt  is empty. Skipping analysing it.\n",
      "TLOut.txt  is empty. Skipping analysing it.\n",
      "invTLOut.txt  is empty. Skipping analysing it.\n",
      "ctxOut.txt is empty. Skipping analysing it.\n",
      "inv Out.txt is empty. Skipping analysing it.\n",
      "TL Out.txt is empty. Skipping analysing it.\n",
      "invTL Out.txt is empty. Skipping analysing it.\n",
      "dup Out.txt is empty. Skipping analysing it.\n",
      "invDup Out.txt is empty. Skipping analysing it.\n",
      "ctxOut.txt is empty. Skipping analysing it.\n",
      "invOut.txt  is empty. Skipping analysing it.\n",
      "TLOut.txt  is empty. Skipping analysing it.\n",
      "invTLOut.txt  is empty. Skipping analysing it.\n",
      "ctxOut.txt is empty. Skipping analysing it.\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "[M::mm_idx_gen::0.058*1.01] collected minimizers\n",
      "[M::mm_idx_gen::0.076*1.00] sorted minimizers\n",
      "[M::main::0.076*1.00] loaded/built the index for 1 target sequence(s)\n",
      "[M::mm_mapopt_update::0.076*1.00] mid_occ = 100\n",
      "[M::mm_idx_stat] kmer size: 19; skip: 10; is_hpc: 1; #seq: 1\n",
      "[M::mm_idx_stat::0.080*1.00] distinct minimizers: 525478 (99.34% are singletons); average occurrences: 1.015; average spacing: 7.461; total length: 3977333\n",
      "[M::worker_pipeline::0.605*0.99] mapped 1 sequences\n",
      "[M::main] Version: 2.19-r1057\n",
      "[M::main] CMD: minimap2 -ax asm20 --eqx -H -f 100 -r1k,10k --rmq=no -t 1 exp/multiple_runs11/tmp/L04-2/is_deleted/L04-2_genome.deleted_IS.0.fasta exp/multiple_runs11/tmp/L04-2/is_deleted/L04-2_genome.deleted_IS.3.fasta\n",
      "[M::main] Real time: 0.608 sec; CPU: 0.602 sec; Peak RSS: 0.088 GB\n",
      "local_variation - WARNING - Finished syri\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "TLOut.txt  is empty. Skipping analysing it.\n",
      "invTLOut.txt  is empty. Skipping analysing it.\n",
      "ctxOut.txt is empty. Skipping analysing it.\n",
      "TL Out.txt is empty. Skipping analysing it.\n",
      "invTL Out.txt is empty. Skipping analysing it.\n",
      "dup Out.txt is empty. Skipping analysing it.\n",
      "invDup Out.txt is empty. Skipping analysing it.\n",
      "ctxOut.txt is empty. Skipping analysing it.\n",
      "TLOut.txt  is empty. Skipping analysing it.\n",
      "invTLOut.txt  is empty. Skipping analysing it.\n",
      "ctxOut.txt is empty. Skipping analysing it.\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "[M::mm_idx_gen::0.063*1.01] collected minimizers\n",
      "[M::mm_idx_gen::0.081*1.00] sorted minimizers\n",
      "[M::main::0.081*1.00] loaded/built the index for 1 target sequence(s)\n",
      "[M::mm_mapopt_update::0.081*1.00] mid_occ = 100\n",
      "[M::mm_idx_stat] kmer size: 19; skip: 10; is_hpc: 1; #seq: 1\n",
      "[M::mm_idx_stat::0.085*1.00] distinct minimizers: 525478 (99.34% are singletons); average occurrences: 1.015; average spacing: 7.461; total length: 3977333\n",
      "[M::worker_pipeline::0.619*1.00] mapped 1 sequences\n",
      "[M::main] Version: 2.19-r1057\n",
      "[M::main] CMD: minimap2 -ax asm20 --eqx -H -f 100 -r1k,10k --rmq=no -t 1 exp/multiple_runs11/tmp/L04-3/is_deleted/L04-3_genome.deleted_IS.0.fasta exp/multiple_runs11/tmp/L04-3/is_deleted/L04-3_genome.deleted_IS.1.fasta\n",
      "[M::main] Real time: 0.622 sec; CPU: 0.620 sec; Peak RSS: 0.086 GB\n",
      "local_variation - WARNING - Finished syri\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "invOut.txt  is empty. Skipping analysing it.\n",
      "TLOut.txt  is empty. Skipping analysing it.\n",
      "invTLOut.txt  is empty. Skipping analysing it.\n",
      "ctxOut.txt is empty. Skipping analysing it.\n",
      "inv Out.txt is empty. Skipping analysing it.\n",
      "TL Out.txt is empty. Skipping analysing it.\n",
      "invTL Out.txt is empty. Skipping analysing it.\n",
      "dup Out.txt is empty. Skipping analysing it.\n",
      "invDup Out.txt is empty. Skipping analysing it.\n",
      "ctxOut.txt is empty. Skipping analysing it.\n",
      "invOut.txt  is empty. Skipping analysing it.\n",
      "TLOut.txt  is empty. Skipping analysing it.\n",
      "invTLOut.txt  is empty. Skipping analysing it.\n",
      "ctxOut.txt is empty. Skipping analysing it.\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "[M::mm_idx_gen::0.062*1.01] collected minimizers\n",
      "[M::mm_idx_gen::0.081*1.00] sorted minimizers\n",
      "[M::main::0.081*1.00] loaded/built the index for 1 target sequence(s)\n",
      "[M::mm_mapopt_update::0.081*1.00] mid_occ = 100\n",
      "[M::mm_idx_stat] kmer size: 19; skip: 10; is_hpc: 1; #seq: 1\n",
      "[M::mm_idx_stat::0.085*1.00] distinct minimizers: 525478 (99.34% are singletons); average occurrences: 1.015; average spacing: 7.461; total length: 3977333\n",
      "[M::worker_pipeline::0.666*0.99] mapped 1 sequences\n",
      "[M::main] Version: 2.19-r1057\n",
      "[M::main] CMD: minimap2 -ax asm20 --eqx -H -f 100 -r1k,10k --rmq=no -t 1 exp/multiple_runs11/tmp/L04-3/is_deleted/L04-3_genome.deleted_IS.0.fasta exp/multiple_runs11/tmp/L04-3/is_deleted/L04-3_genome.deleted_IS.2.fasta\n",
      "[M::main] Real time: 0.668 sec; CPU: 0.661 sec; Peak RSS: 0.088 GB\n",
      "local_variation - WARNING - Finished syri\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "invOut.txt  is empty. Skipping analysing it.\n",
      "TLOut.txt  is empty. Skipping analysing it.\n",
      "invTLOut.txt  is empty. Skipping analysing it.\n",
      "ctxOut.txt is empty. Skipping analysing it.\n",
      "inv Out.txt is empty. Skipping analysing it.\n",
      "TL Out.txt is empty. Skipping analysing it.\n",
      "invTL Out.txt is empty. Skipping analysing it.\n",
      "ctxOut.txt is empty. Skipping analysing it.\n",
      "invOut.txt  is empty. Skipping analysing it.\n",
      "TLOut.txt  is empty. Skipping analysing it.\n",
      "invTLOut.txt  is empty. Skipping analysing it.\n",
      "ctxOut.txt is empty. Skipping analysing it.\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "[M::mm_idx_gen::0.058*1.01] collected minimizers\n",
      "[M::mm_idx_gen::0.076*1.00] sorted minimizers\n",
      "[M::main::0.076*1.00] loaded/built the index for 1 target sequence(s)\n",
      "[M::mm_mapopt_update::0.076*1.00] mid_occ = 100\n",
      "[M::mm_idx_stat] kmer size: 19; skip: 10; is_hpc: 1; #seq: 1\n",
      "[M::mm_idx_stat::0.080*1.00] distinct minimizers: 525478 (99.34% are singletons); average occurrences: 1.015; average spacing: 7.461; total length: 3977333\n",
      "[M::worker_pipeline::0.639*0.99] mapped 1 sequences\n",
      "[M::main] Version: 2.19-r1057\n",
      "[M::main] CMD: minimap2 -ax asm20 --eqx -H -f 100 -r1k,10k --rmq=no -t 1 exp/multiple_runs11/tmp/L04-3/is_deleted/L04-3_genome.deleted_IS.0.fasta exp/multiple_runs11/tmp/L04-3/is_deleted/L04-3_genome.deleted_IS.3.fasta\n",
      "[M::main] Real time: 0.642 sec; CPU: 0.635 sec; Peak RSS: 0.087 GB\n",
      "local_variation - WARNING - Finished syri\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "TLOut.txt  is empty. Skipping analysing it.\n",
      "invTLOut.txt  is empty. Skipping analysing it.\n",
      "ctxOut.txt is empty. Skipping analysing it.\n",
      "TL Out.txt is empty. Skipping analysing it.\n",
      "invTL Out.txt is empty. Skipping analysing it.\n",
      "invDup Out.txt is empty. Skipping analysing it.\n",
      "ctxOut.txt is empty. Skipping analysing it.\n",
      "TLOut.txt  is empty. Skipping analysing it.\n",
      "invTLOut.txt  is empty. Skipping analysing it.\n",
      "ctxOut.txt is empty. Skipping analysing it.\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "[M::mm_idx_gen::0.064*1.00] collected minimizers\n",
      "[M::mm_idx_gen::0.082*1.00] sorted minimizers\n",
      "[M::main::0.082*1.00] loaded/built the index for 1 target sequence(s)\n",
      "[M::mm_mapopt_update::0.082*1.00] mid_occ = 100\n",
      "[M::mm_idx_stat] kmer size: 19; skip: 10; is_hpc: 1; #seq: 1\n",
      "[M::mm_idx_stat::0.086*1.00] distinct minimizers: 525478 (99.34% are singletons); average occurrences: 1.015; average spacing: 7.461; total length: 3977333\n",
      "[M::worker_pipeline::0.613*1.00] mapped 1 sequences\n",
      "[M::main] Version: 2.19-r1057\n",
      "[M::main] CMD: minimap2 -ax asm20 --eqx -H -f 100 -r1k,10k --rmq=no -t 1 exp/multiple_runs11/tmp/L04-4/is_deleted/L04-4_genome.deleted_IS.0.fasta exp/multiple_runs11/tmp/L04-4/is_deleted/L04-4_genome.deleted_IS.1.fasta\n",
      "[M::main] Real time: 0.616 sec; CPU: 0.614 sec; Peak RSS: 0.087 GB\n",
      "local_variation - WARNING - Finished syri\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "invOut.txt  is empty. Skipping analysing it.\n",
      "TLOut.txt  is empty. Skipping analysing it.\n",
      "invTLOut.txt  is empty. Skipping analysing it.\n",
      "ctxOut.txt is empty. Skipping analysing it.\n",
      "inv Out.txt is empty. Skipping analysing it.\n",
      "TL Out.txt is empty. Skipping analysing it.\n",
      "invTL Out.txt is empty. Skipping analysing it.\n",
      "dup Out.txt is empty. Skipping analysing it.\n",
      "invDup Out.txt is empty. Skipping analysing it.\n",
      "ctxOut.txt is empty. Skipping analysing it.\n",
      "invOut.txt  is empty. Skipping analysing it.\n",
      "TLOut.txt  is empty. Skipping analysing it.\n",
      "invTLOut.txt  is empty. Skipping analysing it.\n",
      "ctxOut.txt is empty. Skipping analysing it.\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "[M::mm_idx_gen::0.063*1.01] collected minimizers\n",
      "[M::mm_idx_gen::0.081*1.00] sorted minimizers\n",
      "[M::main::0.081*1.00] loaded/built the index for 1 target sequence(s)\n",
      "[M::mm_mapopt_update::0.082*1.00] mid_occ = 100\n",
      "[M::mm_idx_stat] kmer size: 19; skip: 10; is_hpc: 1; #seq: 1\n",
      "[M::mm_idx_stat::0.085*1.00] distinct minimizers: 525478 (99.34% are singletons); average occurrences: 1.015; average spacing: 7.461; total length: 3977333\n",
      "[M::worker_pipeline::0.617*0.99] mapped 1 sequences\n",
      "[M::main] Version: 2.19-r1057\n",
      "[M::main] CMD: minimap2 -ax asm20 --eqx -H -f 100 -r1k,10k --rmq=no -t 1 exp/multiple_runs11/tmp/L04-4/is_deleted/L04-4_genome.deleted_IS.0.fasta exp/multiple_runs11/tmp/L04-4/is_deleted/L04-4_genome.deleted_IS.2.fasta\n",
      "[M::main] Real time: 0.619 sec; CPU: 0.612 sec; Peak RSS: 0.088 GB\n",
      "local_variation - WARNING - Finished syri\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "invOut.txt  is empty. Skipping analysing it.\n",
      "TLOut.txt  is empty. Skipping analysing it.\n",
      "invTLOut.txt  is empty. Skipping analysing it.\n",
      "ctxOut.txt is empty. Skipping analysing it.\n",
      "inv Out.txt is empty. Skipping analysing it.\n",
      "TL Out.txt is empty. Skipping analysing it.\n",
      "invTL Out.txt is empty. Skipping analysing it.\n",
      "dup Out.txt is empty. Skipping analysing it.\n",
      "ctxOut.txt is empty. Skipping analysing it.\n",
      "invOut.txt  is empty. Skipping analysing it.\n",
      "TLOut.txt  is empty. Skipping analysing it.\n",
      "invTLOut.txt  is empty. Skipping analysing it.\n",
      "ctxOut.txt is empty. Skipping analysing it.\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "[M::mm_idx_gen::0.058*1.01] collected minimizers\n",
      "[M::mm_idx_gen::0.076*1.00] sorted minimizers\n",
      "[M::main::0.076*1.00] loaded/built the index for 1 target sequence(s)\n",
      "[M::mm_mapopt_update::0.076*1.00] mid_occ = 100\n",
      "[M::mm_idx_stat] kmer size: 19; skip: 10; is_hpc: 1; #seq: 1\n",
      "[M::mm_idx_stat::0.080*1.00] distinct minimizers: 525478 (99.34% are singletons); average occurrences: 1.015; average spacing: 7.461; total length: 3977333\n",
      "[M::worker_pipeline::0.629*0.99] mapped 1 sequences\n",
      "[M::main] Version: 2.19-r1057\n",
      "[M::main] CMD: minimap2 -ax asm20 --eqx -H -f 100 -r1k,10k --rmq=no -t 1 exp/multiple_runs11/tmp/L04-4/is_deleted/L04-4_genome.deleted_IS.0.fasta exp/multiple_runs11/tmp/L04-4/is_deleted/L04-4_genome.deleted_IS.3.fasta\n",
      "[M::main] Real time: 0.631 sec; CPU: 0.625 sec; Peak RSS: 0.089 GB\n",
      "local_variation - WARNING - Finished syri\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "TLOut.txt  is empty. Skipping analysing it.\n",
      "invTLOut.txt  is empty. Skipping analysing it.\n",
      "ctxOut.txt is empty. Skipping analysing it.\n",
      "TL Out.txt is empty. Skipping analysing it.\n",
      "invTL Out.txt is empty. Skipping analysing it.\n",
      "ctxOut.txt is empty. Skipping analysing it.\n",
      "TLOut.txt  is empty. Skipping analysing it.\n",
      "invTLOut.txt  is empty. Skipping analysing it.\n",
      "ctxOut.txt is empty. Skipping analysing it.\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "[M::mm_idx_gen::0.062*1.01] collected minimizers\n",
      "[M::mm_idx_gen::0.081*1.00] sorted minimizers\n",
      "[M::main::0.081*1.00] loaded/built the index for 1 target sequence(s)\n",
      "[M::mm_mapopt_update::0.081*1.00] mid_occ = 100\n",
      "[M::mm_idx_stat] kmer size: 19; skip: 10; is_hpc: 1; #seq: 1\n",
      "[M::mm_idx_stat::0.085*1.00] distinct minimizers: 525478 (99.34% are singletons); average occurrences: 1.015; average spacing: 7.461; total length: 3977333\n",
      "[M::worker_pipeline::0.609*0.99] mapped 1 sequences\n",
      "[M::main] Version: 2.19-r1057\n",
      "[M::main] CMD: minimap2 -ax asm20 --eqx -H -f 100 -r1k,10k --rmq=no -t 1 exp/multiple_runs11/tmp/L05-1/is_deleted/L05-1_genome.deleted_IS.0.fasta exp/multiple_runs11/tmp/L05-1/is_deleted/L05-1_genome.deleted_IS.1.fasta\n",
      "[M::main] Real time: 0.612 sec; CPU: 0.606 sec; Peak RSS: 0.086 GB\n",
      "local_variation - WARNING - Finished syri\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "invOut.txt  is empty. Skipping analysing it.\n",
      "TLOut.txt  is empty. Skipping analysing it.\n",
      "invTLOut.txt  is empty. Skipping analysing it.\n",
      "ctxOut.txt is empty. Skipping analysing it.\n",
      "inv Out.txt is empty. Skipping analysing it.\n",
      "TL Out.txt is empty. Skipping analysing it.\n",
      "invTL Out.txt is empty. Skipping analysing it.\n",
      "dup Out.txt is empty. Skipping analysing it.\n",
      "invDup Out.txt is empty. Skipping analysing it.\n",
      "ctxOut.txt is empty. Skipping analysing it.\n",
      "invOut.txt  is empty. Skipping analysing it.\n",
      "TLOut.txt  is empty. Skipping analysing it.\n",
      "invTLOut.txt  is empty. Skipping analysing it.\n",
      "ctxOut.txt is empty. Skipping analysing it.\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "[M::mm_idx_gen::0.061*1.01] collected minimizers\n",
      "[M::mm_idx_gen::0.080*1.00] sorted minimizers\n",
      "[M::main::0.080*1.00] loaded/built the index for 1 target sequence(s)\n",
      "[M::mm_mapopt_update::0.080*1.00] mid_occ = 100\n",
      "[M::mm_idx_stat] kmer size: 19; skip: 10; is_hpc: 1; #seq: 1\n",
      "[M::mm_idx_stat::0.084*1.00] distinct minimizers: 525478 (99.34% are singletons); average occurrences: 1.015; average spacing: 7.461; total length: 3977333\n",
      "[M::worker_pipeline::0.606*0.99] mapped 1 sequences\n",
      "[M::main] Version: 2.19-r1057\n",
      "[M::main] CMD: minimap2 -ax asm20 --eqx -H -f 100 -r1k,10k --rmq=no -t 1 exp/multiple_runs11/tmp/L05-1/is_deleted/L05-1_genome.deleted_IS.0.fasta exp/multiple_runs11/tmp/L05-1/is_deleted/L05-1_genome.deleted_IS.2.fasta\n",
      "[M::main] Real time: 0.610 sec; CPU: 0.604 sec; Peak RSS: 0.087 GB\n",
      "local_variation - WARNING - Finished syri\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "invOut.txt  is empty. Skipping analysing it.\n",
      "TLOut.txt  is empty. Skipping analysing it.\n",
      "invTLOut.txt  is empty. Skipping analysing it.\n",
      "ctxOut.txt is empty. Skipping analysing it.\n",
      "inv Out.txt is empty. Skipping analysing it.\n",
      "TL Out.txt is empty. Skipping analysing it.\n",
      "invTL Out.txt is empty. Skipping analysing it.\n",
      "dup Out.txt is empty. Skipping analysing it.\n",
      "invDup Out.txt is empty. Skipping analysing it.\n",
      "ctxOut.txt is empty. Skipping analysing it.\n",
      "invOut.txt  is empty. Skipping analysing it.\n",
      "TLOut.txt  is empty. Skipping analysing it.\n",
      "invTLOut.txt  is empty. Skipping analysing it.\n",
      "ctxOut.txt is empty. Skipping analysing it.\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "[M::mm_idx_gen::0.059*1.01] collected minimizers\n",
      "[M::mm_idx_gen::0.078*1.00] sorted minimizers\n",
      "[M::main::0.078*1.00] loaded/built the index for 1 target sequence(s)\n",
      "[M::mm_mapopt_update::0.078*1.00] mid_occ = 100\n",
      "[M::mm_idx_stat] kmer size: 19; skip: 10; is_hpc: 1; #seq: 1\n",
      "[M::mm_idx_stat::0.082*1.00] distinct minimizers: 525478 (99.34% are singletons); average occurrences: 1.015; average spacing: 7.461; total length: 3977333\n",
      "[M::worker_pipeline::0.600*0.99] mapped 1 sequences\n",
      "[M::main] Version: 2.19-r1057\n",
      "[M::main] CMD: minimap2 -ax asm20 --eqx -H -f 100 -r1k,10k --rmq=no -t 1 exp/multiple_runs11/tmp/L05-1/is_deleted/L05-1_genome.deleted_IS.0.fasta exp/multiple_runs11/tmp/L05-1/is_deleted/L05-1_genome.deleted_IS.3.fasta\n",
      "[M::main] Real time: 0.604 sec; CPU: 0.598 sec; Peak RSS: 0.089 GB\n",
      "local_variation - WARNING - Finished syri\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "invOut.txt  is empty. Skipping analysing it.\n",
      "TLOut.txt  is empty. Skipping analysing it.\n",
      "invTLOut.txt  is empty. Skipping analysing it.\n",
      "ctxOut.txt is empty. Skipping analysing it.\n",
      "inv Out.txt is empty. Skipping analysing it.\n",
      "TL Out.txt is empty. Skipping analysing it.\n",
      "invTL Out.txt is empty. Skipping analysing it.\n",
      "dup Out.txt is empty. Skipping analysing it.\n",
      "invDup Out.txt is empty. Skipping analysing it.\n",
      "ctxOut.txt is empty. Skipping analysing it.\n",
      "invOut.txt  is empty. Skipping analysing it.\n",
      "TLOut.txt  is empty. Skipping analysing it.\n",
      "invTLOut.txt  is empty. Skipping analysing it.\n",
      "ctxOut.txt is empty. Skipping analysing it.\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "[M::mm_idx_gen::0.057*1.01] collected minimizers\n",
      "[M::mm_idx_gen::0.075*1.00] sorted minimizers\n",
      "[M::main::0.075*1.00] loaded/built the index for 1 target sequence(s)\n",
      "[M::mm_mapopt_update::0.075*1.00] mid_occ = 100\n",
      "[M::mm_idx_stat] kmer size: 19; skip: 10; is_hpc: 1; #seq: 1\n",
      "[M::mm_idx_stat::0.079*1.00] distinct minimizers: 525478 (99.34% are singletons); average occurrences: 1.015; average spacing: 7.461; total length: 3977333\n",
      "[M::worker_pipeline::0.603*0.99] mapped 1 sequences\n",
      "[M::main] Version: 2.19-r1057\n",
      "[M::main] CMD: minimap2 -ax asm20 --eqx -H -f 100 -r1k,10k --rmq=no -t 1 exp/multiple_runs11/tmp/L05-2/is_deleted/L05-2_genome.deleted_IS.0.fasta exp/multiple_runs11/tmp/L05-2/is_deleted/L05-2_genome.deleted_IS.1.fasta\n",
      "[M::main] Real time: 0.605 sec; CPU: 0.600 sec; Peak RSS: 0.087 GB\n",
      "local_variation - WARNING - Finished syri\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "invOut.txt  is empty. Skipping analysing it.\n",
      "TLOut.txt  is empty. Skipping analysing it.\n",
      "invTLOut.txt  is empty. Skipping analysing it.\n",
      "ctxOut.txt is empty. Skipping analysing it.\n",
      "inv Out.txt is empty. Skipping analysing it.\n",
      "TL Out.txt is empty. Skipping analysing it.\n",
      "invTL Out.txt is empty. Skipping analysing it.\n",
      "dup Out.txt is empty. Skipping analysing it.\n",
      "invDup Out.txt is empty. Skipping analysing it.\n",
      "ctxOut.txt is empty. Skipping analysing it.\n",
      "invOut.txt  is empty. Skipping analysing it.\n",
      "TLOut.txt  is empty. Skipping analysing it.\n",
      "invTLOut.txt  is empty. Skipping analysing it.\n",
      "ctxOut.txt is empty. Skipping analysing it.\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "[M::mm_idx_gen::0.062*1.01] collected minimizers\n",
      "[M::mm_idx_gen::0.081*1.00] sorted minimizers\n",
      "[M::main::0.081*1.00] loaded/built the index for 1 target sequence(s)\n",
      "[M::mm_mapopt_update::0.081*1.00] mid_occ = 100\n",
      "[M::mm_idx_stat] kmer size: 19; skip: 10; is_hpc: 1; #seq: 1\n",
      "[M::mm_idx_stat::0.085*1.00] distinct minimizers: 525478 (99.34% are singletons); average occurrences: 1.015; average spacing: 7.461; total length: 3977333\n",
      "[M::worker_pipeline::0.612*0.99] mapped 1 sequences\n",
      "[M::main] Version: 2.19-r1057\n",
      "[M::main] CMD: minimap2 -ax asm20 --eqx -H -f 100 -r1k,10k --rmq=no -t 1 exp/multiple_runs11/tmp/L05-2/is_deleted/L05-2_genome.deleted_IS.0.fasta exp/multiple_runs11/tmp/L05-2/is_deleted/L05-2_genome.deleted_IS.2.fasta\n",
      "[M::main] Real time: 0.616 sec; CPU: 0.608 sec; Peak RSS: 0.087 GB\n",
      "local_variation - WARNING - Finished syri\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "TLOut.txt  is empty. Skipping analysing it.\n",
      "ctxOut.txt is empty. Skipping analysing it.\n",
      "TL Out.txt is empty. Skipping analysing it.\n",
      "dup Out.txt is empty. Skipping analysing it.\n",
      "invDup Out.txt is empty. Skipping analysing it.\n",
      "ctxOut.txt is empty. Skipping analysing it.\n",
      "TLOut.txt  is empty. Skipping analysing it.\n",
      "ctxOut.txt is empty. Skipping analysing it.\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "[M::mm_idx_gen::0.057*1.01] collected minimizers\n",
      "[M::mm_idx_gen::0.076*1.00] sorted minimizers\n",
      "[M::main::0.076*1.00] loaded/built the index for 1 target sequence(s)\n",
      "[M::mm_mapopt_update::0.076*1.00] mid_occ = 100\n",
      "[M::mm_idx_stat] kmer size: 19; skip: 10; is_hpc: 1; #seq: 1\n",
      "[M::mm_idx_stat::0.080*1.00] distinct minimizers: 525478 (99.34% are singletons); average occurrences: 1.015; average spacing: 7.461; total length: 3977333\n",
      "[M::worker_pipeline::0.599*0.99] mapped 1 sequences\n",
      "[M::main] Version: 2.19-r1057\n",
      "[M::main] CMD: minimap2 -ax asm20 --eqx -H -f 100 -r1k,10k --rmq=no -t 1 exp/multiple_runs11/tmp/L05-2/is_deleted/L05-2_genome.deleted_IS.0.fasta exp/multiple_runs11/tmp/L05-2/is_deleted/L05-2_genome.deleted_IS.3.fasta\n",
      "[M::main] Real time: 0.602 sec; CPU: 0.596 sec; Peak RSS: 0.086 GB\n",
      "local_variation - WARNING - Finished syri\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "TLOut.txt  is empty. Skipping analysing it.\n",
      "invTLOut.txt  is empty. Skipping analysing it.\n",
      "ctxOut.txt is empty. Skipping analysing it.\n",
      "TL Out.txt is empty. Skipping analysing it.\n",
      "invTL Out.txt is empty. Skipping analysing it.\n",
      "invDup Out.txt is empty. Skipping analysing it.\n",
      "ctxOut.txt is empty. Skipping analysing it.\n",
      "TLOut.txt  is empty. Skipping analysing it.\n",
      "invTLOut.txt  is empty. Skipping analysing it.\n",
      "ctxOut.txt is empty. Skipping analysing it.\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "[M::mm_idx_gen::0.058*1.01] collected minimizers\n",
      "[M::mm_idx_gen::0.077*1.00] sorted minimizers\n",
      "[M::main::0.077*1.00] loaded/built the index for 1 target sequence(s)\n",
      "[M::mm_mapopt_update::0.077*1.00] mid_occ = 100\n",
      "[M::mm_idx_stat] kmer size: 19; skip: 10; is_hpc: 1; #seq: 1\n",
      "[M::mm_idx_stat::0.081*1.00] distinct minimizers: 525478 (99.34% are singletons); average occurrences: 1.015; average spacing: 7.461; total length: 3977333\n",
      "[M::worker_pipeline::0.604*0.99] mapped 1 sequences\n",
      "[M::main] Version: 2.19-r1057\n",
      "[M::main] CMD: minimap2 -ax asm20 --eqx -H -f 100 -r1k,10k --rmq=no -t 1 exp/multiple_runs11/tmp/L05-3/is_deleted/L05-3_genome.deleted_IS.0.fasta exp/multiple_runs11/tmp/L05-3/is_deleted/L05-3_genome.deleted_IS.1.fasta\n",
      "[M::main] Real time: 0.607 sec; CPU: 0.602 sec; Peak RSS: 0.087 GB\n",
      "local_variation - WARNING - Finished syri\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "invOut.txt  is empty. Skipping analysing it.\n",
      "TLOut.txt  is empty. Skipping analysing it.\n",
      "invTLOut.txt  is empty. Skipping analysing it.\n",
      "ctxOut.txt is empty. Skipping analysing it.\n",
      "inv Out.txt is empty. Skipping analysing it.\n",
      "TL Out.txt is empty. Skipping analysing it.\n",
      "invTL Out.txt is empty. Skipping analysing it.\n",
      "dup Out.txt is empty. Skipping analysing it.\n",
      "invDup Out.txt is empty. Skipping analysing it.\n",
      "ctxOut.txt is empty. Skipping analysing it.\n",
      "invOut.txt  is empty. Skipping analysing it.\n",
      "TLOut.txt  is empty. Skipping analysing it.\n",
      "invTLOut.txt  is empty. Skipping analysing it.\n",
      "ctxOut.txt is empty. Skipping analysing it.\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "[M::mm_idx_gen::0.058*1.01] collected minimizers\n",
      "[M::mm_idx_gen::0.077*1.00] sorted minimizers\n",
      "[M::main::0.077*1.00] loaded/built the index for 1 target sequence(s)\n",
      "[M::mm_mapopt_update::0.077*1.00] mid_occ = 100\n",
      "[M::mm_idx_stat] kmer size: 19; skip: 10; is_hpc: 1; #seq: 1\n",
      "[M::mm_idx_stat::0.080*1.00] distinct minimizers: 525478 (99.34% are singletons); average occurrences: 1.015; average spacing: 7.461; total length: 3977333\n",
      "[M::worker_pipeline::0.602*0.99] mapped 1 sequences\n",
      "[M::main] Version: 2.19-r1057\n",
      "[M::main] CMD: minimap2 -ax asm20 --eqx -H -f 100 -r1k,10k --rmq=no -t 1 exp/multiple_runs11/tmp/L05-3/is_deleted/L05-3_genome.deleted_IS.0.fasta exp/multiple_runs11/tmp/L05-3/is_deleted/L05-3_genome.deleted_IS.2.fasta\n",
      "[M::main] Real time: 0.606 sec; CPU: 0.600 sec; Peak RSS: 0.087 GB\n",
      "local_variation - WARNING - Finished syri\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "TLOut.txt  is empty. Skipping analysing it.\n",
      "invTLOut.txt  is empty. Skipping analysing it.\n",
      "ctxOut.txt is empty. Skipping analysing it.\n",
      "TL Out.txt is empty. Skipping analysing it.\n",
      "invTL Out.txt is empty. Skipping analysing it.\n",
      "dup Out.txt is empty. Skipping analysing it.\n",
      "invDup Out.txt is empty. Skipping analysing it.\n",
      "ctxOut.txt is empty. Skipping analysing it.\n",
      "TLOut.txt  is empty. Skipping analysing it.\n",
      "invTLOut.txt  is empty. Skipping analysing it.\n",
      "ctxOut.txt is empty. Skipping analysing it.\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "[M::mm_idx_gen::0.062*1.01] collected minimizers\n",
      "[M::mm_idx_gen::0.081*1.00] sorted minimizers\n",
      "[M::main::0.081*1.00] loaded/built the index for 1 target sequence(s)\n",
      "[M::mm_mapopt_update::0.081*1.00] mid_occ = 100\n",
      "[M::mm_idx_stat] kmer size: 19; skip: 10; is_hpc: 1; #seq: 1\n",
      "[M::mm_idx_stat::0.085*1.00] distinct minimizers: 525478 (99.34% are singletons); average occurrences: 1.015; average spacing: 7.461; total length: 3977333\n",
      "[M::worker_pipeline::0.583*0.99] mapped 1 sequences\n",
      "[M::main] Version: 2.19-r1057\n",
      "[M::main] CMD: minimap2 -ax asm20 --eqx -H -f 100 -r1k,10k --rmq=no -t 1 exp/multiple_runs11/tmp/L05-3/is_deleted/L05-3_genome.deleted_IS.0.fasta exp/multiple_runs11/tmp/L05-3/is_deleted/L05-3_genome.deleted_IS.3.fasta\n",
      "[M::main] Real time: 0.586 sec; CPU: 0.580 sec; Peak RSS: 0.089 GB\n",
      "local_variation - WARNING - Finished syri\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "TLOut.txt  is empty. Skipping analysing it.\n",
      "invTLOut.txt  is empty. Skipping analysing it.\n",
      "ctxOut.txt is empty. Skipping analysing it.\n",
      "TL Out.txt is empty. Skipping analysing it.\n",
      "invTL Out.txt is empty. Skipping analysing it.\n",
      "dup Out.txt is empty. Skipping analysing it.\n",
      "invDup Out.txt is empty. Skipping analysing it.\n",
      "ctxOut.txt is empty. Skipping analysing it.\n",
      "TLOut.txt  is empty. Skipping analysing it.\n",
      "invTLOut.txt  is empty. Skipping analysing it.\n",
      "ctxOut.txt is empty. Skipping analysing it.\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "[M::mm_idx_gen::0.057*1.01] collected minimizers\n",
      "[M::mm_idx_gen::0.075*1.00] sorted minimizers\n",
      "[M::main::0.075*1.00] loaded/built the index for 1 target sequence(s)\n",
      "[M::mm_mapopt_update::0.075*1.00] mid_occ = 100\n",
      "[M::mm_idx_stat] kmer size: 19; skip: 10; is_hpc: 1; #seq: 1\n",
      "[M::mm_idx_stat::0.079*1.00] distinct minimizers: 525478 (99.34% are singletons); average occurrences: 1.015; average spacing: 7.461; total length: 3977333\n",
      "[M::worker_pipeline::0.604*0.99] mapped 1 sequences\n",
      "[M::main] Version: 2.19-r1057\n",
      "[M::main] CMD: minimap2 -ax asm20 --eqx -H -f 100 -r1k,10k --rmq=no -t 1 exp/multiple_runs11/tmp/L05-4/is_deleted/L05-4_genome.deleted_IS.0.fasta exp/multiple_runs11/tmp/L05-4/is_deleted/L05-4_genome.deleted_IS.1.fasta\n",
      "[M::main] Real time: 0.607 sec; CPU: 0.601 sec; Peak RSS: 0.086 GB\n",
      "local_variation - WARNING - Finished syri\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "invOut.txt  is empty. Skipping analysing it.\n",
      "TLOut.txt  is empty. Skipping analysing it.\n",
      "invTLOut.txt  is empty. Skipping analysing it.\n",
      "ctxOut.txt is empty. Skipping analysing it.\n",
      "inv Out.txt is empty. Skipping analysing it.\n",
      "TL Out.txt is empty. Skipping analysing it.\n",
      "invTL Out.txt is empty. Skipping analysing it.\n",
      "dup Out.txt is empty. Skipping analysing it.\n",
      "invDup Out.txt is empty. Skipping analysing it.\n",
      "ctxOut.txt is empty. Skipping analysing it.\n",
      "invOut.txt  is empty. Skipping analysing it.\n",
      "TLOut.txt  is empty. Skipping analysing it.\n",
      "invTLOut.txt  is empty. Skipping analysing it.\n",
      "ctxOut.txt is empty. Skipping analysing it.\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "[M::mm_idx_gen::0.057*1.01] collected minimizers\n",
      "[M::mm_idx_gen::0.075*1.00] sorted minimizers\n",
      "[M::main::0.075*1.00] loaded/built the index for 1 target sequence(s)\n",
      "[M::mm_mapopt_update::0.075*1.00] mid_occ = 100\n",
      "[M::mm_idx_stat] kmer size: 19; skip: 10; is_hpc: 1; #seq: 1\n",
      "[M::mm_idx_stat::0.079*1.00] distinct minimizers: 525478 (99.34% are singletons); average occurrences: 1.015; average spacing: 7.461; total length: 3977333\n",
      "[M::worker_pipeline::0.610*0.99] mapped 1 sequences\n",
      "[M::main] Version: 2.19-r1057\n",
      "[M::main] CMD: minimap2 -ax asm20 --eqx -H -f 100 -r1k,10k --rmq=no -t 1 exp/multiple_runs11/tmp/L05-4/is_deleted/L05-4_genome.deleted_IS.0.fasta exp/multiple_runs11/tmp/L05-4/is_deleted/L05-4_genome.deleted_IS.2.fasta\n",
      "[M::main] Real time: 0.613 sec; CPU: 0.607 sec; Peak RSS: 0.088 GB\n",
      "Reading Coords - WARNING - Reference chromosome contig_1 has high fraction of inverted alignments with its homologous chromosome in the query genome (contig_1). Ensure that same chromosome-strands are being compared in the two genomes, as different strand can result in unexpected errors.\n",
      "local_variation - WARNING - Finished syri\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "TLOut.txt  is empty. Skipping analysing it.\n",
      "ctxOut.txt is empty. Skipping analysing it.\n",
      "TL Out.txt is empty. Skipping analysing it.\n",
      "invDup Out.txt is empty. Skipping analysing it.\n",
      "ctxOut.txt is empty. Skipping analysing it.\n",
      "TLOut.txt  is empty. Skipping analysing it.\n",
      "ctxOut.txt is empty. Skipping analysing it.\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "[M::mm_idx_gen::0.063*1.01] collected minimizers\n",
      "[M::mm_idx_gen::0.082*1.00] sorted minimizers\n",
      "[M::main::0.082*1.00] loaded/built the index for 1 target sequence(s)\n",
      "[M::mm_mapopt_update::0.082*1.00] mid_occ = 100\n",
      "[M::mm_idx_stat] kmer size: 19; skip: 10; is_hpc: 1; #seq: 1\n",
      "[M::mm_idx_stat::0.086*1.00] distinct minimizers: 525478 (99.34% are singletons); average occurrences: 1.015; average spacing: 7.461; total length: 3977333\n",
      "[M::worker_pipeline::0.605*0.99] mapped 1 sequences\n",
      "[M::main] Version: 2.19-r1057\n",
      "[M::main] CMD: minimap2 -ax asm20 --eqx -H -f 100 -r1k,10k --rmq=no -t 1 exp/multiple_runs11/tmp/L05-4/is_deleted/L05-4_genome.deleted_IS.0.fasta exp/multiple_runs11/tmp/L05-4/is_deleted/L05-4_genome.deleted_IS.3.fasta\n",
      "[M::main] Real time: 0.607 sec; CPU: 0.601 sec; Peak RSS: 0.088 GB\n",
      "local_variation - WARNING - Finished syri\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "TLOut.txt  is empty. Skipping analysing it.\n",
      "ctxOut.txt is empty. Skipping analysing it.\n",
      "TL Out.txt is empty. Skipping analysing it.\n",
      "invDup Out.txt is empty. Skipping analysing it.\n",
      "ctxOut.txt is empty. Skipping analysing it.\n",
      "TLOut.txt  is empty. Skipping analysing it.\n",
      "ctxOut.txt is empty. Skipping analysing it.\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "[M::mm_idx_gen::0.058*1.01] collected minimizers\n",
      "[M::mm_idx_gen::0.076*1.00] sorted minimizers\n",
      "[M::main::0.076*1.00] loaded/built the index for 1 target sequence(s)\n",
      "[M::mm_mapopt_update::0.076*1.00] mid_occ = 100\n",
      "[M::mm_idx_stat] kmer size: 19; skip: 10; is_hpc: 1; #seq: 1\n",
      "[M::mm_idx_stat::0.080*1.00] distinct minimizers: 525478 (99.34% are singletons); average occurrences: 1.015; average spacing: 7.461; total length: 3977333\n",
      "[M::worker_pipeline::0.617*0.99] mapped 1 sequences\n",
      "[M::main] Version: 2.19-r1057\n",
      "[M::main] CMD: minimap2 -ax asm20 --eqx -H -f 100 -r1k,10k --rmq=no -t 1 exp/multiple_runs11/tmp/L06-1/is_deleted/L06-1_genome.deleted_IS.0.fasta exp/multiple_runs11/tmp/L06-1/is_deleted/L06-1_genome.deleted_IS.1.fasta\n",
      "[M::main] Real time: 0.619 sec; CPU: 0.614 sec; Peak RSS: 0.089 GB\n",
      "local_variation - WARNING - Finished syri\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "invOut.txt  is empty. Skipping analysing it.\n",
      "TLOut.txt  is empty. Skipping analysing it.\n",
      "invTLOut.txt  is empty. Skipping analysing it.\n",
      "ctxOut.txt is empty. Skipping analysing it.\n",
      "inv Out.txt is empty. Skipping analysing it.\n",
      "TL Out.txt is empty. Skipping analysing it.\n",
      "invTL Out.txt is empty. Skipping analysing it.\n",
      "dup Out.txt is empty. Skipping analysing it.\n",
      "invDup Out.txt is empty. Skipping analysing it.\n",
      "ctxOut.txt is empty. Skipping analysing it.\n",
      "invOut.txt  is empty. Skipping analysing it.\n",
      "TLOut.txt  is empty. Skipping analysing it.\n",
      "invTLOut.txt  is empty. Skipping analysing it.\n",
      "ctxOut.txt is empty. Skipping analysing it.\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "[M::mm_idx_gen::0.059*1.01] collected minimizers\n",
      "[M::mm_idx_gen::0.078*1.00] sorted minimizers\n",
      "[M::main::0.078*1.00] loaded/built the index for 1 target sequence(s)\n",
      "[M::mm_mapopt_update::0.078*1.00] mid_occ = 100\n",
      "[M::mm_idx_stat] kmer size: 19; skip: 10; is_hpc: 1; #seq: 1\n",
      "[M::mm_idx_stat::0.082*1.00] distinct minimizers: 525478 (99.34% are singletons); average occurrences: 1.015; average spacing: 7.461; total length: 3977333\n",
      "[M::worker_pipeline::0.621*0.99] mapped 1 sequences\n",
      "[M::main] Version: 2.19-r1057\n",
      "[M::main] CMD: minimap2 -ax asm20 --eqx -H -f 100 -r1k,10k --rmq=no -t 1 exp/multiple_runs11/tmp/L06-1/is_deleted/L06-1_genome.deleted_IS.0.fasta exp/multiple_runs11/tmp/L06-1/is_deleted/L06-1_genome.deleted_IS.2.fasta\n",
      "[M::main] Real time: 0.624 sec; CPU: 0.617 sec; Peak RSS: 0.087 GB\n",
      "local_variation - WARNING - Finished syri\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "TLOut.txt  is empty. Skipping analysing it.\n",
      "invTLOut.txt  is empty. Skipping analysing it.\n",
      "ctxOut.txt is empty. Skipping analysing it.\n",
      "TL Out.txt is empty. Skipping analysing it.\n",
      "invTL Out.txt is empty. Skipping analysing it.\n",
      "dup Out.txt is empty. Skipping analysing it.\n",
      "invDup Out.txt is empty. Skipping analysing it.\n",
      "ctxOut.txt is empty. Skipping analysing it.\n",
      "TLOut.txt  is empty. Skipping analysing it.\n",
      "invTLOut.txt  is empty. Skipping analysing it.\n",
      "ctxOut.txt is empty. Skipping analysing it.\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "[M::mm_idx_gen::0.058*1.01] collected minimizers\n",
      "[M::mm_idx_gen::0.077*1.00] sorted minimizers\n",
      "[M::main::0.077*1.00] loaded/built the index for 1 target sequence(s)\n",
      "[M::mm_mapopt_update::0.077*1.00] mid_occ = 100\n",
      "[M::mm_idx_stat] kmer size: 19; skip: 10; is_hpc: 1; #seq: 1\n",
      "[M::mm_idx_stat::0.080*1.00] distinct minimizers: 525478 (99.34% are singletons); average occurrences: 1.015; average spacing: 7.461; total length: 3977333\n",
      "[M::worker_pipeline::0.615*0.99] mapped 1 sequences\n",
      "[M::main] Version: 2.19-r1057\n",
      "[M::main] CMD: minimap2 -ax asm20 --eqx -H -f 100 -r1k,10k --rmq=no -t 1 exp/multiple_runs11/tmp/L06-1/is_deleted/L06-1_genome.deleted_IS.0.fasta exp/multiple_runs11/tmp/L06-1/is_deleted/L06-1_genome.deleted_IS.3.fasta\n",
      "[M::main] Real time: 0.617 sec; CPU: 0.611 sec; Peak RSS: 0.088 GB\n",
      "local_variation - WARNING - Finished syri\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "invTLOut.txt  is empty. Skipping analysing it.\n",
      "ctxOut.txt is empty. Skipping analysing it.\n",
      "invTL Out.txt is empty. Skipping analysing it.\n",
      "dup Out.txt is empty. Skipping analysing it.\n",
      "invDup Out.txt is empty. Skipping analysing it.\n",
      "ctxOut.txt is empty. Skipping analysing it.\n",
      "invTLOut.txt  is empty. Skipping analysing it.\n",
      "ctxOut.txt is empty. Skipping analysing it.\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "[M::mm_idx_gen::0.064*1.01] collected minimizers\n",
      "[M::mm_idx_gen::0.083*1.00] sorted minimizers\n",
      "[M::main::0.083*1.00] loaded/built the index for 1 target sequence(s)\n",
      "[M::mm_mapopt_update::0.083*1.00] mid_occ = 100\n",
      "[M::mm_idx_stat] kmer size: 19; skip: 10; is_hpc: 1; #seq: 1\n",
      "[M::mm_idx_stat::0.086*1.00] distinct minimizers: 525478 (99.34% are singletons); average occurrences: 1.015; average spacing: 7.461; total length: 3977333\n",
      "[M::worker_pipeline::0.624*0.99] mapped 1 sequences\n",
      "[M::main] Version: 2.19-r1057\n",
      "[M::main] CMD: minimap2 -ax asm20 --eqx -H -f 100 -r1k,10k --rmq=no -t 1 exp/multiple_runs11/tmp/L06-2/is_deleted/L06-2_genome.deleted_IS.0.fasta exp/multiple_runs11/tmp/L06-2/is_deleted/L06-2_genome.deleted_IS.1.fasta\n",
      "[M::main] Real time: 0.626 sec; CPU: 0.621 sec; Peak RSS: 0.089 GB\n",
      "local_variation - WARNING - Finished syri\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "invOut.txt  is empty. Skipping analysing it.\n",
      "TLOut.txt  is empty. Skipping analysing it.\n",
      "invTLOut.txt  is empty. Skipping analysing it.\n",
      "ctxOut.txt is empty. Skipping analysing it.\n",
      "inv Out.txt is empty. Skipping analysing it.\n",
      "TL Out.txt is empty. Skipping analysing it.\n",
      "invTL Out.txt is empty. Skipping analysing it.\n",
      "dup Out.txt is empty. Skipping analysing it.\n",
      "invDup Out.txt is empty. Skipping analysing it.\n",
      "ctxOut.txt is empty. Skipping analysing it.\n",
      "invOut.txt  is empty. Skipping analysing it.\n",
      "TLOut.txt  is empty. Skipping analysing it.\n",
      "invTLOut.txt  is empty. Skipping analysing it.\n",
      "ctxOut.txt is empty. Skipping analysing it.\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "[M::mm_idx_gen::0.060*1.01] collected minimizers\n",
      "[M::mm_idx_gen::0.078*1.01] sorted minimizers\n",
      "[M::main::0.078*1.01] loaded/built the index for 1 target sequence(s)\n",
      "[M::mm_mapopt_update::0.078*1.01] mid_occ = 100\n",
      "[M::mm_idx_stat] kmer size: 19; skip: 10; is_hpc: 1; #seq: 1\n",
      "[M::mm_idx_stat::0.082*1.01] distinct minimizers: 525478 (99.34% are singletons); average occurrences: 1.015; average spacing: 7.461; total length: 3977333\n",
      "[M::worker_pipeline::0.702*0.99] mapped 1 sequences\n",
      "[M::main] Version: 2.19-r1057\n",
      "[M::main] CMD: minimap2 -ax asm20 --eqx -H -f 100 -r1k,10k --rmq=no -t 1 exp/multiple_runs11/tmp/L06-2/is_deleted/L06-2_genome.deleted_IS.0.fasta exp/multiple_runs11/tmp/L06-2/is_deleted/L06-2_genome.deleted_IS.2.fasta\n",
      "[M::main] Real time: 0.704 sec; CPU: 0.698 sec; Peak RSS: 0.203 GB\n",
      "local_variation - WARNING - Finished syri\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "TLOut.txt  is empty. Skipping analysing it.\n",
      "invTLOut.txt  is empty. Skipping analysing it.\n",
      "ctxOut.txt is empty. Skipping analysing it.\n",
      "TL Out.txt is empty. Skipping analysing it.\n",
      "invTL Out.txt is empty. Skipping analysing it.\n",
      "dup Out.txt is empty. Skipping analysing it.\n",
      "invDup Out.txt is empty. Skipping analysing it.\n",
      "ctxOut.txt is empty. Skipping analysing it.\n",
      "TLOut.txt  is empty. Skipping analysing it.\n",
      "invTLOut.txt  is empty. Skipping analysing it.\n",
      "ctxOut.txt is empty. Skipping analysing it.\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "[M::mm_idx_gen::0.058*1.01] collected minimizers\n",
      "[M::mm_idx_gen::0.077*1.00] sorted minimizers\n",
      "[M::main::0.077*1.00] loaded/built the index for 1 target sequence(s)\n",
      "[M::mm_mapopt_update::0.077*1.00] mid_occ = 100\n",
      "[M::mm_idx_stat] kmer size: 19; skip: 10; is_hpc: 1; #seq: 1\n",
      "[M::mm_idx_stat::0.080*1.00] distinct minimizers: 525478 (99.34% are singletons); average occurrences: 1.015; average spacing: 7.461; total length: 3977333\n",
      "[M::worker_pipeline::0.617*0.99] mapped 1 sequences\n",
      "[M::main] Version: 2.19-r1057\n",
      "[M::main] CMD: minimap2 -ax asm20 --eqx -H -f 100 -r1k,10k --rmq=no -t 1 exp/multiple_runs11/tmp/L06-2/is_deleted/L06-2_genome.deleted_IS.0.fasta exp/multiple_runs11/tmp/L06-2/is_deleted/L06-2_genome.deleted_IS.3.fasta\n",
      "[M::main] Real time: 0.620 sec; CPU: 0.613 sec; Peak RSS: 0.089 GB\n",
      "local_variation - WARNING - Finished syri\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "TLOut.txt  is empty. Skipping analysing it.\n",
      "ctxOut.txt is empty. Skipping analysing it.\n",
      "TL Out.txt is empty. Skipping analysing it.\n",
      "dup Out.txt is empty. Skipping analysing it.\n",
      "invDup Out.txt is empty. Skipping analysing it.\n",
      "ctxOut.txt is empty. Skipping analysing it.\n",
      "TLOut.txt  is empty. Skipping analysing it.\n",
      "ctxOut.txt is empty. Skipping analysing it.\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "[M::mm_idx_gen::0.058*1.01] collected minimizers\n",
      "[M::mm_idx_gen::0.077*1.00] sorted minimizers\n",
      "[M::main::0.077*1.00] loaded/built the index for 1 target sequence(s)\n",
      "[M::mm_mapopt_update::0.077*1.00] mid_occ = 100\n",
      "[M::mm_idx_stat] kmer size: 19; skip: 10; is_hpc: 1; #seq: 1\n",
      "[M::mm_idx_stat::0.081*1.00] distinct minimizers: 525478 (99.34% are singletons); average occurrences: 1.015; average spacing: 7.461; total length: 3977333\n",
      "[M::worker_pipeline::0.617*0.99] mapped 1 sequences\n",
      "[M::main] Version: 2.19-r1057\n",
      "[M::main] CMD: minimap2 -ax asm20 --eqx -H -f 100 -r1k,10k --rmq=no -t 1 exp/multiple_runs11/tmp/L06-3/is_deleted/L06-3_genome.deleted_IS.0.fasta exp/multiple_runs11/tmp/L06-3/is_deleted/L06-3_genome.deleted_IS.1.fasta\n",
      "[M::main] Real time: 0.620 sec; CPU: 0.614 sec; Peak RSS: 0.089 GB\n",
      "local_variation - WARNING - Finished syri\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "invOut.txt  is empty. Skipping analysing it.\n",
      "TLOut.txt  is empty. Skipping analysing it.\n",
      "invTLOut.txt  is empty. Skipping analysing it.\n",
      "ctxOut.txt is empty. Skipping analysing it.\n",
      "inv Out.txt is empty. Skipping analysing it.\n",
      "TL Out.txt is empty. Skipping analysing it.\n",
      "invTL Out.txt is empty. Skipping analysing it.\n",
      "dup Out.txt is empty. Skipping analysing it.\n",
      "invDup Out.txt is empty. Skipping analysing it.\n",
      "ctxOut.txt is empty. Skipping analysing it.\n",
      "invOut.txt  is empty. Skipping analysing it.\n",
      "TLOut.txt  is empty. Skipping analysing it.\n",
      "invTLOut.txt  is empty. Skipping analysing it.\n",
      "ctxOut.txt is empty. Skipping analysing it.\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "[M::mm_idx_gen::0.057*1.01] collected minimizers\n",
      "[M::mm_idx_gen::0.075*1.00] sorted minimizers\n",
      "[M::main::0.075*1.00] loaded/built the index for 1 target sequence(s)\n",
      "[M::mm_mapopt_update::0.075*1.00] mid_occ = 100\n",
      "[M::mm_idx_stat] kmer size: 19; skip: 10; is_hpc: 1; #seq: 1\n",
      "[M::mm_idx_stat::0.079*1.00] distinct minimizers: 525478 (99.34% are singletons); average occurrences: 1.015; average spacing: 7.461; total length: 3977333\n",
      "[M::worker_pipeline::0.604*0.99] mapped 1 sequences\n",
      "[M::main] Version: 2.19-r1057\n",
      "[M::main] CMD: minimap2 -ax asm20 --eqx -H -f 100 -r1k,10k --rmq=no -t 1 exp/multiple_runs11/tmp/L06-3/is_deleted/L06-3_genome.deleted_IS.0.fasta exp/multiple_runs11/tmp/L06-3/is_deleted/L06-3_genome.deleted_IS.2.fasta\n",
      "[M::main] Real time: 0.607 sec; CPU: 0.601 sec; Peak RSS: 0.087 GB\n",
      "Reading Coords - WARNING - Reference chromosome contig_1 has high fraction of inverted alignments with its homologous chromosome in the query genome (contig_1). Ensure that same chromosome-strands are being compared in the two genomes, as different strand can result in unexpected errors.\n",
      "local_variation - WARNING - Finished syri\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "TLOut.txt  is empty. Skipping analysing it.\n",
      "invTLOut.txt  is empty. Skipping analysing it.\n",
      "ctxOut.txt is empty. Skipping analysing it.\n",
      "TL Out.txt is empty. Skipping analysing it.\n",
      "invTL Out.txt is empty. Skipping analysing it.\n",
      "dup Out.txt is empty. Skipping analysing it.\n",
      "invDup Out.txt is empty. Skipping analysing it.\n",
      "ctxOut.txt is empty. Skipping analysing it.\n",
      "TLOut.txt  is empty. Skipping analysing it.\n",
      "invTLOut.txt  is empty. Skipping analysing it.\n",
      "ctxOut.txt is empty. Skipping analysing it.\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "[M::mm_idx_gen::0.057*1.01] collected minimizers\n",
      "[M::mm_idx_gen::0.075*1.00] sorted minimizers\n",
      "[M::main::0.075*1.00] loaded/built the index for 1 target sequence(s)\n",
      "[M::mm_mapopt_update::0.075*1.00] mid_occ = 100\n",
      "[M::mm_idx_stat] kmer size: 19; skip: 10; is_hpc: 1; #seq: 1\n",
      "[M::mm_idx_stat::0.079*1.00] distinct minimizers: 525478 (99.34% are singletons); average occurrences: 1.015; average spacing: 7.461; total length: 3977333\n",
      "[M::worker_pipeline::0.631*0.99] mapped 1 sequences\n",
      "[M::main] Version: 2.19-r1057\n",
      "[M::main] CMD: minimap2 -ax asm20 --eqx -H -f 100 -r1k,10k --rmq=no -t 1 exp/multiple_runs11/tmp/L06-3/is_deleted/L06-3_genome.deleted_IS.0.fasta exp/multiple_runs11/tmp/L06-3/is_deleted/L06-3_genome.deleted_IS.3.fasta\n",
      "[M::main] Real time: 0.634 sec; CPU: 0.627 sec; Peak RSS: 0.089 GB\n",
      "Reading Coords - WARNING - Reference chromosome contig_1 has high fraction of inverted alignments with its homologous chromosome in the query genome (contig_1). Ensure that same chromosome-strands are being compared in the two genomes, as different strand can result in unexpected errors.\n",
      "local_variation - WARNING - Finished syri\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "TLOut.txt  is empty. Skipping analysing it.\n",
      "invTLOut.txt  is empty. Skipping analysing it.\n",
      "ctxOut.txt is empty. Skipping analysing it.\n",
      "TL Out.txt is empty. Skipping analysing it.\n",
      "invTL Out.txt is empty. Skipping analysing it.\n",
      "dup Out.txt is empty. Skipping analysing it.\n",
      "invDup Out.txt is empty. Skipping analysing it.\n",
      "ctxOut.txt is empty. Skipping analysing it.\n",
      "TLOut.txt  is empty. Skipping analysing it.\n",
      "invTLOut.txt  is empty. Skipping analysing it.\n",
      "ctxOut.txt is empty. Skipping analysing it.\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "[M::mm_idx_gen::0.057*1.00] collected minimizers\n",
      "[M::mm_idx_gen::0.076*1.00] sorted minimizers\n",
      "[M::main::0.076*1.00] loaded/built the index for 1 target sequence(s)\n",
      "[M::mm_mapopt_update::0.076*1.00] mid_occ = 100\n",
      "[M::mm_idx_stat] kmer size: 19; skip: 10; is_hpc: 1; #seq: 1\n",
      "[M::mm_idx_stat::0.079*1.00] distinct minimizers: 525478 (99.34% are singletons); average occurrences: 1.015; average spacing: 7.461; total length: 3977333\n",
      "[M::worker_pipeline::0.607*0.99] mapped 1 sequences\n",
      "[M::main] Version: 2.19-r1057\n",
      "[M::main] CMD: minimap2 -ax asm20 --eqx -H -f 100 -r1k,10k --rmq=no -t 1 exp/multiple_runs11/tmp/L06-4/is_deleted/L06-4_genome.deleted_IS.0.fasta exp/multiple_runs11/tmp/L06-4/is_deleted/L06-4_genome.deleted_IS.1.fasta\n",
      "[M::main] Real time: 0.609 sec; CPU: 0.604 sec; Peak RSS: 0.089 GB\n",
      "local_variation - WARNING - Finished syri\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "invOut.txt  is empty. Skipping analysing it.\n",
      "TLOut.txt  is empty. Skipping analysing it.\n",
      "invTLOut.txt  is empty. Skipping analysing it.\n",
      "ctxOut.txt is empty. Skipping analysing it.\n",
      "inv Out.txt is empty. Skipping analysing it.\n",
      "TL Out.txt is empty. Skipping analysing it.\n",
      "invTL Out.txt is empty. Skipping analysing it.\n",
      "dup Out.txt is empty. Skipping analysing it.\n",
      "invDup Out.txt is empty. Skipping analysing it.\n",
      "ctxOut.txt is empty. Skipping analysing it.\n",
      "invOut.txt  is empty. Skipping analysing it.\n",
      "TLOut.txt  is empty. Skipping analysing it.\n",
      "invTLOut.txt  is empty. Skipping analysing it.\n",
      "ctxOut.txt is empty. Skipping analysing it.\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "[M::mm_idx_gen::0.057*1.01] collected minimizers\n",
      "[M::mm_idx_gen::0.075*1.00] sorted minimizers\n",
      "[M::main::0.075*1.00] loaded/built the index for 1 target sequence(s)\n",
      "[M::mm_mapopt_update::0.075*1.00] mid_occ = 100\n",
      "[M::mm_idx_stat] kmer size: 19; skip: 10; is_hpc: 1; #seq: 1\n",
      "[M::mm_idx_stat::0.079*1.00] distinct minimizers: 525478 (99.34% are singletons); average occurrences: 1.015; average spacing: 7.461; total length: 3977333\n",
      "[M::worker_pipeline::0.604*0.99] mapped 1 sequences\n",
      "[M::main] Version: 2.19-r1057\n",
      "[M::main] CMD: minimap2 -ax asm20 --eqx -H -f 100 -r1k,10k --rmq=no -t 1 exp/multiple_runs11/tmp/L06-4/is_deleted/L06-4_genome.deleted_IS.0.fasta exp/multiple_runs11/tmp/L06-4/is_deleted/L06-4_genome.deleted_IS.2.fasta\n",
      "[M::main] Real time: 0.607 sec; CPU: 0.600 sec; Peak RSS: 0.088 GB\n",
      "local_variation - WARNING - Finished syri\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "TLOut.txt  is empty. Skipping analysing it.\n",
      "invTLOut.txt  is empty. Skipping analysing it.\n",
      "ctxOut.txt is empty. Skipping analysing it.\n",
      "TL Out.txt is empty. Skipping analysing it.\n",
      "invTL Out.txt is empty. Skipping analysing it.\n",
      "dup Out.txt is empty. Skipping analysing it.\n",
      "invDup Out.txt is empty. Skipping analysing it.\n",
      "ctxOut.txt is empty. Skipping analysing it.\n",
      "TLOut.txt  is empty. Skipping analysing it.\n",
      "invTLOut.txt  is empty. Skipping analysing it.\n",
      "ctxOut.txt is empty. Skipping analysing it.\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "[M::mm_idx_gen::0.058*1.01] collected minimizers\n",
      "[M::mm_idx_gen::0.077*1.01] sorted minimizers\n",
      "[M::main::0.077*1.01] loaded/built the index for 1 target sequence(s)\n",
      "[M::mm_mapopt_update::0.077*1.01] mid_occ = 100\n",
      "[M::mm_idx_stat] kmer size: 19; skip: 10; is_hpc: 1; #seq: 1\n",
      "[M::mm_idx_stat::0.081*1.01] distinct minimizers: 525478 (99.34% are singletons); average occurrences: 1.015; average spacing: 7.461; total length: 3977333\n",
      "[M::worker_pipeline::0.597*0.99] mapped 1 sequences\n",
      "[M::main] Version: 2.19-r1057\n",
      "[M::main] CMD: minimap2 -ax asm20 --eqx -H -f 100 -r1k,10k --rmq=no -t 1 exp/multiple_runs11/tmp/L06-4/is_deleted/L06-4_genome.deleted_IS.0.fasta exp/multiple_runs11/tmp/L06-4/is_deleted/L06-4_genome.deleted_IS.3.fasta\n",
      "[M::main] Real time: 0.600 sec; CPU: 0.594 sec; Peak RSS: 0.086 GB\n",
      "local_variation - WARNING - Finished syri\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "TLOut.txt  is empty. Skipping analysing it.\n",
      "invTLOut.txt  is empty. Skipping analysing it.\n",
      "ctxOut.txt is empty. Skipping analysing it.\n",
      "TL Out.txt is empty. Skipping analysing it.\n",
      "invTL Out.txt is empty. Skipping analysing it.\n",
      "dup Out.txt is empty. Skipping analysing it.\n",
      "invDup Out.txt is empty. Skipping analysing it.\n",
      "ctxOut.txt is empty. Skipping analysing it.\n",
      "TLOut.txt  is empty. Skipping analysing it.\n",
      "invTLOut.txt  is empty. Skipping analysing it.\n",
      "ctxOut.txt is empty. Skipping analysing it.\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "[M::mm_idx_gen::0.058*1.01] collected minimizers\n",
      "[M::mm_idx_gen::0.076*1.00] sorted minimizers\n",
      "[M::main::0.077*1.00] loaded/built the index for 1 target sequence(s)\n",
      "[M::mm_mapopt_update::0.077*1.00] mid_occ = 100\n",
      "[M::mm_idx_stat] kmer size: 19; skip: 10; is_hpc: 1; #seq: 1\n",
      "[M::mm_idx_stat::0.080*1.00] distinct minimizers: 525478 (99.34% are singletons); average occurrences: 1.015; average spacing: 7.461; total length: 3977333\n",
      "[M::worker_pipeline::0.642*0.99] mapped 1 sequences\n",
      "[M::main] Version: 2.19-r1057\n",
      "[M::main] CMD: minimap2 -ax asm20 --eqx -H -f 100 -r1k,10k --rmq=no -t 1 exp/multiple_runs11/tmp/L07-1/is_deleted/L07-1_genome.deleted_IS.0.fasta exp/multiple_runs11/tmp/L07-1/is_deleted/L07-1_genome.deleted_IS.1.fasta\n",
      "[M::main] Real time: 0.645 sec; CPU: 0.638 sec; Peak RSS: 0.133 GB\n",
      "local_variation - WARNING - Finished syri\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "invOut.txt  is empty. Skipping analysing it.\n",
      "TLOut.txt  is empty. Skipping analysing it.\n",
      "invTLOut.txt  is empty. Skipping analysing it.\n",
      "ctxOut.txt is empty. Skipping analysing it.\n",
      "inv Out.txt is empty. Skipping analysing it.\n",
      "TL Out.txt is empty. Skipping analysing it.\n",
      "invTL Out.txt is empty. Skipping analysing it.\n",
      "dup Out.txt is empty. Skipping analysing it.\n",
      "invDup Out.txt is empty. Skipping analysing it.\n",
      "ctxOut.txt is empty. Skipping analysing it.\n",
      "invOut.txt  is empty. Skipping analysing it.\n",
      "TLOut.txt  is empty. Skipping analysing it.\n",
      "invTLOut.txt  is empty. Skipping analysing it.\n",
      "ctxOut.txt is empty. Skipping analysing it.\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "[M::mm_idx_gen::0.058*1.01] collected minimizers\n",
      "[M::mm_idx_gen::0.077*1.00] sorted minimizers\n",
      "[M::main::0.077*1.00] loaded/built the index for 1 target sequence(s)\n",
      "[M::mm_mapopt_update::0.077*1.00] mid_occ = 100\n",
      "[M::mm_idx_stat] kmer size: 19; skip: 10; is_hpc: 1; #seq: 1\n",
      "[M::mm_idx_stat::0.081*1.00] distinct minimizers: 525478 (99.34% are singletons); average occurrences: 1.015; average spacing: 7.461; total length: 3977333\n",
      "[M::worker_pipeline::0.653*0.99] mapped 1 sequences\n",
      "[M::main] Version: 2.19-r1057\n",
      "[M::main] CMD: minimap2 -ax asm20 --eqx -H -f 100 -r1k,10k --rmq=no -t 1 exp/multiple_runs11/tmp/L07-1/is_deleted/L07-1_genome.deleted_IS.0.fasta exp/multiple_runs11/tmp/L07-1/is_deleted/L07-1_genome.deleted_IS.2.fasta\n",
      "[M::main] Real time: 0.655 sec; CPU: 0.648 sec; Peak RSS: 0.132 GB\n",
      "local_variation - WARNING - Finished syri\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "invOut.txt  is empty. Skipping analysing it.\n",
      "TLOut.txt  is empty. Skipping analysing it.\n",
      "invTLOut.txt  is empty. Skipping analysing it.\n",
      "ctxOut.txt is empty. Skipping analysing it.\n",
      "inv Out.txt is empty. Skipping analysing it.\n",
      "TL Out.txt is empty. Skipping analysing it.\n",
      "invTL Out.txt is empty. Skipping analysing it.\n",
      "dup Out.txt is empty. Skipping analysing it.\n",
      "invDup Out.txt is empty. Skipping analysing it.\n",
      "ctxOut.txt is empty. Skipping analysing it.\n",
      "invOut.txt  is empty. Skipping analysing it.\n",
      "TLOut.txt  is empty. Skipping analysing it.\n",
      "invTLOut.txt  is empty. Skipping analysing it.\n",
      "ctxOut.txt is empty. Skipping analysing it.\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "[M::mm_idx_gen::0.057*1.01] collected minimizers\n",
      "[M::mm_idx_gen::0.076*1.00] sorted minimizers\n",
      "[M::main::0.076*1.00] loaded/built the index for 1 target sequence(s)\n",
      "[M::mm_mapopt_update::0.076*1.00] mid_occ = 100\n",
      "[M::mm_idx_stat] kmer size: 19; skip: 10; is_hpc: 1; #seq: 1\n",
      "[M::mm_idx_stat::0.080*1.00] distinct minimizers: 525478 (99.34% are singletons); average occurrences: 1.015; average spacing: 7.461; total length: 3977333\n",
      "[M::worker_pipeline::0.642*0.99] mapped 1 sequences\n",
      "[M::main] Version: 2.19-r1057\n",
      "[M::main] CMD: minimap2 -ax asm20 --eqx -H -f 100 -r1k,10k --rmq=no -t 1 exp/multiple_runs11/tmp/L07-1/is_deleted/L07-1_genome.deleted_IS.0.fasta exp/multiple_runs11/tmp/L07-1/is_deleted/L07-1_genome.deleted_IS.3.fasta\n",
      "[M::main] Real time: 0.644 sec; CPU: 0.638 sec; Peak RSS: 0.134 GB\n",
      "local_variation - WARNING - Finished syri\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "TLOut.txt  is empty. Skipping analysing it.\n",
      "invTLOut.txt  is empty. Skipping analysing it.\n",
      "ctxOut.txt is empty. Skipping analysing it.\n",
      "TL Out.txt is empty. Skipping analysing it.\n",
      "invTL Out.txt is empty. Skipping analysing it.\n",
      "dup Out.txt is empty. Skipping analysing it.\n",
      "invDup Out.txt is empty. Skipping analysing it.\n",
      "ctxOut.txt is empty. Skipping analysing it.\n",
      "TLOut.txt  is empty. Skipping analysing it.\n",
      "invTLOut.txt  is empty. Skipping analysing it.\n",
      "ctxOut.txt is empty. Skipping analysing it.\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "[M::mm_idx_gen::0.058*1.01] collected minimizers\n",
      "[M::mm_idx_gen::0.077*1.01] sorted minimizers\n",
      "[M::main::0.077*1.01] loaded/built the index for 1 target sequence(s)\n",
      "[M::mm_mapopt_update::0.077*1.01] mid_occ = 100\n",
      "[M::mm_idx_stat] kmer size: 19; skip: 10; is_hpc: 1; #seq: 1\n",
      "[M::mm_idx_stat::0.080*1.00] distinct minimizers: 525478 (99.34% are singletons); average occurrences: 1.015; average spacing: 7.461; total length: 3977333\n",
      "[M::worker_pipeline::0.655*0.99] mapped 1 sequences\n",
      "[M::main] Version: 2.19-r1057\n",
      "[M::main] CMD: minimap2 -ax asm20 --eqx -H -f 100 -r1k,10k --rmq=no -t 1 exp/multiple_runs11/tmp/L07-2/is_deleted/L07-2_genome.deleted_IS.0.fasta exp/multiple_runs11/tmp/L07-2/is_deleted/L07-2_genome.deleted_IS.1.fasta\n",
      "[M::main] Real time: 0.657 sec; CPU: 0.651 sec; Peak RSS: 0.134 GB\n",
      "local_variation - WARNING - Finished syri\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "invOut.txt  is empty. Skipping analysing it.\n",
      "TLOut.txt  is empty. Skipping analysing it.\n",
      "invTLOut.txt  is empty. Skipping analysing it.\n",
      "ctxOut.txt is empty. Skipping analysing it.\n",
      "inv Out.txt is empty. Skipping analysing it.\n",
      "TL Out.txt is empty. Skipping analysing it.\n",
      "invTL Out.txt is empty. Skipping analysing it.\n",
      "dup Out.txt is empty. Skipping analysing it.\n",
      "invDup Out.txt is empty. Skipping analysing it.\n",
      "ctxOut.txt is empty. Skipping analysing it.\n",
      "invOut.txt  is empty. Skipping analysing it.\n",
      "TLOut.txt  is empty. Skipping analysing it.\n",
      "invTLOut.txt  is empty. Skipping analysing it.\n",
      "ctxOut.txt is empty. Skipping analysing it.\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "[M::mm_idx_gen::0.059*1.01] collected minimizers\n",
      "[M::mm_idx_gen::0.077*1.01] sorted minimizers\n",
      "[M::main::0.077*1.01] loaded/built the index for 1 target sequence(s)\n",
      "[M::mm_mapopt_update::0.077*1.01] mid_occ = 100\n",
      "[M::mm_idx_stat] kmer size: 19; skip: 10; is_hpc: 1; #seq: 1\n",
      "[M::mm_idx_stat::0.081*1.01] distinct minimizers: 525478 (99.34% are singletons); average occurrences: 1.015; average spacing: 7.461; total length: 3977333\n",
      "[M::worker_pipeline::0.641*0.99] mapped 1 sequences\n",
      "[M::main] Version: 2.19-r1057\n",
      "[M::main] CMD: minimap2 -ax asm20 --eqx -H -f 100 -r1k,10k --rmq=no -t 1 exp/multiple_runs11/tmp/L07-2/is_deleted/L07-2_genome.deleted_IS.0.fasta exp/multiple_runs11/tmp/L07-2/is_deleted/L07-2_genome.deleted_IS.2.fasta\n",
      "[M::main] Real time: 0.644 sec; CPU: 0.637 sec; Peak RSS: 0.133 GB\n",
      "local_variation - WARNING - Finished syri\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "invOut.txt  is empty. Skipping analysing it.\n",
      "TLOut.txt  is empty. Skipping analysing it.\n",
      "invTLOut.txt  is empty. Skipping analysing it.\n",
      "ctxOut.txt is empty. Skipping analysing it.\n",
      "inv Out.txt is empty. Skipping analysing it.\n",
      "TL Out.txt is empty. Skipping analysing it.\n",
      "invTL Out.txt is empty. Skipping analysing it.\n",
      "dup Out.txt is empty. Skipping analysing it.\n",
      "invDup Out.txt is empty. Skipping analysing it.\n",
      "ctxOut.txt is empty. Skipping analysing it.\n",
      "invOut.txt  is empty. Skipping analysing it.\n",
      "TLOut.txt  is empty. Skipping analysing it.\n",
      "invTLOut.txt  is empty. Skipping analysing it.\n",
      "ctxOut.txt is empty. Skipping analysing it.\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "[M::mm_idx_gen::0.057*1.01] collected minimizers\n",
      "[M::mm_idx_gen::0.076*1.00] sorted minimizers\n",
      "[M::main::0.076*1.00] loaded/built the index for 1 target sequence(s)\n",
      "[M::mm_mapopt_update::0.076*1.00] mid_occ = 100\n",
      "[M::mm_idx_stat] kmer size: 19; skip: 10; is_hpc: 1; #seq: 1\n",
      "[M::mm_idx_stat::0.080*1.00] distinct minimizers: 525478 (99.34% are singletons); average occurrences: 1.015; average spacing: 7.461; total length: 3977333\n",
      "[M::worker_pipeline::0.639*0.99] mapped 1 sequences\n",
      "[M::main] Version: 2.19-r1057\n",
      "[M::main] CMD: minimap2 -ax asm20 --eqx -H -f 100 -r1k,10k --rmq=no -t 1 exp/multiple_runs11/tmp/L07-2/is_deleted/L07-2_genome.deleted_IS.0.fasta exp/multiple_runs11/tmp/L07-2/is_deleted/L07-2_genome.deleted_IS.3.fasta\n",
      "[M::main] Real time: 0.642 sec; CPU: 0.635 sec; Peak RSS: 0.134 GB\n",
      "local_variation - WARNING - Finished syri\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "invOut.txt  is empty. Skipping analysing it.\n",
      "TLOut.txt  is empty. Skipping analysing it.\n",
      "invTLOut.txt  is empty. Skipping analysing it.\n",
      "ctxOut.txt is empty. Skipping analysing it.\n",
      "inv Out.txt is empty. Skipping analysing it.\n",
      "TL Out.txt is empty. Skipping analysing it.\n",
      "invTL Out.txt is empty. Skipping analysing it.\n",
      "dup Out.txt is empty. Skipping analysing it.\n",
      "invDup Out.txt is empty. Skipping analysing it.\n",
      "ctxOut.txt is empty. Skipping analysing it.\n",
      "invOut.txt  is empty. Skipping analysing it.\n",
      "TLOut.txt  is empty. Skipping analysing it.\n",
      "invTLOut.txt  is empty. Skipping analysing it.\n",
      "ctxOut.txt is empty. Skipping analysing it.\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "[M::mm_idx_gen::0.065*1.01] collected minimizers\n",
      "[M::mm_idx_gen::0.083*1.00] sorted minimizers\n",
      "[M::main::0.083*1.00] loaded/built the index for 1 target sequence(s)\n",
      "[M::mm_mapopt_update::0.083*1.00] mid_occ = 100\n",
      "[M::mm_idx_stat] kmer size: 19; skip: 10; is_hpc: 1; #seq: 1\n",
      "[M::mm_idx_stat::0.087*1.00] distinct minimizers: 525478 (99.34% are singletons); average occurrences: 1.015; average spacing: 7.461; total length: 3977333\n",
      "[M::worker_pipeline::0.649*0.99] mapped 1 sequences\n",
      "[M::main] Version: 2.19-r1057\n",
      "[M::main] CMD: minimap2 -ax asm20 --eqx -H -f 100 -r1k,10k --rmq=no -t 1 exp/multiple_runs11/tmp/L07-3/is_deleted/L07-3_genome.deleted_IS.0.fasta exp/multiple_runs11/tmp/L07-3/is_deleted/L07-3_genome.deleted_IS.1.fasta\n",
      "[M::main] Real time: 0.652 sec; CPU: 0.646 sec; Peak RSS: 0.134 GB\n",
      "local_variation - WARNING - Finished syri\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "invOut.txt  is empty. Skipping analysing it.\n",
      "TLOut.txt  is empty. Skipping analysing it.\n",
      "invTLOut.txt  is empty. Skipping analysing it.\n",
      "ctxOut.txt is empty. Skipping analysing it.\n",
      "inv Out.txt is empty. Skipping analysing it.\n",
      "TL Out.txt is empty. Skipping analysing it.\n",
      "invTL Out.txt is empty. Skipping analysing it.\n",
      "dup Out.txt is empty. Skipping analysing it.\n",
      "invDup Out.txt is empty. Skipping analysing it.\n",
      "ctxOut.txt is empty. Skipping analysing it.\n",
      "invOut.txt  is empty. Skipping analysing it.\n",
      "TLOut.txt  is empty. Skipping analysing it.\n",
      "invTLOut.txt  is empty. Skipping analysing it.\n",
      "ctxOut.txt is empty. Skipping analysing it.\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "[M::mm_idx_gen::0.061*1.00] collected minimizers\n",
      "[M::mm_idx_gen::0.080*1.00] sorted minimizers\n",
      "[M::main::0.080*1.00] loaded/built the index for 1 target sequence(s)\n",
      "[M::mm_mapopt_update::0.080*1.00] mid_occ = 100\n",
      "[M::mm_idx_stat] kmer size: 19; skip: 10; is_hpc: 1; #seq: 1\n",
      "[M::mm_idx_stat::0.084*1.00] distinct minimizers: 525478 (99.34% are singletons); average occurrences: 1.015; average spacing: 7.461; total length: 3977333\n",
      "[M::worker_pipeline::0.642*0.99] mapped 1 sequences\n",
      "[M::main] Version: 2.19-r1057\n",
      "[M::main] CMD: minimap2 -ax asm20 --eqx -H -f 100 -r1k,10k --rmq=no -t 1 exp/multiple_runs11/tmp/L07-3/is_deleted/L07-3_genome.deleted_IS.0.fasta exp/multiple_runs11/tmp/L07-3/is_deleted/L07-3_genome.deleted_IS.2.fasta\n",
      "[M::main] Real time: 0.644 sec; CPU: 0.637 sec; Peak RSS: 0.133 GB\n",
      "local_variation - WARNING - Finished syri\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "TLOut.txt  is empty. Skipping analysing it.\n",
      "invTLOut.txt  is empty. Skipping analysing it.\n",
      "ctxOut.txt is empty. Skipping analysing it.\n",
      "TL Out.txt is empty. Skipping analysing it.\n",
      "invTL Out.txt is empty. Skipping analysing it.\n",
      "dup Out.txt is empty. Skipping analysing it.\n",
      "invDup Out.txt is empty. Skipping analysing it.\n",
      "ctxOut.txt is empty. Skipping analysing it.\n",
      "TLOut.txt  is empty. Skipping analysing it.\n",
      "invTLOut.txt  is empty. Skipping analysing it.\n",
      "ctxOut.txt is empty. Skipping analysing it.\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "[M::mm_idx_gen::0.061*1.01] collected minimizers\n",
      "[M::mm_idx_gen::0.079*1.00] sorted minimizers\n",
      "[M::main::0.079*1.00] loaded/built the index for 1 target sequence(s)\n",
      "[M::mm_mapopt_update::0.079*1.00] mid_occ = 100\n",
      "[M::mm_idx_stat] kmer size: 19; skip: 10; is_hpc: 1; #seq: 1\n",
      "[M::mm_idx_stat::0.083*1.00] distinct minimizers: 525478 (99.34% are singletons); average occurrences: 1.015; average spacing: 7.461; total length: 3977333\n",
      "[M::worker_pipeline::0.643*0.99] mapped 1 sequences\n",
      "[M::main] Version: 2.19-r1057\n",
      "[M::main] CMD: minimap2 -ax asm20 --eqx -H -f 100 -r1k,10k --rmq=no -t 1 exp/multiple_runs11/tmp/L07-3/is_deleted/L07-3_genome.deleted_IS.0.fasta exp/multiple_runs11/tmp/L07-3/is_deleted/L07-3_genome.deleted_IS.3.fasta\n",
      "[M::main] Real time: 0.645 sec; CPU: 0.639 sec; Peak RSS: 0.133 GB\n",
      "local_variation - WARNING - Finished syri\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "TLOut.txt  is empty. Skipping analysing it.\n",
      "invTLOut.txt  is empty. Skipping analysing it.\n",
      "ctxOut.txt is empty. Skipping analysing it.\n",
      "TL Out.txt is empty. Skipping analysing it.\n",
      "invTL Out.txt is empty. Skipping analysing it.\n",
      "dup Out.txt is empty. Skipping analysing it.\n",
      "invDup Out.txt is empty. Skipping analysing it.\n",
      "ctxOut.txt is empty. Skipping analysing it.\n",
      "TLOut.txt  is empty. Skipping analysing it.\n",
      "invTLOut.txt  is empty. Skipping analysing it.\n",
      "ctxOut.txt is empty. Skipping analysing it.\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "[M::mm_idx_gen::0.062*1.01] collected minimizers\n",
      "[M::mm_idx_gen::0.081*1.00] sorted minimizers\n",
      "[M::main::0.081*1.00] loaded/built the index for 1 target sequence(s)\n",
      "[M::mm_mapopt_update::0.081*1.00] mid_occ = 100\n",
      "[M::mm_idx_stat] kmer size: 19; skip: 10; is_hpc: 1; #seq: 1\n",
      "[M::mm_idx_stat::0.085*1.00] distinct minimizers: 525478 (99.34% are singletons); average occurrences: 1.015; average spacing: 7.461; total length: 3977333\n",
      "[M::worker_pipeline::0.647*0.99] mapped 1 sequences\n",
      "[M::main] Version: 2.19-r1057\n",
      "[M::main] CMD: minimap2 -ax asm20 --eqx -H -f 100 -r1k,10k --rmq=no -t 1 exp/multiple_runs11/tmp/L07-4/is_deleted/L07-4_genome.deleted_IS.0.fasta exp/multiple_runs11/tmp/L07-4/is_deleted/L07-4_genome.deleted_IS.1.fasta\n",
      "[M::main] Real time: 0.650 sec; CPU: 0.645 sec; Peak RSS: 0.132 GB\n",
      "local_variation - WARNING - Finished syri\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "invOut.txt  is empty. Skipping analysing it.\n",
      "TLOut.txt  is empty. Skipping analysing it.\n",
      "invTLOut.txt  is empty. Skipping analysing it.\n",
      "ctxOut.txt is empty. Skipping analysing it.\n",
      "inv Out.txt is empty. Skipping analysing it.\n",
      "TL Out.txt is empty. Skipping analysing it.\n",
      "invTL Out.txt is empty. Skipping analysing it.\n",
      "dup Out.txt is empty. Skipping analysing it.\n",
      "invDup Out.txt is empty. Skipping analysing it.\n",
      "ctxOut.txt is empty. Skipping analysing it.\n",
      "invOut.txt  is empty. Skipping analysing it.\n",
      "TLOut.txt  is empty. Skipping analysing it.\n",
      "invTLOut.txt  is empty. Skipping analysing it.\n",
      "ctxOut.txt is empty. Skipping analysing it.\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "[M::mm_idx_gen::0.062*1.01] collected minimizers\n",
      "[M::mm_idx_gen::0.081*1.00] sorted minimizers\n",
      "[M::main::0.081*1.00] loaded/built the index for 1 target sequence(s)\n",
      "[M::mm_mapopt_update::0.081*1.00] mid_occ = 100\n",
      "[M::mm_idx_stat] kmer size: 19; skip: 10; is_hpc: 1; #seq: 1\n",
      "[M::mm_idx_stat::0.085*1.00] distinct minimizers: 525478 (99.34% are singletons); average occurrences: 1.015; average spacing: 7.461; total length: 3977333\n",
      "[M::worker_pipeline::0.649*0.99] mapped 1 sequences\n",
      "[M::main] Version: 2.19-r1057\n",
      "[M::main] CMD: minimap2 -ax asm20 --eqx -H -f 100 -r1k,10k --rmq=no -t 1 exp/multiple_runs11/tmp/L07-4/is_deleted/L07-4_genome.deleted_IS.0.fasta exp/multiple_runs11/tmp/L07-4/is_deleted/L07-4_genome.deleted_IS.2.fasta\n",
      "[M::main] Real time: 0.652 sec; CPU: 0.645 sec; Peak RSS: 0.133 GB\n",
      "local_variation - WARNING - Finished syri\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "invOut.txt  is empty. Skipping analysing it.\n",
      "TLOut.txt  is empty. Skipping analysing it.\n",
      "invTLOut.txt  is empty. Skipping analysing it.\n",
      "ctxOut.txt is empty. Skipping analysing it.\n",
      "inv Out.txt is empty. Skipping analysing it.\n",
      "TL Out.txt is empty. Skipping analysing it.\n",
      "invTL Out.txt is empty. Skipping analysing it.\n",
      "dup Out.txt is empty. Skipping analysing it.\n",
      "invDup Out.txt is empty. Skipping analysing it.\n",
      "ctxOut.txt is empty. Skipping analysing it.\n",
      "invOut.txt  is empty. Skipping analysing it.\n",
      "TLOut.txt  is empty. Skipping analysing it.\n",
      "invTLOut.txt  is empty. Skipping analysing it.\n",
      "ctxOut.txt is empty. Skipping analysing it.\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "[M::mm_idx_gen::0.057*1.01] collected minimizers\n",
      "[M::mm_idx_gen::0.075*1.00] sorted minimizers\n",
      "[M::main::0.075*1.00] loaded/built the index for 1 target sequence(s)\n",
      "[M::mm_mapopt_update::0.075*1.00] mid_occ = 100\n",
      "[M::mm_idx_stat] kmer size: 19; skip: 10; is_hpc: 1; #seq: 1\n",
      "[M::mm_idx_stat::0.079*1.00] distinct minimizers: 525478 (99.34% are singletons); average occurrences: 1.015; average spacing: 7.461; total length: 3977333\n",
      "[M::worker_pipeline::0.636*0.99] mapped 1 sequences\n",
      "[M::main] Version: 2.19-r1057\n",
      "[M::main] CMD: minimap2 -ax asm20 --eqx -H -f 100 -r1k,10k --rmq=no -t 1 exp/multiple_runs11/tmp/L07-4/is_deleted/L07-4_genome.deleted_IS.0.fasta exp/multiple_runs11/tmp/L07-4/is_deleted/L07-4_genome.deleted_IS.3.fasta\n",
      "[M::main] Real time: 0.639 sec; CPU: 0.633 sec; Peak RSS: 0.134 GB\n",
      "local_variation - WARNING - Finished syri\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "TLOut.txt  is empty. Skipping analysing it.\n",
      "invTLOut.txt  is empty. Skipping analysing it.\n",
      "ctxOut.txt is empty. Skipping analysing it.\n",
      "TL Out.txt is empty. Skipping analysing it.\n",
      "invTL Out.txt is empty. Skipping analysing it.\n",
      "dup Out.txt is empty. Skipping analysing it.\n",
      "invDup Out.txt is empty. Skipping analysing it.\n",
      "ctxOut.txt is empty. Skipping analysing it.\n",
      "TLOut.txt  is empty. Skipping analysing it.\n",
      "invTLOut.txt  is empty. Skipping analysing it.\n",
      "ctxOut.txt is empty. Skipping analysing it.\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "[M::mm_idx_gen::0.061*1.01] collected minimizers\n",
      "[M::mm_idx_gen::0.079*1.00] sorted minimizers\n",
      "[M::main::0.079*1.00] loaded/built the index for 1 target sequence(s)\n",
      "[M::mm_mapopt_update::0.079*1.00] mid_occ = 100\n",
      "[M::mm_idx_stat] kmer size: 19; skip: 10; is_hpc: 1; #seq: 1\n",
      "[M::mm_idx_stat::0.083*1.00] distinct minimizers: 525478 (99.34% are singletons); average occurrences: 1.015; average spacing: 7.461; total length: 3977333\n",
      "[M::worker_pipeline::0.657*0.99] mapped 1 sequences\n",
      "[M::main] Version: 2.19-r1057\n",
      "[M::main] CMD: minimap2 -ax asm20 --eqx -H -f 100 -r1k,10k --rmq=no -t 1 exp/multiple_runs11/tmp/L08-1/is_deleted/L08-1_genome.deleted_IS.0.fasta exp/multiple_runs11/tmp/L08-1/is_deleted/L08-1_genome.deleted_IS.1.fasta\n",
      "[M::main] Real time: 0.660 sec; CPU: 0.654 sec; Peak RSS: 0.134 GB\n",
      "local_variation - WARNING - Finished syri\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "invOut.txt  is empty. Skipping analysing it.\n",
      "TLOut.txt  is empty. Skipping analysing it.\n",
      "invTLOut.txt  is empty. Skipping analysing it.\n",
      "ctxOut.txt is empty. Skipping analysing it.\n",
      "inv Out.txt is empty. Skipping analysing it.\n",
      "TL Out.txt is empty. Skipping analysing it.\n",
      "invTL Out.txt is empty. Skipping analysing it.\n",
      "dup Out.txt is empty. Skipping analysing it.\n",
      "invDup Out.txt is empty. Skipping analysing it.\n",
      "ctxOut.txt is empty. Skipping analysing it.\n",
      "invOut.txt  is empty. Skipping analysing it.\n",
      "TLOut.txt  is empty. Skipping analysing it.\n",
      "invTLOut.txt  is empty. Skipping analysing it.\n",
      "ctxOut.txt is empty. Skipping analysing it.\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "[M::mm_idx_gen::0.057*1.01] collected minimizers\n",
      "[M::mm_idx_gen::0.075*1.00] sorted minimizers\n",
      "[M::main::0.075*1.00] loaded/built the index for 1 target sequence(s)\n",
      "[M::mm_mapopt_update::0.075*1.00] mid_occ = 100\n",
      "[M::mm_idx_stat] kmer size: 19; skip: 10; is_hpc: 1; #seq: 1\n",
      "[M::mm_idx_stat::0.079*1.00] distinct minimizers: 525478 (99.34% are singletons); average occurrences: 1.015; average spacing: 7.461; total length: 3977333\n",
      "[M::worker_pipeline::0.634*0.99] mapped 1 sequences\n",
      "[M::main] Version: 2.19-r1057\n",
      "[M::main] CMD: minimap2 -ax asm20 --eqx -H -f 100 -r1k,10k --rmq=no -t 1 exp/multiple_runs11/tmp/L08-1/is_deleted/L08-1_genome.deleted_IS.0.fasta exp/multiple_runs11/tmp/L08-1/is_deleted/L08-1_genome.deleted_IS.2.fasta\n",
      "[M::main] Real time: 0.637 sec; CPU: 0.630 sec; Peak RSS: 0.133 GB\n",
      "local_variation - WARNING - Finished syri\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "invOut.txt  is empty. Skipping analysing it.\n",
      "TLOut.txt  is empty. Skipping analysing it.\n",
      "invTLOut.txt  is empty. Skipping analysing it.\n",
      "ctxOut.txt is empty. Skipping analysing it.\n",
      "inv Out.txt is empty. Skipping analysing it.\n",
      "TL Out.txt is empty. Skipping analysing it.\n",
      "invTL Out.txt is empty. Skipping analysing it.\n",
      "dup Out.txt is empty. Skipping analysing it.\n",
      "invDup Out.txt is empty. Skipping analysing it.\n",
      "ctxOut.txt is empty. Skipping analysing it.\n",
      "invOut.txt  is empty. Skipping analysing it.\n",
      "TLOut.txt  is empty. Skipping analysing it.\n",
      "invTLOut.txt  is empty. Skipping analysing it.\n",
      "ctxOut.txt is empty. Skipping analysing it.\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "[M::mm_idx_gen::0.061*1.01] collected minimizers\n",
      "[M::mm_idx_gen::0.079*1.01] sorted minimizers\n",
      "[M::main::0.079*1.01] loaded/built the index for 1 target sequence(s)\n",
      "[M::mm_mapopt_update::0.079*1.01] mid_occ = 100\n",
      "[M::mm_idx_stat] kmer size: 19; skip: 10; is_hpc: 1; #seq: 1\n",
      "[M::mm_idx_stat::0.083*1.01] distinct minimizers: 525478 (99.34% are singletons); average occurrences: 1.015; average spacing: 7.461; total length: 3977333\n",
      "[M::worker_pipeline::0.646*0.99] mapped 1 sequences\n",
      "[M::main] Version: 2.19-r1057\n",
      "[M::main] CMD: minimap2 -ax asm20 --eqx -H -f 100 -r1k,10k --rmq=no -t 1 exp/multiple_runs11/tmp/L08-1/is_deleted/L08-1_genome.deleted_IS.0.fasta exp/multiple_runs11/tmp/L08-1/is_deleted/L08-1_genome.deleted_IS.3.fasta\n",
      "[M::main] Real time: 0.649 sec; CPU: 0.642 sec; Peak RSS: 0.134 GB\n",
      "local_variation - WARNING - Finished syri\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "invOut.txt  is empty. Skipping analysing it.\n",
      "TLOut.txt  is empty. Skipping analysing it.\n",
      "invTLOut.txt  is empty. Skipping analysing it.\n",
      "ctxOut.txt is empty. Skipping analysing it.\n",
      "inv Out.txt is empty. Skipping analysing it.\n",
      "TL Out.txt is empty. Skipping analysing it.\n",
      "invTL Out.txt is empty. Skipping analysing it.\n",
      "dup Out.txt is empty. Skipping analysing it.\n",
      "invDup Out.txt is empty. Skipping analysing it.\n",
      "ctxOut.txt is empty. Skipping analysing it.\n",
      "invOut.txt  is empty. Skipping analysing it.\n",
      "TLOut.txt  is empty. Skipping analysing it.\n",
      "invTLOut.txt  is empty. Skipping analysing it.\n",
      "ctxOut.txt is empty. Skipping analysing it.\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "[M::mm_idx_gen::0.059*1.01] collected minimizers\n",
      "[M::mm_idx_gen::0.078*1.00] sorted minimizers\n",
      "[M::main::0.078*1.00] loaded/built the index for 1 target sequence(s)\n",
      "[M::mm_mapopt_update::0.078*1.00] mid_occ = 100\n",
      "[M::mm_idx_stat] kmer size: 19; skip: 10; is_hpc: 1; #seq: 1\n",
      "[M::mm_idx_stat::0.082*1.00] distinct minimizers: 525478 (99.34% are singletons); average occurrences: 1.015; average spacing: 7.461; total length: 3977333\n",
      "[M::worker_pipeline::0.644*0.99] mapped 1 sequences\n",
      "[M::main] Version: 2.19-r1057\n",
      "[M::main] CMD: minimap2 -ax asm20 --eqx -H -f 100 -r1k,10k --rmq=no -t 1 exp/multiple_runs11/tmp/L08-2/is_deleted/L08-2_genome.deleted_IS.0.fasta exp/multiple_runs11/tmp/L08-2/is_deleted/L08-2_genome.deleted_IS.1.fasta\n",
      "[M::main] Real time: 0.647 sec; CPU: 0.642 sec; Peak RSS: 0.132 GB\n",
      "local_variation - WARNING - Finished syri\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "invOut.txt  is empty. Skipping analysing it.\n",
      "TLOut.txt  is empty. Skipping analysing it.\n",
      "invTLOut.txt  is empty. Skipping analysing it.\n",
      "ctxOut.txt is empty. Skipping analysing it.\n",
      "inv Out.txt is empty. Skipping analysing it.\n",
      "TL Out.txt is empty. Skipping analysing it.\n",
      "invTL Out.txt is empty. Skipping analysing it.\n",
      "dup Out.txt is empty. Skipping analysing it.\n",
      "invDup Out.txt is empty. Skipping analysing it.\n",
      "ctxOut.txt is empty. Skipping analysing it.\n",
      "invOut.txt  is empty. Skipping analysing it.\n",
      "TLOut.txt  is empty. Skipping analysing it.\n",
      "invTLOut.txt  is empty. Skipping analysing it.\n",
      "ctxOut.txt is empty. Skipping analysing it.\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "[M::mm_idx_gen::0.058*1.01] collected minimizers\n",
      "[M::mm_idx_gen::0.077*1.00] sorted minimizers\n",
      "[M::main::0.077*1.00] loaded/built the index for 1 target sequence(s)\n",
      "[M::mm_mapopt_update::0.077*1.00] mid_occ = 100\n",
      "[M::mm_idx_stat] kmer size: 19; skip: 10; is_hpc: 1; #seq: 1\n",
      "[M::mm_idx_stat::0.081*1.00] distinct minimizers: 525478 (99.34% are singletons); average occurrences: 1.015; average spacing: 7.461; total length: 3977333\n",
      "[M::worker_pipeline::0.650*0.99] mapped 1 sequences\n",
      "[M::main] Version: 2.19-r1057\n",
      "[M::main] CMD: minimap2 -ax asm20 --eqx -H -f 100 -r1k,10k --rmq=no -t 1 exp/multiple_runs11/tmp/L08-2/is_deleted/L08-2_genome.deleted_IS.0.fasta exp/multiple_runs11/tmp/L08-2/is_deleted/L08-2_genome.deleted_IS.2.fasta\n",
      "[M::main] Real time: 0.654 sec; CPU: 0.647 sec; Peak RSS: 0.133 GB\n",
      "local_variation - WARNING - Finished syri\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "invOut.txt  is empty. Skipping analysing it.\n",
      "TLOut.txt  is empty. Skipping analysing it.\n",
      "invTLOut.txt  is empty. Skipping analysing it.\n",
      "ctxOut.txt is empty. Skipping analysing it.\n",
      "inv Out.txt is empty. Skipping analysing it.\n",
      "TL Out.txt is empty. Skipping analysing it.\n",
      "invTL Out.txt is empty. Skipping analysing it.\n",
      "dup Out.txt is empty. Skipping analysing it.\n",
      "invDup Out.txt is empty. Skipping analysing it.\n",
      "ctxOut.txt is empty. Skipping analysing it.\n",
      "invOut.txt  is empty. Skipping analysing it.\n",
      "TLOut.txt  is empty. Skipping analysing it.\n",
      "invTLOut.txt  is empty. Skipping analysing it.\n",
      "ctxOut.txt is empty. Skipping analysing it.\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "[M::mm_idx_gen::0.056*1.01] collected minimizers\n",
      "[M::mm_idx_gen::0.075*1.00] sorted minimizers\n",
      "[M::main::0.075*1.00] loaded/built the index for 1 target sequence(s)\n",
      "[M::mm_mapopt_update::0.075*1.00] mid_occ = 100\n",
      "[M::mm_idx_stat] kmer size: 19; skip: 10; is_hpc: 1; #seq: 1\n",
      "[M::mm_idx_stat::0.079*1.00] distinct minimizers: 525478 (99.34% are singletons); average occurrences: 1.015; average spacing: 7.461; total length: 3977333\n",
      "[M::worker_pipeline::0.613*0.99] mapped 1 sequences\n",
      "[M::main] Version: 2.19-r1057\n",
      "[M::main] CMD: minimap2 -ax asm20 --eqx -H -f 100 -r1k,10k --rmq=no -t 1 exp/multiple_runs11/tmp/L08-2/is_deleted/L08-2_genome.deleted_IS.0.fasta exp/multiple_runs11/tmp/L08-2/is_deleted/L08-2_genome.deleted_IS.3.fasta\n",
      "[M::main] Real time: 0.616 sec; CPU: 0.610 sec; Peak RSS: 0.106 GB\n",
      "local_variation - WARNING - Finished syri\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "invOut.txt  is empty. Skipping analysing it.\n",
      "TLOut.txt  is empty. Skipping analysing it.\n",
      "invTLOut.txt  is empty. Skipping analysing it.\n",
      "ctxOut.txt is empty. Skipping analysing it.\n",
      "inv Out.txt is empty. Skipping analysing it.\n",
      "TL Out.txt is empty. Skipping analysing it.\n",
      "invTL Out.txt is empty. Skipping analysing it.\n",
      "invDup Out.txt is empty. Skipping analysing it.\n",
      "ctxOut.txt is empty. Skipping analysing it.\n",
      "invOut.txt  is empty. Skipping analysing it.\n",
      "TLOut.txt  is empty. Skipping analysing it.\n",
      "invTLOut.txt  is empty. Skipping analysing it.\n",
      "ctxOut.txt is empty. Skipping analysing it.\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "[M::mm_idx_gen::0.057*1.01] collected minimizers\n",
      "[M::mm_idx_gen::0.075*1.00] sorted minimizers\n",
      "[M::main::0.075*1.00] loaded/built the index for 1 target sequence(s)\n",
      "[M::mm_mapopt_update::0.075*1.00] mid_occ = 100\n",
      "[M::mm_idx_stat] kmer size: 19; skip: 10; is_hpc: 1; #seq: 1\n",
      "[M::mm_idx_stat::0.079*1.00] distinct minimizers: 525478 (99.34% are singletons); average occurrences: 1.015; average spacing: 7.461; total length: 3977333\n",
      "[M::worker_pipeline::0.642*0.99] mapped 1 sequences\n",
      "[M::main] Version: 2.19-r1057\n",
      "[M::main] CMD: minimap2 -ax asm20 --eqx -H -f 100 -r1k,10k --rmq=no -t 1 exp/multiple_runs11/tmp/L08-3/is_deleted/L08-3_genome.deleted_IS.0.fasta exp/multiple_runs11/tmp/L08-3/is_deleted/L08-3_genome.deleted_IS.1.fasta\n",
      "[M::main] Real time: 0.645 sec; CPU: 0.639 sec; Peak RSS: 0.132 GB\n",
      "local_variation - WARNING - Finished syri\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "invOut.txt  is empty. Skipping analysing it.\n",
      "TLOut.txt  is empty. Skipping analysing it.\n",
      "invTLOut.txt  is empty. Skipping analysing it.\n",
      "ctxOut.txt is empty. Skipping analysing it.\n",
      "inv Out.txt is empty. Skipping analysing it.\n",
      "TL Out.txt is empty. Skipping analysing it.\n",
      "invTL Out.txt is empty. Skipping analysing it.\n",
      "dup Out.txt is empty. Skipping analysing it.\n",
      "invDup Out.txt is empty. Skipping analysing it.\n",
      "ctxOut.txt is empty. Skipping analysing it.\n",
      "invOut.txt  is empty. Skipping analysing it.\n",
      "TLOut.txt  is empty. Skipping analysing it.\n",
      "invTLOut.txt  is empty. Skipping analysing it.\n",
      "ctxOut.txt is empty. Skipping analysing it.\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "[M::mm_idx_gen::0.067*1.01] collected minimizers\n",
      "[M::mm_idx_gen::0.086*1.00] sorted minimizers\n",
      "[M::main::0.086*1.00] loaded/built the index for 1 target sequence(s)\n",
      "[M::mm_mapopt_update::0.086*1.00] mid_occ = 100\n",
      "[M::mm_idx_stat] kmer size: 19; skip: 10; is_hpc: 1; #seq: 1\n",
      "[M::mm_idx_stat::0.090*1.00] distinct minimizers: 525478 (99.34% are singletons); average occurrences: 1.015; average spacing: 7.461; total length: 3977333\n",
      "[M::worker_pipeline::0.644*0.99] mapped 1 sequences\n",
      "[M::main] Version: 2.19-r1057\n",
      "[M::main] CMD: minimap2 -ax asm20 --eqx -H -f 100 -r1k,10k --rmq=no -t 1 exp/multiple_runs11/tmp/L08-3/is_deleted/L08-3_genome.deleted_IS.0.fasta exp/multiple_runs11/tmp/L08-3/is_deleted/L08-3_genome.deleted_IS.2.fasta\n",
      "[M::main] Real time: 0.648 sec; CPU: 0.642 sec; Peak RSS: 0.132 GB\n",
      "local_variation - WARNING - Finished syri\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "invOut.txt  is empty. Skipping analysing it.\n",
      "TLOut.txt  is empty. Skipping analysing it.\n",
      "invTLOut.txt  is empty. Skipping analysing it.\n",
      "ctxOut.txt is empty. Skipping analysing it.\n",
      "inv Out.txt is empty. Skipping analysing it.\n",
      "TL Out.txt is empty. Skipping analysing it.\n",
      "invTL Out.txt is empty. Skipping analysing it.\n",
      "dup Out.txt is empty. Skipping analysing it.\n",
      "invDup Out.txt is empty. Skipping analysing it.\n",
      "ctxOut.txt is empty. Skipping analysing it.\n",
      "invOut.txt  is empty. Skipping analysing it.\n",
      "TLOut.txt  is empty. Skipping analysing it.\n",
      "invTLOut.txt  is empty. Skipping analysing it.\n",
      "ctxOut.txt is empty. Skipping analysing it.\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "[M::mm_idx_gen::0.057*1.01] collected minimizers\n",
      "[M::mm_idx_gen::0.076*1.00] sorted minimizers\n",
      "[M::main::0.076*1.00] loaded/built the index for 1 target sequence(s)\n",
      "[M::mm_mapopt_update::0.076*1.00] mid_occ = 100\n",
      "[M::mm_idx_stat] kmer size: 19; skip: 10; is_hpc: 1; #seq: 1\n",
      "[M::mm_idx_stat::0.079*1.00] distinct minimizers: 525478 (99.34% are singletons); average occurrences: 1.015; average spacing: 7.461; total length: 3977333\n",
      "[M::worker_pipeline::0.645*0.99] mapped 1 sequences\n",
      "[M::main] Version: 2.19-r1057\n",
      "[M::main] CMD: minimap2 -ax asm20 --eqx -H -f 100 -r1k,10k --rmq=no -t 1 exp/multiple_runs11/tmp/L08-3/is_deleted/L08-3_genome.deleted_IS.0.fasta exp/multiple_runs11/tmp/L08-3/is_deleted/L08-3_genome.deleted_IS.3.fasta\n",
      "[M::main] Real time: 0.648 sec; CPU: 0.643 sec; Peak RSS: 0.133 GB\n",
      "local_variation - WARNING - Finished syri\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "invOut.txt  is empty. Skipping analysing it.\n",
      "TLOut.txt  is empty. Skipping analysing it.\n",
      "invTLOut.txt  is empty. Skipping analysing it.\n",
      "ctxOut.txt is empty. Skipping analysing it.\n",
      "inv Out.txt is empty. Skipping analysing it.\n",
      "TL Out.txt is empty. Skipping analysing it.\n",
      "invTL Out.txt is empty. Skipping analysing it.\n",
      "dup Out.txt is empty. Skipping analysing it.\n",
      "invDup Out.txt is empty. Skipping analysing it.\n",
      "ctxOut.txt is empty. Skipping analysing it.\n",
      "invOut.txt  is empty. Skipping analysing it.\n",
      "TLOut.txt  is empty. Skipping analysing it.\n",
      "invTLOut.txt  is empty. Skipping analysing it.\n",
      "ctxOut.txt is empty. Skipping analysing it.\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "[M::mm_idx_gen::0.059*1.01] collected minimizers\n",
      "[M::mm_idx_gen::0.077*1.00] sorted minimizers\n",
      "[M::main::0.077*1.00] loaded/built the index for 1 target sequence(s)\n",
      "[M::mm_mapopt_update::0.077*1.00] mid_occ = 100\n",
      "[M::mm_idx_stat] kmer size: 19; skip: 10; is_hpc: 1; #seq: 1\n",
      "[M::mm_idx_stat::0.081*1.00] distinct minimizers: 525478 (99.34% are singletons); average occurrences: 1.015; average spacing: 7.461; total length: 3977333\n",
      "[M::worker_pipeline::0.655*0.99] mapped 1 sequences\n",
      "[M::main] Version: 2.19-r1057\n",
      "[M::main] CMD: minimap2 -ax asm20 --eqx -H -f 100 -r1k,10k --rmq=no -t 1 exp/multiple_runs11/tmp/L08-4/is_deleted/L08-4_genome.deleted_IS.0.fasta exp/multiple_runs11/tmp/L08-4/is_deleted/L08-4_genome.deleted_IS.1.fasta\n",
      "[M::main] Real time: 0.658 sec; CPU: 0.653 sec; Peak RSS: 0.134 GB\n",
      "local_variation - WARNING - Finished syri\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "invOut.txt  is empty. Skipping analysing it.\n",
      "TLOut.txt  is empty. Skipping analysing it.\n",
      "invTLOut.txt  is empty. Skipping analysing it.\n",
      "ctxOut.txt is empty. Skipping analysing it.\n",
      "inv Out.txt is empty. Skipping analysing it.\n",
      "TL Out.txt is empty. Skipping analysing it.\n",
      "invTL Out.txt is empty. Skipping analysing it.\n",
      "dup Out.txt is empty. Skipping analysing it.\n",
      "invDup Out.txt is empty. Skipping analysing it.\n",
      "ctxOut.txt is empty. Skipping analysing it.\n",
      "invOut.txt  is empty. Skipping analysing it.\n",
      "TLOut.txt  is empty. Skipping analysing it.\n",
      "invTLOut.txt  is empty. Skipping analysing it.\n",
      "ctxOut.txt is empty. Skipping analysing it.\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "[M::mm_idx_gen::0.064*1.01] collected minimizers\n",
      "[M::mm_idx_gen::0.082*1.00] sorted minimizers\n",
      "[M::main::0.082*1.00] loaded/built the index for 1 target sequence(s)\n",
      "[M::mm_mapopt_update::0.082*1.00] mid_occ = 100\n",
      "[M::mm_idx_stat] kmer size: 19; skip: 10; is_hpc: 1; #seq: 1\n",
      "[M::mm_idx_stat::0.086*1.00] distinct minimizers: 525478 (99.34% are singletons); average occurrences: 1.015; average spacing: 7.461; total length: 3977333\n",
      "[M::worker_pipeline::0.659*0.99] mapped 1 sequences\n",
      "[M::main] Version: 2.19-r1057\n",
      "[M::main] CMD: minimap2 -ax asm20 --eqx -H -f 100 -r1k,10k --rmq=no -t 1 exp/multiple_runs11/tmp/L08-4/is_deleted/L08-4_genome.deleted_IS.0.fasta exp/multiple_runs11/tmp/L08-4/is_deleted/L08-4_genome.deleted_IS.2.fasta\n",
      "[M::main] Real time: 0.661 sec; CPU: 0.655 sec; Peak RSS: 0.133 GB\n",
      "local_variation - WARNING - Finished syri\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "invOut.txt  is empty. Skipping analysing it.\n",
      "TLOut.txt  is empty. Skipping analysing it.\n",
      "invTLOut.txt  is empty. Skipping analysing it.\n",
      "ctxOut.txt is empty. Skipping analysing it.\n",
      "inv Out.txt is empty. Skipping analysing it.\n",
      "TL Out.txt is empty. Skipping analysing it.\n",
      "invTL Out.txt is empty. Skipping analysing it.\n",
      "dup Out.txt is empty. Skipping analysing it.\n",
      "invDup Out.txt is empty. Skipping analysing it.\n",
      "ctxOut.txt is empty. Skipping analysing it.\n",
      "invOut.txt  is empty. Skipping analysing it.\n",
      "TLOut.txt  is empty. Skipping analysing it.\n",
      "invTLOut.txt  is empty. Skipping analysing it.\n",
      "ctxOut.txt is empty. Skipping analysing it.\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "[M::mm_idx_gen::0.062*1.01] collected minimizers\n",
      "[M::mm_idx_gen::0.080*1.00] sorted minimizers\n",
      "[M::main::0.080*1.00] loaded/built the index for 1 target sequence(s)\n",
      "[M::mm_mapopt_update::0.080*1.00] mid_occ = 100\n",
      "[M::mm_idx_stat] kmer size: 19; skip: 10; is_hpc: 1; #seq: 1\n",
      "[M::mm_idx_stat::0.084*1.00] distinct minimizers: 525478 (99.34% are singletons); average occurrences: 1.015; average spacing: 7.461; total length: 3977333\n",
      "[M::worker_pipeline::0.656*0.99] mapped 1 sequences\n",
      "[M::main] Version: 2.19-r1057\n",
      "[M::main] CMD: minimap2 -ax asm20 --eqx -H -f 100 -r1k,10k --rmq=no -t 1 exp/multiple_runs11/tmp/L08-4/is_deleted/L08-4_genome.deleted_IS.0.fasta exp/multiple_runs11/tmp/L08-4/is_deleted/L08-4_genome.deleted_IS.3.fasta\n",
      "[M::main] Real time: 0.658 sec; CPU: 0.652 sec; Peak RSS: 0.132 GB\n",
      "local_variation - WARNING - Finished syri\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "invOut.txt  is empty. Skipping analysing it.\n",
      "TLOut.txt  is empty. Skipping analysing it.\n",
      "invTLOut.txt  is empty. Skipping analysing it.\n",
      "ctxOut.txt is empty. Skipping analysing it.\n",
      "inv Out.txt is empty. Skipping analysing it.\n",
      "TL Out.txt is empty. Skipping analysing it.\n",
      "invTL Out.txt is empty. Skipping analysing it.\n",
      "dup Out.txt is empty. Skipping analysing it.\n",
      "invDup Out.txt is empty. Skipping analysing it.\n",
      "ctxOut.txt is empty. Skipping analysing it.\n",
      "invOut.txt  is empty. Skipping analysing it.\n",
      "TLOut.txt  is empty. Skipping analysing it.\n",
      "invTLOut.txt  is empty. Skipping analysing it.\n",
      "ctxOut.txt is empty. Skipping analysing it.\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "[M::mm_idx_gen::0.059*1.01] collected minimizers\n",
      "[M::mm_idx_gen::0.077*1.00] sorted minimizers\n",
      "[M::main::0.077*1.00] loaded/built the index for 1 target sequence(s)\n",
      "[M::mm_mapopt_update::0.077*1.00] mid_occ = 100\n",
      "[M::mm_idx_stat] kmer size: 19; skip: 10; is_hpc: 1; #seq: 1\n",
      "[M::mm_idx_stat::0.081*1.00] distinct minimizers: 525478 (99.34% are singletons); average occurrences: 1.015; average spacing: 7.461; total length: 3977333\n",
      "[M::worker_pipeline::0.653*0.99] mapped 1 sequences\n",
      "[M::main] Version: 2.19-r1057\n",
      "[M::main] CMD: minimap2 -ax asm20 --eqx -H -f 100 -r1k,10k --rmq=no -t 1 exp/multiple_runs11/tmp/L09-1/is_deleted/L09-1_genome.deleted_IS.0.fasta exp/multiple_runs11/tmp/L09-1/is_deleted/L09-1_genome.deleted_IS.1.fasta\n",
      "[M::main] Real time: 0.655 sec; CPU: 0.650 sec; Peak RSS: 0.131 GB\n",
      "local_variation - WARNING - Finished syri\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "invOut.txt  is empty. Skipping analysing it.\n",
      "TLOut.txt  is empty. Skipping analysing it.\n",
      "invTLOut.txt  is empty. Skipping analysing it.\n",
      "ctxOut.txt is empty. Skipping analysing it.\n",
      "inv Out.txt is empty. Skipping analysing it.\n",
      "TL Out.txt is empty. Skipping analysing it.\n",
      "invTL Out.txt is empty. Skipping analysing it.\n",
      "dup Out.txt is empty. Skipping analysing it.\n",
      "invDup Out.txt is empty. Skipping analysing it.\n",
      "ctxOut.txt is empty. Skipping analysing it.\n",
      "invOut.txt  is empty. Skipping analysing it.\n",
      "TLOut.txt  is empty. Skipping analysing it.\n",
      "invTLOut.txt  is empty. Skipping analysing it.\n",
      "ctxOut.txt is empty. Skipping analysing it.\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "[M::mm_idx_gen::0.058*1.01] collected minimizers\n",
      "[M::mm_idx_gen::0.077*1.00] sorted minimizers\n",
      "[M::main::0.077*1.00] loaded/built the index for 1 target sequence(s)\n",
      "[M::mm_mapopt_update::0.077*1.00] mid_occ = 100\n",
      "[M::mm_idx_stat] kmer size: 19; skip: 10; is_hpc: 1; #seq: 1\n",
      "[M::mm_idx_stat::0.080*1.00] distinct minimizers: 525478 (99.34% are singletons); average occurrences: 1.015; average spacing: 7.461; total length: 3977333\n",
      "[M::worker_pipeline::0.648*0.99] mapped 1 sequences\n",
      "[M::main] Version: 2.19-r1057\n",
      "[M::main] CMD: minimap2 -ax asm20 --eqx -H -f 100 -r1k,10k --rmq=no -t 1 exp/multiple_runs11/tmp/L09-1/is_deleted/L09-1_genome.deleted_IS.0.fasta exp/multiple_runs11/tmp/L09-1/is_deleted/L09-1_genome.deleted_IS.2.fasta\n",
      "[M::main] Real time: 0.651 sec; CPU: 0.646 sec; Peak RSS: 0.131 GB\n",
      "local_variation - WARNING - Finished syri\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "invOut.txt  is empty. Skipping analysing it.\n",
      "TLOut.txt  is empty. Skipping analysing it.\n",
      "invTLOut.txt  is empty. Skipping analysing it.\n",
      "ctxOut.txt is empty. Skipping analysing it.\n",
      "inv Out.txt is empty. Skipping analysing it.\n",
      "TL Out.txt is empty. Skipping analysing it.\n",
      "invTL Out.txt is empty. Skipping analysing it.\n",
      "dup Out.txt is empty. Skipping analysing it.\n",
      "invDup Out.txt is empty. Skipping analysing it.\n",
      "ctxOut.txt is empty. Skipping analysing it.\n",
      "invOut.txt  is empty. Skipping analysing it.\n",
      "TLOut.txt  is empty. Skipping analysing it.\n",
      "invTLOut.txt  is empty. Skipping analysing it.\n",
      "ctxOut.txt is empty. Skipping analysing it.\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "[M::mm_idx_gen::0.058*1.01] collected minimizers\n",
      "[M::mm_idx_gen::0.076*1.00] sorted minimizers\n",
      "[M::main::0.076*1.00] loaded/built the index for 1 target sequence(s)\n",
      "[M::mm_mapopt_update::0.076*1.00] mid_occ = 100\n",
      "[M::mm_idx_stat] kmer size: 19; skip: 10; is_hpc: 1; #seq: 1\n",
      "[M::mm_idx_stat::0.080*1.00] distinct minimizers: 525478 (99.34% are singletons); average occurrences: 1.015; average spacing: 7.461; total length: 3977333\n",
      "[M::worker_pipeline::0.649*0.99] mapped 1 sequences\n",
      "[M::main] Version: 2.19-r1057\n",
      "[M::main] CMD: minimap2 -ax asm20 --eqx -H -f 100 -r1k,10k --rmq=no -t 1 exp/multiple_runs11/tmp/L09-1/is_deleted/L09-1_genome.deleted_IS.0.fasta exp/multiple_runs11/tmp/L09-1/is_deleted/L09-1_genome.deleted_IS.3.fasta\n",
      "[M::main] Real time: 0.651 sec; CPU: 0.645 sec; Peak RSS: 0.132 GB\n",
      "local_variation - WARNING - Finished syri\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "invOut.txt  is empty. Skipping analysing it.\n",
      "TLOut.txt  is empty. Skipping analysing it.\n",
      "invTLOut.txt  is empty. Skipping analysing it.\n",
      "ctxOut.txt is empty. Skipping analysing it.\n",
      "inv Out.txt is empty. Skipping analysing it.\n",
      "TL Out.txt is empty. Skipping analysing it.\n",
      "invTL Out.txt is empty. Skipping analysing it.\n",
      "dup Out.txt is empty. Skipping analysing it.\n",
      "invDup Out.txt is empty. Skipping analysing it.\n",
      "ctxOut.txt is empty. Skipping analysing it.\n",
      "invOut.txt  is empty. Skipping analysing it.\n",
      "TLOut.txt  is empty. Skipping analysing it.\n",
      "invTLOut.txt  is empty. Skipping analysing it.\n",
      "ctxOut.txt is empty. Skipping analysing it.\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "[M::mm_idx_gen::0.059*1.01] collected minimizers\n",
      "[M::mm_idx_gen::0.078*1.00] sorted minimizers\n",
      "[M::main::0.078*1.00] loaded/built the index for 1 target sequence(s)\n",
      "[M::mm_mapopt_update::0.078*1.00] mid_occ = 100\n",
      "[M::mm_idx_stat] kmer size: 19; skip: 10; is_hpc: 1; #seq: 1\n",
      "[M::mm_idx_stat::0.081*1.00] distinct minimizers: 525478 (99.34% are singletons); average occurrences: 1.015; average spacing: 7.461; total length: 3977333\n",
      "[M::worker_pipeline::0.651*0.99] mapped 1 sequences\n",
      "[M::main] Version: 2.19-r1057\n",
      "[M::main] CMD: minimap2 -ax asm20 --eqx -H -f 100 -r1k,10k --rmq=no -t 1 exp/multiple_runs11/tmp/L09-2/is_deleted/L09-2_genome.deleted_IS.0.fasta exp/multiple_runs11/tmp/L09-2/is_deleted/L09-2_genome.deleted_IS.1.fasta\n",
      "[M::main] Real time: 0.654 sec; CPU: 0.649 sec; Peak RSS: 0.131 GB\n",
      "local_variation - WARNING - Finished syri\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "invOut.txt  is empty. Skipping analysing it.\n",
      "TLOut.txt  is empty. Skipping analysing it.\n",
      "invTLOut.txt  is empty. Skipping analysing it.\n",
      "ctxOut.txt is empty. Skipping analysing it.\n",
      "inv Out.txt is empty. Skipping analysing it.\n",
      "TL Out.txt is empty. Skipping analysing it.\n",
      "invTL Out.txt is empty. Skipping analysing it.\n",
      "dup Out.txt is empty. Skipping analysing it.\n",
      "invDup Out.txt is empty. Skipping analysing it.\n",
      "ctxOut.txt is empty. Skipping analysing it.\n",
      "invOut.txt  is empty. Skipping analysing it.\n",
      "TLOut.txt  is empty. Skipping analysing it.\n",
      "invTLOut.txt  is empty. Skipping analysing it.\n",
      "ctxOut.txt is empty. Skipping analysing it.\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "[M::mm_idx_gen::0.058*1.01] collected minimizers\n",
      "[M::mm_idx_gen::0.076*1.00] sorted minimizers\n",
      "[M::main::0.076*1.00] loaded/built the index for 1 target sequence(s)\n",
      "[M::mm_mapopt_update::0.076*1.00] mid_occ = 100\n",
      "[M::mm_idx_stat] kmer size: 19; skip: 10; is_hpc: 1; #seq: 1\n",
      "[M::mm_idx_stat::0.080*1.00] distinct minimizers: 525478 (99.34% are singletons); average occurrences: 1.015; average spacing: 7.461; total length: 3977333\n",
      "[M::worker_pipeline::0.650*0.99] mapped 1 sequences\n",
      "[M::main] Version: 2.19-r1057\n",
      "[M::main] CMD: minimap2 -ax asm20 --eqx -H -f 100 -r1k,10k --rmq=no -t 1 exp/multiple_runs11/tmp/L09-2/is_deleted/L09-2_genome.deleted_IS.0.fasta exp/multiple_runs11/tmp/L09-2/is_deleted/L09-2_genome.deleted_IS.2.fasta\n",
      "[M::main] Real time: 0.652 sec; CPU: 0.646 sec; Peak RSS: 0.133 GB\n",
      "local_variation - WARNING - Finished syri\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "TLOut.txt  is empty. Skipping analysing it.\n",
      "invTLOut.txt  is empty. Skipping analysing it.\n",
      "ctxOut.txt is empty. Skipping analysing it.\n",
      "TL Out.txt is empty. Skipping analysing it.\n",
      "invTL Out.txt is empty. Skipping analysing it.\n",
      "dup Out.txt is empty. Skipping analysing it.\n",
      "invDup Out.txt is empty. Skipping analysing it.\n",
      "ctxOut.txt is empty. Skipping analysing it.\n",
      "TLOut.txt  is empty. Skipping analysing it.\n",
      "invTLOut.txt  is empty. Skipping analysing it.\n",
      "ctxOut.txt is empty. Skipping analysing it.\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "[M::mm_idx_gen::0.058*1.01] collected minimizers\n",
      "[M::mm_idx_gen::0.077*1.00] sorted minimizers\n",
      "[M::main::0.077*1.00] loaded/built the index for 1 target sequence(s)\n",
      "[M::mm_mapopt_update::0.077*1.00] mid_occ = 100\n",
      "[M::mm_idx_stat] kmer size: 19; skip: 10; is_hpc: 1; #seq: 1\n",
      "[M::mm_idx_stat::0.080*1.00] distinct minimizers: 525478 (99.34% are singletons); average occurrences: 1.015; average spacing: 7.461; total length: 3977333\n",
      "[M::worker_pipeline::0.649*0.99] mapped 1 sequences\n",
      "[M::main] Version: 2.19-r1057\n",
      "[M::main] CMD: minimap2 -ax asm20 --eqx -H -f 100 -r1k,10k --rmq=no -t 1 exp/multiple_runs11/tmp/L09-2/is_deleted/L09-2_genome.deleted_IS.0.fasta exp/multiple_runs11/tmp/L09-2/is_deleted/L09-2_genome.deleted_IS.3.fasta\n",
      "[M::main] Real time: 0.652 sec; CPU: 0.646 sec; Peak RSS: 0.133 GB\n",
      "local_variation - WARNING - Finished syri\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "TLOut.txt  is empty. Skipping analysing it.\n",
      "invTLOut.txt  is empty. Skipping analysing it.\n",
      "ctxOut.txt is empty. Skipping analysing it.\n",
      "TL Out.txt is empty. Skipping analysing it.\n",
      "invTL Out.txt is empty. Skipping analysing it.\n",
      "dup Out.txt is empty. Skipping analysing it.\n",
      "invDup Out.txt is empty. Skipping analysing it.\n",
      "ctxOut.txt is empty. Skipping analysing it.\n",
      "TLOut.txt  is empty. Skipping analysing it.\n",
      "invTLOut.txt  is empty. Skipping analysing it.\n",
      "ctxOut.txt is empty. Skipping analysing it.\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "[M::mm_idx_gen::0.057*1.01] collected minimizers\n",
      "[M::mm_idx_gen::0.076*1.00] sorted minimizers\n",
      "[M::main::0.076*1.00] loaded/built the index for 1 target sequence(s)\n",
      "[M::mm_mapopt_update::0.076*1.00] mid_occ = 100\n",
      "[M::mm_idx_stat] kmer size: 19; skip: 10; is_hpc: 1; #seq: 1\n",
      "[M::mm_idx_stat::0.080*1.00] distinct minimizers: 525478 (99.34% are singletons); average occurrences: 1.015; average spacing: 7.461; total length: 3977333\n",
      "[M::worker_pipeline::0.652*0.99] mapped 1 sequences\n",
      "[M::main] Version: 2.19-r1057\n",
      "[M::main] CMD: minimap2 -ax asm20 --eqx -H -f 100 -r1k,10k --rmq=no -t 1 exp/multiple_runs11/tmp/L09-3/is_deleted/L09-3_genome.deleted_IS.0.fasta exp/multiple_runs11/tmp/L09-3/is_deleted/L09-3_genome.deleted_IS.1.fasta\n",
      "[M::main] Real time: 0.654 sec; CPU: 0.649 sec; Peak RSS: 0.131 GB\n",
      "local_variation - WARNING - Finished syri\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "invOut.txt  is empty. Skipping analysing it.\n",
      "TLOut.txt  is empty. Skipping analysing it.\n",
      "invTLOut.txt  is empty. Skipping analysing it.\n",
      "ctxOut.txt is empty. Skipping analysing it.\n",
      "inv Out.txt is empty. Skipping analysing it.\n",
      "TL Out.txt is empty. Skipping analysing it.\n",
      "invTL Out.txt is empty. Skipping analysing it.\n",
      "dup Out.txt is empty. Skipping analysing it.\n",
      "invDup Out.txt is empty. Skipping analysing it.\n",
      "ctxOut.txt is empty. Skipping analysing it.\n",
      "invOut.txt  is empty. Skipping analysing it.\n",
      "TLOut.txt  is empty. Skipping analysing it.\n",
      "invTLOut.txt  is empty. Skipping analysing it.\n",
      "ctxOut.txt is empty. Skipping analysing it.\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "[M::mm_idx_gen::0.058*1.01] collected minimizers\n",
      "[M::mm_idx_gen::0.077*1.00] sorted minimizers\n",
      "[M::main::0.077*1.00] loaded/built the index for 1 target sequence(s)\n",
      "[M::mm_mapopt_update::0.077*1.00] mid_occ = 100\n",
      "[M::mm_idx_stat] kmer size: 19; skip: 10; is_hpc: 1; #seq: 1\n",
      "[M::mm_idx_stat::0.081*1.00] distinct minimizers: 525478 (99.34% are singletons); average occurrences: 1.015; average spacing: 7.461; total length: 3977333\n",
      "[M::worker_pipeline::0.652*0.99] mapped 1 sequences\n",
      "[M::main] Version: 2.19-r1057\n",
      "[M::main] CMD: minimap2 -ax asm20 --eqx -H -f 100 -r1k,10k --rmq=no -t 1 exp/multiple_runs11/tmp/L09-3/is_deleted/L09-3_genome.deleted_IS.0.fasta exp/multiple_runs11/tmp/L09-3/is_deleted/L09-3_genome.deleted_IS.2.fasta\n",
      "[M::main] Real time: 0.655 sec; CPU: 0.648 sec; Peak RSS: 0.133 GB\n",
      "local_variation - WARNING - Finished syri\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "invOut.txt  is empty. Skipping analysing it.\n",
      "TLOut.txt  is empty. Skipping analysing it.\n",
      "invTLOut.txt  is empty. Skipping analysing it.\n",
      "ctxOut.txt is empty. Skipping analysing it.\n",
      "inv Out.txt is empty. Skipping analysing it.\n",
      "TL Out.txt is empty. Skipping analysing it.\n",
      "invTL Out.txt is empty. Skipping analysing it.\n",
      "dup Out.txt is empty. Skipping analysing it.\n",
      "invDup Out.txt is empty. Skipping analysing it.\n",
      "ctxOut.txt is empty. Skipping analysing it.\n",
      "invOut.txt  is empty. Skipping analysing it.\n",
      "TLOut.txt  is empty. Skipping analysing it.\n",
      "invTLOut.txt  is empty. Skipping analysing it.\n",
      "ctxOut.txt is empty. Skipping analysing it.\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "[M::mm_idx_gen::0.059*1.01] collected minimizers\n",
      "[M::mm_idx_gen::0.078*1.00] sorted minimizers\n",
      "[M::main::0.078*1.00] loaded/built the index for 1 target sequence(s)\n",
      "[M::mm_mapopt_update::0.078*1.00] mid_occ = 100\n",
      "[M::mm_idx_stat] kmer size: 19; skip: 10; is_hpc: 1; #seq: 1\n",
      "[M::mm_idx_stat::0.082*1.00] distinct minimizers: 525478 (99.34% are singletons); average occurrences: 1.015; average spacing: 7.461; total length: 3977333\n",
      "[M::worker_pipeline::0.653*0.99] mapped 1 sequences\n",
      "[M::main] Version: 2.19-r1057\n",
      "[M::main] CMD: minimap2 -ax asm20 --eqx -H -f 100 -r1k,10k --rmq=no -t 1 exp/multiple_runs11/tmp/L09-3/is_deleted/L09-3_genome.deleted_IS.0.fasta exp/multiple_runs11/tmp/L09-3/is_deleted/L09-3_genome.deleted_IS.3.fasta\n",
      "[M::main] Real time: 0.655 sec; CPU: 0.649 sec; Peak RSS: 0.132 GB\n",
      "local_variation - WARNING - Finished syri\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "invOut.txt  is empty. Skipping analysing it.\n",
      "TLOut.txt  is empty. Skipping analysing it.\n",
      "invTLOut.txt  is empty. Skipping analysing it.\n",
      "ctxOut.txt is empty. Skipping analysing it.\n",
      "inv Out.txt is empty. Skipping analysing it.\n",
      "TL Out.txt is empty. Skipping analysing it.\n",
      "invTL Out.txt is empty. Skipping analysing it.\n",
      "dup Out.txt is empty. Skipping analysing it.\n",
      "invDup Out.txt is empty. Skipping analysing it.\n",
      "ctxOut.txt is empty. Skipping analysing it.\n",
      "invOut.txt  is empty. Skipping analysing it.\n",
      "TLOut.txt  is empty. Skipping analysing it.\n",
      "invTLOut.txt  is empty. Skipping analysing it.\n",
      "ctxOut.txt is empty. Skipping analysing it.\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "[M::mm_idx_gen::0.063*1.01] collected minimizers\n",
      "[M::mm_idx_gen::0.082*1.00] sorted minimizers\n",
      "[M::main::0.082*1.00] loaded/built the index for 1 target sequence(s)\n",
      "[M::mm_mapopt_update::0.082*1.00] mid_occ = 100\n",
      "[M::mm_idx_stat] kmer size: 19; skip: 10; is_hpc: 1; #seq: 1\n",
      "[M::mm_idx_stat::0.085*1.00] distinct minimizers: 525478 (99.34% are singletons); average occurrences: 1.015; average spacing: 7.461; total length: 3977333\n",
      "[M::worker_pipeline::0.654*0.99] mapped 1 sequences\n",
      "[M::main] Version: 2.19-r1057\n",
      "[M::main] CMD: minimap2 -ax asm20 --eqx -H -f 100 -r1k,10k --rmq=no -t 1 exp/multiple_runs11/tmp/L09-4/is_deleted/L09-4_genome.deleted_IS.0.fasta exp/multiple_runs11/tmp/L09-4/is_deleted/L09-4_genome.deleted_IS.1.fasta\n",
      "[M::main] Real time: 0.657 sec; CPU: 0.652 sec; Peak RSS: 0.131 GB\n",
      "local_variation - WARNING - Finished syri\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "invOut.txt  is empty. Skipping analysing it.\n",
      "TLOut.txt  is empty. Skipping analysing it.\n",
      "invTLOut.txt  is empty. Skipping analysing it.\n",
      "ctxOut.txt is empty. Skipping analysing it.\n",
      "inv Out.txt is empty. Skipping analysing it.\n",
      "TL Out.txt is empty. Skipping analysing it.\n",
      "invTL Out.txt is empty. Skipping analysing it.\n",
      "dup Out.txt is empty. Skipping analysing it.\n",
      "invDup Out.txt is empty. Skipping analysing it.\n",
      "ctxOut.txt is empty. Skipping analysing it.\n",
      "invOut.txt  is empty. Skipping analysing it.\n",
      "TLOut.txt  is empty. Skipping analysing it.\n",
      "invTLOut.txt  is empty. Skipping analysing it.\n",
      "ctxOut.txt is empty. Skipping analysing it.\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "[M::mm_idx_gen::0.062*1.01] collected minimizers\n",
      "[M::mm_idx_gen::0.081*1.00] sorted minimizers\n",
      "[M::main::0.081*1.00] loaded/built the index for 1 target sequence(s)\n",
      "[M::mm_mapopt_update::0.081*1.00] mid_occ = 100\n",
      "[M::mm_idx_stat] kmer size: 19; skip: 10; is_hpc: 1; #seq: 1\n",
      "[M::mm_idx_stat::0.085*1.00] distinct minimizers: 525478 (99.34% are singletons); average occurrences: 1.015; average spacing: 7.461; total length: 3977333\n",
      "[M::worker_pipeline::0.655*0.99] mapped 1 sequences\n",
      "[M::main] Version: 2.19-r1057\n",
      "[M::main] CMD: minimap2 -ax asm20 --eqx -H -f 100 -r1k,10k --rmq=no -t 1 exp/multiple_runs11/tmp/L09-4/is_deleted/L09-4_genome.deleted_IS.0.fasta exp/multiple_runs11/tmp/L09-4/is_deleted/L09-4_genome.deleted_IS.2.fasta\n",
      "[M::main] Real time: 0.657 sec; CPU: 0.652 sec; Peak RSS: 0.132 GB\n",
      "local_variation - WARNING - Finished syri\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "invOut.txt  is empty. Skipping analysing it.\n",
      "TLOut.txt  is empty. Skipping analysing it.\n",
      "invTLOut.txt  is empty. Skipping analysing it.\n",
      "ctxOut.txt is empty. Skipping analysing it.\n",
      "inv Out.txt is empty. Skipping analysing it.\n",
      "TL Out.txt is empty. Skipping analysing it.\n",
      "invTL Out.txt is empty. Skipping analysing it.\n",
      "dup Out.txt is empty. Skipping analysing it.\n",
      "invDup Out.txt is empty. Skipping analysing it.\n",
      "ctxOut.txt is empty. Skipping analysing it.\n",
      "invOut.txt  is empty. Skipping analysing it.\n",
      "TLOut.txt  is empty. Skipping analysing it.\n",
      "invTLOut.txt  is empty. Skipping analysing it.\n",
      "ctxOut.txt is empty. Skipping analysing it.\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "[M::mm_idx_gen::0.062*1.01] collected minimizers\n",
      "[M::mm_idx_gen::0.081*1.00] sorted minimizers\n",
      "[M::main::0.081*1.00] loaded/built the index for 1 target sequence(s)\n",
      "[M::mm_mapopt_update::0.081*1.00] mid_occ = 100\n",
      "[M::mm_idx_stat] kmer size: 19; skip: 10; is_hpc: 1; #seq: 1\n",
      "[M::mm_idx_stat::0.085*1.00] distinct minimizers: 525478 (99.34% are singletons); average occurrences: 1.015; average spacing: 7.461; total length: 3977333\n",
      "[M::worker_pipeline::0.653*0.99] mapped 1 sequences\n",
      "[M::main] Version: 2.19-r1057\n",
      "[M::main] CMD: minimap2 -ax asm20 --eqx -H -f 100 -r1k,10k --rmq=no -t 1 exp/multiple_runs11/tmp/L09-4/is_deleted/L09-4_genome.deleted_IS.0.fasta exp/multiple_runs11/tmp/L09-4/is_deleted/L09-4_genome.deleted_IS.3.fasta\n",
      "[M::main] Real time: 0.655 sec; CPU: 0.650 sec; Peak RSS: 0.132 GB\n",
      "local_variation - WARNING - Finished syri\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "invOut.txt  is empty. Skipping analysing it.\n",
      "TLOut.txt  is empty. Skipping analysing it.\n",
      "invTLOut.txt  is empty. Skipping analysing it.\n",
      "ctxOut.txt is empty. Skipping analysing it.\n",
      "inv Out.txt is empty. Skipping analysing it.\n",
      "TL Out.txt is empty. Skipping analysing it.\n",
      "invTL Out.txt is empty. Skipping analysing it.\n",
      "dup Out.txt is empty. Skipping analysing it.\n",
      "invDup Out.txt is empty. Skipping analysing it.\n",
      "ctxOut.txt is empty. Skipping analysing it.\n",
      "invOut.txt  is empty. Skipping analysing it.\n",
      "TLOut.txt  is empty. Skipping analysing it.\n",
      "invTLOut.txt  is empty. Skipping analysing it.\n",
      "ctxOut.txt is empty. Skipping analysing it.\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "[M::mm_idx_gen::0.057*1.01] collected minimizers\n",
      "[M::mm_idx_gen::0.076*1.00] sorted minimizers\n",
      "[M::main::0.076*1.00] loaded/built the index for 1 target sequence(s)\n",
      "[M::mm_mapopt_update::0.076*1.00] mid_occ = 100\n",
      "[M::mm_idx_stat] kmer size: 19; skip: 10; is_hpc: 1; #seq: 1\n",
      "[M::mm_idx_stat::0.080*1.00] distinct minimizers: 525478 (99.34% are singletons); average occurrences: 1.015; average spacing: 7.461; total length: 3977333\n",
      "[M::worker_pipeline::0.651*0.99] mapped 1 sequences\n",
      "[M::main] Version: 2.19-r1057\n",
      "[M::main] CMD: minimap2 -ax asm20 --eqx -H -f 100 -r1k,10k --rmq=no -t 1 exp/multiple_runs11/tmp/L10-1/is_deleted/L10-1_genome.deleted_IS.0.fasta exp/multiple_runs11/tmp/L10-1/is_deleted/L10-1_genome.deleted_IS.1.fasta\n",
      "[M::main] Real time: 0.653 sec; CPU: 0.647 sec; Peak RSS: 0.133 GB\n",
      "local_variation - WARNING - Finished syri\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "invOut.txt  is empty. Skipping analysing it.\n",
      "TLOut.txt  is empty. Skipping analysing it.\n",
      "invTLOut.txt  is empty. Skipping analysing it.\n",
      "ctxOut.txt is empty. Skipping analysing it.\n",
      "inv Out.txt is empty. Skipping analysing it.\n",
      "TL Out.txt is empty. Skipping analysing it.\n",
      "invTL Out.txt is empty. Skipping analysing it.\n",
      "dup Out.txt is empty. Skipping analysing it.\n",
      "invDup Out.txt is empty. Skipping analysing it.\n",
      "ctxOut.txt is empty. Skipping analysing it.\n",
      "invOut.txt  is empty. Skipping analysing it.\n",
      "TLOut.txt  is empty. Skipping analysing it.\n",
      "invTLOut.txt  is empty. Skipping analysing it.\n",
      "ctxOut.txt is empty. Skipping analysing it.\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "[M::mm_idx_gen::0.057*1.01] collected minimizers\n",
      "[M::mm_idx_gen::0.076*1.00] sorted minimizers\n",
      "[M::main::0.076*1.00] loaded/built the index for 1 target sequence(s)\n",
      "[M::mm_mapopt_update::0.076*1.00] mid_occ = 100\n",
      "[M::mm_idx_stat] kmer size: 19; skip: 10; is_hpc: 1; #seq: 1\n",
      "[M::mm_idx_stat::0.080*1.00] distinct minimizers: 525478 (99.34% are singletons); average occurrences: 1.015; average spacing: 7.461; total length: 3977333\n",
      "[M::worker_pipeline::0.641*0.99] mapped 1 sequences\n",
      "[M::main] Version: 2.19-r1057\n",
      "[M::main] CMD: minimap2 -ax asm20 --eqx -H -f 100 -r1k,10k --rmq=no -t 1 exp/multiple_runs11/tmp/L10-1/is_deleted/L10-1_genome.deleted_IS.0.fasta exp/multiple_runs11/tmp/L10-1/is_deleted/L10-1_genome.deleted_IS.2.fasta\n",
      "[M::main] Real time: 0.643 sec; CPU: 0.637 sec; Peak RSS: 0.132 GB\n",
      "local_variation - WARNING - Finished syri\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "invOut.txt  is empty. Skipping analysing it.\n",
      "TLOut.txt  is empty. Skipping analysing it.\n",
      "invTLOut.txt  is empty. Skipping analysing it.\n",
      "ctxOut.txt is empty. Skipping analysing it.\n",
      "inv Out.txt is empty. Skipping analysing it.\n",
      "TL Out.txt is empty. Skipping analysing it.\n",
      "invTL Out.txt is empty. Skipping analysing it.\n",
      "dup Out.txt is empty. Skipping analysing it.\n",
      "invDup Out.txt is empty. Skipping analysing it.\n",
      "ctxOut.txt is empty. Skipping analysing it.\n",
      "invOut.txt  is empty. Skipping analysing it.\n",
      "TLOut.txt  is empty. Skipping analysing it.\n",
      "invTLOut.txt  is empty. Skipping analysing it.\n",
      "ctxOut.txt is empty. Skipping analysing it.\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "[M::mm_idx_gen::0.069*1.01] collected minimizers\n",
      "[M::mm_idx_gen::0.088*1.01] sorted minimizers\n",
      "[M::main::0.088*1.01] loaded/built the index for 1 target sequence(s)\n",
      "[M::mm_mapopt_update::0.088*1.01] mid_occ = 100\n",
      "[M::mm_idx_stat] kmer size: 19; skip: 10; is_hpc: 1; #seq: 1\n",
      "[M::mm_idx_stat::0.092*1.01] distinct minimizers: 525478 (99.34% are singletons); average occurrences: 1.015; average spacing: 7.461; total length: 3977333\n",
      "[M::worker_pipeline::0.649*0.99] mapped 1 sequences\n",
      "[M::main] Version: 2.19-r1057\n",
      "[M::main] CMD: minimap2 -ax asm20 --eqx -H -f 100 -r1k,10k --rmq=no -t 1 exp/multiple_runs11/tmp/L10-1/is_deleted/L10-1_genome.deleted_IS.0.fasta exp/multiple_runs11/tmp/L10-1/is_deleted/L10-1_genome.deleted_IS.3.fasta\n",
      "[M::main] Real time: 0.653 sec; CPU: 0.646 sec; Peak RSS: 0.132 GB\n",
      "local_variation - WARNING - Finished syri\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "invOut.txt  is empty. Skipping analysing it.\n",
      "TLOut.txt  is empty. Skipping analysing it.\n",
      "invTLOut.txt  is empty. Skipping analysing it.\n",
      "ctxOut.txt is empty. Skipping analysing it.\n",
      "inv Out.txt is empty. Skipping analysing it.\n",
      "TL Out.txt is empty. Skipping analysing it.\n",
      "invTL Out.txt is empty. Skipping analysing it.\n",
      "dup Out.txt is empty. Skipping analysing it.\n",
      "invDup Out.txt is empty. Skipping analysing it.\n",
      "ctxOut.txt is empty. Skipping analysing it.\n",
      "invOut.txt  is empty. Skipping analysing it.\n",
      "TLOut.txt  is empty. Skipping analysing it.\n",
      "invTLOut.txt  is empty. Skipping analysing it.\n",
      "ctxOut.txt is empty. Skipping analysing it.\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "[M::mm_idx_gen::0.058*1.01] collected minimizers\n",
      "[M::mm_idx_gen::0.076*1.00] sorted minimizers\n",
      "[M::main::0.076*1.00] loaded/built the index for 1 target sequence(s)\n",
      "[M::mm_mapopt_update::0.076*1.00] mid_occ = 100\n",
      "[M::mm_idx_stat] kmer size: 19; skip: 10; is_hpc: 1; #seq: 1\n",
      "[M::mm_idx_stat::0.080*1.00] distinct minimizers: 525478 (99.34% are singletons); average occurrences: 1.015; average spacing: 7.461; total length: 3977333\n",
      "[M::worker_pipeline::0.654*0.99] mapped 1 sequences\n",
      "[M::main] Version: 2.19-r1057\n",
      "[M::main] CMD: minimap2 -ax asm20 --eqx -H -f 100 -r1k,10k --rmq=no -t 1 exp/multiple_runs11/tmp/L10-2/is_deleted/L10-2_genome.deleted_IS.0.fasta exp/multiple_runs11/tmp/L10-2/is_deleted/L10-2_genome.deleted_IS.1.fasta\n",
      "[M::main] Real time: 0.656 sec; CPU: 0.650 sec; Peak RSS: 0.133 GB\n",
      "local_variation - WARNING - Finished syri\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "invOut.txt  is empty. Skipping analysing it.\n",
      "TLOut.txt  is empty. Skipping analysing it.\n",
      "invTLOut.txt  is empty. Skipping analysing it.\n",
      "ctxOut.txt is empty. Skipping analysing it.\n",
      "inv Out.txt is empty. Skipping analysing it.\n",
      "TL Out.txt is empty. Skipping analysing it.\n",
      "invTL Out.txt is empty. Skipping analysing it.\n",
      "dup Out.txt is empty. Skipping analysing it.\n",
      "invDup Out.txt is empty. Skipping analysing it.\n",
      "ctxOut.txt is empty. Skipping analysing it.\n",
      "invOut.txt  is empty. Skipping analysing it.\n",
      "TLOut.txt  is empty. Skipping analysing it.\n",
      "invTLOut.txt  is empty. Skipping analysing it.\n",
      "ctxOut.txt is empty. Skipping analysing it.\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "[M::mm_idx_gen::0.058*1.01] collected minimizers\n",
      "[M::mm_idx_gen::0.076*1.00] sorted minimizers\n",
      "[M::main::0.076*1.00] loaded/built the index for 1 target sequence(s)\n",
      "[M::mm_mapopt_update::0.076*1.00] mid_occ = 100\n",
      "[M::mm_idx_stat] kmer size: 19; skip: 10; is_hpc: 1; #seq: 1\n",
      "[M::mm_idx_stat::0.080*1.00] distinct minimizers: 525478 (99.34% are singletons); average occurrences: 1.015; average spacing: 7.461; total length: 3977333\n",
      "[M::worker_pipeline::0.762*0.99] mapped 1 sequences\n",
      "[M::main] Version: 2.19-r1057\n",
      "[M::main] CMD: minimap2 -ax asm20 --eqx -H -f 100 -r1k,10k --rmq=no -t 1 exp/multiple_runs11/tmp/L10-2/is_deleted/L10-2_genome.deleted_IS.0.fasta exp/multiple_runs11/tmp/L10-2/is_deleted/L10-2_genome.deleted_IS.2.fasta\n",
      "[M::main] Real time: 0.765 sec; CPU: 0.759 sec; Peak RSS: 0.132 GB\n",
      "local_variation - WARNING - Finished syri\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "invOut.txt  is empty. Skipping analysing it.\n",
      "TLOut.txt  is empty. Skipping analysing it.\n",
      "invTLOut.txt  is empty. Skipping analysing it.\n",
      "ctxOut.txt is empty. Skipping analysing it.\n",
      "inv Out.txt is empty. Skipping analysing it.\n",
      "TL Out.txt is empty. Skipping analysing it.\n",
      "invTL Out.txt is empty. Skipping analysing it.\n",
      "dup Out.txt is empty. Skipping analysing it.\n",
      "ctxOut.txt is empty. Skipping analysing it.\n",
      "invOut.txt  is empty. Skipping analysing it.\n",
      "TLOut.txt  is empty. Skipping analysing it.\n",
      "invTLOut.txt  is empty. Skipping analysing it.\n",
      "ctxOut.txt is empty. Skipping analysing it.\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "[M::mm_idx_gen::0.058*1.01] collected minimizers\n",
      "[M::mm_idx_gen::0.077*1.00] sorted minimizers\n",
      "[M::main::0.077*1.00] loaded/built the index for 1 target sequence(s)\n",
      "[M::mm_mapopt_update::0.077*1.00] mid_occ = 100\n",
      "[M::mm_idx_stat] kmer size: 19; skip: 10; is_hpc: 1; #seq: 1\n",
      "[M::mm_idx_stat::0.081*1.00] distinct minimizers: 525478 (99.34% are singletons); average occurrences: 1.015; average spacing: 7.461; total length: 3977333\n",
      "[M::worker_pipeline::0.650*0.99] mapped 1 sequences\n",
      "[M::main] Version: 2.19-r1057\n",
      "[M::main] CMD: minimap2 -ax asm20 --eqx -H -f 100 -r1k,10k --rmq=no -t 1 exp/multiple_runs11/tmp/L10-2/is_deleted/L10-2_genome.deleted_IS.0.fasta exp/multiple_runs11/tmp/L10-2/is_deleted/L10-2_genome.deleted_IS.3.fasta\n",
      "[M::main] Real time: 0.652 sec; CPU: 0.646 sec; Peak RSS: 0.135 GB\n",
      "local_variation - WARNING - Finished syri\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "TLOut.txt  is empty. Skipping analysing it.\n",
      "ctxOut.txt is empty. Skipping analysing it.\n",
      "TL Out.txt is empty. Skipping analysing it.\n",
      "dup Out.txt is empty. Skipping analysing it.\n",
      "invDup Out.txt is empty. Skipping analysing it.\n",
      "ctxOut.txt is empty. Skipping analysing it.\n",
      "TLOut.txt  is empty. Skipping analysing it.\n",
      "ctxOut.txt is empty. Skipping analysing it.\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "[M::mm_idx_gen::0.058*1.01] collected minimizers\n",
      "[M::mm_idx_gen::0.077*1.00] sorted minimizers\n",
      "[M::main::0.077*1.00] loaded/built the index for 1 target sequence(s)\n",
      "[M::mm_mapopt_update::0.077*1.00] mid_occ = 100\n",
      "[M::mm_idx_stat] kmer size: 19; skip: 10; is_hpc: 1; #seq: 1\n",
      "[M::mm_idx_stat::0.081*1.00] distinct minimizers: 525478 (99.34% are singletons); average occurrences: 1.015; average spacing: 7.461; total length: 3977333\n",
      "[M::worker_pipeline::0.653*0.99] mapped 1 sequences\n",
      "[M::main] Version: 2.19-r1057\n",
      "[M::main] CMD: minimap2 -ax asm20 --eqx -H -f 100 -r1k,10k --rmq=no -t 1 exp/multiple_runs11/tmp/L10-3/is_deleted/L10-3_genome.deleted_IS.0.fasta exp/multiple_runs11/tmp/L10-3/is_deleted/L10-3_genome.deleted_IS.1.fasta\n",
      "[M::main] Real time: 0.655 sec; CPU: 0.649 sec; Peak RSS: 0.133 GB\n",
      "local_variation - WARNING - Finished syri\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "invOut.txt  is empty. Skipping analysing it.\n",
      "TLOut.txt  is empty. Skipping analysing it.\n",
      "invTLOut.txt  is empty. Skipping analysing it.\n",
      "ctxOut.txt is empty. Skipping analysing it.\n",
      "inv Out.txt is empty. Skipping analysing it.\n",
      "TL Out.txt is empty. Skipping analysing it.\n",
      "invTL Out.txt is empty. Skipping analysing it.\n",
      "dup Out.txt is empty. Skipping analysing it.\n",
      "invDup Out.txt is empty. Skipping analysing it.\n",
      "ctxOut.txt is empty. Skipping analysing it.\n",
      "invOut.txt  is empty. Skipping analysing it.\n",
      "TLOut.txt  is empty. Skipping analysing it.\n",
      "invTLOut.txt  is empty. Skipping analysing it.\n",
      "ctxOut.txt is empty. Skipping analysing it.\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "[M::mm_idx_gen::0.058*1.01] collected minimizers\n",
      "[M::mm_idx_gen::0.076*1.00] sorted minimizers\n",
      "[M::main::0.076*1.00] loaded/built the index for 1 target sequence(s)\n",
      "[M::mm_mapopt_update::0.076*1.00] mid_occ = 100\n",
      "[M::mm_idx_stat] kmer size: 19; skip: 10; is_hpc: 1; #seq: 1\n",
      "[M::mm_idx_stat::0.080*1.00] distinct minimizers: 525478 (99.34% are singletons); average occurrences: 1.015; average spacing: 7.461; total length: 3977333\n",
      "[M::worker_pipeline::0.658*0.99] mapped 1 sequences\n",
      "[M::main] Version: 2.19-r1057\n",
      "[M::main] CMD: minimap2 -ax asm20 --eqx -H -f 100 -r1k,10k --rmq=no -t 1 exp/multiple_runs11/tmp/L10-3/is_deleted/L10-3_genome.deleted_IS.0.fasta exp/multiple_runs11/tmp/L10-3/is_deleted/L10-3_genome.deleted_IS.2.fasta\n",
      "[M::main] Real time: 0.660 sec; CPU: 0.653 sec; Peak RSS: 0.133 GB\n",
      "local_variation - WARNING - Finished syri\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "TLOut.txt  is empty. Skipping analysing it.\n",
      "invTLOut.txt  is empty. Skipping analysing it.\n",
      "ctxOut.txt is empty. Skipping analysing it.\n",
      "TL Out.txt is empty. Skipping analysing it.\n",
      "invTL Out.txt is empty. Skipping analysing it.\n",
      "dup Out.txt is empty. Skipping analysing it.\n",
      "invDup Out.txt is empty. Skipping analysing it.\n",
      "ctxOut.txt is empty. Skipping analysing it.\n",
      "TLOut.txt  is empty. Skipping analysing it.\n",
      "invTLOut.txt  is empty. Skipping analysing it.\n",
      "ctxOut.txt is empty. Skipping analysing it.\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "[M::mm_idx_gen::0.058*1.01] collected minimizers\n",
      "[M::mm_idx_gen::0.077*1.00] sorted minimizers\n",
      "[M::main::0.077*1.00] loaded/built the index for 1 target sequence(s)\n",
      "[M::mm_mapopt_update::0.077*1.00] mid_occ = 100\n",
      "[M::mm_idx_stat] kmer size: 19; skip: 10; is_hpc: 1; #seq: 1\n",
      "[M::mm_idx_stat::0.081*1.00] distinct minimizers: 525478 (99.34% are singletons); average occurrences: 1.015; average spacing: 7.461; total length: 3977333\n",
      "[M::worker_pipeline::0.654*0.99] mapped 1 sequences\n",
      "[M::main] Version: 2.19-r1057\n",
      "[M::main] CMD: minimap2 -ax asm20 --eqx -H -f 100 -r1k,10k --rmq=no -t 1 exp/multiple_runs11/tmp/L10-3/is_deleted/L10-3_genome.deleted_IS.0.fasta exp/multiple_runs11/tmp/L10-3/is_deleted/L10-3_genome.deleted_IS.3.fasta\n",
      "[M::main] Real time: 0.657 sec; CPU: 0.650 sec; Peak RSS: 0.131 GB\n",
      "local_variation - WARNING - Finished syri\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "TLOut.txt  is empty. Skipping analysing it.\n",
      "invTLOut.txt  is empty. Skipping analysing it.\n",
      "ctxOut.txt is empty. Skipping analysing it.\n",
      "TL Out.txt is empty. Skipping analysing it.\n",
      "invTL Out.txt is empty. Skipping analysing it.\n",
      "dup Out.txt is empty. Skipping analysing it.\n",
      "invDup Out.txt is empty. Skipping analysing it.\n",
      "ctxOut.txt is empty. Skipping analysing it.\n",
      "TLOut.txt  is empty. Skipping analysing it.\n",
      "invTLOut.txt  is empty. Skipping analysing it.\n",
      "ctxOut.txt is empty. Skipping analysing it.\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "[M::mm_idx_gen::0.058*1.01] collected minimizers\n",
      "[M::mm_idx_gen::0.077*1.00] sorted minimizers\n",
      "[M::main::0.077*1.00] loaded/built the index for 1 target sequence(s)\n",
      "[M::mm_mapopt_update::0.077*1.00] mid_occ = 100\n",
      "[M::mm_idx_stat] kmer size: 19; skip: 10; is_hpc: 1; #seq: 1\n",
      "[M::mm_idx_stat::0.081*1.00] distinct minimizers: 525478 (99.34% are singletons); average occurrences: 1.015; average spacing: 7.461; total length: 3977333\n",
      "[M::worker_pipeline::0.652*0.99] mapped 1 sequences\n",
      "[M::main] Version: 2.19-r1057\n",
      "[M::main] CMD: minimap2 -ax asm20 --eqx -H -f 100 -r1k,10k --rmq=no -t 1 exp/multiple_runs11/tmp/L10-4/is_deleted/L10-4_genome.deleted_IS.0.fasta exp/multiple_runs11/tmp/L10-4/is_deleted/L10-4_genome.deleted_IS.1.fasta\n",
      "[M::main] Real time: 0.654 sec; CPU: 0.649 sec; Peak RSS: 0.133 GB\n",
      "local_variation - WARNING - Finished syri\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "invOut.txt  is empty. Skipping analysing it.\n",
      "TLOut.txt  is empty. Skipping analysing it.\n",
      "invTLOut.txt  is empty. Skipping analysing it.\n",
      "ctxOut.txt is empty. Skipping analysing it.\n",
      "inv Out.txt is empty. Skipping analysing it.\n",
      "TL Out.txt is empty. Skipping analysing it.\n",
      "invTL Out.txt is empty. Skipping analysing it.\n",
      "dup Out.txt is empty. Skipping analysing it.\n",
      "invDup Out.txt is empty. Skipping analysing it.\n",
      "ctxOut.txt is empty. Skipping analysing it.\n",
      "invOut.txt  is empty. Skipping analysing it.\n",
      "TLOut.txt  is empty. Skipping analysing it.\n",
      "invTLOut.txt  is empty. Skipping analysing it.\n",
      "ctxOut.txt is empty. Skipping analysing it.\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "[M::mm_idx_gen::0.058*1.01] collected minimizers\n",
      "[M::mm_idx_gen::0.076*1.00] sorted minimizers\n",
      "[M::main::0.077*1.00] loaded/built the index for 1 target sequence(s)\n",
      "[M::mm_mapopt_update::0.077*1.00] mid_occ = 100\n",
      "[M::mm_idx_stat] kmer size: 19; skip: 10; is_hpc: 1; #seq: 1\n",
      "[M::mm_idx_stat::0.080*1.00] distinct minimizers: 525478 (99.34% are singletons); average occurrences: 1.015; average spacing: 7.461; total length: 3977333\n",
      "[M::worker_pipeline::0.650*0.99] mapped 1 sequences\n",
      "[M::main] Version: 2.19-r1057\n",
      "[M::main] CMD: minimap2 -ax asm20 --eqx -H -f 100 -r1k,10k --rmq=no -t 1 exp/multiple_runs11/tmp/L10-4/is_deleted/L10-4_genome.deleted_IS.0.fasta exp/multiple_runs11/tmp/L10-4/is_deleted/L10-4_genome.deleted_IS.2.fasta\n",
      "[M::main] Real time: 0.652 sec; CPU: 0.646 sec; Peak RSS: 0.133 GB\n",
      "local_variation - WARNING - Finished syri\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "TLOut.txt  is empty. Skipping analysing it.\n",
      "invTLOut.txt  is empty. Skipping analysing it.\n",
      "ctxOut.txt is empty. Skipping analysing it.\n",
      "TL Out.txt is empty. Skipping analysing it.\n",
      "invTL Out.txt is empty. Skipping analysing it.\n",
      "dup Out.txt is empty. Skipping analysing it.\n",
      "invDup Out.txt is empty. Skipping analysing it.\n",
      "ctxOut.txt is empty. Skipping analysing it.\n",
      "TLOut.txt  is empty. Skipping analysing it.\n",
      "invTLOut.txt  is empty. Skipping analysing it.\n",
      "ctxOut.txt is empty. Skipping analysing it.\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "[M::mm_idx_gen::0.058*1.01] collected minimizers\n",
      "[M::mm_idx_gen::0.076*1.00] sorted minimizers\n",
      "[M::main::0.076*1.00] loaded/built the index for 1 target sequence(s)\n",
      "[M::mm_mapopt_update::0.076*1.00] mid_occ = 100\n",
      "[M::mm_idx_stat] kmer size: 19; skip: 10; is_hpc: 1; #seq: 1\n",
      "[M::mm_idx_stat::0.080*1.00] distinct minimizers: 525478 (99.34% are singletons); average occurrences: 1.015; average spacing: 7.461; total length: 3977333\n",
      "[M::worker_pipeline::0.656*0.99] mapped 1 sequences\n",
      "[M::main] Version: 2.19-r1057\n",
      "[M::main] CMD: minimap2 -ax asm20 --eqx -H -f 100 -r1k,10k --rmq=no -t 1 exp/multiple_runs11/tmp/L10-4/is_deleted/L10-4_genome.deleted_IS.0.fasta exp/multiple_runs11/tmp/L10-4/is_deleted/L10-4_genome.deleted_IS.3.fasta\n",
      "[M::main] Real time: 0.659 sec; CPU: 0.653 sec; Peak RSS: 0.134 GB\n",
      "local_variation - WARNING - Finished syri\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "TLOut.txt  is empty. Skipping analysing it.\n",
      "invTLOut.txt  is empty. Skipping analysing it.\n",
      "ctxOut.txt is empty. Skipping analysing it.\n",
      "TL Out.txt is empty. Skipping analysing it.\n",
      "invTL Out.txt is empty. Skipping analysing it.\n",
      "invDup Out.txt is empty. Skipping analysing it.\n",
      "ctxOut.txt is empty. Skipping analysing it.\n",
      "TLOut.txt  is empty. Skipping analysing it.\n",
      "invTLOut.txt  is empty. Skipping analysing it.\n",
      "ctxOut.txt is empty. Skipping analysing it.\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "[M::mm_idx_gen::0.058*1.01] collected minimizers\n",
      "[M::mm_idx_gen::0.077*1.00] sorted minimizers\n",
      "[M::main::0.077*1.00] loaded/built the index for 1 target sequence(s)\n",
      "[M::mm_mapopt_update::0.077*1.00] mid_occ = 100\n",
      "[M::mm_idx_stat] kmer size: 19; skip: 10; is_hpc: 1; #seq: 1\n",
      "[M::mm_idx_stat::0.081*1.00] distinct minimizers: 525478 (99.34% are singletons); average occurrences: 1.015; average spacing: 7.461; total length: 3977333\n",
      "[M::worker_pipeline::0.658*0.99] mapped 1 sequences\n",
      "[M::main] Version: 2.19-r1057\n",
      "[M::main] CMD: minimap2 -ax asm20 --eqx -H -f 100 -r1k,10k --rmq=no -t 1 exp/multiple_runs11/tmp/L11-1/is_deleted/L11-1_genome.deleted_IS.0.fasta exp/multiple_runs11/tmp/L11-1/is_deleted/L11-1_genome.deleted_IS.1.fasta\n",
      "[M::main] Real time: 0.661 sec; CPU: 0.654 sec; Peak RSS: 0.173 GB\n",
      "local_variation - WARNING - Finished syri\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "invOut.txt  is empty. Skipping analysing it.\n",
      "TLOut.txt  is empty. Skipping analysing it.\n",
      "invTLOut.txt  is empty. Skipping analysing it.\n",
      "ctxOut.txt is empty. Skipping analysing it.\n",
      "inv Out.txt is empty. Skipping analysing it.\n",
      "TL Out.txt is empty. Skipping analysing it.\n",
      "invTL Out.txt is empty. Skipping analysing it.\n",
      "dup Out.txt is empty. Skipping analysing it.\n",
      "invDup Out.txt is empty. Skipping analysing it.\n",
      "ctxOut.txt is empty. Skipping analysing it.\n",
      "invOut.txt  is empty. Skipping analysing it.\n",
      "TLOut.txt  is empty. Skipping analysing it.\n",
      "invTLOut.txt  is empty. Skipping analysing it.\n",
      "ctxOut.txt is empty. Skipping analysing it.\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "[M::mm_idx_gen::0.058*1.01] collected minimizers\n",
      "[M::mm_idx_gen::0.076*1.00] sorted minimizers\n",
      "[M::main::0.076*1.00] loaded/built the index for 1 target sequence(s)\n",
      "[M::mm_mapopt_update::0.076*1.00] mid_occ = 100\n",
      "[M::mm_idx_stat] kmer size: 19; skip: 10; is_hpc: 1; #seq: 1\n",
      "[M::mm_idx_stat::0.080*1.00] distinct minimizers: 525478 (99.34% are singletons); average occurrences: 1.015; average spacing: 7.461; total length: 3977333\n",
      "[M::worker_pipeline::0.673*0.99] mapped 1 sequences\n",
      "[M::main] Version: 2.19-r1057\n",
      "[M::main] CMD: minimap2 -ax asm20 --eqx -H -f 100 -r1k,10k --rmq=no -t 1 exp/multiple_runs11/tmp/L11-1/is_deleted/L11-1_genome.deleted_IS.0.fasta exp/multiple_runs11/tmp/L11-1/is_deleted/L11-1_genome.deleted_IS.2.fasta\n",
      "[M::main] Real time: 0.675 sec; CPU: 0.670 sec; Peak RSS: 0.170 GB\n",
      "local_variation - WARNING - Finished syri\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "invOut.txt  is empty. Skipping analysing it.\n",
      "TLOut.txt  is empty. Skipping analysing it.\n",
      "invTLOut.txt  is empty. Skipping analysing it.\n",
      "ctxOut.txt is empty. Skipping analysing it.\n",
      "inv Out.txt is empty. Skipping analysing it.\n",
      "TL Out.txt is empty. Skipping analysing it.\n",
      "invTL Out.txt is empty. Skipping analysing it.\n",
      "dup Out.txt is empty. Skipping analysing it.\n",
      "invDup Out.txt is empty. Skipping analysing it.\n",
      "ctxOut.txt is empty. Skipping analysing it.\n",
      "invOut.txt  is empty. Skipping analysing it.\n",
      "TLOut.txt  is empty. Skipping analysing it.\n",
      "invTLOut.txt  is empty. Skipping analysing it.\n",
      "ctxOut.txt is empty. Skipping analysing it.\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "[M::mm_idx_gen::0.062*1.01] collected minimizers\n",
      "[M::mm_idx_gen::0.081*1.00] sorted minimizers\n",
      "[M::main::0.081*1.00] loaded/built the index for 1 target sequence(s)\n",
      "[M::mm_mapopt_update::0.081*1.00] mid_occ = 100\n",
      "[M::mm_idx_stat] kmer size: 19; skip: 10; is_hpc: 1; #seq: 1\n",
      "[M::mm_idx_stat::0.085*1.00] distinct minimizers: 525478 (99.34% are singletons); average occurrences: 1.015; average spacing: 7.461; total length: 3977333\n",
      "[M::worker_pipeline::0.708*0.99] mapped 1 sequences\n",
      "[M::main] Version: 2.19-r1057\n",
      "[M::main] CMD: minimap2 -ax asm20 --eqx -H -f 100 -r1k,10k --rmq=no -t 1 exp/multiple_runs11/tmp/L11-1/is_deleted/L11-1_genome.deleted_IS.0.fasta exp/multiple_runs11/tmp/L11-1/is_deleted/L11-1_genome.deleted_IS.3.fasta\n",
      "[M::main] Real time: 0.710 sec; CPU: 0.704 sec; Peak RSS: 0.266 GB\n",
      "local_variation - WARNING - Finished syri\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "invOut.txt  is empty. Skipping analysing it.\n",
      "TLOut.txt  is empty. Skipping analysing it.\n",
      "invTLOut.txt  is empty. Skipping analysing it.\n",
      "ctxOut.txt is empty. Skipping analysing it.\n",
      "inv Out.txt is empty. Skipping analysing it.\n",
      "TL Out.txt is empty. Skipping analysing it.\n",
      "invTL Out.txt is empty. Skipping analysing it.\n",
      "dup Out.txt is empty. Skipping analysing it.\n",
      "invDup Out.txt is empty. Skipping analysing it.\n",
      "ctxOut.txt is empty. Skipping analysing it.\n",
      "invOut.txt  is empty. Skipping analysing it.\n",
      "TLOut.txt  is empty. Skipping analysing it.\n",
      "invTLOut.txt  is empty. Skipping analysing it.\n",
      "ctxOut.txt is empty. Skipping analysing it.\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "[M::mm_idx_gen::0.064*1.01] collected minimizers\n",
      "[M::mm_idx_gen::0.083*1.01] sorted minimizers\n",
      "[M::main::0.083*1.01] loaded/built the index for 1 target sequence(s)\n",
      "[M::mm_mapopt_update::0.083*1.01] mid_occ = 100\n",
      "[M::mm_idx_stat] kmer size: 19; skip: 10; is_hpc: 1; #seq: 1\n",
      "[M::mm_idx_stat::0.087*1.01] distinct minimizers: 525478 (99.34% are singletons); average occurrences: 1.015; average spacing: 7.461; total length: 3977333\n",
      "[M::worker_pipeline::0.663*0.99] mapped 1 sequences\n",
      "[M::main] Version: 2.19-r1057\n",
      "[M::main] CMD: minimap2 -ax asm20 --eqx -H -f 100 -r1k,10k --rmq=no -t 1 exp/multiple_runs11/tmp/L11-2/is_deleted/L11-2_genome.deleted_IS.0.fasta exp/multiple_runs11/tmp/L11-2/is_deleted/L11-2_genome.deleted_IS.1.fasta\n",
      "[M::main] Real time: 0.666 sec; CPU: 0.660 sec; Peak RSS: 0.173 GB\n",
      "local_variation - WARNING - Finished syri\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "invOut.txt  is empty. Skipping analysing it.\n",
      "TLOut.txt  is empty. Skipping analysing it.\n",
      "invTLOut.txt  is empty. Skipping analysing it.\n",
      "ctxOut.txt is empty. Skipping analysing it.\n",
      "inv Out.txt is empty. Skipping analysing it.\n",
      "TL Out.txt is empty. Skipping analysing it.\n",
      "invTL Out.txt is empty. Skipping analysing it.\n",
      "dup Out.txt is empty. Skipping analysing it.\n",
      "invDup Out.txt is empty. Skipping analysing it.\n",
      "ctxOut.txt is empty. Skipping analysing it.\n",
      "invOut.txt  is empty. Skipping analysing it.\n",
      "TLOut.txt  is empty. Skipping analysing it.\n",
      "invTLOut.txt  is empty. Skipping analysing it.\n",
      "ctxOut.txt is empty. Skipping analysing it.\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "[M::mm_idx_gen::0.063*1.01] collected minimizers\n",
      "[M::mm_idx_gen::0.081*1.00] sorted minimizers\n",
      "[M::main::0.081*1.00] loaded/built the index for 1 target sequence(s)\n",
      "[M::mm_mapopt_update::0.081*1.00] mid_occ = 100\n",
      "[M::mm_idx_stat] kmer size: 19; skip: 10; is_hpc: 1; #seq: 1\n",
      "[M::mm_idx_stat::0.085*1.00] distinct minimizers: 525478 (99.34% are singletons); average occurrences: 1.015; average spacing: 7.461; total length: 3977333\n",
      "[M::worker_pipeline::0.696*0.99] mapped 1 sequences\n",
      "[M::main] Version: 2.19-r1057\n",
      "[M::main] CMD: minimap2 -ax asm20 --eqx -H -f 100 -r1k,10k --rmq=no -t 1 exp/multiple_runs11/tmp/L11-2/is_deleted/L11-2_genome.deleted_IS.0.fasta exp/multiple_runs11/tmp/L11-2/is_deleted/L11-2_genome.deleted_IS.2.fasta\n",
      "[M::main] Real time: 0.698 sec; CPU: 0.692 sec; Peak RSS: 0.215 GB\n",
      "local_variation - WARNING - Finished syri\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "invOut.txt  is empty. Skipping analysing it.\n",
      "TLOut.txt  is empty. Skipping analysing it.\n",
      "invTLOut.txt  is empty. Skipping analysing it.\n",
      "ctxOut.txt is empty. Skipping analysing it.\n",
      "inv Out.txt is empty. Skipping analysing it.\n",
      "TL Out.txt is empty. Skipping analysing it.\n",
      "invTL Out.txt is empty. Skipping analysing it.\n",
      "ctxOut.txt is empty. Skipping analysing it.\n",
      "invOut.txt  is empty. Skipping analysing it.\n",
      "TLOut.txt  is empty. Skipping analysing it.\n",
      "invTLOut.txt  is empty. Skipping analysing it.\n",
      "ctxOut.txt is empty. Skipping analysing it.\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "[M::mm_idx_gen::0.063*1.01] collected minimizers\n",
      "[M::mm_idx_gen::0.082*1.00] sorted minimizers\n",
      "[M::main::0.082*1.00] loaded/built the index for 1 target sequence(s)\n",
      "[M::mm_mapopt_update::0.082*1.00] mid_occ = 100\n",
      "[M::mm_idx_stat] kmer size: 19; skip: 10; is_hpc: 1; #seq: 1\n",
      "[M::mm_idx_stat::0.086*1.00] distinct minimizers: 525478 (99.34% are singletons); average occurrences: 1.015; average spacing: 7.461; total length: 3977333\n",
      "[M::worker_pipeline::0.703*0.99] mapped 1 sequences\n",
      "[M::main] Version: 2.19-r1057\n",
      "[M::main] CMD: minimap2 -ax asm20 --eqx -H -f 100 -r1k,10k --rmq=no -t 1 exp/multiple_runs11/tmp/L11-2/is_deleted/L11-2_genome.deleted_IS.0.fasta exp/multiple_runs11/tmp/L11-2/is_deleted/L11-2_genome.deleted_IS.3.fasta\n",
      "[M::main] Real time: 0.705 sec; CPU: 0.698 sec; Peak RSS: 0.217 GB\n",
      "local_variation - WARNING - Finished syri\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "TLOut.txt  is empty. Skipping analysing it.\n",
      "invTLOut.txt  is empty. Skipping analysing it.\n",
      "ctxOut.txt is empty. Skipping analysing it.\n",
      "TL Out.txt is empty. Skipping analysing it.\n",
      "invTL Out.txt is empty. Skipping analysing it.\n",
      "ctxOut.txt is empty. Skipping analysing it.\n",
      "TLOut.txt  is empty. Skipping analysing it.\n",
      "invTLOut.txt  is empty. Skipping analysing it.\n",
      "ctxOut.txt is empty. Skipping analysing it.\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "[M::mm_idx_gen::0.058*1.01] collected minimizers\n",
      "[M::mm_idx_gen::0.077*1.00] sorted minimizers\n",
      "[M::main::0.077*1.00] loaded/built the index for 1 target sequence(s)\n",
      "[M::mm_mapopt_update::0.077*1.00] mid_occ = 100\n",
      "[M::mm_idx_stat] kmer size: 19; skip: 10; is_hpc: 1; #seq: 1\n",
      "[M::mm_idx_stat::0.081*1.00] distinct minimizers: 525478 (99.34% are singletons); average occurrences: 1.015; average spacing: 7.461; total length: 3977333\n",
      "[M::worker_pipeline::0.658*0.99] mapped 1 sequences\n",
      "[M::main] Version: 2.19-r1057\n",
      "[M::main] CMD: minimap2 -ax asm20 --eqx -H -f 100 -r1k,10k --rmq=no -t 1 exp/multiple_runs11/tmp/L11-3/is_deleted/L11-3_genome.deleted_IS.0.fasta exp/multiple_runs11/tmp/L11-3/is_deleted/L11-3_genome.deleted_IS.1.fasta\n",
      "[M::main] Real time: 0.661 sec; CPU: 0.655 sec; Peak RSS: 0.173 GB\n",
      "local_variation - WARNING - Finished syri\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "invOut.txt  is empty. Skipping analysing it.\n",
      "TLOut.txt  is empty. Skipping analysing it.\n",
      "invTLOut.txt  is empty. Skipping analysing it.\n",
      "ctxOut.txt is empty. Skipping analysing it.\n",
      "inv Out.txt is empty. Skipping analysing it.\n",
      "TL Out.txt is empty. Skipping analysing it.\n",
      "invTL Out.txt is empty. Skipping analysing it.\n",
      "dup Out.txt is empty. Skipping analysing it.\n",
      "invDup Out.txt is empty. Skipping analysing it.\n",
      "ctxOut.txt is empty. Skipping analysing it.\n",
      "invOut.txt  is empty. Skipping analysing it.\n",
      "TLOut.txt  is empty. Skipping analysing it.\n",
      "invTLOut.txt  is empty. Skipping analysing it.\n",
      "ctxOut.txt is empty. Skipping analysing it.\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "[M::mm_idx_gen::0.062*1.01] collected minimizers\n",
      "[M::mm_idx_gen::0.080*1.00] sorted minimizers\n",
      "[M::main::0.080*1.00] loaded/built the index for 1 target sequence(s)\n",
      "[M::mm_mapopt_update::0.080*1.00] mid_occ = 100\n",
      "[M::mm_idx_stat] kmer size: 19; skip: 10; is_hpc: 1; #seq: 1\n",
      "[M::mm_idx_stat::0.084*1.00] distinct minimizers: 525478 (99.34% are singletons); average occurrences: 1.015; average spacing: 7.461; total length: 3977333\n",
      "[M::worker_pipeline::0.674*0.99] mapped 1 sequences\n",
      "[M::main] Version: 2.19-r1057\n",
      "[M::main] CMD: minimap2 -ax asm20 --eqx -H -f 100 -r1k,10k --rmq=no -t 1 exp/multiple_runs11/tmp/L11-3/is_deleted/L11-3_genome.deleted_IS.0.fasta exp/multiple_runs11/tmp/L11-3/is_deleted/L11-3_genome.deleted_IS.2.fasta\n",
      "[M::main] Real time: 0.677 sec; CPU: 0.671 sec; Peak RSS: 0.170 GB\n",
      "local_variation - WARNING - Finished syri\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "invOut.txt  is empty. Skipping analysing it.\n",
      "TLOut.txt  is empty. Skipping analysing it.\n",
      "invTLOut.txt  is empty. Skipping analysing it.\n",
      "ctxOut.txt is empty. Skipping analysing it.\n",
      "inv Out.txt is empty. Skipping analysing it.\n",
      "TL Out.txt is empty. Skipping analysing it.\n",
      "invTL Out.txt is empty. Skipping analysing it.\n",
      "dup Out.txt is empty. Skipping analysing it.\n",
      "invDup Out.txt is empty. Skipping analysing it.\n",
      "ctxOut.txt is empty. Skipping analysing it.\n",
      "invOut.txt  is empty. Skipping analysing it.\n",
      "TLOut.txt  is empty. Skipping analysing it.\n",
      "invTLOut.txt  is empty. Skipping analysing it.\n",
      "ctxOut.txt is empty. Skipping analysing it.\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "[M::mm_idx_gen::0.058*1.00] collected minimizers\n",
      "[M::mm_idx_gen::0.077*1.00] sorted minimizers\n",
      "[M::main::0.077*1.00] loaded/built the index for 1 target sequence(s)\n",
      "[M::mm_mapopt_update::0.077*1.00] mid_occ = 100\n",
      "[M::mm_idx_stat] kmer size: 19; skip: 10; is_hpc: 1; #seq: 1\n",
      "[M::mm_idx_stat::0.081*1.00] distinct minimizers: 525478 (99.34% are singletons); average occurrences: 1.015; average spacing: 7.461; total length: 3977333\n",
      "[M::worker_pipeline::0.674*0.99] mapped 1 sequences\n",
      "[M::main] Version: 2.19-r1057\n",
      "[M::main] CMD: minimap2 -ax asm20 --eqx -H -f 100 -r1k,10k --rmq=no -t 1 exp/multiple_runs11/tmp/L11-3/is_deleted/L11-3_genome.deleted_IS.0.fasta exp/multiple_runs11/tmp/L11-3/is_deleted/L11-3_genome.deleted_IS.3.fasta\n",
      "[M::main] Real time: 0.676 sec; CPU: 0.670 sec; Peak RSS: 0.170 GB\n",
      "local_variation - WARNING - Finished syri\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "invOut.txt  is empty. Skipping analysing it.\n",
      "TLOut.txt  is empty. Skipping analysing it.\n",
      "invTLOut.txt  is empty. Skipping analysing it.\n",
      "ctxOut.txt is empty. Skipping analysing it.\n",
      "inv Out.txt is empty. Skipping analysing it.\n",
      "TL Out.txt is empty. Skipping analysing it.\n",
      "invTL Out.txt is empty. Skipping analysing it.\n",
      "dup Out.txt is empty. Skipping analysing it.\n",
      "invDup Out.txt is empty. Skipping analysing it.\n",
      "ctxOut.txt is empty. Skipping analysing it.\n",
      "invOut.txt  is empty. Skipping analysing it.\n",
      "TLOut.txt  is empty. Skipping analysing it.\n",
      "invTLOut.txt  is empty. Skipping analysing it.\n",
      "ctxOut.txt is empty. Skipping analysing it.\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "[M::mm_idx_gen::0.058*1.01] collected minimizers\n",
      "[M::mm_idx_gen::0.077*1.00] sorted minimizers\n",
      "[M::main::0.077*1.00] loaded/built the index for 1 target sequence(s)\n",
      "[M::mm_mapopt_update::0.077*1.00] mid_occ = 100\n",
      "[M::mm_idx_stat] kmer size: 19; skip: 10; is_hpc: 1; #seq: 1\n",
      "[M::mm_idx_stat::0.081*1.00] distinct minimizers: 525478 (99.34% are singletons); average occurrences: 1.015; average spacing: 7.461; total length: 3977333\n",
      "[M::worker_pipeline::0.658*0.99] mapped 1 sequences\n",
      "[M::main] Version: 2.19-r1057\n",
      "[M::main] CMD: minimap2 -ax asm20 --eqx -H -f 100 -r1k,10k --rmq=no -t 1 exp/multiple_runs11/tmp/L11-4/is_deleted/L11-4_genome.deleted_IS.0.fasta exp/multiple_runs11/tmp/L11-4/is_deleted/L11-4_genome.deleted_IS.1.fasta\n",
      "[M::main] Real time: 0.660 sec; CPU: 0.654 sec; Peak RSS: 0.173 GB\n",
      "local_variation - WARNING - Finished syri\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "invOut.txt  is empty. Skipping analysing it.\n",
      "TLOut.txt  is empty. Skipping analysing it.\n",
      "invTLOut.txt  is empty. Skipping analysing it.\n",
      "ctxOut.txt is empty. Skipping analysing it.\n",
      "inv Out.txt is empty. Skipping analysing it.\n",
      "TL Out.txt is empty. Skipping analysing it.\n",
      "invTL Out.txt is empty. Skipping analysing it.\n",
      "dup Out.txt is empty. Skipping analysing it.\n",
      "invDup Out.txt is empty. Skipping analysing it.\n",
      "ctxOut.txt is empty. Skipping analysing it.\n",
      "invOut.txt  is empty. Skipping analysing it.\n",
      "TLOut.txt  is empty. Skipping analysing it.\n",
      "invTLOut.txt  is empty. Skipping analysing it.\n",
      "ctxOut.txt is empty. Skipping analysing it.\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "[M::mm_idx_gen::0.058*1.01] collected minimizers\n",
      "[M::mm_idx_gen::0.076*1.00] sorted minimizers\n",
      "[M::main::0.076*1.00] loaded/built the index for 1 target sequence(s)\n",
      "[M::mm_mapopt_update::0.076*1.00] mid_occ = 100\n",
      "[M::mm_idx_stat] kmer size: 19; skip: 10; is_hpc: 1; #seq: 1\n",
      "[M::mm_idx_stat::0.080*1.00] distinct minimizers: 525478 (99.34% are singletons); average occurrences: 1.015; average spacing: 7.461; total length: 3977333\n",
      "[M::worker_pipeline::0.673*0.99] mapped 1 sequences\n",
      "[M::main] Version: 2.19-r1057\n",
      "[M::main] CMD: minimap2 -ax asm20 --eqx -H -f 100 -r1k,10k --rmq=no -t 1 exp/multiple_runs11/tmp/L11-4/is_deleted/L11-4_genome.deleted_IS.0.fasta exp/multiple_runs11/tmp/L11-4/is_deleted/L11-4_genome.deleted_IS.2.fasta\n",
      "[M::main] Real time: 0.675 sec; CPU: 0.669 sec; Peak RSS: 0.170 GB\n",
      "local_variation - WARNING - Finished syri\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "TLOut.txt  is empty. Skipping analysing it.\n",
      "invTLOut.txt  is empty. Skipping analysing it.\n",
      "ctxOut.txt is empty. Skipping analysing it.\n",
      "TL Out.txt is empty. Skipping analysing it.\n",
      "invTL Out.txt is empty. Skipping analysing it.\n",
      "dup Out.txt is empty. Skipping analysing it.\n",
      "invDup Out.txt is empty. Skipping analysing it.\n",
      "ctxOut.txt is empty. Skipping analysing it.\n",
      "TLOut.txt  is empty. Skipping analysing it.\n",
      "invTLOut.txt  is empty. Skipping analysing it.\n",
      "ctxOut.txt is empty. Skipping analysing it.\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "[M::mm_idx_gen::0.058*1.01] collected minimizers\n",
      "[M::mm_idx_gen::0.077*1.00] sorted minimizers\n",
      "[M::main::0.077*1.00] loaded/built the index for 1 target sequence(s)\n",
      "[M::mm_mapopt_update::0.077*1.00] mid_occ = 100\n",
      "[M::mm_idx_stat] kmer size: 19; skip: 10; is_hpc: 1; #seq: 1\n",
      "[M::mm_idx_stat::0.081*1.00] distinct minimizers: 525478 (99.34% are singletons); average occurrences: 1.015; average spacing: 7.461; total length: 3977333\n",
      "[M::worker_pipeline::0.707*0.99] mapped 1 sequences\n",
      "[M::main] Version: 2.19-r1057\n",
      "[M::main] CMD: minimap2 -ax asm20 --eqx -H -f 100 -r1k,10k --rmq=no -t 1 exp/multiple_runs11/tmp/L11-4/is_deleted/L11-4_genome.deleted_IS.0.fasta exp/multiple_runs11/tmp/L11-4/is_deleted/L11-4_genome.deleted_IS.3.fasta\n",
      "[M::main] Real time: 0.710 sec; CPU: 0.703 sec; Peak RSS: 0.172 GB\n",
      "local_variation - WARNING - Finished syri\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "TLOut.txt  is empty. Skipping analysing it.\n",
      "invTLOut.txt  is empty. Skipping analysing it.\n",
      "ctxOut.txt is empty. Skipping analysing it.\n",
      "TL Out.txt is empty. Skipping analysing it.\n",
      "invTL Out.txt is empty. Skipping analysing it.\n",
      "dup Out.txt is empty. Skipping analysing it.\n",
      "ctxOut.txt is empty. Skipping analysing it.\n",
      "TLOut.txt  is empty. Skipping analysing it.\n",
      "invTLOut.txt  is empty. Skipping analysing it.\n",
      "ctxOut.txt is empty. Skipping analysing it.\n"
     ]
    }
   ],
   "source": [
    "# I decided to include data for comparison of FACS generation as a negative control of IS mediated deletion\n",
    "import is_event_classification_func as iecf\n",
    "#lines_ = ['L05-1', 'L05-2', 'L05-3', 'L05-4']\n",
    "lines_ = lines\n",
    "for i, line in enumerate(lines_):\n",
    "\tC_ = 'python3 ../src/SyRI_IS/classify_IS_events.py '\n",
    "\tC_ += ' --prefix ' + line \n",
    "\tC_ += ' --out ' + os.path.join(exportdir, 'classify_IS_events_MDS')\n",
    "\tC_ += ' --log ' + logdir\n",
    "\tC_ += ' --input ' + os.path.join(tmpdir, line)\n",
    "\tC_ += ' --calc_depth_interval '\n",
    "\tfor gen_id, g in enumerate(file_df[file_df.Prefix == line].file_name):\n",
    "\t\tC = C_ + ' --rid ' + str(0) + ' --qid ' + str(gen_id+1)\n",
    "\t\tif i ==0 and gen_id == 0:\n",
    "\t\t\tprint(C)\n",
    "\t\tiecf.run_syri_isrm_for_specific_id(tmpdir, line, 0, gen_id+1)\n",
    "\t\tos.system(C)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 15,
   "metadata": {},
   "outputs": [],
   "source": [
    "depth_interval_df = pd.DataFrame()\n",
    "for line in lines_:\n",
    "\tfor qid in [1, 2,3]:\n",
    "\t\tdf_ = pd.read_csv(os.path.join(exportdir, 'classify_IS_events_MDS', line, f'{line}_genome.{qid}.depth_interval.csv'))\n",
    "\t\tdf_['line'] = line\n",
    "\t\tdf_['gen_id'] = qid\n",
    "\t\tdepth_interval_df = pd.concat([depth_interval_df, df_], axis=0)\n",
    "depth_interval_df= depth_interval_df.reset_index(drop=True)\n",
    "depth_interval_df['gen_id'] = depth_interval_df['gen_id'].astype(int)\n",
    "depth_interval_df.to_csv(os.path.join(exportdir, 'classify_IS_events_MDS', 'depth_interval.asm20.csv'), index=False)"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "# Obtain data for analysis\n",
    "### Genomesize"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 16,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>Prefix</th>\n",
       "      <th>Gen</th>\n",
       "      <th>File</th>\n",
       "      <th>genome_size</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>L01-1</td>\n",
       "      <td>1</td>\n",
       "      <td>../fasta/20231004/L01_Anc_m1.fasta</td>\n",
       "      <td>4003851</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>L01-1</td>\n",
       "      <td>2</td>\n",
       "      <td>../fasta/20231013/L01-1_G8_m.fasta</td>\n",
       "      <td>4014776</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>L01-1</td>\n",
       "      <td>3</td>\n",
       "      <td>../fasta/20231003/L01-1_G20.fasta</td>\n",
       "      <td>3923637</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>L01-2</td>\n",
       "      <td>1</td>\n",
       "      <td>../fasta/20231004/L01_Anc_m1.fasta</td>\n",
       "      <td>4003851</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>L01-2</td>\n",
       "      <td>2</td>\n",
       "      <td>../fasta/20231003/L01-2_G8.fasta</td>\n",
       "      <td>4007666</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "  Prefix  Gen                                File  genome_size\n",
       "0  L01-1    1  ../fasta/20231004/L01_Anc_m1.fasta      4003851\n",
       "1  L01-1    2  ../fasta/20231013/L01-1_G8_m.fasta      4014776\n",
       "2  L01-1    3   ../fasta/20231003/L01-1_G20.fasta      3923637\n",
       "3  L01-2    1  ../fasta/20231004/L01_Anc_m1.fasta      4003851\n",
       "4  L01-2    2    ../fasta/20231003/L01-2_G8.fasta      4007666"
      ]
     },
     "execution_count": 16,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "import SyRI_IS\n",
    "#genome_size_df = pd.DataFrame({'Prefix': prefixes, 'Gen': gen_ids, 'File': ma_fasta_files})\n",
    "lines_ = lines\n",
    "genome_size_df = pd.DataFrame({'Prefix': [line for line in lines_ for _ in range(3)], 'Gen': gen_ids, 'File': ma_fasta_files})\n",
    "genome_size_df['Gen'] = genome_size_df.Gen.apply(lambda x: x-1)\n",
    "genome_size_df['genome_size'] = genome_size_df.File.apply(lambda x: SyRI_IS.get_fasta_length(x, rmN=True))\n",
    "genome_size_df.head()"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "### insertion distance"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 17,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "(904, 8)\n"
     ]
    },
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>insert_id</th>\n",
       "      <th>pos</th>\n",
       "      <th>position_status</th>\n",
       "      <th>sv</th>\n",
       "      <th>closest_original_id</th>\n",
       "      <th>closest_original_distance</th>\n",
       "      <th>Line</th>\n",
       "      <th>Gen</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>10</th>\n",
       "      <td>31</td>\n",
       "      <td>2953083</td>\n",
       "      <td>new</td>\n",
       "      <td>simple_insertion</td>\n",
       "      <td>30</td>\n",
       "      <td>54630</td>\n",
       "      <td>L11-4</td>\n",
       "      <td>2</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>11</th>\n",
       "      <td>32</td>\n",
       "      <td>3390500</td>\n",
       "      <td>new</td>\n",
       "      <td>simple_insertion</td>\n",
       "      <td>33</td>\n",
       "      <td>271391</td>\n",
       "      <td>L11-4</td>\n",
       "      <td>2</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>12</th>\n",
       "      <td>37</td>\n",
       "      <td>3980006</td>\n",
       "      <td>new</td>\n",
       "      <td>simple_insertion</td>\n",
       "      <td>36</td>\n",
       "      <td>57312</td>\n",
       "      <td>L11-4</td>\n",
       "      <td>2</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>13</td>\n",
       "      <td>1244239</td>\n",
       "      <td>new</td>\n",
       "      <td>simple_insertion</td>\n",
       "      <td>14</td>\n",
       "      <td>45894</td>\n",
       "      <td>L11-4</td>\n",
       "      <td>3</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>18</td>\n",
       "      <td>1576177</td>\n",
       "      <td>new</td>\n",
       "      <td>simple_insertion</td>\n",
       "      <td>17</td>\n",
       "      <td>136310</td>\n",
       "      <td>L11-4</td>\n",
       "      <td>3</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>21</td>\n",
       "      <td>1873190</td>\n",
       "      <td>new</td>\n",
       "      <td>unknown</td>\n",
       "      <td>22</td>\n",
       "      <td>188</td>\n",
       "      <td>L11-4</td>\n",
       "      <td>3</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>26</td>\n",
       "      <td>2072935</td>\n",
       "      <td>new</td>\n",
       "      <td>simple_insertion</td>\n",
       "      <td>27</td>\n",
       "      <td>54595</td>\n",
       "      <td>L11-4</td>\n",
       "      <td>3</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>31</td>\n",
       "      <td>2365293</td>\n",
       "      <td>new</td>\n",
       "      <td>simple_insertion</td>\n",
       "      <td>32</td>\n",
       "      <td>80918</td>\n",
       "      <td>L11-4</td>\n",
       "      <td>3</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>5</th>\n",
       "      <td>46</td>\n",
       "      <td>3757398</td>\n",
       "      <td>new</td>\n",
       "      <td>unknown</td>\n",
       "      <td>45</td>\n",
       "      <td>91996</td>\n",
       "      <td>L11-4</td>\n",
       "      <td>3</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>6</th>\n",
       "      <td>49</td>\n",
       "      <td>3978936</td>\n",
       "      <td>new</td>\n",
       "      <td>unknown</td>\n",
       "      <td>50</td>\n",
       "      <td>1470</td>\n",
       "      <td>L11-4</td>\n",
       "      <td>3</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "    insert_id      pos position_status                sv  closest_original_id  \\\n",
       "10         31  2953083             new  simple_insertion                   30   \n",
       "11         32  3390500             new  simple_insertion                   33   \n",
       "12         37  3980006             new  simple_insertion                   36   \n",
       "0          13  1244239             new  simple_insertion                   14   \n",
       "1          18  1576177             new  simple_insertion                   17   \n",
       "2          21  1873190             new           unknown                   22   \n",
       "3          26  2072935             new  simple_insertion                   27   \n",
       "4          31  2365293             new  simple_insertion                   32   \n",
       "5          46  3757398             new           unknown                   45   \n",
       "6          49  3978936             new           unknown                   50   \n",
       "\n",
       "    closest_original_distance   Line  Gen  \n",
       "10                      54630  L11-4    2  \n",
       "11                     271391  L11-4    2  \n",
       "12                      57312  L11-4    2  \n",
       "0                       45894  L11-4    3  \n",
       "1                      136310  L11-4    3  \n",
       "2                         188  L11-4    3  \n",
       "3                       54595  L11-4    3  \n",
       "4                       80918  L11-4    3  \n",
       "5                       91996  L11-4    3  \n",
       "6                        1470  L11-4    3  "
      ]
     },
     "execution_count": 17,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "import analyse_stats\n",
    "insertion_distance_file = analyse_stats.get_insertion_distances(lines_, exportdir, [2, 3])\n",
    "print(insertion_distance_file.shape)\n",
    "insertion_distance_file.to_csv(os.path.join(exportdir, 'classify_IS_events', 'simple_insertion_distances.csv'), index=False)\n",
    "insertion_distance_file.tail(10)"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "### position of iss in the gneome"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 18,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "<module 'analyse_stats' from '/home/kanai/documents/analysis/myTELR/syri/ipynb/../src/SyRI_IS/analyse_stats.py'>"
      ]
     },
     "execution_count": 18,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "import importlib\n",
    "importlib.reload(analyse_stats)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 19,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "(2369, 8)\n"
     ]
    },
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>start</th>\n",
       "      <th>end</th>\n",
       "      <th>IS_strand</th>\n",
       "      <th>max_alignment_length</th>\n",
       "      <th>length</th>\n",
       "      <th>cluster_id</th>\n",
       "      <th>Line</th>\n",
       "      <th>Gen</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>19</th>\n",
       "      <td>2829024</td>\n",
       "      <td>2832116</td>\n",
       "      <td>forward</td>\n",
       "      <td>3094</td>\n",
       "      <td>3093</td>\n",
       "      <td>19</td>\n",
       "      <td>L11-4</td>\n",
       "      <td>3</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>20</th>\n",
       "      <td>2902961</td>\n",
       "      <td>2906053</td>\n",
       "      <td>forward</td>\n",
       "      <td>3093</td>\n",
       "      <td>3093</td>\n",
       "      <td>20</td>\n",
       "      <td>L11-4</td>\n",
       "      <td>3</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>21</th>\n",
       "      <td>2960630</td>\n",
       "      <td>2963721</td>\n",
       "      <td>reverse</td>\n",
       "      <td>3093</td>\n",
       "      <td>3092</td>\n",
       "      <td>21</td>\n",
       "      <td>L11-4</td>\n",
       "      <td>3</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>22</th>\n",
       "      <td>3401008</td>\n",
       "      <td>3404097</td>\n",
       "      <td>forward</td>\n",
       "      <td>3093</td>\n",
       "      <td>3090</td>\n",
       "      <td>22</td>\n",
       "      <td>L11-4</td>\n",
       "      <td>3</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>23</th>\n",
       "      <td>3675429</td>\n",
       "      <td>3678520</td>\n",
       "      <td>forward</td>\n",
       "      <td>3093</td>\n",
       "      <td>3092</td>\n",
       "      <td>23</td>\n",
       "      <td>L11-4</td>\n",
       "      <td>3</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>24</th>\n",
       "      <td>3770500</td>\n",
       "      <td>3773590</td>\n",
       "      <td>forward</td>\n",
       "      <td>3093</td>\n",
       "      <td>3091</td>\n",
       "      <td>24</td>\n",
       "      <td>L11-4</td>\n",
       "      <td>3</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>25</th>\n",
       "      <td>3854577</td>\n",
       "      <td>3857669</td>\n",
       "      <td>reverse</td>\n",
       "      <td>3093</td>\n",
       "      <td>3093</td>\n",
       "      <td>25</td>\n",
       "      <td>L11-4</td>\n",
       "      <td>3</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>26</th>\n",
       "      <td>3996588</td>\n",
       "      <td>3999678</td>\n",
       "      <td>forward</td>\n",
       "      <td>3092</td>\n",
       "      <td>3091</td>\n",
       "      <td>26</td>\n",
       "      <td>L11-4</td>\n",
       "      <td>3</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>27</th>\n",
       "      <td>4162242</td>\n",
       "      <td>4165333</td>\n",
       "      <td>forward</td>\n",
       "      <td>3092</td>\n",
       "      <td>3092</td>\n",
       "      <td>27</td>\n",
       "      <td>L11-4</td>\n",
       "      <td>3</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>28</th>\n",
       "      <td>4221165</td>\n",
       "      <td>4224256</td>\n",
       "      <td>forward</td>\n",
       "      <td>3093</td>\n",
       "      <td>3092</td>\n",
       "      <td>28</td>\n",
       "      <td>L11-4</td>\n",
       "      <td>3</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "      start      end IS_strand  max_alignment_length  length  cluster_id  \\\n",
       "19  2829024  2832116   forward                  3094    3093          19   \n",
       "20  2902961  2906053   forward                  3093    3093          20   \n",
       "21  2960630  2963721   reverse                  3093    3092          21   \n",
       "22  3401008  3404097   forward                  3093    3090          22   \n",
       "23  3675429  3678520   forward                  3093    3092          23   \n",
       "24  3770500  3773590   forward                  3093    3091          24   \n",
       "25  3854577  3857669   reverse                  3093    3093          25   \n",
       "26  3996588  3999678   forward                  3092    3091          26   \n",
       "27  4162242  4165333   forward                  3092    3092          27   \n",
       "28  4221165  4224256   forward                  3093    3092          28   \n",
       "\n",
       "     Line  Gen  \n",
       "19  L11-4    3  \n",
       "20  L11-4    3  \n",
       "21  L11-4    3  \n",
       "22  L11-4    3  \n",
       "23  L11-4    3  \n",
       "24  L11-4    3  \n",
       "25  L11-4    3  \n",
       "26  L11-4    3  \n",
       "27  L11-4    3  \n",
       "28  L11-4    3  "
      ]
     },
     "execution_count": 19,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "import analyse_stats\n",
    "is_positions = analyse_stats.get_is_positions(lines_, tmpdir, [1, 2, 3])\n",
    "print(is_positions.shape)\n",
    "is_positions.to_csv(os.path.join(exportdir, 'classify_IS_events', 'IS_positions.csv'), index=False)\n",
    "is_positions.tail(10)"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "## Find identical mutations in different runs to avoid over counting"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "Due to high mutation rate, there seems to be some mutations happening after sequencing directly after IS enrichment, and passage by MA. Here, I detect those events by identifying simlilar IS insertions in predecessors. This data and data exported by plotsr are used to check similar events to avoid overcounting.\n",
    "1. get all the insertions in terms of the FACS generation\n",
    "2. Cluster the positions\n",
    "3. list up all the mutations shared among multiple samples\n",
    "4. fix the ancestor sequences by hand"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 20,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "<module 'analyse_stats' from '/home/kanai/documents/analysis/myTELR/syri/ipynb/../src/SyRI_IS/analyse_stats.py'>"
      ]
     },
     "execution_count": 20,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "import analyse_stats\n",
    "importlib.reload(analyse_stats)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 21,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "(6357, 12)\n"
     ]
    },
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>pos_id</th>\n",
       "      <th>IS_strand</th>\n",
       "      <th>flank</th>\n",
       "      <th>pos</th>\n",
       "      <th>genome</th>\n",
       "      <th>cluster_id</th>\n",
       "      <th>insert_id</th>\n",
       "      <th>flankMatchDir</th>\n",
       "      <th>position_status</th>\n",
       "      <th>sv</th>\n",
       "      <th>Line</th>\n",
       "      <th>Gen</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>6352</th>\n",
       "      <td>26</td>\n",
       "      <td>True</td>\n",
       "      <td>r</td>\n",
       "      <td>3757392</td>\n",
       "      <td>Query</td>\n",
       "      <td>6</td>\n",
       "      <td>46</td>\n",
       "      <td>False</td>\n",
       "      <td>new</td>\n",
       "      <td>unknown</td>\n",
       "      <td>L11-4</td>\n",
       "      <td>3</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>6353</th>\n",
       "      <td>27</td>\n",
       "      <td>True</td>\n",
       "      <td>f</td>\n",
       "      <td>3919996</td>\n",
       "      <td>Query</td>\n",
       "      <td>4</td>\n",
       "      <td>47</td>\n",
       "      <td>True</td>\n",
       "      <td>original</td>\n",
       "      <td>pristine</td>\n",
       "      <td>L11-4</td>\n",
       "      <td>3</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>6354</th>\n",
       "      <td>27</td>\n",
       "      <td>True</td>\n",
       "      <td>r</td>\n",
       "      <td>3923084</td>\n",
       "      <td>Query</td>\n",
       "      <td>48</td>\n",
       "      <td>48</td>\n",
       "      <td>False</td>\n",
       "      <td>original</td>\n",
       "      <td>pristine</td>\n",
       "      <td>L11-4</td>\n",
       "      <td>3</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>6355</th>\n",
       "      <td>28</td>\n",
       "      <td>True</td>\n",
       "      <td>f</td>\n",
       "      <td>3978936</td>\n",
       "      <td>Query</td>\n",
       "      <td>49</td>\n",
       "      <td>49</td>\n",
       "      <td>True</td>\n",
       "      <td>new</td>\n",
       "      <td>unknown</td>\n",
       "      <td>L11-4</td>\n",
       "      <td>3</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>6356</th>\n",
       "      <td>28</td>\n",
       "      <td>True</td>\n",
       "      <td>r</td>\n",
       "      <td>3983498</td>\n",
       "      <td>Query</td>\n",
       "      <td>23</td>\n",
       "      <td>51</td>\n",
       "      <td>False</td>\n",
       "      <td>original</td>\n",
       "      <td>unknown</td>\n",
       "      <td>L11-4</td>\n",
       "      <td>3</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "      pos_id  IS_strand flank      pos genome  cluster_id  insert_id  \\\n",
       "6352      26       True     r  3757392  Query           6         46   \n",
       "6353      27       True     f  3919996  Query           4         47   \n",
       "6354      27       True     r  3923084  Query          48         48   \n",
       "6355      28       True     f  3978936  Query          49         49   \n",
       "6356      28       True     r  3983498  Query          23         51   \n",
       "\n",
       "      flankMatchDir position_status        sv   Line  Gen  \n",
       "6352          False             new   unknown  L11-4    3  \n",
       "6353           True        original  pristine  L11-4    3  \n",
       "6354          False        original  pristine  L11-4    3  \n",
       "6355           True             new   unknown  L11-4    3  \n",
       "6356          False        original   unknown  L11-4    3  "
      ]
     },
     "execution_count": 21,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "lines_ = lines\n",
    "is_in_r_df = analyse_stats.get_is_positions_in_ref_genome(lines_, os.path.join(exportdir, 'classify_IS_events'), [1, 2, 3])\n",
    "print(is_in_r_df.shape)\n",
    "is_in_r_df.to_csv(os.path.join(exportdir, 'classify_IS_events', 'IS_positions_in_ref_genome.csv'), index=False)\n",
    "is_in_r_df.tail(5)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 22,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "(2, 15)\n"
     ]
    },
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>pos_id</th>\n",
       "      <th>IS_strand</th>\n",
       "      <th>flank</th>\n",
       "      <th>pos</th>\n",
       "      <th>genome</th>\n",
       "      <th>cluster_id</th>\n",
       "      <th>insert_id</th>\n",
       "      <th>flankMatchDir</th>\n",
       "      <th>position_status</th>\n",
       "      <th>sv</th>\n",
       "      <th>Line</th>\n",
       "      <th>Gen</th>\n",
       "      <th>pos_cluster</th>\n",
       "      <th>cluster_id_cluster</th>\n",
       "      <th>insert_id_cluster</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>32</th>\n",
       "      <td>1</td>\n",
       "      <td>False</td>\n",
       "      <td>r</td>\n",
       "      <td>1232077</td>\n",
       "      <td>Query</td>\n",
       "      <td>34</td>\n",
       "      <td>2</td>\n",
       "      <td>True</td>\n",
       "      <td>new</td>\n",
       "      <td>unknown</td>\n",
       "      <td>L02-1</td>\n",
       "      <td>2</td>\n",
       "      <td>1232077</td>\n",
       "      <td>0</td>\n",
       "      <td>11</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>187</th>\n",
       "      <td>2</td>\n",
       "      <td>False</td>\n",
       "      <td>f</td>\n",
       "      <td>1232095</td>\n",
       "      <td>Query</td>\n",
       "      <td>24</td>\n",
       "      <td>4</td>\n",
       "      <td>False</td>\n",
       "      <td>new</td>\n",
       "      <td>unknown</td>\n",
       "      <td>L02-3</td>\n",
       "      <td>2</td>\n",
       "      <td>1232095</td>\n",
       "      <td>0</td>\n",
       "      <td>11</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "     pos_id  IS_strand flank      pos genome  cluster_id  insert_id  \\\n",
       "32        1      False     r  1232077  Query          34          2   \n",
       "187       2      False     f  1232095  Query          24          4   \n",
       "\n",
       "     flankMatchDir position_status       sv   Line  Gen  pos_cluster  \\\n",
       "32            True             new  unknown  L02-1    2      1232077   \n",
       "187          False             new  unknown  L02-3    2      1232095   \n",
       "\n",
       "     cluster_id_cluster  insert_id_cluster  \n",
       "32                    0                 11  \n",
       "187                   0                 11  "
      ]
     },
     "execution_count": 22,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "is_in_r_df_identical = analyse_stats.get_identical_inserts(is_in_r_df, ['L01', 'L02', 'L03', 'L04', 'L05', 'L06', 'L07', 'L08', 'L09', 'L10', 'L11'])\n",
    "print(is_in_r_df_identical[0].shape)\n",
    "is_in_r_df_identical[0].to_csv(os.path.join(exportdir, 'classify_IS_events', 'IS_positions_in_ref_genome_identical.csv'), index=False)\n",
    "is_in_r_df_identical[0].head()"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "If similar insertions are detected, I manually checked if they seem to be identical insertions, incorporated into the fasta files and reran the analysis. L02-1 and L02-3 had close insertions but the insertions were in slightly different locus that was the same in the 20th passage so they were considered independent insertions."
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "## Get statistics of each genome"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "Input specification of lines\n",
    "Output Insertion, IS-related events, loss, Copy number change"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 23,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>IS_Detect_ID</th>\n",
       "      <th>ParentLine</th>\n",
       "      <th>SubLine</th>\n",
       "      <th>gen</th>\n",
       "      <th>file_name</th>\n",
       "      <th>Anc</th>\n",
       "      <th>RecA</th>\n",
       "      <th>Prefix</th>\n",
       "      <th>sample_name_raw</th>\n",
       "      <th>Contig_Date</th>\n",
       "      <th>Complete</th>\n",
       "      <th>gen_id</th>\n",
       "      <th>File</th>\n",
       "      <th>genome_size</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>L01_Anc</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>FACS</td>\n",
       "      <td>20231004/L01_Anc_m1.fasta</td>\n",
       "      <td>R01</td>\n",
       "      <td>1</td>\n",
       "      <td>L01-1</td>\n",
       "      <td>L01_Anc</td>\n",
       "      <td>20220418</td>\n",
       "      <td>TRUE</td>\n",
       "      <td>1</td>\n",
       "      <td>../fasta/20231004/L01_Anc_m1.fasta</td>\n",
       "      <td>4003851</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>L01_Anc</td>\n",
       "      <td>1</td>\n",
       "      <td>2</td>\n",
       "      <td>FACS</td>\n",
       "      <td>20231004/L01_Anc_m1.fasta</td>\n",
       "      <td>R01</td>\n",
       "      <td>1</td>\n",
       "      <td>L01-2</td>\n",
       "      <td>L01_Anc</td>\n",
       "      <td>20220418</td>\n",
       "      <td>TRUE</td>\n",
       "      <td>1</td>\n",
       "      <td>../fasta/20231004/L01_Anc_m1.fasta</td>\n",
       "      <td>4003851</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>L01_Anc</td>\n",
       "      <td>1</td>\n",
       "      <td>3</td>\n",
       "      <td>FACS</td>\n",
       "      <td>20231004/L01_Anc_m1.fasta</td>\n",
       "      <td>R01</td>\n",
       "      <td>1</td>\n",
       "      <td>L01-3</td>\n",
       "      <td>L01_Anc</td>\n",
       "      <td>20220418</td>\n",
       "      <td>TRUE</td>\n",
       "      <td>1</td>\n",
       "      <td>../fasta/20231004/L01_Anc_m1.fasta</td>\n",
       "      <td>4003851</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>L01_Anc</td>\n",
       "      <td>1</td>\n",
       "      <td>4</td>\n",
       "      <td>FACS</td>\n",
       "      <td>20231004/L01_Anc_m1.fasta</td>\n",
       "      <td>R01</td>\n",
       "      <td>1</td>\n",
       "      <td>L01-4</td>\n",
       "      <td>L01_Anc</td>\n",
       "      <td>20220418</td>\n",
       "      <td>TRUE</td>\n",
       "      <td>1</td>\n",
       "      <td>../fasta/20231004/L01_Anc_m1.fasta</td>\n",
       "      <td>4003851</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>L02_Anc</td>\n",
       "      <td>2</td>\n",
       "      <td>1</td>\n",
       "      <td>FACS</td>\n",
       "      <td>20231003/L02_Anc.fasta</td>\n",
       "      <td>R02</td>\n",
       "      <td>1</td>\n",
       "      <td>L02-1</td>\n",
       "      <td>L02_Anc</td>\n",
       "      <td>20220418</td>\n",
       "      <td>TRUE</td>\n",
       "      <td>1</td>\n",
       "      <td>../fasta/20231003/L02_Anc.fasta</td>\n",
       "      <td>4026846</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "  IS_Detect_ID  ParentLine  SubLine   gen                  file_name  Anc  \\\n",
       "0      L01_Anc           1        1  FACS  20231004/L01_Anc_m1.fasta  R01   \n",
       "1      L01_Anc           1        2  FACS  20231004/L01_Anc_m1.fasta  R01   \n",
       "2      L01_Anc           1        3  FACS  20231004/L01_Anc_m1.fasta  R01   \n",
       "3      L01_Anc           1        4  FACS  20231004/L01_Anc_m1.fasta  R01   \n",
       "4      L02_Anc           2        1  FACS     20231003/L02_Anc.fasta  R02   \n",
       "\n",
       "   RecA Prefix sample_name_raw Contig_Date Complete  gen_id  \\\n",
       "0     1  L01-1         L01_Anc    20220418     TRUE       1   \n",
       "1     1  L01-2         L01_Anc    20220418     TRUE       1   \n",
       "2     1  L01-3         L01_Anc    20220418     TRUE       1   \n",
       "3     1  L01-4         L01_Anc    20220418     TRUE       1   \n",
       "4     1  L02-1         L02_Anc    20220418     TRUE       1   \n",
       "\n",
       "                                 File  genome_size  \n",
       "0  ../fasta/20231004/L01_Anc_m1.fasta      4003851  \n",
       "1  ../fasta/20231004/L01_Anc_m1.fasta      4003851  \n",
       "2  ../fasta/20231004/L01_Anc_m1.fasta      4003851  \n",
       "3  ../fasta/20231004/L01_Anc_m1.fasta      4003851  \n",
       "4     ../fasta/20231003/L02_Anc.fasta      4026846  "
      ]
     },
     "execution_count": 23,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "genome_stats = file_df.drop(file_df.columns[11:], axis =1).drop(file_df.index[0], axis=0)\n",
    "gen_to_gen_id = pd.DataFrame({'gen': ['Anc', 'FACS', '8', '20'],\n",
    "\t'gen_id': [0, 1, 2, 3]})\n",
    "genome_stats = genome_stats.merge(gen_to_gen_id, on='gen')\n",
    "genome_stats = genome_stats.merge(genome_size_df.rename(columns={'Gen': 'gen_id'}),\n",
    "\ton=['Prefix', 'gen_id'])\n",
    "genome_stats.head()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 24,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>IS_Detect_ID</th>\n",
       "      <th>ParentLine</th>\n",
       "      <th>SubLine</th>\n",
       "      <th>gen</th>\n",
       "      <th>file_name</th>\n",
       "      <th>Anc</th>\n",
       "      <th>RecA</th>\n",
       "      <th>Prefix</th>\n",
       "      <th>sample_name_raw</th>\n",
       "      <th>Contig_Date</th>\n",
       "      <th>Complete</th>\n",
       "      <th>gen_id</th>\n",
       "      <th>File</th>\n",
       "      <th>genome_size</th>\n",
       "      <th>is_cnt</th>\n",
       "      <th>is_length</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>130</th>\n",
       "      <td>L11-3_G20</td>\n",
       "      <td>11</td>\n",
       "      <td>3</td>\n",
       "      <td>20</td>\n",
       "      <td>20231003/L11-3_G20.fasta</td>\n",
       "      <td>r05</td>\n",
       "      <td>0</td>\n",
       "      <td>L11-3</td>\n",
       "      <td>L11-3_G20</td>\n",
       "      <td>20230909</td>\n",
       "      <td>TRUE</td>\n",
       "      <td>3</td>\n",
       "      <td>../fasta/20231003/L11-3_G20.fasta</td>\n",
       "      <td>3958158</td>\n",
       "      <td>14</td>\n",
       "      <td>43295</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>131</th>\n",
       "      <td>L11-4_G20</td>\n",
       "      <td>11</td>\n",
       "      <td>4</td>\n",
       "      <td>20</td>\n",
       "      <td>20231003/L11-4_G20.fasta</td>\n",
       "      <td>r05</td>\n",
       "      <td>0</td>\n",
       "      <td>L11-4</td>\n",
       "      <td>L11-4_G20</td>\n",
       "      <td>20230909</td>\n",
       "      <td>FALSE</td>\n",
       "      <td>3</td>\n",
       "      <td>../fasta/20231003/L11-4_G20.fasta</td>\n",
       "      <td>4248613</td>\n",
       "      <td>29</td>\n",
       "      <td>92682</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "    IS_Detect_ID  ParentLine  SubLine gen                 file_name  Anc  \\\n",
       "130    L11-3_G20          11        3  20  20231003/L11-3_G20.fasta  r05   \n",
       "131    L11-4_G20          11        4  20  20231003/L11-4_G20.fasta  r05   \n",
       "\n",
       "     RecA Prefix sample_name_raw Contig_Date Complete  gen_id  \\\n",
       "130     0  L11-3       L11-3_G20    20230909     TRUE       3   \n",
       "131     0  L11-4       L11-4_G20    20230909    FALSE       3   \n",
       "\n",
       "                                  File  genome_size  is_cnt  is_length  \n",
       "130  ../fasta/20231003/L11-3_G20.fasta      3958158      14      43295  \n",
       "131  ../fasta/20231003/L11-4_G20.fasta      4248613      29      92682  "
      ]
     },
     "execution_count": 24,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "genome_stats = pd.merge(genome_stats, is_positions.groupby(['Line', 'Gen']).\\\n",
    "\tagg(is_cnt = ('length', 'count'), is_length = ('length', 'sum')).reset_index(),\n",
    " \thow='left', left_on=['Prefix', 'gen_id'], right_on=['Line', 'Gen']).\\\n",
    "\t\t\t\t \tdrop(['Line', 'Gen'], axis=1)\n",
    "genome_stats.tail(2)"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "insert_id_ : insertion poses of query in ancestor \\\n",
    "new_ins_id : insertion pos in ref genome that is not in the ref \\\n",
    "lost_id: insertion pos in ancestor that is not in the query genome\\\n",
    "is_insert_id: insertion pos if true simple insertion or transposition \\\n",
    "\n",
    "note that lost here accounts for lost is-genome junctions and not the loss of is copies"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 25,
   "metadata": {},
   "outputs": [],
   "source": [
    "def create_aggregate_data(is_in_r_df, suffix = ''):\n",
    "\tis_in_r_df_ = is_in_r_df.copy()\n",
    "\t## add corresponding ref pos_id for rows with original iss\n",
    "\tis_in_r_df_ = pd.merge(is_in_r_df_, is_in_r_df_.query('genome == \"Ref\"').loc[:, ['pos_id', 'insert_id', 'Line', 'Gen']].\\\n",
    "\t\trename(columns={'pos_id': 'pos_id_ref'}), on=['insert_id', 'Line', 'Gen'], how='left').\\\n",
    "\t\t\tassign(pos_id_ref=lambda x: x.pos_id_ref.apply(lambda y: int(y) if not np.isnan(y) else -1))\n",
    "\tis_in_r_df_['insert_id_'] = is_in_r_df_.apply(lambda x: x.insert_id if x.genome == 'Query' else -1, axis=1)\n",
    "\t# for those with corresponding reference IS, assign the ID of the reference IS to overcount the number of insertions by counting both ends of the reference\n",
    "\tis_in_r_df_['insert_id_ref_check'] = is_in_r_df_.apply(lambda x: str(x.insert_id) if x.genome == 'Query' and x.position_status != 'original' else 'r' + str(x.pos_id_ref), axis=1)\n",
    "\tis_in_r_df_['new_ins_id'] = is_in_r_df_.apply(lambda x: x.insert_id if x.position_status == 'new' else -1, axis=1)\n",
    "\tis_in_r_df_['lost_id'] = is_in_r_df_.apply(lambda x: x.insert_id if x.position_status == 'lost' else -1, axis=1)\n",
    "\tis_in_r_df_['is_insertion'] = is_in_r_df_.apply(lambda x: x.insert_id if x.sv in ['simple_insertion', 'simple_transposition'] else -1, axis=1)\n",
    "\n",
    "\taggregated_df = is_in_r_df_.groupby(['Line', 'Gen']).agg(\n",
    "\t\tins_pos_cnt_raw=('insert_id', 'nunique'),\n",
    "\t\tins_pos_cnt = ('insert_id_ref_check', 'nunique'),\n",
    "\t\tnew_cnt = ('new_ins_id', 'nunique'),\n",
    "\t\tlost_end_cnt = ('lost_id', 'nunique'),\n",
    "\t\tsimple_insertion_cnt=('is_insertion', 'nunique'),\n",
    "\t).eval('ins_pos_cnt_raw = ins_pos_cnt_raw - 1').\\\n",
    "\t\teval('ins_pos_cnt = ins_pos_cnt - 1').\\\n",
    "\t\teval('new_cnt = new_cnt - 1').\\\n",
    "\t\teval('lost_end_cnt = lost_end_cnt - 1').\\\n",
    "\t\teval('simple_insertion_cnt = simple_insertion_cnt - 1').\\\n",
    "\t\treset_index()\n",
    "\t# Get a list of all columns, excluding 'Line' and 'Gen'\n",
    "\tcols_to_change = [col for col in aggregated_df.columns if col not in ['Line', 'Gen']]\n",
    "\trename_dict = {col: col + suffix for col in cols_to_change}\n",
    "\t# Rename the selected columns with suffixes\n",
    "\taggregated_df.rename(columns=rename_dict, inplace=True)\n",
    "\n",
    "\treturn aggregated_df"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 26,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>IS_Detect_ID</th>\n",
       "      <th>ParentLine</th>\n",
       "      <th>SubLine</th>\n",
       "      <th>gen</th>\n",
       "      <th>file_name</th>\n",
       "      <th>Anc</th>\n",
       "      <th>RecA</th>\n",
       "      <th>Prefix</th>\n",
       "      <th>sample_name_raw</th>\n",
       "      <th>Contig_Date</th>\n",
       "      <th>...</th>\n",
       "      <th>gen_id</th>\n",
       "      <th>File</th>\n",
       "      <th>genome_size</th>\n",
       "      <th>is_cnt</th>\n",
       "      <th>is_length</th>\n",
       "      <th>ins_pos_cnt_raw</th>\n",
       "      <th>ins_pos_cnt</th>\n",
       "      <th>new_cnt</th>\n",
       "      <th>lost_end_cnt</th>\n",
       "      <th>simple_insertion_cnt</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>127</th>\n",
       "      <td>L10-4_G20</td>\n",
       "      <td>10</td>\n",
       "      <td>4</td>\n",
       "      <td>20</td>\n",
       "      <td>20231003/L10-4_G20.fasta</td>\n",
       "      <td>r04</td>\n",
       "      <td>0</td>\n",
       "      <td>L10-4</td>\n",
       "      <td>L10-4_G20</td>\n",
       "      <td>20230909</td>\n",
       "      <td>...</td>\n",
       "      <td>3</td>\n",
       "      <td>../fasta/20231003/L10-4_G20.fasta</td>\n",
       "      <td>4074199</td>\n",
       "      <td>25</td>\n",
       "      <td>99353</td>\n",
       "      <td>40.0</td>\n",
       "      <td>26.0</td>\n",
       "      <td>13.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>12.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>128</th>\n",
       "      <td>L11-1_G20</td>\n",
       "      <td>11</td>\n",
       "      <td>1</td>\n",
       "      <td>20</td>\n",
       "      <td>20231003/L11-1_G20.fasta</td>\n",
       "      <td>r05</td>\n",
       "      <td>0</td>\n",
       "      <td>L11-1</td>\n",
       "      <td>L11-1_G20</td>\n",
       "      <td>20230909</td>\n",
       "      <td>...</td>\n",
       "      <td>3</td>\n",
       "      <td>../fasta/20231003/L11-1_G20.fasta</td>\n",
       "      <td>4002703</td>\n",
       "      <td>20</td>\n",
       "      <td>61848</td>\n",
       "      <td>40.0</td>\n",
       "      <td>23.0</td>\n",
       "      <td>7.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>2.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>129</th>\n",
       "      <td>L11-2_G20</td>\n",
       "      <td>11</td>\n",
       "      <td>2</td>\n",
       "      <td>20</td>\n",
       "      <td>20231003/L11-2_G20.fasta</td>\n",
       "      <td>r05</td>\n",
       "      <td>0</td>\n",
       "      <td>L11-2</td>\n",
       "      <td>L11-2_G20</td>\n",
       "      <td>20230909</td>\n",
       "      <td>...</td>\n",
       "      <td>3</td>\n",
       "      <td>../fasta/20231003/L11-2_G20.fasta</td>\n",
       "      <td>4100257</td>\n",
       "      <td>25</td>\n",
       "      <td>77313</td>\n",
       "      <td>45.0</td>\n",
       "      <td>25.0</td>\n",
       "      <td>6.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>3.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>130</th>\n",
       "      <td>L11-3_G20</td>\n",
       "      <td>11</td>\n",
       "      <td>3</td>\n",
       "      <td>20</td>\n",
       "      <td>20231003/L11-3_G20.fasta</td>\n",
       "      <td>r05</td>\n",
       "      <td>0</td>\n",
       "      <td>L11-3</td>\n",
       "      <td>L11-3_G20</td>\n",
       "      <td>20230909</td>\n",
       "      <td>...</td>\n",
       "      <td>3</td>\n",
       "      <td>../fasta/20231003/L11-3_G20.fasta</td>\n",
       "      <td>3958158</td>\n",
       "      <td>14</td>\n",
       "      <td>43295</td>\n",
       "      <td>26.0</td>\n",
       "      <td>14.0</td>\n",
       "      <td>3.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>2.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>131</th>\n",
       "      <td>L11-4_G20</td>\n",
       "      <td>11</td>\n",
       "      <td>4</td>\n",
       "      <td>20</td>\n",
       "      <td>20231003/L11-4_G20.fasta</td>\n",
       "      <td>r05</td>\n",
       "      <td>0</td>\n",
       "      <td>L11-4</td>\n",
       "      <td>L11-4_G20</td>\n",
       "      <td>20230909</td>\n",
       "      <td>...</td>\n",
       "      <td>3</td>\n",
       "      <td>../fasta/20231003/L11-4_G20.fasta</td>\n",
       "      <td>4248613</td>\n",
       "      <td>29</td>\n",
       "      <td>92682</td>\n",
       "      <td>50.0</td>\n",
       "      <td>28.0</td>\n",
       "      <td>7.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>4.0</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>5 rows × 21 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "    IS_Detect_ID  ParentLine  SubLine gen                 file_name  Anc  \\\n",
       "127    L10-4_G20          10        4  20  20231003/L10-4_G20.fasta  r04   \n",
       "128    L11-1_G20          11        1  20  20231003/L11-1_G20.fasta  r05   \n",
       "129    L11-2_G20          11        2  20  20231003/L11-2_G20.fasta  r05   \n",
       "130    L11-3_G20          11        3  20  20231003/L11-3_G20.fasta  r05   \n",
       "131    L11-4_G20          11        4  20  20231003/L11-4_G20.fasta  r05   \n",
       "\n",
       "     RecA Prefix sample_name_raw Contig_Date  ... gen_id  \\\n",
       "127     0  L10-4       L10-4_G20    20230909  ...      3   \n",
       "128     0  L11-1       L11-1_G20    20230909  ...      3   \n",
       "129     0  L11-2       L11-2_G20    20230909  ...      3   \n",
       "130     0  L11-3       L11-3_G20    20230909  ...      3   \n",
       "131     0  L11-4       L11-4_G20    20230909  ...      3   \n",
       "\n",
       "                                  File genome_size  is_cnt  is_length  \\\n",
       "127  ../fasta/20231003/L10-4_G20.fasta     4074199      25      99353   \n",
       "128  ../fasta/20231003/L11-1_G20.fasta     4002703      20      61848   \n",
       "129  ../fasta/20231003/L11-2_G20.fasta     4100257      25      77313   \n",
       "130  ../fasta/20231003/L11-3_G20.fasta     3958158      14      43295   \n",
       "131  ../fasta/20231003/L11-4_G20.fasta     4248613      29      92682   \n",
       "\n",
       "     ins_pos_cnt_raw  ins_pos_cnt  new_cnt  lost_end_cnt  simple_insertion_cnt  \n",
       "127             40.0         26.0     13.0           0.0                  12.0  \n",
       "128             40.0         23.0      7.0           0.0                   2.0  \n",
       "129             45.0         25.0      6.0           0.0                   3.0  \n",
       "130             26.0         14.0      3.0           0.0                   2.0  \n",
       "131             50.0         28.0      7.0           0.0                   4.0  \n",
       "\n",
       "[5 rows x 21 columns]"
      ]
     },
     "execution_count": 26,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "genome_stats = pd.merge(genome_stats, create_aggregate_data(is_in_r_df), how='left', left_on=['Prefix', 'gen_id'], right_on=['Line', 'Gen']).\\\n",
    "                     \tdrop(['Line', 'Gen'], axis=1).reset_index(drop=True)\n",
    "genome_stats.to_csv(os.path.join(exportdir, 'classify_IS_events', 'genome_stats.csv'), index=False)\n",
    "genome_stats.tail(5)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 27,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "(5779, 12)\n"
     ]
    },
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>pos_id</th>\n",
       "      <th>IS_strand</th>\n",
       "      <th>flank</th>\n",
       "      <th>pos</th>\n",
       "      <th>genome</th>\n",
       "      <th>cluster_id</th>\n",
       "      <th>insert_id</th>\n",
       "      <th>flankMatchDir</th>\n",
       "      <th>position_status</th>\n",
       "      <th>sv</th>\n",
       "      <th>Line</th>\n",
       "      <th>Gen</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>5774</th>\n",
       "      <td>26</td>\n",
       "      <td>True</td>\n",
       "      <td>r</td>\n",
       "      <td>3756961</td>\n",
       "      <td>Query</td>\n",
       "      <td>6</td>\n",
       "      <td>40</td>\n",
       "      <td>False</td>\n",
       "      <td>new</td>\n",
       "      <td>unknown</td>\n",
       "      <td>L11-4</td>\n",
       "      <td>3</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>5775</th>\n",
       "      <td>27</td>\n",
       "      <td>True</td>\n",
       "      <td>f</td>\n",
       "      <td>3919604</td>\n",
       "      <td>Query</td>\n",
       "      <td>16</td>\n",
       "      <td>41</td>\n",
       "      <td>True</td>\n",
       "      <td>original</td>\n",
       "      <td>pristine</td>\n",
       "      <td>L11-4</td>\n",
       "      <td>3</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>5776</th>\n",
       "      <td>27</td>\n",
       "      <td>True</td>\n",
       "      <td>r</td>\n",
       "      <td>3922694</td>\n",
       "      <td>Query</td>\n",
       "      <td>13</td>\n",
       "      <td>42</td>\n",
       "      <td>False</td>\n",
       "      <td>original</td>\n",
       "      <td>pristine</td>\n",
       "      <td>L11-4</td>\n",
       "      <td>3</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>5777</th>\n",
       "      <td>28</td>\n",
       "      <td>True</td>\n",
       "      <td>f</td>\n",
       "      <td>3978541</td>\n",
       "      <td>Query</td>\n",
       "      <td>42</td>\n",
       "      <td>43</td>\n",
       "      <td>True</td>\n",
       "      <td>new</td>\n",
       "      <td>unknown</td>\n",
       "      <td>L11-4</td>\n",
       "      <td>3</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>5778</th>\n",
       "      <td>28</td>\n",
       "      <td>True</td>\n",
       "      <td>r</td>\n",
       "      <td>3980000</td>\n",
       "      <td>Query</td>\n",
       "      <td>21</td>\n",
       "      <td>44</td>\n",
       "      <td>False</td>\n",
       "      <td>new</td>\n",
       "      <td>unknown</td>\n",
       "      <td>L11-4</td>\n",
       "      <td>3</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "      pos_id  IS_strand flank      pos genome  cluster_id  insert_id  \\\n",
       "5774      26       True     r  3756961  Query           6         40   \n",
       "5775      27       True     f  3919604  Query          16         41   \n",
       "5776      27       True     r  3922694  Query          13         42   \n",
       "5777      28       True     f  3978541  Query          42         43   \n",
       "5778      28       True     r  3980000  Query          21         44   \n",
       "\n",
       "      flankMatchDir position_status        sv   Line  Gen  \n",
       "5774          False             new   unknown  L11-4    3  \n",
       "5775           True        original  pristine  L11-4    3  \n",
       "5776          False        original  pristine  L11-4    3  \n",
       "5777           True             new   unknown  L11-4    3  \n",
       "5778          False             new   unknown  L11-4    3  "
      ]
     },
     "execution_count": 27,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "is_in_anc_df = analyse_stats.get_is_positions_in_ref_genome(lines_, os.path.join(exportdir, 'classify_IS_events_Anc'), [1, 2, 3])\n",
    "print(is_in_anc_df.shape)\n",
    "is_in_anc_df.to_csv(os.path.join(exportdir, 'classify_IS_events_Anc', 'IS_positions_in_ref_genome.csv'), index=False)\n",
    "is_in_anc_df.tail(5)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 28,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>Line</th>\n",
       "      <th>Gen</th>\n",
       "      <th>ins_pos_cnt_raw_anc</th>\n",
       "      <th>ins_pos_cnt_anc</th>\n",
       "      <th>new_cnt_anc</th>\n",
       "      <th>lost_end_cnt_anc</th>\n",
       "      <th>simple_insertion_cnt_anc</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>83</th>\n",
       "      <td>L11-2</td>\n",
       "      <td>3</td>\n",
       "      <td>37</td>\n",
       "      <td>25</td>\n",
       "      <td>14</td>\n",
       "      <td>0</td>\n",
       "      <td>8</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>84</th>\n",
       "      <td>L11-3</td>\n",
       "      <td>2</td>\n",
       "      <td>26</td>\n",
       "      <td>14</td>\n",
       "      <td>3</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>85</th>\n",
       "      <td>L11-3</td>\n",
       "      <td>3</td>\n",
       "      <td>29</td>\n",
       "      <td>17</td>\n",
       "      <td>6</td>\n",
       "      <td>0</td>\n",
       "      <td>2</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>86</th>\n",
       "      <td>L11-4</td>\n",
       "      <td>2</td>\n",
       "      <td>36</td>\n",
       "      <td>24</td>\n",
       "      <td>13</td>\n",
       "      <td>0</td>\n",
       "      <td>8</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>87</th>\n",
       "      <td>L11-4</td>\n",
       "      <td>3</td>\n",
       "      <td>43</td>\n",
       "      <td>31</td>\n",
       "      <td>20</td>\n",
       "      <td>0</td>\n",
       "      <td>10</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "     Line  Gen  ins_pos_cnt_raw_anc  ins_pos_cnt_anc  new_cnt_anc  \\\n",
       "83  L11-2    3                   37               25           14   \n",
       "84  L11-3    2                   26               14            3   \n",
       "85  L11-3    3                   29               17            6   \n",
       "86  L11-4    2                   36               24           13   \n",
       "87  L11-4    3                   43               31           20   \n",
       "\n",
       "    lost_end_cnt_anc  simple_insertion_cnt_anc  \n",
       "83                 0                         8  \n",
       "84                 0                         0  \n",
       "85                 0                         2  \n",
       "86                 0                         8  \n",
       "87                 0                        10  "
      ]
     },
     "execution_count": 28,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "create_aggregate_data(is_in_anc_df, '_anc').tail(5)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 29,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>IS_Detect_ID</th>\n",
       "      <th>ParentLine</th>\n",
       "      <th>SubLine</th>\n",
       "      <th>gen</th>\n",
       "      <th>file_name</th>\n",
       "      <th>Anc</th>\n",
       "      <th>RecA</th>\n",
       "      <th>Prefix</th>\n",
       "      <th>sample_name_raw</th>\n",
       "      <th>Contig_Date</th>\n",
       "      <th>...</th>\n",
       "      <th>ins_pos_cnt_raw</th>\n",
       "      <th>ins_pos_cnt</th>\n",
       "      <th>new_cnt</th>\n",
       "      <th>lost_end_cnt</th>\n",
       "      <th>simple_insertion_cnt</th>\n",
       "      <th>ins_pos_cnt_raw_anc</th>\n",
       "      <th>ins_pos_cnt_anc</th>\n",
       "      <th>new_cnt_anc</th>\n",
       "      <th>lost_end_cnt_anc</th>\n",
       "      <th>simple_insertion_cnt_anc</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>127</th>\n",
       "      <td>L10-4_G20</td>\n",
       "      <td>10</td>\n",
       "      <td>4</td>\n",
       "      <td>20</td>\n",
       "      <td>20231003/L10-4_G20.fasta</td>\n",
       "      <td>r04</td>\n",
       "      <td>0</td>\n",
       "      <td>L10-4</td>\n",
       "      <td>L10-4_G20</td>\n",
       "      <td>20230909</td>\n",
       "      <td>...</td>\n",
       "      <td>40.0</td>\n",
       "      <td>26.0</td>\n",
       "      <td>13.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>12.0</td>\n",
       "      <td>39.0</td>\n",
       "      <td>29.0</td>\n",
       "      <td>20.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>11.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>128</th>\n",
       "      <td>L11-1_G20</td>\n",
       "      <td>11</td>\n",
       "      <td>1</td>\n",
       "      <td>20</td>\n",
       "      <td>20231003/L11-1_G20.fasta</td>\n",
       "      <td>r05</td>\n",
       "      <td>0</td>\n",
       "      <td>L11-1</td>\n",
       "      <td>L11-1_G20</td>\n",
       "      <td>20230909</td>\n",
       "      <td>...</td>\n",
       "      <td>40.0</td>\n",
       "      <td>23.0</td>\n",
       "      <td>7.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>2.0</td>\n",
       "      <td>35.0</td>\n",
       "      <td>23.0</td>\n",
       "      <td>12.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>6.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>129</th>\n",
       "      <td>L11-2_G20</td>\n",
       "      <td>11</td>\n",
       "      <td>2</td>\n",
       "      <td>20</td>\n",
       "      <td>20231003/L11-2_G20.fasta</td>\n",
       "      <td>r05</td>\n",
       "      <td>0</td>\n",
       "      <td>L11-2</td>\n",
       "      <td>L11-2_G20</td>\n",
       "      <td>20230909</td>\n",
       "      <td>...</td>\n",
       "      <td>45.0</td>\n",
       "      <td>25.0</td>\n",
       "      <td>6.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>3.0</td>\n",
       "      <td>37.0</td>\n",
       "      <td>25.0</td>\n",
       "      <td>14.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>8.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>130</th>\n",
       "      <td>L11-3_G20</td>\n",
       "      <td>11</td>\n",
       "      <td>3</td>\n",
       "      <td>20</td>\n",
       "      <td>20231003/L11-3_G20.fasta</td>\n",
       "      <td>r05</td>\n",
       "      <td>0</td>\n",
       "      <td>L11-3</td>\n",
       "      <td>L11-3_G20</td>\n",
       "      <td>20230909</td>\n",
       "      <td>...</td>\n",
       "      <td>26.0</td>\n",
       "      <td>14.0</td>\n",
       "      <td>3.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>2.0</td>\n",
       "      <td>29.0</td>\n",
       "      <td>17.0</td>\n",
       "      <td>6.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>2.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>131</th>\n",
       "      <td>L11-4_G20</td>\n",
       "      <td>11</td>\n",
       "      <td>4</td>\n",
       "      <td>20</td>\n",
       "      <td>20231003/L11-4_G20.fasta</td>\n",
       "      <td>r05</td>\n",
       "      <td>0</td>\n",
       "      <td>L11-4</td>\n",
       "      <td>L11-4_G20</td>\n",
       "      <td>20230909</td>\n",
       "      <td>...</td>\n",
       "      <td>50.0</td>\n",
       "      <td>28.0</td>\n",
       "      <td>7.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>4.0</td>\n",
       "      <td>43.0</td>\n",
       "      <td>31.0</td>\n",
       "      <td>20.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>10.0</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>5 rows × 26 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "    IS_Detect_ID  ParentLine  SubLine gen                 file_name  Anc  \\\n",
       "127    L10-4_G20          10        4  20  20231003/L10-4_G20.fasta  r04   \n",
       "128    L11-1_G20          11        1  20  20231003/L11-1_G20.fasta  r05   \n",
       "129    L11-2_G20          11        2  20  20231003/L11-2_G20.fasta  r05   \n",
       "130    L11-3_G20          11        3  20  20231003/L11-3_G20.fasta  r05   \n",
       "131    L11-4_G20          11        4  20  20231003/L11-4_G20.fasta  r05   \n",
       "\n",
       "     RecA Prefix sample_name_raw Contig_Date  ... ins_pos_cnt_raw  \\\n",
       "127     0  L10-4       L10-4_G20    20230909  ...            40.0   \n",
       "128     0  L11-1       L11-1_G20    20230909  ...            40.0   \n",
       "129     0  L11-2       L11-2_G20    20230909  ...            45.0   \n",
       "130     0  L11-3       L11-3_G20    20230909  ...            26.0   \n",
       "131     0  L11-4       L11-4_G20    20230909  ...            50.0   \n",
       "\n",
       "     ins_pos_cnt new_cnt  lost_end_cnt  simple_insertion_cnt  \\\n",
       "127         26.0    13.0           0.0                  12.0   \n",
       "128         23.0     7.0           0.0                   2.0   \n",
       "129         25.0     6.0           0.0                   3.0   \n",
       "130         14.0     3.0           0.0                   2.0   \n",
       "131         28.0     7.0           0.0                   4.0   \n",
       "\n",
       "     ins_pos_cnt_raw_anc  ins_pos_cnt_anc  new_cnt_anc  lost_end_cnt_anc  \\\n",
       "127                 39.0             29.0         20.0               0.0   \n",
       "128                 35.0             23.0         12.0               0.0   \n",
       "129                 37.0             25.0         14.0               0.0   \n",
       "130                 29.0             17.0          6.0               0.0   \n",
       "131                 43.0             31.0         20.0               0.0   \n",
       "\n",
       "     simple_insertion_cnt_anc  \n",
       "127                      11.0  \n",
       "128                       6.0  \n",
       "129                       8.0  \n",
       "130                       2.0  \n",
       "131                      10.0  \n",
       "\n",
       "[5 rows x 26 columns]"
      ]
     },
     "execution_count": 29,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "genome_stats = pd.merge(genome_stats,\n",
    "                        create_aggregate_data(is_in_anc_df, '_anc'),\n",
    "                        how='left', left_on=['Prefix', 'gen_id'], right_on=['Line', 'Gen']).\\\n",
    "                     \tdrop(['Line', 'Gen'], axis=1).reset_index(drop=True)\n",
    "genome_stats.tail(5)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 30,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "(5000, 12)\n"
     ]
    },
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>pos_id</th>\n",
       "      <th>IS_strand</th>\n",
       "      <th>flank</th>\n",
       "      <th>pos</th>\n",
       "      <th>genome</th>\n",
       "      <th>cluster_id</th>\n",
       "      <th>insert_id</th>\n",
       "      <th>flankMatchDir</th>\n",
       "      <th>position_status</th>\n",
       "      <th>sv</th>\n",
       "      <th>Line</th>\n",
       "      <th>Gen</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>4995</th>\n",
       "      <td>26</td>\n",
       "      <td>True</td>\n",
       "      <td>r</td>\n",
       "      <td>3727985</td>\n",
       "      <td>Query</td>\n",
       "      <td>17</td>\n",
       "      <td>32</td>\n",
       "      <td>False</td>\n",
       "      <td>new</td>\n",
       "      <td>unknown</td>\n",
       "      <td>L11-4</td>\n",
       "      <td>3</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4996</th>\n",
       "      <td>27</td>\n",
       "      <td>True</td>\n",
       "      <td>f</td>\n",
       "      <td>3890676</td>\n",
       "      <td>Query</td>\n",
       "      <td>6</td>\n",
       "      <td>33</td>\n",
       "      <td>True</td>\n",
       "      <td>new</td>\n",
       "      <td>unknown</td>\n",
       "      <td>L11-4</td>\n",
       "      <td>3</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4997</th>\n",
       "      <td>27</td>\n",
       "      <td>True</td>\n",
       "      <td>r</td>\n",
       "      <td>3898750</td>\n",
       "      <td>Query</td>\n",
       "      <td>26</td>\n",
       "      <td>34</td>\n",
       "      <td>False</td>\n",
       "      <td>new</td>\n",
       "      <td>unknown</td>\n",
       "      <td>L11-4</td>\n",
       "      <td>3</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4998</th>\n",
       "      <td>28</td>\n",
       "      <td>True</td>\n",
       "      <td>f</td>\n",
       "      <td>3954602</td>\n",
       "      <td>Query</td>\n",
       "      <td>32</td>\n",
       "      <td>35</td>\n",
       "      <td>True</td>\n",
       "      <td>new</td>\n",
       "      <td>unknown</td>\n",
       "      <td>L11-4</td>\n",
       "      <td>3</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4999</th>\n",
       "      <td>28</td>\n",
       "      <td>True</td>\n",
       "      <td>r</td>\n",
       "      <td>3956061</td>\n",
       "      <td>Query</td>\n",
       "      <td>18</td>\n",
       "      <td>36</td>\n",
       "      <td>False</td>\n",
       "      <td>new</td>\n",
       "      <td>unknown</td>\n",
       "      <td>L11-4</td>\n",
       "      <td>3</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "      pos_id  IS_strand flank      pos genome  cluster_id  insert_id  \\\n",
       "4995      26       True     r  3727985  Query          17         32   \n",
       "4996      27       True     f  3890676  Query           6         33   \n",
       "4997      27       True     r  3898750  Query          26         34   \n",
       "4998      28       True     f  3954602  Query          32         35   \n",
       "4999      28       True     r  3956061  Query          18         36   \n",
       "\n",
       "      flankMatchDir position_status       sv   Line  Gen  \n",
       "4995          False             new  unknown  L11-4    3  \n",
       "4996           True             new  unknown  L11-4    3  \n",
       "4997          False             new  unknown  L11-4    3  \n",
       "4998           True             new  unknown  L11-4    3  \n",
       "4999          False             new  unknown  L11-4    3  "
      ]
     },
     "execution_count": 30,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "is_in_mds_df = analyse_stats.get_is_positions_in_ref_genome(lines_, os.path.join(exportdir, 'classify_IS_events_MDS'), [1, 2, 3])\n",
    "print(is_in_mds_df.shape)\n",
    "is_in_mds_df.to_csv(os.path.join(exportdir, 'classify_IS_events_MDS', 'IS_positions_in_ref_genome.csv'), index=False)\n",
    "is_in_mds_df.tail(5)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 31,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>pos_id</th>\n",
       "      <th>IS_strand</th>\n",
       "      <th>flank</th>\n",
       "      <th>pos</th>\n",
       "      <th>genome</th>\n",
       "      <th>cluster_id</th>\n",
       "      <th>insert_id</th>\n",
       "      <th>flankMatchDir</th>\n",
       "      <th>position_status</th>\n",
       "      <th>sv</th>\n",
       "      <th>Line</th>\n",
       "      <th>Gen</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>0</td>\n",
       "      <td>False</td>\n",
       "      <td>f</td>\n",
       "      <td>1272610</td>\n",
       "      <td>Ref</td>\n",
       "      <td>5</td>\n",
       "      <td>4</td>\n",
       "      <td>False</td>\n",
       "      <td>original</td>\n",
       "      <td>pristine</td>\n",
       "      <td>L01-1</td>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>0</td>\n",
       "      <td>False</td>\n",
       "      <td>r</td>\n",
       "      <td>1275701</td>\n",
       "      <td>Ref</td>\n",
       "      <td>6</td>\n",
       "      <td>5</td>\n",
       "      <td>True</td>\n",
       "      <td>original</td>\n",
       "      <td>pristine</td>\n",
       "      <td>L01-1</td>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>0</td>\n",
       "      <td>False</td>\n",
       "      <td>r</td>\n",
       "      <td>476936</td>\n",
       "      <td>Query</td>\n",
       "      <td>2</td>\n",
       "      <td>1</td>\n",
       "      <td>True</td>\n",
       "      <td>new</td>\n",
       "      <td>simple_insertion</td>\n",
       "      <td>L01-1</td>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>0</td>\n",
       "      <td>False</td>\n",
       "      <td>f</td>\n",
       "      <td>476923</td>\n",
       "      <td>Query</td>\n",
       "      <td>2</td>\n",
       "      <td>1</td>\n",
       "      <td>False</td>\n",
       "      <td>new</td>\n",
       "      <td>simple_insertion</td>\n",
       "      <td>L01-1</td>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>1</td>\n",
       "      <td>False</td>\n",
       "      <td>r</td>\n",
       "      <td>554076</td>\n",
       "      <td>Query</td>\n",
       "      <td>4</td>\n",
       "      <td>2</td>\n",
       "      <td>True</td>\n",
       "      <td>new</td>\n",
       "      <td>simple_insertion</td>\n",
       "      <td>L01-1</td>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "   pos_id  IS_strand flank      pos genome  cluster_id  insert_id  \\\n",
       "0       0      False     f  1272610    Ref           5          4   \n",
       "1       0      False     r  1275701    Ref           6          5   \n",
       "2       0      False     r   476936  Query           2          1   \n",
       "3       0      False     f   476923  Query           2          1   \n",
       "4       1      False     r   554076  Query           4          2   \n",
       "\n",
       "   flankMatchDir position_status                sv   Line  Gen  \n",
       "0          False        original          pristine  L01-1    1  \n",
       "1           True        original          pristine  L01-1    1  \n",
       "2           True             new  simple_insertion  L01-1    1  \n",
       "3          False             new  simple_insertion  L01-1    1  \n",
       "4           True             new  simple_insertion  L01-1    1  "
      ]
     },
     "execution_count": 31,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "is_in_mds_df.head()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 32,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>IS_Detect_ID</th>\n",
       "      <th>ParentLine</th>\n",
       "      <th>SubLine</th>\n",
       "      <th>gen</th>\n",
       "      <th>file_name</th>\n",
       "      <th>Anc</th>\n",
       "      <th>RecA</th>\n",
       "      <th>Prefix</th>\n",
       "      <th>sample_name_raw</th>\n",
       "      <th>Contig_Date</th>\n",
       "      <th>...</th>\n",
       "      <th>ins_pos_cnt_raw_anc</th>\n",
       "      <th>ins_pos_cnt_anc</th>\n",
       "      <th>new_cnt_anc</th>\n",
       "      <th>lost_end_cnt_anc</th>\n",
       "      <th>simple_insertion_cnt_anc</th>\n",
       "      <th>ins_pos_cnt_raw_mds</th>\n",
       "      <th>ins_pos_cnt_mds</th>\n",
       "      <th>new_cnt_mds</th>\n",
       "      <th>lost_end_cnt_mds</th>\n",
       "      <th>simple_insertion_cnt_mds</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>127</th>\n",
       "      <td>L10-4_G20</td>\n",
       "      <td>10</td>\n",
       "      <td>4</td>\n",
       "      <td>20</td>\n",
       "      <td>20231003/L10-4_G20.fasta</td>\n",
       "      <td>r04</td>\n",
       "      <td>0</td>\n",
       "      <td>L10-4</td>\n",
       "      <td>L10-4_G20</td>\n",
       "      <td>20230909</td>\n",
       "      <td>...</td>\n",
       "      <td>39.0</td>\n",
       "      <td>29.0</td>\n",
       "      <td>20.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>11.0</td>\n",
       "      <td>30</td>\n",
       "      <td>29</td>\n",
       "      <td>29</td>\n",
       "      <td>0</td>\n",
       "      <td>15</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>128</th>\n",
       "      <td>L11-1_G20</td>\n",
       "      <td>11</td>\n",
       "      <td>1</td>\n",
       "      <td>20</td>\n",
       "      <td>20231003/L11-1_G20.fasta</td>\n",
       "      <td>r05</td>\n",
       "      <td>0</td>\n",
       "      <td>L11-1</td>\n",
       "      <td>L11-1_G20</td>\n",
       "      <td>20230909</td>\n",
       "      <td>...</td>\n",
       "      <td>35.0</td>\n",
       "      <td>23.0</td>\n",
       "      <td>12.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>6.0</td>\n",
       "      <td>26</td>\n",
       "      <td>25</td>\n",
       "      <td>25</td>\n",
       "      <td>0</td>\n",
       "      <td>13</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>129</th>\n",
       "      <td>L11-2_G20</td>\n",
       "      <td>11</td>\n",
       "      <td>2</td>\n",
       "      <td>20</td>\n",
       "      <td>20231003/L11-2_G20.fasta</td>\n",
       "      <td>r05</td>\n",
       "      <td>0</td>\n",
       "      <td>L11-2</td>\n",
       "      <td>L11-2_G20</td>\n",
       "      <td>20230909</td>\n",
       "      <td>...</td>\n",
       "      <td>37.0</td>\n",
       "      <td>25.0</td>\n",
       "      <td>14.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>8.0</td>\n",
       "      <td>28</td>\n",
       "      <td>27</td>\n",
       "      <td>27</td>\n",
       "      <td>0</td>\n",
       "      <td>13</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>130</th>\n",
       "      <td>L11-3_G20</td>\n",
       "      <td>11</td>\n",
       "      <td>3</td>\n",
       "      <td>20</td>\n",
       "      <td>20231003/L11-3_G20.fasta</td>\n",
       "      <td>r05</td>\n",
       "      <td>0</td>\n",
       "      <td>L11-3</td>\n",
       "      <td>L11-3_G20</td>\n",
       "      <td>20230909</td>\n",
       "      <td>...</td>\n",
       "      <td>29.0</td>\n",
       "      <td>17.0</td>\n",
       "      <td>6.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>2.0</td>\n",
       "      <td>20</td>\n",
       "      <td>19</td>\n",
       "      <td>19</td>\n",
       "      <td>0</td>\n",
       "      <td>7</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>131</th>\n",
       "      <td>L11-4_G20</td>\n",
       "      <td>11</td>\n",
       "      <td>4</td>\n",
       "      <td>20</td>\n",
       "      <td>20231003/L11-4_G20.fasta</td>\n",
       "      <td>r05</td>\n",
       "      <td>0</td>\n",
       "      <td>L11-4</td>\n",
       "      <td>L11-4_G20</td>\n",
       "      <td>20230909</td>\n",
       "      <td>...</td>\n",
       "      <td>43.0</td>\n",
       "      <td>31.0</td>\n",
       "      <td>20.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>10.0</td>\n",
       "      <td>35</td>\n",
       "      <td>34</td>\n",
       "      <td>34</td>\n",
       "      <td>0</td>\n",
       "      <td>16</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>5 rows × 31 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "    IS_Detect_ID  ParentLine  SubLine gen                 file_name  Anc  \\\n",
       "127    L10-4_G20          10        4  20  20231003/L10-4_G20.fasta  r04   \n",
       "128    L11-1_G20          11        1  20  20231003/L11-1_G20.fasta  r05   \n",
       "129    L11-2_G20          11        2  20  20231003/L11-2_G20.fasta  r05   \n",
       "130    L11-3_G20          11        3  20  20231003/L11-3_G20.fasta  r05   \n",
       "131    L11-4_G20          11        4  20  20231003/L11-4_G20.fasta  r05   \n",
       "\n",
       "     RecA Prefix sample_name_raw Contig_Date  ... ins_pos_cnt_raw_anc  \\\n",
       "127     0  L10-4       L10-4_G20    20230909  ...                39.0   \n",
       "128     0  L11-1       L11-1_G20    20230909  ...                35.0   \n",
       "129     0  L11-2       L11-2_G20    20230909  ...                37.0   \n",
       "130     0  L11-3       L11-3_G20    20230909  ...                29.0   \n",
       "131     0  L11-4       L11-4_G20    20230909  ...                43.0   \n",
       "\n",
       "     ins_pos_cnt_anc new_cnt_anc  lost_end_cnt_anc  simple_insertion_cnt_anc  \\\n",
       "127             29.0        20.0               0.0                      11.0   \n",
       "128             23.0        12.0               0.0                       6.0   \n",
       "129             25.0        14.0               0.0                       8.0   \n",
       "130             17.0         6.0               0.0                       2.0   \n",
       "131             31.0        20.0               0.0                      10.0   \n",
       "\n",
       "     ins_pos_cnt_raw_mds  ins_pos_cnt_mds  new_cnt_mds  lost_end_cnt_mds  \\\n",
       "127                   30               29           29                 0   \n",
       "128                   26               25           25                 0   \n",
       "129                   28               27           27                 0   \n",
       "130                   20               19           19                 0   \n",
       "131                   35               34           34                 0   \n",
       "\n",
       "     simple_insertion_cnt_mds  \n",
       "127                        15  \n",
       "128                        13  \n",
       "129                        13  \n",
       "130                         7  \n",
       "131                        16  \n",
       "\n",
       "[5 rows x 31 columns]"
      ]
     },
     "execution_count": 32,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "genome_stats = pd.merge(genome_stats,\n",
    "                        create_aggregate_data(is_in_mds_df, '_mds'),\n",
    "                        how='left', left_on=['Prefix', 'gen_id'], right_on=['Line', 'Gen']).\\\n",
    "                     \tdrop(['Line', 'Gen'], axis=1).reset_index(drop=True)\n",
    "genome_stats.to_csv(os.path.join(exportdir, 'classify_IS_events', 'genome_stats_merged.csv'), index=False)\n",
    "genome_stats.tail(5)"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "# Aggregate sv data (MDS)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 35,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>chr1</th>\n",
       "      <th>start1</th>\n",
       "      <th>end1</th>\n",
       "      <th>REF</th>\n",
       "      <th>ALT</th>\n",
       "      <th>chr2</th>\n",
       "      <th>start2</th>\n",
       "      <th>end2</th>\n",
       "      <th>ID</th>\n",
       "      <th>parent</th>\n",
       "      <th>...</th>\n",
       "      <th>end2_is_status_r</th>\n",
       "      <th>new_cnt</th>\n",
       "      <th>dstart1_</th>\n",
       "      <th>dend1_</th>\n",
       "      <th>dstart2_</th>\n",
       "      <th>dend2_</th>\n",
       "      <th>SVLEN</th>\n",
       "      <th>SVLEN_fixed</th>\n",
       "      <th>Line</th>\n",
       "      <th>gen_id</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>2691</th>\n",
       "      <td>contig_1</td>\n",
       "      <td>3364619</td>\n",
       "      <td>3364619</td>\n",
       "      <td>t</td>\n",
       "      <td>tCGGTCAGTG</td>\n",
       "      <td>contig_1</td>\n",
       "      <td>3400998</td>\n",
       "      <td>3401007</td>\n",
       "      <td>INS651</td>\n",
       "      <td>SYN2</td>\n",
       "      <td>...</td>\n",
       "      <td>new</td>\n",
       "      <td>2</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>10</td>\n",
       "      <td>10</td>\n",
       "      <td>L11-4</td>\n",
       "      <td>3</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2692</th>\n",
       "      <td>contig_1</td>\n",
       "      <td>3635997</td>\n",
       "      <td>3635997</td>\n",
       "      <td>t</td>\n",
       "      <td>tCACCACCGG</td>\n",
       "      <td>contig_1</td>\n",
       "      <td>3675419</td>\n",
       "      <td>3675428</td>\n",
       "      <td>INS713</td>\n",
       "      <td>SYN2</td>\n",
       "      <td>...</td>\n",
       "      <td>new</td>\n",
       "      <td>2</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>10</td>\n",
       "      <td>10</td>\n",
       "      <td>L11-4</td>\n",
       "      <td>3</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2693</th>\n",
       "      <td>contig_1</td>\n",
       "      <td>3870530</td>\n",
       "      <td>3870612</td>\n",
       "      <td>ccgccttctgcgtcgctacaatggattcgattcccctcgggccaga...</td>\n",
       "      <td>c</td>\n",
       "      <td>contig_1</td>\n",
       "      <td>4142185</td>\n",
       "      <td>4142185</td>\n",
       "      <td>DEL772</td>\n",
       "      <td>SYN3</td>\n",
       "      <td>...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>83</td>\n",
       "      <td>83</td>\n",
       "      <td>L11-4</td>\n",
       "      <td>3</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2694</th>\n",
       "      <td>contig_1</td>\n",
       "      <td>3890673</td>\n",
       "      <td>3898751</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>contig_1</td>\n",
       "      <td>4162241</td>\n",
       "      <td>4162241</td>\n",
       "      <td>DEL4</td>\n",
       "      <td>SYN3</td>\n",
       "      <td>...</td>\n",
       "      <td>new</td>\n",
       "      <td>2</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>8079</td>\n",
       "      <td>8079</td>\n",
       "      <td>L11-4</td>\n",
       "      <td>3</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2695</th>\n",
       "      <td>contig_1</td>\n",
       "      <td>3954599</td>\n",
       "      <td>3956062</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>contig_1</td>\n",
       "      <td>4221164</td>\n",
       "      <td>4221164</td>\n",
       "      <td>DEL5</td>\n",
       "      <td>SYN3</td>\n",
       "      <td>...</td>\n",
       "      <td>new</td>\n",
       "      <td>2</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>1464</td>\n",
       "      <td>1464</td>\n",
       "      <td>L11-4</td>\n",
       "      <td>3</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>5 rows × 42 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "          chr1   start1     end1  \\\n",
       "2691  contig_1  3364619  3364619   \n",
       "2692  contig_1  3635997  3635997   \n",
       "2693  contig_1  3870530  3870612   \n",
       "2694  contig_1  3890673  3898751   \n",
       "2695  contig_1  3954599  3956062   \n",
       "\n",
       "                                                    REF         ALT      chr2  \\\n",
       "2691                                                  t  tCGGTCAGTG  contig_1   \n",
       "2692                                                  t  tCACCACCGG  contig_1   \n",
       "2693  ccgccttctgcgtcgctacaatggattcgattcccctcgggccaga...           c  contig_1   \n",
       "2694                                                  -           -  contig_1   \n",
       "2695                                                  -           -  contig_1   \n",
       "\n",
       "       start2     end2      ID parent  ... end2_is_status_r new_cnt  dstart1_  \\\n",
       "2691  3400998  3401007  INS651   SYN2  ...              new       2         0   \n",
       "2692  3675419  3675428  INS713   SYN2  ...              new       2         0   \n",
       "2693  4142185  4142185  DEL772   SYN3  ...              NaN       0         0   \n",
       "2694  4162241  4162241    DEL4   SYN3  ...              new       2         0   \n",
       "2695  4221164  4221164    DEL5   SYN3  ...              new       2         0   \n",
       "\n",
       "      dend1_ dstart2_ dend2_  SVLEN  SVLEN_fixed   Line  gen_id  \n",
       "2691       0      0.0    0.0     10           10  L11-4       3  \n",
       "2692       0      0.0    0.0     10           10  L11-4       3  \n",
       "2693       0      0.0    0.0     83           83  L11-4       3  \n",
       "2694       0      0.0    0.0   8079         8079  L11-4       3  \n",
       "2695       0      0.0    0.0   1464         1464  L11-4       3  \n",
       "\n",
       "[5 rows x 42 columns]"
      ]
     },
     "execution_count": 35,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "#C_ += ' --out ' + os.path.join(exportdir, 'classify_IS_events_MDS')\n",
    "\n",
    "sv_df_mds = pd.DataFrame()\n",
    "for line_ in lines:\n",
    "\tfor qid in [1,2,3]:\n",
    "\t\tdf_ = pd.read_csv(os.path.join(exportdir, 'classify_IS_events_MDS', line_, f'{line_}_genome.{qid}.sv_df.csv'))\n",
    "\t\tdf_['Line'] = line_\n",
    "\t\tdf_['gen_id'] = qid\n",
    "\t\t# inversion sizes should not exceed half the genome size\n",
    "\t\tgenome_size_ = 3980425\n",
    "\t\tdf_['SVLEN'] = df_.apply(lambda x: x.SVLEN if x.SVTYPE != 'INV' or x.SVLEN < genome_size_/2 else genome_size_ - x.SVLEN, axis=1)\n",
    "\t\tsv_df_mds = pd.concat([sv_df_mds, df_], axis=0)\n",
    "sv_df_mds = sv_df_mds.reset_index(drop=True)\n",
    "sv_df_mds.to_csv(os.path.join(exportdir, 'classify_IS_events_MDS', 'sv_df.csv'), index=False)\n",
    "sv_df_mds.tail()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 36,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>Line</th>\n",
       "      <th>gen_id</th>\n",
       "      <th>SVLEN_ISDEL</th>\n",
       "      <th>start1</th>\n",
       "      <th>end1</th>\n",
       "      <th>start2</th>\n",
       "      <th>end2</th>\n",
       "      <th>ID</th>\n",
       "      <th>start1_is_rm</th>\n",
       "      <th>end1_is_rm</th>\n",
       "      <th>start2_is_rm</th>\n",
       "      <th>end2_is_rm</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>261</th>\n",
       "      <td>L02-1</td>\n",
       "      <td>2</td>\n",
       "      <td>2398</td>\n",
       "      <td>3854237</td>\n",
       "      <td>3856634</td>\n",
       "      <td>3865103</td>\n",
       "      <td>3868203</td>\n",
       "      <td>HDR3</td>\n",
       "      <td>3851145</td>\n",
       "      <td>3853542</td>\n",
       "      <td>3805050</td>\n",
       "      <td>3805058</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>355</th>\n",
       "      <td>L02-2</td>\n",
       "      <td>3</td>\n",
       "      <td>49766</td>\n",
       "      <td>3670033</td>\n",
       "      <td>3719798</td>\n",
       "      <td>3408176</td>\n",
       "      <td>3440807</td>\n",
       "      <td>HDR4</td>\n",
       "      <td>3666941</td>\n",
       "      <td>3716706</td>\n",
       "      <td>3342640</td>\n",
       "      <td>3375271</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>772</th>\n",
       "      <td>L04-1</td>\n",
       "      <td>2</td>\n",
       "      <td>75492</td>\n",
       "      <td>2243177</td>\n",
       "      <td>2318668</td>\n",
       "      <td>1685689</td>\n",
       "      <td>1764283</td>\n",
       "      <td>HDR3</td>\n",
       "      <td>2240085</td>\n",
       "      <td>2315576</td>\n",
       "      <td>1657857</td>\n",
       "      <td>1733359</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1114</th>\n",
       "      <td>L05-2</td>\n",
       "      <td>3</td>\n",
       "      <td>684004</td>\n",
       "      <td>1272609</td>\n",
       "      <td>1959704</td>\n",
       "      <td>2352785</td>\n",
       "      <td>2994379</td>\n",
       "      <td>HDR3</td>\n",
       "      <td>1272609</td>\n",
       "      <td>1956612</td>\n",
       "      <td>2289605</td>\n",
       "      <td>2911491</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1125</th>\n",
       "      <td>L05-2</td>\n",
       "      <td>3</td>\n",
       "      <td>11366</td>\n",
       "      <td>3012354</td>\n",
       "      <td>3023719</td>\n",
       "      <td>1119735</td>\n",
       "      <td>1320078</td>\n",
       "      <td>HDR8</td>\n",
       "      <td>3009262</td>\n",
       "      <td>3020627</td>\n",
       "      <td>1093666</td>\n",
       "      <td>1287824</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1515</th>\n",
       "      <td>L07-1</td>\n",
       "      <td>1</td>\n",
       "      <td>1157</td>\n",
       "      <td>2958682</td>\n",
       "      <td>2959838</td>\n",
       "      <td>2963669</td>\n",
       "      <td>2963822</td>\n",
       "      <td>HDR3</td>\n",
       "      <td>2955590</td>\n",
       "      <td>2956746</td>\n",
       "      <td>2932742</td>\n",
       "      <td>2932895</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1695</th>\n",
       "      <td>L07-4</td>\n",
       "      <td>2</td>\n",
       "      <td>2282</td>\n",
       "      <td>846657</td>\n",
       "      <td>848938</td>\n",
       "      <td>849783</td>\n",
       "      <td>849790</td>\n",
       "      <td>HDR1</td>\n",
       "      <td>846657</td>\n",
       "      <td>848938</td>\n",
       "      <td>846689</td>\n",
       "      <td>846696</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1899</th>\n",
       "      <td>L08-3</td>\n",
       "      <td>3</td>\n",
       "      <td>21308</td>\n",
       "      <td>1844426</td>\n",
       "      <td>1865733</td>\n",
       "      <td>1862503</td>\n",
       "      <td>1865595</td>\n",
       "      <td>HDR1</td>\n",
       "      <td>1841334</td>\n",
       "      <td>1862641</td>\n",
       "      <td>1840858</td>\n",
       "      <td>1840866</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2442</th>\n",
       "      <td>L11-1</td>\n",
       "      <td>1</td>\n",
       "      <td>687</td>\n",
       "      <td>1786522</td>\n",
       "      <td>1787208</td>\n",
       "      <td>1798301</td>\n",
       "      <td>1799054</td>\n",
       "      <td>HDR1</td>\n",
       "      <td>1783430</td>\n",
       "      <td>1784116</td>\n",
       "      <td>1782835</td>\n",
       "      <td>1783588</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "       Line  gen_id  SVLEN_ISDEL   start1     end1   start2     end2    ID  \\\n",
       "261   L02-1       2         2398  3854237  3856634  3865103  3868203  HDR3   \n",
       "355   L02-2       3        49766  3670033  3719798  3408176  3440807  HDR4   \n",
       "772   L04-1       2        75492  2243177  2318668  1685689  1764283  HDR3   \n",
       "1114  L05-2       3       684004  1272609  1959704  2352785  2994379  HDR3   \n",
       "1125  L05-2       3        11366  3012354  3023719  1119735  1320078  HDR8   \n",
       "1515  L07-1       1         1157  2958682  2959838  2963669  2963822  HDR3   \n",
       "1695  L07-4       2         2282   846657   848938   849783   849790  HDR1   \n",
       "1899  L08-3       3        21308  1844426  1865733  1862503  1865595  HDR1   \n",
       "2442  L11-1       1          687  1786522  1787208  1798301  1799054  HDR1   \n",
       "\n",
       "      start1_is_rm  end1_is_rm start2_is_rm end2_is_rm  \n",
       "261        3851145     3853542      3805050    3805058  \n",
       "355        3666941     3716706      3342640    3375271  \n",
       "772        2240085     2315576      1657857    1733359  \n",
       "1114       1272609     1956612      2289605    2911491  \n",
       "1125       3009262     3020627      1093666    1287824  \n",
       "1515       2955590     2956746      2932742    2932895  \n",
       "1695        846657      848938       846689     846696  \n",
       "1899       1841334     1862641      1840858    1840866  \n",
       "2442       1783430     1784116      1782835    1783588  "
      ]
     },
     "execution_count": 36,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "sv_df_mds.query('SVTYPE == \"HDR\"').drop_duplicates(['start1', 'end1'], keep = 'first').\\\n",
    "    loc[:,['Line', 'gen_id', 'SVLEN_ISDEL', 'start1', 'end1', 'start2', 'end2', 'ID', 'start1_is_rm', 'end1_is_rm', 'start2_is_rm', 'end2_is_rm']]"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 37,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>Prefix</th>\n",
       "      <th>Gen</th>\n",
       "      <th>File</th>\n",
       "      <th>genome_size</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>L01-1</td>\n",
       "      <td>1</td>\n",
       "      <td>../fasta/20231004/L01_Anc_m1.fasta</td>\n",
       "      <td>4003851</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>L01-1</td>\n",
       "      <td>2</td>\n",
       "      <td>../fasta/20231013/L01-1_G8_m.fasta</td>\n",
       "      <td>4014776</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>L01-1</td>\n",
       "      <td>3</td>\n",
       "      <td>../fasta/20231003/L01-1_G20.fasta</td>\n",
       "      <td>3923637</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>L01-2</td>\n",
       "      <td>1</td>\n",
       "      <td>../fasta/20231004/L01_Anc_m1.fasta</td>\n",
       "      <td>4003851</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>L01-2</td>\n",
       "      <td>2</td>\n",
       "      <td>../fasta/20231003/L01-2_G8.fasta</td>\n",
       "      <td>4007666</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>127</th>\n",
       "      <td>L11-3</td>\n",
       "      <td>2</td>\n",
       "      <td>../fasta/20231003/L11-3_G8.fasta</td>\n",
       "      <td>3975273</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>128</th>\n",
       "      <td>L11-3</td>\n",
       "      <td>3</td>\n",
       "      <td>../fasta/20231003/L11-3_G20.fasta</td>\n",
       "      <td>3958158</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>129</th>\n",
       "      <td>L11-4</td>\n",
       "      <td>1</td>\n",
       "      <td>../fasta/20231003/L11_Anc.fasta</td>\n",
       "      <td>4004368</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>130</th>\n",
       "      <td>L11-4</td>\n",
       "      <td>2</td>\n",
       "      <td>../fasta/20231003/L11-4_G8.fasta</td>\n",
       "      <td>4007862</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>131</th>\n",
       "      <td>L11-4</td>\n",
       "      <td>3</td>\n",
       "      <td>../fasta/20231003/L11-4_G20.fasta</td>\n",
       "      <td>4248613</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>132 rows × 4 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "    Prefix  Gen                                File  genome_size\n",
       "0    L01-1    1  ../fasta/20231004/L01_Anc_m1.fasta      4003851\n",
       "1    L01-1    2  ../fasta/20231013/L01-1_G8_m.fasta      4014776\n",
       "2    L01-1    3   ../fasta/20231003/L01-1_G20.fasta      3923637\n",
       "3    L01-2    1  ../fasta/20231004/L01_Anc_m1.fasta      4003851\n",
       "4    L01-2    2    ../fasta/20231003/L01-2_G8.fasta      4007666\n",
       "..     ...  ...                                 ...          ...\n",
       "127  L11-3    2    ../fasta/20231003/L11-3_G8.fasta      3975273\n",
       "128  L11-3    3   ../fasta/20231003/L11-3_G20.fasta      3958158\n",
       "129  L11-4    1     ../fasta/20231003/L11_Anc.fasta      4004368\n",
       "130  L11-4    2    ../fasta/20231003/L11-4_G8.fasta      4007862\n",
       "131  L11-4    3   ../fasta/20231003/L11-4_G20.fasta      4248613\n",
       "\n",
       "[132 rows x 4 columns]"
      ]
     },
     "execution_count": 37,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "genome_size_df"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 38,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>chr1</th>\n",
       "      <th>start1</th>\n",
       "      <th>end1</th>\n",
       "      <th>REF</th>\n",
       "      <th>ALT</th>\n",
       "      <th>chr2</th>\n",
       "      <th>start2</th>\n",
       "      <th>end2</th>\n",
       "      <th>ID</th>\n",
       "      <th>parent</th>\n",
       "      <th>...</th>\n",
       "      <th>new_cnt</th>\n",
       "      <th>dstart1_</th>\n",
       "      <th>dend1_</th>\n",
       "      <th>dstart2_</th>\n",
       "      <th>dend2_</th>\n",
       "      <th>SVLEN</th>\n",
       "      <th>SVLEN_fixed</th>\n",
       "      <th>Line</th>\n",
       "      <th>gen_id</th>\n",
       "      <th>genome_size</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>989</th>\n",
       "      <td>contig_1</td>\n",
       "      <td>1873186</td>\n",
       "      <td>1873376</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>contig_1</td>\n",
       "      <td>1880592</td>\n",
       "      <td>1880592</td>\n",
       "      <td>DEL360</td>\n",
       "      <td>SYN1</td>\n",
       "      <td>...</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>191</td>\n",
       "      <td>192</td>\n",
       "      <td>L11-4</td>\n",
       "      <td>3</td>\n",
       "      <td>4007862</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>990</th>\n",
       "      <td>contig_1</td>\n",
       "      <td>2072930</td>\n",
       "      <td>2072930</td>\n",
       "      <td>G</td>\n",
       "      <td>GGGCATCGAC</td>\n",
       "      <td>contig_1</td>\n",
       "      <td>2080112</td>\n",
       "      <td>2080121</td>\n",
       "      <td>INS406</td>\n",
       "      <td>SYN1</td>\n",
       "      <td>...</td>\n",
       "      <td>2</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>10</td>\n",
       "      <td>10</td>\n",
       "      <td>L11-4</td>\n",
       "      <td>3</td>\n",
       "      <td>4007862</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>991</th>\n",
       "      <td>contig_1</td>\n",
       "      <td>2365287</td>\n",
       "      <td>2365287</td>\n",
       "      <td>T</td>\n",
       "      <td>TACGCGAAGA</td>\n",
       "      <td>contig_1</td>\n",
       "      <td>2375527</td>\n",
       "      <td>2375536</td>\n",
       "      <td>INS476</td>\n",
       "      <td>SYN1</td>\n",
       "      <td>...</td>\n",
       "      <td>2</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>10</td>\n",
       "      <td>10</td>\n",
       "      <td>L11-4</td>\n",
       "      <td>3</td>\n",
       "      <td>4007862</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>992</th>\n",
       "      <td>contig_1</td>\n",
       "      <td>167287</td>\n",
       "      <td>390321</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>contig_1</td>\n",
       "      <td>3773591</td>\n",
       "      <td>3996587</td>\n",
       "      <td>INVDP3</td>\n",
       "      <td>-</td>\n",
       "      <td>...</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>223035</td>\n",
       "      <td>223035</td>\n",
       "      <td>L11-4</td>\n",
       "      <td>3</td>\n",
       "      <td>4007862</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>993</th>\n",
       "      <td>contig_1</td>\n",
       "      <td>3978933</td>\n",
       "      <td>3980405</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>contig_1</td>\n",
       "      <td>4221164</td>\n",
       "      <td>4221164</td>\n",
       "      <td>DEL1</td>\n",
       "      <td>SYN2</td>\n",
       "      <td>...</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>1473</td>\n",
       "      <td>1473</td>\n",
       "      <td>L11-4</td>\n",
       "      <td>3</td>\n",
       "      <td>4007862</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>5 rows × 43 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "         chr1   start1     end1 REF         ALT      chr2   start2     end2  \\\n",
       "989  contig_1  1873186  1873376   -           -  contig_1  1880592  1880592   \n",
       "990  contig_1  2072930  2072930   G  GGGCATCGAC  contig_1  2080112  2080121   \n",
       "991  contig_1  2365287  2365287   T  TACGCGAAGA  contig_1  2375527  2375536   \n",
       "992  contig_1   167287   390321   -           -  contig_1  3773591  3996587   \n",
       "993  contig_1  3978933  3980405   -           -  contig_1  4221164  4221164   \n",
       "\n",
       "         ID parent  ... new_cnt dstart1_  dend1_  dstart2_ dend2_   SVLEN  \\\n",
       "989  DEL360   SYN1  ...       1        0       1       0.0    0.0     191   \n",
       "990  INS406   SYN1  ...       2        0       0       0.0    0.0      10   \n",
       "991  INS476   SYN1  ...       2        0       0       0.0    0.0      10   \n",
       "992  INVDP3      -  ...       0        0       0       0.0    0.0  223035   \n",
       "993    DEL1   SYN2  ...       1        0       0       0.0    0.0    1473   \n",
       "\n",
       "     SVLEN_fixed   Line  gen_id  genome_size  \n",
       "989          192  L11-4       3      4007862  \n",
       "990           10  L11-4       3      4007862  \n",
       "991           10  L11-4       3      4007862  \n",
       "992       223035  L11-4       3      4007862  \n",
       "993         1473  L11-4       3      4007862  \n",
       "\n",
       "[5 rows x 43 columns]"
      ]
     },
     "execution_count": 38,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "sv_df_evo = pd.DataFrame()\n",
    "for line_ in lines:\n",
    "\tfor qid in [2,3]:\n",
    "\t\tdf_ = pd.read_csv(os.path.join(exportdir, 'classify_IS_events', line_, f'{line_}_genome.{qid}.sv_df.csv'))\n",
    "\t\tdf_['Line'] = line_\n",
    "\t\tdf_['gen_id'] = qid\n",
    "\t\t# genome size of reference genome\n",
    "\t\tgenome_size_ = genome_size_df.query('Prefix == @line_ and Gen == @qid -1').genome_size.iloc[0]\n",
    "\t\t# inversion sizes should not exceed half the genome size\n",
    "\t\tdf_['genome_size'] = genome_size_\n",
    "\t\tdf_['SVLEN'] = df_.apply(lambda x: x.SVLEN if x.SVTYPE != 'INV' or x.SVLEN < genome_size_/2 else genome_size_ - x.SVLEN, axis=1)\n",
    "\t\tdf_['SVLEN_fixed'] = df_.apply(lambda x: x.SVLEN_fixed if x.SVTYPE != 'INV' or x.SVLEN_fixed < genome_size_/2 else genome_size_ - x.SVLEN_fixed, axis=1)\n",
    "\t\tsv_df_evo = pd.concat([sv_df_evo, df_], axis=0)\n",
    "sv_df_evo = sv_df_evo.reset_index(drop=True)\n",
    "sv_df_evo.to_csv(os.path.join(exportdir, 'classify_IS_events', 'sv_df.csv'), index=False)\n",
    "sv_df_evo.tail()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 39,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "INV cnt 55\n",
      "max 1719098\n",
      "min 2174\n",
      "mean 243863\n",
      "median 92322\n"
     ]
    }
   ],
   "source": [
    "print(\"INV cnt %d\" % len(sv_df_evo.query('SVTYPE == \"INV\"')))\n",
    "print(\"max %d\" % sv_df_evo.query('SVTYPE == \"INV\"').SVLEN_fixed.max())\n",
    "print(\"min %d\" % sv_df_evo.query('SVTYPE == \"INV\"').SVLEN_fixed.min())\n",
    "print(\"mean %d\" % sv_df_evo.query('SVTYPE == \"INV\"').SVLEN_fixed.mean())\n",
    "print(\"median %d\" % sv_df_evo.query('SVTYPE == \"INV\"').SVLEN_fixed.median())"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 40,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>chr1</th>\n",
       "      <th>start1</th>\n",
       "      <th>end1</th>\n",
       "      <th>REF</th>\n",
       "      <th>ALT</th>\n",
       "      <th>chr2</th>\n",
       "      <th>start2</th>\n",
       "      <th>end2</th>\n",
       "      <th>ID</th>\n",
       "      <th>parent</th>\n",
       "      <th>...</th>\n",
       "      <th>new_cnt</th>\n",
       "      <th>dstart1_</th>\n",
       "      <th>dend1_</th>\n",
       "      <th>dstart2_</th>\n",
       "      <th>dend2_</th>\n",
       "      <th>SVLEN</th>\n",
       "      <th>SVLEN_fixed</th>\n",
       "      <th>Line</th>\n",
       "      <th>gen_id</th>\n",
       "      <th>genome_size</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>602</th>\n",
       "      <td>contig_1</td>\n",
       "      <td>883796</td>\n",
       "      <td>3146395</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>contig_1</td>\n",
       "      <td>888326</td>\n",
       "      <td>3122332</td>\n",
       "      <td>INV3</td>\n",
       "      <td>-</td>\n",
       "      <td>...</td>\n",
       "      <td>1</td>\n",
       "      <td>3093</td>\n",
       "      <td>0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>-3094.0</td>\n",
       "      <td>1716005</td>\n",
       "      <td>1719098</td>\n",
       "      <td>L06-3</td>\n",
       "      <td>2</td>\n",
       "      <td>3978605</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>464</th>\n",
       "      <td>contig_1</td>\n",
       "      <td>884867</td>\n",
       "      <td>3399721</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>contig_1</td>\n",
       "      <td>891049</td>\n",
       "      <td>3397214</td>\n",
       "      <td>INV3</td>\n",
       "      <td>-</td>\n",
       "      <td>...</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>2.0</td>\n",
       "      <td>1419696</td>\n",
       "      <td>1419696</td>\n",
       "      <td>L05-2</td>\n",
       "      <td>3</td>\n",
       "      <td>3934551</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>531</th>\n",
       "      <td>contig_1</td>\n",
       "      <td>383001</td>\n",
       "      <td>3214409</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>contig_1</td>\n",
       "      <td>381924</td>\n",
       "      <td>3228692</td>\n",
       "      <td>INV5</td>\n",
       "      <td>-</td>\n",
       "      <td>...</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>-3092</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>1146661</td>\n",
       "      <td>1149753</td>\n",
       "      <td>L05-4</td>\n",
       "      <td>3</td>\n",
       "      <td>3978070</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>3 rows × 43 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "         chr1  start1     end1 REF ALT      chr2  start2     end2    ID  \\\n",
       "602  contig_1  883796  3146395   -   -  contig_1  888326  3122332  INV3   \n",
       "464  contig_1  884867  3399721   -   -  contig_1  891049  3397214  INV3   \n",
       "531  contig_1  383001  3214409   -   -  contig_1  381924  3228692  INV5   \n",
       "\n",
       "    parent  ... new_cnt dstart1_  dend1_  dstart2_  dend2_    SVLEN  \\\n",
       "602      -  ...       1     3093       0       0.0 -3094.0  1716005   \n",
       "464      -  ...       1        0       0       0.0     2.0  1419696   \n",
       "531      -  ...       1        0   -3092       0.0     0.0  1146661   \n",
       "\n",
       "     SVLEN_fixed   Line  gen_id  genome_size  \n",
       "602      1719098  L06-3       2      3978605  \n",
       "464      1419696  L05-2       3      3934551  \n",
       "531      1149753  L05-4       3      3978070  \n",
       "\n",
       "[3 rows x 43 columns]"
      ]
     },
     "execution_count": 40,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "sv_df_evo.query('SVTYPE == \"INV\"').sort_values('SVLEN_fixed', ascending=False).head(3)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 41,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>chr1</th>\n",
       "      <th>start1</th>\n",
       "      <th>end1</th>\n",
       "      <th>REF</th>\n",
       "      <th>ALT</th>\n",
       "      <th>chr2</th>\n",
       "      <th>start2</th>\n",
       "      <th>end2</th>\n",
       "      <th>ID</th>\n",
       "      <th>parent</th>\n",
       "      <th>...</th>\n",
       "      <th>new_cnt</th>\n",
       "      <th>dstart1_</th>\n",
       "      <th>dend1_</th>\n",
       "      <th>dstart2_</th>\n",
       "      <th>dend2_</th>\n",
       "      <th>SVLEN</th>\n",
       "      <th>SVLEN_fixed</th>\n",
       "      <th>Line</th>\n",
       "      <th>gen_id</th>\n",
       "      <th>genome_size</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>61</th>\n",
       "      <td>contig_1</td>\n",
       "      <td>1835423</td>\n",
       "      <td>1835600</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>contig_1</td>\n",
       "      <td>1836332</td>\n",
       "      <td>1848272</td>\n",
       "      <td>HDR2</td>\n",
       "      <td>SYN1</td>\n",
       "      <td>...</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>178</td>\n",
       "      <td>178</td>\n",
       "      <td>L01-3</td>\n",
       "      <td>3</td>\n",
       "      <td>4008366</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>103</th>\n",
       "      <td>contig_1</td>\n",
       "      <td>3894537</td>\n",
       "      <td>3900034</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>contig_1</td>\n",
       "      <td>3865103</td>\n",
       "      <td>3868203</td>\n",
       "      <td>HDR3</td>\n",
       "      <td>SYN1</td>\n",
       "      <td>...</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>-3098</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>5498</td>\n",
       "      <td>2400</td>\n",
       "      <td>L02-1</td>\n",
       "      <td>2</td>\n",
       "      <td>4026846</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>308</th>\n",
       "      <td>contig_1</td>\n",
       "      <td>2243314</td>\n",
       "      <td>2318818</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>contig_1</td>\n",
       "      <td>1685689</td>\n",
       "      <td>1761190</td>\n",
       "      <td>HDR2</td>\n",
       "      <td>INV3</td>\n",
       "      <td>...</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>-1.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>75505</td>\n",
       "      <td>75506</td>\n",
       "      <td>L04-1</td>\n",
       "      <td>2</td>\n",
       "      <td>3992834</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>400</th>\n",
       "      <td>contig_1</td>\n",
       "      <td>1859078</td>\n",
       "      <td>1944207</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>contig_1</td>\n",
       "      <td>1854139</td>\n",
       "      <td>1863203</td>\n",
       "      <td>HDR2</td>\n",
       "      <td>SYN1</td>\n",
       "      <td>...</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>2</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>85130</td>\n",
       "      <td>85132</td>\n",
       "      <td>L04-4</td>\n",
       "      <td>3</td>\n",
       "      <td>4025553</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>510</th>\n",
       "      <td>contig_1</td>\n",
       "      <td>2950687</td>\n",
       "      <td>2961668</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>contig_1</td>\n",
       "      <td>2862686</td>\n",
       "      <td>2865781</td>\n",
       "      <td>HDR8</td>\n",
       "      <td>SYN1</td>\n",
       "      <td>...</td>\n",
       "      <td>1</td>\n",
       "      <td>3093</td>\n",
       "      <td>0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>10982</td>\n",
       "      <td>7889</td>\n",
       "      <td>L05-3</td>\n",
       "      <td>3</td>\n",
       "      <td>3869555</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>532</th>\n",
       "      <td>contig_1</td>\n",
       "      <td>473895</td>\n",
       "      <td>502346</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>contig_1</td>\n",
       "      <td>3127219</td>\n",
       "      <td>3137797</td>\n",
       "      <td>HDR1</td>\n",
       "      <td>INV5</td>\n",
       "      <td>...</td>\n",
       "      <td>2</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>-1.0</td>\n",
       "      <td>-3093.0</td>\n",
       "      <td>28452</td>\n",
       "      <td>28452</td>\n",
       "      <td>L05-4</td>\n",
       "      <td>3</td>\n",
       "      <td>3978070</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>538</th>\n",
       "      <td>contig_1</td>\n",
       "      <td>1701361</td>\n",
       "      <td>1722725</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>contig_1</td>\n",
       "      <td>1890157</td>\n",
       "      <td>2051835</td>\n",
       "      <td>HDR5</td>\n",
       "      <td>INV5</td>\n",
       "      <td>...</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>21365</td>\n",
       "      <td>21365</td>\n",
       "      <td>L05-4</td>\n",
       "      <td>3</td>\n",
       "      <td>3978070</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>539</th>\n",
       "      <td>contig_1</td>\n",
       "      <td>2004895</td>\n",
       "      <td>2008224</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>contig_1</td>\n",
       "      <td>1597401</td>\n",
       "      <td>1607980</td>\n",
       "      <td>HDR6</td>\n",
       "      <td>INV5</td>\n",
       "      <td>...</td>\n",
       "      <td>2</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>-3093.0</td>\n",
       "      <td>3330</td>\n",
       "      <td>3330</td>\n",
       "      <td>L05-4</td>\n",
       "      <td>3</td>\n",
       "      <td>3978070</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>684</th>\n",
       "      <td>contig_1</td>\n",
       "      <td>849738</td>\n",
       "      <td>852028</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>contig_1</td>\n",
       "      <td>849783</td>\n",
       "      <td>849790</td>\n",
       "      <td>HDR1</td>\n",
       "      <td>SYN1</td>\n",
       "      <td>...</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>2291</td>\n",
       "      <td>2291</td>\n",
       "      <td>L07-4</td>\n",
       "      <td>2</td>\n",
       "      <td>3998098</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>908</th>\n",
       "      <td>contig_1</td>\n",
       "      <td>1798301</td>\n",
       "      <td>1799054</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>contig_1</td>\n",
       "      <td>1807133</td>\n",
       "      <td>1807819</td>\n",
       "      <td>HDR1</td>\n",
       "      <td>SYN1</td>\n",
       "      <td>...</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>754</td>\n",
       "      <td>754</td>\n",
       "      <td>L11-1</td>\n",
       "      <td>2</td>\n",
       "      <td>4004368</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>10 rows × 43 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "         chr1   start1     end1 REF ALT      chr2   start2     end2    ID  \\\n",
       "61   contig_1  1835423  1835600   -   -  contig_1  1836332  1848272  HDR2   \n",
       "103  contig_1  3894537  3900034   -   -  contig_1  3865103  3868203  HDR3   \n",
       "308  contig_1  2243314  2318818   -   -  contig_1  1685689  1761190  HDR2   \n",
       "400  contig_1  1859078  1944207   -   -  contig_1  1854139  1863203  HDR2   \n",
       "510  contig_1  2950687  2961668   -   -  contig_1  2862686  2865781  HDR8   \n",
       "532  contig_1   473895   502346   -   -  contig_1  3127219  3137797  HDR1   \n",
       "538  contig_1  1701361  1722725   -   -  contig_1  1890157  2051835  HDR5   \n",
       "539  contig_1  2004895  2008224   -   -  contig_1  1597401  1607980  HDR6   \n",
       "684  contig_1   849738   852028   -   -  contig_1   849783   849790  HDR1   \n",
       "908  contig_1  1798301  1799054   -   -  contig_1  1807133  1807819  HDR1   \n",
       "\n",
       "    parent  ... new_cnt dstart1_  dend1_  dstart2_  dend2_  SVLEN  \\\n",
       "61    SYN1  ...       1        0       0       0.0     0.0    178   \n",
       "103   SYN1  ...       1        0   -3098       0.0     0.0   5498   \n",
       "308   INV3  ...       0        0       1      -1.0     0.0  75505   \n",
       "400   SYN1  ...       1        0       2       0.0     0.0  85130   \n",
       "510   SYN1  ...       1     3093       0       0.0     0.0  10982   \n",
       "532   INV5  ...       2        0       0      -1.0 -3093.0  28452   \n",
       "538   INV5  ...       1        0       0       0.0     0.0  21365   \n",
       "539   INV5  ...       2        0       0       0.0 -3093.0   3330   \n",
       "684   SYN1  ...       1        0       0       0.0     0.0   2291   \n",
       "908   SYN1  ...       0        0       0       0.0     0.0    754   \n",
       "\n",
       "     SVLEN_fixed   Line  gen_id  genome_size  \n",
       "61           178  L01-3       3      4008366  \n",
       "103         2400  L02-1       2      4026846  \n",
       "308        75506  L04-1       2      3992834  \n",
       "400        85132  L04-4       3      4025553  \n",
       "510         7889  L05-3       3      3869555  \n",
       "532        28452  L05-4       3      3978070  \n",
       "538        21365  L05-4       3      3978070  \n",
       "539         3330  L05-4       3      3978070  \n",
       "684         2291  L07-4       2      3998098  \n",
       "908          754  L11-1       2      4004368  \n",
       "\n",
       "[10 rows x 43 columns]"
      ]
     },
     "execution_count": 41,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "sv_df_evo.query('SVTYPE == \"HDR\"').drop_duplicates(['start1', 'end1'], keep = 'first')"
   ]
  }
 ],
 "metadata": {
  "kernelspec": {
   "display_name": "Python 3 (ipykernel)",
   "language": "python",
   "name": "python3"
  },
  "language_info": {
   "codemirror_mode": {
    "name": "ipython",
    "version": 3
   },
   "file_extension": ".py",
   "mimetype": "text/x-python",
   "name": "python",
   "nbconvert_exporter": "python",
   "pygments_lexer": "ipython3",
   "version": "3.9.17"
  }
 },
 "nbformat": 4,
 "nbformat_minor": 2
}
