{
 "cells": [
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "I considered analyzing sequence divergence of ISs but decided to go with plotting all and visual inspection.\n",
    "\n",
    "Here, I annotate the sequences and visualization is done in the next (07) ipynb. ipynb."
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "# Load the data \n",
    "\n",
    "1. load IS borders\n",
    "2. annotate the genes within as fasta\n",
    "3. save as df\n",
    "4. visualize "
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 1,
   "metadata": {},
   "outputs": [],
   "source": [
    "%load_ext autoreload\n",
    "%autoreload 2\n",
    "\n",
    "import pandas as pd\n",
    "import numpy as np\n",
    "import sys\n",
    "import os\n",
    "import glob\n",
    "from Bio import SeqIO\n",
    "from Bio.Seq import Seq\n",
    "from Bio.SeqRecord import SeqRecord\n",
    "sys.path.append('../src/SyRI_IS/')\n",
    "import importlib, subprocess\n",
    "import run_utils\n",
    "from sklearn.cluster import AgglomerativeClustering\n",
    "import SyRI_IS"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 2,
   "metadata": {},
   "outputs": [],
   "source": [
    "analysis_dir = './is_variants_analysis/full'\n",
    "os.makedirs(analysis_dir, exist_ok=True)\n",
    "\n",
    "master_dir = 'exp/multiple_runs11/'\n",
    "exportdir = os.path.join(master_dir, 'export')\n",
    "tmpdir = os.path.join(master_dir, 'tmp')\n",
    "fastadir = '../fasta/'"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 3,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\n",
      "\n",
      "Building a new DB, current time: 07/03/2024 13:53:51\n",
      "New DB name:   /home/kanai/documents/analysis/myTELR/syri/ipynb/is_variants_analysis/full/blastdb/iss.fasta\n",
      "New DB title:  ./is_variants_analysis/full/iss.fasta\n",
      "Sequence type: Nucleotide\n",
      "Deleted existing Nucleotide BLAST database named /home/kanai/documents/analysis/myTELR/syri/ipynb/is_variants_analysis/full/blastdb/iss.fasta\n",
      "Keep MBits: T\n",
      "Maximum file size: 3000000000B\n",
      "Adding sequences from FASTA; added 2369 sequences in 0.041986 seconds.\n",
      "\n",
      "\n"
     ]
    },
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>sseqid</th>\n",
       "      <th>qseqid</th>\n",
       "      <th>qstart</th>\n",
       "      <th>qend</th>\n",
       "      <th>sstart</th>\n",
       "      <th>send</th>\n",
       "      <th>length</th>\n",
       "      <th>evalue</th>\n",
       "      <th>bitscore</th>\n",
       "      <th>pident</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>L10-2.3.19</td>\n",
       "      <td>IS1-IRL</td>\n",
       "      <td>1</td>\n",
       "      <td>23</td>\n",
       "      <td>1</td>\n",
       "      <td>23</td>\n",
       "      <td>23</td>\n",
       "      <td>0.000005</td>\n",
       "      <td>42.8</td>\n",
       "      <td>100.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>L10-2.3.19</td>\n",
       "      <td>IS1-IRL</td>\n",
       "      <td>1</td>\n",
       "      <td>23</td>\n",
       "      <td>696</td>\n",
       "      <td>718</td>\n",
       "      <td>23</td>\n",
       "      <td>0.000005</td>\n",
       "      <td>42.8</td>\n",
       "      <td>100.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>L10-2.2.20</td>\n",
       "      <td>IS1-IRL</td>\n",
       "      <td>1</td>\n",
       "      <td>23</td>\n",
       "      <td>1</td>\n",
       "      <td>23</td>\n",
       "      <td>23</td>\n",
       "      <td>0.000005</td>\n",
       "      <td>42.8</td>\n",
       "      <td>100.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>L10-2.2.20</td>\n",
       "      <td>IS1-IRL</td>\n",
       "      <td>1</td>\n",
       "      <td>23</td>\n",
       "      <td>695</td>\n",
       "      <td>717</td>\n",
       "      <td>23</td>\n",
       "      <td>0.000005</td>\n",
       "      <td>42.8</td>\n",
       "      <td>100.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>L02-2.2.22</td>\n",
       "      <td>IS1-IRL</td>\n",
       "      <td>1</td>\n",
       "      <td>23</td>\n",
       "      <td>1</td>\n",
       "      <td>23</td>\n",
       "      <td>23</td>\n",
       "      <td>0.000005</td>\n",
       "      <td>42.8</td>\n",
       "      <td>100.0</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "       sseqid   qseqid  qstart  qend  sstart  send  length    evalue  \\\n",
       "0  L10-2.3.19  IS1-IRL       1    23       1    23      23  0.000005   \n",
       "1  L10-2.3.19  IS1-IRL       1    23     696   718      23  0.000005   \n",
       "2  L10-2.2.20  IS1-IRL       1    23       1    23      23  0.000005   \n",
       "3  L10-2.2.20  IS1-IRL       1    23     695   717      23  0.000005   \n",
       "4  L02-2.2.22  IS1-IRL       1    23       1    23      23  0.000005   \n",
       "\n",
       "   bitscore  pident  \n",
       "0      42.8   100.0  \n",
       "1      42.8   100.0  \n",
       "2      42.8   100.0  \n",
       "3      42.8   100.0  \n",
       "4      42.8   100.0  "
      ]
     },
     "execution_count": 3,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "# fasta file of all detected iss\n",
    "is_fasta = os.path.join(analysis_dir, 'iss.fasta')\n",
    "# blastn output of genes inside iss\n",
    "blastn_out = os.path.join(analysis_dir, 'annotation_of_variants.blastn')\n",
    "\n",
    "# make fasta file of all iss and ideally, its cluster ID\n",
    "## input is positions\n",
    "is_positions = pd.read_csv(os.path.join(exportdir, 'classify_IS_events', 'IS_positions.csv'))\n",
    "is_positions = is_positions.sort_values(['length', 'Line', 'Gen', 'start'])\n",
    "is_positions['global_is_id'] = np.arange(is_positions.shape[0])\n",
    "\n",
    "## cluster iss\n",
    "is1 = os.path.join(fastadir, 'IS1.fasta')\n",
    "is1_seq = SeqIO.read(is1, 'fasta').seq\n",
    "# cluster by lengths and get unique variants\n",
    "clustering = AgglomerativeClustering(n_clusters=None, distance_threshold=100, linkage='complete')\n",
    "clustering = clustering.fit(is_positions.length.values.reshape(-1, 1))\n",
    "is_positions['variant_cluster'] = clustering.labels_\n",
    "\n",
    "records = []\n",
    "for i, row in is_positions.iterrows():\n",
    "\tseq = SeqIO.read(os.path.join(tmpdir, row.Line, f'{row.Line}_genome.{row.Gen}.fasta'), 'fasta')\n",
    "\tseq_ = seq.seq[row.start-1:row.end]\n",
    "\tif row.IS_strand == 'reverse':\n",
    "\t\tseq_ = seq_.reverse_complement()\n",
    "\trecords.append(SeqRecord(seq_, id=f'{row.Line}.{row.Gen}.{row.cluster_id}', description='', name=''))\n",
    "SeqIO.write(records, is_fasta, 'fasta') # make fasta file\n",
    "is_positions.to_csv(os.path.join(analysis_dir, 'IS_positions.csv'), index=False) # save positions\n",
    "dbdir = SyRI_IS.makedb(is_fasta, 'NA', analysis_dir)\n",
    "\n",
    "# annotate the genes inside the iss\n",
    "C = 'blastn -task blastn -query ' + os.path.join('../../test/refs/IS1_internal.fasta') + ' -evalue 1e-5 -db ' + dbdir\n",
    "# max_target_seqs is set to 100000 to get all hits this is essential when using all\n",
    "C += ' -outfmt \"7 sseqid qseqid qstart qend sstart send length evalue bitscore pident \" -max_target_seqs 100000 -out ' \n",
    "C += blastn_out\n",
    "!{C}\n",
    "\n",
    "variant_annotation_df = pd.read_csv(blastn_out, sep='\\t', comment='#', header=None)\n",
    "variant_annotation_df.columns = ['sseqid', 'qseqid', 'qstart', 'qend', 'sstart', 'send', 'length', 'evalue', 'bitscore', 'pident']\n",
    "variant_annotation_df.head()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 4,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>sseqid</th>\n",
       "      <th>qseqid</th>\n",
       "      <th>qstart</th>\n",
       "      <th>qend</th>\n",
       "      <th>sstart</th>\n",
       "      <th>send</th>\n",
       "      <th>length</th>\n",
       "      <th>evalue</th>\n",
       "      <th>bitscore</th>\n",
       "      <th>pident</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>L10-2.3.19</td>\n",
       "      <td>IS1FS</td>\n",
       "      <td>1</td>\n",
       "      <td>699</td>\n",
       "      <td>1059</td>\n",
       "      <td>1757</td>\n",
       "      <td>699</td>\n",
       "      <td>0.0</td>\n",
       "      <td>1291.0</td>\n",
       "      <td>100.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>L10-2.3.19</td>\n",
       "      <td>IS1FS</td>\n",
       "      <td>323</td>\n",
       "      <td>699</td>\n",
       "      <td>3789</td>\n",
       "      <td>4165</td>\n",
       "      <td>377</td>\n",
       "      <td>0.0</td>\n",
       "      <td>697.0</td>\n",
       "      <td>100.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>L10-2.3.19</td>\n",
       "      <td>IS1FS</td>\n",
       "      <td>323</td>\n",
       "      <td>699</td>\n",
       "      <td>6197</td>\n",
       "      <td>6573</td>\n",
       "      <td>377</td>\n",
       "      <td>0.0</td>\n",
       "      <td>697.0</td>\n",
       "      <td>100.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>L10-2.3.19</td>\n",
       "      <td>IS1FS</td>\n",
       "      <td>323</td>\n",
       "      <td>699</td>\n",
       "      <td>8605</td>\n",
       "      <td>8981</td>\n",
       "      <td>377</td>\n",
       "      <td>0.0</td>\n",
       "      <td>697.0</td>\n",
       "      <td>100.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>L10-2.3.19</td>\n",
       "      <td>IS1FS</td>\n",
       "      <td>323</td>\n",
       "      <td>699</td>\n",
       "      <td>11013</td>\n",
       "      <td>11389</td>\n",
       "      <td>377</td>\n",
       "      <td>0.0</td>\n",
       "      <td>697.0</td>\n",
       "      <td>100.0</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "       sseqid qseqid  qstart  qend  sstart   send  length  evalue  bitscore  \\\n",
       "0  L10-2.3.19  IS1FS       1   699    1059   1757     699     0.0    1291.0   \n",
       "1  L10-2.3.19  IS1FS     323   699    3789   4165     377     0.0     697.0   \n",
       "2  L10-2.3.19  IS1FS     323   699    6197   6573     377     0.0     697.0   \n",
       "3  L10-2.3.19  IS1FS     323   699    8605   8981     377     0.0     697.0   \n",
       "4  L10-2.3.19  IS1FS     323   699   11013  11389     377     0.0     697.0   \n",
       "\n",
       "   pident  \n",
       "0   100.0  \n",
       "1   100.0  \n",
       "2   100.0  \n",
       "3   100.0  \n",
       "4   100.0  "
      ]
     },
     "execution_count": 4,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "# annotate the genes inside the iss\n",
    "dbdir = os.path.join(analysis_dir, 'blastdb', 'iss.fasta')\n",
    "blastn_out = os.path.join(analysis_dir, 'annotation_of_variants.megablast')\n",
    "C = 'blastn -task megablast -query ' + os.path.join('../../test/refs/IS1_internal.fasta') + ' -db ' + dbdir\n",
    "C += ' -outfmt \"7 sseqid qseqid qstart qend sstart send length evalue bitscore pident \" -out '\n",
    "C += blastn_out\n",
    "!{C}\n",
    "\n",
    "variant_annotation_df = pd.read_csv(blastn_out, sep='\\t', comment='#', header=None)\n",
    "variant_annotation_df.columns = ['sseqid', 'qseqid', 'qstart', 'qend', 'sstart', 'send', 'length', 'evalue', 'bitscore', 'pident']\n",
    "variant_annotation_df.head()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 5,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>sseqid</th>\n",
       "      <th>qseqid</th>\n",
       "      <th>qstart</th>\n",
       "      <th>qend</th>\n",
       "      <th>sstart</th>\n",
       "      <th>send</th>\n",
       "      <th>length</th>\n",
       "      <th>evalue</th>\n",
       "      <th>bitscore</th>\n",
       "      <th>pident</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>4090</th>\n",
       "      <td>L04-1.3.6</td>\n",
       "      <td>Full_IS1</td>\n",
       "      <td>2072</td>\n",
       "      <td>2108</td>\n",
       "      <td>2106</td>\n",
       "      <td>2070</td>\n",
       "      <td>37</td>\n",
       "      <td>1.360000e-09</td>\n",
       "      <td>63.9</td>\n",
       "      <td>97.297</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4091</th>\n",
       "      <td>L04-1.3.5</td>\n",
       "      <td>Full_IS1</td>\n",
       "      <td>1</td>\n",
       "      <td>3092</td>\n",
       "      <td>1</td>\n",
       "      <td>3093</td>\n",
       "      <td>3093</td>\n",
       "      <td>0.000000e+00</td>\n",
       "      <td>5705.0</td>\n",
       "      <td>99.968</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4092</th>\n",
       "      <td>L04-1.3.5</td>\n",
       "      <td>Full_IS1</td>\n",
       "      <td>2918</td>\n",
       "      <td>2959</td>\n",
       "      <td>2006</td>\n",
       "      <td>1965</td>\n",
       "      <td>42</td>\n",
       "      <td>4.850000e-14</td>\n",
       "      <td>78.7</td>\n",
       "      <td>100.000</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4093</th>\n",
       "      <td>L04-1.3.5</td>\n",
       "      <td>Full_IS1</td>\n",
       "      <td>1964</td>\n",
       "      <td>2005</td>\n",
       "      <td>2960</td>\n",
       "      <td>2919</td>\n",
       "      <td>42</td>\n",
       "      <td>4.850000e-14</td>\n",
       "      <td>78.7</td>\n",
       "      <td>100.000</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4094</th>\n",
       "      <td>L04-1.3.5</td>\n",
       "      <td>Full_IS1</td>\n",
       "      <td>2072</td>\n",
       "      <td>2108</td>\n",
       "      <td>2106</td>\n",
       "      <td>2070</td>\n",
       "      <td>37</td>\n",
       "      <td>1.360000e-09</td>\n",
       "      <td>63.9</td>\n",
       "      <td>97.297</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "         sseqid    qseqid  qstart  qend  sstart  send  length        evalue  \\\n",
       "4090  L04-1.3.6  Full_IS1    2072  2108    2106  2070      37  1.360000e-09   \n",
       "4091  L04-1.3.5  Full_IS1       1  3092       1  3093    3093  0.000000e+00   \n",
       "4092  L04-1.3.5  Full_IS1    2918  2959    2006  1965      42  4.850000e-14   \n",
       "4093  L04-1.3.5  Full_IS1    1964  2005    2960  2919      42  4.850000e-14   \n",
       "4094  L04-1.3.5  Full_IS1    2072  2108    2106  2070      37  1.360000e-09   \n",
       "\n",
       "      bitscore   pident  \n",
       "4090      63.9   97.297  \n",
       "4091    5705.0   99.968  \n",
       "4092      78.7  100.000  \n",
       "4093      78.7  100.000  \n",
       "4094      63.9   97.297  "
      ]
     },
     "execution_count": 5,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "variant_annotation_df.tail()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 6,
   "metadata": {},
   "outputs": [],
   "source": [
    "# length of ISs\n",
    "is_fasta = os.path.join(analysis_dir, 'iss.fasta')\n",
    "pd.DataFrame({'id': [s.id for s in SeqIO.parse(is_fasta, 'fasta')],\n",
    "              \t\t'length': [len(s.seq) for s in SeqIO.parse(is_fasta, 'fasta')]}).\\\n",
    "\t\t\t\t\t\tto_csv(os.path.join(analysis_dir, 'is_variant_lengths.csv'), index=False)"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "The remaining analysis is performed in the following notebook (in R).\n",
    "07_visualize_all_iss.ipynb"
   ]
  }
 ],
 "metadata": {
  "kernelspec": {
   "display_name": "Python 3 (ipykernel)",
   "language": "python",
   "name": "python3"
  },
  "language_info": {
   "codemirror_mode": {
    "name": "ipython",
    "version": 3
   },
   "file_extension": ".py",
   "mimetype": "text/x-python",
   "name": "python",
   "nbconvert_exporter": "python",
   "pygments_lexer": "ipython3",
   "version": "3.9.17"
  }
 },
 "nbformat": 4,
 "nbformat_minor": 2
}
